<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sam White">
<meta name="dcterms.date" content="2025-10-15">

<title>Data Exploration - E5 Timeseries Multi-species Multi-omics Using workflow-stdm – Sam’s Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 2,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 100,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-7483T61T11"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-7483T61T11', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Sam’s Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" target="_blank">
 <span class="dropdown-text">Rstudio Server - Raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu/Atumefaciens/" target="_blank">
 <span class="dropdown-text">My Gannet Directory</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" target="_blank">
 <span class="dropdown-text">Steven’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shedurkin.github.io/Roberts-LabNotebook/" target="_blank">
 <span class="dropdown-text">Kathleen’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://meganewing.github.io/mewing-notebook/" target="_blank">
 <span class="dropdown-text">Megan’s Notebook</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/sams-notebook" target="_blank">
 <span class="dropdown-text">Notebook repo</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" target="_blank">
 <span class="dropdown-text">Lab Organization (GitHub)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" target="_blank">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" target="_blank"> <i class="bi bi-calendar-week" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-right">
      <h1 class="title">Data Exploration - E5 Timeseries Multi-species Multi-omics Using workflow-stdm</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2025</div>
                <div class="quarto-category">Acropora pulchra</div>
                <div class="quarto-category">Porites evermanni</div>
                <div class="quarto-category">Pocillopora tuahiniensis</div>
                <div class="quarto-category">E5</div>
                <div class="quarto-category">coral</div>
                <div class="quarto-category">timeseries_molecular</div>
                <div class="quarto-category">workflow-stdm</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-right">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Sam White </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">October 15, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#intro" id="toc-intro" class="nav-link active" data-scroll-target="#intro">INTRO</a></li>
  <li><a href="#background" id="toc-background" class="nav-link" data-scroll-target="#background">1 BACKGROUND</a>
  <ul>
  <li><a href="#analysis-overview" id="toc-analysis-overview" class="nav-link" data-scroll-target="#analysis-overview">1.1 Analysis Overview</a></li>
  <li><a href="#input-files" id="toc-input-files" class="nav-link" data-scroll-target="#input-files">1.2 Input Files</a></li>
  <li><a href="#expected-output-structure" id="toc-expected-output-structure" class="nav-link" data-scroll-target="#expected-output-structure">1.3 Expected Output Structure</a></li>
  </ul></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup">2 SETUP</a>
  <ul>
  <li><a href="#libraries" id="toc-libraries" class="nav-link" data-scroll-target="#libraries">2.1 Libraries</a></li>
  <li><a href="#set-r-variables" id="toc-set-r-variables" class="nav-link" data-scroll-target="#set-r-variables">2.2 Set R variables</a></li>
  <li><a href="#load-project-python-environment" id="toc-load-project-python-environment" class="nav-link" data-scroll-target="#load-project-python-environment">2.3 Load project Python environment</a></li>
  <li><a href="#test-python-environment" id="toc-test-python-environment" class="nav-link" data-scroll-target="#test-python-environment">2.4 Test Python Environment</a></li>
  </ul></li>
  <li><a href="#stdm-analysis" id="toc-stdm-analysis" class="nav-link" data-scroll-target="#stdm-analysis">3 STDM ANALYSIS</a>
  <ul>
  <li><a href="#transform-data-to-stdm-format" id="toc-transform-data-to-stdm-format" class="nav-link" data-scroll-target="#transform-data-to-stdm-format">3.1 Transform data to stdm format</a></li>
  <li><a href="#rank-comparison-analysis" id="toc-rank-comparison-analysis" class="nav-link" data-scroll-target="#rank-comparison-analysis">3.2 Rank comparison analysis</a></li>
  <li><a href="#analyze-rank-comparison-results" id="toc-analyze-rank-comparison-results" class="nav-link" data-scroll-target="#analyze-rank-comparison-results">3.3 Analyze rank comparison results</a></li>
  <li><a href="#analyze-optimal-decomposition-results" id="toc-analyze-optimal-decomposition-results" class="nav-link" data-scroll-target="#analyze-optimal-decomposition-results">3.4 Analyze optimal decomposition results</a></li>
  </ul></li>
  <li><a href="#visualizations" id="toc-visualizations" class="nav-link" data-scroll-target="#visualizations">4 VISUALIZATIONS</a>
  <ul>
  <li><a href="#generate-visualizations-for-all-tested-ranks" id="toc-generate-visualizations-for-all-tested-ranks" class="nav-link" data-scroll-target="#generate-visualizations-for-all-tested-ranks">4.1 Generate visualizations for all tested ranks</a></li>
  <li><a href="#visualize-optimal-decomposition-results" id="toc-visualize-optimal-decomposition-results" class="nav-link" data-scroll-target="#visualize-optimal-decomposition-results">4.2 Visualize optimal decomposition results</a></li>
  <li><a href="#individual-plot-1-component-strengths" id="toc-individual-plot-1-component-strengths" class="nav-link" data-scroll-target="#individual-plot-1-component-strengths">4.3 Individual Plot 1: Component Strengths</a></li>
  <li><a href="#individual-plot-2-temporal-patterns-heatmap" id="toc-individual-plot-2-temporal-patterns-heatmap" class="nav-link" data-scroll-target="#individual-plot-2-temporal-patterns-heatmap">4.4 Individual Plot 2: Temporal Patterns Heatmap</a></li>
  <li><a href="#individual-plot-3-species-patterns-heatmap" id="toc-individual-plot-3-species-patterns-heatmap" class="nav-link" data-scroll-target="#individual-plot-3-species-patterns-heatmap">4.5 Individual Plot 3: Species Patterns Heatmap</a></li>
  <li><a href="#individual-plot-4-temporal-trajectories" id="toc-individual-plot-4-temporal-trajectories" class="nav-link" data-scroll-target="#individual-plot-4-temporal-trajectories">4.6 Individual Plot 4: Temporal Trajectories</a></li>
  <li><a href="#individual-plot-5-species-clustering-pca" id="toc-individual-plot-5-species-clustering-pca" class="nav-link" data-scroll-target="#individual-plot-5-species-clustering-pca">4.7 Individual Plot 5: Species Clustering PCA</a></li>
  <li><a href="#individual-plot-6-gene-factor-distribution" id="toc-individual-plot-6-gene-factor-distribution" class="nav-link" data-scroll-target="#individual-plot-6-gene-factor-distribution">4.8 Individual Plot 6: Gene Factor Distribution</a></li>
  <li><a href="#individual-plot-7-component-correlations" id="toc-individual-plot-7-component-correlations" class="nav-link" data-scroll-target="#individual-plot-7-component-correlations">4.9 Individual Plot 7: Component Correlations</a></li>
  <li><a href="#individual-plot-8-species-type-comparison" id="toc-individual-plot-8-species-type-comparison" class="nav-link" data-scroll-target="#individual-plot-8-species-type-comparison">4.10 Individual Plot 8: Species Type Comparison</a></li>
  <li><a href="#individual-plot-9-temporal-variance-per-component" id="toc-individual-plot-9-temporal-variance-per-component" class="nav-link" data-scroll-target="#individual-plot-9-temporal-variance-per-component">4.11 Individual Plot 9: Temporal Variance per Component</a></li>
  <li><a href="#individual-plot-10-species-variance-per-component" id="toc-individual-plot-10-species-variance-per-component" class="nav-link" data-scroll-target="#individual-plot-10-species-variance-per-component">4.12 Individual Plot 10: Species Variance per Component</a></li>
  <li><a href="#individual-plot-11-top-genes-heatmap" id="toc-individual-plot-11-top-genes-heatmap" class="nav-link" data-scroll-target="#individual-plot-11-top-genes-heatmap">4.13 Individual Plot 11: Top Genes Heatmap</a></li>
  <li><a href="#individual-plot-12-reconstruction-quality" id="toc-individual-plot-12-reconstruction-quality" class="nav-link" data-scroll-target="#individual-plot-12-reconstruction-quality">4.14 Individual Plot 12: Reconstruction Quality</a></li>
  <li><a href="#interpret-species-clustering-patterns" id="toc-interpret-species-clustering-patterns" class="nav-link" data-scroll-target="#interpret-species-clustering-patterns">4.15 Interpret species clustering patterns</a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-right" id="quarto-document-content">





<section id="intro" class="level1">
<h1>INTRO</h1>
<p>This notebook is an introductory exploration of the E5 coral <a href="https://github.com/urol-e5/timeseries_molecular"><code>urol-e5/timeseries_molecular</code></a> (GitHub repo) multi-species data using a tensor decomposition approach, using <a href="https://github.com/sr320/workflow-stdm"><code>workflow-stdm</code></a> (GitHub Repo), a tensor decomposition package Steven created using an AI agent.</p>
<p>I misunderstood what the goal was, until well after I went through the trouble of running this.</p>
<p>The primary goal was simply to run the package and see what types of files/visualizations it generated. It successfully generates the expected output files; primarily numpy binary files for each factor:</p>
<ul>
<li><code>gene_factors.npy</code></li>
<li><code>recontstructed_tensory.npy</code></li>
<li><code>species_factorys.npy</code></li>
<li><code>summary.json</code></li>
<li><code>time_factors.npy</code></li>
</ul>
<p>I think Steven would’ve liked if this package had also output some simple tables of the results that were human-readable and could be used to easy glance at the results, instead of all these numpy binary files.</p>
<p>Not realizing this at the time, I ran the package and worked heavily to assess/process the output data into something understandable. In the notebook below, I added steps to improve the input data formatting, process a range of ranks, compare those ranks, and select the “best” rank. Additionally, I added a number of visualizations to help better understand the results. However, like the default settings of the package, I did <em>not</em> add any steps to output results in any sort of table formats.</p>
<p>Overall, this analysis is likely similar to that of using <a href="https://github.com/blasks/barnacle">barnacle</a> (GitHub Repo) for this multi-omics analysis <a href="../../../posts/2025/2025-10-08-Data-Exploration---E5-Timeseries-A.pulchra-Multi-species-Multi-omics-Using-Barnacle/index.html">that I ran on 20251008</a> (notebook entry). Both greatly differ than <a href="../../../posts/2025/2025-09-19-Data-Exploration---E5-Timeseries-A.pulchra-Multi-omics-Using-MOFA2/index.html">the analysis I messed around with using MOFA2</a> (notebook entry).</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The contents below are from knitted markdown <a href="https://github.com/urol-e5/timeseries_molecular/blob/84515d3db6e8d3a3459764fb9a81b14e8f8b3426/M-multi-species/scripts/13.00-multiomics-stdm.Rmd"><code>13.00-multiomics-stdm.Rmd</code></a> (commit <code>84515d3</code>).</p>
</div>
</div>
</section>
<section id="background" class="level1">
<h1>1 BACKGROUND</h1>
<p>This notebook performs tensor decomposition analysis on multi-species gene expression timeseries data using Steven’s <a href="https://github.com/sr320/workflow-stdm"><code>workflow-stdm</code></a> (GitHub repo) tensor decomposition project.</p>
<p>The analysis identifies optimal tensor rank through reconstruction error analysis and provides biological interpretations of gene expression patterns across species and time.</p>
<section id="analysis-overview" class="level2">
<h2 class="anchored" data-anchor-id="analysis-overview">1.1 Analysis Overview</h2>
<p>The notebook conducts the following analyses:</p>
<ol type="1">
<li><strong>Data Transformation</strong>: Converts wide-format gene expression data to long format required by the stdm package</li>
<li><strong>Rank Comparison</strong>: Tests tensor decomposition across multiple ranks (2-15) to identify optimal number of components</li>
<li><strong>Tensor Decomposition</strong>: Performs PARAFAC decomposition to identify gene, species, and temporal patterns</li>
<li><strong>Visualization</strong>: Creates comprehensive plots to interpret biological patterns in the decomposition results</li>
</ol>
</section>
<section id="input-files" class="level2">
<h2 class="anchored" data-anchor-id="input-files">1.2 Input Files</h2>
<ul>
<li><strong>Primary Input</strong>: <code>/home/shared/16TB_HDD_01/sam/gitrepos/sr320/workflow-stdm/input-data/vst_counts_matrix.csv</code>
<ul>
<li>Wide-format VST-normalized gene expression data</li>
<li>Columns: gene IDs and sample names (format: ACR-139-TP1, POR-216-TP2, etc.)</li>
<li>Sample names encode: Species (ACR/POR/POC) - Individual ID - Timepoint (TP1-TP4)</li>
</ul></li>
</ul>
</section>
<section id="expected-output-structure" class="level2">
<h2 class="anchored" data-anchor-id="expected-output-structure">1.3 Expected Output Structure</h2>
<pre><code>../output/13.00-multiomics-stdm/
├── vst_counts_matrix_long_format.csv                    # Transformed input data
├── YYYYMMDD_HHMMSS/                                     # Main timestamped results directory
│   ├── rank_analysis.png                               # Rank selection visualization
│   ├── rank_selection_summary.json                     # Optimal rank determination
│   ├── optimal_tensor_decomposition_visualization_rank_X.png  # Comprehensive plots
│   ├── plot_01_component_strengths_optimal_rank_X.png   # Individual visualizations
│   ├── plot_02_temporal_heatmap_optimal_rank_X.png
│   ├── plot_03_species_heatmap_optimal_rank_X.png
│   ├── plot_04_temporal_trajectories_optimal_rank_X.png
│   ├── plot_05_species_clustering_optimal_rank_X.png
│   └── rank_comparison/                                 # Rank comparison results
│       ├── rank_comparison_results.json                # Summary of all ranks tested
│       ├── rank_02/                                    # Results for each rank
│       │   └── YYYYMMDD_HHMMSS/                        # Timestamped stdm output
│       │       ├── summary.json                        # Decomposition metrics
│       │       ├── gene_factors.npy                    # Gene factor matrix
│       │       ├── species_factors.npy                 # Species factor matrix
│       │       ├── time_factors.npy                    # Time factor matrix
│       │       ├── reconstructed_tensor.npy            # Reconstructed data
│       │       └── rank_02_comprehensive_visualization.png
│       ├── rank_03/
│       │   └── YYYYMMDD_HHMMSS/
│       │       └── [same structure as rank_02]
│       └── ...                                         # Additional ranks (up to rank_15)</code></pre>
</section>
</section>
<section id="setup" class="level1">
<h1>2 SETUP</h1>
<section id="libraries" class="level2">
<h2 class="anchored" data-anchor-id="libraries">2.1 Libraries</h2>
</section>
<section id="set-r-variables" class="level2">
<h2 class="anchored" data-anchor-id="set-r-variables">2.2 Set R variables</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># OUTPUT DIRECTORY</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>output_dir <span class="ot">&lt;-</span> <span class="st">"../output/13.00-multiomics-stdm"</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#INPUT FILE(S)</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># PYTHON ENVIRONMENT - Use the project's .venv environment</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>python_path <span class="ot">&lt;-</span> <span class="st">"/home/shared/16TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/.venv/bin/python"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="load-project-python-environment" class="level2">
<h2 class="anchored" data-anchor-id="load-project-python-environment">2.3 Load project Python environment</h2>
<p>The project uses a virtual environment (<code>.venv</code>) with all required packages including <code>stdm</code>. If this is successful, the Python path should show the <code>.venv</code> environment.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the project's .venv Python environment directly</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reticulate)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Set Python path to the project's .venv</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">use_python</span>(python_path, <span class="at">required =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Show final configuration</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">py_config</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## python:         /home/shared/16TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/.venv/bin/python
## libpython:      /usr/lib/python3.12/config-3.12-x86_64-linux-gnu/libpython3.12.so
## pythonhome:     /home/shared/16TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/.venv:/home/shared/16TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/.venv
## version:        3.12.3 (main, Aug 14 2025, 17:47:21) [GCC 13.3.0]
## numpy:          /home/shared/16TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/.venv/lib/python3.12/site-packages/numpy
## numpy_version:  2.3.3
## 
## NOTE: Python version was forced by use_python() function</code></pre>
</section>
<section id="test-python-environment" class="level2">
<h2 class="anchored" data-anchor-id="test-python-environment">2.4 Test Python Environment</h2>
<p>Let’s verify the Python environment has all required libraries:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> sys</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Python executable: </span><span class="sc">{</span>sys<span class="sc">.</span>executable<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Test problematic imports</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="cf">try</span>:</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>    <span class="im">import</span> PIL</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"✓ PIL imported successfully"</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="cf">except</span> <span class="pp">ImportError</span> <span class="im">as</span> e:</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"✗ PIL import failed: </span><span class="sc">{</span>e<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="cf">try</span>:</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>    <span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"✓ matplotlib.pyplot imported successfully"</span>)</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="cf">except</span> <span class="pp">ImportError</span> <span class="im">as</span> e:</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"✗ matplotlib.pyplot import failed: </span><span class="sc">{</span>e<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="cf">try</span>:</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>    <span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>    <span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"✓ pandas and numpy imported successfully"</span>)</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="cf">except</span> <span class="pp">ImportError</span> <span class="im">as</span> e:</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"✗ pandas/numpy import failed: </span><span class="sc">{</span>e<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Environment test completed."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Python executable: /home/shared/16TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/.venv/bin/python
## ✓ PIL imported successfully
## ✓ matplotlib.pyplot imported successfully
## ✓ pandas and numpy imported successfully
## Environment test completed.</code></pre>
</section>
</section>
<section id="stdm-analysis" class="level1">
<h1>3 STDM ANALYSIS</h1>
<section id="transform-data-to-stdm-format" class="level2">
<h2 class="anchored" data-anchor-id="transform-data-to-stdm-format">3.1 Transform data to stdm format</h2>
<p>The stdm package expects data in long format with columns: <code>gene</code>, <code>species</code>, <code>timepoint</code>, <code>expression</code>. Our current data is in wide format, so we need to transform it first.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> re</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Create output directory if it doesn't exist</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>os.makedirs(output_dir, exist_ok<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the wide-format data</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Reading wide-format data..."</span>)</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>input_file <span class="op">=</span> <span class="st">"/home/shared/16TB_HDD_01/sam/gitrepos/sr320/workflow-stdm/input-data/vst_counts_matrix.csv"</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>df_wide <span class="op">=</span> pd.read_csv(input_file)</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Original data shape: </span><span class="sc">{</span>df_wide<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Columns: </span><span class="sc">{</span><span class="bu">list</span>(df_wide.columns[:<span class="dv">5</span>])<span class="sc">}</span><span class="ss">... (showing first 5)"</span>)</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Create long-format data</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Transforming to long format..."</span>)</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>data_long <span class="op">=</span> []</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Get gene column (first column)</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>gene_col <span class="op">=</span> df_wide.columns[<span class="dv">0</span>]  <span class="co"># Should be 'group_id'</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>sample_cols <span class="op">=</span> df_wide.columns[<span class="dv">1</span>:]  <span class="co"># All sample columns</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> _, row <span class="kw">in</span> df_wide.iterrows():</span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>    gene <span class="op">=</span> row[gene_col]</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> sample_col <span class="kw">in</span> sample_cols:</span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>        expression_value <span class="op">=</span> row[sample_col]</span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Parse sample name to extract species and timepoint</span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Expected format: ACR-139-TP1, POR-216-TP2, POC-201-TP3, etc.</span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>        match <span class="op">=</span> re.match(<span class="vs">r'([A-Z]+)-(\d+)-(TP\d+)'</span>, sample_col)</span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> match:</span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>            species_prefix <span class="op">=</span> match.group(<span class="dv">1</span>)  <span class="co"># ACR, POR, POC (species designation)</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>            individual_id <span class="op">=</span> match.group(<span class="dv">2</span>)   <span class="co"># 139, 216, 201 (individual/sample ID)</span></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>            timepoint <span class="op">=</span> match.group(<span class="dv">3</span>)       <span class="co"># TP1, TP2, TP3, TP4</span></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Create individual identifier (species-individual combination)</span></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>            species <span class="op">=</span> <span class="ss">f"</span><span class="sc">{</span>species_prefix<span class="sc">}</span><span class="ss">-</span><span class="sc">{</span>individual_id<span class="sc">}</span><span class="ss">"</span></span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a>            data_long.append({</span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>                <span class="st">'gene'</span>: gene,</span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>                <span class="st">'species'</span>: species,</span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a>                <span class="st">'timepoint'</span>: timepoint,</span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a>                <span class="st">'expression'</span>: expression_value</span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>            })</span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Warning: Could not parse sample name: </span><span class="sc">{</span>sample_col<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a><span class="co"># Create DataFrame</span></span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a>df_long <span class="op">=</span> pd.DataFrame(data_long)</span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Transformed data shape: </span><span class="sc">{</span>df_long<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Number of unique genes: </span><span class="sc">{</span>df_long[<span class="st">'gene'</span>]<span class="sc">.</span>nunique()<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Number of unique species: </span><span class="sc">{</span>df_long[<span class="st">'species'</span>]<span class="sc">.</span>nunique()<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Number of unique timepoints: </span><span class="sc">{</span>df_long[<span class="st">'timepoint'</span>]<span class="sc">.</span>nunique()<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Show sample of transformed data</span></span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">Sample of transformed data:"</span>)</span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(df_long.head(<span class="dv">10</span>))</span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Save transformed data</span></span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a>output_file <span class="op">=</span> os.path.join(output_dir, <span class="st">"vst_counts_matrix_long_format.csv"</span>)</span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a>df_long.to_csv(output_file, index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Transformed data saved to: </span><span class="sc">{</span>output_file<span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Reading wide-format data...
## Original data shape: (9800, 118)
## Columns: ['group_id', 'ACR-139-TP1', 'ACR-139-TP2', 'ACR-139-TP3', 'ACR-139-TP4']... (showing first 5)
## Transforming to long format...
## Transformed data shape: (1146600, 4)
## Number of unique genes: 9800
## Number of unique species: 30
## Number of unique timepoints: 4
## 
## Sample of transformed data:
##        gene  species timepoint  expression
## 0  OG_00686  ACR-139       TP1    5.971679
## 1  OG_00686  ACR-139       TP2    6.206077
## 2  OG_00686  ACR-139       TP3    6.379771
## 3  OG_00686  ACR-139       TP4    5.882078
## 4  OG_00686  ACR-145       TP1    5.414779
## 5  OG_00686  ACR-145       TP2    7.260837
## 6  OG_00686  ACR-145       TP3    6.630233
## 7  OG_00686  ACR-145       TP4    5.533403
## 8  OG_00686  ACR-150       TP1    5.922190
## 9  OG_00686  ACR-150       TP2    6.834550
## 
## Transformed data saved to: ../output/13.00-multiomics-stdm/vst_counts_matrix_long_format.csv</code></pre>
</section>
<section id="rank-comparison-analysis" class="level2">
<h2 class="anchored" data-anchor-id="rank-comparison-analysis">3.2 Rank comparison analysis</h2>
<p>First, let’s run tensor decomposition across multiple ranks to find the optimal number of components.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> subprocess</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> datetime <span class="im">import</span> datetime</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Define ranks to test (2-15 should cover the useful range)</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>ranks_to_test <span class="op">=</span> <span class="bu">list</span>(<span class="bu">range</span>(<span class="dv">2</span>, <span class="dv">16</span>))</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the transformed data file</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>input_file <span class="op">=</span> os.path.join(output_dir, <span class="st">"vst_counts_matrix_long_format.csv"</span>)</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Running stdm decomposition across </span><span class="sc">{</span><span class="bu">len</span>(ranks_to_test)<span class="sc">}</span><span class="ss"> different ranks..."</span>)</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Ranks to test: </span><span class="sc">{</span>ranks_to_test<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Input file: </span><span class="sc">{</span>input_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Create main timestamped directory for all results</span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>main_timestamp <span class="op">=</span> datetime.now().strftime(<span class="st">"%Y%m</span><span class="sc">%d</span><span class="st">_%H%M%S"</span>)</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>main_results_dir <span class="op">=</span> os.path.join(output_dir, main_timestamp)</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>os.makedirs(main_results_dir, exist_ok<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a subdirectory for rank comparison within main results</span></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>os.makedirs(comparison_dir, exist_ok<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Dictionary to store results</span></span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>rank_results <span class="op">=</span> {}</span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Run decomposition for each rank</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> rank <span class="kw">in</span> ranks_to_test:</span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="sc">{</span><span class="st">'='</span><span class="op">*</span><span class="dv">50</span><span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Processing rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">..."</span>)</span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span><span class="st">'='</span><span class="op">*</span><span class="dv">50</span><span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create rank-specific subdirectory</span></span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>    rank_subdir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a>    os.makedirs(rank_subdir, exist_ok<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Run stdm decomposition</span></span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a>    <span class="cf">try</span>:</span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a>        result <span class="op">=</span> subprocess.run([</span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>            <span class="st">"stdm"</span>, <span class="st">"run"</span>,</span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a>            <span class="st">"--input"</span>, input_file,</span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>            <span class="st">"--output"</span>, rank_subdir,</span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>            <span class="st">"--rank"</span>, <span class="bu">str</span>(rank),</span>
<span id="cb9-52"><a href="#cb9-52" aria-hidden="true" tabindex="-1"></a>            <span class="st">"--method"</span>, <span class="st">"parafac"</span>,</span>
<span id="cb9-53"><a href="#cb9-53" aria-hidden="true" tabindex="-1"></a>            <span class="st">"--sparsity-threshold"</span>, <span class="st">"0.01"</span>,</span>
<span id="cb9-54"><a href="#cb9-54" aria-hidden="true" tabindex="-1"></a>            <span class="st">"--no-log-transform"</span>,</span>
<span id="cb9-55"><a href="#cb9-55" aria-hidden="true" tabindex="-1"></a>            <span class="st">"--standardize"</span>,</span>
<span id="cb9-56"><a href="#cb9-56" aria-hidden="true" tabindex="-1"></a>            <span class="st">"--verbose"</span></span>
<span id="cb9-57"><a href="#cb9-57" aria-hidden="true" tabindex="-1"></a>        ], capture_output<span class="op">=</span><span class="va">True</span>, text<span class="op">=</span><span class="va">True</span>, check<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb9-58"><a href="#cb9-58" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb9-59"><a href="#cb9-59" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the results directory (should be timestamped)</span></span>
<span id="cb9-60"><a href="#cb9-60" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(rank_subdir, <span class="st">"20*"</span>))</span>
<span id="cb9-61"><a href="#cb9-61" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb9-62"><a href="#cb9-62" aria-hidden="true" tabindex="-1"></a>            latest_result_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb9-63"><a href="#cb9-63" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb9-64"><a href="#cb9-64" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Load summary statistics</span></span>
<span id="cb9-65"><a href="#cb9-65" aria-hidden="true" tabindex="-1"></a>            summary_file <span class="op">=</span> os.path.join(latest_result_dir, <span class="st">"summary.json"</span>)</span>
<span id="cb9-66"><a href="#cb9-66" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb9-67"><a href="#cb9-67" aria-hidden="true" tabindex="-1"></a>                <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb9-68"><a href="#cb9-68" aria-hidden="true" tabindex="-1"></a>                    summary <span class="op">=</span> json.load(f)</span>
<span id="cb9-69"><a href="#cb9-69" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb9-70"><a href="#cb9-70" aria-hidden="true" tabindex="-1"></a>                rank_results[rank] <span class="op">=</span> {</span>
<span id="cb9-71"><a href="#cb9-71" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'reconstruction_error'</span>: summary[<span class="st">'reconstruction_error'</span>],</span>
<span id="cb9-72"><a href="#cb9-72" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'rank'</span>: summary[<span class="st">'rank'</span>],</span>
<span id="cb9-73"><a href="#cb9-73" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'method'</span>: summary[<span class="st">'method'</span>],</span>
<span id="cb9-74"><a href="#cb9-74" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'result_dir'</span>: latest_result_dir</span>
<span id="cb9-75"><a href="#cb9-75" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb9-76"><a href="#cb9-76" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb9-77"><a href="#cb9-77" aria-hidden="true" tabindex="-1"></a>                <span class="bu">print</span>(<span class="ss">f"  Rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">: Reconstruction error = </span><span class="sc">{</span>summary[<span class="st">'reconstruction_error'</span>]<span class="sc">:.6f}</span><span class="ss">"</span>)</span>
<span id="cb9-78"><a href="#cb9-78" aria-hidden="true" tabindex="-1"></a>            <span class="cf">else</span>:</span>
<span id="cb9-79"><a href="#cb9-79" aria-hidden="true" tabindex="-1"></a>                <span class="bu">print</span>(<span class="ss">f"  Warning: No summary file found for rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-80"><a href="#cb9-80" aria-hidden="true" tabindex="-1"></a>                rank_results[rank] <span class="op">=</span> {<span class="st">'reconstruction_error'</span>: np.nan, <span class="st">'rank'</span>: rank, <span class="st">'result_dir'</span>: <span class="va">None</span>}</span>
<span id="cb9-81"><a href="#cb9-81" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb9-82"><a href="#cb9-82" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  Warning: No results directory found for rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-83"><a href="#cb9-83" aria-hidden="true" tabindex="-1"></a>            rank_results[rank] <span class="op">=</span> {<span class="st">'reconstruction_error'</span>: np.nan, <span class="st">'rank'</span>: rank, <span class="st">'result_dir'</span>: <span class="va">None</span>}</span>
<span id="cb9-84"><a href="#cb9-84" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb9-85"><a href="#cb9-85" aria-hidden="true" tabindex="-1"></a>    <span class="cf">except</span> subprocess.CalledProcessError <span class="im">as</span> e:</span>
<span id="cb9-86"><a href="#cb9-86" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"  Error running decomposition for rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">: </span><span class="sc">{</span>e<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-87"><a href="#cb9-87" aria-hidden="true" tabindex="-1"></a>        rank_results[rank] <span class="op">=</span> {<span class="st">'reconstruction_error'</span>: np.nan, <span class="st">'rank'</span>: rank, <span class="st">'result_dir'</span>: <span class="va">None</span>}</span>
<span id="cb9-88"><a href="#cb9-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-89"><a href="#cb9-89" aria-hidden="true" tabindex="-1"></a><span class="co"># Save rank comparison results</span></span>
<span id="cb9-90"><a href="#cb9-90" aria-hidden="true" tabindex="-1"></a>comparison_results_file <span class="op">=</span> os.path.join(comparison_dir, <span class="st">"rank_comparison_results.json"</span>)</span>
<span id="cb9-91"><a href="#cb9-91" aria-hidden="true" tabindex="-1"></a><span class="cf">with</span> <span class="bu">open</span>(comparison_results_file, <span class="st">'w'</span>) <span class="im">as</span> f:</span>
<span id="cb9-92"><a href="#cb9-92" aria-hidden="true" tabindex="-1"></a>    json.dump(rank_results, f, indent<span class="op">=</span><span class="dv">2</span>)</span>
<span id="cb9-93"><a href="#cb9-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-94"><a href="#cb9-94" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="sc">{</span><span class="st">'='</span><span class="op">*</span><span class="dv">60</span><span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-95"><a href="#cb9-95" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"RANK COMPARISON COMPLETED"</span>)</span>
<span id="cb9-96"><a href="#cb9-96" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span><span class="st">'='</span><span class="op">*</span><span class="dv">60</span><span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-97"><a href="#cb9-97" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Main results directory: </span><span class="sc">{</span>main_results_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-98"><a href="#cb9-98" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Rank comparison data: </span><span class="sc">{</span>comparison_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-99"><a href="#cb9-99" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Comparison results file: </span><span class="sc">{</span>comparison_results_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-100"><a href="#cb9-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-101"><a href="#cb9-101" aria-hidden="true" tabindex="-1"></a><span class="co"># Display summary table</span></span>
<span id="cb9-102"><a href="#cb9-102" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">SUMMARY TABLE:"</span>)</span>
<span id="cb9-103"><a href="#cb9-103" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span><span class="st">'Rank'</span><span class="sc">:&lt;6}</span><span class="ss"> </span><span class="sc">{</span><span class="st">'Reconstruction Error'</span><span class="sc">:&lt;20}</span><span class="ss"> </span><span class="sc">{</span><span class="st">'Status'</span><span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-104"><a href="#cb9-104" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span><span class="st">'-'</span><span class="op">*</span><span class="dv">40</span><span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-105"><a href="#cb9-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-106"><a href="#cb9-106" aria-hidden="true" tabindex="-1"></a>valid_results <span class="op">=</span> {}</span>
<span id="cb9-107"><a href="#cb9-107" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> rank <span class="kw">in</span> ranks_to_test:</span>
<span id="cb9-108"><a href="#cb9-108" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> rank <span class="kw">in</span> rank_results <span class="kw">and</span> <span class="kw">not</span> np.isnan(rank_results[rank][<span class="st">'reconstruction_error'</span>]):</span>
<span id="cb9-109"><a href="#cb9-109" aria-hidden="true" tabindex="-1"></a>        error <span class="op">=</span> rank_results[rank][<span class="st">'reconstruction_error'</span>]</span>
<span id="cb9-110"><a href="#cb9-110" aria-hidden="true" tabindex="-1"></a>        valid_results[rank] <span class="op">=</span> error</span>
<span id="cb9-111"><a href="#cb9-111" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span>rank<span class="sc">:&lt;6}</span><span class="ss"> </span><span class="sc">{</span>error<span class="sc">:&lt;20.6f}</span><span class="ss"> </span><span class="sc">{</span><span class="st">'Success'</span><span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-112"><a href="#cb9-112" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb9-113"><a href="#cb9-113" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span>rank<span class="sc">:&lt;6}</span><span class="ss"> </span><span class="sc">{</span><span class="st">'N/A'</span><span class="sc">:&lt;20}</span><span class="ss"> </span><span class="sc">{</span><span class="st">'Failed'</span><span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb9-114"><a href="#cb9-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-115"><a href="#cb9-115" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> valid_results:</span>
<span id="cb9-116"><a href="#cb9-116" aria-hidden="true" tabindex="-1"></a>    best_rank <span class="op">=</span> <span class="bu">min</span>(valid_results.keys(), key<span class="op">=</span><span class="kw">lambda</span> k: valid_results[k])</span>
<span id="cb9-117"><a href="#cb9-117" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Best rank by reconstruction error: </span><span class="sc">{</span>best_rank<span class="sc">}</span><span class="ss"> (error: </span><span class="sc">{</span>valid_results[best_rank]<span class="sc">:.6f}</span><span class="ss">)"</span>)</span>
<span id="cb9-118"><a href="#cb9-118" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb9-119"><a href="#cb9-119" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">No successful decompositions found!"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Running stdm decomposition across 14 different ranks...
## Ranks to test: [2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
## Input file: ../output/13.00-multiomics-stdm/vst_counts_matrix_long_format.csv
## 
## ==================================================
## Processing rank 2...
## ==================================================
##   Rank 2: Reconstruction error = 0.643202
## 
## ==================================================
## Processing rank 3...
## ==================================================
##   Rank 3: Reconstruction error = 0.544191
## 
## ==================================================
## Processing rank 4...
## ==================================================
##   Rank 4: Reconstruction error = 0.470061
## 
## ==================================================
## Processing rank 5...
## ==================================================
##   Rank 5: Reconstruction error = 0.451016
## 
## ==================================================
## Processing rank 6...
## ==================================================
##   Rank 6: Reconstruction error = 0.434595
## 
## ==================================================
## Processing rank 7...
## ==================================================
##   Rank 7: Reconstruction error = 0.423404
## 
## ==================================================
## Processing rank 8...
## ==================================================
##   Rank 8: Reconstruction error = 0.410814
## 
## ==================================================
## Processing rank 9...
## ==================================================
##   Rank 9: Reconstruction error = 0.408422
## 
## ==================================================
## Processing rank 10...
## ==================================================
##   Rank 10: Reconstruction error = 0.395594
## 
## ==================================================
## Processing rank 11...
## ==================================================
##   Rank 11: Reconstruction error = 0.385213
## 
## ==================================================
## Processing rank 12...
## ==================================================
##   Rank 12: Reconstruction error = 0.384416
## 
## ==================================================
## Processing rank 13...
## ==================================================
##   Rank 13: Reconstruction error = 0.372751
## 
## ==================================================
## Processing rank 14...
## ==================================================
##   Rank 14: Reconstruction error = 0.364498
## 
## ==================================================
## Processing rank 15...
## ==================================================
##   Rank 15: Reconstruction error = 0.360396
## 
## ============================================================
## RANK COMPARISON COMPLETED
## ============================================================
## Main results directory: ../output/13.00-multiomics-stdm/20251015_190157
## Rank comparison data: ../output/13.00-multiomics-stdm/20251015_190157/rank_comparison
## Comparison results file: ../output/13.00-multiomics-stdm/20251015_190157/rank_comparison/rank_comparison_results.json
## 
## SUMMARY TABLE:
## Rank   Reconstruction Error Status
## ----------------------------------------
## 2      0.643202             Success
## 3      0.544191             Success
## 4      0.470061             Success
## 5      0.451016             Success
## 6      0.434595             Success
## 7      0.423404             Success
## 8      0.410814             Success
## 9      0.408422             Success
## 10     0.395594             Success
## 11     0.385213             Success
## 12     0.384416             Success
## 13     0.372751             Success
## 14     0.364498             Success
## 15     0.360396             Success
## 
## Best rank by reconstruction error: 15 (error: 0.360396)</code></pre>
</section>
<section id="analyze-rank-comparison-results" class="level2">
<h2 class="anchored" data-anchor-id="analyze-rank-comparison-results">3.3 Analyze rank comparison results</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory and find the main results directory</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Find the most recent main results directory (timestamped)</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_result_dirs:</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Fallback to old structure if new structure doesn't exist yet</span></span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(output_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> output_dir</span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Load rank comparison results</span></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>comparison_results_file <span class="op">=</span> os.path.join(comparison_dir, <span class="st">"rank_comparison_results.json"</span>)</span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> os.path.exists(comparison_results_file):</span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>    <span class="cf">with</span> <span class="bu">open</span>(comparison_results_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>        rank_results <span class="op">=</span> json.load(f)</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Convert to DataFrame for easier analysis</span></span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>    results_data <span class="op">=</span> []</span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> rank_str, data <span class="kw">in</span> rank_results.items():</span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a>        rank <span class="op">=</span> <span class="bu">int</span>(rank_str)</span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="kw">not</span> np.isnan(data[<span class="st">'reconstruction_error'</span>]):</span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a>            results_data.append({</span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a>                <span class="st">'rank'</span>: rank,</span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a>                <span class="st">'reconstruction_error'</span>: data[<span class="st">'reconstruction_error'</span>],</span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>                <span class="st">'result_dir'</span>: data[<span class="st">'result_dir'</span>]</span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a>            })</span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> results_data:</span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a>        df_results <span class="op">=</span> pd.DataFrame(results_data)</span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a>        df_results <span class="op">=</span> df_results.sort_values(<span class="st">'rank'</span>)</span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"="</span><span class="op">*</span><span class="dv">60</span>)</span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"RANK SELECTION ANALYSIS"</span>)</span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"="</span><span class="op">*</span><span class="dv">60</span>)</span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Calculate additional metrics</span></span>
<span id="cb11-54"><a href="#cb11-54" aria-hidden="true" tabindex="-1"></a>        df_results[<span class="st">'error_change'</span>] <span class="op">=</span> df_results[<span class="st">'reconstruction_error'</span>].diff()</span>
<span id="cb11-55"><a href="#cb11-55" aria-hidden="true" tabindex="-1"></a>        df_results[<span class="st">'error_change_pct'</span>] <span class="op">=</span> df_results[<span class="st">'reconstruction_error'</span>].pct_change() <span class="op">*</span> <span class="dv">100</span></span>
<span id="cb11-56"><a href="#cb11-56" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-57"><a href="#cb11-57" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find elbow point using second derivative</span></span>
<span id="cb11-58"><a href="#cb11-58" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="bu">len</span>(df_results) <span class="op">&gt;=</span> <span class="dv">3</span>:</span>
<span id="cb11-59"><a href="#cb11-59" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Calculate second derivative (acceleration of error decrease)</span></span>
<span id="cb11-60"><a href="#cb11-60" aria-hidden="true" tabindex="-1"></a>            df_results[<span class="st">'second_derivative'</span>] <span class="op">=</span> df_results[<span class="st">'error_change'</span>].diff()</span>
<span id="cb11-61"><a href="#cb11-61" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb11-62"><a href="#cb11-62" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Find elbow (where second derivative is maximum - greatest change in slope)</span></span>
<span id="cb11-63"><a href="#cb11-63" aria-hidden="true" tabindex="-1"></a>            elbow_idx <span class="op">=</span> df_results[<span class="st">'second_derivative'</span>].idxmax()</span>
<span id="cb11-64"><a href="#cb11-64" aria-hidden="true" tabindex="-1"></a>            elbow_rank <span class="op">=</span> df_results.loc[elbow_idx, <span class="st">'rank'</span>] <span class="cf">if</span> <span class="kw">not</span> pd.isna(elbow_idx) <span class="cf">else</span> <span class="va">None</span></span>
<span id="cb11-65"><a href="#cb11-65" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb11-66"><a href="#cb11-66" aria-hidden="true" tabindex="-1"></a>            elbow_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb11-67"><a href="#cb11-67" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-68"><a href="#cb11-68" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Create comprehensive visualization</span></span>
<span id="cb11-69"><a href="#cb11-69" aria-hidden="true" tabindex="-1"></a>        fig, axes <span class="op">=</span> plt.subplots(<span class="dv">2</span>, <span class="dv">2</span>, figsize<span class="op">=</span>(<span class="dv">15</span>, <span class="dv">12</span>))</span>
<span id="cb11-70"><a href="#cb11-70" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-71"><a href="#cb11-71" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 1. Scree plot - Reconstruction error vs rank</span></span>
<span id="cb11-72"><a href="#cb11-72" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">0</span>, <span class="dv">0</span>].plot(df_results[<span class="st">'rank'</span>], df_results[<span class="st">'reconstruction_error'</span>], </span>
<span id="cb11-73"><a href="#cb11-73" aria-hidden="true" tabindex="-1"></a>                        <span class="st">'bo-'</span>, linewidth<span class="op">=</span><span class="dv">2</span>, markersize<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb11-74"><a href="#cb11-74" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">0</span>, <span class="dv">0</span>].set_xlabel(<span class="st">'Rank (Number of Components)'</span>)</span>
<span id="cb11-75"><a href="#cb11-75" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">0</span>, <span class="dv">0</span>].set_ylabel(<span class="st">'Reconstruction Error'</span>)</span>
<span id="cb11-76"><a href="#cb11-76" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">0</span>, <span class="dv">0</span>].set_title(<span class="st">'Scree Plot: Reconstruction Error vs Rank'</span>)</span>
<span id="cb11-77"><a href="#cb11-77" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">0</span>, <span class="dv">0</span>].grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb11-78"><a href="#cb11-78" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-79"><a href="#cb11-79" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Highlight elbow point</span></span>
<span id="cb11-80"><a href="#cb11-80" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> elbow_rank:</span>
<span id="cb11-81"><a href="#cb11-81" aria-hidden="true" tabindex="-1"></a>            elbow_error <span class="op">=</span> df_results[df_results[<span class="st">'rank'</span>] <span class="op">==</span> elbow_rank][<span class="st">'reconstruction_error'</span>].iloc[<span class="dv">0</span>]</span>
<span id="cb11-82"><a href="#cb11-82" aria-hidden="true" tabindex="-1"></a>            axes[<span class="dv">0</span>, <span class="dv">0</span>].plot(elbow_rank, elbow_error, <span class="st">'ro'</span>, markersize<span class="op">=</span><span class="dv">12</span>, </span>
<span id="cb11-83"><a href="#cb11-83" aria-hidden="true" tabindex="-1"></a>                           label<span class="op">=</span><span class="ss">f'Elbow Point (Rank </span><span class="sc">{</span>elbow_rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb11-84"><a href="#cb11-84" aria-hidden="true" tabindex="-1"></a>            axes[<span class="dv">0</span>, <span class="dv">0</span>].legend()</span>
<span id="cb11-85"><a href="#cb11-85" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-86"><a href="#cb11-86" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 2. Error reduction rate</span></span>
<span id="cb11-87"><a href="#cb11-87" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">0</span>, <span class="dv">1</span>].plot(df_results[<span class="st">'rank'</span>][<span class="dv">1</span>:], df_results[<span class="st">'error_change'</span>][<span class="dv">1</span>:], </span>
<span id="cb11-88"><a href="#cb11-88" aria-hidden="true" tabindex="-1"></a>                        <span class="st">'go-'</span>, linewidth<span class="op">=</span><span class="dv">2</span>, markersize<span class="op">=</span><span class="dv">6</span>)</span>
<span id="cb11-89"><a href="#cb11-89" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">0</span>, <span class="dv">1</span>].set_xlabel(<span class="st">'Rank'</span>)</span>
<span id="cb11-90"><a href="#cb11-90" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">0</span>, <span class="dv">1</span>].set_ylabel(<span class="st">'Error Change from Previous Rank'</span>)</span>
<span id="cb11-91"><a href="#cb11-91" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">0</span>, <span class="dv">1</span>].set_title(<span class="st">'Error Reduction Rate'</span>)</span>
<span id="cb11-92"><a href="#cb11-92" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">0</span>, <span class="dv">1</span>].grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb11-93"><a href="#cb11-93" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">0</span>, <span class="dv">1</span>].axhline(y<span class="op">=</span><span class="dv">0</span>, color<span class="op">=</span><span class="st">'r'</span>, linestyle<span class="op">=</span><span class="st">'--'</span>, alpha<span class="op">=</span><span class="fl">0.5</span>)</span>
<span id="cb11-94"><a href="#cb11-94" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-95"><a href="#cb11-95" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 3. Percentage change in error</span></span>
<span id="cb11-96"><a href="#cb11-96" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">1</span>, <span class="dv">0</span>].plot(df_results[<span class="st">'rank'</span>][<span class="dv">1</span>:], df_results[<span class="st">'error_change_pct'</span>][<span class="dv">1</span>:], </span>
<span id="cb11-97"><a href="#cb11-97" aria-hidden="true" tabindex="-1"></a>                        <span class="st">'mo-'</span>, linewidth<span class="op">=</span><span class="dv">2</span>, markersize<span class="op">=</span><span class="dv">6</span>)</span>
<span id="cb11-98"><a href="#cb11-98" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">1</span>, <span class="dv">0</span>].set_xlabel(<span class="st">'Rank'</span>)</span>
<span id="cb11-99"><a href="#cb11-99" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">1</span>, <span class="dv">0</span>].set_ylabel(<span class="st">'Error Change (%)'</span>)</span>
<span id="cb11-100"><a href="#cb11-100" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">1</span>, <span class="dv">0</span>].set_title(<span class="st">'Percentage Error Reduction'</span>)</span>
<span id="cb11-101"><a href="#cb11-101" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">1</span>, <span class="dv">0</span>].grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb11-102"><a href="#cb11-102" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">1</span>, <span class="dv">0</span>].axhline(y<span class="op">=</span><span class="dv">0</span>, color<span class="op">=</span><span class="st">'r'</span>, linestyle<span class="op">=</span><span class="st">'--'</span>, alpha<span class="op">=</span><span class="fl">0.5</span>)</span>
<span id="cb11-103"><a href="#cb11-103" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-104"><a href="#cb11-104" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 4. Second derivative (elbow detection)</span></span>
<span id="cb11-105"><a href="#cb11-105" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="bu">len</span>(df_results) <span class="op">&gt;=</span> <span class="dv">3</span>:</span>
<span id="cb11-106"><a href="#cb11-106" aria-hidden="true" tabindex="-1"></a>            axes[<span class="dv">1</span>, <span class="dv">1</span>].plot(df_results[<span class="st">'rank'</span>][<span class="dv">2</span>:], df_results[<span class="st">'second_derivative'</span>][<span class="dv">2</span>:], </span>
<span id="cb11-107"><a href="#cb11-107" aria-hidden="true" tabindex="-1"></a>                            <span class="st">'co-'</span>, linewidth<span class="op">=</span><span class="dv">2</span>, markersize<span class="op">=</span><span class="dv">6</span>)</span>
<span id="cb11-108"><a href="#cb11-108" aria-hidden="true" tabindex="-1"></a>            axes[<span class="dv">1</span>, <span class="dv">1</span>].set_xlabel(<span class="st">'Rank'</span>)</span>
<span id="cb11-109"><a href="#cb11-109" aria-hidden="true" tabindex="-1"></a>            axes[<span class="dv">1</span>, <span class="dv">1</span>].set_ylabel(<span class="st">'Second Derivative'</span>)</span>
<span id="cb11-110"><a href="#cb11-110" aria-hidden="true" tabindex="-1"></a>            axes[<span class="dv">1</span>, <span class="dv">1</span>].set_title(<span class="st">'Elbow Detection (Second Derivative)'</span>)</span>
<span id="cb11-111"><a href="#cb11-111" aria-hidden="true" tabindex="-1"></a>            axes[<span class="dv">1</span>, <span class="dv">1</span>].grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb11-112"><a href="#cb11-112" aria-hidden="true" tabindex="-1"></a>            axes[<span class="dv">1</span>, <span class="dv">1</span>].axhline(y<span class="op">=</span><span class="dv">0</span>, color<span class="op">=</span><span class="st">'r'</span>, linestyle<span class="op">=</span><span class="st">'--'</span>, alpha<span class="op">=</span><span class="fl">0.5</span>)</span>
<span id="cb11-113"><a href="#cb11-113" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb11-114"><a href="#cb11-114" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> elbow_rank <span class="kw">and</span> elbow_rank <span class="op">&gt;=</span> <span class="dv">3</span>:</span>
<span id="cb11-115"><a href="#cb11-115" aria-hidden="true" tabindex="-1"></a>                elbow_second_deriv <span class="op">=</span> df_results[df_results[<span class="st">'rank'</span>] <span class="op">==</span> elbow_rank][<span class="st">'second_derivative'</span>].iloc[<span class="dv">0</span>]</span>
<span id="cb11-116"><a href="#cb11-116" aria-hidden="true" tabindex="-1"></a>                axes[<span class="dv">1</span>, <span class="dv">1</span>].plot(elbow_rank, elbow_second_deriv, <span class="st">'ro'</span>, markersize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb11-117"><a href="#cb11-117" aria-hidden="true" tabindex="-1"></a>                               label<span class="op">=</span><span class="ss">f'Elbow Point (Rank </span><span class="sc">{</span>elbow_rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb11-118"><a href="#cb11-118" aria-hidden="true" tabindex="-1"></a>                axes[<span class="dv">1</span>, <span class="dv">1</span>].legend()</span>
<span id="cb11-119"><a href="#cb11-119" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb11-120"><a href="#cb11-120" aria-hidden="true" tabindex="-1"></a>            axes[<span class="dv">1</span>, <span class="dv">1</span>].text(<span class="fl">0.5</span>, <span class="fl">0.5</span>, <span class="st">'Not enough data points</span><span class="ch">\n</span><span class="st">for elbow detection'</span>, </span>
<span id="cb11-121"><a href="#cb11-121" aria-hidden="true" tabindex="-1"></a>                           transform<span class="op">=</span>axes[<span class="dv">1</span>, <span class="dv">1</span>].transAxes, ha<span class="op">=</span><span class="st">'center'</span>, va<span class="op">=</span><span class="st">'center'</span>)</span>
<span id="cb11-122"><a href="#cb11-122" aria-hidden="true" tabindex="-1"></a>            axes[<span class="dv">1</span>, <span class="dv">1</span>].set_title(<span class="st">'Elbow Detection (Insufficient Data)'</span>)</span>
<span id="cb11-123"><a href="#cb11-123" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-124"><a href="#cb11-124" aria-hidden="true" tabindex="-1"></a>        plt.tight_layout()</span>
<span id="cb11-125"><a href="#cb11-125" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-126"><a href="#cb11-126" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Save rank analysis plot</span></span>
<span id="cb11-127"><a href="#cb11-127" aria-hidden="true" tabindex="-1"></a>        rank_analysis_plot <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_analysis.png"</span>)</span>
<span id="cb11-128"><a href="#cb11-128" aria-hidden="true" tabindex="-1"></a>        plt.savefig(rank_analysis_plot, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb11-129"><a href="#cb11-129" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Rank analysis plot saved to: </span><span class="sc">{</span>rank_analysis_plot<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb11-130"><a href="#cb11-130" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-131"><a href="#cb11-131" aria-hidden="true" tabindex="-1"></a>        plt.show()</span>
<span id="cb11-132"><a href="#cb11-132" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-133"><a href="#cb11-133" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Print detailed analysis</span></span>
<span id="cb11-134"><a href="#cb11-134" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">DETAILED RANK ANALYSIS:"</span>)</span>
<span id="cb11-135"><a href="#cb11-135" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span><span class="st">'Rank'</span><span class="sc">:&lt;6}</span><span class="ss"> </span><span class="sc">{</span><span class="st">'Error'</span><span class="sc">:&lt;12}</span><span class="ss"> </span><span class="sc">{</span><span class="st">'Change'</span><span class="sc">:&lt;12}</span><span class="ss"> </span><span class="sc">{</span><span class="st">'Change %'</span><span class="sc">:&lt;12}</span><span class="ss"> </span><span class="sc">{</span><span class="st">'2nd Deriv'</span><span class="sc">:&lt;12}</span><span class="ss">"</span>)</span>
<span id="cb11-136"><a href="#cb11-136" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span><span class="st">'-'</span><span class="op">*</span><span class="dv">60</span><span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb11-137"><a href="#cb11-137" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-138"><a href="#cb11-138" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> _, row <span class="kw">in</span> df_results.iterrows():</span>
<span id="cb11-139"><a href="#cb11-139" aria-hidden="true" tabindex="-1"></a>            rank <span class="op">=</span> <span class="bu">int</span>(row[<span class="st">'rank'</span>])</span>
<span id="cb11-140"><a href="#cb11-140" aria-hidden="true" tabindex="-1"></a>            error <span class="op">=</span> row[<span class="st">'reconstruction_error'</span>]</span>
<span id="cb11-141"><a href="#cb11-141" aria-hidden="true" tabindex="-1"></a>            change <span class="op">=</span> row[<span class="st">'error_change'</span>] <span class="cf">if</span> <span class="kw">not</span> pd.isna(row[<span class="st">'error_change'</span>]) <span class="cf">else</span> <span class="dv">0</span></span>
<span id="cb11-142"><a href="#cb11-142" aria-hidden="true" tabindex="-1"></a>            change_pct <span class="op">=</span> row[<span class="st">'error_change_pct'</span>] <span class="cf">if</span> <span class="kw">not</span> pd.isna(row[<span class="st">'error_change_pct'</span>]) <span class="cf">else</span> <span class="dv">0</span></span>
<span id="cb11-143"><a href="#cb11-143" aria-hidden="true" tabindex="-1"></a>            second_deriv <span class="op">=</span> row[<span class="st">'second_derivative'</span>] <span class="cf">if</span> <span class="kw">not</span> pd.isna(row[<span class="st">'second_derivative'</span>]) <span class="cf">else</span> <span class="dv">0</span></span>
<span id="cb11-144"><a href="#cb11-144" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb11-145"><a href="#cb11-145" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span>rank<span class="sc">:&lt;6}</span><span class="ss"> </span><span class="sc">{</span>error<span class="sc">:&lt;12.6f}</span><span class="ss"> </span><span class="sc">{</span>change<span class="sc">:&lt;12.6f}</span><span class="ss"> </span><span class="sc">{</span>change_pct<span class="sc">:&lt;12.2f}</span><span class="ss"> </span><span class="sc">{</span>second_deriv<span class="sc">:&lt;12.6f}</span><span class="ss">"</span>)</span>
<span id="cb11-146"><a href="#cb11-146" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-147"><a href="#cb11-147" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Recommendations</span></span>
<span id="cb11-148"><a href="#cb11-148" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">RECOMMENDATIONS:"</span>)</span>
<span id="cb11-149"><a href="#cb11-149" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-150"><a href="#cb11-150" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Best by reconstruction error</span></span>
<span id="cb11-151"><a href="#cb11-151" aria-hidden="true" tabindex="-1"></a>        best_error_rank <span class="op">=</span> df_results.loc[df_results[<span class="st">'reconstruction_error'</span>].idxmin(), <span class="st">'rank'</span>]</span>
<span id="cb11-152"><a href="#cb11-152" aria-hidden="true" tabindex="-1"></a>        best_error <span class="op">=</span> df_results[<span class="st">'reconstruction_error'</span>].<span class="bu">min</span>()</span>
<span id="cb11-153"><a href="#cb11-153" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"  Best reconstruction error: Rank </span><span class="sc">{</span><span class="bu">int</span>(best_error_rank)<span class="sc">}</span><span class="ss"> (error: </span><span class="sc">{</span>best_error<span class="sc">:.6f}</span><span class="ss">)"</span>)</span>
<span id="cb11-154"><a href="#cb11-154" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-155"><a href="#cb11-155" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Elbow point recommendation</span></span>
<span id="cb11-156"><a href="#cb11-156" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> elbow_rank:</span>
<span id="cb11-157"><a href="#cb11-157" aria-hidden="true" tabindex="-1"></a>            elbow_error <span class="op">=</span> df_results[df_results[<span class="st">'rank'</span>] <span class="op">==</span> elbow_rank][<span class="st">'reconstruction_error'</span>].iloc[<span class="dv">0</span>]</span>
<span id="cb11-158"><a href="#cb11-158" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  Elbow point (best trade-off): Rank </span><span class="sc">{</span><span class="bu">int</span>(elbow_rank)<span class="sc">}</span><span class="ss"> (error: </span><span class="sc">{</span>elbow_error<span class="sc">:.6f}</span><span class="ss">)"</span>)</span>
<span id="cb11-159"><a href="#cb11-159" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-160"><a href="#cb11-160" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Conservative recommendation (where improvement becomes marginal)</span></span>
<span id="cb11-161"><a href="#cb11-161" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find where error reduction becomes &lt; 5% of previous rank</span></span>
<span id="cb11-162"><a href="#cb11-162" aria-hidden="true" tabindex="-1"></a>        marginal_ranks <span class="op">=</span> df_results[df_results[<span class="st">'error_change_pct'</span>] <span class="op">&gt;</span> <span class="op">-</span><span class="dv">5</span>][<span class="st">'rank'</span>]</span>
<span id="cb11-163"><a href="#cb11-163" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="bu">len</span>(marginal_ranks) <span class="op">&gt;</span> <span class="dv">0</span>:</span>
<span id="cb11-164"><a href="#cb11-164" aria-hidden="true" tabindex="-1"></a>            conservative_rank <span class="op">=</span> marginal_ranks.iloc[<span class="dv">0</span>]</span>
<span id="cb11-165"><a href="#cb11-165" aria-hidden="true" tabindex="-1"></a>            conservative_error <span class="op">=</span> df_results[df_results[<span class="st">'rank'</span>] <span class="op">==</span> conservative_rank][<span class="st">'reconstruction_error'</span>].iloc[<span class="dv">0</span>]</span>
<span id="cb11-166"><a href="#cb11-166" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  Conservative choice (&lt; 5% improvement): Rank </span><span class="sc">{</span><span class="bu">int</span>(conservative_rank)<span class="sc">}</span><span class="ss"> (error: </span><span class="sc">{</span>conservative_error<span class="sc">:.6f}</span><span class="ss">)"</span>)</span>
<span id="cb11-167"><a href="#cb11-167" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-168"><a href="#cb11-168" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">  SUGGESTED RANK: </span><span class="sc">{</span><span class="bu">int</span>(elbow_rank) <span class="cf">if</span> elbow_rank <span class="cf">else</span> <span class="bu">int</span>(best_error_rank)<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb11-169"><a href="#cb11-169" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"  RATIONALE: </span><span class="sc">{</span><span class="st">'Elbow point provides best complexity/accuracy trade-off'</span> <span class="cf">if</span> elbow_rank <span class="cf">else</span> <span class="st">'Best reconstruction error'</span><span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb11-170"><a href="#cb11-170" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-171"><a href="#cb11-171" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Save summary</span></span>
<span id="cb11-172"><a href="#cb11-172" aria-hidden="true" tabindex="-1"></a>        summary_data <span class="op">=</span> {</span>
<span id="cb11-173"><a href="#cb11-173" aria-hidden="true" tabindex="-1"></a>            <span class="st">'best_rank_by_error'</span>: <span class="bu">int</span>(best_error_rank),</span>
<span id="cb11-174"><a href="#cb11-174" aria-hidden="true" tabindex="-1"></a>            <span class="st">'best_error'</span>: <span class="bu">float</span>(best_error),</span>
<span id="cb11-175"><a href="#cb11-175" aria-hidden="true" tabindex="-1"></a>            <span class="st">'elbow_rank'</span>: <span class="bu">int</span>(elbow_rank) <span class="cf">if</span> elbow_rank <span class="cf">else</span> <span class="va">None</span>,</span>
<span id="cb11-176"><a href="#cb11-176" aria-hidden="true" tabindex="-1"></a>            <span class="st">'suggested_rank'</span>: <span class="bu">int</span>(elbow_rank) <span class="cf">if</span> elbow_rank <span class="cf">else</span> <span class="bu">int</span>(best_error_rank),</span>
<span id="cb11-177"><a href="#cb11-177" aria-hidden="true" tabindex="-1"></a>            <span class="st">'total_ranks_tested'</span>: <span class="bu">len</span>(df_results),</span>
<span id="cb11-178"><a href="#cb11-178" aria-hidden="true" tabindex="-1"></a>            <span class="st">'analysis_date'</span>: datetime.now().isoformat()</span>
<span id="cb11-179"><a href="#cb11-179" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb11-180"><a href="#cb11-180" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-181"><a href="#cb11-181" aria-hidden="true" tabindex="-1"></a>        summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb11-182"><a href="#cb11-182" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'w'</span>) <span class="im">as</span> f:</span>
<span id="cb11-183"><a href="#cb11-183" aria-hidden="true" tabindex="-1"></a>            json.dump(summary_data, f, indent<span class="op">=</span><span class="dv">2</span>)</span>
<span id="cb11-184"><a href="#cb11-184" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-185"><a href="#cb11-185" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Rank selection summary saved to: </span><span class="sc">{</span>summary_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb11-186"><a href="#cb11-186" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-187"><a href="#cb11-187" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Store main results directory for subsequent analyses</span></span>
<span id="cb11-188"><a href="#cb11-188" aria-hidden="true" tabindex="-1"></a>        <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>] <span class="op">=</span> main_results_dir</span>
<span id="cb11-189"><a href="#cb11-189" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-190"><a href="#cb11-190" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb11-191"><a href="#cb11-191" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No valid results found in rank comparison!"</span>)</span>
<span id="cb11-192"><a href="#cb11-192" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb11-193"><a href="#cb11-193" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb11-194"><a href="#cb11-194" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Rank comparison results file not found: </span><span class="sc">{</span>comparison_results_file<span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/analyze-rank-comparison-1.png" width="1440"></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="analyze-optimal-decomposition-results" class="level2">
<h2 class="anchored" data-anchor-id="analyze-optimal-decomposition-results">3.4 Analyze optimal decomposition results</h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Find the main results directory</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Fallback to find the most recent main results directory</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> output_dir</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Load rank selection summary to get optimal rank</span></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>    <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>        rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Analyzing optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> results from rank comparison data"</span>)</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Find the optimal rank data in rank_comparison directory</span></span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Analyzing optimal results from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Load summary</span></span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>            <span class="cf">with</span> <span class="bu">open</span>(os.path.join(latest_dir, <span class="st">"summary.json"</span>)) <span class="im">as</span> f:</span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>                summary <span class="op">=</span> json.load(f)</span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">"</span> <span class="op">+</span> <span class="st">"="</span><span class="op">*</span><span class="dv">60</span>)</span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="st">"OPTIMAL DECOMPOSITION QUALITY ASSESSMENT"</span>)</span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="st">"="</span><span class="op">*</span><span class="dv">60</span>)</span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Print key metrics</span></span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Method: </span><span class="sc">{</span>summary[<span class="st">'method'</span>]<span class="sc">.</span>upper()<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Rank: </span><span class="sc">{</span>summary[<span class="st">'rank'</span>]<span class="sc">}</span><span class="ss"> (data-driven selection)"</span>)</span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Reconstruction Error: </span><span class="sc">{</span>summary[<span class="st">'reconstruction_error'</span>]<span class="sc">:.6f}</span><span class="ss">"</span>)</span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Quality interpretation</span></span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a>            error <span class="op">=</span> summary[<span class="st">'reconstruction_error'</span>]</span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> error <span class="op">&lt;</span> <span class="fl">0.1</span>:</span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a>                quality <span class="op">=</span> <span class="st">"Excellent (may be overfitting)"</span></span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a>            <span class="cf">elif</span> error <span class="op">&lt;</span> <span class="fl">0.3</span>:</span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a>                quality <span class="op">=</span> <span class="st">"Good"</span></span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a>            <span class="cf">elif</span> error <span class="op">&lt;</span> <span class="fl">0.5</span>:</span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a>                quality <span class="op">=</span> <span class="st">"Acceptable"</span></span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a>            <span class="cf">else</span>:</span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a>                quality <span class="op">=</span> <span class="st">"Poor - consider increasing rank"</span></span>
<span id="cb13-65"><a href="#cb13-65" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-66"><a href="#cb13-66" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Quality Assessment: </span><span class="sc">{</span>quality<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb13-67"><a href="#cb13-67" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-68"><a href="#cb13-68" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Load factor matrices</span></span>
<span id="cb13-69"><a href="#cb13-69" aria-hidden="true" tabindex="-1"></a>            gene_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb13-70"><a href="#cb13-70" aria-hidden="true" tabindex="-1"></a>            species_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"species_factors.npy"</span>))</span>
<span id="cb13-71"><a href="#cb13-71" aria-hidden="true" tabindex="-1"></a>            time_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"time_factors.npy"</span>))</span>
<span id="cb13-72"><a href="#cb13-72" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-73"><a href="#cb13-73" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Factor Matrix Shapes:"</span>)</span>
<span id="cb13-74"><a href="#cb13-74" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  Gene factors: </span><span class="sc">{</span>gene_factors<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb13-75"><a href="#cb13-75" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  Species factors: </span><span class="sc">{</span>species_factors<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb13-76"><a href="#cb13-76" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  Time factors: </span><span class="sc">{</span>time_factors<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb13-77"><a href="#cb13-77" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-78"><a href="#cb13-78" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Convergence Check:"</span>)</span>
<span id="cb13-79"><a href="#cb13-79" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  All factors finite: </span><span class="sc">{</span>np<span class="sc">.</span><span class="bu">all</span>(np.isfinite(gene_factors)) <span class="kw">and</span> np<span class="sc">.</span><span class="bu">all</span>(np.isfinite(species_factors)) <span class="kw">and</span> np<span class="sc">.</span><span class="bu">all</span>(np.isfinite(time_factors))<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb13-80"><a href="#cb13-80" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  No NaN values: </span><span class="sc">{</span><span class="kw">not</span> (np.<span class="bu">any</span>(np.isnan(gene_factors)) <span class="kw">or</span> np.<span class="bu">any</span>(np.isnan(species_factors)) <span class="kw">or</span> np.<span class="bu">any</span>(np.isnan(time_factors)))<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb13-81"><a href="#cb13-81" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-82"><a href="#cb13-82" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Show component strengths</span></span>
<span id="cb13-83"><a href="#cb13-83" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Component Analysis:"</span>)</span>
<span id="cb13-84"><a href="#cb13-84" aria-hidden="true" tabindex="-1"></a>            component_strengths <span class="op">=</span> []</span>
<span id="cb13-85"><a href="#cb13-85" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(summary[<span class="st">'rank'</span>]):</span>
<span id="cb13-86"><a href="#cb13-86" aria-hidden="true" tabindex="-1"></a>                gene_norm <span class="op">=</span> np.linalg.norm(gene_factors[:, i])</span>
<span id="cb13-87"><a href="#cb13-87" aria-hidden="true" tabindex="-1"></a>                species_norm <span class="op">=</span> np.linalg.norm(species_factors[:, i])</span>
<span id="cb13-88"><a href="#cb13-88" aria-hidden="true" tabindex="-1"></a>                time_norm <span class="op">=</span> np.linalg.norm(time_factors[:, i])</span>
<span id="cb13-89"><a href="#cb13-89" aria-hidden="true" tabindex="-1"></a>                total_strength <span class="op">=</span> gene_norm <span class="op">*</span> species_norm <span class="op">*</span> time_norm</span>
<span id="cb13-90"><a href="#cb13-90" aria-hidden="true" tabindex="-1"></a>                component_strengths.append(total_strength)</span>
<span id="cb13-91"><a href="#cb13-91" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-92"><a href="#cb13-92" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Sort components by strength</span></span>
<span id="cb13-93"><a href="#cb13-93" aria-hidden="true" tabindex="-1"></a>            sorted_components <span class="op">=</span> np.argsort(component_strengths)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb13-94"><a href="#cb13-94" aria-hidden="true" tabindex="-1"></a>            n_top <span class="op">=</span> <span class="bu">min</span>(<span class="dv">5</span>, <span class="bu">len</span>(sorted_components))</span>
<span id="cb13-95"><a href="#cb13-95" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  Top </span><span class="sc">{</span>n_top<span class="sc">}</span><span class="ss"> strongest components: </span><span class="sc">{</span>sorted_components[:n_top] <span class="op">+</span> <span class="dv">1</span><span class="sc">}</span><span class="ss">"</span>)  <span class="co"># +1 for 1-based indexing</span></span>
<span id="cb13-96"><a href="#cb13-96" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-97"><a href="#cb13-97" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Find most variable timepoints</span></span>
<span id="cb13-98"><a href="#cb13-98" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Temporal Patterns:"</span>)</span>
<span id="cb13-99"><a href="#cb13-99" aria-hidden="true" tabindex="-1"></a>            time_variance <span class="op">=</span> np.var(time_factors, axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb13-100"><a href="#cb13-100" aria-hidden="true" tabindex="-1"></a>            most_temporal <span class="op">=</span> np.argsort(time_variance)[::<span class="op">-</span><span class="dv">1</span>][:n_top]</span>
<span id="cb13-101"><a href="#cb13-101" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  Components with strongest temporal patterns: </span><span class="sc">{</span>most_temporal <span class="op">+</span> <span class="dv">1</span><span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb13-102"><a href="#cb13-102" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-103"><a href="#cb13-103" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Species diversity</span></span>
<span id="cb13-104"><a href="#cb13-104" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Species Patterns:"</span>)</span>
<span id="cb13-105"><a href="#cb13-105" aria-hidden="true" tabindex="-1"></a>            species_variance <span class="op">=</span> np.var(species_factors, axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb13-106"><a href="#cb13-106" aria-hidden="true" tabindex="-1"></a>            most_species_specific <span class="op">=</span> np.argsort(species_variance)[::<span class="op">-</span><span class="dv">1</span>][:n_top]</span>
<span id="cb13-107"><a href="#cb13-107" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  Components with strongest species differences: </span><span class="sc">{</span>most_species_specific <span class="op">+</span> <span class="dv">1</span><span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb13-108"><a href="#cb13-108" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-109"><a href="#cb13-109" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">"</span> <span class="op">+</span> <span class="st">"="</span><span class="op">*</span><span class="dv">60</span>)</span>
<span id="cb13-110"><a href="#cb13-110" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="st">"CONCLUSION: The optimal decomposition successfully converged!"</span>)</span>
<span id="cb13-111"><a href="#cb13-111" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Results saved in: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb13-112"><a href="#cb13-112" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="st">"="</span><span class="op">*</span><span class="dv">60</span>)</span>
<span id="cb13-113"><a href="#cb13-113" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-114"><a href="#cb13-114" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Store latest_dir for subsequent analyses (this is now the rank_comparison data)</span></span>
<span id="cb13-115"><a href="#cb13-115" aria-hidden="true" tabindex="-1"></a>            <span class="bu">globals</span>()[<span class="st">'optimal_results_dir'</span>] <span class="op">=</span> latest_dir</span>
<span id="cb13-116"><a href="#cb13-116" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb13-117"><a href="#cb13-117" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb13-118"><a href="#cb13-118" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb13-119"><a href="#cb13-119" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb13-120"><a href="#cb13-120" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found in rank comparison!"</span>)</span>
<span id="cb13-121"><a href="#cb13-121" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb13-122"><a href="#cb13-122" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No rank selection summary found!"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Analyzing optimal rank 5 results from rank comparison data
## Analyzing optimal results from: ../output/13.00-multiomics-stdm/20251015_190157/rank_comparison/rank_05/20251015_190351
## 
## ============================================================
## OPTIMAL DECOMPOSITION QUALITY ASSESSMENT
## ============================================================
## Method: PARAFAC
## Rank: 5 (data-driven selection)
## Reconstruction Error: 0.451016
## Quality Assessment: Acceptable
## 
## Factor Matrix Shapes:
##   Gene factors: (9800, 5)
##   Species factors: (30, 5)
##   Time factors: (4, 5)
## 
## Convergence Check:
##   All factors finite: True
##   No NaN values: True
## 
## Component Analysis:
##   Top 5 strongest components: [3 5 2 1 4]
## 
## Temporal Patterns:
##   Components with strongest temporal patterns: [5 4 3 2 1]
## 
## Species Patterns:
##   Components with strongest species differences: [1 2 3 4 5]
## 
## ============================================================
## CONCLUSION: The optimal decomposition successfully converged!
## Results saved in: ../output/13.00-multiomics-stdm/20251015_190157/rank_comparison/rank_05/20251015_190351
## ============================================================</code></pre>
</section>
</section>
<section id="visualizations" class="level1">
<h1>4 VISUALIZATIONS</h1>
<section id="generate-visualizations-for-all-tested-ranks" class="level2">
<h2 class="anchored" data-anchor-id="generate-visualizations-for-all-tested-ranks">4.1 Generate visualizations for all tested ranks</h2>
<p>This section creates comprehensive visualizations for each rank tested, allowing comparison across different numbers of components.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Must be before importing pyplot</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> sklearn.decomposition <span class="im">import</span> PCA</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting style  </span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>plt.style.use(<span class="st">'default'</span>)</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>sns.set_palette(<span class="st">"husl"</span>)</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory and find the main results directory</span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Find the most recent main results directory (timestamped)</span></span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> output_dir</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Load rank comparison results</span></span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>comparison_results_file <span class="op">=</span> os.path.join(comparison_dir, <span class="st">"rank_comparison_results.json"</span>)</span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> os.path.exists(comparison_results_file):</span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>    <span class="cf">with</span> <span class="bu">open</span>(comparison_results_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>        rank_results <span class="op">=</span> json.load(f)</span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load the transformed data</span></span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>    data_file <span class="op">=</span> os.path.join(output_dir, <span class="st">"vst_counts_matrix_long_format.csv"</span>)</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>    df_long <span class="op">=</span> pd.read_csv(data_file)</span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>    species_names <span class="op">=</span> <span class="bu">sorted</span>(df_long[<span class="st">'species'</span>].unique())</span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>    timepoint_names <span class="op">=</span> <span class="bu">sorted</span>(df_long[<span class="st">'timepoint'</span>].unique())</span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>    species_types <span class="op">=</span> [name.split(<span class="st">'-'</span>)[<span class="dv">0</span>] <span class="cf">for</span> name <span class="kw">in</span> species_names]</span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a>    unique_types <span class="op">=</span> <span class="bu">list</span>(<span class="bu">set</span>(species_types))</span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"="</span><span class="op">*</span><span class="dv">60</span>)</span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"GENERATING VISUALIZATIONS FOR ALL TESTED RANKS"</span>)</span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"="</span><span class="op">*</span><span class="dv">60</span>)</span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a>    successful_ranks <span class="op">=</span> []</span>
<span id="cb15-52"><a href="#cb15-52" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb15-53"><a href="#cb15-53" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Process each rank that completed successfully</span></span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> rank_str, data <span class="kw">in</span> rank_results.items():</span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a>        rank <span class="op">=</span> <span class="bu">int</span>(rank_str)</span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a>        result_dir <span class="op">=</span> data.get(<span class="st">'result_dir'</span>)</span>
<span id="cb15-57"><a href="#cb15-57" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb15-58"><a href="#cb15-58" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dir <span class="kw">and</span> <span class="kw">not</span> np.isnan(data[<span class="st">'reconstruction_error'</span>]):</span>
<span id="cb15-59"><a href="#cb15-59" aria-hidden="true" tabindex="-1"></a>            successful_ranks.append(rank)</span>
<span id="cb15-60"><a href="#cb15-60" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb15-61"><a href="#cb15-61" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Processing visualizations for rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">..."</span>)</span>
<span id="cb15-62"><a href="#cb15-62" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb15-63"><a href="#cb15-63" aria-hidden="true" tabindex="-1"></a>            <span class="cf">try</span>:</span>
<span id="cb15-64"><a href="#cb15-64" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Load factor matrices</span></span>
<span id="cb15-65"><a href="#cb15-65" aria-hidden="true" tabindex="-1"></a>                gene_factors <span class="op">=</span> np.load(os.path.join(result_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb15-66"><a href="#cb15-66" aria-hidden="true" tabindex="-1"></a>                species_factors <span class="op">=</span> np.load(os.path.join(result_dir, <span class="st">"species_factors.npy"</span>))</span>
<span id="cb15-67"><a href="#cb15-67" aria-hidden="true" tabindex="-1"></a>                time_factors <span class="op">=</span> np.load(os.path.join(result_dir, <span class="st">"time_factors.npy"</span>))</span>
<span id="cb15-68"><a href="#cb15-68" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-69"><a href="#cb15-69" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Calculate component strengths</span></span>
<span id="cb15-70"><a href="#cb15-70" aria-hidden="true" tabindex="-1"></a>                component_strengths <span class="op">=</span> []</span>
<span id="cb15-71"><a href="#cb15-71" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(gene_factors.shape[<span class="dv">1</span>]):</span>
<span id="cb15-72"><a href="#cb15-72" aria-hidden="true" tabindex="-1"></a>                    gene_norm <span class="op">=</span> np.linalg.norm(gene_factors[:, i])</span>
<span id="cb15-73"><a href="#cb15-73" aria-hidden="true" tabindex="-1"></a>                    species_norm <span class="op">=</span> np.linalg.norm(species_factors[:, i])</span>
<span id="cb15-74"><a href="#cb15-74" aria-hidden="true" tabindex="-1"></a>                    time_norm <span class="op">=</span> np.linalg.norm(time_factors[:, i])</span>
<span id="cb15-75"><a href="#cb15-75" aria-hidden="true" tabindex="-1"></a>                    total_strength <span class="op">=</span> gene_norm <span class="op">*</span> species_norm <span class="op">*</span> time_norm</span>
<span id="cb15-76"><a href="#cb15-76" aria-hidden="true" tabindex="-1"></a>                    component_strengths.append(total_strength)</span>
<span id="cb15-77"><a href="#cb15-77" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-78"><a href="#cb15-78" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Sort components by strength</span></span>
<span id="cb15-79"><a href="#cb15-79" aria-hidden="true" tabindex="-1"></a>                sorted_components <span class="op">=</span> np.argsort(component_strengths)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb15-80"><a href="#cb15-80" aria-hidden="true" tabindex="-1"></a>                n_top_display <span class="op">=</span> <span class="bu">min</span>(<span class="dv">5</span>, rank)</span>
<span id="cb15-81"><a href="#cb15-81" aria-hidden="true" tabindex="-1"></a>                top_components <span class="op">=</span> sorted_components[:n_top_display]</span>
<span id="cb15-82"><a href="#cb15-82" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-83"><a href="#cb15-83" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Create comprehensive visualization for this rank</span></span>
<span id="cb15-84"><a href="#cb15-84" aria-hidden="true" tabindex="-1"></a>                fig <span class="op">=</span> plt.figure(figsize<span class="op">=</span>(<span class="dv">16</span>, <span class="dv">12</span>))</span>
<span id="cb15-85"><a href="#cb15-85" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-86"><a href="#cb15-86" aria-hidden="true" tabindex="-1"></a>                <span class="co"># 1. Component strengths</span></span>
<span id="cb15-87"><a href="#cb15-87" aria-hidden="true" tabindex="-1"></a>                plt.subplot(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">1</span>)</span>
<span id="cb15-88"><a href="#cb15-88" aria-hidden="true" tabindex="-1"></a>                plt.bar(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(component_strengths) <span class="op">+</span> <span class="dv">1</span>), </span>
<span id="cb15-89"><a href="#cb15-89" aria-hidden="true" tabindex="-1"></a>                        [component_strengths[i] <span class="cf">for</span> i <span class="kw">in</span> sorted_components])</span>
<span id="cb15-90"><a href="#cb15-90" aria-hidden="true" tabindex="-1"></a>                plt.xlabel(<span class="st">'Component (ranked by strength)'</span>)</span>
<span id="cb15-91"><a href="#cb15-91" aria-hidden="true" tabindex="-1"></a>                plt.ylabel(<span class="st">'Component Strength'</span>)</span>
<span id="cb15-92"><a href="#cb15-92" aria-hidden="true" tabindex="-1"></a>                plt.title(<span class="ss">f'Component Strengths (Rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb15-93"><a href="#cb15-93" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> rank <span class="op">&lt;=</span> <span class="dv">10</span>:</span>
<span id="cb15-94"><a href="#cb15-94" aria-hidden="true" tabindex="-1"></a>                    plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, rank <span class="op">+</span> <span class="dv">1</span>))</span>
<span id="cb15-95"><a href="#cb15-95" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-96"><a href="#cb15-96" aria-hidden="true" tabindex="-1"></a>                <span class="co"># 2. Temporal patterns heatmap</span></span>
<span id="cb15-97"><a href="#cb15-97" aria-hidden="true" tabindex="-1"></a>                plt.subplot(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">2</span>)</span>
<span id="cb15-98"><a href="#cb15-98" aria-hidden="true" tabindex="-1"></a>                time_data <span class="op">=</span> time_factors[:, top_components]</span>
<span id="cb15-99"><a href="#cb15-99" aria-hidden="true" tabindex="-1"></a>                sns.heatmap(time_data.T, </span>
<span id="cb15-100"><a href="#cb15-100" aria-hidden="true" tabindex="-1"></a>                            xticklabels<span class="op">=</span>timepoint_names,</span>
<span id="cb15-101"><a href="#cb15-101" aria-hidden="true" tabindex="-1"></a>                            yticklabels<span class="op">=</span>[<span class="ss">f'C</span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> top_components],</span>
<span id="cb15-102"><a href="#cb15-102" aria-hidden="true" tabindex="-1"></a>                            cmap<span class="op">=</span><span class="st">'RdBu_r'</span>, center<span class="op">=</span><span class="dv">0</span>, annot<span class="op">=</span><span class="va">True</span>, fmt<span class="op">=</span><span class="st">'.2f'</span>)</span>
<span id="cb15-103"><a href="#cb15-103" aria-hidden="true" tabindex="-1"></a>                plt.title(<span class="ss">f'Top </span><span class="sc">{</span>n_top_display<span class="sc">}</span><span class="ss"> Components: Temporal'</span>)</span>
<span id="cb15-104"><a href="#cb15-104" aria-hidden="true" tabindex="-1"></a>                plt.xlabel(<span class="st">'Timepoint'</span>)</span>
<span id="cb15-105"><a href="#cb15-105" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-106"><a href="#cb15-106" aria-hidden="true" tabindex="-1"></a>                <span class="co"># 3. Species clustering PCA</span></span>
<span id="cb15-107"><a href="#cb15-107" aria-hidden="true" tabindex="-1"></a>                plt.subplot(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">3</span>)</span>
<span id="cb15-108"><a href="#cb15-108" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> rank <span class="op">&gt;=</span> <span class="dv">2</span>:</span>
<span id="cb15-109"><a href="#cb15-109" aria-hidden="true" tabindex="-1"></a>                    n_pca_components <span class="op">=</span> <span class="bu">min</span>(rank, <span class="dv">10</span>)</span>
<span id="cb15-110"><a href="#cb15-110" aria-hidden="true" tabindex="-1"></a>                    species_subset <span class="op">=</span> species_factors[:, sorted_components[:n_pca_components]]</span>
<span id="cb15-111"><a href="#cb15-111" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb15-112"><a href="#cb15-112" aria-hidden="true" tabindex="-1"></a>                    pca <span class="op">=</span> PCA(n_components<span class="op">=</span><span class="dv">2</span>)</span>
<span id="cb15-113"><a href="#cb15-113" aria-hidden="true" tabindex="-1"></a>                    species_pca <span class="op">=</span> pca.fit_transform(species_subset)</span>
<span id="cb15-114"><a href="#cb15-114" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb15-115"><a href="#cb15-115" aria-hidden="true" tabindex="-1"></a>                    colors <span class="op">=</span> [<span class="st">'red'</span>, <span class="st">'blue'</span>, <span class="st">'green'</span>, <span class="st">'orange'</span>, <span class="st">'purple'</span>][:<span class="bu">len</span>(unique_types)]</span>
<span id="cb15-116"><a href="#cb15-116" aria-hidden="true" tabindex="-1"></a>                    species_type_map <span class="op">=</span> {stype: i <span class="cf">for</span> i, stype <span class="kw">in</span> <span class="bu">enumerate</span>(unique_types)}</span>
<span id="cb15-117"><a href="#cb15-117" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb15-118"><a href="#cb15-118" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span> i, (x, y) <span class="kw">in</span> <span class="bu">enumerate</span>(species_pca):</span>
<span id="cb15-119"><a href="#cb15-119" aria-hidden="true" tabindex="-1"></a>                        stype <span class="op">=</span> species_types[i]</span>
<span id="cb15-120"><a href="#cb15-120" aria-hidden="true" tabindex="-1"></a>                        plt.scatter(x, y, c<span class="op">=</span>colors[species_type_map[stype]], </span>
<span id="cb15-121"><a href="#cb15-121" aria-hidden="true" tabindex="-1"></a>                                   alpha<span class="op">=</span><span class="fl">0.7</span>, s<span class="op">=</span><span class="dv">40</span>, </span>
<span id="cb15-122"><a href="#cb15-122" aria-hidden="true" tabindex="-1"></a>                                   label<span class="op">=</span>stype <span class="cf">if</span> i <span class="op">==</span> species_types.index(stype) <span class="cf">else</span> <span class="st">""</span>)</span>
<span id="cb15-123"><a href="#cb15-123" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb15-124"><a href="#cb15-124" aria-hidden="true" tabindex="-1"></a>                    plt.xlabel(<span class="ss">f'PC1 (</span><span class="sc">{</span>pca<span class="sc">.</span>explained_variance_ratio_[<span class="dv">0</span>]<span class="sc">:.1%}</span><span class="ss">)'</span>)</span>
<span id="cb15-125"><a href="#cb15-125" aria-hidden="true" tabindex="-1"></a>                    plt.ylabel(<span class="ss">f'PC2 (</span><span class="sc">{</span>pca<span class="sc">.</span>explained_variance_ratio_[<span class="dv">1</span>]<span class="sc">:.1%}</span><span class="ss">)'</span>)</span>
<span id="cb15-126"><a href="#cb15-126" aria-hidden="true" tabindex="-1"></a>                    plt.title(<span class="ss">f'Species PCA (Rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb15-127"><a href="#cb15-127" aria-hidden="true" tabindex="-1"></a>                    plt.legend()</span>
<span id="cb15-128"><a href="#cb15-128" aria-hidden="true" tabindex="-1"></a>                    plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb15-129"><a href="#cb15-129" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-130"><a href="#cb15-130" aria-hidden="true" tabindex="-1"></a>                <span class="co"># 4. Temporal trajectories</span></span>
<span id="cb15-131"><a href="#cb15-131" aria-hidden="true" tabindex="-1"></a>                plt.subplot(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">4</span>)</span>
<span id="cb15-132"><a href="#cb15-132" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span> i, comp_idx <span class="kw">in</span> <span class="bu">enumerate</span>(top_components):</span>
<span id="cb15-133"><a href="#cb15-133" aria-hidden="true" tabindex="-1"></a>                    plt.plot(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(timepoint_names) <span class="op">+</span> <span class="dv">1</span>), </span>
<span id="cb15-134"><a href="#cb15-134" aria-hidden="true" tabindex="-1"></a>                            time_factors[:, comp_idx], </span>
<span id="cb15-135"><a href="#cb15-135" aria-hidden="true" tabindex="-1"></a>                            marker<span class="op">=</span><span class="st">'o'</span>, linewidth<span class="op">=</span><span class="dv">2</span>, label<span class="op">=</span><span class="ss">f'C</span><span class="sc">{</span>comp_idx <span class="op">+</span> <span class="dv">1</span><span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb15-136"><a href="#cb15-136" aria-hidden="true" tabindex="-1"></a>                plt.xlabel(<span class="st">'Timepoint'</span>)</span>
<span id="cb15-137"><a href="#cb15-137" aria-hidden="true" tabindex="-1"></a>                plt.ylabel(<span class="st">'Factor Value'</span>)</span>
<span id="cb15-138"><a href="#cb15-138" aria-hidden="true" tabindex="-1"></a>                plt.title(<span class="ss">f'Temporal Trajectories (Rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb15-139"><a href="#cb15-139" aria-hidden="true" tabindex="-1"></a>                plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(timepoint_names) <span class="op">+</span> <span class="dv">1</span>), timepoint_names)</span>
<span id="cb15-140"><a href="#cb15-140" aria-hidden="true" tabindex="-1"></a>                plt.legend()</span>
<span id="cb15-141"><a href="#cb15-141" aria-hidden="true" tabindex="-1"></a>                plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb15-142"><a href="#cb15-142" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-143"><a href="#cb15-143" aria-hidden="true" tabindex="-1"></a>                <span class="co"># 5. Component correlations</span></span>
<span id="cb15-144"><a href="#cb15-144" aria-hidden="true" tabindex="-1"></a>                plt.subplot(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">5</span>)</span>
<span id="cb15-145"><a href="#cb15-145" aria-hidden="true" tabindex="-1"></a>                gene_corr <span class="op">=</span> np.corrcoef(gene_factors.T)</span>
<span id="cb15-146"><a href="#cb15-146" aria-hidden="true" tabindex="-1"></a>                sns.heatmap(gene_corr, cmap<span class="op">=</span><span class="st">'RdBu_r'</span>, center<span class="op">=</span><span class="dv">0</span>, </span>
<span id="cb15-147"><a href="#cb15-147" aria-hidden="true" tabindex="-1"></a>                            xticklabels<span class="op">=</span><span class="bu">range</span>(<span class="dv">1</span>, rank <span class="op">+</span> <span class="dv">1</span>),</span>
<span id="cb15-148"><a href="#cb15-148" aria-hidden="true" tabindex="-1"></a>                            yticklabels<span class="op">=</span><span class="bu">range</span>(<span class="dv">1</span>, rank <span class="op">+</span> <span class="dv">1</span>),</span>
<span id="cb15-149"><a href="#cb15-149" aria-hidden="true" tabindex="-1"></a>                            square<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb15-150"><a href="#cb15-150" aria-hidden="true" tabindex="-1"></a>                plt.title(<span class="ss">f'Component Correlations (Rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb15-151"><a href="#cb15-151" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-152"><a href="#cb15-152" aria-hidden="true" tabindex="-1"></a>                <span class="co"># 6. Species type comparison</span></span>
<span id="cb15-153"><a href="#cb15-153" aria-hidden="true" tabindex="-1"></a>                plt.subplot(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">6</span>)</span>
<span id="cb15-154"><a href="#cb15-154" aria-hidden="true" tabindex="-1"></a>                species_by_type <span class="op">=</span> {}</span>
<span id="cb15-155"><a href="#cb15-155" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span> i, name <span class="kw">in</span> <span class="bu">enumerate</span>(species_names):</span>
<span id="cb15-156"><a href="#cb15-156" aria-hidden="true" tabindex="-1"></a>                    stype <span class="op">=</span> name.split(<span class="st">'-'</span>)[<span class="dv">0</span>]</span>
<span id="cb15-157"><a href="#cb15-157" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span> stype <span class="kw">not</span> <span class="kw">in</span> species_by_type:</span>
<span id="cb15-158"><a href="#cb15-158" aria-hidden="true" tabindex="-1"></a>                        species_by_type[stype] <span class="op">=</span> []</span>
<span id="cb15-159"><a href="#cb15-159" aria-hidden="true" tabindex="-1"></a>                    species_by_type[stype].append(i)</span>
<span id="cb15-160"><a href="#cb15-160" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-161"><a href="#cb15-161" aria-hidden="true" tabindex="-1"></a>                type_means <span class="op">=</span> {}</span>
<span id="cb15-162"><a href="#cb15-162" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span> stype, indices <span class="kw">in</span> species_by_type.items():</span>
<span id="cb15-163"><a href="#cb15-163" aria-hidden="true" tabindex="-1"></a>                    type_means[stype] <span class="op">=</span> np.mean(species_factors[indices, :], axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb15-164"><a href="#cb15-164" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-165"><a href="#cb15-165" aria-hidden="true" tabindex="-1"></a>                x_pos <span class="op">=</span> np.arange(<span class="bu">len</span>(top_components))</span>
<span id="cb15-166"><a href="#cb15-166" aria-hidden="true" tabindex="-1"></a>                width <span class="op">=</span> <span class="fl">0.8</span> <span class="op">/</span> <span class="bu">len</span>(type_means)</span>
<span id="cb15-167"><a href="#cb15-167" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-168"><a href="#cb15-168" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span> i, (stype, means) <span class="kw">in</span> <span class="bu">enumerate</span>(type_means.items()):</span>
<span id="cb15-169"><a href="#cb15-169" aria-hidden="true" tabindex="-1"></a>                    plt.bar(x_pos <span class="op">+</span> i <span class="op">*</span> width, means[top_components], </span>
<span id="cb15-170"><a href="#cb15-170" aria-hidden="true" tabindex="-1"></a>                           width, label<span class="op">=</span>stype, alpha<span class="op">=</span><span class="fl">0.8</span>)</span>
<span id="cb15-171"><a href="#cb15-171" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-172"><a href="#cb15-172" aria-hidden="true" tabindex="-1"></a>                plt.xlabel(<span class="st">'Component'</span>)</span>
<span id="cb15-173"><a href="#cb15-173" aria-hidden="true" tabindex="-1"></a>                plt.ylabel(<span class="st">'Mean Factor Value'</span>)</span>
<span id="cb15-174"><a href="#cb15-174" aria-hidden="true" tabindex="-1"></a>                plt.title(<span class="ss">f'Species Comparison (Rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb15-175"><a href="#cb15-175" aria-hidden="true" tabindex="-1"></a>                plt.xticks(x_pos <span class="op">+</span> width <span class="op">*</span> (<span class="bu">len</span>(type_means) <span class="op">-</span> <span class="dv">1</span>) <span class="op">/</span> <span class="dv">2</span>, </span>
<span id="cb15-176"><a href="#cb15-176" aria-hidden="true" tabindex="-1"></a>                           [<span class="ss">f'C</span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> top_components])</span>
<span id="cb15-177"><a href="#cb15-177" aria-hidden="true" tabindex="-1"></a>                plt.legend()</span>
<span id="cb15-178"><a href="#cb15-178" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-179"><a href="#cb15-179" aria-hidden="true" tabindex="-1"></a>                <span class="co"># 7. Temporal variance</span></span>
<span id="cb15-180"><a href="#cb15-180" aria-hidden="true" tabindex="-1"></a>                plt.subplot(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">7</span>)</span>
<span id="cb15-181"><a href="#cb15-181" aria-hidden="true" tabindex="-1"></a>                time_variances <span class="op">=</span> np.var(time_factors, axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb15-182"><a href="#cb15-182" aria-hidden="true" tabindex="-1"></a>                sorted_time_vars <span class="op">=</span> np.argsort(time_variances)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb15-183"><a href="#cb15-183" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-184"><a href="#cb15-184" aria-hidden="true" tabindex="-1"></a>                plt.bar(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(time_variances) <span class="op">+</span> <span class="dv">1</span>), </span>
<span id="cb15-185"><a href="#cb15-185" aria-hidden="true" tabindex="-1"></a>                       [time_variances[i] <span class="cf">for</span> i <span class="kw">in</span> sorted_time_vars])</span>
<span id="cb15-186"><a href="#cb15-186" aria-hidden="true" tabindex="-1"></a>                plt.xlabel(<span class="st">'Component (by temporal variance)'</span>)</span>
<span id="cb15-187"><a href="#cb15-187" aria-hidden="true" tabindex="-1"></a>                plt.ylabel(<span class="st">'Temporal Variance'</span>)</span>
<span id="cb15-188"><a href="#cb15-188" aria-hidden="true" tabindex="-1"></a>                plt.title(<span class="ss">f'Temporal Variability (Rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb15-189"><a href="#cb15-189" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> rank <span class="op">&lt;=</span> <span class="dv">10</span>:</span>
<span id="cb15-190"><a href="#cb15-190" aria-hidden="true" tabindex="-1"></a>                    plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, rank <span class="op">+</span> <span class="dv">1</span>))</span>
<span id="cb15-191"><a href="#cb15-191" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-192"><a href="#cb15-192" aria-hidden="true" tabindex="-1"></a>                <span class="co"># 8. Species variance</span></span>
<span id="cb15-193"><a href="#cb15-193" aria-hidden="true" tabindex="-1"></a>                plt.subplot(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">8</span>)</span>
<span id="cb15-194"><a href="#cb15-194" aria-hidden="true" tabindex="-1"></a>                species_variances <span class="op">=</span> np.var(species_factors, axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb15-195"><a href="#cb15-195" aria-hidden="true" tabindex="-1"></a>                sorted_species_vars <span class="op">=</span> np.argsort(species_variances)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb15-196"><a href="#cb15-196" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-197"><a href="#cb15-197" aria-hidden="true" tabindex="-1"></a>                plt.bar(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(species_variances) <span class="op">+</span> <span class="dv">1</span>), </span>
<span id="cb15-198"><a href="#cb15-198" aria-hidden="true" tabindex="-1"></a>                       [species_variances[i] <span class="cf">for</span> i <span class="kw">in</span> sorted_species_vars])</span>
<span id="cb15-199"><a href="#cb15-199" aria-hidden="true" tabindex="-1"></a>                plt.xlabel(<span class="st">'Component (by species variance)'</span>)</span>
<span id="cb15-200"><a href="#cb15-200" aria-hidden="true" tabindex="-1"></a>                plt.ylabel(<span class="st">'Species Variance'</span>)</span>
<span id="cb15-201"><a href="#cb15-201" aria-hidden="true" tabindex="-1"></a>                plt.title(<span class="ss">f'Species Variability (Rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb15-202"><a href="#cb15-202" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> rank <span class="op">&lt;=</span> <span class="dv">10</span>:</span>
<span id="cb15-203"><a href="#cb15-203" aria-hidden="true" tabindex="-1"></a>                    plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, rank <span class="op">+</span> <span class="dv">1</span>))</span>
<span id="cb15-204"><a href="#cb15-204" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-205"><a href="#cb15-205" aria-hidden="true" tabindex="-1"></a>                <span class="co"># 9. Gene factor distribution</span></span>
<span id="cb15-206"><a href="#cb15-206" aria-hidden="true" tabindex="-1"></a>                plt.subplot(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">9</span>)</span>
<span id="cb15-207"><a href="#cb15-207" aria-hidden="true" tabindex="-1"></a>                gene_variances <span class="op">=</span> np.var(gene_factors, axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb15-208"><a href="#cb15-208" aria-hidden="true" tabindex="-1"></a>                plt.hist(gene_variances, bins<span class="op">=</span><span class="dv">30</span>, alpha<span class="op">=</span><span class="fl">0.7</span>, edgecolor<span class="op">=</span><span class="st">'black'</span>)</span>
<span id="cb15-209"><a href="#cb15-209" aria-hidden="true" tabindex="-1"></a>                plt.xlabel(<span class="st">'Gene Factor Variance'</span>)</span>
<span id="cb15-210"><a href="#cb15-210" aria-hidden="true" tabindex="-1"></a>                plt.ylabel(<span class="st">'Number of Genes'</span>)</span>
<span id="cb15-211"><a href="#cb15-211" aria-hidden="true" tabindex="-1"></a>                plt.title(<span class="ss">f'Gene Variability (Rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb15-212"><a href="#cb15-212" aria-hidden="true" tabindex="-1"></a>                plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb15-213"><a href="#cb15-213" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-214"><a href="#cb15-214" aria-hidden="true" tabindex="-1"></a>                plt.tight_layout()</span>
<span id="cb15-215"><a href="#cb15-215" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-216"><a href="#cb15-216" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Save rank-specific visualization in the main results directory for easy access</span></span>
<span id="cb15-217"><a href="#cb15-217" aria-hidden="true" tabindex="-1"></a>                rank_plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"rank_</span><span class="sc">{</span>rank<span class="sc">:02d}</span><span class="ss">_comprehensive_visualization.png"</span>)</span>
<span id="cb15-218"><a href="#cb15-218" aria-hidden="true" tabindex="-1"></a>                plt.savefig(rank_plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb15-219"><a href="#cb15-219" aria-hidden="true" tabindex="-1"></a>                <span class="bu">print</span>(<span class="ss">f"  Saved: </span><span class="sc">{</span>rank_plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb15-220"><a href="#cb15-220" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-221"><a href="#cb15-221" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Also save in the rank-specific subdirectory</span></span>
<span id="cb15-222"><a href="#cb15-222" aria-hidden="true" tabindex="-1"></a>                rank_subdir_plot <span class="op">=</span> os.path.join(result_dir, <span class="ss">f"rank_</span><span class="sc">{</span>rank<span class="sc">:02d}</span><span class="ss">_comprehensive_visualization.png"</span>)</span>
<span id="cb15-223"><a href="#cb15-223" aria-hidden="true" tabindex="-1"></a>                plt.savefig(rank_subdir_plot, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb15-224"><a href="#cb15-224" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-225"><a href="#cb15-225" aria-hidden="true" tabindex="-1"></a>                plt.close()  <span class="co"># Close the figure to free memory instead of showing all plots</span></span>
<span id="cb15-226"><a href="#cb15-226" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb15-227"><a href="#cb15-227" aria-hidden="true" tabindex="-1"></a>            <span class="cf">except</span> <span class="pp">Exception</span> <span class="im">as</span> e:</span>
<span id="cb15-228"><a href="#cb15-228" aria-hidden="true" tabindex="-1"></a>                <span class="bu">print</span>(<span class="ss">f"  Error processing rank </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">: </span><span class="sc">{</span>e<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb15-229"><a href="#cb15-229" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb15-230"><a href="#cb15-230" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Successfully generated visualizations for ranks: </span><span class="sc">{</span>successful_ranks<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb15-231"><a href="#cb15-231" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb15-232"><a href="#cb15-232" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb15-233"><a href="#cb15-233" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Rank comparison results file not found!"</span>)</span>
<span id="cb15-234"><a href="#cb15-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-235"><a href="#cb15-235" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb15-236"><a href="#cb15-236" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="visualize-optimal-decomposition-results" class="level2">
<h2 class="anchored" data-anchor-id="visualize-optimal-decomposition-results">4.2 Visualize optimal decomposition results</h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> sklearn.cluster <span class="im">import</span> KMeans</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> sklearn.decomposition <span class="im">import</span> PCA</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting style</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>plt.style.use(<span class="st">'default'</span>)</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>sns.set_palette(<span class="st">"husl"</span>)</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Find the main results directory</span></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> output_dir</span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Load rank selection summary to get optimal rank</span></span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>    <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>        rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Visualizing optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> results from rank comparison data"</span>)</span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Find the optimal rank data in rank_comparison directory</span></span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Using optimal analysis data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found in rank comparison!"</span>)</span>
<span id="cb16-55"><a href="#cb16-55" aria-hidden="true" tabindex="-1"></a>        latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb16-56"><a href="#cb16-56" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb16-57"><a href="#cb16-57" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No rank selection summary found!"</span>)</span>
<span id="cb16-58"><a href="#cb16-58" aria-hidden="true" tabindex="-1"></a>    latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb16-59"><a href="#cb16-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-60"><a href="#cb16-60" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb16-61"><a href="#cb16-61" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Visualizing results from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb16-62"><a href="#cb16-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-63"><a href="#cb16-63" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load all data</span></span>
<span id="cb16-64"><a href="#cb16-64" aria-hidden="true" tabindex="-1"></a>    gene_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb16-65"><a href="#cb16-65" aria-hidden="true" tabindex="-1"></a>    species_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"species_factors.npy"</span>))</span>
<span id="cb16-66"><a href="#cb16-66" aria-hidden="true" tabindex="-1"></a>    time_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"time_factors.npy"</span>))</span>
<span id="cb16-67"><a href="#cb16-67" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-68"><a href="#cb16-68" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load the transformed data to get species names</span></span>
<span id="cb16-69"><a href="#cb16-69" aria-hidden="true" tabindex="-1"></a>    data_file <span class="op">=</span> os.path.join(output_dir, <span class="st">"vst_counts_matrix_long_format.csv"</span>)</span>
<span id="cb16-70"><a href="#cb16-70" aria-hidden="true" tabindex="-1"></a>    df_long <span class="op">=</span> pd.read_csv(data_file)</span>
<span id="cb16-71"><a href="#cb16-71" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-72"><a href="#cb16-72" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get unique species and timepoint names</span></span>
<span id="cb16-73"><a href="#cb16-73" aria-hidden="true" tabindex="-1"></a>    species_names <span class="op">=</span> <span class="bu">sorted</span>(df_long[<span class="st">'species'</span>].unique())</span>
<span id="cb16-74"><a href="#cb16-74" aria-hidden="true" tabindex="-1"></a>    timepoint_names <span class="op">=</span> <span class="bu">sorted</span>(df_long[<span class="st">'timepoint'</span>].unique())</span>
<span id="cb16-75"><a href="#cb16-75" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-76"><a href="#cb16-76" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate component strengths</span></span>
<span id="cb16-77"><a href="#cb16-77" aria-hidden="true" tabindex="-1"></a>    component_strengths <span class="op">=</span> []</span>
<span id="cb16-78"><a href="#cb16-78" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(gene_factors.shape[<span class="dv">1</span>]):</span>
<span id="cb16-79"><a href="#cb16-79" aria-hidden="true" tabindex="-1"></a>        gene_norm <span class="op">=</span> np.linalg.norm(gene_factors[:, i])</span>
<span id="cb16-80"><a href="#cb16-80" aria-hidden="true" tabindex="-1"></a>        species_norm <span class="op">=</span> np.linalg.norm(species_factors[:, i])</span>
<span id="cb16-81"><a href="#cb16-81" aria-hidden="true" tabindex="-1"></a>        time_norm <span class="op">=</span> np.linalg.norm(time_factors[:, i])</span>
<span id="cb16-82"><a href="#cb16-82" aria-hidden="true" tabindex="-1"></a>        total_strength <span class="op">=</span> gene_norm <span class="op">*</span> species_norm <span class="op">*</span> time_norm</span>
<span id="cb16-83"><a href="#cb16-83" aria-hidden="true" tabindex="-1"></a>        component_strengths.append(total_strength)</span>
<span id="cb16-84"><a href="#cb16-84" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-85"><a href="#cb16-85" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Sort components by strength</span></span>
<span id="cb16-86"><a href="#cb16-86" aria-hidden="true" tabindex="-1"></a>    sorted_components <span class="op">=</span> np.argsort(component_strengths)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb16-87"><a href="#cb16-87" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-88"><a href="#cb16-88" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Determine how many top components to show (max 5, or all if fewer)</span></span>
<span id="cb16-89"><a href="#cb16-89" aria-hidden="true" tabindex="-1"></a>    n_components <span class="op">=</span> gene_factors.shape[<span class="dv">1</span>]</span>
<span id="cb16-90"><a href="#cb16-90" aria-hidden="true" tabindex="-1"></a>    n_top_display <span class="op">=</span> <span class="bu">min</span>(<span class="dv">5</span>, n_components)</span>
<span id="cb16-91"><a href="#cb16-91" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-92"><a href="#cb16-92" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create figure with multiple subplots</span></span>
<span id="cb16-93"><a href="#cb16-93" aria-hidden="true" tabindex="-1"></a>    fig <span class="op">=</span> plt.figure(figsize<span class="op">=</span>(<span class="dv">20</span>, <span class="dv">24</span>))</span>
<span id="cb16-94"><a href="#cb16-94" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-95"><a href="#cb16-95" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 1. Component strengths bar plot</span></span>
<span id="cb16-96"><a href="#cb16-96" aria-hidden="true" tabindex="-1"></a>    plt.subplot(<span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">1</span>)</span>
<span id="cb16-97"><a href="#cb16-97" aria-hidden="true" tabindex="-1"></a>    plt.bar(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(component_strengths) <span class="op">+</span> <span class="dv">1</span>), </span>
<span id="cb16-98"><a href="#cb16-98" aria-hidden="true" tabindex="-1"></a>            [component_strengths[i] <span class="cf">for</span> i <span class="kw">in</span> sorted_components])</span>
<span id="cb16-99"><a href="#cb16-99" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Component (ranked by strength)'</span>)</span>
<span id="cb16-100"><a href="#cb16-100" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Component Strength'</span>)</span>
<span id="cb16-101"><a href="#cb16-101" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="st">'Component Strengths (Gene × Species × Time norms)'</span>)</span>
<span id="cb16-102"><a href="#cb16-102" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> n_components <span class="op">&lt;=</span> <span class="dv">10</span>:</span>
<span id="cb16-103"><a href="#cb16-103" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>))</span>
<span id="cb16-104"><a href="#cb16-104" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb16-105"><a href="#cb16-105" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>, <span class="bu">max</span>(<span class="dv">1</span>, n_components <span class="op">//</span> <span class="dv">10</span>)))</span>
<span id="cb16-106"><a href="#cb16-106" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-107"><a href="#cb16-107" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 2. Time factors heatmap for top components</span></span>
<span id="cb16-108"><a href="#cb16-108" aria-hidden="true" tabindex="-1"></a>    plt.subplot(<span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">2</span>)</span>
<span id="cb16-109"><a href="#cb16-109" aria-hidden="true" tabindex="-1"></a>    top_components <span class="op">=</span> sorted_components[:n_top_display]</span>
<span id="cb16-110"><a href="#cb16-110" aria-hidden="true" tabindex="-1"></a>    time_data <span class="op">=</span> time_factors[:, top_components]</span>
<span id="cb16-111"><a href="#cb16-111" aria-hidden="true" tabindex="-1"></a>    sns.heatmap(time_data.T, </span>
<span id="cb16-112"><a href="#cb16-112" aria-hidden="true" tabindex="-1"></a>                xticklabels<span class="op">=</span>timepoint_names,</span>
<span id="cb16-113"><a href="#cb16-113" aria-hidden="true" tabindex="-1"></a>                yticklabels<span class="op">=</span>[<span class="ss">f'Comp </span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> top_components],</span>
<span id="cb16-114"><a href="#cb16-114" aria-hidden="true" tabindex="-1"></a>                cmap<span class="op">=</span><span class="st">'RdBu_r'</span>, center<span class="op">=</span><span class="dv">0</span>, annot<span class="op">=</span><span class="va">True</span>, fmt<span class="op">=</span><span class="st">'.3f'</span>)</span>
<span id="cb16-115"><a href="#cb16-115" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Top </span><span class="sc">{</span>n_top_display<span class="sc">}</span><span class="ss"> Components: Temporal Patterns'</span>)</span>
<span id="cb16-116"><a href="#cb16-116" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Timepoint'</span>)</span>
<span id="cb16-117"><a href="#cb16-117" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Component'</span>)</span>
<span id="cb16-118"><a href="#cb16-118" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-119"><a href="#cb16-119" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 3. Species factors heatmap for top components</span></span>
<span id="cb16-120"><a href="#cb16-120" aria-hidden="true" tabindex="-1"></a>    plt.subplot(<span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">3</span>)</span>
<span id="cb16-121"><a href="#cb16-121" aria-hidden="true" tabindex="-1"></a>    species_data <span class="op">=</span> species_factors[:, top_components]</span>
<span id="cb16-122"><a href="#cb16-122" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-123"><a href="#cb16-123" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create species type labels for better visualization</span></span>
<span id="cb16-124"><a href="#cb16-124" aria-hidden="true" tabindex="-1"></a>    species_types <span class="op">=</span> [name.split(<span class="st">'-'</span>)[<span class="dv">0</span>] <span class="cf">for</span> name <span class="kw">in</span> species_names]</span>
<span id="cb16-125"><a href="#cb16-125" aria-hidden="true" tabindex="-1"></a>    unique_types <span class="op">=</span> <span class="bu">list</span>(<span class="bu">set</span>(species_types))</span>
<span id="cb16-126"><a href="#cb16-126" aria-hidden="true" tabindex="-1"></a>    type_colors <span class="op">=</span> plt.cm.Set3(np.linspace(<span class="dv">0</span>, <span class="dv">1</span>, <span class="bu">len</span>(unique_types)))</span>
<span id="cb16-127"><a href="#cb16-127" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-128"><a href="#cb16-128" aria-hidden="true" tabindex="-1"></a>    sns.heatmap(species_data.T,</span>
<span id="cb16-129"><a href="#cb16-129" aria-hidden="true" tabindex="-1"></a>                xticklabels<span class="op">=</span>[<span class="ss">f"</span><span class="sc">{</span>name[:<span class="dv">8</span>]<span class="sc">}</span><span class="ss">..."</span> <span class="cf">if</span> <span class="bu">len</span>(name) <span class="op">&gt;</span> <span class="dv">8</span> <span class="cf">else</span> name <span class="cf">for</span> name <span class="kw">in</span> species_names],</span>
<span id="cb16-130"><a href="#cb16-130" aria-hidden="true" tabindex="-1"></a>                yticklabels<span class="op">=</span>[<span class="ss">f'Comp </span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> top_components],</span>
<span id="cb16-131"><a href="#cb16-131" aria-hidden="true" tabindex="-1"></a>                cmap<span class="op">=</span><span class="st">'RdBu_r'</span>, center<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb16-132"><a href="#cb16-132" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Top </span><span class="sc">{</span>n_top_display<span class="sc">}</span><span class="ss"> Components: Species Patterns'</span>)</span>
<span id="cb16-133"><a href="#cb16-133" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Individual'</span>)</span>
<span id="cb16-134"><a href="#cb16-134" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Component'</span>)</span>
<span id="cb16-135"><a href="#cb16-135" aria-hidden="true" tabindex="-1"></a>    plt.xticks(rotation<span class="op">=</span><span class="dv">90</span>)</span>
<span id="cb16-136"><a href="#cb16-136" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-137"><a href="#cb16-137" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 4. Temporal patterns for strongest components</span></span>
<span id="cb16-138"><a href="#cb16-138" aria-hidden="true" tabindex="-1"></a>    plt.subplot(<span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">4</span>)</span>
<span id="cb16-139"><a href="#cb16-139" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i, comp_idx <span class="kw">in</span> <span class="bu">enumerate</span>(top_components):</span>
<span id="cb16-140"><a href="#cb16-140" aria-hidden="true" tabindex="-1"></a>        plt.plot(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(timepoint_names) <span class="op">+</span> <span class="dv">1</span>), </span>
<span id="cb16-141"><a href="#cb16-141" aria-hidden="true" tabindex="-1"></a>                time_factors[:, comp_idx], </span>
<span id="cb16-142"><a href="#cb16-142" aria-hidden="true" tabindex="-1"></a>                marker<span class="op">=</span><span class="st">'o'</span>, linewidth<span class="op">=</span><span class="dv">2</span>, label<span class="op">=</span><span class="ss">f'Component </span><span class="sc">{</span>comp_idx <span class="op">+</span> <span class="dv">1</span><span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb16-143"><a href="#cb16-143" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Timepoint'</span>)</span>
<span id="cb16-144"><a href="#cb16-144" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Factor Value'</span>)</span>
<span id="cb16-145"><a href="#cb16-145" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Temporal Trajectories (Top </span><span class="sc">{</span>n_top_display<span class="sc">}</span><span class="ss"> Components)'</span>)</span>
<span id="cb16-146"><a href="#cb16-146" aria-hidden="true" tabindex="-1"></a>    plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(timepoint_names) <span class="op">+</span> <span class="dv">1</span>), timepoint_names)</span>
<span id="cb16-147"><a href="#cb16-147" aria-hidden="true" tabindex="-1"></a>    plt.legend()</span>
<span id="cb16-148"><a href="#cb16-148" aria-hidden="true" tabindex="-1"></a>    plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb16-149"><a href="#cb16-149" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-150"><a href="#cb16-150" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 5. Species clustering based on factor loadings</span></span>
<span id="cb16-151"><a href="#cb16-151" aria-hidden="true" tabindex="-1"></a>    plt.subplot(<span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">5</span>)</span>
<span id="cb16-152"><a href="#cb16-152" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use all components for clustering if &lt;= 10, otherwise top 10</span></span>
<span id="cb16-153"><a href="#cb16-153" aria-hidden="true" tabindex="-1"></a>    n_cluster_components <span class="op">=</span> <span class="bu">min</span>(<span class="dv">10</span>, n_components)</span>
<span id="cb16-154"><a href="#cb16-154" aria-hidden="true" tabindex="-1"></a>    species_subset <span class="op">=</span> species_factors[:, sorted_components[:n_cluster_components]]</span>
<span id="cb16-155"><a href="#cb16-155" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-156"><a href="#cb16-156" aria-hidden="true" tabindex="-1"></a>    <span class="co"># PCA for visualization</span></span>
<span id="cb16-157"><a href="#cb16-157" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> species_subset.shape[<span class="dv">1</span>] <span class="op">&gt;=</span> <span class="dv">2</span>:</span>
<span id="cb16-158"><a href="#cb16-158" aria-hidden="true" tabindex="-1"></a>        pca <span class="op">=</span> PCA(n_components<span class="op">=</span><span class="dv">2</span>)</span>
<span id="cb16-159"><a href="#cb16-159" aria-hidden="true" tabindex="-1"></a>        species_pca <span class="op">=</span> pca.fit_transform(species_subset)</span>
<span id="cb16-160"><a href="#cb16-160" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-161"><a href="#cb16-161" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Color by species type</span></span>
<span id="cb16-162"><a href="#cb16-162" aria-hidden="true" tabindex="-1"></a>        colors <span class="op">=</span> [<span class="st">'red'</span>, <span class="st">'blue'</span>, <span class="st">'green'</span>, <span class="st">'orange'</span>, <span class="st">'purple'</span>, <span class="st">'brown'</span>][:<span class="bu">len</span>(unique_types)]</span>
<span id="cb16-163"><a href="#cb16-163" aria-hidden="true" tabindex="-1"></a>        species_type_map <span class="op">=</span> {stype: i <span class="cf">for</span> i, stype <span class="kw">in</span> <span class="bu">enumerate</span>(unique_types)}</span>
<span id="cb16-164"><a href="#cb16-164" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-165"><a href="#cb16-165" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> i, (x, y) <span class="kw">in</span> <span class="bu">enumerate</span>(species_pca):</span>
<span id="cb16-166"><a href="#cb16-166" aria-hidden="true" tabindex="-1"></a>            stype <span class="op">=</span> species_types[i]</span>
<span id="cb16-167"><a href="#cb16-167" aria-hidden="true" tabindex="-1"></a>            plt.scatter(x, y, c<span class="op">=</span>colors[species_type_map[stype]], </span>
<span id="cb16-168"><a href="#cb16-168" aria-hidden="true" tabindex="-1"></a>                       alpha<span class="op">=</span><span class="fl">0.7</span>, s<span class="op">=</span><span class="dv">60</span>, label<span class="op">=</span>stype <span class="cf">if</span> i <span class="op">==</span> species_types.index(stype) <span class="cf">else</span> <span class="st">""</span>)</span>
<span id="cb16-169"><a href="#cb16-169" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-170"><a href="#cb16-170" aria-hidden="true" tabindex="-1"></a>        plt.xlabel(<span class="ss">f'PC1 (</span><span class="sc">{</span>pca<span class="sc">.</span>explained_variance_ratio_[<span class="dv">0</span>]<span class="sc">:.1%}</span><span class="ss"> variance)'</span>)</span>
<span id="cb16-171"><a href="#cb16-171" aria-hidden="true" tabindex="-1"></a>        plt.ylabel(<span class="ss">f'PC2 (</span><span class="sc">{</span>pca<span class="sc">.</span>explained_variance_ratio_[<span class="dv">1</span>]<span class="sc">:.1%}</span><span class="ss"> variance)'</span>)</span>
<span id="cb16-172"><a href="#cb16-172" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="st">'Species Clustering (PCA of Factor Loadings)'</span>)</span>
<span id="cb16-173"><a href="#cb16-173" aria-hidden="true" tabindex="-1"></a>        plt.legend()</span>
<span id="cb16-174"><a href="#cb16-174" aria-hidden="true" tabindex="-1"></a>        plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb16-175"><a href="#cb16-175" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb16-176"><a href="#cb16-176" aria-hidden="true" tabindex="-1"></a>        plt.text(<span class="fl">0.5</span>, <span class="fl">0.5</span>, <span class="st">'Not enough components</span><span class="ch">\n</span><span class="st">for PCA visualization'</span>, </span>
<span id="cb16-177"><a href="#cb16-177" aria-hidden="true" tabindex="-1"></a>                transform<span class="op">=</span>plt.gca().transAxes, ha<span class="op">=</span><span class="st">'center'</span>, va<span class="op">=</span><span class="st">'center'</span>)</span>
<span id="cb16-178"><a href="#cb16-178" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="st">'Species Clustering (Insufficient Components)'</span>)</span>
<span id="cb16-179"><a href="#cb16-179" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-180"><a href="#cb16-180" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 6. Gene factor distribution</span></span>
<span id="cb16-181"><a href="#cb16-181" aria-hidden="true" tabindex="-1"></a>    plt.subplot(<span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">6</span>)</span>
<span id="cb16-182"><a href="#cb16-182" aria-hidden="true" tabindex="-1"></a>    gene_variances <span class="op">=</span> np.var(gene_factors, axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb16-183"><a href="#cb16-183" aria-hidden="true" tabindex="-1"></a>    plt.hist(gene_variances, bins<span class="op">=</span><span class="dv">50</span>, alpha<span class="op">=</span><span class="fl">0.7</span>, edgecolor<span class="op">=</span><span class="st">'black'</span>)</span>
<span id="cb16-184"><a href="#cb16-184" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Gene Factor Variance'</span>)</span>
<span id="cb16-185"><a href="#cb16-185" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Number of Genes'</span>)</span>
<span id="cb16-186"><a href="#cb16-186" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="st">'Distribution of Gene Factor Variability'</span>)</span>
<span id="cb16-187"><a href="#cb16-187" aria-hidden="true" tabindex="-1"></a>    plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb16-188"><a href="#cb16-188" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-189"><a href="#cb16-189" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 7. Component correlation matrix</span></span>
<span id="cb16-190"><a href="#cb16-190" aria-hidden="true" tabindex="-1"></a>    plt.subplot(<span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">7</span>)</span>
<span id="cb16-191"><a href="#cb16-191" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate correlations between components across genes</span></span>
<span id="cb16-192"><a href="#cb16-192" aria-hidden="true" tabindex="-1"></a>    gene_corr <span class="op">=</span> np.corrcoef(gene_factors.T)</span>
<span id="cb16-193"><a href="#cb16-193" aria-hidden="true" tabindex="-1"></a>    sns.heatmap(gene_corr, cmap<span class="op">=</span><span class="st">'RdBu_r'</span>, center<span class="op">=</span><span class="dv">0</span>, </span>
<span id="cb16-194"><a href="#cb16-194" aria-hidden="true" tabindex="-1"></a>                xticklabels<span class="op">=</span><span class="bu">range</span>(<span class="dv">1</span>, gene_factors.shape[<span class="dv">1</span>] <span class="op">+</span> <span class="dv">1</span>),</span>
<span id="cb16-195"><a href="#cb16-195" aria-hidden="true" tabindex="-1"></a>                yticklabels<span class="op">=</span><span class="bu">range</span>(<span class="dv">1</span>, gene_factors.shape[<span class="dv">1</span>] <span class="op">+</span> <span class="dv">1</span>))</span>
<span id="cb16-196"><a href="#cb16-196" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="st">'Component Correlations (Gene Space)'</span>)</span>
<span id="cb16-197"><a href="#cb16-197" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Component'</span>)</span>
<span id="cb16-198"><a href="#cb16-198" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Component'</span>)</span>
<span id="cb16-199"><a href="#cb16-199" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-200"><a href="#cb16-200" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 8. Species type comparison</span></span>
<span id="cb16-201"><a href="#cb16-201" aria-hidden="true" tabindex="-1"></a>    plt.subplot(<span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">8</span>)</span>
<span id="cb16-202"><a href="#cb16-202" aria-hidden="true" tabindex="-1"></a>    species_by_type <span class="op">=</span> {}</span>
<span id="cb16-203"><a href="#cb16-203" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i, name <span class="kw">in</span> <span class="bu">enumerate</span>(species_names):</span>
<span id="cb16-204"><a href="#cb16-204" aria-hidden="true" tabindex="-1"></a>        stype <span class="op">=</span> name.split(<span class="st">'-'</span>)[<span class="dv">0</span>]</span>
<span id="cb16-205"><a href="#cb16-205" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> stype <span class="kw">not</span> <span class="kw">in</span> species_by_type:</span>
<span id="cb16-206"><a href="#cb16-206" aria-hidden="true" tabindex="-1"></a>            species_by_type[stype] <span class="op">=</span> []</span>
<span id="cb16-207"><a href="#cb16-207" aria-hidden="true" tabindex="-1"></a>        species_by_type[stype].append(i)</span>
<span id="cb16-208"><a href="#cb16-208" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-209"><a href="#cb16-209" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate mean factor values per species type for top components</span></span>
<span id="cb16-210"><a href="#cb16-210" aria-hidden="true" tabindex="-1"></a>    type_means <span class="op">=</span> {}</span>
<span id="cb16-211"><a href="#cb16-211" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> stype, indices <span class="kw">in</span> species_by_type.items():</span>
<span id="cb16-212"><a href="#cb16-212" aria-hidden="true" tabindex="-1"></a>        type_means[stype] <span class="op">=</span> np.mean(species_factors[indices, :], axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb16-213"><a href="#cb16-213" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-214"><a href="#cb16-214" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Plot for top components</span></span>
<span id="cb16-215"><a href="#cb16-215" aria-hidden="true" tabindex="-1"></a>    x_pos <span class="op">=</span> np.arange(<span class="bu">len</span>(top_components))</span>
<span id="cb16-216"><a href="#cb16-216" aria-hidden="true" tabindex="-1"></a>    width <span class="op">=</span> <span class="fl">0.8</span> <span class="op">/</span> <span class="bu">len</span>(type_means)  <span class="co"># Adjust width based on number of species types</span></span>
<span id="cb16-217"><a href="#cb16-217" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-218"><a href="#cb16-218" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i, (stype, means) <span class="kw">in</span> <span class="bu">enumerate</span>(type_means.items()):</span>
<span id="cb16-219"><a href="#cb16-219" aria-hidden="true" tabindex="-1"></a>        plt.bar(x_pos <span class="op">+</span> i <span class="op">*</span> width, means[top_components], </span>
<span id="cb16-220"><a href="#cb16-220" aria-hidden="true" tabindex="-1"></a>               width, label<span class="op">=</span>stype, alpha<span class="op">=</span><span class="fl">0.8</span>)</span>
<span id="cb16-221"><a href="#cb16-221" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-222"><a href="#cb16-222" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Component'</span>)</span>
<span id="cb16-223"><a href="#cb16-223" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Mean Factor Value'</span>)</span>
<span id="cb16-224"><a href="#cb16-224" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Species Type Comparison (Top </span><span class="sc">{</span>n_top_display<span class="sc">}</span><span class="ss"> Components)'</span>)</span>
<span id="cb16-225"><a href="#cb16-225" aria-hidden="true" tabindex="-1"></a>    plt.xticks(x_pos <span class="op">+</span> width <span class="op">*</span> (<span class="bu">len</span>(type_means) <span class="op">-</span> <span class="dv">1</span>) <span class="op">/</span> <span class="dv">2</span>, </span>
<span id="cb16-226"><a href="#cb16-226" aria-hidden="true" tabindex="-1"></a>               [<span class="ss">f'Comp </span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> top_components])</span>
<span id="cb16-227"><a href="#cb16-227" aria-hidden="true" tabindex="-1"></a>    plt.legend()</span>
<span id="cb16-228"><a href="#cb16-228" aria-hidden="true" tabindex="-1"></a>    plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb16-229"><a href="#cb16-229" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-230"><a href="#cb16-230" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 9. Temporal variance per component</span></span>
<span id="cb16-231"><a href="#cb16-231" aria-hidden="true" tabindex="-1"></a>    plt.subplot(<span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">9</span>)</span>
<span id="cb16-232"><a href="#cb16-232" aria-hidden="true" tabindex="-1"></a>    time_variances <span class="op">=</span> np.var(time_factors, axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb16-233"><a href="#cb16-233" aria-hidden="true" tabindex="-1"></a>    sorted_time_vars <span class="op">=</span> np.argsort(time_variances)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb16-234"><a href="#cb16-234" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-235"><a href="#cb16-235" aria-hidden="true" tabindex="-1"></a>    plt.bar(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(time_variances) <span class="op">+</span> <span class="dv">1</span>), </span>
<span id="cb16-236"><a href="#cb16-236" aria-hidden="true" tabindex="-1"></a>           [time_variances[i] <span class="cf">for</span> i <span class="kw">in</span> sorted_time_vars])</span>
<span id="cb16-237"><a href="#cb16-237" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Component (ranked by temporal variance)'</span>)</span>
<span id="cb16-238"><a href="#cb16-238" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Temporal Variance'</span>)</span>
<span id="cb16-239"><a href="#cb16-239" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="st">'Components by Temporal Variability'</span>)</span>
<span id="cb16-240"><a href="#cb16-240" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> n_components <span class="op">&lt;=</span> <span class="dv">10</span>:</span>
<span id="cb16-241"><a href="#cb16-241" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>))</span>
<span id="cb16-242"><a href="#cb16-242" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb16-243"><a href="#cb16-243" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>, <span class="bu">max</span>(<span class="dv">1</span>, n_components <span class="op">//</span> <span class="dv">10</span>)))</span>
<span id="cb16-244"><a href="#cb16-244" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-245"><a href="#cb16-245" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 10. Species variance per component</span></span>
<span id="cb16-246"><a href="#cb16-246" aria-hidden="true" tabindex="-1"></a>    plt.subplot(<span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">10</span>)</span>
<span id="cb16-247"><a href="#cb16-247" aria-hidden="true" tabindex="-1"></a>    species_variances <span class="op">=</span> np.var(species_factors, axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb16-248"><a href="#cb16-248" aria-hidden="true" tabindex="-1"></a>    sorted_species_vars <span class="op">=</span> np.argsort(species_variances)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb16-249"><a href="#cb16-249" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-250"><a href="#cb16-250" aria-hidden="true" tabindex="-1"></a>    plt.bar(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(species_variances) <span class="op">+</span> <span class="dv">1</span>), </span>
<span id="cb16-251"><a href="#cb16-251" aria-hidden="true" tabindex="-1"></a>           [species_variances[i] <span class="cf">for</span> i <span class="kw">in</span> sorted_species_vars])</span>
<span id="cb16-252"><a href="#cb16-252" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Component (ranked by species variance)'</span>)</span>
<span id="cb16-253"><a href="#cb16-253" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Species Variance'</span>)</span>
<span id="cb16-254"><a href="#cb16-254" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="st">'Components by Species Variability'</span>)</span>
<span id="cb16-255"><a href="#cb16-255" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> n_components <span class="op">&lt;=</span> <span class="dv">10</span>:</span>
<span id="cb16-256"><a href="#cb16-256" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>))</span>
<span id="cb16-257"><a href="#cb16-257" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb16-258"><a href="#cb16-258" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>, <span class="bu">max</span>(<span class="dv">1</span>, n_components <span class="op">//</span> <span class="dv">10</span>)))</span>
<span id="cb16-259"><a href="#cb16-259" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-260"><a href="#cb16-260" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 11. Top genes per component heatmap</span></span>
<span id="cb16-261"><a href="#cb16-261" aria-hidden="true" tabindex="-1"></a>    plt.subplot(<span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">11</span>)</span>
<span id="cb16-262"><a href="#cb16-262" aria-hidden="true" tabindex="-1"></a>    n_top_genes <span class="op">=</span> <span class="dv">10</span></span>
<span id="cb16-263"><a href="#cb16-263" aria-hidden="true" tabindex="-1"></a>    top_gene_indices <span class="op">=</span> []</span>
<span id="cb16-264"><a href="#cb16-264" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-265"><a href="#cb16-265" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get top genes for each of the top components</span></span>
<span id="cb16-266"><a href="#cb16-266" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> comp_idx <span class="kw">in</span> top_components:</span>
<span id="cb16-267"><a href="#cb16-267" aria-hidden="true" tabindex="-1"></a>        top_genes <span class="op">=</span> np.argsort(np.<span class="bu">abs</span>(gene_factors[:, comp_idx]))[<span class="op">-</span>n_top_genes:]</span>
<span id="cb16-268"><a href="#cb16-268" aria-hidden="true" tabindex="-1"></a>        top_gene_indices.extend(top_genes)</span>
<span id="cb16-269"><a href="#cb16-269" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-270"><a href="#cb16-270" aria-hidden="true" tabindex="-1"></a>    top_gene_indices <span class="op">=</span> <span class="bu">list</span>(<span class="bu">set</span>(top_gene_indices))  <span class="co"># Remove duplicates</span></span>
<span id="cb16-271"><a href="#cb16-271" aria-hidden="true" tabindex="-1"></a>    top_gene_data <span class="op">=</span> gene_factors[top_gene_indices, :][:, top_components]</span>
<span id="cb16-272"><a href="#cb16-272" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-273"><a href="#cb16-273" aria-hidden="true" tabindex="-1"></a>    sns.heatmap(top_gene_data, </span>
<span id="cb16-274"><a href="#cb16-274" aria-hidden="true" tabindex="-1"></a>                cmap<span class="op">=</span><span class="st">'RdBu_r'</span>, center<span class="op">=</span><span class="dv">0</span>,</span>
<span id="cb16-275"><a href="#cb16-275" aria-hidden="true" tabindex="-1"></a>                yticklabels<span class="op">=</span>[<span class="ss">f'Gene </span><span class="sc">{</span>i<span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> top_gene_indices],</span>
<span id="cb16-276"><a href="#cb16-276" aria-hidden="true" tabindex="-1"></a>                xticklabels<span class="op">=</span>[<span class="ss">f'Comp </span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> top_components])</span>
<span id="cb16-277"><a href="#cb16-277" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Top </span><span class="sc">{</span><span class="bu">len</span>(top_gene_indices)<span class="sc">}</span><span class="ss"> Genes × Top </span><span class="sc">{</span>n_top_display<span class="sc">}</span><span class="ss"> Components'</span>)</span>
<span id="cb16-278"><a href="#cb16-278" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Component'</span>)</span>
<span id="cb16-279"><a href="#cb16-279" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Gene'</span>)</span>
<span id="cb16-280"><a href="#cb16-280" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-281"><a href="#cb16-281" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 12. Reconstruction quality assessment</span></span>
<span id="cb16-282"><a href="#cb16-282" aria-hidden="true" tabindex="-1"></a>    plt.subplot(<span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">12</span>)</span>
<span id="cb16-283"><a href="#cb16-283" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load reconstructed tensor</span></span>
<span id="cb16-284"><a href="#cb16-284" aria-hidden="true" tabindex="-1"></a>    reconstructed <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"reconstructed_tensor.npy"</span>))</span>
<span id="cb16-285"><a href="#cb16-285" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-286"><a href="#cb16-286" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create original tensor from the long format data</span></span>
<span id="cb16-287"><a href="#cb16-287" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Pivot to get the original tensor structure</span></span>
<span id="cb16-288"><a href="#cb16-288" aria-hidden="true" tabindex="-1"></a>    original_pivot <span class="op">=</span> df_long.pivot_table(values<span class="op">=</span><span class="st">'expression'</span>, </span>
<span id="cb16-289"><a href="#cb16-289" aria-hidden="true" tabindex="-1"></a>                                       index<span class="op">=</span><span class="st">'gene'</span>, </span>
<span id="cb16-290"><a href="#cb16-290" aria-hidden="true" tabindex="-1"></a>                                       columns<span class="op">=</span>[<span class="st">'species'</span>, <span class="st">'timepoint'</span>], </span>
<span id="cb16-291"><a href="#cb16-291" aria-hidden="true" tabindex="-1"></a>                                       fill_value<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb16-292"><a href="#cb16-292" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-293"><a href="#cb16-293" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Debug: Original pivot shape: </span><span class="sc">{</span>original_pivot<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb16-294"><a href="#cb16-294" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Debug: Reconstructed tensor shape: </span><span class="sc">{</span>reconstructed<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb16-295"><a href="#cb16-295" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Debug: Expected shape: (</span><span class="sc">{</span>gene_factors<span class="sc">.</span>shape[<span class="dv">0</span>]<span class="sc">}</span><span class="ss">, </span><span class="sc">{</span>species_factors<span class="sc">.</span>shape[<span class="dv">0</span>]<span class="sc">}</span><span class="ss">, </span><span class="sc">{</span>time_factors<span class="sc">.</span>shape[<span class="dv">0</span>]<span class="sc">}</span><span class="ss">)"</span>)</span>
<span id="cb16-296"><a href="#cb16-296" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-297"><a href="#cb16-297" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if we can safely reshape</span></span>
<span id="cb16-298"><a href="#cb16-298" aria-hidden="true" tabindex="-1"></a>    expected_size <span class="op">=</span> gene_factors.shape[<span class="dv">0</span>] <span class="op">*</span> species_factors.shape[<span class="dv">0</span>] <span class="op">*</span> time_factors.shape[<span class="dv">0</span>]</span>
<span id="cb16-299"><a href="#cb16-299" aria-hidden="true" tabindex="-1"></a>    actual_size <span class="op">=</span> original_pivot.size</span>
<span id="cb16-300"><a href="#cb16-300" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-301"><a href="#cb16-301" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> actual_size <span class="op">==</span> expected_size:</span>
<span id="cb16-302"><a href="#cb16-302" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Reshape the pivot table to match tensor format</span></span>
<span id="cb16-303"><a href="#cb16-303" aria-hidden="true" tabindex="-1"></a>        n_genes, n_species, n_timepoints <span class="op">=</span> gene_factors.shape[<span class="dv">0</span>], species_factors.shape[<span class="dv">0</span>], time_factors.shape[<span class="dv">0</span>]</span>
<span id="cb16-304"><a href="#cb16-304" aria-hidden="true" tabindex="-1"></a>        original_reshaped <span class="op">=</span> original_pivot.values.reshape(n_genes, n_species, n_timepoints)</span>
<span id="cb16-305"><a href="#cb16-305" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-306"><a href="#cb16-306" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Calculate per-gene reconstruction error</span></span>
<span id="cb16-307"><a href="#cb16-307" aria-hidden="true" tabindex="-1"></a>        gene_errors <span class="op">=</span> np.mean(np.<span class="bu">abs</span>(original_reshaped <span class="op">-</span> reconstructed), axis<span class="op">=</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb16-308"><a href="#cb16-308" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-309"><a href="#cb16-309" aria-hidden="true" tabindex="-1"></a>        plt.hist(gene_errors, bins<span class="op">=</span><span class="dv">50</span>, alpha<span class="op">=</span><span class="fl">0.7</span>, edgecolor<span class="op">=</span><span class="st">'black'</span>)</span>
<span id="cb16-310"><a href="#cb16-310" aria-hidden="true" tabindex="-1"></a>        plt.xlabel(<span class="st">'Mean Absolute Error per Gene'</span>)</span>
<span id="cb16-311"><a href="#cb16-311" aria-hidden="true" tabindex="-1"></a>        plt.ylabel(<span class="st">'Number of Genes'</span>)</span>
<span id="cb16-312"><a href="#cb16-312" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="st">'Reconstruction Error Distribution'</span>)</span>
<span id="cb16-313"><a href="#cb16-313" aria-hidden="true" tabindex="-1"></a>        plt.axvline(np.mean(gene_errors), color<span class="op">=</span><span class="st">'red'</span>, linestyle<span class="op">=</span><span class="st">'--'</span>, </span>
<span id="cb16-314"><a href="#cb16-314" aria-hidden="true" tabindex="-1"></a>                   label<span class="op">=</span><span class="ss">f'Mean: </span><span class="sc">{</span>np<span class="sc">.</span>mean(gene_errors)<span class="sc">:.3f}</span><span class="ss">'</span>)</span>
<span id="cb16-315"><a href="#cb16-315" aria-hidden="true" tabindex="-1"></a>        plt.legend()</span>
<span id="cb16-316"><a href="#cb16-316" aria-hidden="true" tabindex="-1"></a>        plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb16-317"><a href="#cb16-317" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb16-318"><a href="#cb16-318" aria-hidden="true" tabindex="-1"></a>        <span class="co"># If reshape fails, show a simpler quality metric</span></span>
<span id="cb16-319"><a href="#cb16-319" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Calculate overall reconstruction quality using Frobenius norm</span></span>
<span id="cb16-320"><a href="#cb16-320" aria-hidden="true" tabindex="-1"></a>        original_flat <span class="op">=</span> original_pivot.values.flatten()</span>
<span id="cb16-321"><a href="#cb16-321" aria-hidden="true" tabindex="-1"></a>        reconstructed_flat <span class="op">=</span> reconstructed.flatten()</span>
<span id="cb16-322"><a href="#cb16-322" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-323"><a href="#cb16-323" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Ensure both arrays are the same size by taking the minimum</span></span>
<span id="cb16-324"><a href="#cb16-324" aria-hidden="true" tabindex="-1"></a>        min_size <span class="op">=</span> <span class="bu">min</span>(<span class="bu">len</span>(original_flat), <span class="bu">len</span>(reconstructed_flat))</span>
<span id="cb16-325"><a href="#cb16-325" aria-hidden="true" tabindex="-1"></a>        original_flat <span class="op">=</span> original_flat[:min_size]</span>
<span id="cb16-326"><a href="#cb16-326" aria-hidden="true" tabindex="-1"></a>        reconstructed_flat <span class="op">=</span> reconstructed_flat[:min_size]</span>
<span id="cb16-327"><a href="#cb16-327" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-328"><a href="#cb16-328" aria-hidden="true" tabindex="-1"></a>        residuals <span class="op">=</span> original_flat <span class="op">-</span> reconstructed_flat</span>
<span id="cb16-329"><a href="#cb16-329" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-330"><a href="#cb16-330" aria-hidden="true" tabindex="-1"></a>        plt.hist(residuals, bins<span class="op">=</span><span class="dv">50</span>, alpha<span class="op">=</span><span class="fl">0.7</span>, edgecolor<span class="op">=</span><span class="st">'black'</span>)</span>
<span id="cb16-331"><a href="#cb16-331" aria-hidden="true" tabindex="-1"></a>        plt.xlabel(<span class="st">'Reconstruction Residuals'</span>)</span>
<span id="cb16-332"><a href="#cb16-332" aria-hidden="true" tabindex="-1"></a>        plt.ylabel(<span class="st">'Frequency'</span>)</span>
<span id="cb16-333"><a href="#cb16-333" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="st">'Reconstruction Residuals Distribution'</span>)</span>
<span id="cb16-334"><a href="#cb16-334" aria-hidden="true" tabindex="-1"></a>        plt.axvline(np.mean(residuals), color<span class="op">=</span><span class="st">'red'</span>, linestyle<span class="op">=</span><span class="st">'--'</span>, </span>
<span id="cb16-335"><a href="#cb16-335" aria-hidden="true" tabindex="-1"></a>                   label<span class="op">=</span><span class="ss">f'Mean: </span><span class="sc">{</span>np<span class="sc">.</span>mean(residuals)<span class="sc">:.3f}</span><span class="ss">'</span>)</span>
<span id="cb16-336"><a href="#cb16-336" aria-hidden="true" tabindex="-1"></a>        plt.axvline(<span class="dv">0</span>, color<span class="op">=</span><span class="st">'black'</span>, linestyle<span class="op">=</span><span class="st">':'</span>, alpha<span class="op">=</span><span class="fl">0.5</span>, label<span class="op">=</span><span class="st">'Perfect fit'</span>)</span>
<span id="cb16-337"><a href="#cb16-337" aria-hidden="true" tabindex="-1"></a>        plt.legend()</span>
<span id="cb16-338"><a href="#cb16-338" aria-hidden="true" tabindex="-1"></a>        plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb16-339"><a href="#cb16-339" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-340"><a href="#cb16-340" aria-hidden="true" tabindex="-1"></a>    plt.tight_layout()</span>
<span id="cb16-341"><a href="#cb16-341" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-342"><a href="#cb16-342" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save the plot in main results directory</span></span>
<span id="cb16-343"><a href="#cb16-343" aria-hidden="true" tabindex="-1"></a>    plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"optimal_tensor_decomposition_visualization_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb16-344"><a href="#cb16-344" aria-hidden="true" tabindex="-1"></a>    plt.savefig(plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb16-345"><a href="#cb16-345" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Optimal visualization saved to: </span><span class="sc">{</span>plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb16-346"><a href="#cb16-346" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-347"><a href="#cb16-347" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb16-348"><a href="#cb16-348" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-349"><a href="#cb16-349" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Print interpretation guide</span></span>
<span id="cb16-350"><a href="#cb16-350" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">"</span> <span class="op">+</span> <span class="st">"="</span><span class="op">*</span><span class="dv">60</span>)</span>
<span id="cb16-351"><a href="#cb16-351" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"OPTIMAL VISUALIZATION INTERPRETATION GUIDE"</span>)</span>
<span id="cb16-352"><a href="#cb16-352" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"="</span><span class="op">*</span><span class="dv">60</span>)</span>
<span id="cb16-353"><a href="#cb16-353" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"1. Component Strengths: Shows which components capture most variation"</span>)</span>
<span id="cb16-354"><a href="#cb16-354" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"2. Temporal Patterns: How each component changes over time"</span>)</span>
<span id="cb16-355"><a href="#cb16-355" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"3. Species Patterns: How each individual loads on components"</span>)</span>
<span id="cb16-356"><a href="#cb16-356" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"4. Temporal Trajectories: Line plots of time patterns"</span>)</span>
<span id="cb16-357"><a href="#cb16-357" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"5. Species Clustering: PCA showing individual similarities"</span>)</span>
<span id="cb16-358"><a href="#cb16-358" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"6. Gene Variability: Distribution of gene response diversity"</span>)</span>
<span id="cb16-359"><a href="#cb16-359" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"7. Component Correlations: Independence of components"</span>)</span>
<span id="cb16-360"><a href="#cb16-360" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"8. Species Type Comparison: Average patterns by species type"</span>)</span>
<span id="cb16-361"><a href="#cb16-361" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"9. Temporal Variance: Which components show most time variation"</span>)</span>
<span id="cb16-362"><a href="#cb16-362" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"10. Species Variance: Which components show most individual variation"</span>)</span>
<span id="cb16-363"><a href="#cb16-363" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"11. Top Genes Heatmap: Key genes driving each component"</span>)</span>
<span id="cb16-364"><a href="#cb16-364" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"12. Reconstruction Quality: How well the optimal model fits the data"</span>)</span>
<span id="cb16-365"><a href="#cb16-365" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-366"><a href="#cb16-366" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb16-367"><a href="#cb16-367" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No optimal rank results found!"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/visualize-optimal-results-3.png" width="1920"></p>
<div class="sourceCode" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="individual-plot-1-component-strengths" class="level2">
<h2 class="anchored" data-anchor-id="individual-plot-1-component-strengths">4.3 Individual Plot 1: Component Strengths</h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the optimal rank from the rank selection summary</span></span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir:</span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>    summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>            rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> (from rank selection analysis)"</span>)</span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No rank selection summary found! Cannot determine optimal rank."</span>)</span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No main results directory found!"</span>)</span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for the optimal rank data in rank_comparison directory</span></span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a>latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir <span class="kw">and</span> optimal_rank:</span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb18-54"><a href="#cb18-54" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb18-55"><a href="#cb18-55" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb18-56"><a href="#cb18-56" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb18-57"><a href="#cb18-57" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found!"</span>)</span>
<span id="cb18-58"><a href="#cb18-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-59"><a href="#cb18-59" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb18-60"><a href="#cb18-60" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-61"><a href="#cb18-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load data</span></span>
<span id="cb18-62"><a href="#cb18-62" aria-hidden="true" tabindex="-1"></a>    gene_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb18-63"><a href="#cb18-63" aria-hidden="true" tabindex="-1"></a>    species_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"species_factors.npy"</span>))</span>
<span id="cb18-64"><a href="#cb18-64" aria-hidden="true" tabindex="-1"></a>    time_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"time_factors.npy"</span>))</span>
<span id="cb18-65"><a href="#cb18-65" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-66"><a href="#cb18-66" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate component strengths</span></span>
<span id="cb18-67"><a href="#cb18-67" aria-hidden="true" tabindex="-1"></a>    component_strengths <span class="op">=</span> []</span>
<span id="cb18-68"><a href="#cb18-68" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(gene_factors.shape[<span class="dv">1</span>]):</span>
<span id="cb18-69"><a href="#cb18-69" aria-hidden="true" tabindex="-1"></a>        gene_norm <span class="op">=</span> np.linalg.norm(gene_factors[:, i])</span>
<span id="cb18-70"><a href="#cb18-70" aria-hidden="true" tabindex="-1"></a>        species_norm <span class="op">=</span> np.linalg.norm(species_factors[:, i])</span>
<span id="cb18-71"><a href="#cb18-71" aria-hidden="true" tabindex="-1"></a>        time_norm <span class="op">=</span> np.linalg.norm(time_factors[:, i])</span>
<span id="cb18-72"><a href="#cb18-72" aria-hidden="true" tabindex="-1"></a>        total_strength <span class="op">=</span> gene_norm <span class="op">*</span> species_norm <span class="op">*</span> time_norm</span>
<span id="cb18-73"><a href="#cb18-73" aria-hidden="true" tabindex="-1"></a>        component_strengths.append(total_strength)</span>
<span id="cb18-74"><a href="#cb18-74" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-75"><a href="#cb18-75" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Sort components by strength</span></span>
<span id="cb18-76"><a href="#cb18-76" aria-hidden="true" tabindex="-1"></a>    sorted_components <span class="op">=</span> np.argsort(component_strengths)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb18-77"><a href="#cb18-77" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-78"><a href="#cb18-78" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create plot</span></span>
<span id="cb18-79"><a href="#cb18-79" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">12</span>, <span class="dv">6</span>))</span>
<span id="cb18-80"><a href="#cb18-80" aria-hidden="true" tabindex="-1"></a>    plt.bar(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(component_strengths) <span class="op">+</span> <span class="dv">1</span>), </span>
<span id="cb18-81"><a href="#cb18-81" aria-hidden="true" tabindex="-1"></a>            [component_strengths[i] <span class="cf">for</span> i <span class="kw">in</span> sorted_components])</span>
<span id="cb18-82"><a href="#cb18-82" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Component (ranked by strength)'</span>)</span>
<span id="cb18-83"><a href="#cb18-83" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Component Strength'</span>)</span>
<span id="cb18-84"><a href="#cb18-84" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Component Strengths (Optimal Rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb18-85"><a href="#cb18-85" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-86"><a href="#cb18-86" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set x-ticks to show all components if &lt;= 10, otherwise show every 2nd or 5th</span></span>
<span id="cb18-87"><a href="#cb18-87" aria-hidden="true" tabindex="-1"></a>    n_components <span class="op">=</span> <span class="bu">len</span>(component_strengths)</span>
<span id="cb18-88"><a href="#cb18-88" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> n_components <span class="op">&lt;=</span> <span class="dv">10</span>:</span>
<span id="cb18-89"><a href="#cb18-89" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>))</span>
<span id="cb18-90"><a href="#cb18-90" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> n_components <span class="op">&lt;=</span> <span class="dv">20</span>:</span>
<span id="cb18-91"><a href="#cb18-91" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb18-92"><a href="#cb18-92" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb18-93"><a href="#cb18-93" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>, <span class="dv">5</span>))</span>
<span id="cb18-94"><a href="#cb18-94" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-95"><a href="#cb18-95" aria-hidden="true" tabindex="-1"></a>    plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb18-96"><a href="#cb18-96" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-97"><a href="#cb18-97" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save individual plot in main results directory</span></span>
<span id="cb18-98"><a href="#cb18-98" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_results_dir:</span>
<span id="cb18-99"><a href="#cb18-99" aria-hidden="true" tabindex="-1"></a>        plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"plot_01_component_strengths_optimal_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb18-100"><a href="#cb18-100" aria-hidden="true" tabindex="-1"></a>        plt.savefig(plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb18-101"><a href="#cb18-101" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Component strengths plot saved to: </span><span class="sc">{</span>plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb18-102"><a href="#cb18-102" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-103"><a href="#cb18-103" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb18-104"><a href="#cb18-104" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb18-105"><a href="#cb18-105" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Could not load data for optimal rank </span><span class="sc">{</span>optimal_rank <span class="cf">if</span> optimal_rank <span class="cf">else</span> <span class="st">'unknown'</span><span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/plot-component-strengths-5.png" width="1152"></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="individual-plot-2-temporal-patterns-heatmap" class="level2">
<h2 class="anchored" data-anchor-id="individual-plot-2-temporal-patterns-heatmap">4.4 Individual Plot 2: Temporal Patterns Heatmap</h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory</span></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the optimal rank from the rank selection summary</span></span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir:</span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a>    summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a>            rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> (from rank selection analysis)"</span>)</span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No rank selection summary found! Cannot determine optimal rank."</span>)</span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb20-38"><a href="#cb20-38" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb20-39"><a href="#cb20-39" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb20-40"><a href="#cb20-40" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No main results directory found!"</span>)</span>
<span id="cb20-41"><a href="#cb20-41" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb20-42"><a href="#cb20-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-43"><a href="#cb20-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for the optimal rank data in rank_comparison directory</span></span>
<span id="cb20-44"><a href="#cb20-44" aria-hidden="true" tabindex="-1"></a>latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb20-45"><a href="#cb20-45" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir <span class="kw">and</span> optimal_rank:</span>
<span id="cb20-46"><a href="#cb20-46" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb20-47"><a href="#cb20-47" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb20-48"><a href="#cb20-48" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb20-49"><a href="#cb20-49" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb20-50"><a href="#cb20-50" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb20-51"><a href="#cb20-51" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb20-52"><a href="#cb20-52" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb20-53"><a href="#cb20-53" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb20-54"><a href="#cb20-54" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb20-55"><a href="#cb20-55" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb20-56"><a href="#cb20-56" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb20-57"><a href="#cb20-57" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb20-58"><a href="#cb20-58" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found!"</span>)</span>
<span id="cb20-59"><a href="#cb20-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-60"><a href="#cb20-60" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb20-61"><a href="#cb20-61" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb20-62"><a href="#cb20-62" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load data</span></span>
<span id="cb20-63"><a href="#cb20-63" aria-hidden="true" tabindex="-1"></a>    gene_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb20-64"><a href="#cb20-64" aria-hidden="true" tabindex="-1"></a>    species_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"species_factors.npy"</span>))</span>
<span id="cb20-65"><a href="#cb20-65" aria-hidden="true" tabindex="-1"></a>    time_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"time_factors.npy"</span>))</span>
<span id="cb20-66"><a href="#cb20-66" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb20-67"><a href="#cb20-67" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load the transformed data to get timepoint names</span></span>
<span id="cb20-68"><a href="#cb20-68" aria-hidden="true" tabindex="-1"></a>    data_file <span class="op">=</span> os.path.join(output_dir, <span class="st">"vst_counts_matrix_long_format.csv"</span>)</span>
<span id="cb20-69"><a href="#cb20-69" aria-hidden="true" tabindex="-1"></a>    df_long <span class="op">=</span> pd.read_csv(data_file)</span>
<span id="cb20-70"><a href="#cb20-70" aria-hidden="true" tabindex="-1"></a>    timepoint_names <span class="op">=</span> <span class="bu">sorted</span>(df_long[<span class="st">'timepoint'</span>].unique())</span>
<span id="cb20-71"><a href="#cb20-71" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb20-72"><a href="#cb20-72" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate component strengths and get top 5</span></span>
<span id="cb20-73"><a href="#cb20-73" aria-hidden="true" tabindex="-1"></a>    component_strengths <span class="op">=</span> []</span>
<span id="cb20-74"><a href="#cb20-74" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(gene_factors.shape[<span class="dv">1</span>]):</span>
<span id="cb20-75"><a href="#cb20-75" aria-hidden="true" tabindex="-1"></a>        gene_norm <span class="op">=</span> np.linalg.norm(gene_factors[:, i])</span>
<span id="cb20-76"><a href="#cb20-76" aria-hidden="true" tabindex="-1"></a>        species_norm <span class="op">=</span> np.linalg.norm(species_factors[:, i])</span>
<span id="cb20-77"><a href="#cb20-77" aria-hidden="true" tabindex="-1"></a>        time_norm <span class="op">=</span> np.linalg.norm(time_factors[:, i])</span>
<span id="cb20-78"><a href="#cb20-78" aria-hidden="true" tabindex="-1"></a>        total_strength <span class="op">=</span> gene_norm <span class="op">*</span> species_norm <span class="op">*</span> time_norm</span>
<span id="cb20-79"><a href="#cb20-79" aria-hidden="true" tabindex="-1"></a>        component_strengths.append(total_strength)</span>
<span id="cb20-80"><a href="#cb20-80" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb20-81"><a href="#cb20-81" aria-hidden="true" tabindex="-1"></a>    sorted_components <span class="op">=</span> np.argsort(component_strengths)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb20-82"><a href="#cb20-82" aria-hidden="true" tabindex="-1"></a>    top_5_components <span class="op">=</span> sorted_components[:<span class="bu">min</span>(<span class="dv">5</span>, optimal_rank)]  <span class="co"># Use min to handle ranks &lt; 5</span></span>
<span id="cb20-83"><a href="#cb20-83" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb20-84"><a href="#cb20-84" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create heatmap</span></span>
<span id="cb20-85"><a href="#cb20-85" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">6</span>))</span>
<span id="cb20-86"><a href="#cb20-86" aria-hidden="true" tabindex="-1"></a>    time_data <span class="op">=</span> time_factors[:, top_5_components]</span>
<span id="cb20-87"><a href="#cb20-87" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add a labeled colorbar so readers know what the colors represent</span></span>
<span id="cb20-88"><a href="#cb20-88" aria-hidden="true" tabindex="-1"></a>    sns.heatmap(time_data.T, </span>
<span id="cb20-89"><a href="#cb20-89" aria-hidden="true" tabindex="-1"></a>                xticklabels<span class="op">=</span>timepoint_names,</span>
<span id="cb20-90"><a href="#cb20-90" aria-hidden="true" tabindex="-1"></a>                yticklabels<span class="op">=</span>[<span class="ss">f'Component </span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> top_5_components],</span>
<span id="cb20-91"><a href="#cb20-91" aria-hidden="true" tabindex="-1"></a>                cmap<span class="op">=</span><span class="st">'RdBu_r'</span>, center<span class="op">=</span><span class="dv">0</span>, annot<span class="op">=</span><span class="va">True</span>, fmt<span class="op">=</span><span class="st">'.3f'</span>,</span>
<span id="cb20-92"><a href="#cb20-92" aria-hidden="true" tabindex="-1"></a>                cbar<span class="op">=</span><span class="va">True</span>, cbar_kws<span class="op">=</span>{<span class="st">'label'</span>: <span class="st">'Time factor value (loading)'</span>} ,</span>
<span id="cb20-93"><a href="#cb20-93" aria-hidden="true" tabindex="-1"></a>                linewidths<span class="op">=</span><span class="fl">0.5</span>)</span>
<span id="cb20-94"><a href="#cb20-94" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Top </span><span class="sc">{</span><span class="bu">len</span>(top_5_components)<span class="sc">}</span><span class="ss"> Components: Temporal Patterns (Optimal Rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb20-95"><a href="#cb20-95" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Timepoint'</span>)</span>
<span id="cb20-96"><a href="#cb20-96" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Component'</span>)</span>
<span id="cb20-97"><a href="#cb20-97" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb20-98"><a href="#cb20-98" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save individual plot in main results directory</span></span>
<span id="cb20-99"><a href="#cb20-99" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_results_dir:</span>
<span id="cb20-100"><a href="#cb20-100" aria-hidden="true" tabindex="-1"></a>        plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"plot_02_temporal_heatmap_optimal_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb20-101"><a href="#cb20-101" aria-hidden="true" tabindex="-1"></a>        plt.savefig(plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb20-102"><a href="#cb20-102" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Temporal patterns heatmap saved to: </span><span class="sc">{</span>plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb20-103"><a href="#cb20-103" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb20-104"><a href="#cb20-104" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb20-105"><a href="#cb20-105" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb20-106"><a href="#cb20-106" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Could not load data for optimal rank </span><span class="sc">{</span>optimal_rank <span class="cf">if</span> optimal_rank <span class="cf">else</span> <span class="st">'unknown'</span><span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/plot-temporal-heatmap-7.png" width="960"></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="individual-plot-3-species-patterns-heatmap" class="level2">
<h2 class="anchored" data-anchor-id="individual-plot-3-species-patterns-heatmap">4.5 Individual Plot 3: Species Patterns Heatmap</h2>
<div class="sourceCode" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the optimal rank from the rank selection summary</span></span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir:</span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>    summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a>            rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> (from rank selection analysis)"</span>)</span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No rank selection summary found! Cannot determine optimal rank."</span>)</span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No main results directory found!"</span>)</span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for the optimal rank data in rank_comparison directory</span></span>
<span id="cb22-44"><a href="#cb22-44" aria-hidden="true" tabindex="-1"></a>latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb22-45"><a href="#cb22-45" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir <span class="kw">and</span> optimal_rank:</span>
<span id="cb22-46"><a href="#cb22-46" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb22-47"><a href="#cb22-47" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb22-48"><a href="#cb22-48" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb22-49"><a href="#cb22-49" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb22-50"><a href="#cb22-50" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb22-51"><a href="#cb22-51" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb22-52"><a href="#cb22-52" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb22-53"><a href="#cb22-53" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb22-54"><a href="#cb22-54" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb22-55"><a href="#cb22-55" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb22-56"><a href="#cb22-56" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb22-57"><a href="#cb22-57" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb22-58"><a href="#cb22-58" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found!"</span>)</span>
<span id="cb22-59"><a href="#cb22-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-60"><a href="#cb22-60" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb22-61"><a href="#cb22-61" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb22-62"><a href="#cb22-62" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load data</span></span>
<span id="cb22-63"><a href="#cb22-63" aria-hidden="true" tabindex="-1"></a>    gene_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb22-64"><a href="#cb22-64" aria-hidden="true" tabindex="-1"></a>    species_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"species_factors.npy"</span>))</span>
<span id="cb22-65"><a href="#cb22-65" aria-hidden="true" tabindex="-1"></a>    time_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"time_factors.npy"</span>))</span>
<span id="cb22-66"><a href="#cb22-66" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb22-67"><a href="#cb22-67" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load the transformed data to get species names</span></span>
<span id="cb22-68"><a href="#cb22-68" aria-hidden="true" tabindex="-1"></a>    data_file <span class="op">=</span> os.path.join(output_dir, <span class="st">"vst_counts_matrix_long_format.csv"</span>)</span>
<span id="cb22-69"><a href="#cb22-69" aria-hidden="true" tabindex="-1"></a>    df_long <span class="op">=</span> pd.read_csv(data_file)</span>
<span id="cb22-70"><a href="#cb22-70" aria-hidden="true" tabindex="-1"></a>    species_names <span class="op">=</span> <span class="bu">sorted</span>(df_long[<span class="st">'species'</span>].unique())</span>
<span id="cb22-71"><a href="#cb22-71" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb22-72"><a href="#cb22-72" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate component strengths and get top 5</span></span>
<span id="cb22-73"><a href="#cb22-73" aria-hidden="true" tabindex="-1"></a>    component_strengths <span class="op">=</span> []</span>
<span id="cb22-74"><a href="#cb22-74" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(gene_factors.shape[<span class="dv">1</span>]):</span>
<span id="cb22-75"><a href="#cb22-75" aria-hidden="true" tabindex="-1"></a>        gene_norm <span class="op">=</span> np.linalg.norm(gene_factors[:, i])</span>
<span id="cb22-76"><a href="#cb22-76" aria-hidden="true" tabindex="-1"></a>        species_norm <span class="op">=</span> np.linalg.norm(species_factors[:, i])</span>
<span id="cb22-77"><a href="#cb22-77" aria-hidden="true" tabindex="-1"></a>        time_norm <span class="op">=</span> np.linalg.norm(time_factors[:, i])</span>
<span id="cb22-78"><a href="#cb22-78" aria-hidden="true" tabindex="-1"></a>        total_strength <span class="op">=</span> gene_norm <span class="op">*</span> species_norm <span class="op">*</span> time_norm</span>
<span id="cb22-79"><a href="#cb22-79" aria-hidden="true" tabindex="-1"></a>        component_strengths.append(total_strength)</span>
<span id="cb22-80"><a href="#cb22-80" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb22-81"><a href="#cb22-81" aria-hidden="true" tabindex="-1"></a>    sorted_components <span class="op">=</span> np.argsort(component_strengths)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb22-82"><a href="#cb22-82" aria-hidden="true" tabindex="-1"></a>    top_5_components <span class="op">=</span> sorted_components[:<span class="bu">min</span>(<span class="dv">5</span>, optimal_rank)]  <span class="co"># Use min to handle ranks &lt; 5</span></span>
<span id="cb22-83"><a href="#cb22-83" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb22-84"><a href="#cb22-84" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create heatmap</span></span>
<span id="cb22-85"><a href="#cb22-85" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">16</span>, <span class="dv">6</span>))</span>
<span id="cb22-86"><a href="#cb22-86" aria-hidden="true" tabindex="-1"></a>    species_data <span class="op">=</span> species_factors[:, top_5_components]</span>
<span id="cb22-87"><a href="#cb22-87" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb22-88"><a href="#cb22-88" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add a labeled colorbar so readers know what the colors represent</span></span>
<span id="cb22-89"><a href="#cb22-89" aria-hidden="true" tabindex="-1"></a>    sns.heatmap(species_data.T,</span>
<span id="cb22-90"><a href="#cb22-90" aria-hidden="true" tabindex="-1"></a>                xticklabels<span class="op">=</span>[<span class="ss">f"</span><span class="sc">{</span>name[:<span class="dv">8</span>]<span class="sc">}</span><span class="ss">..."</span> <span class="cf">if</span> <span class="bu">len</span>(name) <span class="op">&gt;</span> <span class="dv">8</span> <span class="cf">else</span> name <span class="cf">for</span> name <span class="kw">in</span> species_names],</span>
<span id="cb22-91"><a href="#cb22-91" aria-hidden="true" tabindex="-1"></a>                yticklabels<span class="op">=</span>[<span class="ss">f'Component </span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> top_5_components],</span>
<span id="cb22-92"><a href="#cb22-92" aria-hidden="true" tabindex="-1"></a>                cmap<span class="op">=</span><span class="st">'RdBu_r'</span>, center<span class="op">=</span><span class="dv">0</span>,</span>
<span id="cb22-93"><a href="#cb22-93" aria-hidden="true" tabindex="-1"></a>                cbar<span class="op">=</span><span class="va">True</span>, cbar_kws<span class="op">=</span>{<span class="st">'label'</span>: <span class="st">'Species factor value (loading)'</span>},</span>
<span id="cb22-94"><a href="#cb22-94" aria-hidden="true" tabindex="-1"></a>                linewidths<span class="op">=</span><span class="fl">0.5</span>)</span>
<span id="cb22-95"><a href="#cb22-95" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Top </span><span class="sc">{</span><span class="bu">len</span>(top_5_components)<span class="sc">}</span><span class="ss"> Components: Species Patterns (Optimal Rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb22-96"><a href="#cb22-96" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Individual'</span>)</span>
<span id="cb22-97"><a href="#cb22-97" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Component'</span>)</span>
<span id="cb22-98"><a href="#cb22-98" aria-hidden="true" tabindex="-1"></a>    plt.xticks(rotation<span class="op">=</span><span class="dv">90</span>)</span>
<span id="cb22-99"><a href="#cb22-99" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb22-100"><a href="#cb22-100" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save individual plot in main results directory</span></span>
<span id="cb22-101"><a href="#cb22-101" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_results_dir:</span>
<span id="cb22-102"><a href="#cb22-102" aria-hidden="true" tabindex="-1"></a>        plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"plot_03_species_heatmap_optimal_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb22-103"><a href="#cb22-103" aria-hidden="true" tabindex="-1"></a>        plt.savefig(plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb22-104"><a href="#cb22-104" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Species patterns heatmap saved to: </span><span class="sc">{</span>plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb22-105"><a href="#cb22-105" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb22-106"><a href="#cb22-106" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb22-107"><a href="#cb22-107" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb22-108"><a href="#cb22-108" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Could not load data for optimal rank </span><span class="sc">{</span>optimal_rank <span class="cf">if</span> optimal_rank <span class="cf">else</span> <span class="st">'unknown'</span><span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/plot-species-heatmap-9.png" width="1536"></p>
<div class="sourceCode" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="individual-plot-4-temporal-trajectories" class="level2">
<h2 class="anchored" data-anchor-id="individual-plot-4-temporal-trajectories">4.6 Individual Plot 4: Temporal Trajectories</h2>
<div class="sourceCode" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the optimal rank from the rank selection summary</span></span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir:</span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>    summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>            rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> (from rank selection analysis)"</span>)</span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No rank selection summary found! Cannot determine optimal rank."</span>)</span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No main results directory found!"</span>)</span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for the optimal rank data in rank_comparison directory</span></span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir <span class="kw">and</span> optimal_rank:</span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb24-54"><a href="#cb24-54" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb24-55"><a href="#cb24-55" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb24-56"><a href="#cb24-56" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb24-57"><a href="#cb24-57" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found!"</span>)</span>
<span id="cb24-58"><a href="#cb24-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-59"><a href="#cb24-59" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb24-60"><a href="#cb24-60" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load data</span></span>
<span id="cb24-61"><a href="#cb24-61" aria-hidden="true" tabindex="-1"></a>    gene_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb24-62"><a href="#cb24-62" aria-hidden="true" tabindex="-1"></a>    species_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"species_factors.npy"</span>))</span>
<span id="cb24-63"><a href="#cb24-63" aria-hidden="true" tabindex="-1"></a>    time_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"time_factors.npy"</span>))</span>
<span id="cb24-64"><a href="#cb24-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-65"><a href="#cb24-65" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load the transformed data to get timepoint names</span></span>
<span id="cb24-66"><a href="#cb24-66" aria-hidden="true" tabindex="-1"></a>    data_file <span class="op">=</span> os.path.join(output_dir, <span class="st">"vst_counts_matrix_long_format.csv"</span>)</span>
<span id="cb24-67"><a href="#cb24-67" aria-hidden="true" tabindex="-1"></a>    df_long <span class="op">=</span> pd.read_csv(data_file)</span>
<span id="cb24-68"><a href="#cb24-68" aria-hidden="true" tabindex="-1"></a>    timepoint_names <span class="op">=</span> <span class="bu">sorted</span>(df_long[<span class="st">'timepoint'</span>].unique())</span>
<span id="cb24-69"><a href="#cb24-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-70"><a href="#cb24-70" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate component strengths and get top 5</span></span>
<span id="cb24-71"><a href="#cb24-71" aria-hidden="true" tabindex="-1"></a>    component_strengths <span class="op">=</span> []</span>
<span id="cb24-72"><a href="#cb24-72" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(gene_factors.shape[<span class="dv">1</span>]):</span>
<span id="cb24-73"><a href="#cb24-73" aria-hidden="true" tabindex="-1"></a>        gene_norm <span class="op">=</span> np.linalg.norm(gene_factors[:, i])</span>
<span id="cb24-74"><a href="#cb24-74" aria-hidden="true" tabindex="-1"></a>        species_norm <span class="op">=</span> np.linalg.norm(species_factors[:, i])</span>
<span id="cb24-75"><a href="#cb24-75" aria-hidden="true" tabindex="-1"></a>        time_norm <span class="op">=</span> np.linalg.norm(time_factors[:, i])</span>
<span id="cb24-76"><a href="#cb24-76" aria-hidden="true" tabindex="-1"></a>        total_strength <span class="op">=</span> gene_norm <span class="op">*</span> species_norm <span class="op">*</span> time_norm</span>
<span id="cb24-77"><a href="#cb24-77" aria-hidden="true" tabindex="-1"></a>        component_strengths.append(total_strength)</span>
<span id="cb24-78"><a href="#cb24-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-79"><a href="#cb24-79" aria-hidden="true" tabindex="-1"></a>    sorted_components <span class="op">=</span> np.argsort(component_strengths)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb24-80"><a href="#cb24-80" aria-hidden="true" tabindex="-1"></a>    top_5_components <span class="op">=</span> sorted_components[:<span class="bu">min</span>(<span class="dv">5</span>, optimal_rank)]  <span class="co"># Use min to handle ranks &lt; 5</span></span>
<span id="cb24-81"><a href="#cb24-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-82"><a href="#cb24-82" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create line plot</span></span>
<span id="cb24-83"><a href="#cb24-83" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">12</span>, <span class="dv">6</span>))</span>
<span id="cb24-84"><a href="#cb24-84" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i, comp_idx <span class="kw">in</span> <span class="bu">enumerate</span>(top_5_components):</span>
<span id="cb24-85"><a href="#cb24-85" aria-hidden="true" tabindex="-1"></a>        plt.plot(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(timepoint_names) <span class="op">+</span> <span class="dv">1</span>), </span>
<span id="cb24-86"><a href="#cb24-86" aria-hidden="true" tabindex="-1"></a>                time_factors[:, comp_idx], </span>
<span id="cb24-87"><a href="#cb24-87" aria-hidden="true" tabindex="-1"></a>                marker<span class="op">=</span><span class="st">'o'</span>, linewidth<span class="op">=</span><span class="dv">2</span>, label<span class="op">=</span><span class="ss">f'Component </span><span class="sc">{</span>comp_idx <span class="op">+</span> <span class="dv">1</span><span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb24-88"><a href="#cb24-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-89"><a href="#cb24-89" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Timepoint'</span>)</span>
<span id="cb24-90"><a href="#cb24-90" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Factor Value'</span>)</span>
<span id="cb24-91"><a href="#cb24-91" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Temporal Trajectories (Top </span><span class="sc">{</span><span class="bu">len</span>(top_5_components)<span class="sc">}</span><span class="ss"> Components, Optimal Rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb24-92"><a href="#cb24-92" aria-hidden="true" tabindex="-1"></a>    plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(timepoint_names) <span class="op">+</span> <span class="dv">1</span>), timepoint_names)</span>
<span id="cb24-93"><a href="#cb24-93" aria-hidden="true" tabindex="-1"></a>    plt.legend()</span>
<span id="cb24-94"><a href="#cb24-94" aria-hidden="true" tabindex="-1"></a>    plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb24-95"><a href="#cb24-95" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb24-96"><a href="#cb24-96" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save individual plot in main results directory</span></span>
<span id="cb24-97"><a href="#cb24-97" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_results_dir:</span>
<span id="cb24-98"><a href="#cb24-98" aria-hidden="true" tabindex="-1"></a>        plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"plot_04_temporal_trajectories_optimal_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb24-99"><a href="#cb24-99" aria-hidden="true" tabindex="-1"></a>        plt.savefig(plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb24-100"><a href="#cb24-100" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Temporal trajectories plot saved to: </span><span class="sc">{</span>plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb24-101"><a href="#cb24-101" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb24-102"><a href="#cb24-102" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb24-103"><a href="#cb24-103" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb24-104"><a href="#cb24-104" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Could not load data for optimal rank </span><span class="sc">{</span>optimal_rank <span class="cf">if</span> optimal_rank <span class="cf">else</span> <span class="st">'unknown'</span><span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/plot-temporal-trajectories-11.png" width="1152"></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="individual-plot-5-species-clustering-pca" class="level2">
<h2 class="anchored" data-anchor-id="individual-plot-5-species-clustering-pca">4.7 Individual Plot 5: Species Clustering PCA</h2>
<div class="sourceCode" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> sklearn.cluster <span class="im">import</span> KMeans</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> sklearn.decomposition <span class="im">import</span> PCA</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting</span></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory</span></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the optimal rank from the rank selection summary</span></span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir:</span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>    summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>            rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> (from rank selection analysis)"</span>)</span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No rank selection summary found! Cannot determine optimal rank."</span>)</span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No main results directory found!"</span>)</span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for the optimal rank data in rank_comparison directory</span></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a>latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb26-46"><a href="#cb26-46" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir <span class="kw">and</span> optimal_rank:</span>
<span id="cb26-47"><a href="#cb26-47" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb26-48"><a href="#cb26-48" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb26-49"><a href="#cb26-49" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-50"><a href="#cb26-50" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb26-51"><a href="#cb26-51" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb26-52"><a href="#cb26-52" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb26-53"><a href="#cb26-53" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb26-54"><a href="#cb26-54" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb26-55"><a href="#cb26-55" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb26-56"><a href="#cb26-56" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb26-57"><a href="#cb26-57" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb26-58"><a href="#cb26-58" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb26-59"><a href="#cb26-59" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found!"</span>)</span>
<span id="cb26-60"><a href="#cb26-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-61"><a href="#cb26-61" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb26-62"><a href="#cb26-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-63"><a href="#cb26-63" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load data</span></span>
<span id="cb26-64"><a href="#cb26-64" aria-hidden="true" tabindex="-1"></a>    gene_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb26-65"><a href="#cb26-65" aria-hidden="true" tabindex="-1"></a>    species_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"species_factors.npy"</span>))</span>
<span id="cb26-66"><a href="#cb26-66" aria-hidden="true" tabindex="-1"></a>    time_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"time_factors.npy"</span>))</span>
<span id="cb26-67"><a href="#cb26-67" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-68"><a href="#cb26-68" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load the transformed data to get species names</span></span>
<span id="cb26-69"><a href="#cb26-69" aria-hidden="true" tabindex="-1"></a>    data_file <span class="op">=</span> os.path.join(output_dir, <span class="st">"vst_counts_matrix_long_format.csv"</span>)</span>
<span id="cb26-70"><a href="#cb26-70" aria-hidden="true" tabindex="-1"></a>    df_long <span class="op">=</span> pd.read_csv(data_file)</span>
<span id="cb26-71"><a href="#cb26-71" aria-hidden="true" tabindex="-1"></a>    species_names <span class="op">=</span> <span class="bu">sorted</span>(df_long[<span class="st">'species'</span>].unique())</span>
<span id="cb26-72"><a href="#cb26-72" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-73"><a href="#cb26-73" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create species type labels (same as comprehensive visualization)</span></span>
<span id="cb26-74"><a href="#cb26-74" aria-hidden="true" tabindex="-1"></a>    species_types <span class="op">=</span> [name.split(<span class="st">'-'</span>)[<span class="dv">0</span>] <span class="cf">for</span> name <span class="kw">in</span> species_names]</span>
<span id="cb26-75"><a href="#cb26-75" aria-hidden="true" tabindex="-1"></a>    unique_types <span class="op">=</span> <span class="bu">list</span>(<span class="bu">set</span>(species_types))</span>
<span id="cb26-76"><a href="#cb26-76" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-77"><a href="#cb26-77" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate component strengths (same as comprehensive visualization)</span></span>
<span id="cb26-78"><a href="#cb26-78" aria-hidden="true" tabindex="-1"></a>    component_strengths <span class="op">=</span> []</span>
<span id="cb26-79"><a href="#cb26-79" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(gene_factors.shape[<span class="dv">1</span>]):</span>
<span id="cb26-80"><a href="#cb26-80" aria-hidden="true" tabindex="-1"></a>        gene_norm <span class="op">=</span> np.linalg.norm(gene_factors[:, i])</span>
<span id="cb26-81"><a href="#cb26-81" aria-hidden="true" tabindex="-1"></a>        species_norm <span class="op">=</span> np.linalg.norm(species_factors[:, i])</span>
<span id="cb26-82"><a href="#cb26-82" aria-hidden="true" tabindex="-1"></a>        time_norm <span class="op">=</span> np.linalg.norm(time_factors[:, i])</span>
<span id="cb26-83"><a href="#cb26-83" aria-hidden="true" tabindex="-1"></a>        total_strength <span class="op">=</span> gene_norm <span class="op">*</span> species_norm <span class="op">*</span> time_norm</span>
<span id="cb26-84"><a href="#cb26-84" aria-hidden="true" tabindex="-1"></a>        component_strengths.append(total_strength)</span>
<span id="cb26-85"><a href="#cb26-85" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-86"><a href="#cb26-86" aria-hidden="true" tabindex="-1"></a>    sorted_components <span class="op">=</span> np.argsort(component_strengths)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb26-87"><a href="#cb26-87" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-88"><a href="#cb26-88" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use the same logic as comprehensive visualization</span></span>
<span id="cb26-89"><a href="#cb26-89" aria-hidden="true" tabindex="-1"></a>    rank <span class="op">=</span> gene_factors.shape[<span class="dv">1</span>]  <span class="co"># This is the actual rank (number of components)</span></span>
<span id="cb26-90"><a href="#cb26-90" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-91"><a href="#cb26-91" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> rank <span class="op">&gt;=</span> <span class="dv">2</span>:</span>
<span id="cb26-92"><a href="#cb26-92" aria-hidden="true" tabindex="-1"></a>        n_pca_components <span class="op">=</span> <span class="bu">min</span>(rank, <span class="dv">10</span>)</span>
<span id="cb26-93"><a href="#cb26-93" aria-hidden="true" tabindex="-1"></a>        species_subset <span class="op">=</span> species_factors[:, sorted_components[:n_pca_components]]</span>
<span id="cb26-94"><a href="#cb26-94" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb26-95"><a href="#cb26-95" aria-hidden="true" tabindex="-1"></a>        pca <span class="op">=</span> PCA(n_components<span class="op">=</span><span class="dv">2</span>)</span>
<span id="cb26-96"><a href="#cb26-96" aria-hidden="true" tabindex="-1"></a>        species_pca <span class="op">=</span> pca.fit_transform(species_subset)</span>
<span id="cb26-97"><a href="#cb26-97" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb26-98"><a href="#cb26-98" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Create plot (same styling as comprehensive visualization)</span></span>
<span id="cb26-99"><a href="#cb26-99" aria-hidden="true" tabindex="-1"></a>        plt.figure(figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">8</span>))</span>
<span id="cb26-100"><a href="#cb26-100" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb26-101"><a href="#cb26-101" aria-hidden="true" tabindex="-1"></a>        colors <span class="op">=</span> [<span class="st">'red'</span>, <span class="st">'blue'</span>, <span class="st">'green'</span>, <span class="st">'orange'</span>, <span class="st">'purple'</span>][:<span class="bu">len</span>(unique_types)]</span>
<span id="cb26-102"><a href="#cb26-102" aria-hidden="true" tabindex="-1"></a>        species_type_map <span class="op">=</span> {stype: i <span class="cf">for</span> i, stype <span class="kw">in</span> <span class="bu">enumerate</span>(unique_types)}</span>
<span id="cb26-103"><a href="#cb26-103" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb26-104"><a href="#cb26-104" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> i, (x, y) <span class="kw">in</span> <span class="bu">enumerate</span>(species_pca):</span>
<span id="cb26-105"><a href="#cb26-105" aria-hidden="true" tabindex="-1"></a>            stype <span class="op">=</span> species_types[i]</span>
<span id="cb26-106"><a href="#cb26-106" aria-hidden="true" tabindex="-1"></a>            plt.scatter(x, y, c<span class="op">=</span>colors[species_type_map[stype]], </span>
<span id="cb26-107"><a href="#cb26-107" aria-hidden="true" tabindex="-1"></a>                       alpha<span class="op">=</span><span class="fl">0.7</span>, s<span class="op">=</span><span class="dv">40</span>, </span>
<span id="cb26-108"><a href="#cb26-108" aria-hidden="true" tabindex="-1"></a>                       label<span class="op">=</span>stype <span class="cf">if</span> i <span class="op">==</span> species_types.index(stype) <span class="cf">else</span> <span class="st">""</span>)</span>
<span id="cb26-109"><a href="#cb26-109" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb26-110"><a href="#cb26-110" aria-hidden="true" tabindex="-1"></a>        plt.xlabel(<span class="ss">f'PC1 (</span><span class="sc">{</span>pca<span class="sc">.</span>explained_variance_ratio_[<span class="dv">0</span>]<span class="sc">:.1%}</span><span class="ss">)'</span>)</span>
<span id="cb26-111"><a href="#cb26-111" aria-hidden="true" tabindex="-1"></a>        plt.ylabel(<span class="ss">f'PC2 (</span><span class="sc">{</span>pca<span class="sc">.</span>explained_variance_ratio_[<span class="dv">1</span>]<span class="sc">:.1%}</span><span class="ss">)'</span>)</span>
<span id="cb26-112"><a href="#cb26-112" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="ss">f'Species PCA (Optimal Rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb26-113"><a href="#cb26-113" aria-hidden="true" tabindex="-1"></a>        plt.legend()</span>
<span id="cb26-114"><a href="#cb26-114" aria-hidden="true" tabindex="-1"></a>        plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb26-115"><a href="#cb26-115" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb26-116"><a href="#cb26-116" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Save to main results directory for consistency</span></span>
<span id="cb26-117"><a href="#cb26-117" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> main_results_dir:</span>
<span id="cb26-118"><a href="#cb26-118" aria-hidden="true" tabindex="-1"></a>            plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"plot_05_species_clustering_optimal_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb26-119"><a href="#cb26-119" aria-hidden="true" tabindex="-1"></a>            plt.savefig(plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb26-120"><a href="#cb26-120" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Species clustering PCA plot saved to: </span><span class="sc">{</span>plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb26-121"><a href="#cb26-121" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb26-122"><a href="#cb26-122" aria-hidden="true" tabindex="-1"></a>        plt.show()</span>
<span id="cb26-123"><a href="#cb26-123" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb26-124"><a href="#cb26-124" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> has insufficient components for PCA visualization"</span>)</span>
<span id="cb26-125"><a href="#cb26-125" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb26-126"><a href="#cb26-126" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Could not load data for optimal rank </span><span class="sc">{</span>optimal_rank <span class="cf">if</span> optimal_rank <span class="cf">else</span> <span class="st">'unknown'</span><span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/plot-species-clustering-13.png" width="960"></p>
<div class="sourceCode" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="individual-plot-6-gene-factor-distribution" class="level2">
<h2 class="anchored" data-anchor-id="individual-plot-6-gene-factor-distribution">4.8 Individual Plot 6: Gene Factor Distribution</h2>
<div class="sourceCode" id="cb28"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory</span></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the optimal rank from the rank selection summary</span></span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir:</span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a>    summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a>            rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> (from rank selection analysis)"</span>)</span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No rank selection summary found! Cannot determine optimal rank."</span>)</span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No main results directory found!"</span>)</span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for the optimal rank data in rank_comparison directory</span></span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a>latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir <span class="kw">and</span> optimal_rank:</span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb28-48"><a href="#cb28-48" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb28-49"><a href="#cb28-49" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb28-50"><a href="#cb28-50" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb28-51"><a href="#cb28-51" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb28-52"><a href="#cb28-52" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb28-53"><a href="#cb28-53" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb28-54"><a href="#cb28-54" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb28-55"><a href="#cb28-55" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb28-56"><a href="#cb28-56" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found!"</span>)</span>
<span id="cb28-57"><a href="#cb28-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-58"><a href="#cb28-58" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb28-59"><a href="#cb28-59" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb28-60"><a href="#cb28-60" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load data</span></span>
<span id="cb28-61"><a href="#cb28-61" aria-hidden="true" tabindex="-1"></a>    gene_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb28-62"><a href="#cb28-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb28-63"><a href="#cb28-63" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate gene variances</span></span>
<span id="cb28-64"><a href="#cb28-64" aria-hidden="true" tabindex="-1"></a>    gene_variances <span class="op">=</span> np.var(gene_factors, axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb28-65"><a href="#cb28-65" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb28-66"><a href="#cb28-66" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create histogram</span></span>
<span id="cb28-67"><a href="#cb28-67" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">6</span>))</span>
<span id="cb28-68"><a href="#cb28-68" aria-hidden="true" tabindex="-1"></a>    plt.hist(gene_variances, bins<span class="op">=</span><span class="dv">50</span>, alpha<span class="op">=</span><span class="fl">0.7</span>, edgecolor<span class="op">=</span><span class="st">'black'</span>)</span>
<span id="cb28-69"><a href="#cb28-69" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Gene Factor Variance'</span>)</span>
<span id="cb28-70"><a href="#cb28-70" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Number of Genes'</span>)</span>
<span id="cb28-71"><a href="#cb28-71" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Distribution of Gene Factor Variability (Optimal Rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb28-72"><a href="#cb28-72" aria-hidden="true" tabindex="-1"></a>    plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb28-73"><a href="#cb28-73" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb28-74"><a href="#cb28-74" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add statistics</span></span>
<span id="cb28-75"><a href="#cb28-75" aria-hidden="true" tabindex="-1"></a>    plt.axvline(np.mean(gene_variances), color<span class="op">=</span><span class="st">'red'</span>, linestyle<span class="op">=</span><span class="st">'--'</span>, </span>
<span id="cb28-76"><a href="#cb28-76" aria-hidden="true" tabindex="-1"></a>               label<span class="op">=</span><span class="ss">f'Mean: </span><span class="sc">{</span>np<span class="sc">.</span>mean(gene_variances)<span class="sc">:.3f}</span><span class="ss">'</span>)</span>
<span id="cb28-77"><a href="#cb28-77" aria-hidden="true" tabindex="-1"></a>    plt.axvline(np.median(gene_variances), color<span class="op">=</span><span class="st">'orange'</span>, linestyle<span class="op">=</span><span class="st">'--'</span>, </span>
<span id="cb28-78"><a href="#cb28-78" aria-hidden="true" tabindex="-1"></a>               label<span class="op">=</span><span class="ss">f'Median: </span><span class="sc">{</span>np<span class="sc">.</span>median(gene_variances)<span class="sc">:.3f}</span><span class="ss">'</span>)</span>
<span id="cb28-79"><a href="#cb28-79" aria-hidden="true" tabindex="-1"></a>    plt.legend()</span>
<span id="cb28-80"><a href="#cb28-80" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb28-81"><a href="#cb28-81" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save individual plot in main results directory</span></span>
<span id="cb28-82"><a href="#cb28-82" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_results_dir:</span>
<span id="cb28-83"><a href="#cb28-83" aria-hidden="true" tabindex="-1"></a>        plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"plot_06_gene_distribution_optimal_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb28-84"><a href="#cb28-84" aria-hidden="true" tabindex="-1"></a>        plt.savefig(plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb28-85"><a href="#cb28-85" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Gene factor distribution plot saved to: </span><span class="sc">{</span>plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb28-86"><a href="#cb28-86" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb28-87"><a href="#cb28-87" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb28-88"><a href="#cb28-88" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb28-89"><a href="#cb28-89" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Could not load data for optimal rank </span><span class="sc">{</span>optimal_rank <span class="cf">if</span> optimal_rank <span class="cf">else</span> <span class="st">'unknown'</span><span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/plot-gene-distribution-15.png" width="960"></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="individual-plot-7-component-correlations" class="level2">
<h2 class="anchored" data-anchor-id="individual-plot-7-component-correlations">4.9 Individual Plot 7: Component Correlations</h2>
<div class="sourceCode" id="cb30"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory</span></span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the optimal rank from the rank selection summary</span></span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir:</span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a>    summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>            rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> (from rank selection analysis)"</span>)</span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No rank selection summary found! Cannot determine optimal rank."</span>)</span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No main results directory found!"</span>)</span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-42"><a href="#cb30-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for the optimal rank data in rank_comparison directory</span></span>
<span id="cb30-43"><a href="#cb30-43" aria-hidden="true" tabindex="-1"></a>latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb30-44"><a href="#cb30-44" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir <span class="kw">and</span> optimal_rank:</span>
<span id="cb30-45"><a href="#cb30-45" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb30-46"><a href="#cb30-46" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb30-47"><a href="#cb30-47" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-48"><a href="#cb30-48" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb30-49"><a href="#cb30-49" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb30-50"><a href="#cb30-50" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb30-51"><a href="#cb30-51" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb30-52"><a href="#cb30-52" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb30-53"><a href="#cb30-53" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb30-54"><a href="#cb30-54" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb30-55"><a href="#cb30-55" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb30-56"><a href="#cb30-56" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb30-57"><a href="#cb30-57" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found!"</span>)</span>
<span id="cb30-58"><a href="#cb30-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-59"><a href="#cb30-59" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb30-60"><a href="#cb30-60" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-61"><a href="#cb30-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load data</span></span>
<span id="cb30-62"><a href="#cb30-62" aria-hidden="true" tabindex="-1"></a>    gene_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb30-63"><a href="#cb30-63" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-64"><a href="#cb30-64" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate correlations between components across genes</span></span>
<span id="cb30-65"><a href="#cb30-65" aria-hidden="true" tabindex="-1"></a>    gene_corr <span class="op">=</span> np.corrcoef(gene_factors.T)</span>
<span id="cb30-66"><a href="#cb30-66" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-67"><a href="#cb30-67" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create heatmap</span></span>
<span id="cb30-68"><a href="#cb30-68" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">12</span>, <span class="dv">10</span>))</span>
<span id="cb30-69"><a href="#cb30-69" aria-hidden="true" tabindex="-1"></a>    sns.heatmap(gene_corr, cmap<span class="op">=</span><span class="st">'RdBu_r'</span>, center<span class="op">=</span><span class="dv">0</span>, </span>
<span id="cb30-70"><a href="#cb30-70" aria-hidden="true" tabindex="-1"></a>                xticklabels<span class="op">=</span><span class="bu">range</span>(<span class="dv">1</span>, gene_factors.shape[<span class="dv">1</span>] <span class="op">+</span> <span class="dv">1</span>),</span>
<span id="cb30-71"><a href="#cb30-71" aria-hidden="true" tabindex="-1"></a>                yticklabels<span class="op">=</span><span class="bu">range</span>(<span class="dv">1</span>, gene_factors.shape[<span class="dv">1</span>] <span class="op">+</span> <span class="dv">1</span>),</span>
<span id="cb30-72"><a href="#cb30-72" aria-hidden="true" tabindex="-1"></a>                square<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb30-73"><a href="#cb30-73" aria-hidden="true" tabindex="-1"></a>                cbar_kws<span class="op">=</span>{<span class="st">'label'</span>: <span class="st">'Correlation coefficient'</span>})</span>
<span id="cb30-74"><a href="#cb30-74" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Component Correlations (Gene Space) - Optimal Rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb30-75"><a href="#cb30-75" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Component'</span>)</span>
<span id="cb30-76"><a href="#cb30-76" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Component'</span>)</span>
<span id="cb30-77"><a href="#cb30-77" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-78"><a href="#cb30-78" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save individual plot in main results directory</span></span>
<span id="cb30-79"><a href="#cb30-79" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_results_dir:</span>
<span id="cb30-80"><a href="#cb30-80" aria-hidden="true" tabindex="-1"></a>        plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"plot_07_component_correlations_optimal_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb30-81"><a href="#cb30-81" aria-hidden="true" tabindex="-1"></a>        plt.savefig(plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb30-82"><a href="#cb30-82" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Component correlations plot saved to: </span><span class="sc">{</span>plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb30-83"><a href="#cb30-83" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-84"><a href="#cb30-84" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb30-85"><a href="#cb30-85" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb30-86"><a href="#cb30-86" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Could not load data for optimal rank </span><span class="sc">{</span>optimal_rank <span class="cf">if</span> optimal_rank <span class="cf">else</span> <span class="st">'unknown'</span><span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/plot-component-correlations-17.png" width="1152"></p>
<div class="sourceCode" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="individual-plot-8-species-type-comparison" class="level2">
<h2 class="anchored" data-anchor-id="individual-plot-8-species-type-comparison">4.10 Individual Plot 8: Species Type Comparison</h2>
<div class="sourceCode" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory</span></span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the optimal rank from the rank selection summary</span></span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir:</span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a>    summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true" tabindex="-1"></a>            rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> (from rank selection analysis)"</span>)</span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No rank selection summary found! Cannot determine optimal rank."</span>)</span>
<span id="cb32-36"><a href="#cb32-36" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb32-37"><a href="#cb32-37" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb32-38"><a href="#cb32-38" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb32-39"><a href="#cb32-39" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No main results directory found!"</span>)</span>
<span id="cb32-40"><a href="#cb32-40" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb32-41"><a href="#cb32-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-42"><a href="#cb32-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for the optimal rank data in rank_comparison directory</span></span>
<span id="cb32-43"><a href="#cb32-43" aria-hidden="true" tabindex="-1"></a>latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb32-44"><a href="#cb32-44" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir <span class="kw">and</span> optimal_rank:</span>
<span id="cb32-45"><a href="#cb32-45" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb32-46"><a href="#cb32-46" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb32-47"><a href="#cb32-47" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-48"><a href="#cb32-48" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb32-49"><a href="#cb32-49" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb32-50"><a href="#cb32-50" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb32-51"><a href="#cb32-51" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb32-52"><a href="#cb32-52" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb32-53"><a href="#cb32-53" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb32-54"><a href="#cb32-54" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb32-55"><a href="#cb32-55" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb32-56"><a href="#cb32-56" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb32-57"><a href="#cb32-57" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found!"</span>)</span>
<span id="cb32-58"><a href="#cb32-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-59"><a href="#cb32-59" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb32-60"><a href="#cb32-60" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-61"><a href="#cb32-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load data</span></span>
<span id="cb32-62"><a href="#cb32-62" aria-hidden="true" tabindex="-1"></a>    gene_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb32-63"><a href="#cb32-63" aria-hidden="true" tabindex="-1"></a>    species_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"species_factors.npy"</span>))</span>
<span id="cb32-64"><a href="#cb32-64" aria-hidden="true" tabindex="-1"></a>    time_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"time_factors.npy"</span>))</span>
<span id="cb32-65"><a href="#cb32-65" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-66"><a href="#cb32-66" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load the transformed data to get species names</span></span>
<span id="cb32-67"><a href="#cb32-67" aria-hidden="true" tabindex="-1"></a>    data_file <span class="op">=</span> os.path.join(output_dir, <span class="st">"vst_counts_matrix_long_format.csv"</span>)</span>
<span id="cb32-68"><a href="#cb32-68" aria-hidden="true" tabindex="-1"></a>    df_long <span class="op">=</span> pd.read_csv(data_file)</span>
<span id="cb32-69"><a href="#cb32-69" aria-hidden="true" tabindex="-1"></a>    species_names <span class="op">=</span> <span class="bu">sorted</span>(df_long[<span class="st">'species'</span>].unique())</span>
<span id="cb32-70"><a href="#cb32-70" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-71"><a href="#cb32-71" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate component strengths and get top 5</span></span>
<span id="cb32-72"><a href="#cb32-72" aria-hidden="true" tabindex="-1"></a>    component_strengths <span class="op">=</span> []</span>
<span id="cb32-73"><a href="#cb32-73" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(gene_factors.shape[<span class="dv">1</span>]):</span>
<span id="cb32-74"><a href="#cb32-74" aria-hidden="true" tabindex="-1"></a>        gene_norm <span class="op">=</span> np.linalg.norm(gene_factors[:, i])</span>
<span id="cb32-75"><a href="#cb32-75" aria-hidden="true" tabindex="-1"></a>        species_norm <span class="op">=</span> np.linalg.norm(species_factors[:, i])</span>
<span id="cb32-76"><a href="#cb32-76" aria-hidden="true" tabindex="-1"></a>        time_norm <span class="op">=</span> np.linalg.norm(time_factors[:, i])</span>
<span id="cb32-77"><a href="#cb32-77" aria-hidden="true" tabindex="-1"></a>        total_strength <span class="op">=</span> gene_norm <span class="op">*</span> species_norm <span class="op">*</span> time_norm</span>
<span id="cb32-78"><a href="#cb32-78" aria-hidden="true" tabindex="-1"></a>        component_strengths.append(total_strength)</span>
<span id="cb32-79"><a href="#cb32-79" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-80"><a href="#cb32-80" aria-hidden="true" tabindex="-1"></a>    sorted_components <span class="op">=</span> np.argsort(component_strengths)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb32-81"><a href="#cb32-81" aria-hidden="true" tabindex="-1"></a>    top_5_components <span class="op">=</span> sorted_components[:<span class="bu">min</span>(<span class="dv">5</span>, optimal_rank)]  <span class="co"># Use min to handle ranks &lt; 5</span></span>
<span id="cb32-82"><a href="#cb32-82" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-83"><a href="#cb32-83" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Group species by type</span></span>
<span id="cb32-84"><a href="#cb32-84" aria-hidden="true" tabindex="-1"></a>    species_by_type <span class="op">=</span> {}</span>
<span id="cb32-85"><a href="#cb32-85" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i, name <span class="kw">in</span> <span class="bu">enumerate</span>(species_names):</span>
<span id="cb32-86"><a href="#cb32-86" aria-hidden="true" tabindex="-1"></a>        stype <span class="op">=</span> name.split(<span class="st">'-'</span>)[<span class="dv">0</span>]</span>
<span id="cb32-87"><a href="#cb32-87" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> stype <span class="kw">not</span> <span class="kw">in</span> species_by_type:</span>
<span id="cb32-88"><a href="#cb32-88" aria-hidden="true" tabindex="-1"></a>            species_by_type[stype] <span class="op">=</span> []</span>
<span id="cb32-89"><a href="#cb32-89" aria-hidden="true" tabindex="-1"></a>        species_by_type[stype].append(i)</span>
<span id="cb32-90"><a href="#cb32-90" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-91"><a href="#cb32-91" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate mean factor values per species type for top components</span></span>
<span id="cb32-92"><a href="#cb32-92" aria-hidden="true" tabindex="-1"></a>    type_means <span class="op">=</span> {}</span>
<span id="cb32-93"><a href="#cb32-93" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> stype, indices <span class="kw">in</span> species_by_type.items():</span>
<span id="cb32-94"><a href="#cb32-94" aria-hidden="true" tabindex="-1"></a>        type_means[stype] <span class="op">=</span> np.mean(species_factors[indices, :], axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb32-95"><a href="#cb32-95" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-96"><a href="#cb32-96" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create bar plot</span></span>
<span id="cb32-97"><a href="#cb32-97" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">12</span>, <span class="dv">6</span>))</span>
<span id="cb32-98"><a href="#cb32-98" aria-hidden="true" tabindex="-1"></a>    x_pos <span class="op">=</span> np.arange(<span class="bu">len</span>(top_5_components))</span>
<span id="cb32-99"><a href="#cb32-99" aria-hidden="true" tabindex="-1"></a>    width <span class="op">=</span> <span class="fl">0.8</span> <span class="op">/</span> <span class="bu">len</span>(type_means)  <span class="co"># Adjust width based on number of species types</span></span>
<span id="cb32-100"><a href="#cb32-100" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-101"><a href="#cb32-101" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i, (stype, means) <span class="kw">in</span> <span class="bu">enumerate</span>(type_means.items()):</span>
<span id="cb32-102"><a href="#cb32-102" aria-hidden="true" tabindex="-1"></a>        plt.bar(x_pos <span class="op">+</span> i <span class="op">*</span> width, means[top_5_components], </span>
<span id="cb32-103"><a href="#cb32-103" aria-hidden="true" tabindex="-1"></a>               width, label<span class="op">=</span>stype, alpha<span class="op">=</span><span class="fl">0.8</span>)</span>
<span id="cb32-104"><a href="#cb32-104" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-105"><a href="#cb32-105" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Component'</span>)</span>
<span id="cb32-106"><a href="#cb32-106" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Mean Factor Value'</span>)</span>
<span id="cb32-107"><a href="#cb32-107" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Species Type Comparison (Top </span><span class="sc">{</span><span class="bu">len</span>(top_5_components)<span class="sc">}</span><span class="ss"> Components) - Optimal Rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb32-108"><a href="#cb32-108" aria-hidden="true" tabindex="-1"></a>    plt.xticks(x_pos <span class="op">+</span> width <span class="op">*</span> (<span class="bu">len</span>(type_means) <span class="op">-</span> <span class="dv">1</span>) <span class="op">/</span> <span class="dv">2</span>, </span>
<span id="cb32-109"><a href="#cb32-109" aria-hidden="true" tabindex="-1"></a>               [<span class="ss">f'Comp </span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> top_5_components])</span>
<span id="cb32-110"><a href="#cb32-110" aria-hidden="true" tabindex="-1"></a>    plt.legend()</span>
<span id="cb32-111"><a href="#cb32-111" aria-hidden="true" tabindex="-1"></a>    plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb32-112"><a href="#cb32-112" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-113"><a href="#cb32-113" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save individual plot in main results directory</span></span>
<span id="cb32-114"><a href="#cb32-114" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_results_dir:</span>
<span id="cb32-115"><a href="#cb32-115" aria-hidden="true" tabindex="-1"></a>        plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"plot_08_species_comparison_optimal_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb32-116"><a href="#cb32-116" aria-hidden="true" tabindex="-1"></a>        plt.savefig(plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb32-117"><a href="#cb32-117" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Species type comparison plot saved to: </span><span class="sc">{</span>plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb32-118"><a href="#cb32-118" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-119"><a href="#cb32-119" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb32-120"><a href="#cb32-120" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb32-121"><a href="#cb32-121" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Could not load data for optimal rank </span><span class="sc">{</span>optimal_rank <span class="cf">if</span> optimal_rank <span class="cf">else</span> <span class="st">'unknown'</span><span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/plot-species-comparison-19.png" width="1152"></p>
<div class="sourceCode" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="individual-plot-9-temporal-variance-per-component" class="level2">
<h2 class="anchored" data-anchor-id="individual-plot-9-temporal-variance-per-component">4.11 Individual Plot 9: Temporal Variance per Component</h2>
<div class="sourceCode" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting</span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory</span></span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the optimal rank from the rank selection summary</span></span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir:</span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a>    summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a>            rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> (from rank selection analysis)"</span>)</span>
<span id="cb34-33"><a href="#cb34-33" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb34-34"><a href="#cb34-34" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No rank selection summary found! Cannot determine optimal rank."</span>)</span>
<span id="cb34-35"><a href="#cb34-35" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb34-36"><a href="#cb34-36" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb34-37"><a href="#cb34-37" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb34-38"><a href="#cb34-38" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No main results directory found!"</span>)</span>
<span id="cb34-39"><a href="#cb34-39" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb34-40"><a href="#cb34-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-41"><a href="#cb34-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for the optimal rank data in rank_comparison directory</span></span>
<span id="cb34-42"><a href="#cb34-42" aria-hidden="true" tabindex="-1"></a>latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb34-43"><a href="#cb34-43" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir <span class="kw">and</span> optimal_rank:</span>
<span id="cb34-44"><a href="#cb34-44" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb34-45"><a href="#cb34-45" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb34-46"><a href="#cb34-46" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-47"><a href="#cb34-47" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb34-48"><a href="#cb34-48" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb34-49"><a href="#cb34-49" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb34-50"><a href="#cb34-50" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb34-51"><a href="#cb34-51" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb34-52"><a href="#cb34-52" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb34-53"><a href="#cb34-53" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb34-54"><a href="#cb34-54" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb34-55"><a href="#cb34-55" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb34-56"><a href="#cb34-56" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found!"</span>)</span>
<span id="cb34-57"><a href="#cb34-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-58"><a href="#cb34-58" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb34-59"><a href="#cb34-59" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-60"><a href="#cb34-60" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load data</span></span>
<span id="cb34-61"><a href="#cb34-61" aria-hidden="true" tabindex="-1"></a>    time_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"time_factors.npy"</span>))</span>
<span id="cb34-62"><a href="#cb34-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-63"><a href="#cb34-63" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate temporal variances</span></span>
<span id="cb34-64"><a href="#cb34-64" aria-hidden="true" tabindex="-1"></a>    time_variances <span class="op">=</span> np.var(time_factors, axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb34-65"><a href="#cb34-65" aria-hidden="true" tabindex="-1"></a>    sorted_time_vars <span class="op">=</span> np.argsort(time_variances)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb34-66"><a href="#cb34-66" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-67"><a href="#cb34-67" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create bar plot</span></span>
<span id="cb34-68"><a href="#cb34-68" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">12</span>, <span class="dv">6</span>))</span>
<span id="cb34-69"><a href="#cb34-69" aria-hidden="true" tabindex="-1"></a>    plt.bar(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(time_variances) <span class="op">+</span> <span class="dv">1</span>), </span>
<span id="cb34-70"><a href="#cb34-70" aria-hidden="true" tabindex="-1"></a>           [time_variances[i] <span class="cf">for</span> i <span class="kw">in</span> sorted_time_vars])</span>
<span id="cb34-71"><a href="#cb34-71" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Component (ranked by temporal variance)'</span>)</span>
<span id="cb34-72"><a href="#cb34-72" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Temporal Variance'</span>)</span>
<span id="cb34-73"><a href="#cb34-73" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Components by Temporal Variability - Optimal Rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb34-74"><a href="#cb34-74" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-75"><a href="#cb34-75" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set x-ticks to show all components if &lt;= 10, otherwise show every 2nd or 5th</span></span>
<span id="cb34-76"><a href="#cb34-76" aria-hidden="true" tabindex="-1"></a>    n_components <span class="op">=</span> <span class="bu">len</span>(time_variances)</span>
<span id="cb34-77"><a href="#cb34-77" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> n_components <span class="op">&lt;=</span> <span class="dv">10</span>:</span>
<span id="cb34-78"><a href="#cb34-78" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>))</span>
<span id="cb34-79"><a href="#cb34-79" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> n_components <span class="op">&lt;=</span> <span class="dv">20</span>:</span>
<span id="cb34-80"><a href="#cb34-80" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb34-81"><a href="#cb34-81" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb34-82"><a href="#cb34-82" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>, <span class="dv">5</span>))</span>
<span id="cb34-83"><a href="#cb34-83" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-84"><a href="#cb34-84" aria-hidden="true" tabindex="-1"></a>    plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb34-85"><a href="#cb34-85" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-86"><a href="#cb34-86" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save individual plot in main results directory</span></span>
<span id="cb34-87"><a href="#cb34-87" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_results_dir:</span>
<span id="cb34-88"><a href="#cb34-88" aria-hidden="true" tabindex="-1"></a>        plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"plot_09_temporal_variance_optimal_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb34-89"><a href="#cb34-89" aria-hidden="true" tabindex="-1"></a>        plt.savefig(plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb34-90"><a href="#cb34-90" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Temporal variance plot saved to: </span><span class="sc">{</span>plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb34-91"><a href="#cb34-91" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-92"><a href="#cb34-92" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb34-93"><a href="#cb34-93" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb34-94"><a href="#cb34-94" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Could not load data for optimal rank </span><span class="sc">{</span>optimal_rank <span class="cf">if</span> optimal_rank <span class="cf">else</span> <span class="st">'unknown'</span><span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/plot-temporal-variance-21.png" width="1152"></p>
<div class="sourceCode" id="cb35"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="individual-plot-10-species-variance-per-component" class="level2">
<h2 class="anchored" data-anchor-id="individual-plot-10-species-variance-per-component">4.12 Individual Plot 10: Species Variance per Component</h2>
<div class="sourceCode" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory</span></span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-25"><a href="#cb36-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the optimal rank from the rank selection summary</span></span>
<span id="cb36-26"><a href="#cb36-26" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir:</span>
<span id="cb36-27"><a href="#cb36-27" aria-hidden="true" tabindex="-1"></a>    summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb36-28"><a href="#cb36-28" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb36-29"><a href="#cb36-29" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb36-30"><a href="#cb36-30" aria-hidden="true" tabindex="-1"></a>            rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb36-31"><a href="#cb36-31" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb36-32"><a href="#cb36-32" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> (from rank selection analysis)"</span>)</span>
<span id="cb36-33"><a href="#cb36-33" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb36-34"><a href="#cb36-34" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No rank selection summary found! Cannot determine optimal rank."</span>)</span>
<span id="cb36-35"><a href="#cb36-35" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb36-36"><a href="#cb36-36" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb36-37"><a href="#cb36-37" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb36-38"><a href="#cb36-38" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No main results directory found!"</span>)</span>
<span id="cb36-39"><a href="#cb36-39" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb36-40"><a href="#cb36-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-41"><a href="#cb36-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for the optimal rank data in rank_comparison directory</span></span>
<span id="cb36-42"><a href="#cb36-42" aria-hidden="true" tabindex="-1"></a>latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb36-43"><a href="#cb36-43" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir <span class="kw">and</span> optimal_rank:</span>
<span id="cb36-44"><a href="#cb36-44" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb36-45"><a href="#cb36-45" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb36-46"><a href="#cb36-46" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-47"><a href="#cb36-47" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb36-48"><a href="#cb36-48" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb36-49"><a href="#cb36-49" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb36-50"><a href="#cb36-50" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb36-51"><a href="#cb36-51" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb36-52"><a href="#cb36-52" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb36-53"><a href="#cb36-53" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb36-54"><a href="#cb36-54" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb36-55"><a href="#cb36-55" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb36-56"><a href="#cb36-56" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found!"</span>)</span>
<span id="cb36-57"><a href="#cb36-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-58"><a href="#cb36-58" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb36-59"><a href="#cb36-59" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-60"><a href="#cb36-60" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load data</span></span>
<span id="cb36-61"><a href="#cb36-61" aria-hidden="true" tabindex="-1"></a>    species_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"species_factors.npy"</span>))</span>
<span id="cb36-62"><a href="#cb36-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-63"><a href="#cb36-63" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate species variances</span></span>
<span id="cb36-64"><a href="#cb36-64" aria-hidden="true" tabindex="-1"></a>    species_variances <span class="op">=</span> np.var(species_factors, axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb36-65"><a href="#cb36-65" aria-hidden="true" tabindex="-1"></a>    sorted_species_vars <span class="op">=</span> np.argsort(species_variances)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb36-66"><a href="#cb36-66" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-67"><a href="#cb36-67" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create bar plot</span></span>
<span id="cb36-68"><a href="#cb36-68" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">12</span>, <span class="dv">6</span>))</span>
<span id="cb36-69"><a href="#cb36-69" aria-hidden="true" tabindex="-1"></a>    plt.bar(<span class="bu">range</span>(<span class="dv">1</span>, <span class="bu">len</span>(species_variances) <span class="op">+</span> <span class="dv">1</span>), </span>
<span id="cb36-70"><a href="#cb36-70" aria-hidden="true" tabindex="-1"></a>           [species_variances[i] <span class="cf">for</span> i <span class="kw">in</span> sorted_species_vars])</span>
<span id="cb36-71"><a href="#cb36-71" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Component (ranked by species variance)'</span>)</span>
<span id="cb36-72"><a href="#cb36-72" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Species Variance'</span>)</span>
<span id="cb36-73"><a href="#cb36-73" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Components by Species Variability - Optimal Rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb36-74"><a href="#cb36-74" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-75"><a href="#cb36-75" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set x-ticks to show all components if &lt;= 10, otherwise show every 2nd or 5th</span></span>
<span id="cb36-76"><a href="#cb36-76" aria-hidden="true" tabindex="-1"></a>    n_components <span class="op">=</span> <span class="bu">len</span>(species_variances)</span>
<span id="cb36-77"><a href="#cb36-77" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> n_components <span class="op">&lt;=</span> <span class="dv">10</span>:</span>
<span id="cb36-78"><a href="#cb36-78" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>))</span>
<span id="cb36-79"><a href="#cb36-79" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> n_components <span class="op">&lt;=</span> <span class="dv">20</span>:</span>
<span id="cb36-80"><a href="#cb36-80" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb36-81"><a href="#cb36-81" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb36-82"><a href="#cb36-82" aria-hidden="true" tabindex="-1"></a>        plt.xticks(<span class="bu">range</span>(<span class="dv">1</span>, n_components <span class="op">+</span> <span class="dv">1</span>, <span class="dv">5</span>))</span>
<span id="cb36-83"><a href="#cb36-83" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-84"><a href="#cb36-84" aria-hidden="true" tabindex="-1"></a>    plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb36-85"><a href="#cb36-85" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-86"><a href="#cb36-86" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save individual plot in main results directory</span></span>
<span id="cb36-87"><a href="#cb36-87" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_results_dir:</span>
<span id="cb36-88"><a href="#cb36-88" aria-hidden="true" tabindex="-1"></a>        plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"plot_10_species_variance_optimal_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb36-89"><a href="#cb36-89" aria-hidden="true" tabindex="-1"></a>        plt.savefig(plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb36-90"><a href="#cb36-90" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Species variance plot saved to: </span><span class="sc">{</span>plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb36-91"><a href="#cb36-91" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-92"><a href="#cb36-92" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb36-93"><a href="#cb36-93" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb36-94"><a href="#cb36-94" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Could not load data for optimal rank </span><span class="sc">{</span>optimal_rank <span class="cf">if</span> optimal_rank <span class="cf">else</span> <span class="st">'unknown'</span><span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/plot-species-variance-23.png" width="1152"></p>
<div class="sourceCode" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="individual-plot-11-top-genes-heatmap" class="level2">
<h2 class="anchored" data-anchor-id="individual-plot-11-top-genes-heatmap">4.13 Individual Plot 11: Top Genes Heatmap</h2>
<div class="sourceCode" id="cb38"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting</span></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory</span></span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb38-23"><a href="#cb38-23" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb38-24"><a href="#cb38-24" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb38-25"><a href="#cb38-25" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb38-26"><a href="#cb38-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-27"><a href="#cb38-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the optimal rank from the rank selection summary</span></span>
<span id="cb38-28"><a href="#cb38-28" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir:</span>
<span id="cb38-29"><a href="#cb38-29" aria-hidden="true" tabindex="-1"></a>    summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb38-30"><a href="#cb38-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb38-31"><a href="#cb38-31" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb38-32"><a href="#cb38-32" aria-hidden="true" tabindex="-1"></a>            rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb38-33"><a href="#cb38-33" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb38-34"><a href="#cb38-34" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> (from rank selection analysis)"</span>)</span>
<span id="cb38-35"><a href="#cb38-35" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb38-36"><a href="#cb38-36" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No rank selection summary found! Cannot determine optimal rank."</span>)</span>
<span id="cb38-37"><a href="#cb38-37" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb38-38"><a href="#cb38-38" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb38-39"><a href="#cb38-39" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb38-40"><a href="#cb38-40" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No main results directory found!"</span>)</span>
<span id="cb38-41"><a href="#cb38-41" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb38-42"><a href="#cb38-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-43"><a href="#cb38-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for the optimal rank data in rank_comparison directory</span></span>
<span id="cb38-44"><a href="#cb38-44" aria-hidden="true" tabindex="-1"></a>latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb38-45"><a href="#cb38-45" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir <span class="kw">and</span> optimal_rank:</span>
<span id="cb38-46"><a href="#cb38-46" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb38-47"><a href="#cb38-47" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb38-48"><a href="#cb38-48" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-49"><a href="#cb38-49" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb38-50"><a href="#cb38-50" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb38-51"><a href="#cb38-51" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb38-52"><a href="#cb38-52" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb38-53"><a href="#cb38-53" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb38-54"><a href="#cb38-54" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb38-55"><a href="#cb38-55" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb38-56"><a href="#cb38-56" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb38-57"><a href="#cb38-57" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb38-58"><a href="#cb38-58" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found!"</span>)</span>
<span id="cb38-59"><a href="#cb38-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-60"><a href="#cb38-60" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb38-61"><a href="#cb38-61" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-62"><a href="#cb38-62" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load data</span></span>
<span id="cb38-63"><a href="#cb38-63" aria-hidden="true" tabindex="-1"></a>    gene_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb38-64"><a href="#cb38-64" aria-hidden="true" tabindex="-1"></a>    species_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"species_factors.npy"</span>))</span>
<span id="cb38-65"><a href="#cb38-65" aria-hidden="true" tabindex="-1"></a>    time_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"time_factors.npy"</span>))</span>
<span id="cb38-66"><a href="#cb38-66" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-67"><a href="#cb38-67" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load the transformed data to get actual gene names</span></span>
<span id="cb38-68"><a href="#cb38-68" aria-hidden="true" tabindex="-1"></a>    data_file <span class="op">=</span> os.path.join(output_dir, <span class="st">"vst_counts_matrix_long_format.csv"</span>)</span>
<span id="cb38-69"><a href="#cb38-69" aria-hidden="true" tabindex="-1"></a>    df_long <span class="op">=</span> pd.read_csv(data_file)</span>
<span id="cb38-70"><a href="#cb38-70" aria-hidden="true" tabindex="-1"></a>    gene_names <span class="op">=</span> <span class="bu">sorted</span>(df_long[<span class="st">'gene'</span>].unique())</span>
<span id="cb38-71"><a href="#cb38-71" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-72"><a href="#cb38-72" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate component strengths and get top components</span></span>
<span id="cb38-73"><a href="#cb38-73" aria-hidden="true" tabindex="-1"></a>    component_strengths <span class="op">=</span> []</span>
<span id="cb38-74"><a href="#cb38-74" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(gene_factors.shape[<span class="dv">1</span>]):</span>
<span id="cb38-75"><a href="#cb38-75" aria-hidden="true" tabindex="-1"></a>        gene_norm <span class="op">=</span> np.linalg.norm(gene_factors[:, i])</span>
<span id="cb38-76"><a href="#cb38-76" aria-hidden="true" tabindex="-1"></a>        species_norm <span class="op">=</span> np.linalg.norm(species_factors[:, i])</span>
<span id="cb38-77"><a href="#cb38-77" aria-hidden="true" tabindex="-1"></a>        time_norm <span class="op">=</span> np.linalg.norm(time_factors[:, i])</span>
<span id="cb38-78"><a href="#cb38-78" aria-hidden="true" tabindex="-1"></a>        total_strength <span class="op">=</span> gene_norm <span class="op">*</span> species_norm <span class="op">*</span> time_norm</span>
<span id="cb38-79"><a href="#cb38-79" aria-hidden="true" tabindex="-1"></a>        component_strengths.append(total_strength)</span>
<span id="cb38-80"><a href="#cb38-80" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-81"><a href="#cb38-81" aria-hidden="true" tabindex="-1"></a>    sorted_components <span class="op">=</span> np.argsort(component_strengths)[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb38-82"><a href="#cb38-82" aria-hidden="true" tabindex="-1"></a>    top_components <span class="op">=</span> sorted_components[:<span class="bu">min</span>(<span class="dv">5</span>, optimal_rank)]  <span class="co"># Use min to handle ranks &lt; 5</span></span>
<span id="cb38-83"><a href="#cb38-83" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-84"><a href="#cb38-84" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get top genes for each of the top components</span></span>
<span id="cb38-85"><a href="#cb38-85" aria-hidden="true" tabindex="-1"></a>    n_top_genes <span class="op">=</span> <span class="dv">10</span></span>
<span id="cb38-86"><a href="#cb38-86" aria-hidden="true" tabindex="-1"></a>    top_gene_indices <span class="op">=</span> []</span>
<span id="cb38-87"><a href="#cb38-87" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-88"><a href="#cb38-88" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> comp_idx <span class="kw">in</span> top_components:</span>
<span id="cb38-89"><a href="#cb38-89" aria-hidden="true" tabindex="-1"></a>        top_genes <span class="op">=</span> np.argsort(np.<span class="bu">abs</span>(gene_factors[:, comp_idx]))[<span class="op">-</span>n_top_genes:]</span>
<span id="cb38-90"><a href="#cb38-90" aria-hidden="true" tabindex="-1"></a>        top_gene_indices.extend(top_genes)</span>
<span id="cb38-91"><a href="#cb38-91" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-92"><a href="#cb38-92" aria-hidden="true" tabindex="-1"></a>    top_gene_indices <span class="op">=</span> <span class="bu">list</span>(<span class="bu">set</span>(top_gene_indices))  <span class="co"># Remove duplicates</span></span>
<span id="cb38-93"><a href="#cb38-93" aria-hidden="true" tabindex="-1"></a>    top_gene_data <span class="op">=</span> gene_factors[top_gene_indices, :][:, top_components]</span>
<span id="cb38-94"><a href="#cb38-94" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-95"><a href="#cb38-95" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get actual gene names for the selected indices</span></span>
<span id="cb38-96"><a href="#cb38-96" aria-hidden="true" tabindex="-1"></a>    top_gene_names <span class="op">=</span> [gene_names[i] <span class="cf">for</span> i <span class="kw">in</span> top_gene_indices]</span>
<span id="cb38-97"><a href="#cb38-97" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-98"><a href="#cb38-98" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create heatmap</span></span>
<span id="cb38-99"><a href="#cb38-99" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">12</span>))</span>
<span id="cb38-100"><a href="#cb38-100" aria-hidden="true" tabindex="-1"></a>    sns.heatmap(top_gene_data, </span>
<span id="cb38-101"><a href="#cb38-101" aria-hidden="true" tabindex="-1"></a>                cmap<span class="op">=</span><span class="st">'RdBu_r'</span>, center<span class="op">=</span><span class="dv">0</span>,</span>
<span id="cb38-102"><a href="#cb38-102" aria-hidden="true" tabindex="-1"></a>                yticklabels<span class="op">=</span>top_gene_names,</span>
<span id="cb38-103"><a href="#cb38-103" aria-hidden="true" tabindex="-1"></a>                xticklabels<span class="op">=</span>[<span class="ss">f'Comp </span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> top_components],</span>
<span id="cb38-104"><a href="#cb38-104" aria-hidden="true" tabindex="-1"></a>                cbar_kws<span class="op">=</span>{<span class="st">'label'</span>: <span class="st">'Gene factor loading'</span>})</span>
<span id="cb38-105"><a href="#cb38-105" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Top </span><span class="sc">{</span><span class="bu">len</span>(top_gene_indices)<span class="sc">}</span><span class="ss"> Genes × Top </span><span class="sc">{</span><span class="bu">len</span>(top_components)<span class="sc">}</span><span class="ss"> Components - Optimal Rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb38-106"><a href="#cb38-106" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Component'</span>)</span>
<span id="cb38-107"><a href="#cb38-107" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Gene'</span>)</span>
<span id="cb38-108"><a href="#cb38-108" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-109"><a href="#cb38-109" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save individual plot in main results directory</span></span>
<span id="cb38-110"><a href="#cb38-110" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_results_dir:</span>
<span id="cb38-111"><a href="#cb38-111" aria-hidden="true" tabindex="-1"></a>        plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"plot_11_top_genes_optimal_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb38-112"><a href="#cb38-112" aria-hidden="true" tabindex="-1"></a>        plt.savefig(plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb38-113"><a href="#cb38-113" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Top genes heatmap saved to: </span><span class="sc">{</span>plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb38-114"><a href="#cb38-114" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-115"><a href="#cb38-115" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb38-116"><a href="#cb38-116" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb38-117"><a href="#cb38-117" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Could not load data for optimal rank </span><span class="sc">{</span>optimal_rank <span class="cf">if</span> optimal_rank <span class="cf">else</span> <span class="st">'unknown'</span><span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/plot-top-genes-25.png" width="960"></p>
<div class="sourceCode" id="cb39"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="individual-plot-12-reconstruction-quality" class="level2">
<h2 class="anchored" data-anchor-id="individual-plot-12-reconstruction-quality">4.14 Individual Plot 12: Reconstruction Quality</h2>
<div class="sourceCode" id="cb40"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting</span></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>matplotlib.use(<span class="st">'Agg'</span>)  <span class="co"># Use non-interactive backend</span></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>plt.ioff()  <span class="co"># Turn off interactive mode</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory</span></span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the optimal rank from the rank selection summary</span></span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir:</span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a>    summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a>            rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> (from rank selection analysis)"</span>)</span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No rank selection summary found! Cannot determine optimal rank."</span>)</span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb40-38"><a href="#cb40-38" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb40-39"><a href="#cb40-39" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No main results directory found!"</span>)</span>
<span id="cb40-40"><a href="#cb40-40" aria-hidden="true" tabindex="-1"></a>    optimal_rank <span class="op">=</span> <span class="va">None</span></span>
<span id="cb40-41"><a href="#cb40-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-42"><a href="#cb40-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for the optimal rank data in rank_comparison directory</span></span>
<span id="cb40-43"><a href="#cb40-43" aria-hidden="true" tabindex="-1"></a>latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb40-44"><a href="#cb40-44" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir <span class="kw">and</span> optimal_rank:</span>
<span id="cb40-45"><a href="#cb40-45" aria-hidden="true" tabindex="-1"></a>    comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb40-46"><a href="#cb40-46" aria-hidden="true" tabindex="-1"></a>    optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb40-47"><a href="#cb40-47" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-48"><a href="#cb40-48" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb40-49"><a href="#cb40-49" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb40-50"><a href="#cb40-50" aria-hidden="true" tabindex="-1"></a>        result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb40-51"><a href="#cb40-51" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> result_dirs:</span>
<span id="cb40-52"><a href="#cb40-52" aria-hidden="true" tabindex="-1"></a>            latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb40-53"><a href="#cb40-53" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb40-54"><a href="#cb40-54" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb40-55"><a href="#cb40-55" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb40-56"><a href="#cb40-56" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb40-57"><a href="#cb40-57" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found!"</span>)</span>
<span id="cb40-58"><a href="#cb40-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-59"><a href="#cb40-59" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb40-60"><a href="#cb40-60" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-61"><a href="#cb40-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load data</span></span>
<span id="cb40-62"><a href="#cb40-62" aria-hidden="true" tabindex="-1"></a>    gene_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb40-63"><a href="#cb40-63" aria-hidden="true" tabindex="-1"></a>    species_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"species_factors.npy"</span>))</span>
<span id="cb40-64"><a href="#cb40-64" aria-hidden="true" tabindex="-1"></a>    time_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"time_factors.npy"</span>))</span>
<span id="cb40-65"><a href="#cb40-65" aria-hidden="true" tabindex="-1"></a>    reconstructed <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"reconstructed_tensor.npy"</span>))</span>
<span id="cb40-66"><a href="#cb40-66" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-67"><a href="#cb40-67" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load the transformed data</span></span>
<span id="cb40-68"><a href="#cb40-68" aria-hidden="true" tabindex="-1"></a>    data_file <span class="op">=</span> os.path.join(output_dir, <span class="st">"vst_counts_matrix_long_format.csv"</span>)</span>
<span id="cb40-69"><a href="#cb40-69" aria-hidden="true" tabindex="-1"></a>    df_long <span class="op">=</span> pd.read_csv(data_file)</span>
<span id="cb40-70"><a href="#cb40-70" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-71"><a href="#cb40-71" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create original tensor from the long format data</span></span>
<span id="cb40-72"><a href="#cb40-72" aria-hidden="true" tabindex="-1"></a>    original_pivot <span class="op">=</span> df_long.pivot_table(values<span class="op">=</span><span class="st">'expression'</span>, </span>
<span id="cb40-73"><a href="#cb40-73" aria-hidden="true" tabindex="-1"></a>                                       index<span class="op">=</span><span class="st">'gene'</span>, </span>
<span id="cb40-74"><a href="#cb40-74" aria-hidden="true" tabindex="-1"></a>                                       columns<span class="op">=</span>[<span class="st">'species'</span>, <span class="st">'timepoint'</span>], </span>
<span id="cb40-75"><a href="#cb40-75" aria-hidden="true" tabindex="-1"></a>                                       fill_value<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb40-76"><a href="#cb40-76" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-77"><a href="#cb40-77" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if we can safely reshape</span></span>
<span id="cb40-78"><a href="#cb40-78" aria-hidden="true" tabindex="-1"></a>    expected_size <span class="op">=</span> gene_factors.shape[<span class="dv">0</span>] <span class="op">*</span> species_factors.shape[<span class="dv">0</span>] <span class="op">*</span> time_factors.shape[<span class="dv">0</span>]</span>
<span id="cb40-79"><a href="#cb40-79" aria-hidden="true" tabindex="-1"></a>    actual_size <span class="op">=</span> original_pivot.size</span>
<span id="cb40-80"><a href="#cb40-80" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-81"><a href="#cb40-81" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">6</span>))</span>
<span id="cb40-82"><a href="#cb40-82" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-83"><a href="#cb40-83" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> actual_size <span class="op">==</span> expected_size:</span>
<span id="cb40-84"><a href="#cb40-84" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Reshape the pivot table to match tensor format</span></span>
<span id="cb40-85"><a href="#cb40-85" aria-hidden="true" tabindex="-1"></a>        n_genes, n_species, n_timepoints <span class="op">=</span> gene_factors.shape[<span class="dv">0</span>], species_factors.shape[<span class="dv">0</span>], time_factors.shape[<span class="dv">0</span>]</span>
<span id="cb40-86"><a href="#cb40-86" aria-hidden="true" tabindex="-1"></a>        original_reshaped <span class="op">=</span> original_pivot.values.reshape(n_genes, n_species, n_timepoints)</span>
<span id="cb40-87"><a href="#cb40-87" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb40-88"><a href="#cb40-88" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Calculate per-gene reconstruction error</span></span>
<span id="cb40-89"><a href="#cb40-89" aria-hidden="true" tabindex="-1"></a>        gene_errors <span class="op">=</span> np.mean(np.<span class="bu">abs</span>(original_reshaped <span class="op">-</span> reconstructed), axis<span class="op">=</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb40-90"><a href="#cb40-90" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb40-91"><a href="#cb40-91" aria-hidden="true" tabindex="-1"></a>        plt.hist(gene_errors, bins<span class="op">=</span><span class="dv">50</span>, alpha<span class="op">=</span><span class="fl">0.7</span>, edgecolor<span class="op">=</span><span class="st">'black'</span>)</span>
<span id="cb40-92"><a href="#cb40-92" aria-hidden="true" tabindex="-1"></a>        plt.xlabel(<span class="st">'Mean Absolute Error per Gene'</span>)</span>
<span id="cb40-93"><a href="#cb40-93" aria-hidden="true" tabindex="-1"></a>        plt.ylabel(<span class="st">'Number of Genes'</span>)</span>
<span id="cb40-94"><a href="#cb40-94" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="ss">f'Reconstruction Error Distribution - Optimal Rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb40-95"><a href="#cb40-95" aria-hidden="true" tabindex="-1"></a>        plt.axvline(np.mean(gene_errors), color<span class="op">=</span><span class="st">'red'</span>, linestyle<span class="op">=</span><span class="st">'--'</span>, </span>
<span id="cb40-96"><a href="#cb40-96" aria-hidden="true" tabindex="-1"></a>                   label<span class="op">=</span><span class="ss">f'Mean: </span><span class="sc">{</span>np<span class="sc">.</span>mean(gene_errors)<span class="sc">:.3f}</span><span class="ss">'</span>)</span>
<span id="cb40-97"><a href="#cb40-97" aria-hidden="true" tabindex="-1"></a>        plt.legend()</span>
<span id="cb40-98"><a href="#cb40-98" aria-hidden="true" tabindex="-1"></a>        plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb40-99"><a href="#cb40-99" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb40-100"><a href="#cb40-100" aria-hidden="true" tabindex="-1"></a>        <span class="co"># If reshape fails, show a simpler quality metric</span></span>
<span id="cb40-101"><a href="#cb40-101" aria-hidden="true" tabindex="-1"></a>        original_flat <span class="op">=</span> original_pivot.values.flatten()</span>
<span id="cb40-102"><a href="#cb40-102" aria-hidden="true" tabindex="-1"></a>        reconstructed_flat <span class="op">=</span> reconstructed.flatten()</span>
<span id="cb40-103"><a href="#cb40-103" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb40-104"><a href="#cb40-104" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Ensure both arrays are the same size by taking the minimum</span></span>
<span id="cb40-105"><a href="#cb40-105" aria-hidden="true" tabindex="-1"></a>        min_size <span class="op">=</span> <span class="bu">min</span>(<span class="bu">len</span>(original_flat), <span class="bu">len</span>(reconstructed_flat))</span>
<span id="cb40-106"><a href="#cb40-106" aria-hidden="true" tabindex="-1"></a>        original_flat <span class="op">=</span> original_flat[:min_size]</span>
<span id="cb40-107"><a href="#cb40-107" aria-hidden="true" tabindex="-1"></a>        reconstructed_flat <span class="op">=</span> reconstructed_flat[:min_size]</span>
<span id="cb40-108"><a href="#cb40-108" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb40-109"><a href="#cb40-109" aria-hidden="true" tabindex="-1"></a>        residuals <span class="op">=</span> original_flat <span class="op">-</span> reconstructed_flat</span>
<span id="cb40-110"><a href="#cb40-110" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb40-111"><a href="#cb40-111" aria-hidden="true" tabindex="-1"></a>        plt.hist(residuals, bins<span class="op">=</span><span class="dv">50</span>, alpha<span class="op">=</span><span class="fl">0.7</span>, edgecolor<span class="op">=</span><span class="st">'black'</span>)</span>
<span id="cb40-112"><a href="#cb40-112" aria-hidden="true" tabindex="-1"></a>        plt.xlabel(<span class="st">'Reconstruction Residuals'</span>)</span>
<span id="cb40-113"><a href="#cb40-113" aria-hidden="true" tabindex="-1"></a>        plt.ylabel(<span class="st">'Frequency'</span>)</span>
<span id="cb40-114"><a href="#cb40-114" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="ss">f'Reconstruction Residuals Distribution - Optimal Rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb40-115"><a href="#cb40-115" aria-hidden="true" tabindex="-1"></a>        plt.axvline(np.mean(residuals), color<span class="op">=</span><span class="st">'red'</span>, linestyle<span class="op">=</span><span class="st">'--'</span>, </span>
<span id="cb40-116"><a href="#cb40-116" aria-hidden="true" tabindex="-1"></a>                   label<span class="op">=</span><span class="ss">f'Mean: </span><span class="sc">{</span>np<span class="sc">.</span>mean(residuals)<span class="sc">:.3f}</span><span class="ss">'</span>)</span>
<span id="cb40-117"><a href="#cb40-117" aria-hidden="true" tabindex="-1"></a>        plt.axvline(<span class="dv">0</span>, color<span class="op">=</span><span class="st">'black'</span>, linestyle<span class="op">=</span><span class="st">':'</span>, alpha<span class="op">=</span><span class="fl">0.5</span>, label<span class="op">=</span><span class="st">'Perfect fit'</span>)</span>
<span id="cb40-118"><a href="#cb40-118" aria-hidden="true" tabindex="-1"></a>        plt.legend()</span>
<span id="cb40-119"><a href="#cb40-119" aria-hidden="true" tabindex="-1"></a>        plt.grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb40-120"><a href="#cb40-120" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-121"><a href="#cb40-121" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save individual plot in main results directory</span></span>
<span id="cb40-122"><a href="#cb40-122" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_results_dir:</span>
<span id="cb40-123"><a href="#cb40-123" aria-hidden="true" tabindex="-1"></a>        plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"plot_12_reconstruction_quality_optimal_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb40-124"><a href="#cb40-124" aria-hidden="true" tabindex="-1"></a>        plt.savefig(plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb40-125"><a href="#cb40-125" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Reconstruction quality plot saved to: </span><span class="sc">{</span>plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb40-126"><a href="#cb40-126" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-127"><a href="#cb40-127" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb40-128"><a href="#cb40-128" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb40-129"><a href="#cb40-129" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Could not load data for optimal rank </span><span class="sc">{</span>optimal_rank <span class="cf">if</span> optimal_rank <span class="cf">else</span> <span class="st">'unknown'</span><span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/plot-reconstruction-quality-27.png" width="960"></p>
<div class="sourceCode" id="cb41"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Suppress output</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="va">None</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="interpret-species-clustering-patterns" class="level2">
<h2 class="anchored" data-anchor-id="interpret-species-clustering-patterns">4.15 Interpret species clustering patterns</h2>
<div class="sourceCode" id="cb42"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scipy <span class="im">import</span> stats</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> glob</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Get output directory from R environment</span></span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory and optimal rank data (consistent with other plots)</span></span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>latest_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for main results directory</span></span>
<span id="cb42-16"><a href="#cb42-16" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="st">'main_results_dir'</span> <span class="kw">in</span> <span class="bu">globals</span>():</span>
<span id="cb42-17"><a href="#cb42-17" aria-hidden="true" tabindex="-1"></a>    main_results_dir <span class="op">=</span> <span class="bu">globals</span>()[<span class="st">'main_results_dir'</span>]</span>
<span id="cb42-18"><a href="#cb42-18" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb42-19"><a href="#cb42-19" aria-hidden="true" tabindex="-1"></a>    main_result_dirs <span class="op">=</span> glob.glob(os.path.join(output_dir, <span class="st">"20*"</span>))</span>
<span id="cb42-20"><a href="#cb42-20" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_result_dirs:</span>
<span id="cb42-21"><a href="#cb42-21" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="bu">max</span>(main_result_dirs)</span>
<span id="cb42-22"><a href="#cb42-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb42-23"><a href="#cb42-23" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb42-24"><a href="#cb42-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-25"><a href="#cb42-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the optimal rank from the rank selection summary (same as other plots)</span></span>
<span id="cb42-26"><a href="#cb42-26" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> main_results_dir:</span>
<span id="cb42-27"><a href="#cb42-27" aria-hidden="true" tabindex="-1"></a>    summary_file <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_selection_summary.json"</span>)</span>
<span id="cb42-28"><a href="#cb42-28" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> os.path.exists(summary_file):</span>
<span id="cb42-29"><a href="#cb42-29" aria-hidden="true" tabindex="-1"></a>        <span class="cf">with</span> <span class="bu">open</span>(summary_file, <span class="st">'r'</span>) <span class="im">as</span> f:</span>
<span id="cb42-30"><a href="#cb42-30" aria-hidden="true" tabindex="-1"></a>            rank_summary <span class="op">=</span> json.load(f)</span>
<span id="cb42-31"><a href="#cb42-31" aria-hidden="true" tabindex="-1"></a>        optimal_rank <span class="op">=</span> rank_summary[<span class="st">'suggested_rank'</span>]</span>
<span id="cb42-32"><a href="#cb42-32" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> (from rank selection analysis)"</span>)</span>
<span id="cb42-33"><a href="#cb42-33" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb42-34"><a href="#cb42-34" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Look for the optimal rank data in rank_comparison directory</span></span>
<span id="cb42-35"><a href="#cb42-35" aria-hidden="true" tabindex="-1"></a>        comparison_dir <span class="op">=</span> os.path.join(main_results_dir, <span class="st">"rank_comparison"</span>)</span>
<span id="cb42-36"><a href="#cb42-36" aria-hidden="true" tabindex="-1"></a>        optimal_rank_dir <span class="op">=</span> os.path.join(comparison_dir, <span class="ss">f"rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss">"</span>)</span>
<span id="cb42-37"><a href="#cb42-37" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb42-38"><a href="#cb42-38" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> os.path.exists(optimal_rank_dir):</span>
<span id="cb42-39"><a href="#cb42-39" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Find the timestamped results within the rank directory</span></span>
<span id="cb42-40"><a href="#cb42-40" aria-hidden="true" tabindex="-1"></a>            result_dirs <span class="op">=</span> glob.glob(os.path.join(optimal_rank_dir, <span class="st">"20*"</span>))</span>
<span id="cb42-41"><a href="#cb42-41" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> result_dirs:</span>
<span id="cb42-42"><a href="#cb42-42" aria-hidden="true" tabindex="-1"></a>                latest_dir <span class="op">=</span> <span class="bu">max</span>(result_dirs)</span>
<span id="cb42-43"><a href="#cb42-43" aria-hidden="true" tabindex="-1"></a>                <span class="bu">print</span>(<span class="ss">f"Using optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> data from: </span><span class="sc">{</span>latest_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb42-44"><a href="#cb42-44" aria-hidden="true" tabindex="-1"></a>            <span class="cf">else</span>:</span>
<span id="cb42-45"><a href="#cb42-45" aria-hidden="true" tabindex="-1"></a>                <span class="bu">print</span>(<span class="ss">f"No timestamped results found in rank_</span><span class="sc">{</span>optimal_rank<span class="sc">:02d}</span><span class="ss"> directory!"</span>)</span>
<span id="cb42-46"><a href="#cb42-46" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb42-47"><a href="#cb42-47" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Optimal rank </span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss"> directory not found!"</span>)</span>
<span id="cb42-48"><a href="#cb42-48" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb42-49"><a href="#cb42-49" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"No rank selection summary found! Cannot determine optimal rank."</span>)</span>
<span id="cb42-50"><a href="#cb42-50" aria-hidden="true" tabindex="-1"></a>        main_results_dir <span class="op">=</span> <span class="va">None</span></span>
<span id="cb42-51"><a href="#cb42-51" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb42-52"><a href="#cb42-52" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No main results directory found!"</span>)</span>
<span id="cb42-53"><a href="#cb42-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-54"><a href="#cb42-54" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> latest_dir:</span>
<span id="cb42-55"><a href="#cb42-55" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-56"><a href="#cb42-56" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load data</span></span>
<span id="cb42-57"><a href="#cb42-57" aria-hidden="true" tabindex="-1"></a>    gene_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"gene_factors.npy"</span>))</span>
<span id="cb42-58"><a href="#cb42-58" aria-hidden="true" tabindex="-1"></a>    species_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"species_factors.npy"</span>))</span>
<span id="cb42-59"><a href="#cb42-59" aria-hidden="true" tabindex="-1"></a>    time_factors <span class="op">=</span> np.load(os.path.join(latest_dir, <span class="st">"time_factors.npy"</span>))</span>
<span id="cb42-60"><a href="#cb42-60" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-61"><a href="#cb42-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load the transformed data</span></span>
<span id="cb42-62"><a href="#cb42-62" aria-hidden="true" tabindex="-1"></a>    data_file <span class="op">=</span> os.path.join(output_dir, <span class="st">"vst_counts_matrix_long_format.csv"</span>)</span>
<span id="cb42-63"><a href="#cb42-63" aria-hidden="true" tabindex="-1"></a>    df_long <span class="op">=</span> pd.read_csv(data_file)</span>
<span id="cb42-64"><a href="#cb42-64" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-65"><a href="#cb42-65" aria-hidden="true" tabindex="-1"></a>    species_names <span class="op">=</span> <span class="bu">sorted</span>(df_long[<span class="st">'species'</span>].unique())</span>
<span id="cb42-66"><a href="#cb42-66" aria-hidden="true" tabindex="-1"></a>    species_types <span class="op">=</span> [name.split(<span class="st">'-'</span>)[<span class="dv">0</span>] <span class="cf">for</span> name <span class="kw">in</span> species_names]</span>
<span id="cb42-67"><a href="#cb42-67" aria-hidden="true" tabindex="-1"></a>    unique_types <span class="op">=</span> <span class="bu">sorted</span>(<span class="bu">list</span>(<span class="bu">set</span>(species_types)))</span>
<span id="cb42-68"><a href="#cb42-68" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-69"><a href="#cb42-69" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"="</span><span class="op">*</span><span class="dv">60</span>)</span>
<span id="cb42-70"><a href="#cb42-70" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"SPECIES CLUSTERING INTERPRETATION"</span>)</span>
<span id="cb42-71"><a href="#cb42-71" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"="</span><span class="op">*</span><span class="dv">60</span>)</span>
<span id="cb42-72"><a href="#cb42-72" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-73"><a href="#cb42-73" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 1. Quantify species separation</span></span>
<span id="cb42-74"><a href="#cb42-74" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">1. SPECIES SEPARATION ANALYSIS:"</span>)</span>
<span id="cb42-75"><a href="#cb42-75" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-76"><a href="#cb42-76" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Group individuals by species type</span></span>
<span id="cb42-77"><a href="#cb42-77" aria-hidden="true" tabindex="-1"></a>    species_groups <span class="op">=</span> {}</span>
<span id="cb42-78"><a href="#cb42-78" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i, name <span class="kw">in</span> <span class="bu">enumerate</span>(species_names):</span>
<span id="cb42-79"><a href="#cb42-79" aria-hidden="true" tabindex="-1"></a>        stype <span class="op">=</span> name.split(<span class="st">'-'</span>)[<span class="dv">0</span>]</span>
<span id="cb42-80"><a href="#cb42-80" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> stype <span class="kw">not</span> <span class="kw">in</span> species_groups:</span>
<span id="cb42-81"><a href="#cb42-81" aria-hidden="true" tabindex="-1"></a>            species_groups[stype] <span class="op">=</span> []</span>
<span id="cb42-82"><a href="#cb42-82" aria-hidden="true" tabindex="-1"></a>        species_groups[stype].append(i)</span>
<span id="cb42-83"><a href="#cb42-83" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-84"><a href="#cb42-84" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate within-species vs between-species distances</span></span>
<span id="cb42-85"><a href="#cb42-85" aria-hidden="true" tabindex="-1"></a>    <span class="im">from</span> scipy.spatial.distance <span class="im">import</span> pdist, squareform</span>
<span id="cb42-86"><a href="#cb42-86" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-87"><a href="#cb42-87" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use all components for distance calculation</span></span>
<span id="cb42-88"><a href="#cb42-88" aria-hidden="true" tabindex="-1"></a>    distances <span class="op">=</span> squareform(pdist(species_factors, metric<span class="op">=</span><span class="st">'euclidean'</span>))</span>
<span id="cb42-89"><a href="#cb42-89" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-90"><a href="#cb42-90" aria-hidden="true" tabindex="-1"></a>    within_species_distances <span class="op">=</span> []</span>
<span id="cb42-91"><a href="#cb42-91" aria-hidden="true" tabindex="-1"></a>    between_species_distances <span class="op">=</span> []</span>
<span id="cb42-92"><a href="#cb42-92" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-93"><a href="#cb42-93" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i, species_i <span class="kw">in</span> <span class="bu">enumerate</span>(species_names):</span>
<span id="cb42-94"><a href="#cb42-94" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> j, species_j <span class="kw">in</span> <span class="bu">enumerate</span>(species_names):</span>
<span id="cb42-95"><a href="#cb42-95" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> i <span class="op">&lt;</span> j:  <span class="co"># Avoid duplicates</span></span>
<span id="cb42-96"><a href="#cb42-96" aria-hidden="true" tabindex="-1"></a>                type_i <span class="op">=</span> species_i.split(<span class="st">'-'</span>)[<span class="dv">0</span>]</span>
<span id="cb42-97"><a href="#cb42-97" aria-hidden="true" tabindex="-1"></a>                type_j <span class="op">=</span> species_j.split(<span class="st">'-'</span>)[<span class="dv">0</span>]</span>
<span id="cb42-98"><a href="#cb42-98" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb42-99"><a href="#cb42-99" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> type_i <span class="op">==</span> type_j:</span>
<span id="cb42-100"><a href="#cb42-100" aria-hidden="true" tabindex="-1"></a>                    within_species_distances.append(distances[i, j])</span>
<span id="cb42-101"><a href="#cb42-101" aria-hidden="true" tabindex="-1"></a>                <span class="cf">else</span>:</span>
<span id="cb42-102"><a href="#cb42-102" aria-hidden="true" tabindex="-1"></a>                    between_species_distances.append(distances[i, j])</span>
<span id="cb42-103"><a href="#cb42-103" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-104"><a href="#cb42-104" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"  Within-species distances: </span><span class="sc">{</span>np<span class="sc">.</span>mean(within_species_distances)<span class="sc">:.3f}</span><span class="ss"> ± </span><span class="sc">{</span>np<span class="sc">.</span>std(within_species_distances)<span class="sc">:.3f}</span><span class="ss">"</span>)</span>
<span id="cb42-105"><a href="#cb42-105" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"  Between-species distances: </span><span class="sc">{</span>np<span class="sc">.</span>mean(between_species_distances)<span class="sc">:.3f}</span><span class="ss"> ± </span><span class="sc">{</span>np<span class="sc">.</span>std(between_species_distances)<span class="sc">:.3f}</span><span class="ss">"</span>)</span>
<span id="cb42-106"><a href="#cb42-106" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-107"><a href="#cb42-107" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Statistical test</span></span>
<span id="cb42-108"><a href="#cb42-108" aria-hidden="true" tabindex="-1"></a>    t_stat, p_value <span class="op">=</span> stats.ttest_ind(between_species_distances, within_species_distances)</span>
<span id="cb42-109"><a href="#cb42-109" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"  T-test p-value: </span><span class="sc">{</span>p_value<span class="sc">:.3e}</span><span class="ss">"</span>)</span>
<span id="cb42-110"><a href="#cb42-110" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-111"><a href="#cb42-111" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> p_value <span class="op">&lt;</span> <span class="fl">0.05</span>:</span>
<span id="cb42-112"><a href="#cb42-112" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"  → Species ARE significantly separated (but effect might be small)"</span>)</span>
<span id="cb42-113"><a href="#cb42-113" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb42-114"><a href="#cb42-114" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"  → Species are NOT significantly separated"</span>)</span>
<span id="cb42-115"><a href="#cb42-115" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-116"><a href="#cb42-116" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 2. Variance decomposition</span></span>
<span id="cb42-117"><a href="#cb42-117" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">2. VARIANCE DECOMPOSITION:"</span>)</span>
<span id="cb42-118"><a href="#cb42-118" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-119"><a href="#cb42-119" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate total variance and partition it</span></span>
<span id="cb42-120"><a href="#cb42-120" aria-hidden="true" tabindex="-1"></a>    total_var <span class="op">=</span> np.var(species_factors, axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb42-121"><a href="#cb42-121" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-122"><a href="#cb42-122" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate within-species variance</span></span>
<span id="cb42-123"><a href="#cb42-123" aria-hidden="true" tabindex="-1"></a>    within_var <span class="op">=</span> np.zeros(species_factors.shape[<span class="dv">1</span>])</span>
<span id="cb42-124"><a href="#cb42-124" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> stype, indices <span class="kw">in</span> species_groups.items():</span>
<span id="cb42-125"><a href="#cb42-125" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="bu">len</span>(indices) <span class="op">&gt;</span> <span class="dv">1</span>:  <span class="co"># Need at least 2 individuals</span></span>
<span id="cb42-126"><a href="#cb42-126" aria-hidden="true" tabindex="-1"></a>            group_var <span class="op">=</span> np.var(species_factors[indices, :], axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb42-127"><a href="#cb42-127" aria-hidden="true" tabindex="-1"></a>            within_var <span class="op">+=</span> group_var <span class="op">*</span> <span class="bu">len</span>(indices) <span class="op">/</span> <span class="bu">len</span>(species_names)</span>
<span id="cb42-128"><a href="#cb42-128" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-129"><a href="#cb42-129" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Between-species variance</span></span>
<span id="cb42-130"><a href="#cb42-130" aria-hidden="true" tabindex="-1"></a>    between_var <span class="op">=</span> total_var <span class="op">-</span> within_var</span>
<span id="cb42-131"><a href="#cb42-131" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-132"><a href="#cb42-132" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate percentages</span></span>
<span id="cb42-133"><a href="#cb42-133" aria-hidden="true" tabindex="-1"></a>    within_percent <span class="op">=</span> (np.mean(within_var) <span class="op">/</span> np.mean(total_var)) <span class="op">*</span> <span class="dv">100</span></span>
<span id="cb42-134"><a href="#cb42-134" aria-hidden="true" tabindex="-1"></a>    between_percent <span class="op">=</span> (np.mean(between_var) <span class="op">/</span> np.mean(total_var)) <span class="op">*</span> <span class="dv">100</span></span>
<span id="cb42-135"><a href="#cb42-135" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-136"><a href="#cb42-136" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"  Within-species variance: </span><span class="sc">{</span>within_percent<span class="sc">:.1f}</span><span class="ss">%"</span>)</span>
<span id="cb42-137"><a href="#cb42-137" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"  Between-species variance: </span><span class="sc">{</span>between_percent<span class="sc">:.1f}</span><span class="ss">%"</span>)</span>
<span id="cb42-138"><a href="#cb42-138" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-139"><a href="#cb42-139" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 3. Components that best separate species</span></span>
<span id="cb42-140"><a href="#cb42-140" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">3. COMPONENTS WITH STRONGEST SPECIES EFFECTS:"</span>)</span>
<span id="cb42-141"><a href="#cb42-141" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-142"><a href="#cb42-142" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate F-statistics for each component</span></span>
<span id="cb42-143"><a href="#cb42-143" aria-hidden="true" tabindex="-1"></a>    f_stats <span class="op">=</span> []</span>
<span id="cb42-144"><a href="#cb42-144" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> comp <span class="kw">in</span> <span class="bu">range</span>(species_factors.shape[<span class="dv">1</span>]):</span>
<span id="cb42-145"><a href="#cb42-145" aria-hidden="true" tabindex="-1"></a>        groups <span class="op">=</span> [species_factors[indices, comp] <span class="cf">for</span> indices <span class="kw">in</span> species_groups.values()]</span>
<span id="cb42-146"><a href="#cb42-146" aria-hidden="true" tabindex="-1"></a>        f_stat, p_val <span class="op">=</span> stats.f_oneway(<span class="op">*</span>groups)</span>
<span id="cb42-147"><a href="#cb42-147" aria-hidden="true" tabindex="-1"></a>        f_stats.append((comp, f_stat, p_val))</span>
<span id="cb42-148"><a href="#cb42-148" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-149"><a href="#cb42-149" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Sort by F-statistic</span></span>
<span id="cb42-150"><a href="#cb42-150" aria-hidden="true" tabindex="-1"></a>    f_stats.sort(key<span class="op">=</span><span class="kw">lambda</span> x: x[<span class="dv">1</span>], reverse<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb42-151"><a href="#cb42-151" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-152"><a href="#cb42-152" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"  Top 5 components by species separation:"</span>)</span>
<span id="cb42-153"><a href="#cb42-153" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i, (comp, f_stat, p_val) <span class="kw">in</span> <span class="bu">enumerate</span>(f_stats[:<span class="dv">5</span>]):</span>
<span id="cb42-154"><a href="#cb42-154" aria-hidden="true" tabindex="-1"></a>        significance <span class="op">=</span> <span class="st">"***"</span> <span class="cf">if</span> p_val <span class="op">&lt;</span> <span class="fl">0.001</span> <span class="cf">else</span> <span class="st">"**"</span> <span class="cf">if</span> p_val <span class="op">&lt;</span> <span class="fl">0.01</span> <span class="cf">else</span> <span class="st">"*"</span> <span class="cf">if</span> p_val <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="cf">else</span> <span class="st">""</span></span>
<span id="cb42-155"><a href="#cb42-155" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"    Component </span><span class="sc">{</span>comp<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">: F=</span><span class="sc">{</span>f_stat<span class="sc">:.2f}</span><span class="ss">, p=</span><span class="sc">{</span>p_val<span class="sc">:.3e}</span><span class="ss"> </span><span class="sc">{</span>significance<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb42-156"><a href="#cb42-156" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-157"><a href="#cb42-157" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 4. Create detailed visualization</span></span>
<span id="cb42-158"><a href="#cb42-158" aria-hidden="true" tabindex="-1"></a>    fig, axes <span class="op">=</span> plt.subplots(<span class="dv">2</span>, <span class="dv">2</span>, figsize<span class="op">=</span>(<span class="dv">15</span>, <span class="dv">12</span>))</span>
<span id="cb42-159"><a href="#cb42-159" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-160"><a href="#cb42-160" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Distance distributions</span></span>
<span id="cb42-161"><a href="#cb42-161" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">0</span>, <span class="dv">0</span>].hist(within_species_distances, alpha<span class="op">=</span><span class="fl">0.7</span>, label<span class="op">=</span><span class="st">'Within-species'</span>, bins<span class="op">=</span><span class="dv">20</span>)</span>
<span id="cb42-162"><a href="#cb42-162" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">0</span>, <span class="dv">0</span>].hist(between_species_distances, alpha<span class="op">=</span><span class="fl">0.7</span>, label<span class="op">=</span><span class="st">'Between-species'</span>, bins<span class="op">=</span><span class="dv">20</span>)</span>
<span id="cb42-163"><a href="#cb42-163" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">0</span>, <span class="dv">0</span>].set_xlabel(<span class="st">'Euclidean Distance'</span>)</span>
<span id="cb42-164"><a href="#cb42-164" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">0</span>, <span class="dv">0</span>].set_ylabel(<span class="st">'Frequency'</span>)</span>
<span id="cb42-165"><a href="#cb42-165" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">0</span>, <span class="dv">0</span>].set_title(<span class="st">'Distance Distributions'</span>)</span>
<span id="cb42-166"><a href="#cb42-166" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">0</span>, <span class="dv">0</span>].legend()</span>
<span id="cb42-167"><a href="#cb42-167" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">0</span>, <span class="dv">0</span>].grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb42-168"><a href="#cb42-168" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-169"><a href="#cb42-169" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Variance decomposition pie chart</span></span>
<span id="cb42-170"><a href="#cb42-170" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">0</span>, <span class="dv">1</span>].pie([within_percent, between_percent], </span>
<span id="cb42-171"><a href="#cb42-171" aria-hidden="true" tabindex="-1"></a>                   labels<span class="op">=</span>[<span class="ss">f'Within-species</span><span class="ch">\n</span><span class="ss">(</span><span class="sc">{</span>within_percent<span class="sc">:.1f}</span><span class="ss">%)'</span>, </span>
<span id="cb42-172"><a href="#cb42-172" aria-hidden="true" tabindex="-1"></a>                          <span class="ss">f'Between-species</span><span class="ch">\n</span><span class="ss">(</span><span class="sc">{</span>between_percent<span class="sc">:.1f}</span><span class="ss">%)'</span>],</span>
<span id="cb42-173"><a href="#cb42-173" aria-hidden="true" tabindex="-1"></a>                   autopct<span class="op">=</span><span class="st">'</span><span class="sc">%1.1f%%</span><span class="st">'</span>, startangle<span class="op">=</span><span class="dv">90</span>)</span>
<span id="cb42-174"><a href="#cb42-174" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">0</span>, <span class="dv">1</span>].set_title(<span class="st">'Variance Decomposition'</span>)</span>
<span id="cb42-175"><a href="#cb42-175" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-176"><a href="#cb42-176" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Box plots for top separating component</span></span>
<span id="cb42-177"><a href="#cb42-177" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> f_stats:</span>
<span id="cb42-178"><a href="#cb42-178" aria-hidden="true" tabindex="-1"></a>        top_comp <span class="op">=</span> f_stats[<span class="dv">0</span>][<span class="dv">0</span>]</span>
<span id="cb42-179"><a href="#cb42-179" aria-hidden="true" tabindex="-1"></a>        species_data <span class="op">=</span> []</span>
<span id="cb42-180"><a href="#cb42-180" aria-hidden="true" tabindex="-1"></a>        species_labels <span class="op">=</span> []</span>
<span id="cb42-181"><a href="#cb42-181" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb42-182"><a href="#cb42-182" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> stype, indices <span class="kw">in</span> species_groups.items():</span>
<span id="cb42-183"><a href="#cb42-183" aria-hidden="true" tabindex="-1"></a>            species_data.extend(species_factors[indices, top_comp])</span>
<span id="cb42-184"><a href="#cb42-184" aria-hidden="true" tabindex="-1"></a>            species_labels.extend([stype] <span class="op">*</span> <span class="bu">len</span>(indices))</span>
<span id="cb42-185"><a href="#cb42-185" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb42-186"><a href="#cb42-186" aria-hidden="true" tabindex="-1"></a>        df_comp <span class="op">=</span> pd.DataFrame({<span class="st">'Species'</span>: species_labels, <span class="st">'Factor_Value'</span>: species_data})</span>
<span id="cb42-187"><a href="#cb42-187" aria-hidden="true" tabindex="-1"></a>        sns.boxplot(data<span class="op">=</span>df_comp, x<span class="op">=</span><span class="st">'Species'</span>, y<span class="op">=</span><span class="st">'Factor_Value'</span>, ax<span class="op">=</span>axes[<span class="dv">1</span>, <span class="dv">0</span>])</span>
<span id="cb42-188"><a href="#cb42-188" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">1</span>, <span class="dv">0</span>].set_title(<span class="ss">f'Component </span><span class="sc">{</span>top_comp<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss"> by Species</span><span class="ch">\n</span><span class="ss">(Best Separating Component)'</span>)</span>
<span id="cb42-189"><a href="#cb42-189" aria-hidden="true" tabindex="-1"></a>        axes[<span class="dv">1</span>, <span class="dv">0</span>].grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb42-190"><a href="#cb42-190" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-191"><a href="#cb42-191" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Individual count per species</span></span>
<span id="cb42-192"><a href="#cb42-192" aria-hidden="true" tabindex="-1"></a>    species_counts <span class="op">=</span> [<span class="bu">len</span>(indices) <span class="cf">for</span> indices <span class="kw">in</span> species_groups.values()]</span>
<span id="cb42-193"><a href="#cb42-193" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">1</span>, <span class="dv">1</span>].bar(species_groups.keys(), species_counts)</span>
<span id="cb42-194"><a href="#cb42-194" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">1</span>, <span class="dv">1</span>].set_xlabel(<span class="st">'Species'</span>)</span>
<span id="cb42-195"><a href="#cb42-195" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">1</span>, <span class="dv">1</span>].set_ylabel(<span class="st">'Number of Individuals'</span>)</span>
<span id="cb42-196"><a href="#cb42-196" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">1</span>, <span class="dv">1</span>].set_title(<span class="st">'Sample Size per Species'</span>)</span>
<span id="cb42-197"><a href="#cb42-197" aria-hidden="true" tabindex="-1"></a>    axes[<span class="dv">1</span>, <span class="dv">1</span>].grid(<span class="va">True</span>, alpha<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb42-198"><a href="#cb42-198" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-199"><a href="#cb42-199" aria-hidden="true" tabindex="-1"></a>    plt.tight_layout()</span>
<span id="cb42-200"><a href="#cb42-200" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-201"><a href="#cb42-201" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save the interpretation plot in the main results directory (consistent with other plots)</span></span>
<span id="cb42-202"><a href="#cb42-202" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> main_results_dir:</span>
<span id="cb42-203"><a href="#cb42-203" aria-hidden="true" tabindex="-1"></a>        interp_plot_file <span class="op">=</span> os.path.join(main_results_dir, <span class="ss">f"species_clustering_interpretation_optimal_rank_</span><span class="sc">{</span>optimal_rank<span class="sc">}</span><span class="ss">.png"</span>)</span>
<span id="cb42-204"><a href="#cb42-204" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb42-205"><a href="#cb42-205" aria-hidden="true" tabindex="-1"></a>        interp_plot_file <span class="op">=</span> os.path.join(latest_dir, <span class="st">"species_clustering_interpretation.png"</span>)</span>
<span id="cb42-206"><a href="#cb42-206" aria-hidden="true" tabindex="-1"></a>    plt.savefig(interp_plot_file, dpi<span class="op">=</span><span class="dv">300</span>, bbox_inches<span class="op">=</span><span class="st">'tight'</span>)</span>
<span id="cb42-207"><a href="#cb42-207" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Interpretation plot saved to: </span><span class="sc">{</span>interp_plot_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb42-208"><a href="#cb42-208" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-209"><a href="#cb42-209" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb42-210"><a href="#cb42-210" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-211"><a href="#cb42-211" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 5. Biological interpretation</span></span>
<span id="cb42-212"><a href="#cb42-212" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">4. BIOLOGICAL INTERPRETATION:"</span>)</span>
<span id="cb42-213"><a href="#cb42-213" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-214"><a href="#cb42-214" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> between_percent <span class="op">&lt;</span> <span class="dv">20</span>:</span>
<span id="cb42-215"><a href="#cb42-215" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"  → STRONG INDIVIDUAL VARIATION: Gene expression varies more between"</span>)</span>
<span id="cb42-216"><a href="#cb42-216" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"    individuals than between species. This suggests:"</span>)</span>
<span id="cb42-217"><a href="#cb42-217" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"    • Individual phenotypic plasticity is high"</span>)</span>
<span id="cb42-218"><a href="#cb42-218" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"    • Species have similar fundamental gene regulatory networks"</span>)</span>
<span id="cb42-219"><a href="#cb42-219" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"    • Environmental or developmental factors affect individuals"</span>)</span>
<span id="cb42-220"><a href="#cb42-220" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"      more than species identity"</span>)</span>
<span id="cb42-221"><a href="#cb42-221" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> between_percent <span class="op">&lt;</span> <span class="dv">40</span>:</span>
<span id="cb42-222"><a href="#cb42-222" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"  → MODERATE SPECIES EFFECTS: Some species differences exist but"</span>)</span>
<span id="cb42-223"><a href="#cb42-223" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"    individual variation is still dominant. This suggests:"</span>)</span>
<span id="cb42-224"><a href="#cb42-224" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"    • Species have some distinct expression patterns"</span>)</span>
<span id="cb42-225"><a href="#cb42-225" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"    • Individual variation remains important"</span>)</span>
<span id="cb42-226"><a href="#cb42-226" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"    • Mixed evolutionary/ecological pressures"</span>)</span>
<span id="cb42-227"><a href="#cb42-227" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb42-228"><a href="#cb42-228" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"  → STRONG SPECIES EFFECTS: Clear species-specific expression"</span>)</span>
<span id="cb42-229"><a href="#cb42-229" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"    patterns exist. This suggests:"</span>)</span>
<span id="cb42-230"><a href="#cb42-230" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"    • Species have distinct gene regulatory strategies"</span>)</span>
<span id="cb42-231"><a href="#cb42-231" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"    • Strong evolutionary divergence"</span>)</span>
<span id="cb42-232"><a href="#cb42-232" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"    • Species-specific adaptations"</span>)</span>
<span id="cb42-233"><a href="#cb42-233" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-234"><a href="#cb42-234" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">  Sample sizes: </span><span class="sc">{</span><span class="bu">dict</span>(<span class="bu">zip</span>(species_groups.keys(), [<span class="bu">len</span>(indices) <span class="cf">for</span> indices <span class="kw">in</span> species_groups.values()]))<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb42-235"><a href="#cb42-235" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-236"><a href="#cb42-236" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="bu">any</span>(<span class="bu">len</span>(indices) <span class="op">&lt;</span> <span class="dv">5</span> <span class="cf">for</span> indices <span class="kw">in</span> species_groups.values()):</span>
<span id="cb42-237"><a href="#cb42-237" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"  ⚠️  WARNING: Some species have small sample sizes, which can"</span>)</span>
<span id="cb42-238"><a href="#cb42-238" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"     affect the ability to detect species-specific patterns."</span>)</span>
<span id="cb42-239"><a href="#cb42-239" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-240"><a href="#cb42-240" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb42-241"><a href="#cb42-241" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"No results directories found!"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Using optimal rank 5 (from rank selection analysis)
## Using optimal rank 5 data from: ../output/13.00-multiomics-stdm/20251015_190157/rank_comparison/rank_05/20251015_190351
## ============================================================
## SPECIES CLUSTERING INTERPRETATION
## ============================================================
## 
## 1. SPECIES SEPARATION ANALYSIS:
##   Within-species distances: 1.007 ± 1.225
##   Between-species distances: 3.390 ± 0.860
##   T-test p-value: 4.930e-78
##   → Species ARE significantly separated (but effect might be small)
## 
## 2. VARIANCE DECOMPOSITION:
##   Within-species variance: 25.4%
##   Between-species variance: 74.6%
## 
## 3. COMPONENTS WITH STRONGEST SPECIES EFFECTS:
##   Top 5 components by species separation:
##     Component 1: F=2220.60, p=1.113e-30 ***
##     Component 2: F=1124.54, p=1.003e-26 ***
##     Component 4: F=39.46, p=9.691e-09 ***
##     Component 5: F=3.78, p=3.575e-02 *
##     Component 3: F=0.80, p=4.590e-01 
## (array([39., 32., 22., 11.,  5.,  4.,  3.,  1.,  0.,  0.,  0.,  0.,  0.,
##         0.,  7.,  2.,  0.,  0.,  0.,  9.]), array([0.10695279, 0.33056048, 0.55416817, 0.77777586, 1.00138356,
##        1.22499125, 1.44859894, 1.67220664, 1.89581433, 2.11942202,
##        2.34302971, 2.56663741, 2.7902451 , 3.01385279, 3.23746049,
##        3.46106818, 3.68467587, 3.90828356, 4.13189126, 4.35549895,
##        4.57910664]), &lt;BarContainer object of 20 artists&gt;)
## (array([10., 32., 45., 41., 19., 11., 16., 15., 17., 17., 19., 15.,  6.,
##         3.,  3.,  3.,  4., 12.,  7.,  5.]), array([2.21788468, 2.38773948, 2.55759428, 2.72744908, 2.89730388,
##        3.06715869, 3.23701349, 3.40686829, 3.57672309, 3.74657789,
##        3.91643269, 4.08628749, 4.25614229, 4.4259971 , 4.5958519 ,
##        4.7657067 , 4.9355615 , 5.1054163 , 5.2752711 , 5.4451259 ,
##        5.6149807 ]), &lt;BarContainer object of 20 artists&gt;)
## Text(0.5, 0, 'Euclidean Distance')
## Text(0, 0.5, 'Frequency')
## Text(0.5, 1.0, 'Distance Distributions')
## &lt;matplotlib.legend.Legend object at 0x7c874132a690&gt;
## ([&lt;matplotlib.patches.Wedge object at 0x7c874136ab70&gt;, &lt;matplotlib.patches.Wedge object at 0x7c874208a300&gt;], [Text(-0.7877726729698648, 0.7677331670065548, 'Within-species\n(25.4%)'), Text(0.7877729231290416, -0.7677329103175957, 'Between-species\n(74.6%)')], [Text(-0.42969418525628983, 0.41876354563993895, '25.4%'), Text(0.42969432170674987, -0.4187634056277794, '74.6%')])
## Text(0.5, 1.0, 'Variance Decomposition')
## &lt;Axes: xlabel='Species', ylabel='Factor_Value'&gt;
## Text(0.5, 1.0, 'Component 1 by Species\n(Best Separating Component)')
## &lt;BarContainer object of 3 artists&gt;
## Text(0.5, 0, 'Species')
## Text(0, 0.5, 'Number of Individuals')
## Text(0.5, 1.0, 'Sample Size per Species')
## 
## Interpretation plot saved to: ../output/13.00-multiomics-stdm/20251015_190157/species_clustering_interpretation_optimal_rank_5.png
## 
## 4. BIOLOGICAL INTERPRETATION:
##   → STRONG SPECIES EFFECTS: Clear species-specific expression
##     patterns exist. This suggests:
##     • Species have distinct gene regulatory strategies
##     • Strong evolutionary divergence
##     • Species-specific adaptations
## 
##   Sample sizes: {'ACR': 10, 'POC': 10, 'POR': 10}</code></pre>
<p><img src="13.00-multiomics-stdm_files/figure-gfm/interpret-clustering-29.png" width="1440"></p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/robertslab\.github\.io\/sams-notebook");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>