<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.552">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sam White">
<meta name="dcterms.date" content="2023-12-06">

<title>Sam’s Notebook - Data Investigation - Mysterious Case of Excessive Exons in CEABIGR</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-7483T61T11"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-7483T61T11', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Sam’s Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" target="_blank">
 <span class="dropdown-text">Rstudio Server - Raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu/Atumefaciens/" target="_blank">
 <span class="dropdown-text">My Gannet Directory</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" target="_blank">
 <span class="dropdown-text">Steven’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shedurkin.github.io/Roberts-LabNotebook/" target="_blank">
 <span class="dropdown-text">Kathleen’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://meganewing.github.io/mewing-notebook/" target="_blank">
 <span class="dropdown-text">Megan’s Notebook</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/sams-notebook" target="_blank">
 <span class="dropdown-text">Notebook repo</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" target="_blank">
 <span class="dropdown-text">Lab Organization (GitHub)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" target="_blank">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" target="_blank"> <i class="bi bi-calendar-week" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-right">
      <h1 class="title">Data Investigation - Mysterious Case of Excessive Exons in CEABIGR</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">CEABIGR</div>
                <div class="quarto-category">Crassostrea viriginica</div>
                <div class="quarto-category">Eastern oyster</div>
                <div class="quarto-category">2023</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-right">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Sam White </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">December 6, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#investigate-genome-featuresc_virginica-3.0_gnomon_exon-geneid.bed" id="toc-investigate-genome-featuresc_virginica-3.0_gnomon_exon-geneid.bed" class="nav-link active" data-scroll-target="#investigate-genome-featuresc_virginica-3.0_gnomon_exon-geneid.bed">Investigate <code>genome-features/C_virginica-3.0_Gnomon_exon-geneID.bed</code></a>
  <ul>
  <li><a href="#inspect-exon-geneid-mapping-file" id="toc-inspect-exon-geneid-mapping-file" class="nav-link" data-scroll-target="#inspect-exon-geneid-mapping-file">Inspect exon-geneID mapping file</a></li>
  <li><a href="#count-exons-per-gene-in-c_virginica-3.0_gnomon_exon-geneid.bed" id="toc-count-exons-per-gene-in-c_virginica-3.0_gnomon_exon-geneid.bed" class="nav-link" data-scroll-target="#count-exons-per-gene-in-c_virginica-3.0_gnomon_exon-geneid.bed">Count exons per gene in <code>C_virginica-3.0_Gnomon_exon-geneID.bed</code></a></li>
  </ul></li>
  <li><a href="#inspect-ncbi-gff" id="toc-inspect-ncbi-gff" class="nav-link" data-scroll-target="#inspect-ncbi-gff">Inspect NCBI GFF</a>
  <ul>
  <li><a href="#decompress-ncbi-gff" id="toc-decompress-ncbi-gff" class="nav-link" data-scroll-target="#decompress-ncbi-gff">Decompress NCBI GFF</a></li>
  <li><a href="#search-for-genes-with-high-number-of-exons" id="toc-search-for-genes-with-high-number-of-exons" class="nav-link" data-scroll-target="#search-for-genes-with-high-number-of-exons">Search for gene(s) with high number of exons</a></li>
  </ul></li>
  <li><a href="#inspect-output19-exon-expressions12m-exon_expression.tab" id="toc-inspect-output19-exon-expressions12m-exon_expression.tab" class="nav-link" data-scroll-target="#inspect-output19-exon-expressions12m-exon_expression.tab">Inspect <code>output/19-exon-expression/S12M-exon_expression.tab</code></a>
  <ul>
  <li><a href="#count-colums-in-output19-exon-expressions12m-exon_expression.tab" id="toc-count-colums-in-output19-exon-expressions12m-exon_expression.tab" class="nav-link" data-scroll-target="#count-colums-in-output19-exon-expressions12m-exon_expression.tab">Count colums in <code>output/19-exon-expression/S12M-exon_expression.tab</code></a></li>
  <li><a href="#compare-number-of-exons-in-gene-loc111119012-between-s12m-exon_expression.tab-and-gcf_002022765.2_c_virginica-3.0_genomic.gff" id="toc-compare-number-of-exons-in-gene-loc111119012-between-s12m-exon_expression.tab-and-gcf_002022765.2_c_virginica-3.0_genomic.gff" class="nav-link" data-scroll-target="#compare-number-of-exons-in-gene-loc111119012-between-s12m-exon_expression.tab-and-gcf_002022765.2_c_virginica-3.0_genomic.gff">Compare number of exons in gene-LOC111119012 between <code>S12M-exon_expression.tab</code> and <code>GCF_002022765.2_C_virginica-3.0_genomic.gff</code></a></li>
  </ul></li>
  <li><a href="#compare-total-exons-in-e_data.ctab-and-gcf_002022765.2_c_virginica-3.0_genomic.gff" id="toc-compare-total-exons-in-e_data.ctab-and-gcf_002022765.2_c_virginica-3.0_genomic.gff" class="nav-link" data-scroll-target="#compare-total-exons-in-e_data.ctab-and-gcf_002022765.2_c_virginica-3.0_genomic.gff">Compare total exons in <code>e_data.ctab</code> and <code>GCF_002022765.2_C_virginica-3.0_genomic.gff</code></a>
  <ul>
  <li><a href="#count-exons-in-s12m-e_data.ctab." id="toc-count-exons-in-s12m-e_data.ctab." class="nav-link" data-scroll-target="#count-exons-in-s12m-e_data.ctab.">Count exons in S12M <code>e_data.ctab</code>.</a></li>
  <li><a href="#count-exons-in-gcf_002022765.2_c_virginica-3.0_genomic.gff" id="toc-count-exons-in-gcf_002022765.2_c_virginica-3.0_genomic.gff" class="nav-link" data-scroll-target="#count-exons-in-gcf_002022765.2_c_virginica-3.0_genomic.gff">Count exons in <code>GCF_002022765.2_C_virginica-3.0_genomic.gff</code></a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-right" id="quarto-document-content">





<p>After Steven <a href="https://github.com/sr320/ceabigr/issues/86#issuecomment-1816557386">ran through some prelimiary exon expression analyses</a> (GitHub Isse), he noticed that there were some genes with a seemingly excessive number of exons (like over 800!). Seeing that, I’ve decided to do some investigation to see if these gene feature annotations stem from the source (i.e.&nbsp;from NCBI GFF) or were introduced accidentally through some subsequent data analysis/manipulations.</p>
<p>First, let’s look at a file which appears to provide mappings of GeneIDs and exons:</p>
<section id="investigate-genome-featuresc_virginica-3.0_gnomon_exon-geneid.bed" class="level1">
<h1>Investigate <code>genome-features/C_virginica-3.0_Gnomon_exon-geneID.bed</code></h1>
<section id="inspect-exon-geneid-mapping-file" class="level2">
<h2 class="anchored" data-anchor-id="inspect-exon-geneid-mapping-file">Inspect exon-geneID mapping file</h2>
<p>This might be a poor place to start, as this file exists in a directory with no README. As such, I’m not certain of the source of this file…</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> /home/sam/gitrepos/ceabigr/genome-features/C_virginica-3.0_Gnomon_exon-geneID.bed</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>NC_035780.1 13578   13603   gene-LOC111116054
NC_035780.1 14237   14290   gene-LOC111116054
NC_035780.1 14557   14594   gene-LOC111116054
NC_035780.1 28961   29073   gene-LOC111126949
NC_035780.1 30524   31557   gene-LOC111126949
NC_035780.1 31736   31887   gene-LOC111126949
NC_035780.1 31977   32565   gene-LOC111126949
NC_035780.1 32959   33324   gene-LOC111126949
NC_035780.1 66869   66897   gene-LOC111110729
NC_035780.1 64123   64334   gene-LOC111110729</code></pre>
</div>
</div>
<p>This appears to have all exons start/stop locations for each gene. At least, that’s my assumption, based off of the filename.</p>
</section>
<section id="count-exons-per-gene-in-c_virginica-3.0_gnomon_exon-geneid.bed" class="level2">
<h2 class="anchored" data-anchor-id="count-exons-per-gene-in-c_virginica-3.0_gnomon_exon-geneid.bed">Count exons per gene in <code>C_virginica-3.0_Gnomon_exon-geneID.bed</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'{print $4}'</span> /home/sam/gitrepos/ceabigr/genome-features/C_virginica-3.0_Gnomon_exon-geneID.bed <span class="dt">\</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">uniq</span> <span class="at">--count</span> <span class="dt">\</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">sort</span> <span class="at">--numeric-sort</span> <span class="at">--reverse</span>  <span class="at">-k1,1</span> <span class="dt">\</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">head</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   2889 gene-LOC111108212
   2600 gene-LOC111125463
   2237 gene-LOC111125330
   1559 gene-LOC111130297
   1557 gene-LOC111112605
   1555 gene-LOC111128998
   1540 gene-LOC111122636
   1506 gene-LOC111107250
   1384 gene-LOC111125942
   1383 gene-LOC111133666</code></pre>
</div>
</div>
<p>Well, this shows us that this file indeed has many, many genes with <em>thousands</em> of exons! This is highly dubious, so let’s glance at the NCBI GFF to see if this is “real” or not.</p>
</section>
</section>
<section id="inspect-ncbi-gff" class="level1">
<h1>Inspect NCBI GFF</h1>
<section id="decompress-ncbi-gff" class="level2">
<h2 class="anchored" data-anchor-id="decompress-ncbi-gff">Decompress NCBI GFF</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gunzip</span> <span class="at">--keep</span> /home/sam/gitrepos/ceabigr/genome-features/GCF_002022765.2_C_virginica-3.0_genomic.gff.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="search-for-genes-with-high-number-of-exons" class="level2">
<h2 class="anchored" data-anchor-id="search-for-genes-with-high-number-of-exons">Search for gene(s) with high number of exons</h2>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Exons in the GFF do <em>not</em> refer to the full parent gene ID!</p>
<p>Need to use truncated version. E.g. <code>LOC111108212</code></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="at">-F</span> <span class="st">"\t"</span> <span class="st">'$3 == "exon"'</span> /home/sam/gitrepos/ceabigr/genome-features/GCF_002022765.2_C_virginica-3.0_genomic.gff <span class="dt">\</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">grep</span> <span class="st">"gene=LOC111110729"</span> <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>5</code></pre>
</div>
</div>
<p>So, the exon counts in <code>genome-features/C_virginica-3.0_Gnomon_exon-geneID.bed</code> are indeed artificial!</p>
<p>Now that I’ve investigated that, I probably should’ve just started with the input files that Steven was using in the first place. However, this certainly identifies a problematic file that should likely be removed from the repo…</p>
<p>Let’s look at the data Steven was working with and see what we can figure out.</p>
</section>
</section>
<section id="inspect-output19-exon-expressions12m-exon_expression.tab" class="level1">
<h1>Inspect <code>output/19-exon-expression/S12M-exon_expression.tab</code></h1>
<section id="count-colums-in-output19-exon-expressions12m-exon_expression.tab" class="level2">
<h2 class="anchored" data-anchor-id="count-colums-in-output19-exon-expressions12m-exon_expression.tab">Count colums in <code>output/19-exon-expression/S12M-exon_expression.tab</code></h2>
<p>Due to the layout of this file (a column for each possible number of exons across <em>all</em> genes), the head command will produce an unwieldy output. Instead, we’ll just count the number of columns (fields) to get an idea of what we’re dealing with.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>We subtract <code>1</code> to account for the first column</p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>S12M.columns <span class="ot">&lt;-</span> <span class="fu">system</span>(<span class="st">"awk '{print NF-1}' /home/sam/gitrepos/ceabigr/output/19-exon-expression/S12M-exon_expression.tab | sort --unique"</span>,</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">intern =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Well, we’re seeing that there is a <em>maximum</em> number of exons across <em>all</em> genes: 462</p>
<p>Next let’s see if we can ID the gene with 462 exons.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>S12M.df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/home/sam/gitrepos/ceabigr/output/19-exon-expression/S12M-exon_expression.tab"</span>,</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(S12M.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>'data.frame':   38838 obs. of  463 variables:
 $ gene_name: chr  "gene-LOC111116054" "gene-LOC111126949" "gene-LOC111110729" "gene-LOC111112434" ...
 $ exon_1   : int  130 22 282 24 69 416 193 23 23 3 ...
 $ exon_2   : int  103 180 63 40 39 NA 106 173 173 1 ...
 $ exon_3   : int  9 35 55 123 86 NA NA 3 NA 2 ...
 $ exon_4   : int  NA 110 11 NA 21 NA NA 1 NA 2 ...
 $ exon_5   : int  NA 47 NA NA 85 NA NA 2 NA NA ...
 $ exon_6   : int  NA NA NA NA NA NA NA 2 NA NA ...
 $ exon_7   : int  NA NA NA NA NA NA NA 75 NA NA ...
 $ exon_8   : int  NA NA NA NA NA NA NA 214 NA NA ...
 $ exon_9   : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_10  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_11  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_12  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_13  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_14  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_15  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_16  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_17  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_18  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_19  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_20  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_21  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_22  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_23  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_24  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_25  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_26  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_27  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_28  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_29  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_30  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_31  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_32  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_33  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_34  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_35  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_36  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_37  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_38  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_39  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_40  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_41  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_42  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_43  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_44  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_45  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_46  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_47  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_48  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_49  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_50  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_51  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_52  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_53  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_54  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_55  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_56  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_57  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_58  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_59  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_60  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_61  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_62  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_63  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_64  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_65  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_66  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_67  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_68  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_69  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_70  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_71  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_72  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_73  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_74  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_75  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_76  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_77  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_78  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_79  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_80  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_81  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_82  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_83  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_84  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_85  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_86  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_87  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_88  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_89  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_90  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_91  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_92  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_93  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_94  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_95  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_96  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_97  : int  NA NA NA NA NA NA NA NA NA NA ...
 $ exon_98  : int  NA NA NA NA NA NA NA NA NA NA ...
  [list output truncated]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> S12M.df <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(exon_462) <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"[0-9]"</span>, exon_462))</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(result<span class="sc">$</span>gene_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "gene-LOC111119012"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>selected_gene_names <span class="ot">&lt;-</span> result<span class="sc">$</span>gene_name</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Okay, we’ve identified gene-LOC111119012 as the gene which contains 462. Let’s see if this matches the original NCBI GFF.</p>
</section>
<section id="compare-number-of-exons-in-gene-loc111119012-between-s12m-exon_expression.tab-and-gcf_002022765.2_c_virginica-3.0_genomic.gff" class="level2">
<h2 class="anchored" data-anchor-id="compare-number-of-exons-in-gene-loc111119012-between-s12m-exon_expression.tab-and-gcf_002022765.2_c_virginica-3.0_genomic.gff">Compare number of exons in gene-LOC111119012 between <code>S12M-exon_expression.tab</code> and <code>GCF_002022765.2_C_virginica-3.0_genomic.gff</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="at">-F</span> <span class="st">"\t"</span> <span class="st">'$3 == "exon"'</span> /home/sam/gitrepos/ceabigr/genome-features/GCF_002022765.2_C_virginica-3.0_genomic.gff <span class="dt">\</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">grep</span> <span class="st">"gene=LOC111119012"</span> <span class="dt">\</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>268</code></pre>
</div>
</div>
<p>Okay, they do <em>not</em> match, which points to an issue with <code>S12M-exon_expression.tab</code>! I generated that file in <a href="https://github.com/sr320/ceabigr/blob/main/code/19-exon-expression.Rmd"><code>19-exon-expression.Rmd</code></a> using the Ballgown exon file (<code>e_data.ctab</code>) file as a source. Let’s check that first before I re-examine <a href="https://github.com/sr320/ceabigr/blob/main/code/19-exon-expression.Rmd"><code>19-exon-expression.Rmd</code></a>.</p>
</section>
</section>
<section id="compare-total-exons-in-e_data.ctab-and-gcf_002022765.2_c_virginica-3.0_genomic.gff" class="level1">
<h1>Compare total exons in <code>e_data.ctab</code> and <code>GCF_002022765.2_C_virginica-3.0_genomic.gff</code></h1>
<section id="count-exons-in-s12m-e_data.ctab." class="level2">
<h2 class="anchored" data-anchor-id="count-exons-in-s12m-e_data.ctab.">Count exons in S12M <code>e_data.ctab</code>.</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>S12M.edata.df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/home/sam/gitrepos/ceabigr/data/ballgown/S12M/e_data.ctab"</span>,</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(S12M.edata.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>'data.frame':   352757 obs. of  12 variables:
 $ e_id   : int  1 2 3 4 5 6 7 8 9 10 ...
 $ chr    : chr  "NC_007175.2" "NC_007175.2" "NC_007175.2" "NC_007175.2" ...
 $ strand : chr  "+" "+" "+" "+" ...
 $ start  : int  1 1710 8250 2645 3430 3499 3578 3647 4897 5673 ...
 $ end    : int  1623 2430 8997 3429 3495 3567 3646 4859 5589 5743 ...
 $ rcount : int  3406 26400 15868 1336 52 53 41 2586 1190 38 ...
 $ ucount : int  3406 26400 15868 1336 52 53 41 2585 1190 38 ...
 $ mrcount: num  3406 26400 15868 1336 52 ...
 $ cov    : num  232.4 4148.4 2338.9 181.2 68.7 ...
 $ cov_sd : num  159.7 3089.9 2148.5 67.5 3.4 ...
 $ mcov   : num  232.4 4148.4 2338.9 181.2 68.7 ...
 $ mcov_sd: num  159.7 3089.9 2148.5 67.5 3.4 ...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>exon.count.S12M.edata <span class="ot">&lt;-</span> <span class="fu">nrow</span>(S12M.edata.df)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(exon.count.S12M.edata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 352757</code></pre>
</div>
</div>
</section>
<section id="count-exons-in-gcf_002022765.2_c_virginica-3.0_genomic.gff" class="level2">
<h2 class="anchored" data-anchor-id="count-exons-in-gcf_002022765.2_c_virginica-3.0_genomic.gff">Count exons in <code>GCF_002022765.2_C_virginica-3.0_genomic.gff</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>exon.count.ncbi.gff <span class="ot">&lt;-</span> <span class="fu">system</span>(<span class="st">"awk -F '</span><span class="sc">\t</span><span class="st">' '$3 == </span><span class="sc">\"</span><span class="st">exon</span><span class="sc">\"</span><span class="st">' /home/sam/gitrepos/ceabigr/genome-features/GCF_002022765.2_C_virginica-3.0_genomic.gff | wc -l"</span>, <span class="at">intern =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(exon.count.ncbi.gff)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "731916"</code></pre>
</div>
</div>
<p>Number of exons in <code>e_data.ctab</code>: 352757</p>
<p>Number of exons in NCBI GFF: 731916</p>
<p>So, we’re seeing that S12M <code>e_data.ctab</code> has ~50% of the total number of exons in the <a href="https://en.wikipedia.org/wiki/Eastern_oyster"><em>Crassostrea virginica</em> (Eastern oyster)</a> genome (as annotated by NCBI). This makes sense to me, as the data we’re using is RNAseq. So, not all genes are going to be expressed and, thus, we won’t have data for each gene/exon in the genome.</p>
<p>Regardless, based off of the difference in exon counts of gene-LOC111119012 in <code>GCF_002022765.2_C_virginica-3.0_genomic.gff</code> and <code>S12M-exon_expression.tab</code>, I will revisit <a href="https://github.com/sr320/ceabigr/blob/main/code/19-exon-expression.Rmd"><code>19-exon-expression.Rmd</code></a> to see where things might have gone wrong.</p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/robertslab\.github\.io\/sams-notebook");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>