---
author: Sam White
toc-title: Contents
toc-depth: 5
toc-location: left
title: qPCR Analysis - M.gigas PolyIC Data from Valentinas Project
date: '2026-02-02'
draft: false
engine: knitr
code-fold: show
categories: 
  - "2026"
  - "qPCR"
  - "Pacific oyster"
  - "Magallana gigas"
  - "Crassostrea gigas"
---
# INTRO

This is an analysis of qPCR data from _Magallana gigas_ (Pacific oyster) samples treated with PolyIC (a synthetic analog of double-stranded RNA that mimics viral infection) and subjected to different stress types (Temperature, Mechanical, Control). The goal was to assess gene expression changes in response to these treatments.

The qPCR data was generated by me on January 27-29, 2026, as part of Valentina's project on immune responses in oysters. Eight genes were targeted: ATP Synthase, Citrate Synthase, DNMT1, HSP70, HSP90, cGAS, and VIPERIN. GAPDH was used as the normalizing gene. Notebook entries documenting the qPCR runs and data exports can be found in the following posts:

- [2026-01-27-qPCRs - M.gigas Valentina PolyIC cGAS, citrate_synthase, and DNMT1](../2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/)
- [2026-01-28-qPCRs - M.gigas Valentina PolyIC ATP synthase, GAPDH, HSP70, and HSP90](../2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/)
- [2026-01-29-qPCR - M.gigas Valentina PolyIC VIPERIN](../2026-01-29-qPCR---M.gigas-Valentina-PolyIC-VIPERIN/)

I also managed to track down the qPCRs run on the RNA I isolated on [isolated on 20250804](../2025-08-04-RNA-Isolation-and-Quantification---C.gigas-PolyIC-Mechanical-Trials-for-Valentina-Using-Quick-DNA-RNA-MiniPrep-Plus-Kit/). After RNA isolation, they were passed to Valentina for reverse transcription and qPCR. I believe those qPCRs were run/analyzed by Aakriti [on 20250814](https://genefish.wordpress.com/2025/08/14/results-from-qpcr-with-valentina/) (Notebook entry). I found the files here:

- https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/admin_2025-08-14%2011-11-18_Connect_ATPase_vv.pcrd

- https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/admin_2025-08-12%2014-19-37_Connect_HSP70_vv.pcrd

- https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/admin_2025-08-12%2011-18-15_Connect_citrate_VV.pcrd

- https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/admin_2025-08-08%2014-18-00_Connect_GAPDH.pcrd


::: {.callout-important}
I have not taken the time to evaluate the qPCRs run by Aakriti in detail. I've relied on the processing below to filter out samples with poor technical replicates. Maybe these can be revisited at a later date to see if any samples need to be re-run.
:::


```{r setup}
#| hold: true
#| label: setup
#| results: hold
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = TRUE,        # Evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  comment = ""         # Prevents appending '##' to beginning of lines in code output
)

library(tidyverse)
library(ggplot2)
library(car)
library(emmeans)
library(multcomp)
```

# LOAD DATA
```{r load-data}
#| hold: true
#| results: hold
#| label: load-data
# Load qPCR data

# Define file URLs and target names in a structured format
file_list <- list(
  # CITRATE SYNTHASE
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_15-32-08_Connect-citrate_synthase-01--Quantification-Cq_Results.csv", 
       target = "Citrate.Synthase", gene = "citrate_synthase"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_15-30-47_CFX96-citrate_synthase-02-Quantification-Cq_Results.csv", 
       target = "Citrate.Synthase", gene = "citrate_synthase"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_16-31-12_Connect-citrate_synthase-03-Quantification-Cq_Results.csv", 
       target = "Citrate.Synthase", gene = "citrate_synthase"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-02-02-qPCR-Analysis---M.gigas-PolyIC-Data-from-Valentinas-Project/admin_2025-08-12%2011-18-15_Connect_citrate_VV-SJW-Quantification-Cq_Results.csv", 
       target = "Citrate.Synthase", gene = "citrate_synthase"),
  
  # DNMT1
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_13-34-02_Connect-DNMT1-01-Quantification-Cq_Results.csv", 
       target = "DNMT1", gene = "DNMT1"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_13-29-34_CFX96-DNMT1-02%20Quantification-Cq_Results.csv", 
       target = "DNMT1", gene = "DNMT1"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_14-33-31_Connect-DNMT1-03-Quantification-Cq_Results.csv", 
       target = "DNMT1", gene = "DNMT1"),
  
  # cGAS
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_11-25-14_Connect-cGAS-01-Quantification-Cq_Results.csv", 
       target = "cGAS", gene = "cGAS"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_11-49-08_CFX96-cGAS-02-Quantification-Cq_Results.csv", 
       target = "cGAS", gene = "cGAS"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_12-35-04_Connect-cGAS-03-QuantificationCq_Results.csv", 
       target = "cGAS", gene = "cGAS"),
  
  # ATP SYNTHASE
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_09-56-00_Connect-ATP_synthase-01-QuantificationCq_Results.csv", 
       target = "ATP.Synthase", gene = "ATP_synthase"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_10-13-15_CFX96-ATP_synthase-02-Quantification-Cq_Results.csv", 
       target = "ATP.Synthase", gene = "ATP_synthase"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_10-55-49_Connect-ATP_synthase-03-Quantification-Cq_Results.csv", 
       target = "ATP.Synthase", gene = "ATP_synthase"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-02-02-qPCR-Analysis---M.gigas-PolyIC-Data-from-Valentinas-Project/admin_2025-08-14%2011-11-18_Connect_ATPase_vv-SJW-Quantification-Cq_Results.csv", 
       target = "ATP.Synthase", gene = "ATP_synthase"),
  
  # GAPDH
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_12-06-16_Connect-GAPDH-01-Quantification-Cq_Results.csv", 
       target = "GAPDH", gene = "GAPDH"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_12-07-20_CFX96-GAPDH-02-Quantification-Cq_Results.csv", 
       target = "GAPDH", gene = "GAPDH"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_13-05-59_Connect-GAPDH-03-Quantification-Cq_Results.csv", 
       target = "GAPDH", gene = "GAPDH"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-02-02-qPCR-Analysis---M.gigas-PolyIC-Data-from-Valentinas-Project/admin_2025-08-08%2014-18-00_Connect_GAPD-SJW-Quantification-Cq_Results.csv", 
       target = "GAPDH", gene = "GAPDH"),
  
  # HSP70
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_14-03-35_Connect-HSP70-01-Quantification-Cq_Results.csv", 
       target = "HSP70", gene = "HSP70"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_14-25-31_CFX96-HSP70-02-Quantification-Cq_Results.csv", 
       target = "HSP70", gene = "HSP70"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_15-00-57_Connect-HSP70-03-Quantification-Cq_Results.csv", 
       target = "HSP70", gene = "HSP70"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-02-02-qPCR-Analysis---M.gigas-PolyIC-Data-from-Valentinas-Project/admin_2025-08-12%2014-19-37_Connect_HSP70_vv-SJW-Quantification-Cq_Results.csv", 
       target = "HSP70", gene = "HSP70"),
  
  # HSP90
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28-16-06-43_Connect-HSP90-01-QuantificationCq_Results.csv", 
       target = "HSP90", gene = "HSP90"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_16-07-47_CFX96-HSP90-02-Quantification-Cq_Results.csv", 
       target = "HSP90", gene = "HSP90"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_17-07-09_Connect-HSP90-03-Quantification-Cq_Results.csv", 
       target = "HSP90", gene = "HSP90"),
  
  # VIPERIN
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-29-qPCR---M.gigas-Valentina-PolyIC-VIPERIN/sam_2026-01-29_07-21-16_Connect-VIPERIN-01-QuantificationCq_Results.csv", 
       target = "VIPERIN", gene = "VIPERIN"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-29-qPCR---M.gigas-Valentina-PolyIC-VIPERIN/sam_2026-01-29_07-39-06_CFX96-VIPERIN-02-Quantification-Cq_Results.csv", 
       target = "VIPERIN", gene = "VIPERIN"),
  list(url = "https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-29-qPCR---M.gigas-Valentina-PolyIC-VIPERIN/sam_2026-01-29_08-24-59_Connect-VIPERIN-03-QuantificationCq_Results.csv", 
       target = "VIPERIN", gene = "VIPERIN")
)

# Load all files and combine automatically
qpcr_data <- map_df(file_list, function(file_info) {
  read.csv(file_info$url) %>%
    mutate(Target = file_info$target)
})

```

# Functions

## Calculate delta Cq

Normalized to designated normalizing gene
```{r function-delta-Cq}
#| hold: true
#| label: function-delta-Cq
calculate_delta_Cq <- function(df) {
  df <- df %>%
    group_by(Sample) %>%
    mutate(delta_Cq = Cq.Mean - Cq.Mean[Target == "GAPDH"]) %>%
    ungroup()
  
  return(df)
}
```

# Data Cleaning

## Remove NTC samples

```{r remove-ntc}
#| hold: true
#| label: remove-ntc
# Remove rows with Sample name "NTC" or empty/NA Sample names
qpcr_data <- qpcr_data %>%
  filter(Sample != "NTC" & !is.na(Sample) & Sample != "")

str(qpcr_data)
```

## Check for high standard deviations

```{r check-std-dev}
#| hold: true
#| label: check-std-dev
# Filter out rows where Cq.Std..Dev is NA
qpcr_data <- qpcr_data[!is.na(qpcr_data$Cq.Std..Dev), ]

# Filter rows where Cq.Std..Dev is greater than 0.5
high_cq_std_dev <- qpcr_data[qpcr_data$Cq.Std..Dev > 0.5, ]

# Print the filtered rows with specified columns, without row names
print(high_cq_std_dev[, c("Target", "Sample", "Cq", "Cq.Std..Dev")], row.names = FALSE)
```

## Remove outlier technical replicates

```{r remove-outliers}
#| hold: true
#| label: remove-outliers
# Group by Sample and Target, then filter out the outlier replicate
qpcr_filtered <- qpcr_data %>%
  group_by(Sample, Target) %>%
  filter(n() == 1 | abs(Cq - median(Cq, na.rm = TRUE)) <= 0.5) %>%
  ungroup()

str(qpcr_filtered)
```

# Group and Summarize Data

```{r group-summarize}
#| hold: true
#| label: group-summarize
# Group by Sample and Target, then summarize to get unique rows for each sample
grouped_df <- qpcr_filtered %>%
  group_by(Sample, Target) %>%
  summarise(
    Cq.Mean = mean(Cq, na.rm = TRUE),
    Cq.Std..Dev = sd(Cq, na.rm = TRUE),
    .groups = 'drop'
  )

str(grouped_df)
```

# Parse Sample Names and Add Treatment Columns

Sample naming convention: Letter + Number + P (optional) + Treatment

- Letter: Biological replicate
- Number: Replicate number
- P: PolyIC treatment (absent if no PolyIC)
- C/T/M: Control/Temperature/Mechanical stress

```{r parse-samples}
#| hold: true
#| label: parse-samples
# Parse sample names to extract treatment information
grouped_df <- grouped_df %>%
  mutate(
    # Extract biological replicate (first letter)
    bio_replicate = str_extract(Sample, "^[A-Z]"),
    
    # Extract replicate number
    replicate_num = str_extract(Sample, "\\d+"),
    
    # Check if PolyIC treatment (contains P)
    polyIC = ifelse(str_detect(Sample, "P"), "PolyIC", "No_PolyIC"),
    
    # Extract stress type (last letter: C, T, or M)
    stress_type = case_when(
      str_detect(Sample, "C$") ~ "Control",
      str_detect(Sample, "T$") ~ "Temperature",
      str_detect(Sample, "M$") ~ "Mechanical",
      TRUE ~ NA_character_
    ),
    
    # Create combined treatment group
    treatment_group = paste(polyIC, stress_type, sep = "_")
  )

str(grouped_df)
```

## View treatment groups

```{r view-groups}
#| hold: true
#| label: view-groups
# Check the unique treatment groups
unique(grouped_df$treatment_group)

# Count samples per treatment group
grouped_df %>%
  group_by(treatment_group, Target) %>%
  summarise(n_samples = n(), .groups = 'drop') %>%
  pivot_wider(names_from = Target, values_from = n_samples) %>%
  knitr::kable()
```

# Calculate Delta Cq

```{r calculate-delta-cq}
#| hold: true
#| label: calculate-delta-cq
# Calculate delta Cq by subtracting GAPDH Cq.Mean from each corresponding Sample Cq.Mean
delta_Cq_df <- calculate_delta_Cq(grouped_df)

# Filter out normalizing gene (GAPDH), since no need to compare normalizing gene to itself
delta_Cq_df <- delta_Cq_df %>%
  filter(Target != "GAPDH")

str(delta_Cq_df)
```

# Statistical Comparisons

## T-tests for treatment comparisons

### Comparison 1: Basic Control vs Stressors (No PolyIC)

```{r ttest-control-vs-stress-no-polyIC}
#| hold: true
#| label: ttest-control-vs-stress-no-polyIC
# Filter data for samples without PolyIC treatment
no_polyIC_data <- delta_Cq_df %>%
  filter(polyIC == "No_PolyIC")

# Perform t-tests for each target comparing Control vs each stressor
comparison1_results <- no_polyIC_data %>%
  filter(stress_type %in% c("Control", "Temperature", "Mechanical")) %>%
  group_by(Target) %>%
  summarise(
    # Control vs Temperature
    temp_pvalue = if(sum(stress_type == "Control") > 0 & sum(stress_type == "Temperature") > 0) {
      t.test(delta_Cq[stress_type == "Control"], 
             delta_Cq[stress_type == "Temperature"])$p.value
    } else NA_real_,
    
    # Control vs Mechanical
    mech_pvalue = if(sum(stress_type == "Control") > 0 & sum(stress_type == "Mechanical") > 0) {
      t.test(delta_Cq[stress_type == "Control"], 
             delta_Cq[stress_type == "Mechanical"])$p.value
    } else NA_real_,
    
    .groups = 'drop'
  )

print(comparison1_results)
```

#### Significant genes (p ≤ 0.05)

```{r sig-genes-no-polyIC}
#| hold: true
#| label: sig-genes-no-polyIC
# Identify genes with significant p-values for Temperature comparison
sig_temp <- comparison1_results %>%
  ungroup() %>%
  filter(temp_pvalue <= 0.05 & !is.na(temp_pvalue)) %>%
  dplyr::select(Target, pvalue = temp_pvalue) %>%
  mutate(comparison = "Control vs Temperature")

# Identify genes with significant p-values for Mechanical comparison
sig_mech <- comparison1_results %>%
  ungroup() %>%
  filter(mech_pvalue <= 0.05 & !is.na(mech_pvalue)) %>%
  dplyr::select(Target, pvalue = mech_pvalue) %>%
  mutate(comparison = "Control vs Mechanical")

# Combine and display
sig_genes_comp1 <- bind_rows(
  sig_temp,
  sig_mech
)

if(nrow(sig_genes_comp1) > 0) {
  print("Genes with p ≤ 0.05:")
  sig_genes_comp1 %>%
    mutate(pvalue = format(pvalue, scientific = TRUE, digits = 3)) %>%
    knitr::kable()
} else {
  print("No genes had p-values ≤ 0.05")
}
```

### Comparison 2: Basic Control vs PolyIC Control

```{r ttest-control-vs-polyIC-control}
#| hold: true
#| results: hold
#| label: ttest-control-vs-polyIC-control
# Filter data for control samples only (with and without PolyIC)
control_data <- delta_Cq_df %>%
  filter(stress_type == "Control")

# Perform t-tests for each target comparing No_PolyIC vs PolyIC controls
comparison2_results <- control_data %>%
  group_by(Target) %>%
  summarise(
    n_no_polyIC = sum(polyIC == "No_PolyIC"),
    n_polyIC = sum(polyIC == "PolyIC"),
    
    pvalue = if(sum(polyIC == "No_PolyIC") > 0 & sum(polyIC == "PolyIC") > 0) {
      t.test(delta_Cq[polyIC == "No_PolyIC"], 
             delta_Cq[polyIC == "PolyIC"])$p.value
    } else NA_real_,
    
    mean_no_polyIC = mean(delta_Cq[polyIC == "No_PolyIC"], na.rm = TRUE),
    sd_no_polyIC = sd(delta_Cq[polyIC == "No_PolyIC"], na.rm = TRUE),
    mean_polyIC = mean(delta_Cq[polyIC == "PolyIC"], na.rm = TRUE),
    sd_polyIC = sd(delta_Cq[polyIC == "PolyIC"], na.rm = TRUE),
    mean_diff = mean_polyIC - mean_no_polyIC,
    
    .groups = 'drop'
  )

print(comparison2_results)
```

#### Significant genes (p ≤ 0.05)

```{r sig-genes-polyIC-control}
#| hold: true
#| label: sig-genes-polyIC-control
# Identify genes with significant p-values
sig_genes_comp2 <- comparison2_results %>%
  filter(pvalue <= 0.05 & !is.na(pvalue)) %>%
  dplyr::select(Target, pvalue)

if(nrow(sig_genes_comp2) > 0) {
  print("Genes with p ≤ 0.05:")
  sig_genes_comp2 %>%
    mutate(pvalue = format(pvalue, scientific = TRUE, digits = 3)) %>%
    knitr::kable()
} else {
  print("No genes had p-values ≤ 0.05")
}
```

### Comparison 3: PolyIC Control vs PolyIC Stressors

```{r ttest-polyIC-control-vs-stress}
#| hold: true
#| label: ttest-polyIC-control-vs-stress
# Filter data for PolyIC samples only
polyIC_data <- delta_Cq_df %>%
  filter(polyIC == "PolyIC")

# Perform t-tests for each target comparing PolyIC Control vs each stressor
comparison3_results <- polyIC_data %>%
  filter(stress_type %in% c("Control", "Temperature", "Mechanical")) %>%
  group_by(Target) %>%
  summarise(
    # PolyIC Control vs Temperature
    temp_pvalue = if(sum(stress_type == "Control") > 0 & sum(stress_type == "Temperature") > 0) {
      t.test(delta_Cq[stress_type == "Control"], 
             delta_Cq[stress_type == "Temperature"])$p.value
    } else NA_real_,
    
    # PolyIC Control vs Mechanical
    mech_pvalue = if(sum(stress_type == "Control") > 0 & sum(stress_type == "Mechanical") > 0) {
      t.test(delta_Cq[stress_type == "Control"], 
             delta_Cq[stress_type == "Mechanical"])$p.value
    } else NA_real_,
    
    .groups = 'drop'
  )

print(comparison3_results)
```

#### Significant genes (p ≤ 0.05)

```{r sig-genes-polyIC-stress}
#| hold: true
#| label: sig-genes-polyIC-stress
# Identify genes with significant p-values for Temperature comparison
sig_temp_polyIC <- comparison3_results %>%
  filter(temp_pvalue <= 0.05 & !is.na(temp_pvalue)) %>%
  dplyr::select(Target, pvalue = temp_pvalue) %>%
  mutate(comparison = "PolyIC Control vs Temperature")

# Identify genes with significant p-values for Mechanical comparison
sig_mech_polyIC <- comparison3_results %>%
  filter(mech_pvalue <= 0.05 & !is.na(mech_pvalue)) %>%
  dplyr::select(Target, pvalue = mech_pvalue) %>%
  mutate(comparison = "PolyIC Control vs Mechanical")

# Combine and display
sig_genes_comp3 <- bind_rows(
  sig_temp_polyIC,
  sig_mech_polyIC
)

if(nrow(sig_genes_comp3) > 0) {
  print("Genes with p ≤ 0.05:")
  sig_genes_comp3 %>%
    mutate(pvalue = format(pvalue, scientific = TRUE, digits = 3)) %>%
    knitr::kable()
} else {
  print("No genes had p-values ≤ 0.05")
}
```

## ANOVA Models

Two-way ANOVA (PolyIC × Stress Type) for each gene to comprehensively test treatment effects and interactions.

```{r anova-models-run}
#| hold: true
#| label: anova-models-run

# Run ANOVA models for all genes and store emmeans objects for plotting
# Detailed assumption checks and interpretations are in the "ANOVA Models" section below

# ATP Synthase
model_atp <- delta_Cq_df %>%
  filter(Target == "ATP.Synthase") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)
if(any(summary(model_atp)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_atp <- emmeans(model_atp, ~ polyIC * stress_type)
}

# Citrate Synthase
model_citrate <- delta_Cq_df %>%
  filter(Target == "Citrate.Synthase") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)
if(any(summary(model_citrate)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_citrate <- emmeans(model_citrate, ~ polyIC * stress_type)
}

# DNMT1
model_dnmt1 <- delta_Cq_df %>%
  filter(Target == "DNMT1") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)
if(any(summary(model_dnmt1)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_dnmt1 <- emmeans(model_dnmt1, ~ polyIC * stress_type)
}

# HSP70
model_hsp70 <- delta_Cq_df %>%
  filter(Target == "HSP70") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)
if(any(summary(model_hsp70)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_hsp70 <- emmeans(model_hsp70, ~ polyIC * stress_type)
}

# HSP90
model_hsp90 <- delta_Cq_df %>%
  filter(Target == "HSP90") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)
if(any(summary(model_hsp90)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_hsp90 <- emmeans(model_hsp90, ~ polyIC * stress_type)
}

# cGAS
model_cgas <- delta_Cq_df %>%
  filter(Target == "cGAS") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)
if(any(summary(model_cgas)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_cgas <- emmeans(model_cgas, ~ polyIC * stress_type)
}

# VIPERIN
model_viperin <- delta_Cq_df %>%
  filter(Target == "VIPERIN") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)
if(any(summary(model_viperin)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_viperin <- emmeans(model_viperin, ~ polyIC * stress_type)
}

cat("ANOVA models created for all genes. Emmeans objects stored for post-hoc testing and plotting.\n")
cat("See 'ANOVA Models' section below for detailed results, assumption checks, and interpretations.\n")
```

# Visualizations

## Delta Cq Boxplots

### Comparison 1: Control vs Stressors (No PolyIC)

```{r boxplot-no-polyIC}
#| hold: true
#| label: boxplot-no-polyIC
#| fig.width: 10
#| fig.height: 6

# Prepare significance annotations from t-test results
if(exists("sig_genes_comp1") && nrow(sig_genes_comp1) > 0) {
  # Extract significant genes for Temperature
  sig_temp_genes <- sig_genes_comp1 %>%
    filter(comparison == "Control vs Temperature") %>%
    pull(Target)
  
  # Extract significant genes for Mechanical
  sig_mech_genes <- sig_genes_comp1 %>%
    filter(comparison == "Control vs Mechanical") %>%
    pull(Target)
  
  # Create annotation data
  sig_annotations_comp1 <- data.frame(
    Target = character(),
    stress_type = character(),
    label = character(),
    stringsAsFactors = FALSE
  )
  
  # Add Temperature annotations
  if(length(sig_temp_genes) > 0) {
    sig_annotations_comp1 <- rbind(sig_annotations_comp1,
      data.frame(Target = sig_temp_genes,
                 stress_type = "Temperature",
                 label = "*"))
  }
  
  # Add Mechanical annotations
  if(length(sig_mech_genes) > 0) {
    sig_annotations_comp1 <- rbind(sig_annotations_comp1,
      data.frame(Target = sig_mech_genes,
                 stress_type = "Mechanical",
                 label = "*"))
  }
  
  # Get y positions for annotations
  if(nrow(sig_annotations_comp1) > 0) {
    y_pos_comp1 <- no_polyIC_data %>%
      group_by(Target, stress_type) %>%
      summarise(max_y = max(delta_Cq, na.rm = TRUE), .groups = "drop")
    
    sig_annotations_comp1 <- sig_annotations_comp1 %>%
      left_join(y_pos_comp1, by = c("Target", "stress_type")) %>%
      mutate(y_pos = max_y + 0.5)
  }
} else {
  sig_annotations_comp1 <- data.frame()
}

# Create plot
p <- no_polyIC_data %>%
  ggplot(aes(x = stress_type, y = delta_Cq, fill = stress_type)) +
  geom_boxplot() +
  facet_wrap(~Target, scales = "free_y") +
  scale_fill_manual(values = c("Control" = "darkgray", "Temperature" = "salmon", "Mechanical" = "steelblue")) +
  theme_bw() +
  labs(
    title = "Delta Cq: Control vs Stressors (No PolyIC Treatment)",
    x = "Stress Type",
    y = "Delta Cq",
    fill = "Stress Type"
  ) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Add annotations if any exist
if(nrow(sig_annotations_comp1) > 0) {
  p <- p + 
    geom_text(data = sig_annotations_comp1, 
              aes(x = stress_type, y = y_pos, label = label),
              inherit.aes = FALSE, size = 6, vjust = 0, color = "darkorange") +
    labs(subtitle = "Asterisks (*) indicate significant differences from Control (t-test p ≤ 0.05)",
         caption = paste("Significant comparisons:", paste(unique(sig_genes_comp1$Target), collapse = ", ")))
} else {
  p <- p + labs(subtitle = "No significant differences detected in t-tests (all p > 0.05)")
}

print(p)
```

### Comparison 2: Control vs PolyIC Control

```{r boxplot-control-comparison}
#| hold: true
#| label: boxplot-control-comparison
#| fig.width: 10
#| fig.height: 6

# Prepare significance annotations from t-test results
if(exists("sig_genes_comp2") && nrow(sig_genes_comp2) > 0) {
  sig_annotations_comp2 <- data.frame(
    Target = sig_genes_comp2$Target,
    polyIC = "PolyIC",
    label = "*",
    stringsAsFactors = FALSE
  )
  
  # Get y positions for annotations
  y_pos_comp2 <- control_data %>%
    group_by(Target, polyIC) %>%
    summarise(max_y = max(delta_Cq, na.rm = TRUE), .groups = "drop")
  
  sig_annotations_comp2 <- sig_annotations_comp2 %>%
    left_join(y_pos_comp2, by = c("Target", "polyIC")) %>%
    mutate(y_pos = max_y + 0.5)
} else {
  sig_annotations_comp2 <- data.frame()
}

# Create plot
p <- control_data %>%
  ggplot(aes(x = polyIC, y = delta_Cq, fill = polyIC)) +
  geom_boxplot() +
  facet_wrap(~Target, scales = "free_y") +
  scale_fill_manual(values = c("No_PolyIC" = "darkgray", "PolyIC" = "salmon")) +
  coord_cartesian(clip = "off") +
  theme_bw() +
  labs(
    title = "Delta Cq: Basic Control vs PolyIC Control",
    x = "PolyIC Treatment",
    y = "Delta Cq",
    fill = "Treatment"
  ) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    plot.margin = margin(t = 20, r = 10, b = 10, l = 10)
  )

# Add annotations if any exist
if(nrow(sig_annotations_comp2) > 0) {
  # Expand y-axis limits to ensure annotations are visible
  y_limits_comp2 <- control_data %>%
    group_by(Target) %>%
    summarise(
      y_min = min(delta_Cq, na.rm = TRUE) - 0.5,
      y_max = max(delta_Cq, na.rm = TRUE) + 1.5,
      .groups = "drop"
    )
  
  p <- p + 
    geom_blank(data = y_limits_comp2, aes(x = 1, y = y_max), inherit.aes = FALSE) +
    geom_text(data = sig_annotations_comp2, 
              aes(x = polyIC, y = y_pos, label = label),
              inherit.aes = FALSE, size = 6, vjust = 0, color = "darkorange") +
    labs(subtitle = "Asterisks (*) indicate significant differences (t-test p ≤ 0.05)",
         caption = paste("Genes with significant PolyIC effect:", paste(sig_genes_comp2$Target, collapse = ", ")))
} else {
  p <- p + labs(subtitle = "No significant differences detected in t-tests (all p > 0.05)")
}

print(p)
```

### Comparison 3: PolyIC Control vs PolyIC Stressors

```{r boxplot-polyIC-stress}
#| hold: true
#| label: boxplot-polyIC-stress
#| fig.width: 10
#| fig.height: 6

# Prepare significance annotations from t-test results
if(exists("sig_genes_comp3") && nrow(sig_genes_comp3) > 0) {
  # Extract significant genes for Temperature
  sig_temp_genes <- sig_genes_comp3 %>%
    filter(comparison == "PolyIC Control vs Temperature") %>%
    pull(Target)
  
  # Extract significant genes for Mechanical
  sig_mech_genes <- sig_genes_comp3 %>%
    filter(comparison == "PolyIC Control vs Mechanical") %>%
    pull(Target)
  
  # Create annotation data
  sig_annotations_comp3 <- data.frame(
    Target = character(),
    stress_type = character(),
    label = character(),
    stringsAsFactors = FALSE
  )
  
  # Add Temperature annotations
  if(length(sig_temp_genes) > 0) {
    sig_annotations_comp3 <- rbind(sig_annotations_comp3,
      data.frame(Target = sig_temp_genes,
                 stress_type = "Temperature",
                 label = "*"))
  }
  
  # Add Mechanical annotations
  if(length(sig_mech_genes) > 0) {
    sig_annotations_comp3 <- rbind(sig_annotations_comp3,
      data.frame(Target = sig_mech_genes,
                 stress_type = "Mechanical",
                 label = "*"))
  }
  
  # Get y positions for annotations
  if(nrow(sig_annotations_comp3) > 0) {
    y_pos_comp3 <- polyIC_data %>%
      group_by(Target, stress_type) %>%
      summarise(max_y = max(delta_Cq, na.rm = TRUE), .groups = "drop")
    
    sig_annotations_comp3 <- sig_annotations_comp3 %>%
      left_join(y_pos_comp3, by = c("Target", "stress_type")) %>%
      mutate(y_pos = max_y + 0.5)
  }
} else {
  sig_annotations_comp3 <- data.frame()
}

# Create plot
p <- polyIC_data %>%
  ggplot(aes(x = stress_type, y = delta_Cq, fill = stress_type)) +
  geom_boxplot() +
  facet_wrap(~Target, scales = "free_y") +
  scale_fill_manual(values = c("Control" = "darkgray", "Temperature" = "salmon", "Mechanical" = "steelblue")) +
  theme_bw() +
  labs(
    title = "Delta Cq: PolyIC Control vs PolyIC Stressors",
    x = "Stress Type",
    y = "Delta Cq",
    fill = "Stress Type"
  ) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Add annotations if any exist
if(nrow(sig_annotations_comp3) > 0) {
  p <- p + 
    geom_text(data = sig_annotations_comp3, 
              aes(x = stress_type, y = y_pos, label = label),
              inherit.aes = FALSE, size = 6, vjust = 0, color = "darkorange") +
    labs(subtitle = "Asterisks (*) indicate significant differences from PolyIC Control (t-test p ≤ 0.05)",
         caption = paste("Significant comparisons:", paste(unique(sig_genes_comp3$Target), collapse = ", ")))
} else {
  p <- p + labs(subtitle = "No significant differences detected in t-tests (all p > 0.05)")
}

print(p)
```

## All treatments comparison

```{r boxplot-all-treatments}
#| hold: true
#| label: boxplot-all-treatments
#| fig.width: 12
#| fig.height: 8

# Use emmeans objects created in Statistical Comparisons section
all_treat_annotations <- data.frame()

# ATP Synthase
if(exists("emm_atp")) {
  cld_temp <- cld(emm_atp, Letters = letters, alpha = 0.05)
  cld_df <- as.data.frame(cld_temp)
  cld_df$Target <- "ATP.Synthase"
  cld_df$label <- trimws(cld_df$.group)
  cld_df$treatment_group <- paste(cld_df$polyIC, cld_df$stress_type, sep = "_")
  all_treat_annotations <- rbind(all_treat_annotations, 
                                 cld_df[, c("Target", "treatment_group", "label")])
}

# HSP70
if(exists("emm_hsp70")) {
  cld_temp <- cld(emm_hsp70, Letters = letters, alpha = 0.05)
  cld_df <- as.data.frame(cld_temp)
  cld_df$Target <- "HSP70"
  cld_df$label <- trimws(cld_df$.group)
  cld_df$treatment_group <- paste(cld_df$polyIC, cld_df$stress_type, sep = "_")
  all_treat_annotations <- rbind(all_treat_annotations, 
                                 cld_df[, c("Target", "treatment_group", "label")])
}

# HSP90
if(exists("emm_hsp90")) {
  cld_temp <- cld(emm_hsp90, Letters = letters, alpha = 0.05)
  cld_df <- as.data.frame(cld_temp)
  cld_df$Target <- "HSP90"
  cld_df$label <- trimws(cld_df$.group)
  cld_df$treatment_group <- paste(cld_df$polyIC, cld_df$stress_type, sep = "_")
  all_treat_annotations <- rbind(all_treat_annotations, 
                                 cld_df[, c("Target", "treatment_group", "label")])
}

# VIPERIN
if(exists("emm_viperin")) {
  cld_temp <- cld(emm_viperin, Letters = letters, alpha = 0.05)
  cld_df <- as.data.frame(cld_temp)
  cld_df$Target <- "VIPERIN"
  cld_df$label <- trimws(cld_df$.group)
  cld_df$treatment_group <- paste(cld_df$polyIC, cld_df$stress_type, sep = "_")
  all_treat_annotations <- rbind(all_treat_annotations, 
                                 cld_df[, c("Target", "treatment_group", "label")])
}

# Calculate y positions if annotations exist
if(nrow(all_treat_annotations) > 0) {
  y_pos_all <- delta_Cq_df %>%
    group_by(Target, treatment_group) %>%
    summarise(max_y = max(delta_Cq, na.rm = TRUE), .groups = "drop")
  
  # Calculate data range for each gene to use proportional offset
  y_range_all <- delta_Cq_df %>%
    group_by(Target) %>%
    summarise(y_range = max(delta_Cq, na.rm = TRUE) - min(delta_Cq, na.rm = TRUE), .groups = "drop")
  
  all_treat_annotations <- all_treat_annotations %>%
    left_join(y_pos_all, by = c("Target", "treatment_group")) %>%
    left_join(y_range_all, by = "Target") %>%
    mutate(y_pos = max_y + y_range * 0.08)  # 8% of data range above max
}

# Create plot
p <- delta_Cq_df %>%
  ggplot(aes(x = treatment_group, y = delta_Cq, fill = polyIC)) +
  geom_boxplot() +
  facet_wrap(~Target, scales = "free_y", ncol = 2) +
  scale_fill_manual(values = c("No_PolyIC" = "darkgray", "PolyIC" = "salmon")) +
  scale_y_continuous(expand = expansion(mult = c(0.05, 0.15))) +  # Extra space at top for annotations
  theme_bw() +
  labs(
    title = "Delta Cq Across All Treatment Groups",
    x = "Treatment Group",
    y = "Delta Cq",
    fill = "PolyIC"
  ) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Add annotations if they exist
if(nrow(all_treat_annotations) > 0) {
  p <- p +
    geom_text(data = all_treat_annotations,
              aes(x = treatment_group, y = y_pos, label = label),
              inherit.aes = FALSE, size = 4, vjust = 0, 
              color = "darkorange", fontface = "bold") +
    labs(subtitle = "Letters show statistical groupings (p ≤ 0.05). Groups sharing a letter are NOT significantly different from each other",
         caption = "Compact Letter Display shown for genes with significant ANOVA effects. See ANOVA Models section for detailed statistics.")
}

print(p)
```

# Delta Delta Cq and Fold Change Analysis

## Calculate delta delta Cq for each comparison

### Comparison 1: Stressors vs Control (No PolyIC)

```{r delta-delta-cq-no-polyIC}
#| hold: true
#| label: delta-delta-cq-no-polyIC

# Calculate delta delta Cq for Temperature vs Control (No PolyIC)
delta_delta_temp_no_polyIC <- no_polyIC_data %>%
  group_by(Target) %>%
  summarise(
    mean_control = mean(delta_Cq[stress_type == "Control"], na.rm = TRUE),
    mean_temp = mean(delta_Cq[stress_type == "Temperature"], na.rm = TRUE),
    delta_delta_Cq = mean_temp - mean_control,
    fold_change = 2^(-delta_delta_Cq),
    .groups = 'drop'
  )

print("Temperature vs Control (No PolyIC):")
print(delta_delta_temp_no_polyIC)

# Calculate delta delta Cq for Mechanical vs Control (No PolyIC)
delta_delta_mech_no_polyIC <- no_polyIC_data %>%
  group_by(Target) %>%
  summarise(
    mean_control = mean(delta_Cq[stress_type == "Control"], na.rm = TRUE),
    mean_mech = mean(delta_Cq[stress_type == "Mechanical"], na.rm = TRUE),
    delta_delta_Cq = mean_mech - mean_control,
    fold_change = 2^(-delta_delta_Cq),
    .groups = 'drop'
  )

print("Mechanical vs Control (No PolyIC):")
print(delta_delta_mech_no_polyIC)
```

### Comparison 2: PolyIC Control vs Basic Control

```{r delta-delta-cq-polyIC-control}
#| hold: true
#| label: delta-delta-cq-polyIC-control

delta_delta_polyIC_control <- control_data %>%
  group_by(Target) %>%
  summarise(
    mean_no_polyIC = mean(delta_Cq[polyIC == "No_PolyIC"], na.rm = TRUE),
    mean_polyIC = mean(delta_Cq[polyIC == "PolyIC"], na.rm = TRUE),
    delta_delta_Cq = mean_polyIC - mean_no_polyIC,
    fold_change = 2^(-delta_delta_Cq),
    .groups = 'drop'
  )

print("PolyIC Control vs Basic Control:")
print(delta_delta_polyIC_control)
```

### Comparison 3: PolyIC Stressors vs PolyIC Control

```{r delta-delta-cq-polyIC-stress}
#| hold: true
#| label: delta-delta-cq-polyIC-stress

# Calculate delta delta Cq for Temperature vs Control (PolyIC)
delta_delta_temp_polyIC <- polyIC_data %>%
  group_by(Target) %>%
  summarise(
    mean_control = mean(delta_Cq[stress_type == "Control"], na.rm = TRUE),
    mean_temp = mean(delta_Cq[stress_type == "Temperature"], na.rm = TRUE),
    delta_delta_Cq = mean_temp - mean_control,
    fold_change = 2^(-delta_delta_Cq),
    .groups = 'drop'
  )

print("Temperature vs Control (PolyIC):")
print(delta_delta_temp_polyIC)

# Calculate delta delta Cq for Mechanical vs Control (PolyIC)
delta_delta_mech_polyIC <- polyIC_data %>%
  group_by(Target) %>%
  summarise(
    mean_control = mean(delta_Cq[stress_type == "Control"], na.rm = TRUE),
    mean_mech = mean(delta_Cq[stress_type == "Mechanical"], na.rm = TRUE),
    delta_delta_Cq = mean_mech - mean_control,
    fold_change = 2^(-delta_delta_Cq),
    .groups = 'drop'
  )

print("Mechanical vs Control (PolyIC):")
print(delta_delta_mech_polyIC)
```

## Fold Change Bar Plots

### No PolyIC: Stressors vs Control

```{r barplot-fold-change-no-polyIC}
#| hold: true
#| label: barplot-fold-change-no-polyIC
#| fig.width: 10
#| fig.height: 6

# Combine temperature and mechanical comparisons
fold_change_no_polyIC <- bind_rows(
  delta_delta_temp_no_polyIC %>% mutate(comparison = "Temperature vs Control"),
  delta_delta_mech_no_polyIC %>% mutate(comparison = "Mechanical vs Control")
)

fold_change_no_polyIC %>%
  ggplot(aes(x = Target, y = fold_change, fill = comparison)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_hline(aes(yintercept = 1, linetype = "No change (FC = 1)"), color = "black") +
  scale_linetype_manual(name = "", values = c("No change (FC = 1)" = "dashed")) +
  theme_bw() +
  labs(
    title = "Fold Change: Stressors vs Control (No PolyIC)",
    x = "Target Gene",
    y = "Fold Change",
    fill = "Comparison"
  ) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

### PolyIC Control vs Basic Control

```{r barplot-fold-change-polyIC-control}
#| hold: true
#| label: barplot-fold-change-polyIC-control
#| fig.width: 10
#| fig.height: 6

delta_delta_polyIC_control %>%
  ggplot(aes(x = Target, y = fold_change, fill = Target)) +
  geom_bar(stat = "identity") +
  geom_hline(aes(yintercept = 1, linetype = "No change (FC = 1)"), color = "black") +
  scale_linetype_manual(name = "", values = c("No change (FC = 1)" = "dashed")) +
  theme_bw() +
  labs(
    title = "Fold Change: PolyIC Control vs Basic Control",
    x = "Target Gene",
    y = "Fold Change"
  ) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

### PolyIC: Stressors vs Control

```{r barplot-fold-change-polyIC-stress}
#| hold: true
#| label: barplot-fold-change-polyIC-stress
#| fig.width: 10
#| fig.height: 6

# Combine temperature and mechanical comparisons for PolyIC
fold_change_polyIC <- bind_rows(
  delta_delta_temp_polyIC %>% mutate(comparison = "Temperature vs Control"),
  delta_delta_mech_polyIC %>% mutate(comparison = "Mechanical vs Control")
)

fold_change_polyIC %>%
  ggplot(aes(x = Target, y = fold_change, fill = comparison)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_hline(aes(yintercept = 1, linetype = "No change (FC = 1)"), color = "black") +
  scale_linetype_manual(name = "", values = c("No change (FC = 1)" = "dashed")) +
  theme_bw() +
  labs(
    title = "Fold Change: Stressors vs Control (PolyIC Treatment)",
    x = "Target Gene",
    y = "Fold Change",
    fill = "Comparison"
  ) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

# ANOVA Models

Run ANOVA models to test for effects of PolyIC treatment and stress type on delta Cq values for each target gene.

**Note:** ANOVA models were created in the Statistical Comparisons section above. This section provides detailed assumption checks and interpretations for each gene.

## Data distribution check

ANOVA assumes that data (or residuals) are approximately normally distributed. This histogram provides a quick visual check of the overall delta Cq distribution across all samples and targets.

**What to look for:**
- **Bell-shaped curve**: Indicates roughly normal distribution, supporting ANOVA use
- **Skewness**: Strong left or right skew may violate normality assumptions
- **Outliers**: Extreme values far from the main distribution
- **Multimodal**: Multiple peaks may indicate distinct subpopulations

While this shows the overall distribution, we'll check normality of residuals for each gene-specific model using QQ plots below.

```{r delta-cq-histogram}
#| hold: true
#| label: delta-cq-histogram
hist(delta_Cq_df$delta_Cq, main = "Distribution of Delta Cq Values", xlab = "Delta Cq")
```

## ATP Synthase

ATP synthase is an enzyme complex that functions to synthesize adenosine triphosphate (ATP) from adenosine diphosphate (ADP) and inorganic phosphate (Pi), essentially generating the cell's primary energy currency by harnessing the energy from a proton gradient across a membrane.

```{r anova-atp-synthase}
#| hold: true
#| label: anova-atp-synthase
model_atp <- delta_Cq_df %>%
  filter(Target == "ATP.Synthase") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)

summary(model_atp)
```

### Assumption Checks for ATP Synthase

**QQ Plot (Quantile-Quantile Plot)**: Tests normality assumption of residuals.

- **What to look for**: Points should fall along the diagonal reference line
- **Good fit**: Points closely follow the line, indicating normally distributed residuals
- **Violations**: Systematic deviations (S-curves, heavy tails) suggest non-normality
- **Minor deviations** at the ends are usually acceptable

```{r anova-atp-synthase-qq}
#| hold: true
#| label: anova-atp-synthase-qq
qqPlot(model_atp$residuals)
```

**Levene's Test**: Tests homogeneity of variance (equal variances across groups).

- **Null hypothesis**: Variances are equal across all treatment groups
- **p-value > 0.05**: Variances are homogeneous (assumption met) ✓
- **p-value ≤ 0.05**: Variances differ significantly (assumption violated)

```{r anova-atp-synthase-levene}
#| results: hold
#| label: anova-atp-synthase-levene
leveneTest(model_atp$residuals ~ polyIC * stress_type, 
           data = delta_Cq_df %>% filter(Target == "ATP.Synthase"))

# If significant effects found, run post-hoc tests
if(any(summary(model_atp)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_atp <- emmeans(model_atp, ~ polyIC * stress_type)
  pairs_atp <- pairs(emm_atp)
  print(pairs_atp)
  
  # Identify significant contrasts
  pairs_df <- as.data.frame(pairs_atp)
  sig_contrasts <- pairs_df[pairs_df$p.value <= 0.05, ]
  
  cat("\n**ATP Synthase - Pairwise Contrast Summary:**\n\n")
  if(nrow(sig_contrasts) > 0) {
    cat("Significant contrasts (p ≤ 0.05):\n")
    for(i in 1:nrow(sig_contrasts)) {
      cat("-", sig_contrasts$contrast[i], "(p =", round(sig_contrasts$p.value[i], 4), ")\n")
    }
  } else {
    cat("No significant pairwise contrasts found (all p > 0.05).\n")
  }
} else {
  cat("No significant effects found; post-hoc tests not performed.\n")
}
```

### ATP Synthase ANOVA Interpretation

```{r anova-atp-synthase-interpret}
#| results: hold
#| label: anova-atp-synthase-interpret
anova_results <- summary(model_atp)[[1]]
# Fix: Trim whitespace from rownames
rownames(anova_results) <- trimws(rownames(anova_results))

levene_results <- leveneTest(model_atp$residuals ~ polyIC * stress_type, 
                              data = delta_Cq_df %>% filter(Target == "ATP.Synthase"))

cat("**ATP Synthase ANOVA Results:**\n\n")
cat("This ANOVA evaluates whether PolyIC treatment and/or stress type affect ATP Synthase expression (delta Cq).\n\n")

cat("**Treatment Effects (ANOVA):**\n\n")

# Check main effects and interaction
if(anova_results["polyIC", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC effect**: SIGNIFICANT (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **PolyIC effect**: Not significant (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
}

if(anova_results["stress_type", "Pr(>F)"] < 0.05) {
  cat("- **Stress type effect**: SIGNIFICANT (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **Stress type effect**: Not significant (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
}

if(anova_results["polyIC:stress_type", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC × Stress interaction**: SIGNIFICANT (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
  cat("  → The effect of stress depends on PolyIC treatment status.\n")
} else {
  cat("- **PolyIC × Stress interaction**: Not significant (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
}

cat("\n**Assumption Checks:**\n\n")

# Check Levene's test
levene_p <- levene_results$`Pr(>F)`[1]
if(levene_p > 0.05) {
  cat("- **Levene's Test** (homogeneity of variance): PASSED (p =", round(levene_p, 4), ") ✓\n")
  cat("  → Variances are equal across groups; ANOVA is appropriate.\n")
} else {
  cat("- **Levene's Test** (homogeneity of variance): FAILED (p =", round(levene_p, 4), ") ⚠️\n")
  cat("  → Variances differ across groups; ANOVA results should be interpreted with caution.\n")
}
```

## Citrate Synthase

Citrate synthase is important for energy production in the TCA cycle and is linked to the electron transport chain. It is also used as an enzyme marker for intact mitochondria.

```{r anova-citrate-synthase}
#| hold: true
#| label: anova-citrate-synthase
model_citrate <- delta_Cq_df %>%
  filter(Target == "Citrate.Synthase") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)

summary(model_citrate)
```

### Assumption Checks for Citrate Synthase

**QQ Plot (Quantile-Quantile Plot)**: Tests normality assumption of residuals.

- **What to look for**: Points should fall along the diagonal reference line
- **Good fit**: Points closely follow the line, indicating normally distributed residuals
- **Violations**: Systematic deviations (S-curves, heavy tails) suggest non-normality
- **Minor deviations** at the ends are usually acceptable

```{r anova-citrate-synthase-qq}
#| hold: true
#| label: anova-citrate-synthase-qq
qqPlot(model_citrate$residuals)
```

**Levene's Test**: Tests homogeneity of variance (equal variances across groups).

- **Null hypothesis**: Variances are equal across all treatment groups
- **p-value > 0.05**: Variances are homogeneous (assumption met) ✓
- **p-value ≤ 0.05**: Variances differ significantly (assumption violated)

```{r anova-citrate-synthase-levene}
#| results: hold
#| label: anova-citrate-synthase-levene
leveneTest(model_citrate$residuals ~ polyIC * stress_type, 
           data = delta_Cq_df %>% filter(Target == "Citrate.Synthase"))

# If significant effects found, run post-hoc tests
if(any(summary(model_citrate)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_citrate <- emmeans(model_citrate, ~ polyIC * stress_type)
  pairs_citrate <- pairs(emm_citrate)
  print(pairs_citrate)
  
  # Identify significant contrasts
  pairs_df <- as.data.frame(pairs_citrate)
  sig_contrasts <- pairs_df[pairs_df$p.value <= 0.05, ]
  
  cat("\n**Citrate Synthase - Pairwise Contrast Summary:**\n\n")
  if(nrow(sig_contrasts) > 0) {
    cat("Significant contrasts (p ≤ 0.05):\n")
    for(i in 1:nrow(sig_contrasts)) {
      cat("-", sig_contrasts$contrast[i], "(p =", round(sig_contrasts$p.value[i], 4), ")\n")
    }
  } else {
    cat("No significant pairwise contrasts found (all p > 0.05).\n")
  }
} else {
  cat("No significant effects found; post-hoc tests not performed.\n")
}
```

### Citrate Synthase ANOVA Interpretation

```{r anova-citrate-synthase-interpret}
#| results: hold
#| label: anova-citrate-synthase-interpret
anova_results <- summary(model_citrate)[[1]]
# Fix: Trim whitespace from rownames
rownames(anova_results) <- trimws(rownames(anova_results))

levene_results <- leveneTest(model_citrate$residuals ~ polyIC * stress_type, 
                              data = delta_Cq_df %>% filter(Target == "Citrate.Synthase"))

cat("**Citrate Synthase ANOVA Results:**\n\n")
cat("This ANOVA evaluates whether PolyIC treatment and/or stress type affect Citrate Synthase expression (delta Cq).\n\n")

cat("**Treatment Effects (ANOVA):**\n\n")

if(anova_results["polyIC", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC effect**: SIGNIFICANT (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **PolyIC effect**: Not significant (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
}

if(anova_results["stress_type", "Pr(>F)"] < 0.05) {
  cat("- **Stress type effect**: SIGNIFICANT (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **Stress type effect**: Not significant (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
}

if(anova_results["polyIC:stress_type", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC × Stress interaction**: SIGNIFICANT (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
  cat("  → The effect of stress depends on PolyIC treatment status.\n")
} else {
  cat("- **PolyIC × Stress interaction**: Not significant (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
}

cat("\n**Assumption Checks:**\n\n")

levene_p <- levene_results$`Pr(>F)`[1]
if(levene_p > 0.05) {
  cat("- **Levene's Test** (homogeneity of variance): PASSED (p =", round(levene_p, 4), ") ✓\n")
  cat("  → Variances are equal across groups; ANOVA is appropriate.\n")
} else {
  cat("- **Levene's Test** (homogeneity of variance): FAILED (p =", round(levene_p, 4), ") ⚠️\n")
  cat("  → Variances differ across groups; ANOVA results should be interpreted with caution.\n")
}
```

## DNMT1

The DNMT1 gene provides instructions for making an enzyme called DNA methyltransferase 1. This enzyme is involved in DNA methylation, which is the addition of methyl groups, consisting of one carbon atom and three hydrogen atoms, to DNA molecules.

```{r anova-dnmt1}
#| hold: true
#| label: anova-dnmt1
model_dnmt1 <- delta_Cq_df %>%
  filter(Target == "DNMT1") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)

summary(model_dnmt1)
```

### Assumption Checks for DNMT1

**QQ Plot (Quantile-Quantile Plot)**: Tests normality assumption of residuals.

- **What to look for**: Points should fall along the diagonal reference line
- **Good fit**: Points closely follow the line, indicating normally distributed residuals
- **Violations**: Systematic deviations (S-curves, heavy tails) suggest non-normality
- **Minor deviations** at the ends are usually acceptable

```{r anova-dnmt1-qq}
#| hold: true
#| label: anova-dnmt1-qq
qqPlot(model_dnmt1$residuals)
```

**Levene's Test**: Tests homogeneity of variance (equal variances across groups).

- **Null hypothesis**: Variances are equal across all treatment groups
- **p-value > 0.05**: Variances are homogeneous (assumption met) ✓
- **p-value ≤ 0.05**: Variances differ significantly (assumption violated)

```{r anova-dnmt1-levene}
#| results: hold
#| label: anova-dnmt1-levene
leveneTest(model_dnmt1$residuals ~ polyIC * stress_type, 
           data = delta_Cq_df %>% filter(Target == "DNMT1"))

# If significant effects found, run post-hoc tests
if(any(summary(model_dnmt1)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_dnmt1 <- emmeans(model_dnmt1, ~ polyIC * stress_type)
  pairs_dnmt1 <- pairs(emm_dnmt1)
  print(pairs_dnmt1)
  
  # Identify significant contrasts
  pairs_df <- as.data.frame(pairs_dnmt1)
  sig_contrasts <- pairs_df[pairs_df$p.value <= 0.05, ]
  
  cat("\n**DNMT1 - Pairwise Contrast Summary:**\n\n")
  if(nrow(sig_contrasts) > 0) {
    cat("Significant contrasts (p ≤ 0.05):\n")
    for(i in 1:nrow(sig_contrasts)) {
      cat("-", sig_contrasts$contrast[i], "(p =", round(sig_contrasts$p.value[i], 4), ")\n")
    }
  } else {
    cat("No significant pairwise contrasts found (all p > 0.05).\n")
  }
} else {
  cat("No significant effects found; post-hoc tests not performed.\n")
}
```

### DNMT1 ANOVA Interpretation

```{r anova-dnmt1-interpret}
#| results: hold
#| label: anova-dnmt1-interpret
anova_results <- summary(model_dnmt1)[[1]]
# Fix: Trim whitespace from rownames
rownames(anova_results) <- trimws(rownames(anova_results))

levene_results <- leveneTest(model_dnmt1$residuals ~ polyIC * stress_type, 
                              data = delta_Cq_df %>% filter(Target == "DNMT1"))

cat("**DNMT1 ANOVA Results:**\n\n")
cat("This ANOVA evaluates whether PolyIC treatment and/or stress type affect DNMT1 expression (delta Cq).\n\n")

cat("**Treatment Effects (ANOVA):**\n\n")

if(anova_results["polyIC", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC effect**: SIGNIFICANT (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **PolyIC effect**: Not significant (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
}

if(anova_results["stress_type", "Pr(>F)"] < 0.05) {
  cat("- **Stress type effect**: SIGNIFICANT (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **Stress type effect**: Not significant (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
}

if(anova_results["polyIC:stress_type", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC × Stress interaction**: SIGNIFICANT (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
  cat("  → The effect of stress depends on PolyIC treatment status.\n")
} else {
  cat("- **PolyIC × Stress interaction**: Not significant (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
}

cat("\n**Assumption Checks:**\n\n")

levene_p <- levene_results$`Pr(>F)`[1]
if(levene_p > 0.05) {
  cat("- **Levene's Test** (homogeneity of variance): PASSED (p =", round(levene_p, 4), ") ✓\n")
  cat("  → Variances are equal across groups; ANOVA is appropriate.\n")
} else {
  cat("- **Levene's Test** (homogeneity of variance): FAILED (p =", round(levene_p, 4), ") ⚠️\n")
  cat("  → Variances differ across groups; ANOVA results should be interpreted with caution.\n")
}
```

## HSP70

Heat Shock Protein 70 (Hsp70) is a molecular chaperone that plays crucial roles in maintaining cellular protein homeostasis and protecting cells from stress.

```{r anova-hsp70}
#| hold: true
#| label: anova-hsp70
model_hsp70 <- delta_Cq_df %>%
  filter(Target == "HSP70") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)

summary(model_hsp70)
```

### Assumption Checks for HSP70

**QQ Plot (Quantile-Quantile Plot)**: Tests normality assumption of residuals.

- **What to look for**: Points should fall along the diagonal reference line
- **Good fit**: Points closely follow the line, indicating normally distributed residuals
- **Violations**: Systematic deviations (S-curves, heavy tails) suggest non-normality
- **Minor deviations** at the ends are usually acceptable

```{r anova-hsp70-qq}
#| hold: true
#| label: anova-hsp70-qq
qqPlot(model_hsp70$residuals)
```

**Levene's Test**: Tests homogeneity of variance (equal variances across groups).

- **Null hypothesis**: Variances are equal across all treatment groups
- **p-value > 0.05**: Variances are homogeneous (assumption met) ✓
- **p-value ≤ 0.05**: Variances differ significantly (assumption violated)

```{r anova-hsp70-levene}
#| results: hold
#| label: anova-hsp70-levene
leveneTest(model_hsp70$residuals ~ polyIC * stress_type, 
           data = delta_Cq_df %>% filter(Target == "HSP70"))

# If significant effects found, run post-hoc tests
if(any(summary(model_hsp70)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_hsp70 <- emmeans(model_hsp70, ~ polyIC * stress_type)
  pairs_hsp70 <- pairs(emm_hsp70)
  print(pairs_hsp70)
  
  # Identify significant contrasts
  pairs_df <- as.data.frame(pairs_hsp70)
  sig_contrasts <- pairs_df[pairs_df$p.value <= 0.05, ]
  
  cat("\n**HSP70 - Pairwise Contrast Summary:**\n\n")
  if(nrow(sig_contrasts) > 0) {
    cat("Significant contrasts (p ≤ 0.05):\n")
    for(i in 1:nrow(sig_contrasts)) {
      cat("-", sig_contrasts$contrast[i], "(p =", round(sig_contrasts$p.value[i], 4), ")\n")
    }
  } else {
    cat("No significant pairwise contrasts found (all p > 0.05).\n")
  }
} else {
  cat("No significant effects found; post-hoc tests not performed.\n")
}
```

### HSP70 ANOVA Interpretation

```{r anova-hsp70-interpret}
#| results: hold
#| label: anova-hsp70-interpret
anova_results <- summary(model_hsp70)[[1]]
# Fix: Trim whitespace from rownames
rownames(anova_results) <- trimws(rownames(anova_results))

levene_results <- leveneTest(model_hsp70$residuals ~ polyIC * stress_type, 
                              data = delta_Cq_df %>% filter(Target == "HSP70"))

cat("**HSP70 ANOVA Results:**\n\n")
cat("This ANOVA evaluates whether PolyIC treatment and/or stress type affect HSP70 expression (delta Cq).\n\n")

cat("**Treatment Effects (ANOVA):**\n\n")

if(anova_results["polyIC", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC effect**: SIGNIFICANT (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **PolyIC effect**: Not significant (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
}

if(anova_results["stress_type", "Pr(>F)"] < 0.05) {
  cat("- **Stress type effect**: SIGNIFICANT (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **Stress type effect**: Not significant (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
}

if(anova_results["polyIC:stress_type", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC × Stress interaction**: SIGNIFICANT (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
  cat("  → The effect of stress depends on PolyIC treatment status.\n")
} else {
  cat("- **PolyIC × Stress interaction**: Not significant (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
}

cat("\n**Assumption Checks:**\n\n")

levene_p <- levene_results$`Pr(>F)`[1]
if(levene_p > 0.05) {
  cat("- **Levene's Test** (homogeneity of variance): PASSED (p =", round(levene_p, 4), ") ✓\n")
  cat("  → Variances are equal across groups; ANOVA is appropriate.\n")
} else {
  cat("- **Levene's Test** (homogeneity of variance): FAILED (p =", round(levene_p, 4), ") ⚠️\n")
  cat("  → Variances differ across groups; ANOVA results should be interpreted with caution.\n")
}
```

## HSP90

Heat shock protein 90 (Hsp90) is a molecular chaperone that helps proteins fold, mature, and remain active. Hsp90 also helps regulate signaling networks and is involved in many cellular processes.

```{r anova-hsp90}
#| hold: true
#| label: anova-hsp90
model_hsp90 <- delta_Cq_df %>%
  filter(Target == "HSP90") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)

summary(model_hsp90)
```

### Assumption Checks for HSP90

**QQ Plot (Quantile-Quantile Plot)**: Tests normality assumption of residuals.

- **What to look for**: Points should fall along the diagonal reference line
- **Good fit**: Points closely follow the line, indicating normally distributed residuals
- **Violations**: Systematic deviations (S-curves, heavy tails) suggest non-normality
- **Minor deviations** at the ends are usually acceptable

```{r anova-hsp90-qq}
#| hold: true
#| label: anova-hsp90-qq
qqPlot(model_hsp90$residuals)
```

**Levene's Test**: Tests homogeneity of variance (equal variances across groups).

- **Null hypothesis**: Variances are equal across all treatment groups
- **p-value > 0.05**: Variances are homogeneous (assumption met) ✓
- **p-value ≤ 0.05**: Variances differ significantly (assumption violated)

```{r anova-hsp90-levene}
#| results: hold
#| label: anova-hsp90-levene
leveneTest(model_hsp90$residuals ~ polyIC * stress_type, 
           data = delta_Cq_df %>% filter(Target == "HSP90"))

# If significant effects found, run post-hoc tests
if(any(summary(model_hsp90)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_hsp90 <- emmeans(model_hsp90, ~ polyIC * stress_type)
  pairs_hsp90 <- pairs(emm_hsp90)
  print(pairs_hsp90)
  
  # Identify significant contrasts
  pairs_df <- as.data.frame(pairs_hsp90)
  sig_contrasts <- pairs_df[pairs_df$p.value <= 0.05, ]
  
  cat("\n**HSP90 - Pairwise Contrast Summary:**\n\n")
  if(nrow(sig_contrasts) > 0) {
    cat("Significant contrasts (p ≤ 0.05):\n")
    for(i in 1:nrow(sig_contrasts)) {
      cat("-", sig_contrasts$contrast[i], "(p =", round(sig_contrasts$p.value[i], 4), ")\n")
    }
  } else {
    cat("No significant pairwise contrasts found (all p > 0.05).\n")
  }
} else {
  cat("No significant effects found; post-hoc tests not performed.\n")
}
```

### HSP90 ANOVA Interpretation

```{r anova-hsp90-interpret}
#| results: hold
#| label: anova-hsp90-interpret
anova_results <- summary(model_hsp90)[[1]]
# Fix: Trim whitespace from rownames
rownames(anova_results) <- trimws(rownames(anova_results))

levene_results <- leveneTest(model_hsp90$residuals ~ polyIC * stress_type, 
                              data = delta_Cq_df %>% filter(Target == "HSP90"))

cat("**HSP90 ANOVA Results:**\n\n")
cat("This ANOVA evaluates whether PolyIC treatment and/or stress type affect HSP90 expression (delta Cq).\n\n")

cat("**Treatment Effects (ANOVA):**\n\n")

if(anova_results["polyIC", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC effect**: SIGNIFICANT (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **PolyIC effect**: Not significant (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
}

if(anova_results["stress_type", "Pr(>F)"] < 0.05) {
  cat("- **Stress type effect**: SIGNIFICANT (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **Stress type effect**: Not significant (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
}

if(anova_results["polyIC:stress_type", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC × Stress interaction**: SIGNIFICANT (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
  cat("  → The effect of stress depends on PolyIC treatment status.\n")
} else {
  cat("- **PolyIC × Stress interaction**: Not significant (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
}

cat("\n**Assumption Checks:**\n\n")

levene_p <- levene_results$`Pr(>F)`[1]
if(levene_p > 0.05) {
  cat("- **Levene's Test** (homogeneity of variance): PASSED (p =", round(levene_p, 4), ") ✓\n")
  cat("  → Variances are equal across groups; ANOVA is appropriate.\n")
} else {
  cat("- **Levene's Test** (homogeneity of variance): FAILED (p =", round(levene_p, 4), ") ⚠️\n")
  cat("  → Variances differ across groups; ANOVA results should be interpreted with caution.\n")
}
```

## cGAS

The cGAS gene is involved in several processes, including cellular response to exogenous dsRNA, positive regulation of intracellular signal transduction, and regulation of defense response.

```{r anova-cgas}
#| hold: true
#| label: anova-cgas
model_cgas <- delta_Cq_df %>%
  filter(Target == "cGAS") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)

summary(model_cgas)
```

### Assumption Checks for cGAS

**QQ Plot (Quantile-Quantile Plot)**: Tests normality assumption of residuals.

- **What to look for**: Points should fall along the diagonal reference line
- **Good fit**: Points closely follow the line, indicating normally distributed residuals
- **Violations**: Systematic deviations (S-curves, heavy tails) suggest non-normality
- **Minor deviations** at the ends are usually acceptable

```{r anova-cgas-qq}
#| hold: true
#| label: anova-cgas-qq
qqPlot(model_cgas$residuals)
```

**Levene's Test**: Tests homogeneity of variance (equal variances across groups).

- **Null hypothesis**: Variances are equal across all treatment groups
- **p-value > 0.05**: Variances are homogeneous (assumption met) ✓
- **p-value ≤ 0.05**: Variances differ significantly (assumption violated)

```{r anova-cgas-levene}
#| results: hold
#| label: anova-cgas-levene
leveneTest(model_cgas$residuals ~ polyIC * stress_type, 
           data = delta_Cq_df %>% filter(Target == "cGAS"))

# If significant effects found, run post-hoc tests
if(any(summary(model_cgas)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_cgas <- emmeans(model_cgas, ~ polyIC * stress_type)
  pairs_cgas <- pairs(emm_cgas)
  print(pairs_cgas)
  
  # Identify significant contrasts
  pairs_df <- as.data.frame(pairs_cgas)
  sig_contrasts <- pairs_df[pairs_df$p.value <= 0.05, ]
  
  cat("\n**cGAS - Pairwise Contrast Summary:**\n\n")
  if(nrow(sig_contrasts) > 0) {
    cat("Significant contrasts (p ≤ 0.05):\n")
    for(i in 1:nrow(sig_contrasts)) {
      cat("-", sig_contrasts$contrast[i], "(p =", round(sig_contrasts$p.value[i], 4), ")\n")
    }
  } else {
    cat("No significant pairwise contrasts found (all p > 0.05).\n")
  }
} else {
  cat("No significant effects found; post-hoc tests not performed.\n")
}
```

### cGAS ANOVA Interpretation

```{r anova-cgas-interpret}
#| results: hold
#| label: anova-cgas-interpret
anova_results <- summary(model_cgas)[[1]]
# Fix: Trim whitespace from rownames
rownames(anova_results) <- trimws(rownames(anova_results))

levene_results <- leveneTest(model_cgas$residuals ~ polyIC * stress_type, 
                              data = delta_Cq_df %>% filter(Target == "cGAS"))

cat("**cGAS ANOVA Results:**\n\n")
cat("This ANOVA evaluates whether PolyIC treatment and/or stress type affect cGAS expression (delta Cq).\n\n")

cat("**Treatment Effects (ANOVA):**\n\n")

if(anova_results["polyIC", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC effect**: SIGNIFICANT (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **PolyIC effect**: Not significant (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
}

if(anova_results["stress_type", "Pr(>F)"] < 0.05) {
  cat("- **Stress type effect**: SIGNIFICANT (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **Stress type effect**: Not significant (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
}

if(anova_results["polyIC:stress_type", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC × Stress interaction**: SIGNIFICANT (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
  cat("  → The effect of stress depends on PolyIC treatment status.\n")
} else {
  cat("- **PolyIC × Stress interaction**: Not significant (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
}

cat("\n**Assumption Checks:**\n\n")

levene_p <- levene_results$`Pr(>F)`[1]
if(levene_p > 0.05) {
  cat("- **Levene's Test** (homogeneity of variance): PASSED (p =", round(levene_p, 4), ") ✓\n")
  cat("  → Variances are equal across groups; ANOVA is appropriate.\n")
} else {
  cat("- **Levene's Test** (homogeneity of variance): FAILED (p =", round(levene_p, 4), ") ⚠️\n")
  cat("  → Variances differ across groups; ANOVA results should be interpreted with caution.\n")
}
```

## VIPERIN

Viperin limits replication of DNA and RNA viruses.

```{r anova-viperin}
#| hold: true
#| label: anova-viperin
model_viperin <- delta_Cq_df %>%
  filter(Target == "VIPERIN") %>%
  aov(delta_Cq ~ polyIC * stress_type, data = .)

summary(model_viperin)
```

### Assumption Checks for VIPERIN

**QQ Plot (Quantile-Quantile Plot)**: Tests normality assumption of residuals.

- **What to look for**: Points should fall along the diagonal reference line
- **Good fit**: Points closely follow the line, indicating normally distributed residuals
- **Violations**: Systematic deviations (S-curves, heavy tails) suggest non-normality
- **Minor deviations** at the ends are usually acceptable

```{r anova-viperin-qq}
#| hold: true
#| label: anova-viperin-qq
qqPlot(model_viperin$residuals)
```

**Levene's Test**: Tests homogeneity of variance (equal variances across groups).

- **Null hypothesis**: Variances are equal across all treatment groups
- **p-value > 0.05**: Variances are homogeneous (assumption met) ✓
- **p-value ≤ 0.05**: Variances differ significantly (assumption violated)

```{r anova-viperin-levene}
#| results: hold
#| label: anova-viperin-levene
leveneTest(model_viperin$residuals ~ polyIC * stress_type, 
           data = delta_Cq_df %>% filter(Target == "VIPERIN"))

# If significant effects found, run post-hoc tests
if(any(summary(model_viperin)[[1]][["Pr(>F)"]][1:3] < 0.05, na.rm = TRUE)) {
  emm_viperin <- emmeans(model_viperin, ~ polyIC * stress_type)
  pairs_viperin <- pairs(emm_viperin)
  print(pairs_viperin)
  
  # Identify significant contrasts
  pairs_df <- as.data.frame(pairs_viperin)
  sig_contrasts <- pairs_df[pairs_df$p.value <= 0.05, ]
  
  cat("\n**VIPERIN - Pairwise Contrast Summary:**\n\n")
  if(nrow(sig_contrasts) > 0) {
    cat("Significant contrasts (p ≤ 0.05):\n")
    for(i in 1:nrow(sig_contrasts)) {
      cat("-", sig_contrasts$contrast[i], "(p =", round(sig_contrasts$p.value[i], 4), ")\n")
    }
  } else {
    cat("No significant pairwise contrasts found (all p > 0.05).\n")
  }
} else {
  cat("No significant effects found; post-hoc tests not performed.\n")
}
```

### VIPERIN ANOVA Interpretation

```{r anova-viperin-interpret}
#| results: hold
#| label: anova-viperin-interpret
anova_results <- summary(model_viperin)[[1]]
# Fix: Trim whitespace from rownames
rownames(anova_results) <- trimws(rownames(anova_results))

levene_results <- leveneTest(model_viperin$residuals ~ polyIC * stress_type, 
                              data = delta_Cq_df %>% filter(Target == "VIPERIN"))

cat("**VIPERIN ANOVA Results:**\n\n")
cat("This ANOVA evaluates whether PolyIC treatment and/or stress type affect VIPERIN expression (delta Cq).\n\n")

cat("**Treatment Effects (ANOVA):**\n\n")

if(anova_results["polyIC", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC effect**: SIGNIFICANT (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **PolyIC effect**: Not significant (p =", round(anova_results["polyIC", "Pr(>F)"], 4), ")\n")
}

if(anova_results["stress_type", "Pr(>F)"] < 0.05) {
  cat("- **Stress type effect**: SIGNIFICANT (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
} else {
  cat("- **Stress type effect**: Not significant (p =", round(anova_results["stress_type", "Pr(>F)"], 4), ")\n")
}

if(anova_results["polyIC:stress_type", "Pr(>F)"] < 0.05) {
  cat("- **PolyIC × Stress interaction**: SIGNIFICANT (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
  cat("  → The effect of stress depends on PolyIC treatment status.\n")
} else {
  cat("- **PolyIC × Stress interaction**: Not significant (p =", round(anova_results["polyIC:stress_type", "Pr(>F)"], 4), ")\n")
}

cat("\n**Assumption Checks:**\n\n")

levene_p <- levene_results$`Pr(>F)`[1]
if(levene_p > 0.05) {
  cat("- **Levene's Test** (homogeneity of variance): PASSED (p =", round(levene_p, 4), ") ✓\n")
  cat("  → Variances are equal across groups; ANOVA is appropriate.\n")
} else {
  cat("- **Levene's Test** (homogeneity of variance): FAILED (p =", round(levene_p, 4), ") ⚠️\n")
  cat("  → Variances differ across groups; ANOVA results should be interpreted with caution.\n")
}
```

## Summary Plots

Visualize delta Cq values by treatment group for each target.

```{r anova-summary-plots}
#| hold: true
#| label: anova-summary-plots
#| fig.width: 12
#| fig.height: 10

# Generate Compact Letter Display (CLD) for each gene with significant effects
# CLD shows which groups are significantly different: groups sharing a letter are NOT significantly different

cld_annotations <- data.frame()

# ATP Synthase
if(exists("emm_atp")) {
  cld_atp <- cld(emm_atp, Letters = letters, alpha = 0.05)
  cld_atp_df <- as.data.frame(cld_atp)
  cld_atp_df$Target <- "ATP.Synthase"
  cld_atp_df$label <- trimws(cld_atp_df$.group)
  cld_atp_df <- cld_atp_df[, c("Target", "polyIC", "stress_type", "label")]
  cld_annotations <- rbind(cld_annotations, cld_atp_df)
}

# HSP70
if(exists("emm_hsp70")) {
  cld_hsp70 <- cld(emm_hsp70, Letters = letters, alpha = 0.05)
  cld_hsp70_df <- as.data.frame(cld_hsp70)
  cld_hsp70_df$Target <- "HSP70"
  cld_hsp70_df$label <- trimws(cld_hsp70_df$.group)
  cld_hsp70_df <- cld_hsp70_df[, c("Target", "polyIC", "stress_type", "label")]
  cld_annotations <- rbind(cld_annotations, cld_hsp70_df)
}

# HSP90
if(exists("emm_hsp90")) {
  cld_hsp90 <- cld(emm_hsp90, Letters = letters, alpha = 0.05)
  cld_hsp90_df <- as.data.frame(cld_hsp90)
  cld_hsp90_df$Target <- "HSP90"
  cld_hsp90_df$label <- trimws(cld_hsp90_df$.group)
  cld_hsp90_df <- cld_hsp90_df[, c("Target", "polyIC", "stress_type", "label")]
  cld_annotations <- rbind(cld_annotations, cld_hsp90_df)
}

# VIPERIN
if(exists("emm_viperin")) {
  cld_viperin <- cld(emm_viperin, Letters = letters, alpha = 0.05)
  cld_viperin_df <- as.data.frame(cld_viperin)
  cld_viperin_df$Target <- "VIPERIN"
  cld_viperin_df$label <- trimws(cld_viperin_df$.group)
  cld_viperin_df <- cld_viperin_df[, c("Target", "polyIC", "stress_type", "label")]
  cld_annotations <- rbind(cld_annotations, cld_viperin_df)
}

# Calculate y positions for annotations (above max values)
if(nrow(cld_annotations) > 0) {
  y_positions <- delta_Cq_df %>%
    group_by(Target, stress_type, polyIC) %>%
    summarise(max_y = max(delta_Cq, na.rm = TRUE), .groups = "drop")
  
  # Calculate data range for each gene to use proportional offset
  y_range_per_gene <- delta_Cq_df %>%
    group_by(Target) %>%
    summarise(y_range = max(delta_Cq, na.rm = TRUE) - min(delta_Cq, na.rm = TRUE), .groups = "drop")
  
  cld_annotations <- cld_annotations %>%
    left_join(y_positions, by = c("Target", "stress_type", "polyIC")) %>%
    left_join(y_range_per_gene, by = "Target") %>%
    mutate(y_pos = max_y + y_range * 0.08)  # 8% of data range above max
}

# Create plot
p <- delta_Cq_df %>%
  ggplot(aes(x = stress_type, y = delta_Cq, fill = polyIC)) +
  geom_boxplot(position = position_dodge(0.8), alpha = 0.7) +
  geom_point(position = position_dodge(0.8), alpha = 0.5, size = 1) +
  facet_wrap(~Target, scales = "free_y", ncol = 2) +
  scale_fill_manual(values = c("No_PolyIC" = "darkgray", "PolyIC" = "salmon")) +
  scale_y_continuous(expand = expansion(mult = c(0.05, 0.15))) +  # Extra space at top for annotations
  geom_hline(yintercept = 0, linetype = "dashed", alpha = 0.5) +
  theme_bw() +
  labs(
    title = "Delta Cq by Treatment Group with ANOVA Significance",
    x = "Stress Type",
    y = "Delta Cq",
    fill = "PolyIC Treatment"
  ) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    legend.position = "bottom"
  )

# Add CLD annotations if any exist
if(nrow(cld_annotations) > 0) {
  p <- p + 
    geom_text(data = cld_annotations, 
              aes(x = stress_type, y = y_pos, label = label, group = polyIC),
              position = position_dodge(0.8), size = 4, vjust = 0, 
              color = "darkorange", fontface = "bold") +
    labs(subtitle = "Letters show statistical groupings (p ≤ 0.05). Groups sharing a letter are NOT significantly different from each other",
         caption = "Compact Letter Display shown for: ATP Synthase, HSP70, HSP90, and VIPERIN. See individual ANOVA sections for detailed p-values.")
}

print(p)
```


# SUMMARY

This analysis examined gene expression responses in *Magallana gigas* to PolyIC treatment (an immune stimulant mimicking viral dsRNA) and different stress types (Temperature, Mechanical, Control) using qPCR. Eight genes were analyzed: ATP Synthase, Citrate Synthase, DNMT1, HSP70, HSP90, cGAS, and VIPERIN. Expression levels were normalized to GAPDH and analyzed using two-way ANOVA (PolyIC × Stress Type) with post-hoc pairwise comparisons.

## Gene-Specific Results

### ATP Synthase
**Function:** Mitochondrial enzyme synthesizing ATP; marker of energy metabolism.

**ANOVA Results:**
- **PolyIC effect:** SIGNIFICANT (p = 7.33e-06) ***
- **Stress type effect:** SIGNIFICANT (p = 0.04) *
- **PolyIC × Stress interaction:** Not significant (p = 0.47)

**Assumption Checks:**
- **Levene's Test:** FAILED (p = 0.0036) - variances are not homogeneous across groups ⚠️

**Significant Pairwise Contrasts (p ≤ 0.05):**
- No_PolyIC Control vs PolyIC Mechanical (p = 0.007)
- No_PolyIC Control vs PolyIC Temperature (p = 0.0001)
- No_PolyIC Mechanical vs PolyIC Temperature (p = 0.0195)
- No_PolyIC Temperature vs PolyIC Temperature (p = 0.0158)

**Key Findings:**
- PolyIC treatment significantly affects ATP Synthase expression
- The "stress type effect" is driven primarily by **PolyIC-treated samples under stress** (all significant contrasts involve PolyIC groups)
- No significant differences between non-PolyIC control and non-PolyIC stressed groups
- Effects are additive rather than interactive (no interaction between PolyIC and stress)
- PolyIC + Temperature shows the strongest effect (lowest p-value), followed by PolyIC + Mechanical
- **Caveat:** Heterogeneous variances suggest results should be interpreted cautiously

### Citrate Synthase
**Function:** TCA cycle enzyme; marker of mitochondrial density and oxidative capacity.

**ANOVA Results:**
- **PolyIC effect:** Not significant (p = 0.693)
- **Stress type effect:** Marginally non-significant (p = 0.0688)
- **PolyIC × Stress interaction:** Not significant (p = 0.464)

**Assumption Checks:**
- **Levene's Test:** PASSED (p = 0.956) ✓

**Key Findings:**
- Citrate Synthase expression is stable across PolyIC treatments
- Stress type shows a trend toward significance (p = 0.0688) but does not reach the p ≤ 0.05 threshold
- No significant pairwise contrasts detected (all p > 0.05)
- This near-significant trend without specific contrasts suggests subtle, distributed differences across stress types rather than a strong specific effect
- Mitochondrial density/function appears relatively unaffected by these experimental conditions

### DNMT1
**Function:** DNA methyltransferase enzyme; maintains DNA methylation patterns and epigenetic regulation.

**ANOVA Results:**
- **PolyIC effect:** Not significant (p = 0.193)
- **Stress type effect:** Not significant (p = 0.179)
- **PolyIC × Stress interaction:** Not significant (p = 0.399)

**Assumption Checks:**
- **Levene's Test:** PASSED (p = 0.617) ✓

**Key Findings:**
- DNMT1 expression does not respond significantly to PolyIC or stress treatments
- DNA methylation machinery appears constitutively expressed under these conditions
- No evidence of stress-induced epigenetic remodeling at the transcriptional level

### HSP70
**Function:** Molecular chaperone; critical for protein folding and cellular stress response.

**ANOVA Results:**
- **PolyIC effect:** SIGNIFICANT (p = 5.7e-13) ***
- **Stress type effect:** SIGNIFICANT (p < 2e-16) ***
- **PolyIC × Stress interaction:** Not significant (p = 0.831)

**Assumption Checks:**
- **Levene's Test:** PASSED (p = 0.745) ✓

**Significant Pairwise Contrasts (p ≤ 0.05):**
- No_PolyIC Control vs No_PolyIC Temperature (p < 0.0001)
- No_PolyIC Control vs PolyIC Temperature (p < 0.0001)
- PolyIC Control vs No_PolyIC Temperature (p < 0.0001)
- PolyIC Control vs PolyIC Temperature (p < 0.0001)
- No_PolyIC Mechanical vs No_PolyIC Temperature (p < 0.0001)
- No_PolyIC Mechanical vs PolyIC Temperature (p < 0.0001)
- PolyIC Mechanical vs No_PolyIC Temperature (p = 0.0017)
- PolyIC Mechanical vs PolyIC Temperature (p < 0.0001)

**Key Findings:**
- HSP70 shows extremely strong responses to both PolyIC and stress type
- **Temperature stress is the primary driver:** All 8 significant contrasts involve temperature-stressed groups vs. non-temperature groups
- Both PolyIC and non-PolyIC oysters show dramatic HSP70 upregulation under temperature stress compared to controls or mechanical stress
- Mechanical stress does NOT significantly induce HSP70 (no significant Control vs Mechanical contrasts)
- No interaction suggests PolyIC and temperature stress act independently (additive effects)
- This is the classic heat shock protein response to thermal stress

### HSP90
**Function:** Molecular chaperone; regulates protein maturation and signaling networks.

**ANOVA Results:**
- **PolyIC effect:** SIGNIFICANT (p = 5.47e-10) ***
- **Stress type effect:** Not significant (p = 0.476)
- **PolyIC × Stress interaction:** Not significant (p = 0.562)

**Assumption Checks:**
- **Levene's Test:** PASSED (p = 0.942) ✓

**Significant Pairwise Contrasts (p ≤ 0.05):**
- No_PolyIC Control vs PolyIC Temperature (p < 0.0001)
- No_PolyIC Mechanical vs PolyIC Temperature (p < 0.0001)
- No_PolyIC Temperature vs PolyIC Temperature (p < 0.0001)

**Key Findings:**
- PolyIC has a highly significant main effect on HSP90 expression
- Unlike HSP70, the overall stress type main effect is not significant
- **All three significant contrasts specifically involve PolyIC + Temperature:**
  - PolyIC Temperature differs from all three non-PolyIC groups (Control, Mechanical, Temperature)
  - No other treatment combinations show significant differences
- The pattern shows **PolyIC + Temperature creates a synergistic-like response** despite the non-significant interaction term (possibly due to sample size/power)
- PolyIC alone (control), temperature alone, and mechanical stress (with/without PolyIC) do NOT significantly induce HSP90
- HSP90 appears to require the combination of immune stimulation (PolyIC) AND thermal stress

### cGAS
**Function:** Cytosolic DNA sensor; involved in innate immune response and dsRNA detection.

**ANOVA Results:**
- **PolyIC effect:** Not significant (p = 0.749)
- **Stress type effect:** Not significant (p = 0.168)
- **PolyIC × Stress interaction:** Not significant (p = 0.831)

**Assumption Checks:**
- **Levene's Test:** PASSED (p = 0.564) ✓

**Key Findings:**
- cGAS does not respond significantly to PolyIC treatment
- No significant response to stress type either
- This is unexpected given cGAS's role in detecting foreign nucleic acids
- May suggest: (1) different immune sensing pathways activated, (2) timing issues (response may occur earlier/later), or (3) PolyIC recognition occurs through alternative receptors in oysters

### VIPERIN
**Function:** Antiviral effector protein; restricts replication of DNA and RNA viruses.

**ANOVA Results:**
- **PolyIC effect:** SIGNIFICANT (p = 0.000165) ***
- **Stress type effect:** Not significant (p = 0.548)
- **PolyIC × Stress interaction:** Not significant (p = 0.505)

**Assumption Checks:**
- **Levene's Test:** PASSED (p = 0.690) ✓

**Significant Pairwise Contrasts (p ≤ 0.05):**
- No_PolyIC Control vs PolyIC Control (p = 0.0242)

**Key Findings:**
- VIPERIN shows a clear response to PolyIC treatment as an immune stimulant
- The only significant contrast is **No_PolyIC Control vs PolyIC Control** (p = 0.024)
- **PolyIC alone is sufficient to induce VIPERIN expression** - no additional stress needed
- Notably, PolyIC + stress combinations (mechanical/temperature) are NOT significantly different from PolyIC control
- This indicates the VIPERIN response is triggered by PolyIC recognition itself, independent of other stressors
- Stress type (temperature/mechanical) does not enhance or suppress the PolyIC-induced VIPERIN response
- As an antiviral effector, VIPERIN appropriately recognizes PolyIC as a viral mimic


# R SESSION INFO

```{r session-info}
#| results: hold
#| label: session-info
sessionInfo()
```
