<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sam White">
<meta name="dcterms.date" content="2024-02-02">

<title>Sam’s Notebook - sRNA-seq Paired Read Merging Exploratios - Coral E5 Deep Dive</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-7483T61T11"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-7483T61T11', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Sam’s Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" rel="" target="_blank">
 <span class="dropdown-text">Rstudio Server - Raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu/Atumefaciens/" rel="" target="_blank">
 <span class="dropdown-text">My Gannet Directory</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" rel="" target="_blank">
 <span class="dropdown-text">Steven’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shedurkin.github.io/Roberts-LabNotebook/" rel="" target="_blank">
 <span class="dropdown-text">Kathleen’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://meganewing.github.io/mewing-notebook/" rel="" target="_blank">
 <span class="dropdown-text">Megan’s Notebook</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/sams-notebook" rel="" target="_blank">
 <span class="dropdown-text">Notebook repo</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" rel="" target="_blank">
 <span class="dropdown-text">Lab Organization (GitHub)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" rel="" target="_blank">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" rel="" target="_blank"><i class="bi bi-calendar-week" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../../index.xml" rel="" target=""><i class="bi bi-rss" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-right">
      <h1 class="title">sRNA-seq Paired Read Merging Exploratios - Coral E5 Deep Dive</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2024</div>
                <div class="quarto-category">E5</div>
                <div class="quarto-category">sRNA-seq</div>
                <div class="quarto-category">merge</div>
                <div class="quarto-category">bbmerge</div>
                <div class="quarto-category">fastp</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Sam White </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 2, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#intro" id="toc-intro" class="nav-link active" data-scroll-target="#intro">INTRO</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup">SETUP</a>
  <ul>
  <li><a href="#create-a-bash-variables-file" id="toc-create-a-bash-variables-file" class="nav-link" data-scroll-target="#create-a-bash-variables-file">Create a Bash variables file</a></li>
  <li><a href="#download-raw-srnaseq-reads" id="toc-download-raw-srnaseq-reads" class="nav-link" data-scroll-target="#download-raw-srnaseq-reads">Download raw sRNAseq reads</a>
  <ul>
  <li><a href="#verify-raw-read-checksums" id="toc-verify-raw-read-checksums" class="nav-link" data-scroll-target="#verify-raw-read-checksums">Verify raw read checksums</a></li>
  </ul></li>
  <li><a href="#create-adapters-fasta" id="toc-create-adapters-fasta" class="nav-link" data-scroll-target="#create-adapters-fasta">Create adapters FastA</a></li>
  <li><a href="#fastp" id="toc-fastp" class="nav-link" data-scroll-target="#fastp"><code>fastp</code></a>
  <ul>
  <li><a href="#paired-end-trimming" id="toc-paired-end-trimming" class="nav-link" data-scroll-target="#paired-end-trimming">Paired-end Trimming</a>
  <ul class="collapse">
  <li><a href="#polyg-trimming-only" id="toc-polyg-trimming-only" class="nav-link" data-scroll-target="#polyg-trimming-only">PolyG Trimming Only</a></li>
  <li><a href="#adapters-polyg-trimming" id="toc-adapters-polyg-trimming" class="nav-link" data-scroll-target="#adapters-polyg-trimming">Adapters &amp; PolyG Trimming</a></li>
  <li><a href="#polyg-trimming-max-length-31" id="toc-polyg-trimming-max-length-31" class="nav-link" data-scroll-target="#polyg-trimming-max-length-31">PolyG Trimming Max Length 31</a></li>
  <li><a href="#adapters-polyg-trimming-max-length-31" id="toc-adapters-polyg-trimming-max-length-31" class="nav-link" data-scroll-target="#adapters-polyg-trimming-max-length-31">Adapters &amp; PolyG Trimming Max Length 31</a></li>
  <li><a href="#adapters-polyg-trimming-max-length-31-with-merge" id="toc-adapters-polyg-trimming-max-length-31-with-merge" class="nav-link" data-scroll-target="#adapters-polyg-trimming-max-length-31-with-merge">Adapters &amp; PolyG Trimming Max Length 31 With Merge</a></li>
  </ul></li>
  <li><a href="#single-end-trimming---r1-reads-only" id="toc-single-end-trimming---r1-reads-only" class="nav-link" data-scroll-target="#single-end-trimming---r1-reads-only">Single-end Trimming - R1 Reads Only</a>
  <ul class="collapse">
  <li><a href="#auto-adapters-polyg-trimming" id="toc-auto-adapters-polyg-trimming" class="nav-link" data-scroll-target="#auto-adapters-polyg-trimming">Auto-adapters &amp; PolyG Trimming</a></li>
  <li><a href="#auto-adapters-polyg-trimming-max-length-31" id="toc-auto-adapters-polyg-trimming-max-length-31" class="nav-link" data-scroll-target="#auto-adapters-polyg-trimming-max-length-31">Auto-adapters &amp; PolyG Trimming Max Length 31</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#bbmerge" id="toc-bbmerge" class="nav-link" data-scroll-target="#bbmerge">BBMerge</a>
  <ul>
  <li><a href="#raw-reads" id="toc-raw-reads" class="nav-link" data-scroll-target="#raw-reads">Raw reads</a></li>
  <li><a href="#trimmed-polyg-only" id="toc-trimmed-polyg-only" class="nav-link" data-scroll-target="#trimmed-polyg-only">Trimmed PolyG Only</a></li>
  <li><a href="#trimmed-adapters-polyg" id="toc-trimmed-adapters-polyg" class="nav-link" data-scroll-target="#trimmed-adapters-polyg">Trimmed Adapters &amp; PolyG</a></li>
  <li><a href="#trimmed-adapters-polyg-max-length-31bp" id="toc-trimmed-adapters-polyg-max-length-31bp" class="nav-link" data-scroll-target="#trimmed-adapters-polyg-max-length-31bp">Trimmed Adapters &amp; PolyG Max Length 31bp</a></li>
  </ul></li>
  <li><a href="#fastqc" id="toc-fastqc" class="nav-link" data-scroll-target="#fastqc">FastQC</a></li>
  <li><a href="#multiqc" id="toc-multiqc" class="nav-link" data-scroll-target="#multiqc">MultiQC</a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-right" id="quarto-document-content">




<section id="intro" class="level1">
<h1>INTRO</h1>
<p>Our previous sRNA-seq data analysis for this project revealed that the R2 reads didn’t seem to match with anything (specifically, no clade assignments when using <a href="https://github.com/friedlanderlab/mirtrace">miRTrace</a>), which seemed odd. Since the insert sizes should be small for these libraries and the read lengths kind of long (150bp PE), we’d expect the resulting reads to overlap. And, in turn, R2 reads should be reverse complements of R1 reads (and vice versa). After discussing things with Azenta (company which performed library construction and sequencing), they indicated that their in-house pipeline discards the R2 reads because they come from the opposite strand and most miRNAs are found on the R1 strand.</p>
<p>It seemed like a bit of waste to discard all the data that exists in the R2 reads, so I wanted to explore merging the reads to see if we could preserve the R2 data. Additionally, I came across <a href="https://github.com/HCGB-IGTP/XICRA">XICRA</a> (GitHub repo) <span class="citation" data-cites="sanchezherrero2021">(<a href="#ref-sanchezherrero2021" role="doc-biblioref">Sanchez Herrero et al. 2021</a>)</span> which describes using paired-end sequence data to identify miRNA isoforms (isomiRs), suggesting that we should try to retain the R2 reads to improve our miRNA analyses.</p>
<p>I decided to explore merging/trimming using two different sets of software:</p>
<ul>
<li><p><a href="https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbmerge-guide/">BBMerge</a> <span class="citation" data-cites="bushnell2017">(<a href="#ref-bushnell2017" role="doc-biblioref">Bushnell, Rood, and Singer 2017</a>)</span></p></li>
<li><p><a href="https://github.com/OpenGene/fastp"><code>fastp</code></a><span class="citation" data-cites="chen2023 chen2018">(<a href="#ref-chen2023" role="doc-biblioref">Chen 2023</a>; <a href="#ref-chen2018" role="doc-biblioref">Chen et al. 2018</a>)</span></p></li>
</ul>
<p><a href="https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbmerge-guide/">BBMerge</a> seemed like a good tool, as it’s specifically designed to handle merging of Illumina PE sequencing reads. Plus, it’s part of the <a href="https://sourceforge.net/projects/bbmap/">BBTols suite</a>, which has a number of other tools, and packages like this usually have nice cross-compatibility between the different tools.</p>
<p><a href="https://github.com/OpenGene/fastp"><code>fastp</code></a> has both trimming <em>and</em> merging built into the tools itself, which I hadn’t previously realized (but, had never explored, since this isn’t how we usually work with PE sequencing data). I was simply going to use <a href="https://github.com/OpenGene/fastp"><code>fastp</code></a> for trimming, as it’s the trimmer thta I’m most familiar with. I honestly thought I’d trim with <a href="https://github.com/OpenGene/fastp"><code>fastp</code></a>, and then merge with <a href="https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbmerge-guide/">BBMerge</a>, but the fact that <a href="https://github.com/OpenGene/fastp"><code>fastp</code></a> had it built-in could make my life easier.</p>
<p>So, with all of that out of the way, let’s start the analysis. This will only use a single set of PE sequencing reads, as an initial exploration. We’ll decide on the best approach and then use that approach to run all the sequencing data through the chosen “pipeline.”</p>
</section>
<section id="setup" class="level1">
<h1>SETUP</h1>
<section id="create-a-bash-variables-file" class="level2">
<h2 class="anchored" data-anchor-id="create-a-bash-variables-file">Create a Bash variables file</h2>
<p>This allows usage of Bash variables across R Markdown chunks.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"#### Assign Variables ####"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"## Data URL"</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export raw_reads_url="https://owl.fish.washington.edu/nightingales/P_evermanni/30-852430235/"'</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# Data directories"</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export fastqc_out_dir="./fastqc"'</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export raw_reads_dir="./raw"'</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export trimmed_fastqs_dir="./trimmed-reads"'</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"## NEB nebnext-small-rna-library-prep-set-for-illumina adapters"</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export first_adapter="AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC"'</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export second_adapter="GATCGTCGGACTGTAGAACTCTGAACGTGTAGATCTCGGTGGTCGCCGTATCATT"'</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# Input/output files"</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export NEB_adapters_fasta=NEB-adapters.fasta'</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# Paths to programs"</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export programs_dir="/home/shared"'</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export bbmerge="${programs_dir}/bbmap-39.06/bbmerge.sh"'</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export fastp="${programs_dir}/fastp"'</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export fastqc="${programs_dir}/FastQC-0.12.1/fastqc"'</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export multiqc=/home/sam/programs/mambaforge/bin/multiqc'</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# Set number of CPUs to use"</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export threads=46'</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# Initialize arrays"</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export trimmed_fastqs_array=()'</span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="kw">}</span> <span class="op">&gt;</span> .bashvars</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> .bashvars</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#### Assign Variables ####

## Data URL
export raw_reads_url="https://owl.fish.washington.edu/nightingales/P_evermanni/30-852430235/"
# Data directories
export fastqc_out_dir="./fastqc"
export raw_reads_dir="./raw"
export trimmed_fastqs_dir="./trimmed-reads"

## NEB nebnext-small-rna-library-prep-set-for-illumina adapters
export first_adapter="AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC"
export second_adapter="GATCGTCGGACTGTAGAACTCTGAACGTGTAGATCTCGGTGGTCGCCGTATCATT"

# Input/output files
export NEB_adapters_fasta=NEB-adapters.fasta

# Paths to programs
export programs_dir="/home/shared"
export bbmerge="${programs_dir}/bbmap-39.06/bbmerge.sh"
export fastp="${programs_dir}/fastp"
export fastqc="${programs_dir}/FastQC-0.12.1/fastqc"
export multiqc=/home/sam/programs/mambaforge/bin/multiqc

# Set number of CPUs to use
export threads=46

# Initialize arrays
export trimmed_fastqs_array=()</code></pre>
</div>
</div>
</section>
<section id="download-raw-srnaseq-reads" class="level2">
<h2 class="anchored" data-anchor-id="download-raw-srnaseq-reads">Download raw sRNAseq reads</h2>
<p>Reads are downloaded from <a href="https://owl.fish.washington.edu/nightingales/P_evermanni/30-852430235/" class="uri">https://owl.fish.washington.edu/nightingales/P_evermanni/30-852430235/</a></p>
<p>The <code>--cut-dirs 3</code> command cuts the preceding directory structure (i.e.&nbsp;<code>nightingales/P_evermanni/30-852430235/</code>) so that we just end up with the reads.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">--parents</span> <span class="st">"</span><span class="va">${raw_reads_dir}</span><span class="st">"</span> <span class="st">"</span><span class="va">${trimmed_fastqs_dir}</span><span class="st">"</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>--directory-prefix <span class="va">${raw_reads_dir}</span> <span class="dt">\</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>--recursive <span class="dt">\</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>--no-check-certificate <span class="dt">\</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>--continue <span class="dt">\</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>--cut-dirs 3 <span class="dt">\</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>--no-host-directories <span class="dt">\</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>--no-parent <span class="dt">\</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>--quiet <span class="dt">\</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>--accept <span class="st">"sRNA-POR-79*,checksums.md5"</span> <span class="va">${raw_reads_url}</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-lh</span> <span class="st">"</span><span class="va">${raw_reads_dir}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>total 1.8G
-rw-r--r-- 1 sam sam  798 May 17  2023 checksums.md5
-rw-r--r-- 1 sam sam 899M May 17  2023 sRNA-POR-79-S1-TP2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 916M May 17  2023 sRNA-POR-79-S1-TP2_R2_001.fastq.gz</code></pre>
</div>
</div>
<section id="verify-raw-read-checksums" class="level3">
<h3 class="anchored" data-anchor-id="verify-raw-read-checksums">Verify raw read checksums</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="st">"</span><span class="va">${raw_reads_dir}</span><span class="st">"</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Checksums file contains other files, so this just looks for the sRNAseq files.</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="st">"sRNA-POR-79"</span> checksums.md5 <span class="kw">|</span> <span class="fu">md5sum</span> <span class="at">--check</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>sRNA-POR-79-S1-TP2_R1_001.fastq.gz: OK
sRNA-POR-79-S1-TP2_R2_001.fastq.gz: OK</code></pre>
</div>
</div>
</section>
</section>
<section id="create-adapters-fasta" class="level2">
<h2 class="anchored" data-anchor-id="create-adapters-fasta">Create adapters FastA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Creating adapters FastA."</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="va">adapter_count</span><span class="op">=</span>0</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Check for adapters file first</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Then create adapters file if doesn't exist</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="bu">[</span> <span class="ot">-f</span> <span class="st">"</span><span class="va">${NEB_adapters_fasta}</span><span class="st">"</span> <span class="bu">]</span><span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"</span><span class="va">${NEB_adapters_fasta}</span><span class="st"> already exists. Nothing to do."</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> adapter <span class="kw">in</span> <span class="st">"</span><span class="va">${first_adapter}</span><span class="st">"</span> <span class="st">"</span><span class="va">${second_adapter}</span><span class="st">"</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">do</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>    <span class="va">adapter_count</span><span class="op">=</span><span class="va">$((adapter_count</span> <span class="op">+</span> <span class="dv">1</span><span class="va">))</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>    <span class="bu">printf</span> <span class="st">"&gt;%s\n%s\n"</span> <span class="st">"adapter_</span><span class="va">${adapter_count}</span><span class="st">"</span> <span class="st">"</span><span class="va">${adapter}</span><span class="st">"</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span> <span class="op">&gt;&gt;</span> <span class="st">"</span><span class="va">${NEB_adapters_fasta}</span><span class="st">"</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="cf">fi</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Adapters FastA:"</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="st">"</span><span class="va">${NEB_adapters_fasta}</span><span class="st">"</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Creating adapters FastA.

NEB-adapters.fasta already exists. Nothing to do.

Adapters FastA:

&gt;adapter_1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC
&gt;adapter_2
GATCGTCGGACTGTAGAACTCTGAACGTGTAGATCTCGGTGGTCGCCGTATCATT</code></pre>
</div>
</div>
</section>
<section id="fastp" class="level2">
<h2 class="anchored" data-anchor-id="fastp"><a href="https://github.com/OpenGene/fastp"><code>fastp</code></a></h2>
<p><a href="https://github.com/OpenGene/fastp"><code>fastp</code></a> has many built-in functions, including before/after trimming plots, and read 1 and read 2 merging.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Setting <code>--overlap_len_require 17</code> forces <a href="https://github.com/OpenGene/fastp"><code>fastp</code></a> to look at reads starting at base 17, instead of the default 30 and results in more accurate analyses, since these are sRNA-seq libraries where the expected/desired read lengths will be 17 - 30bp in length.</p>
</div>
</div>
<section id="paired-end-trimming" class="level3">
<h3 class="anchored" data-anchor-id="paired-end-trimming">Paired-end Trimming</h3>
<p>The paired end trimming will examine how some of the trimming options impact sequencing read lengths, quality, and subsequent merging.</p>
<section id="polyg-trimming-only" class="level4">
<h4 class="anchored" data-anchor-id="polyg-trimming-only">PolyG Trimming Only</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${fastp}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>--in1 ./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz <span class="dt">\</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>--in2 ./raw/sRNA-POR-79-S1-TP2_R2_001.fastq.gz <span class="dt">\</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>--out1 ./trimmed-reads/fastp-polyG_only.R1.fq.gz <span class="dt">\</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>--out2 ./trimmed-reads/fastp-polyG_only.R2.fq.gz <span class="dt">\</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>--disable_adapter_trimming <span class="dt">\</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>--trim_poly_g <span class="dt">\</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>--overlap_len_require 17 <span class="dt">\</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>--thread <span class="va">${threads}</span> <span class="dt">\</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>--html <span class="st">"fastp-polyG-only.html"</span> <span class="dt">\</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>--json <span class="st">"fastp-polyG-only.json"</span> <span class="dt">\</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>--report_title <span class="st">"fastp-polyG-only"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: fastp uses up to 16 threads although you specified 46
Read1 before filtering:
total reads: 19762559
total bases: 2964383850
Q20 bases: 2604771220(87.8689%)
Q30 bases: 2427856243(81.9009%)

Read2 before filtering:
total reads: 19762559
total bases: 2964383850
Q20 bases: 2619314070(88.3595%)
Q30 bases: 2408732242(81.2557%)

Read1 after filtering:
total reads: 19450041
total bases: 1698288792
Q20 bases: 1478970498(87.0859%)
Q30 bases: 1370182460(80.6802%)

Read2 after filtering:
total reads: 19450041
total bases: 1849068926
Q20 bases: 1690862447(91.444%)
Q30 bases: 1596965342(86.3659%)

Filtering result:
reads passed filter: 38900082
reads failed due to low quality: 595592
reads failed due to too many N: 40
reads failed due to too short: 29404

Duplication rate: 10.4184%

Insert size peak (evaluated by paired-end reads): 29

JSON report: fastp-polyG-only.json
HTML report: fastp-polyG-only.html

/home/shared/fastp --in1 ./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz --in2 ./raw/sRNA-POR-79-S1-TP2_R2_001.fastq.gz --out1 ./trimmed-reads/fastp-polyG_only.R1.fq.gz --out2 ./trimmed-reads/fastp-polyG_only.R2.fq.gz --disable_adapter_trimming --trim_poly_g --overlap_len_require 17 --thread 46 --html fastp-polyG-only.html --json fastp-polyG-only.json --report_title fastp-polyG-only 
fastp v0.23.2, time used: 45 seconds</code></pre>
</div>
</div>
</section>
<section id="adapters-polyg-trimming" class="level4">
<h4 class="anchored" data-anchor-id="adapters-polyg-trimming">Adapters &amp; PolyG Trimming</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${fastp}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>--in1 ./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz <span class="dt">\</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>--in2 ./raw/sRNA-POR-79-S1-TP2_R2_001.fastq.gz <span class="dt">\</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>--out1 ./trimmed-reads/fastp-adapters-polyG.R1.fq.gz <span class="dt">\</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>--out2 ./trimmed-reads/fastp-adapters-polyG.R2.fq.gz <span class="dt">\</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>--adapter_fasta NEB-adapters.fasta <span class="dt">\</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>--trim_poly_g <span class="dt">\</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>--overlap_len_require 17 <span class="dt">\</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>--thread <span class="va">${threads}</span> <span class="dt">\</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>--html <span class="st">"fastp-adapters-polyG.html"</span> <span class="dt">\</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>--json <span class="st">"fastp-adapters-polyG.json"</span> <span class="dt">\</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>--report_title <span class="st">"fastp-adapters-polyG-only"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: fastp uses up to 16 threads although you specified 46
Read1 before filtering:
total reads: 19762559
total bases: 2964383850
Q20 bases: 2604771220(87.8689%)
Q30 bases: 2427856243(81.9009%)

Read2 before filtering:
total reads: 19762559
total bases: 2964383850
Q20 bases: 2619314070(88.3595%)
Q30 bases: 2408732242(81.2557%)

Read1 after filtering:
total reads: 16734573
total bases: 489646406
Q20 bases: 480139354(98.0584%)
Q30 bases: 466862298(95.3468%)

Read2 after filtering:
total reads: 16734573
total bases: 492483642
Q20 bases: 484765577(98.4328%)
Q30 bases: 469308001(95.2941%)

Filtering result:
reads passed filter: 33469146
reads failed due to low quality: 232788
reads failed due to too many N: 0
reads failed due to too short: 5823184
reads with adapter trimmed: 30461598
bases trimmed due to adapters: 2013836548

Duplication rate: 10.4184%

Insert size peak (evaluated by paired-end reads): 29

JSON report: fastp-adapters-polyG.json
HTML report: fastp-adapters-polyG.html

/home/shared/fastp --in1 ./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz --in2 ./raw/sRNA-POR-79-S1-TP2_R2_001.fastq.gz --out1 ./trimmed-reads/fastp-adapters-polyG.R1.fq.gz --out2 ./trimmed-reads/fastp-adapters-polyG.R2.fq.gz --adapter_fasta NEB-adapters.fasta --trim_poly_g --overlap_len_require 17 --thread 46 --html fastp-adapters-polyG.html --json fastp-adapters-polyG.json --report_title fastp-adapters-polyG-only 
fastp v0.23.2, time used: 36 seconds</code></pre>
</div>
</div>
</section>
<section id="polyg-trimming-max-length-31" class="level4">
<h4 class="anchored" data-anchor-id="polyg-trimming-max-length-31">PolyG Trimming Max Length 31</h4>
<p>This can’t be run because we know from the polyG only trimming, that most read lengths are &gt;89bp. As such, it will end up discarding all the reads…</p>
</section>
<section id="adapters-polyg-trimming-max-length-31" class="level4">
<h4 class="anchored" data-anchor-id="adapters-polyg-trimming-max-length-31">Adapters &amp; PolyG Trimming Max Length 31</h4>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The max length is based on the <code>fastp</code> insert peak size from the adapter and polyG trimming results, and previous evaluation of mean read lengths via <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"><code>FastQC</code></a> and <a href="https://multiqc.info/"><code>MultiQC</code></a>.</p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${fastp}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>--in1 ./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz <span class="dt">\</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>--in2 ./raw/sRNA-POR-79-S1-TP2_R2_001.fastq.gz <span class="dt">\</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>--out1 ./trimmed-reads/fastp-adapters-polyG-31bp.R1.fq.gz <span class="dt">\</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>--out2 ./trimmed-reads/fastp-adapters-polyG-31bp.R2.fq.gz <span class="dt">\</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>--adapter_fasta NEB-adapters.fasta <span class="dt">\</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>--trim_poly_g <span class="dt">\</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>--overlap_len_require 17 <span class="dt">\</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>--length_limit 31 <span class="dt">\</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>--thread <span class="va">${threads}</span> <span class="dt">\</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>--html <span class="st">"fastp-adapters-polyG-31bp.html"</span> <span class="dt">\</span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>--json <span class="st">"fastp-adapters-polyG-31bp.json"</span> <span class="dt">\</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>--report_title <span class="st">"fastp-adapters-polyG-31bp"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: fastp uses up to 16 threads although you specified 46
Read1 before filtering:
total reads: 19762559
total bases: 2964383850
Q20 bases: 2604771220(87.8689%)
Q30 bases: 2427856243(81.9009%)

Read2 before filtering:
total reads: 19762559
total bases: 2964383850
Q20 bases: 2619314070(88.3595%)
Q30 bases: 2408732242(81.2557%)

Read1 after filtering:
total reads: 13190855
total bases: 339210880
Q20 bases: 335529961(98.9149%)
Q30 bases: 327446980(96.532%)

Read2 after filtering:
total reads: 13190855
total bases: 338558013
Q20 bases: 335504115(99.098%)
Q30 bases: 325352134(96.0994%)

Filtering result:
reads passed filter: 26381710
reads failed due to low quality: 232788
reads failed due to too many N: 0
reads failed due to too short: 5354066
reads failed due to too long: 7556554
reads with adapter trimmed: 30461598
bases trimmed due to adapters: 2013836548

Duplication rate: 10.4184%

Insert size peak (evaluated by paired-end reads): 29

JSON report: fastp-adapters-polyG-31bp.json
HTML report: fastp-adapters-polyG-31bp.html

/home/shared/fastp --in1 ./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz --in2 ./raw/sRNA-POR-79-S1-TP2_R2_001.fastq.gz --out1 ./trimmed-reads/fastp-adapters-polyG-31bp.R1.fq.gz --out2 ./trimmed-reads/fastp-adapters-polyG-31bp.R2.fq.gz --adapter_fasta NEB-adapters.fasta --trim_poly_g --overlap_len_require 17 --length_limit 31 --thread 46 --html fastp-adapters-polyG-31bp.html --json fastp-adapters-polyG-31bp.json --report_title fastp-adapters-polyG-31bp 
fastp v0.23.2, time used: 36 seconds</code></pre>
</div>
</div>
</section>
<section id="adapters-polyg-trimming-max-length-31-with-merge" class="level4">
<h4 class="anchored" data-anchor-id="adapters-polyg-trimming-max-length-31-with-merge">Adapters &amp; PolyG Trimming Max Length 31 With Merge</h4>
<p>This will attempt to merge the R1 and R2 reads.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${fastp}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>--in1 ./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz <span class="dt">\</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>--in2 ./raw/sRNA-POR-79-S1-TP2_R2_001.fastq.gz <span class="dt">\</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>--adapter_fasta NEB-adapters.fasta <span class="dt">\</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>--trim_poly_g <span class="dt">\</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>--overlap_len_require 17 <span class="dt">\</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>--length_limit 31 <span class="dt">\</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>--merge <span class="dt">\</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>--merged_out ./trimmed-reads/fastp-adapters-polyG-31bp-merged.fq.gz <span class="dt">\</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>--thread <span class="va">${threads}</span> <span class="dt">\</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>--html <span class="st">"fastp-adapters-polyG-31bp-merged.html"</span> <span class="dt">\</span></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>--json <span class="st">"fastp-adapters-polyG-31bp-merged.json"</span> <span class="dt">\</span></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>--report_title <span class="st">"fastp-adapters-polyG-31bp-merged"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: fastp uses up to 16 threads although you specified 46
Read1 before filtering:
total reads: 19762559
total bases: 2964383850
Q20 bases: 2604771220(87.8689%)
Q30 bases: 2427856243(81.9009%)

Read2 before filtering:
total reads: 19762559
total bases: 2964383850
Q20 bases: 2619314070(88.3595%)
Q30 bases: 2408732242(81.2557%)

Merged and filtered:
total reads: 12259576
total bases: 324356716
Q20 bases: 320917135(98.9396%)
Q30 bases: 313148708(96.5445%)

Filtering result:
reads passed filter: 26506174
reads failed due to low quality: 179320
reads failed due to too many N: 0
reads failed due to too short: 5774946
reads failed due to too long: 7064678
reads with adapter trimmed: 30461598
bases trimmed due to adapters: 2013836548
reads corrected by overlap analysis: 484376
bases corrected by overlap analysis: 766038

Duplication rate: 10.4184%

Insert size peak (evaluated by paired-end reads): 29

Read pairs merged: 12259576
% of original read pairs: 62.0344%
% in reads after filtering: 100%


JSON report: fastp-adapters-polyG-31bp-merged.json
HTML report: fastp-adapters-polyG-31bp-merged.html

/home/shared/fastp --in1 ./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz --in2 ./raw/sRNA-POR-79-S1-TP2_R2_001.fastq.gz --adapter_fasta NEB-adapters.fasta --trim_poly_g --overlap_len_require 17 --length_limit 31 --merge --merged_out ./trimmed-reads/fastp-adapters-polyG-31bp-merged.fq.gz --thread 46 --html fastp-adapters-polyG-31bp-merged.html --json fastp-adapters-polyG-31bp-merged.json --report_title fastp-adapters-polyG-31bp-merged 
fastp v0.23.2, time used: 45 seconds</code></pre>
</div>
</div>
</section>
</section>
<section id="single-end-trimming---r1-reads-only" class="level3">
<h3 class="anchored" data-anchor-id="single-end-trimming---r1-reads-only">Single-end Trimming - R1 Reads Only</h3>
<p>Using just the R1 reads, per Azenta’s recommendation.</p>
<section id="auto-adapters-polyg-trimming" class="level4">
<h4 class="anchored" data-anchor-id="auto-adapters-polyg-trimming">Auto-adapters &amp; PolyG Trimming</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${fastp}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>--in1 ./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz <span class="dt">\</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>--out1 ./trimmed-reads/fastp-R1-auto_adapters-polyG.fq.gz <span class="dt">\</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>--trim_poly_g <span class="dt">\</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>--thread <span class="va">${threads}</span> <span class="dt">\</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>--html <span class="st">"fastp-R1-auto_adapters-polyG.html"</span> <span class="dt">\</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>--json <span class="st">"fastp-R1-auto_adapters-polyG.json"</span> <span class="dt">\</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>--report_title <span class="st">"fastp-R1-auto_adapters-polyG"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Detecting adapter sequence for read1...
&gt;Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

WARNING: fastp uses up to 16 threads although you specified 46
Read1 before filtering:
total reads: 19762559
total bases: 2964383850
Q20 bases: 2604771220(87.8689%)
Q30 bases: 2427856243(81.9009%)

Read1 after filtering:
total reads: 16984592
total bases: 540469705
Q20 bases: 522006375(96.5838%)
Q30 bases: 504657126(93.3738%)

Filtering result:
reads passed filter: 16984592
reads failed due to low quality: 30923
reads failed due to too many N: 1
reads failed due to too short: 2747043
reads with adapter trimmed: 18545632
bases trimmed due to adapters: 1166285709

Duplication rate (may be overestimated since this is SE data): 29.621%

JSON report: fastp-R1-auto_adapters-polyG.json
HTML report: fastp-R1-auto_adapters-polyG.html

/home/shared/fastp --in1 ./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz --out1 ./trimmed-reads/fastp-R1-auto_adapters-polyG.fq.gz --trim_poly_g --thread 46 --html fastp-R1-auto_adapters-polyG.html --json fastp-R1-auto_adapters-polyG.json --report_title fastp-R1-auto_adapters-polyG 
fastp v0.23.2, time used: 32 seconds</code></pre>
</div>
</div>
</section>
<section id="auto-adapters-polyg-trimming-max-length-31" class="level4">
<h4 class="anchored" data-anchor-id="auto-adapters-polyg-trimming-max-length-31">Auto-adapters &amp; PolyG Trimming Max Length 31</h4>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The max length is based on the <code>fastp</code> insert peak size from the adapter and polyG trimming results, and previous evaluation of mean read lengths via <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"><code>FastQC</code></a> and <a href="https://multiqc.info/"><code>MultiQC</code></a>.</p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${fastp}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>--in1 ./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz <span class="dt">\</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>--out1 ./trimmed-reads/fastp-R1-31bp-auto_adapters-polyG.fq.gz <span class="dt">\</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>--trim_poly_g <span class="dt">\</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>--length_limit 31 <span class="dt">\</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>--thread <span class="va">${threads}</span> <span class="dt">\</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>--html <span class="st">"fastp-R1-auto_adapters-polyG-31bp.html"</span> <span class="dt">\</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>--json <span class="st">"fastp-R1-auto_adapters-polyG-31bp.json"</span> <span class="dt">\</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>--report_title <span class="st">"fastp-R1-auto_adapters-polyG-31bp"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Detecting adapter sequence for read1...
&gt;Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

WARNING: fastp uses up to 16 threads although you specified 46
Read1 before filtering:
total reads: 19762559
total bases: 2964383850
Q20 bases: 2604771220(87.8689%)
Q30 bases: 2427856243(81.9009%)

Read1 after filtering:
total reads: 12801426
total bases: 328647991
Q20 bases: 325069951(98.9113%)
Q30 bases: 317201339(96.517%)

Filtering result:
reads passed filter: 12801426
reads failed due to low quality: 30923
reads failed due to too many N: 1
reads failed due to too short: 2747043
reads failed due to too long: 4183166
reads with adapter trimmed: 18545632
bases trimmed due to adapters: 1166285709

Duplication rate (may be overestimated since this is SE data): 29.621%

JSON report: fastp-R1-auto_adapters-polyG-31bp.json
HTML report: fastp-R1-auto_adapters-polyG-31bp.html

/home/shared/fastp --in1 ./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz --out1 ./trimmed-reads/fastp-R1-31bp-auto_adapters-polyG.fq.gz --trim_poly_g --length_limit 31 --thread 46 --html fastp-R1-auto_adapters-polyG-31bp.html --json fastp-R1-auto_adapters-polyG-31bp.json --report_title fastp-R1-auto_adapters-polyG-31bp 
fastp v0.23.2, time used: 32 seconds</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="bbmerge" class="level2">
<h2 class="anchored" data-anchor-id="bbmerge">BBMerge</h2>
<section id="raw-reads" class="level3">
<h3 class="anchored" data-anchor-id="raw-reads">Raw reads</h3>
<p><a href="https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbmerge-guide/">BBMerge</a> documentations actually indicates that merging performs best using <em>raw</em>, untrimmed reads, so we’ll try that out first…</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="va">${bbmerge}</span> <span class="dt">\</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>in1=./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz <span class="dt">\</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>in2=./raw/sRNA-POR-79-S1-TP2_R2_001.fastq.gz <span class="dt">\</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>adapters=./NEB-adapters.fasta <span class="dt">\</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>mininsert=17 <span class="dt">\</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>out=./trimmed-reads/sRNA-POR-79-S1-TP2-raw-bbmerge.fq.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>java -ea -Xmx1000m -Xms1000m -Djava.library.path=/home/shared/bbmap-39.06/jni/ -cp /home/shared/bbmap-39.06/current/ jgi.BBMerge in1=./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz in2=./raw/sRNA-POR-79-S1-TP2_R2_001.fastq.gz adapters=./NEB-adapters.fasta mininsert=17 out=./trimmed-reads/sRNA-POR-79-S1-TP2-raw-bbmerge.fq.gz
Executing jgi.BBMerge [in1=./raw/sRNA-POR-79-S1-TP2_R1_001.fastq.gz, in2=./raw/sRNA-POR-79-S1-TP2_R2_001.fastq.gz, adapters=./NEB-adapters.fasta, mininsert=17, out=./trimmed-reads/sRNA-POR-79-S1-TP2-raw-bbmerge.fq.gz]
Version 39.06

Writing mergable reads merged.
Started output threads.
Total time: 61.730 seconds.

Pairs:                  19762559
Joined:                 151908      0.769%
Ambiguous:              0           0.000%
No Solution:            19610645    99.231%
Too Short:              6           0.000%
Adapters Expected:      60129       0.152%
Adapters Found:         58428       0.148%

Avg Insert:             131.5
Standard Deviation:     62.2
Mode:                   156

Insert range:           17 - 289
90th percentile:        203
75th percentile:        156
50th percentile:        156
25th percentile:        72
10th percentile:        32</code></pre>
</div>
</div>
</section>
<section id="trimmed-polyg-only" class="level3">
<h3 class="anchored" data-anchor-id="trimmed-polyg-only">Trimmed PolyG Only</h3>
<p>Running <a href="https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbmerge-guide/">BBMerge</a> on just raw reads yielded poor results (&gt;99% of reads could <em>not</em> be merged). Additionally, the average insert size was determined to be 131.5bp, which is not going to be useful for sRNA-seq.</p>
<p>I know that there’s a very large stretch of polyG sequence in the reads (likely due to the small insert size and “long” read length - Gs get added as a “placeholder” base each cycle that exceeds the insert size), so let’s try merging reads <em>after</em> polyG sequence has been trimmed.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="va">${bbmerge}</span> <span class="dt">\</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>in1=./trimmed-reads/fastp-polyG_only.R1.fq.gz <span class="dt">\</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>in2=./trimmed-reads/fastp-polyG_only.R2.fq.gz <span class="dt">\</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>adapters=./NEB-adapters.fasta <span class="dt">\</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>mininsert=17 <span class="dt">\</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>out=./trimmed-reads/sRNA-POR-79-S1-TP2-fastp-polyG_only-bbmerge.fq.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>java -ea -Xmx1000m -Xms1000m -Djava.library.path=/home/shared/bbmap-39.06/jni/ -cp /home/shared/bbmap-39.06/current/ jgi.BBMerge in1=./trimmed-reads/fastp-polyG_only.R1.fq.gz in2=./trimmed-reads/fastp-polyG_only.R2.fq.gz adapters=./NEB-adapters.fasta mininsert=17 out=./trimmed-reads/sRNA-POR-79-S1-TP2-fastp-polyG_only-bbmerge.fq.gz
Executing jgi.BBMerge [in1=./trimmed-reads/fastp-polyG_only.R1.fq.gz, in2=./trimmed-reads/fastp-polyG_only.R2.fq.gz, adapters=./NEB-adapters.fasta, mininsert=17, out=./trimmed-reads/sRNA-POR-79-S1-TP2-fastp-polyG_only-bbmerge.fq.gz]
Version 39.06

Writing mergable reads merged.
Started output threads.
Total time: 62.003 seconds.

Pairs:                  19450041
Joined:                 12649853    65.038%
Ambiguous:              0           0.000%
No Solution:            5981377     30.753%
Too Short:              818811      4.210%
Adapters Expected:      12781441    32.857%
Adapters Found:         12553911    32.272%

Avg Insert:             30.0
Standard Deviation:     14.6
Mode:                   29

Insert range:           17 - 289
90th percentile:        37
75th percentile:        31
50th percentile:        29
25th percentile:        25
10th percentile:        21</code></pre>
</div>
</div>
<p>The results from using polyG-trimmed reads are certainly much better than just merging raw reads (12,649,853 reads could be joined this time). Mean insert size is now determined to be 30bp, which is what we might expect. However, the insert range (17 - 289) is still concerning, as reads &gt;30bp likely aren’t usable. Adapter sequences are still present, which are certainly contributing to some of the inability to merge, as well as the large read lengths. So, let’s try merging reads that have been trimmed of both polyG and adapter sequences.</p>
</section>
<section id="trimmed-adapters-polyg" class="level3">
<h3 class="anchored" data-anchor-id="trimmed-adapters-polyg">Trimmed Adapters &amp; PolyG</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="va">${bbmerge}</span> <span class="dt">\</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>in1=./trimmed-reads/fastp-adapters-polyG.R1.fq.gz <span class="dt">\</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>in2=./trimmed-reads/fastp-adapters-polyG.R2.fq.gz <span class="dt">\</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>adapters=./NEB-adapters.fasta <span class="dt">\</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>mininsert=17 <span class="dt">\</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>out=./trimmed-reads/sRNA-POR-79-S1-TP2-fastp-adapters-polyG-bbmerge.fq.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>java -ea -Xmx1000m -Xms1000m -Djava.library.path=/home/shared/bbmap-39.06/jni/ -cp /home/shared/bbmap-39.06/current/ jgi.BBMerge in1=./trimmed-reads/fastp-adapters-polyG.R1.fq.gz in2=./trimmed-reads/fastp-adapters-polyG.R2.fq.gz adapters=./NEB-adapters.fasta mininsert=17 out=./trimmed-reads/sRNA-POR-79-S1-TP2-fastp-adapters-polyG-bbmerge.fq.gz
Executing jgi.BBMerge [in1=./trimmed-reads/fastp-adapters-polyG.R1.fq.gz, in2=./trimmed-reads/fastp-adapters-polyG.R2.fq.gz, adapters=./NEB-adapters.fasta, mininsert=17, out=./trimmed-reads/sRNA-POR-79-S1-TP2-fastp-adapters-polyG-bbmerge.fq.gz]
Version 39.06

Writing mergable reads merged.
Started output threads.
Total time: 31.592 seconds.

Pairs:                  16734573
Joined:                 15533284    92.822%
Ambiguous:              0           0.000%
No Solution:            603776      3.608%
Too Short:              597513      3.571%
Adapters Expected:      28222       0.084%
Adapters Found:         18307       0.055%

Avg Insert:             29.4
Standard Deviation:     13.2
Mode:                   29

Insert range:           17 - 289
90th percentile:        36
75th percentile:        31
50th percentile:        29
25th percentile:        25
10th percentile:        21</code></pre>
</div>
</div>
<p>Well, trimming adapters certainly made a difference, as well! Now, 15,533,284 reads get joined (as opposed to 12,649,853 when using just polyG-trimmed reads). The average insert size changes very little, but, oddly, the insert range remains the same (17 - 289).</p>
<p>For kicks, let’s throw the reads that have been trimmed to 31bp into <a href="https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbmerge-guide/">BBMerge</a> and see what happens.</p>
</section>
<section id="trimmed-adapters-polyg-max-length-31bp" class="level3">
<h3 class="anchored" data-anchor-id="trimmed-adapters-polyg-max-length-31bp">Trimmed Adapters &amp; PolyG Max Length 31bp</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="va">${bbmerge}</span> <span class="dt">\</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>in1=./trimmed-reads/fastp-adapters-polyG-31bp.R1.fq.gz <span class="dt">\</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>in2=./trimmed-reads/fastp-adapters-polyG-31bp.R2.fq.gz <span class="dt">\</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>adapters=./NEB-adapters.fasta <span class="dt">\</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>mininsert=17 <span class="dt">\</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>out=./trimmed-reads/sRNA-POR-79-S1-TP2-fastp-adapters-polyG-31bp-bbmerge.fq.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>java -ea -Xmx1000m -Xms1000m -Djava.library.path=/home/shared/bbmap-39.06/jni/ -cp /home/shared/bbmap-39.06/current/ jgi.BBMerge in1=./trimmed-reads/fastp-adapters-polyG-31bp.R1.fq.gz in2=./trimmed-reads/fastp-adapters-polyG-31bp.R2.fq.gz adapters=./NEB-adapters.fasta mininsert=17 out=./trimmed-reads/sRNA-POR-79-S1-TP2-fastp-adapters-polyG-31bp-bbmerge.fq.gz
Executing jgi.BBMerge [in1=./trimmed-reads/fastp-adapters-polyG-31bp.R1.fq.gz, in2=./trimmed-reads/fastp-adapters-polyG-31bp.R2.fq.gz, adapters=./NEB-adapters.fasta, mininsert=17, out=./trimmed-reads/sRNA-POR-79-S1-TP2-fastp-adapters-polyG-31bp-bbmerge.fq.gz]
Version 39.06

Writing mergable reads merged.
Started output threads.
Total time: 24.184 seconds.

Pairs:                  13190855
Joined:                 12346728    93.601%
Ambiguous:              0           0.000%
No Solution:            281891      2.137%
Too Short:              562236      4.262%
Adapters Expected:      4           0.000%
Adapters Found:         4           0.000%

Avg Insert:             26.3
Standard Deviation:     4.0
Mode:                   29

Insert range:           17 - 35
90th percentile:        31
75th percentile:        29
50th percentile:        28
25th percentile:        24
10th percentile:        20</code></pre>
</div>
</div>
<p>This resulted in 12,346,728 reads being joined. This is on par with the 12,649,853 reads joined when using just polyG-trimmed reads. Additionally, we see the average insert size is 26.3bp, which is even more inline with what we’d expect for miRNAs (22-24bp), and the insert range is also inline with what we’d expect for these libraries (17 - 35bp). However, this is about 20% fewer reads than the 15,533,284 reads when using polyG- and adapter-trimmed reads.</p>
<p>Maybe I’ll test out the subsequent downstream analyses using these different merging results…</p>
</section>
</section>
<section id="fastqc" class="level2">
<h2 class="anchored" data-anchor-id="fastqc">FastQC</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Populate array with FastQ files</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="va">fastq_array</span><span class="op">=</span><span class="va">(</span>./trimmed-reads/<span class="pp">*</span>.fq.gz<span class="va">)</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Pass array contents to new variable</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="va">fastqc_list</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="st">"</span><span class="va">${fastq_array</span><span class="op">[*]</span><span class="va">}</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Run FastQC</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: Do NOT quote ${fastqc_list}</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a><span class="va">${fastqc}</span> <span class="dt">\</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>--threads <span class="va">${threads}</span> <span class="dt">\</span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>--quiet <span class="dt">\</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>--outdir ./ <span class="dt">\</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a><span class="va">${fastqc_list}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip</code></pre>
</div>
</div>
</section>
<section id="multiqc" class="level2">
<h2 class="anchored" data-anchor-id="multiqc">MultiQC</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="va">${multiqc}</span> .</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
  /// MultiQC 🔍 | v1.14

|           multiqc | MultiQC Version v1.19 now available!
|           multiqc | Search path : /home/shared/8TB_HDD_01/sam/gitrepos/RobertsLab/sams-notebook/posts/2024/2024-02-02-sRNA-seq-Paired-Read-Merging-Exploratios---Coral-E5-Deep-Dive
|         searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 60/60  
|            fastqc | Found 13 reports
|           multiqc | Compressing plot data
|           multiqc | Previous MultiQC output found! Adjusting filenames..
|           multiqc | Use -f or --force to overwrite existing reports instead
|           multiqc | Report      : multiqc_report_1.html
|           multiqc | Data        : multiqc_data_1
|           multiqc | MultiQC complete</code></pre>
</div>
</div>
<p>Check out the <a href="https://multiqc.info/"><code>MultiQC</code></a> results here (HTML):</p>
<ul>
<li><a href="./multiqc_report.html">MultiQC Report</a></li>
</ul>



</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-bushnell2017" class="csl-entry" role="listitem">
Bushnell, Brian, Jonathan Rood, and Esther Singer. 2017. <span>“BBMerge <span></span> Accurate Paired Shotgun Read Merging via Overlap.”</span> Edited by Patrick Jon Biggs. <em>PLOS ONE</em> 12 (10): e0185056. <a href="https://doi.org/10.1371/journal.pone.0185056">https://doi.org/10.1371/journal.pone.0185056</a>.
</div>
<div id="ref-chen2023" class="csl-entry" role="listitem">
Chen, Shifu. 2023. <span>“Ultrafast One<span>-</span>Pass FASTQ Data Preprocessing, Quality Control, and Deduplication Using Fastp.”</span> <em>iMeta</em> 2 (2). <a href="https://doi.org/10.1002/imt2.107">https://doi.org/10.1002/imt2.107</a>.
</div>
<div id="ref-chen2018" class="csl-entry" role="listitem">
Chen, Shifu, Yanqing Zhou, Yaru Chen, and Jia Gu. 2018. <span>“Fastp: An Ultra-Fast All-in-One FASTQ Preprocessor.”</span> <em>Bioinformatics</em> 34 (17): i884–90. <a href="https://doi.org/10.1093/bioinformatics/bty560">https://doi.org/10.1093/bioinformatics/bty560</a>.
</div>
<div id="ref-sanchezherrero2021" class="csl-entry" role="listitem">
Sanchez Herrero, Jose Francisco, Raquel Pluvinet, Antonio Luna de Haro, and Lauro Sumoy. 2021. <span>“Paired-End Small RNA Sequencing Reveals a Possible Overestimation in the isomiR Sequence Repertoire Previously Reported from Conventional Single Read Data Analysis.”</span> <em>BMC Bioinformatics</em> 22 (1). <a href="https://doi.org/10.1186/s12859-021-04128-1">https://doi.org/10.1186/s12859-021-04128-1</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>