---
layout: post
title: Transcript Identification and Alignments - C.virginica RNAseq with NCBI Genome GCF_002022765.2 Using Hisat2 and Stringtie on Mox Again
date: '2023-08-21 14:41'
tags: 
  - hisat2
  - mox
  - Crassostrea virginica
  - stringtie
  - RNAseq
  - Eastern oyster
categories: 
  - Miscellaneous
---
In the process of generating expression matrices for [CEABIGR](https://github.com/sr320/ceabigr/tree/main/data) (GitHub repo), and in turn distance matrices for these samples, I realized a couple of things:

1. [DESeq2](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html) has a function (`assay()`) that will automatically calculate/produce the distance matrices I want.

2. [`StringTie`](https://ccb.jhu.edu/software/stringtie/) comes with a script (`prepDE.py3`) which will spit out expression matrices formatted for import into [DESeq2](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html).

So, this should be easy, right? Of course not. LOL. When trying to run the [`StringTie`](https://ccb.jhu.edu/software/stringtie/) `prepDE.py3` script on the [original output from Stringtie on ](https://robertslab.github.io/sams-notebook/2022/02/25/Transcript-Identification-and-Alignments-C.virginica-RNAseq-with-NCBI-Genome-GCF_002022765.2-Using-Hisat2-and-Stringtie-on-Mox.html) (notebook), I encountered a parsing error for the very first GTF file it encountered. Considering the GTF file(s) was/are generated by [`StringTie`](https://ccb.jhu.edu/software/stringtie/), it seemed odd that it wouldn't be able to parse the GTF. Doing some investigating, I came across a number of [issues in the Stringtie GitHub repo](https://github.com/gpertea/stringtie/issues) indicating that using a version of Stringtie (e.g. <`v2.2.0`), with a certain combination of arguments (specifically, the `-e` argument seemed potentially problematic) could create various errors if using that output with an updated version of [`StringTie`](https://ccb.jhu.edu/software/stringtie/). Well, I was doing just that! Was it causing the parsing error I was experiencing? Don't know, but I'd also like to avoid the other downstream errors potentially associated with the output(s) from the older version of Stringtie, so I've decided to re-run the analysis using a the most current version of [`StringTie`](https://ccb.jhu.edu/software/stringtie/).


I used the [trimmed RNAseq reads from 20220224](https://robertslab.github.io/sams-notebook/2022/02/24/Trimming-Additional-20bp-from-C.virginica-Gonad-RNAseq-with-fastp-on-Mox.html) (notebook). I also needed to identify alternative transcripts in the [_Crassostrea virginica_ (Eastern oyster)](https://en.wikipedia.org/wiki/Eastern_oyster) gonad RNAseq data we have. I previously used [`HISAT2`](https://daehwankimlab.github.io/hisat2/) to index the NCBI [_Crassostrea virginica_ (Eastern oyster)](https://en.wikipedia.org/wiki/Eastern_oyster) genome and identify exon/splice sites [on 20210720](https://robertslab.github.io/sams-notebook/2021/07/20/Genome-Annotations-Splice-Site-and-Exon-Extractions-for-C.virginica-GCF_002022765.2-Genome-Using-Hisat2-on-Mox.html). Then, I used this genome index to run [`StringTie`](https://ccb.jhu.edu/software/stringtie/) on Mox in order to map sequencing reads to the genome/alternative isoforms.

Job was run on Mox.

Skip to [RESULTS section](#results).


SLURM Script (GitHub):

- [20230821-cvir-stringtie-GCF_002022765.2-isoforms.sh](https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20230821-cvir-stringtie-GCF_002022765.2-isoforms.sh)


```bash

```

---

#### RESULTS

Output folder:

- []()

