---
author: Sam White
toc-title: Contents
toc-depth: 5
toc-location: left
date: 2018-02-05 17:29:00+00:00
layout: post
slug: novaseq-assembly-trimmed-geoduck-novaseq-with-meraculous
title: NovaSeq Assembly - Trimmed Geoduck NovaSeq with Meraculous
categories:
  - "2018"
  - "Geoduck Genome Sequencing"
  - "assembly"
  - "geoduck"
  - "Illumina"
  - "kmer"
  - "kmer genie"
  - "meraculous"
  - "NovaSeq"
  - "Panopea generosa"
  - "roadrunner"
---

Attempted to use [Meraculous](https://jgi.doe.gov/data-and-tools/meraculous/) to assemble the [trimmed geoduck NovaSeq data](https://robertslab.github.io/sams-notebook/posts/2018/2018-01-25-adapter-trimming-and-fastqc-illumina-geoduck-novaseq-data/).


After a bunch of various issues (running out of hard drive space - multiple times, config file issues, typos), I've finally given up on running meraculous. It failed, again, saying it couldn't find a file in a directory that _meraculous created_! I've emailed the authors and if they have an easy fix, I'll implement it and see what happens.

Anyway, it's all documented in the Jupyter Notebook below.

One good thing came out of all of it is that I had to run kmergenie to identify an appopriate kmer size to use for assembly, as well as estimated genome size (this info is needed for both meraculous and SOAPdeNovo (which I'll be trying next)):

**kmergenie output folder**: [https://owl.fish.washington.edu/Athaliana/20180125_geoduck_novaseq/20180206_kmergenie/](http://owl.fish.washington.edu/Athaliana/20180125_geoduck_novaseq/20180206_kmergenie/)
kmergenie HTML report (doesn't display histograms for some reason): [20180206_kmergenie/histograms_report.html](https://owl.fish.washington.edu/Athaliana/20180125_geoduck_novaseq/20180206_kmergenie/histograms_report.html)
**kmer size**: 117
**Est. genome size**: 2.17Gbp

**Jupyter Notebook (GitHub)**: [20180205_roadrunner_meraculous_geoduck_novaseq.ipynb](https://github.com/sr320/LabDocs/blob/master/jupyter_nbs/sam/20180205_roadrunner_meraculous_geoduck_novaseq.ipynb)

