---
author: Sam White
toc-title: Contents
toc-depth: 5
toc-location: left
title: qPCR - C.gigas Alternative Oxidases Primer Tests
date: '2025-05-27'
draft: false
engine: knitr
categories: 
  - "2025"
  - "Crassotrea gigas"
  - "Pacific oyster"
  - "Magallana gigas"
  - "primers"
  - "AOX"
  - "alternative oxidase"
  - "XM_066083499.1"
  - "XM_034462847.2"
  - "NM_001305360.1"
---
# INTRO

I previously designed primers for three alternative oxidase (AOX) variants on [20250117](../2025-01-17-Primer-Design---C.gigas-Alternative-Oxidase-AOX-Variants-Using-Primer3/index.qmd) (Notebook entry). Alternative oxidase is a mitochondrial gene/enzyme known to be
responsive to oxidative stress, so this may be a useful biomarker that
we havenâ€™t previously considered.

I needed to test the primer sets to see if they worked and were template-specific. I also used this as an opportunity to introduce some of the undergrads in lab to qPCR. Carolyn, Madeline, and Noah all participated.


## Alternative oxidase variants

The primers were designed on the following NCBI sequences:

- [NP_001292289.1](https://www.ncbi.nlm.nih.gov/protein/NP_001292289.1?report=genbank&log$=protalign&blast_rank=1&RID=SMV9RMT6013):
  alternative oxidase, mitochondrial-like _Magallana gigas_

  - Nucleotide:
    [`NM_001305360.1`](https://www.ncbi.nlm.nih.gov/nuccore/NM_001305360.1)

  - Gene: `LOC105322704`

- [`XP_034318738.2`](https://www.ncbi.nlm.nih.gov/protein/XP_034318738.2?report=genbank&log$=protalign&blast_rank=2&RID=SMV9RMT6013):
  alternative oxidase, mitochondrial-like isoform X2 _Magallana
  gigas_

  - Nucleotide:
    [`XM_034462847.2`](https://www.ncbi.nlm.nih.gov/nuccore/XM_034462847.2)

  - Gene: `LOC105322704`

- [`XP_065939571.1`](https://www.ncbi.nlm.nih.gov/protein/XP_065939571.1?report=genbank&log$=protalign&blast_rank=3&RID=SMV9RMT6013):
  alternative oxidase, mitochondrial-like isoform X1 _Magallana
  gigas_

  - Nucleotide:
    [`XM_066083499.1`](https://www.ncbi.nlm.nih.gov/nuccore/XM_066083499.1)

  - Gene: `LOC105322704`

# MATERIALS & METHODS

## Primers

| Gene    | SR_ID | Primer Name               |
|---------|-------|---------------------------|
| AOX vX1 | 1849  | cgig-AOX-XM_066083499.1-R |
| AOX vX1 | 1848  | cgig-AOX-XM_066083499.1-F |
| AOX vX2 | 1847  | cgig-AOX-XM_034462847.2-R |
| AOX vX2 | 1846  | cgig-AOX-XM_034462847.2-F |
| AOX     | 1845  | cgig-AOX-NM_001305360-R   |
| AOX     | 1844  | cgig-AOX-NM_001305360-F   |

::: {.callout-note}
The `vX1` and `vX2` are NCBI notations to indicate different isoforms.
:::

## Reactions

All samples were run in duplicate, on low-profile, white 96-well plates (USA Scientific) in a CFX Connect (BioRad) real-time thermalcycler. All reactions consisted of the following:

| Component                                          | Stock Concentration | Volume (uL) |
|----------------------------------------------------|---------------------|-------------|
| cDNA                                               | NA                  | 1           |
| SsoAdvanced Universal SYBR Green Supermix (BioRad) | 2x                  | 10          |
| P<sub>F</sub>                                      | 10uM                | 0.5         |
| P<sub>R</sub>                                      | 10uM                | 0.5         |
| H<sub>2</sub>O                                     | NA                  | 8           |
| TOTAL                                              |                     | 20          |

A pool of cDNA from the various samples in the lifestage carryover project was used as the source of cDNA, in hopes of obtaining a source of cDNA with AOX1 expression. 

For cycling parameters, plate layouts, etc. see the [RESULTS](#results) section below.

# RESULTS

## Summary

All three primers work great! Amplfication is seen in all three primers, with AOX (SR IDs 1844/5) coming up the earliest at around 28 Cqs and AOX vX2 (SR IDs 1846/7) coming up the latest at around 33 Cqs. Technically, these values don't really matter, but they do show that AOX expression is likely at high enough levels to be easily detected in future experiments. No amplification is present in the no template controls (NTCs).


### Plots

::: {layout-ncol="2"}
![Amplification plots](./20250527-cgig-aox-amp-plots-sam.png){#fig-amp-plots-sam fig-alt="qPCR amplification plots. AOX (SR IDs 1844/5) is shown in gold. AOX vX2 (SR IDs 1846/7) is shown in light blue. AOX vX1 (SR IDS 1848/9) is shown in dark blue. No amplification seen in the NTCs." .lightbox}

![Melt plots.](./20250527-cgig-aox-melt-plots-sam.png){#fig-melt-plots-sam fig-alt="qPCR melt plots. AOX (SR IDs 1844/5) is shown in gold. AOX vX2 (SR IDs 1846/7) is shown in light blue. AOX vX1 (SR IDS 1848/9) is shown in dark blue. No amplification seen in the NTCs." .lightbox}

- AOX (SR IDs 1844/5) is shown in gold.
- AOX vX2 (SR IDs 1846/7) is shown in light blue.
- AOX vX1 (SR IDS 1848/9) is shown in dark blue.
- NTCs are shown in red.
:::

### Data file

qPCR Data file (CFX Maestro):

- [`admin_2025-05-27_11-07-38_Connect-cgig-AOX-primers.pcrd`](./admin_2025-05-27_11-07-38_Connect-cgig-AOX-primers.pcrd)

## Undergrad results

Undergrads did great! Replicates are tight and results mirror what I obtained. Additionally, despite potential amplification in NTCs (as seen in red in the plots below), this amplification did _not_ occur in both of their replicates and did _not_ occur in any of their samples containing template. Very nice!

### Plots

::: {layout-ncol="2"}
![Undergrad Amplification plots](./20250527-cgig-aox-amp-plots-undergrads.png){#fig-amp-plots-undergrads fig-alt="Undergrad qPCR amplification plots." .lightbox}

![Undergrad Melt plots.](./20250527-cgig-aox-melt-plots-undergrads.png){#fig-melt-plots-undergrads fig-alt="Undergrad qPCR melt plots." .lightbox}

Undergrad qPCR amplification and melt plots. See data file ([`admin_2025-05-27_11-07-38_Connect-cgig-AOX-primers.pcrd`](./admin_2025-05-27_11-07-38_Connect-cgig-AOX-primers.pcrd)) for which samples belong to each person.
:::