<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.552">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sam White">
<meta name="dcterms.date" content="2024-03-27">

<title>Sam’s Notebook - qPCR Analysis - C.gigas Lifestages Carryover from 20240325</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 2,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 100,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-7483T61T11"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-7483T61T11', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Sam’s Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" target="_blank">
 <span class="dropdown-text">Rstudio Server - Raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu/Atumefaciens/" target="_blank">
 <span class="dropdown-text">My Gannet Directory</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" target="_blank">
 <span class="dropdown-text">Steven’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shedurkin.github.io/Roberts-LabNotebook/" target="_blank">
 <span class="dropdown-text">Kathleen’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://meganewing.github.io/mewing-notebook/" target="_blank">
 <span class="dropdown-text">Megan’s Notebook</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/sams-notebook" target="_blank">
 <span class="dropdown-text">Notebook repo</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" target="_blank">
 <span class="dropdown-text">Lab Organization (GitHub)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" target="_blank">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" target="_blank"> <i class="bi bi-calendar-week" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-right">
      <div class="quarto-title-block"><div><h1 class="title">qPCR Analysis - C.gigas Lifestages Carryover from 20240325</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
                                <div class="quarto-categories">
                <div class="quarto-category">2024</div>
                <div class="quarto-category">qPCR</div>
                <div class="quarto-category">Crassotrea gigas</div>
                <div class="quarto-category">Pacific oyster</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-right">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Sam White </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">March 27, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#intro" id="toc-intro" class="nav-link active" data-scroll-target="#intro">INTRO</a></li>
  <li><a href="#analysis" id="toc-analysis" class="nav-link" data-scroll-target="#analysis">ANALYSIS</a>
  <ul>
  <li><a href="#load-libraries" id="toc-load-libraries" class="nav-link" data-scroll-target="#load-libraries">Load libraries</a></li>
  <li><a href="#set-variables" id="toc-set-variables" class="nav-link" data-scroll-target="#set-variables">Set variables</a></li>
  <li><a href="#functions" id="toc-functions" class="nav-link" data-scroll-target="#functions">Functions</a></li>
  <li><a href="#read-in-files" id="toc-read-in-files" class="nav-link" data-scroll-target="#read-in-files">Read in files</a></li>
  <li><a href="#unique-samples-by-target" id="toc-unique-samples-by-target" class="nav-link" data-scroll-target="#unique-samples-by-target">Unique samples by target</a></li>
  <li><a href="#add-life-stage-and-treatment-cols" id="toc-add-life-stage-and-treatment-cols" class="nav-link" data-scroll-target="#add-life-stage-and-treatment-cols">Add life stage and treatment cols</a></li>
  <li><a href="#delta-cq-to-normalizing-gene" id="toc-delta-cq-to-normalizing-gene" class="nav-link" data-scroll-target="#delta-cq-to-normalizing-gene">Delta Cq to Normalizing Gene</a></li>
  <li><a href="#t-tests" id="toc-t-tests" class="nav-link" data-scroll-target="#t-tests">T-tests</a></li>
  <li><a href="#delta-cq-box-plots" id="toc-delta-cq-box-plots" class="nav-link" data-scroll-target="#delta-cq-box-plots">Delta Cq Box Plots</a>
  <ul>
  <li><a href="#seed" id="toc-seed" class="nav-link" data-scroll-target="#seed">Seed</a></li>
  <li><a href="#spat" id="toc-spat" class="nav-link" data-scroll-target="#spat">Spat</a></li>
  </ul></li>
  <li><a href="#delta-delta-cq" id="toc-delta-delta-cq" class="nav-link" data-scroll-target="#delta-delta-cq">Delta delta Cq</a>
  <ul>
  <li><a href="#add-treatment-and-life-stage" id="toc-add-treatment-and-life-stage" class="nav-link" data-scroll-target="#add-treatment-and-life-stage">Add treatment and life stage</a></li>
  <li><a href="#mean-cqs-per-gene-per-treatment-per-life-stage" id="toc-mean-cqs-per-gene-per-treatment-per-life-stage" class="nav-link" data-scroll-target="#mean-cqs-per-gene-per-treatment-per-life-stage">Mean Cqs per gene per treatment per life stage</a></li>
  <li><a href="#delta-cqs" id="toc-delta-cqs" class="nav-link" data-scroll-target="#delta-cqs">Delta Cqs</a></li>
  <li><a href="#delta-delta-cq-1" id="toc-delta-delta-cq-1" class="nav-link" data-scroll-target="#delta-delta-cq-1">Delta delta Cq</a></li>
  <li><a href="#calculate-the-fold-change-for-each-target" id="toc-calculate-the-fold-change-for-each-target" class="nav-link" data-scroll-target="#calculate-the-fold-change-for-each-target">Calculate the fold change for each Target</a></li>
  <li><a href="#plot---seed-fold-change" id="toc-plot---seed-fold-change" class="nav-link" data-scroll-target="#plot---seed-fold-change">Plot - Seed Fold Change</a></li>
  <li><a href="#plot---spat-fold-change" id="toc-plot---spat-fold-change" class="nav-link" data-scroll-target="#plot---spat-fold-change">Plot - Spat Fold Change</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-right" id="quarto-document-content">





<section id="intro" class="level1">
<h1>INTRO</h1>
<p>qPCR analysis of <a href="http://en.wikipedia.org/wiki/Pacific_oyster"><em>Crassostrea gigas</em> (Pacific oyster)</a> spat and seed from the <a href="../../../../project-gigas-carryover/lifestage_carryover/"><code>project-gigas-carryover</code></a> (GitHub Repo).</p>
<p>Metadata:</p>
<ul>
<li><a href="https://github.com/RobertsLab/project-gigas-carryover/blob/82f1a4bcd5e2ce80cea9916f271bf1d54f740bd1/lifestage_carryover/data/sampling_metadata/20240314_rna_extractions.csv">20240314_rna_extractions.csv</a> (GitHub; commit: <code>82f1a4b</code>).</li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Not all of the samples listed in the sheet above were used in the qPCRs, due to lack of RNA <a href="../../../posts/2024/2024-03-19-RNA-Isolation---C.gigas-Lifestage-Carryover-Seed-and-Spat/index.html">during isolation</a> (Notebook).</p>
</div>
</div>
<p>Background info notebooks:</p>
<ul>
<li><p><a href="../../../posts/2024/2024-03-19-RNA-Isolation---C.gigas-Lifestage-Carryover-Seed-and-Spat/index.html">Isolated RNA for gigas carryover project</a> (Notebook)</p></li>
<li><p><a href="../../../posts/2024/2024-03-19-Reverse-Transcription---C.gigas-Lifestage-Carryover-Seed-and-Spat/index.html">Reverse transcribed RNA for gigas carryover project</a> (Notebook)</p></li>
<li><p><a href="../../../posts/2024/2024-03-25-qPCRs---C.gigas-Lifestage-Carryover-cDNA/index.html">qPCRs</a>.</p></li>
</ul>
</section>
<section id="analysis" class="level1">
<h1>ANALYSIS</h1>
<section id="load-libraries" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries">Load libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"tidyverse"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'tidyverse'</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="set-variables" class="level2">
<h2 class="anchored" data-anchor-id="set-variables">Set variables</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>seed.control <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"223"</span>, <span class="st">"243"</span>, <span class="st">"244"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>seed.treated <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"200"</span>, <span class="st">"257"</span>, <span class="st">"285"</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>spat.control <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"206"</span>, <span class="st">"282"</span>, <span class="st">"284"</span>, <span class="st">"289"</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>spat.treated <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"220"</span>, <span class="st">"226"</span>, <span class="st">"242"</span>, <span class="st">"253"</span>, <span class="st">"296"</span>, <span class="st">"298"</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine vectors into a list</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>vector_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">seed.control =</span> seed.control,</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">seed.treated =</span> seed.treated,</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">spat.control =</span> spat.control,</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">spat.treated =</span> spat.treated)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="functions" class="level2">
<h2 class="anchored" data-anchor-id="functions">Functions</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>calculate_delta_Cq <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Sample) <span class="sc">%&gt;%</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">delta_Cq =</span> Cq.Mean <span class="sc">-</span> Cq.Mean[Target <span class="sc">==</span> <span class="st">"GAPDH"</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>()</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-in-files" class="level2">
<h2 class="anchored" data-anchor-id="read-in-files">Read in files</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the directory where your CSV files are located</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>cqs_directory <span class="ot">&lt;-</span> <span class="st">"../2024-03-25-qPCRs---C.gigas-Lifestage-Carryover-cDNA/"</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get a list of all CSV files in the directory with the naming structure "*Cq_Results.csv"</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>cq_file_list <span class="ot">&lt;-</span> <span class="fu">list</span>() <span class="co"># Initialize list</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>cq_file_list <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> cqs_directory, <span class="at">pattern =</span> <span class="st">"Cq_Results</span><span class="sc">\\</span><span class="st">.csv$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize an empty list to store the data frames</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>data_frames <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each file and read it into a data frame, then add it to the list</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (file <span class="cf">in</span> cq_file_list) {</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(file, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  data_frames[[file]] <span class="ot">&lt;-</span> data</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine all data frames into a single data frame</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(data_frames, <span class="at">.id =</span> <span class="st">"data_frame_id"</span>)</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert Sample column to character type</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> combined_df <span class="sc">%&gt;%</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample =</span> <span class="fu">as.character</span>(Sample))</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(combined_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>'data.frame':   272 obs. of  17 variables:
 $ data_frame_id         : chr  "../2024-03-25-qPCRs---C.gigas-Lifestage-Carryover-cDNA//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" "../2024-03-25-qPCRs---C.gigas-Lifestage-Carryover-cDNA//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" "../2024-03-25-qPCRs---C.gigas-Lifestage-Carryover-cDNA//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" "../2024-03-25-qPCRs---C.gigas-Lifestage-Carryover-cDNA//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" ...
 $ X                     : logi  NA NA NA NA NA NA ...
 $ Well                  : chr  "A01" "A02" "A03" "A04" ...
 $ Fluor                 : chr  "SYBR" "SYBR" "SYBR" "SYBR" ...
 $ Target                : chr  "ATPsynthase" "ATPsynthase" "ATPsynthase" "ATPsynthase" ...
 $ Content               : chr  "Unkn-01" "Unkn-01" "Unkn-02" "Unkn-02" ...
 $ Sample                : chr  "206" "206" "220" "220" ...
 $ Biological.Set.Name   : logi  NA NA NA NA NA NA ...
 $ Cq                    : num  26.7 26.7 25.8 25.9 25.1 ...
 $ Cq.Mean               : num  26.7 26.7 25.9 25.9 25.1 ...
 $ Cq.Std..Dev           : num  0.0455 0.0455 0.0239 0.0239 0.0813 ...
 $ Starting.Quantity..SQ.: num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Log.Starting.Quantity : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Mean               : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Std..Dev           : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Set.Point             : int  60 60 60 60 60 60 60 60 60 60 ...
 $ Well.Note             : logi  NA NA NA NA NA NA ...</code></pre>
</div>
</div>
</section>
<section id="unique-samples-by-target" class="level2">
<h2 class="anchored" data-anchor-id="unique-samples-by-target">Unique samples by target</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by Sample and Target, then summarize to get unique rows for each sample</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>aggregated_df <span class="ot">&lt;-</span> combined_df <span class="sc">%&gt;%</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Sample, Target) <span class="sc">%&gt;%</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Cq.Mean =</span> <span class="fu">mean</span>(Cq.Mean, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'Sample'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(aggregated_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [136 × 3] (S3: tbl_df/tbl/data.frame)
 $ Sample : chr [1:136] "200" "200" "200" "200" ...
 $ Target : chr [1:136] "ATPsynthase" "DNMT1" "GAPDH" "HSP70" ...
 $ Cq.Mean: num [1:136] 25.2 33.6 25.4 31.8 26 ...</code></pre>
</div>
</div>
</section>
<section id="add-life-stage-and-treatment-cols" class="level2">
<h2 class="anchored" data-anchor-id="add-life-stage-and-treatment-cols">Add life stage and treatment cols</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize new columns</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>aggregated_df <span class="ot">&lt;-</span> aggregated_df <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">life.stage =</span> <span class="cn">NA_character_</span>,</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">treatment =</span> <span class="cn">NA_character_</span>)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each vector</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (vec_name <span class="cf">in</span> <span class="fu">names</span>(vector_list)) {</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  vec <span class="ot">&lt;-</span> vector_list[[vec_name]]</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  stage <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(vec_name, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  treatment <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(vec_name, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Loop through each row in aggregated_df</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(aggregated_df)) {</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>    sample <span class="ot">&lt;-</span> aggregated_df<span class="sc">$</span>Sample[i]</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if sample is in the vector</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (sample <span class="sc">%in%</span> vec) {</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Update life.stage and treatment columns</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>      aggregated_df<span class="sc">$</span>life.stage[i] <span class="ot">&lt;-</span> stage</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>      aggregated_df<span class="sc">$</span>treatment[i] <span class="ot">&lt;-</span> treatment</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(aggregated_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [136 × 5] (S3: tbl_df/tbl/data.frame)
 $ Sample    : chr [1:136] "200" "200" "200" "200" ...
 $ Target    : chr [1:136] "ATPsynthase" "DNMT1" "GAPDH" "HSP70" ...
 $ Cq.Mean   : num [1:136] 25.2 33.6 25.4 31.8 26 ...
 $ life.stage: chr [1:136] "seed" "seed" "seed" "seed" ...
 $ treatment : chr [1:136] "treated" "treated" "treated" "treated" ...</code></pre>
</div>
</div>
</section>
<section id="delta-cq-to-normalizing-gene" class="level2">
<h2 class="anchored" data-anchor-id="delta-cq-to-normalizing-gene">Delta Cq to Normalizing Gene</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta Cq by subtracting GAPDH Cq.Mean from each corresponding Sample Cq.Mean</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>delta_Cq_df <span class="ot">&lt;-</span> <span class="fu">calculate_delta_Cq</span>(aggregated_df)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_Cq_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [136 × 6] (S3: tbl_df/tbl/data.frame)
 $ Sample    : chr [1:136] "200" "200" "200" "200" ...
 $ Target    : chr [1:136] "ATPsynthase" "DNMT1" "GAPDH" "HSP70" ...
 $ Cq.Mean   : num [1:136] 25.2 33.6 25.4 31.8 26 ...
 $ life.stage: chr [1:136] "seed" "seed" "seed" "seed" ...
 $ treatment : chr [1:136] "treated" "treated" "treated" "treated" ...
 $ delta_Cq  : num [1:136] -0.243 8.168 0 6.349 0.578 ...</code></pre>
</div>
</div>
</section>
<section id="t-tests" class="level2">
<h2 class="anchored" data-anchor-id="t-tests">T-tests</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out groups with missing life.stage or Target</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Caused by NTCs</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Also removes normalizing gene(s)</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>delta_Cq_df_filtered <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(life.stage), <span class="sc">!</span><span class="fu">is.na</span>(Target), Target <span class="sc">!=</span> <span class="st">"GAPDH"</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform t-test for each Target within life.stage</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>t_test_results <span class="ot">&lt;-</span> delta_Cq_df_filtered <span class="sc">%&gt;%</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, Target) <span class="sc">%&gt;%</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">t_test_result =</span> <span class="fu">list</span>(<span class="fu">t.test</span>(delta_Cq <span class="sc">~</span> treatment))</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'life.stage'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract t-test statistics</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>t_test_results <span class="ot">&lt;-</span> t_test_results <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">estimate_diff =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>estimate[<span class="dv">1</span>] <span class="sc">-</span> x<span class="sc">$</span>estimate[<span class="dv">2</span>]),</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_value =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>p.value)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>t_test_result)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Add asterisk information to data frame</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Useful for plotting</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>t_test_results<span class="sc">$</span>asterisk <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(t_test_results<span class="sc">$</span>p_value <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="delta-cq-box-plots" class="level2">
<h2 class="anchored" data-anchor-id="delta-cq-box-plots">Delta Cq Box Plots</h2>
<section id="seed" class="level3">
<h3 class="anchored" data-anchor-id="seed">Seed</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter delta_Cq_df_filtered for seed life stage</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>seed_delta_Cq_df <span class="ot">&lt;-</span> delta_Cq_df_filtered <span class="sc">%&gt;%</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(life.stage <span class="sc">==</span> <span class="st">"seed"</span>)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the box plot</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(seed_delta_Cq_df, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> treatment)) <span class="sc">+</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Delta Cq"</span>)</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Add asterisks</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>boxplot <span class="sc">+</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> t_test_results<span class="sc">$</span>Target, <span class="at">y =</span> <span class="cn">Inf</span>, <span class="at">label =</span> t_test_results<span class="sc">$</span>asterisk,</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>           <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">color =</span> <span class="st">"orange"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/box-plots-seed-delta-Cq-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Show box plot</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(boxplot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/box-plots-seed-delta-Cq-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="spat" class="level3">
<h3 class="anchored" data-anchor-id="spat">Spat</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter data for life.stage = "spat"</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>spat_delta_Cq <span class="ot">&lt;-</span> delta_Cq_df_filtered <span class="sc">%&gt;%</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(life.stage <span class="sc">==</span> <span class="st">"spat"</span>)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the maximum delta_Cq for each Target</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>max_delta_Cq_by_target <span class="ot">&lt;-</span> spat_delta_Cq <span class="sc">%&gt;%</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">max_delta_Cq =</span> <span class="fu">max</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge t_test_results with max_delta_Cq_by_target to get the maximum delta_Cq for each Target</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>t_test_results_with_max_delta_Cq <span class="ot">&lt;-</span> <span class="fu">merge</span>(t_test_results, max_delta_Cq_by_target, <span class="at">by =</span> <span class="st">"Target"</span>)</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the box plot</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(spat_delta_Cq, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> treatment)) <span class="sc">+</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Delta Cq"</span>)</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Add asterisks</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>boxplot <span class="sc">+</span></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> t_test_results_with_max_delta_Cq<span class="sc">$</span>Target, </span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> t_test_results_with_max_delta_Cq<span class="sc">$</span>max_delta_Cq <span class="sc">+</span> <span class="fl">0.5</span>, </span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> t_test_results_with_max_delta_Cq<span class="sc">$</span>asterisk,</span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>           <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">color =</span> <span class="st">"orange"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/box-plots-delta-Cq-spat-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="delta-delta-cq" class="level2">
<h2 class="anchored" data-anchor-id="delta-delta-cq">Delta delta Cq</h2>
<section id="add-treatment-and-life-stage" class="level3">
<h3 class="anchored" data-anchor-id="add-treatment-and-life-stage">Add treatment and life stage</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize empty vectors to store life.stage and treatment</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>life_stage <span class="ot">&lt;-</span> <span class="fu">character</span>(<span class="fu">nrow</span>(combined_df))</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>treatment <span class="ot">&lt;-</span> <span class="fu">character</span>(<span class="fu">nrow</span>(combined_df))</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each row of combined_df</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(combined_df)) {</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  sample_id <span class="ot">&lt;-</span> combined_df<span class="sc">$</span>Sample[i]</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if the sample_id is present in any of the vectors</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  found <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (vec_name <span class="cf">in</span> <span class="fu">names</span>(vector_list)) {</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (sample_id <span class="sc">%in%</span> vector_list[[vec_name]]) {</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>      <span class="co"># If present, extract life.stage and treatment from the vector name</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>      parts <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(vec_name, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>)[[<span class="dv">1</span>]]</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>      life_stage[i] <span class="ot">&lt;-</span> parts[<span class="dv">1</span>]</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>      treatment[i] <span class="ot">&lt;-</span> parts[<span class="dv">2</span>]</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>      found <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span>  <span class="co"># Exit loop once found</span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># If sample_id is not found in any vector, assign NA to both life.stage and treatment</span></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>found) {</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>    life_stage[i] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>    treatment[i] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Add life.stage and treatment columns to combined_df</span></span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> combined_df <span class="sc">%&gt;%</span></span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">life.stage =</span> life_stage,</span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">treatment =</span> treatment)</span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out rows where life.stage is NA</span></span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a>combined_df_filtered <span class="ot">&lt;-</span> combined_df <span class="sc">%&gt;%</span> </span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(life.stage))</span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(combined_df_filtered)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>'data.frame':   256 obs. of  19 variables:
 $ data_frame_id         : chr  "../2024-03-25-qPCRs---C.gigas-Lifestage-Carryover-cDNA//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" "../2024-03-25-qPCRs---C.gigas-Lifestage-Carryover-cDNA//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" "../2024-03-25-qPCRs---C.gigas-Lifestage-Carryover-cDNA//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" "../2024-03-25-qPCRs---C.gigas-Lifestage-Carryover-cDNA//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" ...
 $ X                     : logi  NA NA NA NA NA NA ...
 $ Well                  : chr  "A01" "A02" "A03" "A04" ...
 $ Fluor                 : chr  "SYBR" "SYBR" "SYBR" "SYBR" ...
 $ Target                : chr  "ATPsynthase" "ATPsynthase" "ATPsynthase" "ATPsynthase" ...
 $ Content               : chr  "Unkn-01" "Unkn-01" "Unkn-02" "Unkn-02" ...
 $ Sample                : chr  "206" "206" "220" "220" ...
 $ Biological.Set.Name   : logi  NA NA NA NA NA NA ...
 $ Cq                    : num  26.7 26.7 25.8 25.9 25.1 ...
 $ Cq.Mean               : num  26.7 26.7 25.9 25.9 25.1 ...
 $ Cq.Std..Dev           : num  0.0455 0.0455 0.0239 0.0239 0.0813 ...
 $ Starting.Quantity..SQ.: num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Log.Starting.Quantity : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Mean               : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Std..Dev           : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Set.Point             : int  60 60 60 60 60 60 60 60 60 60 ...
 $ Well.Note             : logi  NA NA NA NA NA NA ...
 $ life.stage            : chr  "spat" "spat" "spat" "spat" ...
 $ treatment             : chr  "control" "control" "treated" "treated" ...</code></pre>
</div>
</div>
</section>
<section id="mean-cqs-per-gene-per-treatment-per-life-stage" class="level3">
<h3 class="anchored" data-anchor-id="mean-cqs-per-gene-per-treatment-per-life-stage">Mean Cqs per gene per treatment per life stage</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by life.stage, treatment, and Target, then calculate the mean Cq</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>mean_Cq_df <span class="ot">&lt;-</span> combined_df_filtered <span class="sc">%&gt;%</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, treatment, Target) <span class="sc">%&gt;%</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean_Cq =</span> <span class="fu">mean</span>(Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'life.stage', 'treatment'. You can override
using the `.groups` argument.</code></pre>
</div>
</div>
</section>
<section id="delta-cqs" class="level3">
<h3 class="anchored" data-anchor-id="delta-cqs">Delta Cqs</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta Cq</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>combined_df_with_delta_Cq <span class="ot">&lt;-</span> mean_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, treatment) <span class="sc">%&gt;%</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">delta_Cq =</span> mean_Cq <span class="sc">-</span> <span class="fu">mean</span>(mean_Cq[Target <span class="sc">==</span> <span class="st">"GAPDH"</span>])) <span class="sc">%&gt;%</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">!=</span> <span class="st">"GAPDH"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="delta-delta-cq-1" class="level3">
<h3 class="anchored" data-anchor-id="delta-delta-cq-1">Delta delta Cq</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta_delta_Cq</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>delta_delta_Cq_df <span class="ot">&lt;-</span> combined_df_with_delta_Cq <span class="sc">%&gt;%</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, Target) <span class="sc">%&gt;%</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">delta_delta_Cq =</span> delta_Cq[treatment <span class="sc">==</span> <span class="st">"treated"</span>] <span class="sc">-</span> delta_Cq[treatment <span class="sc">==</span> <span class="st">"control"</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'life.stage'. You can override using the
`.groups` argument.</code></pre>
</div>
</div>
</section>
<section id="calculate-the-fold-change-for-each-target" class="level3">
<h3 class="anchored" data-anchor-id="calculate-the-fold-change-for-each-target">Calculate the fold change for each Target</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>delta_delta_fold_change <span class="ot">&lt;-</span> delta_delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Target, fold_change)</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_fold_change)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>gropd_df [14 × 3] (S3: grouped_df/tbl_df/tbl/data.frame)
 $ life.stage : chr [1:14] "seed" "seed" "seed" "seed" ...
 $ Target     : chr [1:14] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
 $ fold_change: num [1:14] 1.532 1.146 1.8 0.662 1.365 ...
 - attr(*, "groups")= tibble [2 × 2] (S3: tbl_df/tbl/data.frame)
  ..$ life.stage: chr [1:2] "seed" "spat"
  ..$ .rows     : list&lt;int&gt; [1:2] 
  .. ..$ : int [1:7] 1 2 3 4 5 6 7
  .. ..$ : int [1:7] 8 9 10 11 12 13 14
  .. ..@ ptype: int(0) 
  ..- attr(*, ".drop")= logi TRUE</code></pre>
</div>
</div>
</section>
<section id="plot---seed-fold-change" class="level3">
<h3 class="anchored" data-anchor-id="plot---seed-fold-change">Plot - Seed Fold Change</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter delta_delta_fold_change for seed life stage</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>seed_df <span class="ot">&lt;-</span> delta_delta_fold_change <span class="sc">%&gt;%</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(life.stage <span class="sc">==</span> <span class="st">"seed"</span>)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create bar plot for seed life stage</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the plot with fold changes relative to baseline of 1</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>seed_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(seed_df, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> fold_change <span class="sc">-</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"skyblue"</span>) <span class="sc">+</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span>  <span class="co"># Baseline</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Fold Change"</span>, <span class="at">title =</span> <span class="st">"Seed Life Stage Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.5</span>, <span class="dv">1</span>))</span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Display plot</span></span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>seed_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/plot-seed-fold-change-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="plot---spat-fold-change" class="level3">
<h3 class="anchored" data-anchor-id="plot---spat-fold-change">Plot - Spat Fold Change</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter delta_delta_fold_change for spat life stage</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>spat_df <span class="ot">&lt;-</span> delta_delta_fold_change <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(life.stage <span class="sc">==</span> <span class="st">"spat"</span>)</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create bar plot for spat life stage</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the plot with fold changes relative to baseline of 1</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>spat_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(spat_df, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> fold_change <span class="sc">-</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"salmon"</span>) <span class="sc">+</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span>  <span class="co"># Baseline</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Fold Change"</span>, <span class="at">title =</span> <span class="st">"Spat Life Stage Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">8</span>))</span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Display plot</span></span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>spat_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/plot-spat-fold-change-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>


<!-- -->

</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/robertslab\.github\.io\/sams-notebook");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb33" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> Sam White</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="an">toc-title:</span><span class="co"> Contents</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="an">toc-depth:</span><span class="co"> 5</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="an">toc-location:</span><span class="co"> left</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> qPCR Analysis - C.gigas Lifestages Carryover from 20240325</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> '2024-03-27'</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a><span class="an">draft:</span><span class="co"> false</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a><span class="an">code-fold:</span><span class="co"> false</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a><span class="an">code-tools:</span><span class="co"> true</span></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a><span class="an">engine:</span><span class="co"> knitr</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a><span class="an">categories:</span><span class="co"> </span></span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a><span class="co">  - 2024</span></span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a><span class="co">  - qPCR</span></span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a><span class="co">  - Crassotrea gigas</span></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a><span class="co">  - Pacific oyster</span></span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a><span class="fu"># INTRO</span></span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a>qPCR analysis of <span class="co">[</span><span class="ot">_Crassostrea gigas_ (Pacific oyster)</span><span class="co">](http://en.wikipedia.org/wiki/Pacific_oyster)</span> spat and seed from the <span class="co">[</span><span class="ot">`project-gigas-carryover`</span><span class="co">](../../../../project-gigas-carryover/lifestage_carryover/)</span> (GitHub Repo). </span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a>Metadata:</span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">20240314_rna_extractions.csv</span><span class="co">](https://github.com/RobertsLab/project-gigas-carryover/blob/82f1a4bcd5e2ce80cea9916f271bf1d54f740bd1/lifestage_carryover/data/sampling_metadata/20240314_rna_extractions.csv)</span> (GitHub; commit: <span class="in">`82f1a4b`</span>).</span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>::: {.callout-note}</span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a>Not all of the samples listed in the sheet above were used in the qPCRs, due to lack of RNA <span class="co">[</span><span class="ot">during isolation</span><span class="co">](../2024-03-19-RNA-Isolation---C.gigas-Lifestage-Carryover-Seed-and-Spat/index.qmd)</span> (Notebook).</span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a>Background info notebooks:</span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-32"><a href="#cb33-32" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">Isolated RNA for gigas carryover project</span><span class="co">](../2024-03-19-RNA-Isolation---C.gigas-Lifestage-Carryover-Seed-and-Spat/index.qmd)</span> (Notebook)</span>
<span id="cb33-33"><a href="#cb33-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-34"><a href="#cb33-34" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">Reverse transcribed RNA for gigas carryover project</span><span class="co">](../2024-03-19-Reverse-Transcription---C.gigas-Lifestage-Carryover-Seed-and-Spat/index.qmd)</span> (Notebook)</span>
<span id="cb33-35"><a href="#cb33-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-36"><a href="#cb33-36" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">qPCRs</span><span class="co">](../2024-03-25-qPCRs---C.gigas-Lifestage-Carryover-cDNA/index.qmd)</span>.</span>
<span id="cb33-37"><a href="#cb33-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-38"><a href="#cb33-38" aria-hidden="true" tabindex="-1"></a>::: {.callout-note}</span>
<span id="cb33-39"><a href="#cb33-39" aria-hidden="true" tabindex="-1"></a>Code below was rendered from <span class="co">[</span><span class="ot">20240327-cgig-lifestage-carryover-qpcr-analysis.Rmd</span><span class="co">](https://github.com/RobertsLab/code/blob/f9af1eb3de350e165e65784bbabede2578b71ba2/r_projects/sam/20240327-cgig-lifestage-carryover-qpcr-analysis/code/20240327-cgig-lifestage-carryover-qpcr-analysis.Rmd)</span>, commit <span class="in">`f9af1eb`</span>.</span>
<span id="cb33-40"><a href="#cb33-40" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb33-41"><a href="#cb33-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-42"><a href="#cb33-42" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>&lt;a href="#1-analysis" id="toc-1-analysis"&gt;1 ANALYSIS&lt;/a&gt;</span>
<span id="cb33-43"><a href="#cb33-43" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>&lt;a href="#11-load-libraries" id="toc-11-load-libraries"&gt;1.1 Load</span>
<span id="cb33-44"><a href="#cb33-44" aria-hidden="true" tabindex="-1"></a>    libraries&lt;/a&gt;</span>
<span id="cb33-45"><a href="#cb33-45" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>&lt;a href="#12-set-variables" id="toc-12-set-variables"&gt;1.2 Set</span>
<span id="cb33-46"><a href="#cb33-46" aria-hidden="true" tabindex="-1"></a>    variables&lt;/a&gt;</span>
<span id="cb33-47"><a href="#cb33-47" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>&lt;a href="#13-functions" id="toc-13-functions"&gt;1.3 Functions&lt;/a&gt;</span>
<span id="cb33-48"><a href="#cb33-48" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>&lt;a href="#14-read-in-files" id="toc-14-read-in-files"&gt;1.4 Read in</span>
<span id="cb33-49"><a href="#cb33-49" aria-hidden="true" tabindex="-1"></a>    files&lt;/a&gt;</span>
<span id="cb33-50"><a href="#cb33-50" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>&lt;a href="#15-unique-samples-by-target"</span>
<span id="cb33-51"><a href="#cb33-51" aria-hidden="true" tabindex="-1"></a>    id="toc-15-unique-samples-by-target"&gt;1.5 Unique samples by target&lt;/a&gt;</span>
<span id="cb33-52"><a href="#cb33-52" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>&lt;a href="#16-add-life-stage-and-treatment-cols"</span>
<span id="cb33-53"><a href="#cb33-53" aria-hidden="true" tabindex="-1"></a>    id="toc-16-add-life-stage-and-treatment-cols"&gt;1.6 Add life stage and</span>
<span id="cb33-54"><a href="#cb33-54" aria-hidden="true" tabindex="-1"></a>    treatment cols&lt;/a&gt;</span>
<span id="cb33-55"><a href="#cb33-55" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>&lt;a href="#17-delta-cq-to-normalizing-gene"</span>
<span id="cb33-56"><a href="#cb33-56" aria-hidden="true" tabindex="-1"></a>    id="toc-17-delta-cq-to-normalizing-gene"&gt;1.7 Delta Cq to Normalizing</span>
<span id="cb33-57"><a href="#cb33-57" aria-hidden="true" tabindex="-1"></a>    Gene&lt;/a&gt;</span>
<span id="cb33-58"><a href="#cb33-58" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>&lt;a href="#18-t-tests" id="toc-18-t-tests"&gt;1.8 T-tests&lt;/a&gt;</span>
<span id="cb33-59"><a href="#cb33-59" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>&lt;a href="#19-delta-cq-box-plots" id="toc-19-delta-cq-box-plots"&gt;1.9</span>
<span id="cb33-60"><a href="#cb33-60" aria-hidden="true" tabindex="-1"></a>    Delta Cq Box Plots&lt;/a&gt;</span>
<span id="cb33-61"><a href="#cb33-61" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>&lt;a href="#191-seed" id="toc-191-seed"&gt;1.9.1 Seed&lt;/a&gt;</span>
<span id="cb33-62"><a href="#cb33-62" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>&lt;a href="#192-spat" id="toc-192-spat"&gt;1.9.2 Spat&lt;/a&gt;</span>
<span id="cb33-63"><a href="#cb33-63" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>&lt;a href="#110-delta-delta-cq" id="toc-110-delta-delta-cq"&gt;1.10 Delta</span>
<span id="cb33-64"><a href="#cb33-64" aria-hidden="true" tabindex="-1"></a>    delta Cq&lt;/a&gt;</span>
<span id="cb33-65"><a href="#cb33-65" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>&lt;a href="#1101-add-treatment-and-life-stage"</span>
<span id="cb33-66"><a href="#cb33-66" aria-hidden="true" tabindex="-1"></a>      id="toc-1101-add-treatment-and-life-stage"&gt;1.10.1 Add treatment and life</span>
<span id="cb33-67"><a href="#cb33-67" aria-hidden="true" tabindex="-1"></a>      stage&lt;/a&gt;</span>
<span id="cb33-68"><a href="#cb33-68" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>&lt;a href="#1102-mean-cqs-per-gene-per-treatment-per-life-stage"</span>
<span id="cb33-69"><a href="#cb33-69" aria-hidden="true" tabindex="-1"></a>      id="toc-1102-mean-cqs-per-gene-per-treatment-per-life-stage"&gt;1.10.2 Mean</span>
<span id="cb33-70"><a href="#cb33-70" aria-hidden="true" tabindex="-1"></a>      Cqs per gene per treatment per life stage&lt;/a&gt;</span>
<span id="cb33-71"><a href="#cb33-71" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>&lt;a href="#1103-delta-cqs" id="toc-1103-delta-cqs"&gt;1.10.3 Delta Cqs&lt;/a&gt;</span>
<span id="cb33-72"><a href="#cb33-72" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>&lt;a href="#1104-delta-delta-cq" id="toc-1104-delta-delta-cq"&gt;1.10.4 Delta</span>
<span id="cb33-73"><a href="#cb33-73" aria-hidden="true" tabindex="-1"></a>      delta Cq&lt;/a&gt;</span>
<span id="cb33-74"><a href="#cb33-74" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>&lt;a href="#1105-calculate-the-fold-change-for-each-target"</span>
<span id="cb33-75"><a href="#cb33-75" aria-hidden="true" tabindex="-1"></a>      id="toc-1105-calculate-the-fold-change-for-each-target"&gt;1.10.5 Calculate</span>
<span id="cb33-76"><a href="#cb33-76" aria-hidden="true" tabindex="-1"></a>      the fold change for each Target&lt;/a&gt;</span>
<span id="cb33-77"><a href="#cb33-77" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>&lt;a href="#1106-plot---seed-fold-change"</span>
<span id="cb33-78"><a href="#cb33-78" aria-hidden="true" tabindex="-1"></a>      id="toc-1106-plot---seed-fold-change"&gt;1.10.6 Plot - Seed Fold Change&lt;/a&gt;</span>
<span id="cb33-79"><a href="#cb33-79" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>&lt;a href="#1107-plot---spat-fold-change"</span>
<span id="cb33-80"><a href="#cb33-80" aria-hidden="true" tabindex="-1"></a>      id="toc-1107-plot---spat-fold-change"&gt;1.10.7 Plot - Spat Fold Change&lt;/a&gt;</span>
<span id="cb33-81"><a href="#cb33-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-82"><a href="#cb33-82" aria-hidden="true" tabindex="-1"></a><span class="fu"># 1 ANALYSIS</span></span>
<span id="cb33-83"><a href="#cb33-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-84"><a href="#cb33-84" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.1 Load libraries</span></span>
<span id="cb33-85"><a href="#cb33-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-86"><a href="#cb33-86" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-87"><a href="#cb33-87" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"tidyverse"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'tidyverse'</span>)</span>
<span id="cb33-88"><a href="#cb33-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-89"><a href="#cb33-89" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb33-90"><a href="#cb33-90" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb33-91"><a href="#cb33-91" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-92"><a href="#cb33-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-93"><a href="#cb33-93" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.2 Set variables</span></span>
<span id="cb33-94"><a href="#cb33-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-95"><a href="#cb33-95" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-96"><a href="#cb33-96" aria-hidden="true" tabindex="-1"></a>seed.control <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"223"</span>, <span class="st">"243"</span>, <span class="st">"244"</span>)</span>
<span id="cb33-97"><a href="#cb33-97" aria-hidden="true" tabindex="-1"></a>seed.treated <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"200"</span>, <span class="st">"257"</span>, <span class="st">"285"</span>)</span>
<span id="cb33-98"><a href="#cb33-98" aria-hidden="true" tabindex="-1"></a>spat.control <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"206"</span>, <span class="st">"282"</span>, <span class="st">"284"</span>, <span class="st">"289"</span>)</span>
<span id="cb33-99"><a href="#cb33-99" aria-hidden="true" tabindex="-1"></a>spat.treated <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"220"</span>, <span class="st">"226"</span>, <span class="st">"242"</span>, <span class="st">"253"</span>, <span class="st">"296"</span>, <span class="st">"298"</span>)</span>
<span id="cb33-100"><a href="#cb33-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-101"><a href="#cb33-101" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine vectors into a list</span></span>
<span id="cb33-102"><a href="#cb33-102" aria-hidden="true" tabindex="-1"></a>vector_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">seed.control =</span> seed.control,</span>
<span id="cb33-103"><a href="#cb33-103" aria-hidden="true" tabindex="-1"></a>                    <span class="at">seed.treated =</span> seed.treated,</span>
<span id="cb33-104"><a href="#cb33-104" aria-hidden="true" tabindex="-1"></a>                    <span class="at">spat.control =</span> spat.control,</span>
<span id="cb33-105"><a href="#cb33-105" aria-hidden="true" tabindex="-1"></a>                    <span class="at">spat.treated =</span> spat.treated)</span>
<span id="cb33-106"><a href="#cb33-106" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-107"><a href="#cb33-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-108"><a href="#cb33-108" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.3 Functions</span></span>
<span id="cb33-109"><a href="#cb33-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-110"><a href="#cb33-110" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-111"><a href="#cb33-111" aria-hidden="true" tabindex="-1"></a>calculate_delta_Cq <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb33-112"><a href="#cb33-112" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb33-113"><a href="#cb33-113" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Sample) <span class="sc">%&gt;%</span></span>
<span id="cb33-114"><a href="#cb33-114" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">delta_Cq =</span> Cq.Mean <span class="sc">-</span> Cq.Mean[Target <span class="sc">==</span> <span class="st">"GAPDH"</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb33-115"><a href="#cb33-115" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>()</span>
<span id="cb33-116"><a href="#cb33-116" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb33-117"><a href="#cb33-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb33-118"><a href="#cb33-118" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-119"><a href="#cb33-119" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-120"><a href="#cb33-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-121"><a href="#cb33-121" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.4 Read in files</span></span>
<span id="cb33-122"><a href="#cb33-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-123"><a href="#cb33-123" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-124"><a href="#cb33-124" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the directory where your CSV files are located</span></span>
<span id="cb33-125"><a href="#cb33-125" aria-hidden="true" tabindex="-1"></a>cqs_directory <span class="ot">&lt;-</span> <span class="st">"../data/"</span></span>
<span id="cb33-126"><a href="#cb33-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-127"><a href="#cb33-127" aria-hidden="true" tabindex="-1"></a><span class="co"># Get a list of all CSV files in the directory with the naming structure "*Cq_Results.csv"</span></span>
<span id="cb33-128"><a href="#cb33-128" aria-hidden="true" tabindex="-1"></a>cq_file_list <span class="ot">&lt;-</span> <span class="fu">list</span>() <span class="co"># Initialize list</span></span>
<span id="cb33-129"><a href="#cb33-129" aria-hidden="true" tabindex="-1"></a>cq_file_list <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> cqs_directory, <span class="at">pattern =</span> <span class="st">"Cq_Results</span><span class="sc">\\</span><span class="st">.csv$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-130"><a href="#cb33-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-131"><a href="#cb33-131" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize an empty list to store the data frames</span></span>
<span id="cb33-132"><a href="#cb33-132" aria-hidden="true" tabindex="-1"></a>data_frames <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb33-133"><a href="#cb33-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-134"><a href="#cb33-134" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each file and read it into a data frame, then add it to the list</span></span>
<span id="cb33-135"><a href="#cb33-135" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (file <span class="cf">in</span> cq_file_list) {</span>
<span id="cb33-136"><a href="#cb33-136" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(file, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-137"><a href="#cb33-137" aria-hidden="true" tabindex="-1"></a>  data_frames[[file]] <span class="ot">&lt;-</span> data</span>
<span id="cb33-138"><a href="#cb33-138" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-139"><a href="#cb33-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-140"><a href="#cb33-140" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine all data frames into a single data frame</span></span>
<span id="cb33-141"><a href="#cb33-141" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(data_frames, <span class="at">.id =</span> <span class="st">"data_frame_id"</span>)</span>
<span id="cb33-142"><a href="#cb33-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-143"><a href="#cb33-143" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert Sample column to character type</span></span>
<span id="cb33-144"><a href="#cb33-144" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> combined_df <span class="sc">%&gt;%</span></span>
<span id="cb33-145"><a href="#cb33-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample =</span> <span class="fu">as.character</span>(Sample))</span>
<span id="cb33-146"><a href="#cb33-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-147"><a href="#cb33-147" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(combined_df)</span>
<span id="cb33-148"><a href="#cb33-148" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-149"><a href="#cb33-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-150"><a href="#cb33-150" aria-hidden="true" tabindex="-1"></a><span class="in">    'data.frame':   272 obs. of  17 variables:</span></span>
<span id="cb33-151"><a href="#cb33-151" aria-hidden="true" tabindex="-1"></a><span class="in">     $ data_frame_id         : chr  "../data//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" "../data//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" "../data//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" "../data//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" ...</span></span>
<span id="cb33-152"><a href="#cb33-152" aria-hidden="true" tabindex="-1"></a><span class="in">     $ X                     : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb33-153"><a href="#cb33-153" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Well                  : chr  "A01" "A02" "A03" "A04" ...</span></span>
<span id="cb33-154"><a href="#cb33-154" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Fluor                 : chr  "SYBR" "SYBR" "SYBR" "SYBR" ...</span></span>
<span id="cb33-155"><a href="#cb33-155" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Target                : chr  "ATPsynthase" "ATPsynthase" "ATPsynthase" "ATPsynthase" ...</span></span>
<span id="cb33-156"><a href="#cb33-156" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Content               : chr  "Unkn-01" "Unkn-01" "Unkn-02" "Unkn-02" ...</span></span>
<span id="cb33-157"><a href="#cb33-157" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Sample                : chr  "206" "206" "220" "220" ...</span></span>
<span id="cb33-158"><a href="#cb33-158" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Biological.Set.Name   : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb33-159"><a href="#cb33-159" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Cq                    : num  26.7 26.7 25.8 25.9 25.1 ...</span></span>
<span id="cb33-160"><a href="#cb33-160" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Cq.Mean               : num  26.7 26.7 25.9 25.9 25.1 ...</span></span>
<span id="cb33-161"><a href="#cb33-161" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Cq.Std..Dev           : num  0.0455 0.0455 0.0239 0.0239 0.0813 ...</span></span>
<span id="cb33-162"><a href="#cb33-162" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Starting.Quantity..SQ.: num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...</span></span>
<span id="cb33-163"><a href="#cb33-163" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Log.Starting.Quantity : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...</span></span>
<span id="cb33-164"><a href="#cb33-164" aria-hidden="true" tabindex="-1"></a><span class="in">     $ SQ.Mean               : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...</span></span>
<span id="cb33-165"><a href="#cb33-165" aria-hidden="true" tabindex="-1"></a><span class="in">     $ SQ.Std..Dev           : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...</span></span>
<span id="cb33-166"><a href="#cb33-166" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Set.Point             : int  60 60 60 60 60 60 60 60 60 60 ...</span></span>
<span id="cb33-167"><a href="#cb33-167" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Well.Note             : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb33-168"><a href="#cb33-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-169"><a href="#cb33-169" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.5 Unique samples by target</span></span>
<span id="cb33-170"><a href="#cb33-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-171"><a href="#cb33-171" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-172"><a href="#cb33-172" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by Sample and Target, then summarize to get unique rows for each sample</span></span>
<span id="cb33-173"><a href="#cb33-173" aria-hidden="true" tabindex="-1"></a>aggregated_df <span class="ot">&lt;-</span> combined_df <span class="sc">%&gt;%</span></span>
<span id="cb33-174"><a href="#cb33-174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Sample, Target) <span class="sc">%&gt;%</span></span>
<span id="cb33-175"><a href="#cb33-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Cq.Mean =</span> <span class="fu">mean</span>(Cq.Mean, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb33-176"><a href="#cb33-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb33-177"><a href="#cb33-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-178"><a href="#cb33-178" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(aggregated_df)</span>
<span id="cb33-179"><a href="#cb33-179" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-180"><a href="#cb33-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-181"><a href="#cb33-181" aria-hidden="true" tabindex="-1"></a><span class="in">    tibble [136 × 3] (S3: tbl_df/tbl/data.frame)</span></span>
<span id="cb33-182"><a href="#cb33-182" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Sample : chr [1:136] "200" "200" "200" "200" ...</span></span>
<span id="cb33-183"><a href="#cb33-183" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Target : chr [1:136] "ATPsynthase" "DNMT1" "GAPDH" "HSP70" ...</span></span>
<span id="cb33-184"><a href="#cb33-184" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Cq.Mean: num [1:136] 25.2 33.6 25.4 31.8 26 ...</span></span>
<span id="cb33-185"><a href="#cb33-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-186"><a href="#cb33-186" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.6 Add life stage and treatment cols</span></span>
<span id="cb33-187"><a href="#cb33-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-188"><a href="#cb33-188" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-189"><a href="#cb33-189" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize new columns</span></span>
<span id="cb33-190"><a href="#cb33-190" aria-hidden="true" tabindex="-1"></a>aggregated_df <span class="ot">&lt;-</span> aggregated_df <span class="sc">%&gt;%</span></span>
<span id="cb33-191"><a href="#cb33-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">life.stage =</span> <span class="cn">NA_character_</span>,</span>
<span id="cb33-192"><a href="#cb33-192" aria-hidden="true" tabindex="-1"></a>         <span class="at">treatment =</span> <span class="cn">NA_character_</span>)</span>
<span id="cb33-193"><a href="#cb33-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-194"><a href="#cb33-194" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each vector</span></span>
<span id="cb33-195"><a href="#cb33-195" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (vec_name <span class="cf">in</span> <span class="fu">names</span>(vector_list)) {</span>
<span id="cb33-196"><a href="#cb33-196" aria-hidden="true" tabindex="-1"></a>  vec <span class="ot">&lt;-</span> vector_list[[vec_name]]</span>
<span id="cb33-197"><a href="#cb33-197" aria-hidden="true" tabindex="-1"></a>  stage <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(vec_name, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span>
<span id="cb33-198"><a href="#cb33-198" aria-hidden="true" tabindex="-1"></a>  treatment <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(vec_name, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]</span>
<span id="cb33-199"><a href="#cb33-199" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb33-200"><a href="#cb33-200" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Loop through each row in aggregated_df</span></span>
<span id="cb33-201"><a href="#cb33-201" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(aggregated_df)) {</span>
<span id="cb33-202"><a href="#cb33-202" aria-hidden="true" tabindex="-1"></a>    sample <span class="ot">&lt;-</span> aggregated_df<span class="sc">$</span>Sample[i]</span>
<span id="cb33-203"><a href="#cb33-203" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb33-204"><a href="#cb33-204" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if sample is in the vector</span></span>
<span id="cb33-205"><a href="#cb33-205" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (sample <span class="sc">%in%</span> vec) {</span>
<span id="cb33-206"><a href="#cb33-206" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Update life.stage and treatment columns</span></span>
<span id="cb33-207"><a href="#cb33-207" aria-hidden="true" tabindex="-1"></a>      aggregated_df<span class="sc">$</span>life.stage[i] <span class="ot">&lt;-</span> stage</span>
<span id="cb33-208"><a href="#cb33-208" aria-hidden="true" tabindex="-1"></a>      aggregated_df<span class="sc">$</span>treatment[i] <span class="ot">&lt;-</span> treatment</span>
<span id="cb33-209"><a href="#cb33-209" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb33-210"><a href="#cb33-210" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb33-211"><a href="#cb33-211" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-212"><a href="#cb33-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-213"><a href="#cb33-213" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(aggregated_df)</span>
<span id="cb33-214"><a href="#cb33-214" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-215"><a href="#cb33-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-216"><a href="#cb33-216" aria-hidden="true" tabindex="-1"></a><span class="in">    tibble [136 × 5] (S3: tbl_df/tbl/data.frame)</span></span>
<span id="cb33-217"><a href="#cb33-217" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Sample    : chr [1:136] "200" "200" "200" "200" ...</span></span>
<span id="cb33-218"><a href="#cb33-218" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Target    : chr [1:136] "ATPsynthase" "DNMT1" "GAPDH" "HSP70" ...</span></span>
<span id="cb33-219"><a href="#cb33-219" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Cq.Mean   : num [1:136] 25.2 33.6 25.4 31.8 26 ...</span></span>
<span id="cb33-220"><a href="#cb33-220" aria-hidden="true" tabindex="-1"></a><span class="in">     $ life.stage: chr [1:136] "seed" "seed" "seed" "seed" ...</span></span>
<span id="cb33-221"><a href="#cb33-221" aria-hidden="true" tabindex="-1"></a><span class="in">     $ treatment : chr [1:136] "treated" "treated" "treated" "treated" ...</span></span>
<span id="cb33-222"><a href="#cb33-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-223"><a href="#cb33-223" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.7 Delta Cq to Normalizing Gene</span></span>
<span id="cb33-224"><a href="#cb33-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-225"><a href="#cb33-225" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-226"><a href="#cb33-226" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta Cq by subtracting GAPDH Cq.Mean from each corresponding Sample Cq.Mean</span></span>
<span id="cb33-227"><a href="#cb33-227" aria-hidden="true" tabindex="-1"></a>delta_Cq_df <span class="ot">&lt;-</span> <span class="fu">calculate_delta_Cq</span>(aggregated_df)</span>
<span id="cb33-228"><a href="#cb33-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-229"><a href="#cb33-229" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_Cq_df)</span>
<span id="cb33-230"><a href="#cb33-230" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-231"><a href="#cb33-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-232"><a href="#cb33-232" aria-hidden="true" tabindex="-1"></a><span class="in">    tibble [136 × 6] (S3: tbl_df/tbl/data.frame)</span></span>
<span id="cb33-233"><a href="#cb33-233" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Sample    : chr [1:136] "200" "200" "200" "200" ...</span></span>
<span id="cb33-234"><a href="#cb33-234" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Target    : chr [1:136] "ATPsynthase" "DNMT1" "GAPDH" "HSP70" ...</span></span>
<span id="cb33-235"><a href="#cb33-235" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Cq.Mean   : num [1:136] 25.2 33.6 25.4 31.8 26 ...</span></span>
<span id="cb33-236"><a href="#cb33-236" aria-hidden="true" tabindex="-1"></a><span class="in">     $ life.stage: chr [1:136] "seed" "seed" "seed" "seed" ...</span></span>
<span id="cb33-237"><a href="#cb33-237" aria-hidden="true" tabindex="-1"></a><span class="in">     $ treatment : chr [1:136] "treated" "treated" "treated" "treated" ...</span></span>
<span id="cb33-238"><a href="#cb33-238" aria-hidden="true" tabindex="-1"></a><span class="in">     $ delta_Cq  : num [1:136] -0.243 8.168 0 6.349 0.578 ...</span></span>
<span id="cb33-239"><a href="#cb33-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-240"><a href="#cb33-240" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.8 T-tests</span></span>
<span id="cb33-241"><a href="#cb33-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-242"><a href="#cb33-242" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-243"><a href="#cb33-243" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out groups with missing life.stage or Target</span></span>
<span id="cb33-244"><a href="#cb33-244" aria-hidden="true" tabindex="-1"></a><span class="co"># Caused by NTCs</span></span>
<span id="cb33-245"><a href="#cb33-245" aria-hidden="true" tabindex="-1"></a><span class="co"># Also removes normalizing gene(s)</span></span>
<span id="cb33-246"><a href="#cb33-246" aria-hidden="true" tabindex="-1"></a>delta_Cq_df_filtered <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb33-247"><a href="#cb33-247" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(life.stage), <span class="sc">!</span><span class="fu">is.na</span>(Target), Target <span class="sc">!=</span> <span class="st">"GAPDH"</span>)</span>
<span id="cb33-248"><a href="#cb33-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-249"><a href="#cb33-249" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform t-test for each Target within life.stage</span></span>
<span id="cb33-250"><a href="#cb33-250" aria-hidden="true" tabindex="-1"></a>t_test_results <span class="ot">&lt;-</span> delta_Cq_df_filtered <span class="sc">%&gt;%</span></span>
<span id="cb33-251"><a href="#cb33-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, Target) <span class="sc">%&gt;%</span></span>
<span id="cb33-252"><a href="#cb33-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb33-253"><a href="#cb33-253" aria-hidden="true" tabindex="-1"></a>    <span class="at">t_test_result =</span> <span class="fu">list</span>(<span class="fu">t.test</span>(delta_Cq <span class="sc">~</span> treatment))</span>
<span id="cb33-254"><a href="#cb33-254" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb33-255"><a href="#cb33-255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb33-256"><a href="#cb33-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-257"><a href="#cb33-257" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract t-test statistics</span></span>
<span id="cb33-258"><a href="#cb33-258" aria-hidden="true" tabindex="-1"></a>t_test_results <span class="ot">&lt;-</span> t_test_results <span class="sc">%&gt;%</span></span>
<span id="cb33-259"><a href="#cb33-259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb33-260"><a href="#cb33-260" aria-hidden="true" tabindex="-1"></a>    <span class="at">estimate_diff =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>estimate[<span class="dv">1</span>] <span class="sc">-</span> x<span class="sc">$</span>estimate[<span class="dv">2</span>]),</span>
<span id="cb33-261"><a href="#cb33-261" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_value =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>p.value)</span>
<span id="cb33-262"><a href="#cb33-262" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb33-263"><a href="#cb33-263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>t_test_result)</span>
<span id="cb33-264"><a href="#cb33-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-265"><a href="#cb33-265" aria-hidden="true" tabindex="-1"></a><span class="co"># Add asterisk information to data frame</span></span>
<span id="cb33-266"><a href="#cb33-266" aria-hidden="true" tabindex="-1"></a><span class="co"># Useful for plotting</span></span>
<span id="cb33-267"><a href="#cb33-267" aria-hidden="true" tabindex="-1"></a>t_test_results<span class="sc">$</span>asterisk <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(t_test_results<span class="sc">$</span>p_value <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="st">""</span>)</span>
<span id="cb33-268"><a href="#cb33-268" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-269"><a href="#cb33-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-270"><a href="#cb33-270" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.9 Delta Cq Box Plots</span></span>
<span id="cb33-271"><a href="#cb33-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-272"><a href="#cb33-272" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1.9.1 Seed</span></span>
<span id="cb33-273"><a href="#cb33-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-274"><a href="#cb33-274" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-275"><a href="#cb33-275" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb33-276"><a href="#cb33-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-277"><a href="#cb33-277" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter delta_Cq_df_filtered for seed life stage</span></span>
<span id="cb33-278"><a href="#cb33-278" aria-hidden="true" tabindex="-1"></a>seed_delta_Cq_df <span class="ot">&lt;-</span> delta_Cq_df_filtered <span class="sc">%&gt;%</span></span>
<span id="cb33-279"><a href="#cb33-279" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(life.stage <span class="sc">==</span> <span class="st">"seed"</span>)</span>
<span id="cb33-280"><a href="#cb33-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-281"><a href="#cb33-281" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the box plot</span></span>
<span id="cb33-282"><a href="#cb33-282" aria-hidden="true" tabindex="-1"></a>boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(seed_delta_Cq_df, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> treatment)) <span class="sc">+</span></span>
<span id="cb33-283"><a href="#cb33-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb33-284"><a href="#cb33-284" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb33-285"><a href="#cb33-285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb33-286"><a href="#cb33-286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Delta Cq"</span>)</span>
<span id="cb33-287"><a href="#cb33-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-288"><a href="#cb33-288" aria-hidden="true" tabindex="-1"></a><span class="co"># Add asterisks</span></span>
<span id="cb33-289"><a href="#cb33-289" aria-hidden="true" tabindex="-1"></a>boxplot <span class="sc">+</span></span>
<span id="cb33-290"><a href="#cb33-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> t_test_results<span class="sc">$</span>Target, <span class="at">y =</span> <span class="cn">Inf</span>, <span class="at">label =</span> t_test_results<span class="sc">$</span>asterisk,</span>
<span id="cb33-291"><a href="#cb33-291" aria-hidden="true" tabindex="-1"></a>           <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">color =</span> <span class="st">"orange"</span>)</span>
<span id="cb33-292"><a href="#cb33-292" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-293"><a href="#cb33-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-294"><a href="#cb33-294" aria-hidden="true" tabindex="-1"></a>&lt;div class="figure"&gt;</span>
<span id="cb33-295"><a href="#cb33-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-296"><a href="#cb33-296" aria-hidden="true" tabindex="-1"></a>&lt;img src="20240327-cgig-lifestage-carryover-qpcr-analysis_files/figure-gfm/fig-box-plots-seed-delta-Cq-1.png" alt="Box plots comparing GAPDH-normalized gene expression (delta Cq) between control and treated seed. Blue boxes are treated and red boxes are control. HSP90 is the only gene showing higher expression in treated samples. However, none of the differences in gene expression between controls and treated showed any statistically significant (t-test) differences."  /&gt;</span>
<span id="cb33-297"><a href="#cb33-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-298"><a href="#cb33-298" aria-hidden="true" tabindex="-1"></a>&lt;p class="caption"&gt;</span>
<span id="cb33-299"><a href="#cb33-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-300"><a href="#cb33-300" aria-hidden="true" tabindex="-1"></a>Box plots comparing GAPDH-normalized gene expression (delta Cq) between</span>
<span id="cb33-301"><a href="#cb33-301" aria-hidden="true" tabindex="-1"></a>control and treated seed.</span>
<span id="cb33-302"><a href="#cb33-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-303"><a href="#cb33-303" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb33-304"><a href="#cb33-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-305"><a href="#cb33-305" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;</span>
<span id="cb33-306"><a href="#cb33-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-307"><a href="#cb33-307" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1.9.2 Spat</span></span>
<span id="cb33-308"><a href="#cb33-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-309"><a href="#cb33-309" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-310"><a href="#cb33-310" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter data for life.stage = "spat"</span></span>
<span id="cb33-311"><a href="#cb33-311" aria-hidden="true" tabindex="-1"></a>spat_delta_Cq <span class="ot">&lt;-</span> delta_Cq_df_filtered <span class="sc">%&gt;%</span></span>
<span id="cb33-312"><a href="#cb33-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(life.stage <span class="sc">==</span> <span class="st">"spat"</span>)</span>
<span id="cb33-313"><a href="#cb33-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-314"><a href="#cb33-314" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the maximum delta_Cq for each Target</span></span>
<span id="cb33-315"><a href="#cb33-315" aria-hidden="true" tabindex="-1"></a>max_delta_Cq_by_target <span class="ot">&lt;-</span> spat_delta_Cq <span class="sc">%&gt;%</span></span>
<span id="cb33-316"><a href="#cb33-316" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb33-317"><a href="#cb33-317" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">max_delta_Cq =</span> <span class="fu">max</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb33-318"><a href="#cb33-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-319"><a href="#cb33-319" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge t_test_results with max_delta_Cq_by_target to get the maximum delta_Cq for each Target</span></span>
<span id="cb33-320"><a href="#cb33-320" aria-hidden="true" tabindex="-1"></a>t_test_results_with_max_delta_Cq <span class="ot">&lt;-</span> <span class="fu">merge</span>(t_test_results, max_delta_Cq_by_target, <span class="at">by =</span> <span class="st">"Target"</span>)</span>
<span id="cb33-321"><a href="#cb33-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-322"><a href="#cb33-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-323"><a href="#cb33-323" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the box plot</span></span>
<span id="cb33-324"><a href="#cb33-324" aria-hidden="true" tabindex="-1"></a>boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(spat_delta_Cq, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> treatment)) <span class="sc">+</span></span>
<span id="cb33-325"><a href="#cb33-325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb33-326"><a href="#cb33-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb33-327"><a href="#cb33-327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb33-328"><a href="#cb33-328" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Delta Cq"</span>)</span>
<span id="cb33-329"><a href="#cb33-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-330"><a href="#cb33-330" aria-hidden="true" tabindex="-1"></a><span class="co"># Add asterisks</span></span>
<span id="cb33-331"><a href="#cb33-331" aria-hidden="true" tabindex="-1"></a>boxplot <span class="sc">+</span></span>
<span id="cb33-332"><a href="#cb33-332" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> t_test_results_with_max_delta_Cq<span class="sc">$</span>Target, </span>
<span id="cb33-333"><a href="#cb33-333" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> t_test_results_with_max_delta_Cq<span class="sc">$</span>max_delta_Cq <span class="sc">+</span> <span class="fl">0.5</span>, </span>
<span id="cb33-334"><a href="#cb33-334" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> t_test_results_with_max_delta_Cq<span class="sc">$</span>asterisk,</span>
<span id="cb33-335"><a href="#cb33-335" aria-hidden="true" tabindex="-1"></a>           <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">color =</span> <span class="st">"orange"</span>)</span>
<span id="cb33-336"><a href="#cb33-336" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-337"><a href="#cb33-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-338"><a href="#cb33-338" aria-hidden="true" tabindex="-1"></a>&lt;div class="figure"&gt;</span>
<span id="cb33-339"><a href="#cb33-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-340"><a href="#cb33-340" aria-hidden="true" tabindex="-1"></a>&lt;img src="20240327-cgig-lifestage-carryover-qpcr-analysis_files/figure-gfm/fig-box-plots-spat-delta-Cq-1.png" alt="Box plots comparing GAPDH-normalized gene expression (delta Cq) between control and treated spat. Blue boxes are treated and red boxes are control. HSP90 is the only gene showing higher expression in treated samples. The genes cGAS, HSP70, and HSP90 have orange asterisks above them, indicating the expression was statistically (t-test) significantly different between the control and treatment conditions."  /&gt;</span>
<span id="cb33-341"><a href="#cb33-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-342"><a href="#cb33-342" aria-hidden="true" tabindex="-1"></a>&lt;p class="caption"&gt;</span>
<span id="cb33-343"><a href="#cb33-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-344"><a href="#cb33-344" aria-hidden="true" tabindex="-1"></a>Box plots comparing GAPDH-normalized gene expression (delta Cq) between</span>
<span id="cb33-345"><a href="#cb33-345" aria-hidden="true" tabindex="-1"></a>control and treated spat.</span>
<span id="cb33-346"><a href="#cb33-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-347"><a href="#cb33-347" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb33-348"><a href="#cb33-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-349"><a href="#cb33-349" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;</span>
<span id="cb33-350"><a href="#cb33-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-351"><a href="#cb33-351" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1.10 Delta delta Cq</span></span>
<span id="cb33-352"><a href="#cb33-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-353"><a href="#cb33-353" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1.10.1 Add treatment and life stage</span></span>
<span id="cb33-354"><a href="#cb33-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-355"><a href="#cb33-355" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-356"><a href="#cb33-356" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize empty vectors to store life.stage and treatment</span></span>
<span id="cb33-357"><a href="#cb33-357" aria-hidden="true" tabindex="-1"></a>life_stage <span class="ot">&lt;-</span> <span class="fu">character</span>(<span class="fu">nrow</span>(combined_df))</span>
<span id="cb33-358"><a href="#cb33-358" aria-hidden="true" tabindex="-1"></a>treatment <span class="ot">&lt;-</span> <span class="fu">character</span>(<span class="fu">nrow</span>(combined_df))</span>
<span id="cb33-359"><a href="#cb33-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-360"><a href="#cb33-360" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each row of combined_df</span></span>
<span id="cb33-361"><a href="#cb33-361" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(combined_df)) {</span>
<span id="cb33-362"><a href="#cb33-362" aria-hidden="true" tabindex="-1"></a>  sample_id <span class="ot">&lt;-</span> combined_df<span class="sc">$</span>Sample[i]</span>
<span id="cb33-363"><a href="#cb33-363" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb33-364"><a href="#cb33-364" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if the sample_id is present in any of the vectors</span></span>
<span id="cb33-365"><a href="#cb33-365" aria-hidden="true" tabindex="-1"></a>  found <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb33-366"><a href="#cb33-366" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (vec_name <span class="cf">in</span> <span class="fu">names</span>(vector_list)) {</span>
<span id="cb33-367"><a href="#cb33-367" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (sample_id <span class="sc">%in%</span> vector_list[[vec_name]]) {</span>
<span id="cb33-368"><a href="#cb33-368" aria-hidden="true" tabindex="-1"></a>      <span class="co"># If present, extract life.stage and treatment from the vector name</span></span>
<span id="cb33-369"><a href="#cb33-369" aria-hidden="true" tabindex="-1"></a>      parts <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(vec_name, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>)[[<span class="dv">1</span>]]</span>
<span id="cb33-370"><a href="#cb33-370" aria-hidden="true" tabindex="-1"></a>      life_stage[i] <span class="ot">&lt;-</span> parts[<span class="dv">1</span>]</span>
<span id="cb33-371"><a href="#cb33-371" aria-hidden="true" tabindex="-1"></a>      treatment[i] <span class="ot">&lt;-</span> parts[<span class="dv">2</span>]</span>
<span id="cb33-372"><a href="#cb33-372" aria-hidden="true" tabindex="-1"></a>      found <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb33-373"><a href="#cb33-373" aria-hidden="true" tabindex="-1"></a>      <span class="cf">break</span>  <span class="co"># Exit loop once found</span></span>
<span id="cb33-374"><a href="#cb33-374" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb33-375"><a href="#cb33-375" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb33-376"><a href="#cb33-376" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb33-377"><a href="#cb33-377" aria-hidden="true" tabindex="-1"></a>  <span class="co"># If sample_id is not found in any vector, assign NA to both life.stage and treatment</span></span>
<span id="cb33-378"><a href="#cb33-378" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>found) {</span>
<span id="cb33-379"><a href="#cb33-379" aria-hidden="true" tabindex="-1"></a>    life_stage[i] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb33-380"><a href="#cb33-380" aria-hidden="true" tabindex="-1"></a>    treatment[i] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb33-381"><a href="#cb33-381" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb33-382"><a href="#cb33-382" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-383"><a href="#cb33-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-384"><a href="#cb33-384" aria-hidden="true" tabindex="-1"></a><span class="co"># Add life.stage and treatment columns to combined_df</span></span>
<span id="cb33-385"><a href="#cb33-385" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> combined_df <span class="sc">%&gt;%</span></span>
<span id="cb33-386"><a href="#cb33-386" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">life.stage =</span> life_stage,</span>
<span id="cb33-387"><a href="#cb33-387" aria-hidden="true" tabindex="-1"></a>         <span class="at">treatment =</span> treatment)</span>
<span id="cb33-388"><a href="#cb33-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-389"><a href="#cb33-389" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out rows where life.stage is NA</span></span>
<span id="cb33-390"><a href="#cb33-390" aria-hidden="true" tabindex="-1"></a>combined_df_filtered <span class="ot">&lt;-</span> combined_df <span class="sc">%&gt;%</span> </span>
<span id="cb33-391"><a href="#cb33-391" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(life.stage))</span>
<span id="cb33-392"><a href="#cb33-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-393"><a href="#cb33-393" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(combined_df_filtered)</span>
<span id="cb33-394"><a href="#cb33-394" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-395"><a href="#cb33-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-396"><a href="#cb33-396" aria-hidden="true" tabindex="-1"></a><span class="in">    'data.frame':   256 obs. of  19 variables:</span></span>
<span id="cb33-397"><a href="#cb33-397" aria-hidden="true" tabindex="-1"></a><span class="in">     $ data_frame_id         : chr  "../data//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" "../data//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" "../data//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" "../data//sam_2024-03-25_06-10-54_Connect-Quantification-Cq_Results.csv" ...</span></span>
<span id="cb33-398"><a href="#cb33-398" aria-hidden="true" tabindex="-1"></a><span class="in">     $ X                     : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb33-399"><a href="#cb33-399" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Well                  : chr  "A01" "A02" "A03" "A04" ...</span></span>
<span id="cb33-400"><a href="#cb33-400" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Fluor                 : chr  "SYBR" "SYBR" "SYBR" "SYBR" ...</span></span>
<span id="cb33-401"><a href="#cb33-401" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Target                : chr  "ATPsynthase" "ATPsynthase" "ATPsynthase" "ATPsynthase" ...</span></span>
<span id="cb33-402"><a href="#cb33-402" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Content               : chr  "Unkn-01" "Unkn-01" "Unkn-02" "Unkn-02" ...</span></span>
<span id="cb33-403"><a href="#cb33-403" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Sample                : chr  "206" "206" "220" "220" ...</span></span>
<span id="cb33-404"><a href="#cb33-404" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Biological.Set.Name   : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb33-405"><a href="#cb33-405" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Cq                    : num  26.7 26.7 25.8 25.9 25.1 ...</span></span>
<span id="cb33-406"><a href="#cb33-406" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Cq.Mean               : num  26.7 26.7 25.9 25.9 25.1 ...</span></span>
<span id="cb33-407"><a href="#cb33-407" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Cq.Std..Dev           : num  0.0455 0.0455 0.0239 0.0239 0.0813 ...</span></span>
<span id="cb33-408"><a href="#cb33-408" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Starting.Quantity..SQ.: num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...</span></span>
<span id="cb33-409"><a href="#cb33-409" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Log.Starting.Quantity : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...</span></span>
<span id="cb33-410"><a href="#cb33-410" aria-hidden="true" tabindex="-1"></a><span class="in">     $ SQ.Mean               : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...</span></span>
<span id="cb33-411"><a href="#cb33-411" aria-hidden="true" tabindex="-1"></a><span class="in">     $ SQ.Std..Dev           : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...</span></span>
<span id="cb33-412"><a href="#cb33-412" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Set.Point             : int  60 60 60 60 60 60 60 60 60 60 ...</span></span>
<span id="cb33-413"><a href="#cb33-413" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Well.Note             : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb33-414"><a href="#cb33-414" aria-hidden="true" tabindex="-1"></a><span class="in">     $ life.stage            : chr  "spat" "spat" "spat" "spat" ...</span></span>
<span id="cb33-415"><a href="#cb33-415" aria-hidden="true" tabindex="-1"></a><span class="in">     $ treatment             : chr  "control" "control" "treated" "treated" ...</span></span>
<span id="cb33-416"><a href="#cb33-416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-417"><a href="#cb33-417" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1.10.2 Mean Cqs per gene per treatment per life stage</span></span>
<span id="cb33-418"><a href="#cb33-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-419"><a href="#cb33-419" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-420"><a href="#cb33-420" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by life.stage, treatment, and Target, then calculate the mean Cq</span></span>
<span id="cb33-421"><a href="#cb33-421" aria-hidden="true" tabindex="-1"></a>mean_Cq_df <span class="ot">&lt;-</span> combined_df_filtered <span class="sc">%&gt;%</span></span>
<span id="cb33-422"><a href="#cb33-422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, treatment, Target) <span class="sc">%&gt;%</span></span>
<span id="cb33-423"><a href="#cb33-423" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean_Cq =</span> <span class="fu">mean</span>(Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb33-424"><a href="#cb33-424" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-425"><a href="#cb33-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-426"><a href="#cb33-426" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1.10.3 Delta Cqs</span></span>
<span id="cb33-427"><a href="#cb33-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-428"><a href="#cb33-428" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-429"><a href="#cb33-429" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta Cq</span></span>
<span id="cb33-430"><a href="#cb33-430" aria-hidden="true" tabindex="-1"></a>combined_df_with_delta_Cq <span class="ot">&lt;-</span> mean_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb33-431"><a href="#cb33-431" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, treatment) <span class="sc">%&gt;%</span></span>
<span id="cb33-432"><a href="#cb33-432" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">delta_Cq =</span> mean_Cq <span class="sc">-</span> <span class="fu">mean</span>(mean_Cq[Target <span class="sc">==</span> <span class="st">"GAPDH"</span>])) <span class="sc">%&gt;%</span></span>
<span id="cb33-433"><a href="#cb33-433" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb33-434"><a href="#cb33-434" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">!=</span> <span class="st">"GAPDH"</span>)</span>
<span id="cb33-435"><a href="#cb33-435" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-436"><a href="#cb33-436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-437"><a href="#cb33-437" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1.10.4 Delta delta Cq</span></span>
<span id="cb33-438"><a href="#cb33-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-439"><a href="#cb33-439" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-440"><a href="#cb33-440" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta_delta_Cq</span></span>
<span id="cb33-441"><a href="#cb33-441" aria-hidden="true" tabindex="-1"></a>delta_delta_Cq_df <span class="ot">&lt;-</span> combined_df_with_delta_Cq <span class="sc">%&gt;%</span></span>
<span id="cb33-442"><a href="#cb33-442" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, Target) <span class="sc">%&gt;%</span></span>
<span id="cb33-443"><a href="#cb33-443" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">delta_delta_Cq =</span> delta_Cq[treatment <span class="sc">==</span> <span class="st">"treated"</span>] <span class="sc">-</span> delta_Cq[treatment <span class="sc">==</span> <span class="st">"control"</span>])</span>
<span id="cb33-444"><a href="#cb33-444" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-445"><a href="#cb33-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-446"><a href="#cb33-446" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1.10.5 Calculate the fold change for each Target</span></span>
<span id="cb33-447"><a href="#cb33-447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-448"><a href="#cb33-448" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-449"><a href="#cb33-449" aria-hidden="true" tabindex="-1"></a>delta_delta_fold_change <span class="ot">&lt;-</span> delta_delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb33-450"><a href="#cb33-450" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-451"><a href="#cb33-451" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Target, fold_change)</span>
<span id="cb33-452"><a href="#cb33-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-453"><a href="#cb33-453" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_fold_change)</span>
<span id="cb33-454"><a href="#cb33-454" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-455"><a href="#cb33-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-456"><a href="#cb33-456" aria-hidden="true" tabindex="-1"></a><span class="in">    gropd_df [14 × 3] (S3: grouped_df/tbl_df/tbl/data.frame)</span></span>
<span id="cb33-457"><a href="#cb33-457" aria-hidden="true" tabindex="-1"></a><span class="in">     $ life.stage : chr [1:14] "seed" "seed" "seed" "seed" ...</span></span>
<span id="cb33-458"><a href="#cb33-458" aria-hidden="true" tabindex="-1"></a><span class="in">     $ Target     : chr [1:14] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...</span></span>
<span id="cb33-459"><a href="#cb33-459" aria-hidden="true" tabindex="-1"></a><span class="in">     $ fold_change: num [1:14] 1.532 1.146 1.8 0.662 1.365 ...</span></span>
<span id="cb33-460"><a href="#cb33-460" aria-hidden="true" tabindex="-1"></a><span class="in">     - attr(*, "groups")= tibble [2 × 2] (S3: tbl_df/tbl/data.frame)</span></span>
<span id="cb33-461"><a href="#cb33-461" aria-hidden="true" tabindex="-1"></a><span class="in">      ..$ life.stage: chr [1:2] "seed" "spat"</span></span>
<span id="cb33-462"><a href="#cb33-462" aria-hidden="true" tabindex="-1"></a><span class="in">      ..$ .rows     : list&lt;int&gt; [1:2] </span></span>
<span id="cb33-463"><a href="#cb33-463" aria-hidden="true" tabindex="-1"></a><span class="in">      .. ..$ : int [1:7] 1 2 3 4 5 6 7</span></span>
<span id="cb33-464"><a href="#cb33-464" aria-hidden="true" tabindex="-1"></a><span class="in">      .. ..$ : int [1:7] 8 9 10 11 12 13 14</span></span>
<span id="cb33-465"><a href="#cb33-465" aria-hidden="true" tabindex="-1"></a><span class="in">      .. ..@ ptype: int(0) </span></span>
<span id="cb33-466"><a href="#cb33-466" aria-hidden="true" tabindex="-1"></a><span class="in">      ..- attr(*, ".drop")= logi TRUE</span></span>
<span id="cb33-467"><a href="#cb33-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-468"><a href="#cb33-468" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1.10.6 Plot - Seed Fold Change</span></span>
<span id="cb33-469"><a href="#cb33-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-470"><a href="#cb33-470" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-471"><a href="#cb33-471" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter delta_delta_fold_change for seed life stage</span></span>
<span id="cb33-472"><a href="#cb33-472" aria-hidden="true" tabindex="-1"></a>seed_df <span class="ot">&lt;-</span> delta_delta_fold_change <span class="sc">%&gt;%</span></span>
<span id="cb33-473"><a href="#cb33-473" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(life.stage <span class="sc">==</span> <span class="st">"seed"</span>)</span>
<span id="cb33-474"><a href="#cb33-474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-475"><a href="#cb33-475" aria-hidden="true" tabindex="-1"></a><span class="co"># Create bar plot for seed life stage</span></span>
<span id="cb33-476"><a href="#cb33-476" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the plot with fold changes relative to baseline of 1</span></span>
<span id="cb33-477"><a href="#cb33-477" aria-hidden="true" tabindex="-1"></a>seed_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(seed_df, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> fold_change <span class="sc">-</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb33-478"><a href="#cb33-478" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"skyblue"</span>) <span class="sc">+</span></span>
<span id="cb33-479"><a href="#cb33-479" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span>  <span class="co"># Baseline</span></span>
<span id="cb33-480"><a href="#cb33-480" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb33-481"><a href="#cb33-481" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Fold Change"</span>, <span class="at">title =</span> <span class="st">"Seed Life Stage Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb33-482"><a href="#cb33-482" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.5</span>, <span class="dv">1</span>))</span>
<span id="cb33-483"><a href="#cb33-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-484"><a href="#cb33-484" aria-hidden="true" tabindex="-1"></a><span class="co"># Display plot</span></span>
<span id="cb33-485"><a href="#cb33-485" aria-hidden="true" tabindex="-1"></a>seed_plot</span>
<span id="cb33-486"><a href="#cb33-486" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-487"><a href="#cb33-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-488"><a href="#cb33-488" aria-hidden="true" tabindex="-1"></a>&lt;div class="figure"&gt;</span>
<span id="cb33-489"><a href="#cb33-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-490"><a href="#cb33-490" aria-hidden="true" tabindex="-1"></a>&lt;img src="20240327-cgig-lifestage-carryover-qpcr-analysis_files/figure-gfm/fig-bar-plots-seed-fold-change-1.png" alt="Bar plots showing fold change in expression (2^(-delta delta Cq)) in seed. HSP90 is the only gene which is more highly expressed in treatment relative to controls in seed."  /&gt;</span>
<span id="cb33-491"><a href="#cb33-491" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-492"><a href="#cb33-492" aria-hidden="true" tabindex="-1"></a>&lt;p class="caption"&gt;</span>
<span id="cb33-493"><a href="#cb33-493" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-494"><a href="#cb33-494" aria-hidden="true" tabindex="-1"></a>Bar plots showing fold change in expression (2^(-delta delta Cq)) in</span>
<span id="cb33-495"><a href="#cb33-495" aria-hidden="true" tabindex="-1"></a>seed.</span>
<span id="cb33-496"><a href="#cb33-496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-497"><a href="#cb33-497" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb33-498"><a href="#cb33-498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-499"><a href="#cb33-499" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;</span>
<span id="cb33-500"><a href="#cb33-500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-501"><a href="#cb33-501" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1.10.7 Plot - Spat Fold Change</span></span>
<span id="cb33-502"><a href="#cb33-502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-503"><a href="#cb33-503" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb33-504"><a href="#cb33-504" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter delta_delta_fold_change for spat life stage</span></span>
<span id="cb33-505"><a href="#cb33-505" aria-hidden="true" tabindex="-1"></a>spat_df <span class="ot">&lt;-</span> delta_delta_fold_change <span class="sc">%&gt;%</span></span>
<span id="cb33-506"><a href="#cb33-506" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(life.stage <span class="sc">==</span> <span class="st">"spat"</span>)</span>
<span id="cb33-507"><a href="#cb33-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-508"><a href="#cb33-508" aria-hidden="true" tabindex="-1"></a><span class="co"># Create bar plot for spat life stage</span></span>
<span id="cb33-509"><a href="#cb33-509" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the plot with fold changes relative to baseline of 1</span></span>
<span id="cb33-510"><a href="#cb33-510" aria-hidden="true" tabindex="-1"></a>spat_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(spat_df, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> fold_change <span class="sc">-</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb33-511"><a href="#cb33-511" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"salmon"</span>) <span class="sc">+</span></span>
<span id="cb33-512"><a href="#cb33-512" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span>  <span class="co"># Baseline</span></span>
<span id="cb33-513"><a href="#cb33-513" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb33-514"><a href="#cb33-514" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Fold Change"</span>, <span class="at">title =</span> <span class="st">"Spat Life Stage Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb33-515"><a href="#cb33-515" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">8</span>))</span>
<span id="cb33-516"><a href="#cb33-516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-517"><a href="#cb33-517" aria-hidden="true" tabindex="-1"></a><span class="co"># Display plot</span></span>
<span id="cb33-518"><a href="#cb33-518" aria-hidden="true" tabindex="-1"></a>spat_plot</span>
<span id="cb33-519"><a href="#cb33-519" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb33-520"><a href="#cb33-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-521"><a href="#cb33-521" aria-hidden="true" tabindex="-1"></a>&lt;div class="figure"&gt;</span>
<span id="cb33-522"><a href="#cb33-522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-523"><a href="#cb33-523" aria-hidden="true" tabindex="-1"></a>&lt;img src="20240327-cgig-lifestage-carryover-qpcr-analysis_files/figure-gfm/fig-bar-plots-spat-fold-change-1.png" alt="Bar plots showing fold change in expression (2^(-delta delta Cq)) in spat. HSP90 is the only gene which is more highly expressed in treatment relative to controls in spat."  /&gt;</span>
<span id="cb33-524"><a href="#cb33-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-525"><a href="#cb33-525" aria-hidden="true" tabindex="-1"></a>&lt;p class="caption"&gt;</span>
<span id="cb33-526"><a href="#cb33-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-527"><a href="#cb33-527" aria-hidden="true" tabindex="-1"></a>Bar plots showing fold change in expression (2^(-delta delta Cq)) in</span>
<span id="cb33-528"><a href="#cb33-528" aria-hidden="true" tabindex="-1"></a>spat.</span>
<span id="cb33-529"><a href="#cb33-529" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-530"><a href="#cb33-530" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb33-531"><a href="#cb33-531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-532"><a href="#cb33-532" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>