<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sam White">
<meta name="dcterms.date" content="2025-04-24">

<title>Taxonomy Assignment - Yellow Island 2018 eDNA Samples Using QIIME 2 – Sam’s Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 2,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 100,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-7483T61T11"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-7483T61T11', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Sam’s Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" target="_blank">
 <span class="dropdown-text">Rstudio Server - Raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu/Atumefaciens/" target="_blank">
 <span class="dropdown-text">My Gannet Directory</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" target="_blank">
 <span class="dropdown-text">Steven’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shedurkin.github.io/Roberts-LabNotebook/" target="_blank">
 <span class="dropdown-text">Kathleen’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://meganewing.github.io/mewing-notebook/" target="_blank">
 <span class="dropdown-text">Megan’s Notebook</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/sams-notebook" target="_blank">
 <span class="dropdown-text">Notebook repo</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" target="_blank">
 <span class="dropdown-text">Lab Organization (GitHub)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" target="_blank">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" target="_blank"> <i class="bi bi-calendar-week" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-right">
      <h1 class="title">Taxonomy Assignment - Yellow Island 2018 eDNA Samples Using QIIME 2</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2025</div>
                <div class="quarto-category">taxonomy</div>
                <div class="quarto-category">Yellow Island</div>
                <div class="quarto-category">eDNA</div>
                <div class="quarto-category">QIIME 2</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-right">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Sam White </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">April 24, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#intro" id="toc-intro" class="nav-link active" data-scroll-target="#intro">INTRO</a></li>
  <li><a href="#background" id="toc-background" class="nav-link" data-scroll-target="#background">1 BACKGROUND</a>
  <ul>
  <li><a href="#inputs" id="toc-inputs" class="nav-link" data-scroll-target="#inputs">1.1 Inputs:</a></li>
  <li><a href="#outputs" id="toc-outputs" class="nav-link" data-scroll-target="#outputs">1.2 Outputs:</a></li>
  </ul></li>
  <li><a href="#set-r-variables" id="toc-set-r-variables" class="nav-link" data-scroll-target="#set-r-variables">2 Set R variables</a></li>
  <li><a href="#load-qiime2-amplicon-2024.10-conda-environment" id="toc-load-qiime2-amplicon-2024.10-conda-environment" class="nav-link" data-scroll-target="#load-qiime2-amplicon-2024.10-conda-environment">3 Load <code>qiime2-amplicon-2024.10</code> conda environment</a></li>
  <li><a href="#import-trimmed-reads" id="toc-import-trimmed-reads" class="nav-link" data-scroll-target="#import-trimmed-reads">4 Import trimmed reads</a>
  <ul>
  <li><a href="#transfer-trimmed-reads" id="toc-transfer-trimmed-reads" class="nav-link" data-scroll-target="#transfer-trimmed-reads">4.1 Transfer trimmed reads</a></li>
  <li><a href="#import-to-qiime2-qza-format" id="toc-import-to-qiime2-qza-format" class="nav-link" data-scroll-target="#import-to-qiime2-qza-format">4.2 Import to QIIME2 QZA format</a></li>
  </ul></li>
  <li><a href="#deblur" id="toc-deblur" class="nav-link" data-scroll-target="#deblur">5 Deblur</a>
  <ul>
  <li><a href="#quality-filter" id="toc-quality-filter" class="nav-link" data-scroll-target="#quality-filter">5.1 Quality filter</a></li>
  <li><a href="#denoise" id="toc-denoise" class="nav-link" data-scroll-target="#denoise">5.2 Denoise</a></li>
  <li><a href="#deblur-visualization" id="toc-deblur-visualization" class="nav-link" data-scroll-target="#deblur-visualization">5.3 Deblur visualization</a></li>
  </ul></li>
  <li><a href="#feature-table" id="toc-feature-table" class="nav-link" data-scroll-target="#feature-table">6 Feature Table</a></li>
  <li><a href="#taxonomy" id="toc-taxonomy" class="nav-link" data-scroll-target="#taxonomy">7 Taxonomy</a>
  <ul>
  <li><a href="#classification" id="toc-classification" class="nav-link" data-scroll-target="#classification">7.1 Classification</a></li>
  <li><a href="#bar-plots" id="toc-bar-plots" class="nav-link" data-scroll-target="#bar-plots">7.2 Bar plots</a></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">RESULTS</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">REFERENCES</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-right" id="quarto-document-content">





<section id="intro" class="level1">
<h1>INTRO</h1>
<p>Performed taxonomic assignment for the July 18 Yellow Island sites for <a href="https://github.com/RobertsLab/project-eDNA-yellow">project-eDNA-yellow</a> (GitHub repo) project using QIIME 2 analysis, as <a href="https://github.com/RobertsLab/resources/issues/2167">requested in this GitHub Issue</a>.</p>
<p>The resulting taxonomic output file is a QIIME 2 bar plot visualization. This can be downloaded (either by using the link below or by cloning this project’s repo) and then uploaded to the <a href="https://view.qiime2.org/">QIIME 2 Viewer website</a>. This provides interactive viewing/filtering/sorting of taxonomic assignments.</p>
<ul>
<li><a href="https://github.com/RobertsLab/project-eDNA-yellow/raw/516fd3772577550c1d8f991b116fe1f1081a0cec/output/02.00-qiime2/taxa-bar-plots.qzv"><code>taxa-bar-plots.qzv</code></a></li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The contents below are from markdown knitted from <a href="https://github.com/RobertsLab/project-eDNA-yellow/blob/01f599e4e7072b0a546351e1d1a7206e839a7755/code/02.00-qiime2.Rmd">02.00-qiime2.Rmd</a> (commit <code>01f599e4e</code>).</p>
</div>
</div>
</section>
<section id="background" class="level1">
<h1>1 BACKGROUND</h1>
<p>This notebook will use <a href="https://docs.qiime2.org/2024.10/">QIIME 2 2024.10</a> (Bolyen et al.&nbsp;2019) to taxonomically classify eukaryotic sequences in eDNA samples. Expects trimmed reads to exist in <a href="https://github.com/RobertsLab/project-eDNA-yellow/tree/main/output/01.00-trimming-fastp-fastqc-multiqc"><code>../output/01.00-trimming-fastp-fastqc-multiqc</code></a>.</p>
<section id="inputs" class="level2">
<h2 class="anchored" data-anchor-id="inputs">1.1 Inputs:</h2>
<ul>
<li><p>Trimmed FastQs from <a href="https://github.com/RobertsLab/project-eDNA-yellow/blob/13bdbbec60b0da23266af2e0b409c2b09f3d38fa/code/01.00-trimming-fastp-fastqc-multiqc.Rmd"><code>01.00-trimming-fastp-fastqc-multiqc.Rmd</code></a></p>
<ul>
<li>Filename format: <code>fastp-Yell-July18-Site5e_S95_L001_R1_001.fastq.gz</code></li>
</ul></li>
<li><p>SILVA classifier QZA <code>silva-138-99-nb-classifier.qza</code>.</p>
<ul>
<li>Downloaded from <a href="https://library.qiime2.org/data-resources#naive-bayes-classifiers" class="uri">https://library.qiime2.org/data-resources#naive-bayes-classifiers</a></li>
</ul></li>
</ul>
</section>
<section id="outputs" class="level2">
<h2 class="anchored" data-anchor-id="outputs">1.2 Outputs:</h2>
<p>Files ending in <code>.qza</code> are QIIME zip archives.</p>
<p>Files ending in <code>.qzv</code> are QIIME zip visualization tables.</p>
<ul>
<li><p><code>deblur-stats.qza</code>: Stats after denoise.</p></li>
<li><p><code>deblur-stats.qzv</code>: Stats visualization after denoise.</p></li>
<li><p><code>demux-paired-end-filtered.qza</code>: Quality score filtered archive.</p></li>
<li><p><code>demux-paired-end-filter-stats.qza</code>: Stats archive for quality score filtering.</p></li>
<li><p><code>demux-paired-end.qza</code>: Paired-end reads archive.</p></li>
<li><p><code>rep-seqs-deblur.qza</code>: Representative sequences after denoise.</p></li>
<li><p><code>rep-seqs-deblur.qzv</code>: Representative sequences after denoise visualization.</p></li>
<li><p><code>table-deblur.qza</code>: Table after denoise.</p></li>
<li><p><code>table.qzv</code>: Feature table summary visuallization.</p></li>
<li><p><code>taxa-bar-plots.qzv</code>: Bar plots of taxonomic assignment visualization.</p></li>
<li><p><code>taxonomy.qza</code>: Feature taxonomic classification after applying SILVA classifier.</p></li>
<li><p><code>taxonomy.qzv</code>: Feature taxonomic classification visualization after applying SILVA classifier.</p></li>
</ul>
<p>Software requirements:</p>
<ul>
<li>Utilizes a <a href="https://docs.qiime2.org/2024.10/"><code>qiime2-amplicon-2024.10</code></a> Conda/Mamba environment, per the installation instructions.</li>
</ul>
<p>To use on your own system, replace with name of your qiime2-amplicon-2024.10 environment and the path to the corresponding conda installation (find this <em>after</em> you’ve activated the environment).</p>
<p>E.g.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Activate environment</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate ShortStack4_env</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Find conda path</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">which</span> conda</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<hr>
</section>
</section>
<section id="set-r-variables" class="level1">
<h1>2 Set R variables</h1>
<p>Change <code>repo_dir</code> to match local repo location.</p>
<p>Change <code>threads &lt;- 4</code> to match computer resources for your system.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Conda info</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>qiime2_conda_env_name <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"qiime2-amplicon-2024.10"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>qiime2_conda_path <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"/home/sam/programs/miniforge3/condabin/conda"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Data directories</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>repo_dir <span class="ot">&lt;-</span> <span class="st">"/home/sam/gitrepos/RobertsLab/project-eDNA-yellow"</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(repo_dir, <span class="st">"data"</span>)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>output_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(repo_dir, <span class="st">"output"</span>, <span class="st">"02.00-qiime2"</span>)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>trimmed_reads_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(repo_dir, <span class="st">"output"</span>, <span class="st">"01.00-trimming-fastp-fastqc-multiqc"</span>)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Files</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"raw-fastqs"</span>, <span class="st">"metadata.tsv"</span>)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>silva_classifier <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"silva-138-99-nb-classifier.qza"</span>)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>silva_ref_seqs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"silva-138-99-seqs.qza"</span>)</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Parameters</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="do">## Set Deblur read length</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Based on FastQC results, post-trimming, where quality starts to decline.</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>deblur_ptrim_length <span class="ot">&lt;-</span> <span class="dv">150</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a><span class="co"># CPUs</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>threads <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Export these as environment variables for bash chunks.</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.setenv</span>(</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">data_dir =</span> data_dir,</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">deblur_ptrim_length =</span> deblur_ptrim_length,</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">metadata =</span> metadata,</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">silva_classifier =</span> silva_classifier,</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">silva_ref_seqs =</span> silva_ref_seqs,</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">threads =</span> threads,</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">trimmed_reads_dir =</span> trimmed_reads_dir,</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">output_dir =</span> output_dir</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="load-qiime2-amplicon-2024.10-conda-environment" class="level1">
<h1>3 Load <a href="https://docs.qiime2.org/2024.10/"><code>qiime2-amplicon-2024.10</code></a> conda environment</h1>
<p>If this is successful, the first line of output should show that the Python being used is the one in your <a href="https://docs.qiime2.org/2024.10/"><code>qiime2-amplicon-2024.10</code></a> conda environment path.</p>
<p>E.g.</p>
<p><code>python:         /home/sam/programs/mambaforge/envs/mirmachine_env/bin/python</code></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">use_condaenv</span>(<span class="at">condaenv =</span> qiime2_conda_env_name, <span class="at">conda =</span> qiime2_conda_path)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Check successful env loading</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">py_config</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>python:         /home/sam/programs/miniforge3/envs/qiime2-amplicon-2024.10/bin/python
libpython:      /home/sam/programs/miniforge3/envs/qiime2-amplicon-2024.10/lib/libpython3.10.so
pythonhome:     /home/sam/programs/miniforge3/envs/qiime2-amplicon-2024.10:/home/sam/programs/miniforge3/envs/qiime2-amplicon-2024.10
version:        3.10.14 | packaged by conda-forge | (main, Mar 20 2024, 12:45:18) [GCC 12.3.0]
numpy:          /home/sam/programs/miniforge3/envs/qiime2-amplicon-2024.10/lib/python3.10/site-packages/numpy
numpy_version:  1.26.4

NOTE: Python version was forced by use_python() function</code></pre>
</section>
<section id="import-trimmed-reads" class="level1">
<h1>4 Import trimmed reads</h1>
<section id="transfer-trimmed-reads" class="level2">
<h2 class="anchored" data-anchor-id="transfer-trimmed-reads">4.1 Transfer trimmed reads</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make output directory if it doesn't exist</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">--parents</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rsync</span> <span class="at">-avP</span> <span class="st">"</span><span class="va">${trimmed_reads_dir}</span><span class="st">"</span>/<span class="pp">*</span>.fastq.gz <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-lh</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>sending incremental file list
Site1J_S99_L001_R1_001.fastq.gz
         32,768   3%    0.00kB/s    0:00:00        1,049,054 100%  969.21MB/s    0:00:00 (xfr#1, to-chk=107/108)
Site1J_S99_L001_R2_001.fastq.gz
         32,768   3%   15.62MB/s    0:00:00          870,682 100%  207.59MB/s    0:00:00 (xfr#2, to-chk=106/108)
Site1L_S101_L001_R1_001.fastq.gz
         32,768   1%    7.81MB/s    0:00:00        2,695,538 100%  321.33MB/s    0:00:00 (xfr#3, to-chk=105/108)
Site1L_S101_L001_R2_001.fastq.gz
         32,768   1%    3.91MB/s    0:00:00        3,045,986 100%  223.45MB/s    0:00:00 (xfr#4, to-chk=104/108)
Site1U_S114_L001_R1_001.fastq.gz
         32,768   0%    2.40MB/s    0:00:01        3,303,317 100%  175.02MB/s    0:00:00 (xfr#5, to-chk=103/108)
Site1U_S114_L001_R2_001.fastq.gz
         32,768   0%    1.74MB/s    0:00:02        3,694,663 100%  146.81MB/s    0:00:00 (xfr#6, to-chk=102/108)
Site1a_S83_L001_R1_001.fastq.gz
         32,768   0%    1.30MB/s    0:00:07       10,094,265 100%  178.27MB/s    0:00:00 (xfr#7, to-chk=101/108)
Site1a_S83_L001_R2_001.fastq.gz
         32,768   0%  592.59kB/s    0:00:18       10,924,826 100%  127.06MB/s    0:00:00 (xfr#8, to-chk=100/108)
Site1b_S84_L001_R1_001.fastq.gz
         32,768   0%  390.24kB/s    0:00:24        9,597,931 100%   87.17MB/s    0:00:00 (xfr#9, to-chk=99/108)
Site1b_S84_L001_R2_001.fastq.gz
         32,768   0%  304.76kB/s    0:00:34       10,396,372 100%   71.85MB/s    0:00:00 (xfr#10, to-chk=98/108)
Site1bb_S139_L001_R1_001.fastq.gz
         32,768   0%  231.88kB/s    0:00:31        7,292,087 100%   44.58MB/s    0:00:00 (xfr#11, to-chk=97/108)
Site1bb_S139_L001_R2_001.fastq.gz
         32,768   0%  203.82kB/s    0:00:38        7,812,044 100%   43.82MB/s    0:00:00 (xfr#12, to-chk=96/108)
Site1c_S85_L001_R1_001.fastq.gz
         32,768   0%  188.24kB/s    0:00:37        7,079,940 100%   34.45MB/s    0:00:00 (xfr#13, to-chk=95/108)
Site1c_S85_L001_R2_001.fastq.gz
         32,768   0%  163.27kB/s    0:00:47        7,720,428 100%   34.73MB/s    0:00:00 (xfr#14, to-chk=94/108)
Site1cc_S140_L001_R1_001.fastq.gz
         32,768   0%  150.23kB/s    0:01:55       17,422,172 100%   67.27MB/s    0:00:00 (xfr#15, to-chk=93/108)
Site1cc_S140_L001_R2_001.fastq.gz
         32,768   0%  129.03kB/s    0:02:29       19,271,102 100%   62.09MB/s    0:00:00 (xfr#16, to-chk=92/108)
Site1d_S86_L001_R1_001.fastq.gz
         32,768   0%  107.74kB/s    0:01:31        9,882,039 100%   29.27MB/s    0:00:00 (xfr#17, to-chk=91/108)
Site1d_S86_L001_R2_001.fastq.gz
         32,768   0%   99.38kB/s    0:01:50       10,999,755 100%   30.23MB/s    0:00:00 (xfr#18, to-chk=90/108)
Site1dd_S141_L001_R1_001.fastq.gz
         32,768   0%   91.95kB/s    0:03:24       18,882,154 100%   46.89MB/s    0:00:00 (xfr#19, to-chk=89/108)
Site1dd_S141_L001_R2_001.fastq.gz
         32,768   0%   83.33kB/s    0:03:48       19,062,705 100%   42.98MB/s    0:00:00 (xfr#20, to-chk=88/108)
Site1e_S91_L001_R1_001.fastq.gz
         32,768   2%   75.65kB/s    0:00:21        1,632,564 100%    3.65MB/s    0:00:00 (xfr#21, to-chk=87/108)
Site1e_S91_L001_R2_001.fastq.gz
         32,768   1%   74.94kB/s    0:00:22        1,709,840 100%    3.78MB/s    0:00:00 (xfr#22, to-chk=86/108)
Site1ee_S142_L001_R1_001.fastq.gz
         32,768   0%   74.25kB/s    0:02:35       11,554,889 100%   24.17MB/s    0:00:00 (xfr#23, to-chk=85/108)
Site1ee_S142_L001_R2_001.fastq.gz
         32,768   0%   70.18kB/s    0:02:43       11,489,305 100%   22.73MB/s    0:00:00 (xfr#24, to-chk=84/108)
Site1f_S92_L001_R1_001.fastq.gz
         32,768   0%   66.39kB/s    0:02:39       10,637,646 100%   19.81MB/s    0:00:00 (xfr#25, to-chk=83/108)
Site1f_S92_L001_R2_001.fastq.gz
         32,768   0%   62.50kB/s    0:03:06       11,684,921 100%   20.56MB/s    0:00:00 (xfr#26, to-chk=82/108)
Site1g_S93_L001_R1_001.fastq.gz
         32,768   0%   59.04kB/s    0:02:50       10,116,667 100%   17.11MB/s    0:00:00 (xfr#27, to-chk=81/108)
Site1g_S93_L001_R2_001.fastq.gz
         32,768   0%   56.74kB/s    0:03:14       11,076,204 100%   17.55MB/s    0:00:00 (xfr#28, to-chk=80/108)
Site1h_S94_L001_R1_001.fastq.gz
         32,768   0%   53.16kB/s    0:01:08        3,661,936 100%    5.71MB/s    0:00:00 (xfr#29, to-chk=79/108)
Site1h_S94_L001_R2_001.fastq.gz
         32,768   0%   52.29kB/s    0:01:14        3,944,595 100%    6.01MB/s    0:00:00 (xfr#30, to-chk=78/108)
Site1k_S100_L001_R1_001.fastq.gz
         32,768   0%   51.12kB/s    0:02:24        7,419,423 100%   10.94MB/s    0:00:00 (xfr#31, to-chk=77/108)
Site1k_S100_L001_R2_001.fastq.gz
         32,768   0%   49.46kB/s    0:02:27        7,329,144 100%   10.31MB/s    0:00:00 (xfr#32, to-chk=76/108)
Site1m_S102_L001_R1_001.fastq.gz
         32,768   0%   47.13kB/s    0:03:38       10,344,255 100%   14.05MB/s    0:00:00 (xfr#33, to-chk=75/108)
Site1m_S102_L001_R2_001.fastq.gz
         32,768   0%   45.58kB/s    0:03:58       10,925,572 100%   14.37MB/s    0:00:00 (xfr#34, to-chk=74/108)
Site1n_S103_L001_R1_001.fastq.gz
         32,768   0%   44.14kB/s    0:05:09       13,696,893 100%   16.68MB/s    0:00:00 (xfr#35, to-chk=73/108)
Site1n_S103_L001_R2_001.fastq.gz
         32,768   0%   40.87kB/s    0:06:14       15,317,828 100%   17.27MB/s    0:00:00 (xfr#36, to-chk=72/108)
Site1o_S104_L001_R1_001.fastq.gz
         32,768   0%   37.78kB/s    0:05:24       12,295,963 100%   13.51MB/s    0:00:00 (xfr#37, to-chk=71/108)
Site1o_S104_L001_R2_001.fastq.gz
         32,768   0%   36.87kB/s    0:06:08       13,599,819 100%   14.59MB/s    0:00:00 (xfr#38, to-chk=70/108)
Site1p_S105_L001_R1_001.fastq.gz
         32,768   0%   36.00kB/s    0:26:25       57,109,173 100%   54.68MB/s    0:00:00 (xfr#39, to-chk=69/108)
Site1p_S105_L001_R2_001.fastq.gz
         32,768   0%   32.10kB/s    0:30:03        1,376,256   2%    1.31MB/s    0:00:42       57,912,100 100%   49.01MB/s    0:00:01 (xfr#40, to-chk=68/108)
Site1q_S106_L001_R1_001.fastq.gz
         32,768   0%  250.00kB/s    0:00:23        6,030,256 100%   34.64MB/s    0:00:00 (xfr#41, to-chk=67/108)
Site1q_S106_L001_R2_001.fastq.gz
         32,768   0%  191.62kB/s    0:00:33        6,469,667 100%   33.35MB/s    0:00:00 (xfr#42, to-chk=66/108)
Site1r_S111_L001_R1_001.fastq.gz
         32,768   0%  172.97kB/s    0:00:50        8,838,881 100%   41.73MB/s    0:00:00 (xfr#43, to-chk=65/108)
Site1r_S111_L001_R2_001.fastq.gz
         32,768   0%  158.42kB/s    0:01:02       10,009,537 100%   43.99MB/s    0:00:00 (xfr#44, to-chk=64/108)
Site1s_S112_L001_R1_001.fastq.gz
         32,768   0%  147.47kB/s    0:01:23       12,401,363 100%   49.28MB/s    0:00:00 (xfr#45, to-chk=63/108)
Site1s_S112_L001_R2_001.fastq.gz
         32,768   0%  132.78kB/s    0:01:41       13,539,813 100%   43.48MB/s    0:00:00 (xfr#46, to-chk=62/108)
Site1t_S113_L001_R1_001.fastq.gz
         32,768   0%  107.74kB/s    0:01:48       11,735,076 100%   33.51MB/s    0:00:00 (xfr#47, to-chk=61/108)
Site1t_S113_L001_R2_001.fastq.gz
         32,768   0%   95.52kB/s    0:02:18       13,260,345 100%   27.85MB/s    0:00:00 (xfr#48, to-chk=60/108)
Site1v_S117_L001_R1_001.fastq.gz
         32,768   0%   69.87kB/s    0:02:13        9,342,851 100%   17.82MB/s    0:00:00 (xfr#49, to-chk=59/108)
Site1v_S117_L001_R2_001.fastq.gz
         32,768   0%   64.00kB/s    0:02:41       10,359,447 100%   19.07MB/s    0:00:00 (xfr#50, to-chk=58/108)
Site1w_S118_L001_R1_001.fastq.gz
         32,768   0%   61.66kB/s    0:02:41        9,964,110 100%   17.60MB/s    0:00:00 (xfr#51, to-chk=57/108)
Site1w_S118_L001_R2_001.fastq.gz
         32,768   0%   59.26kB/s    0:03:06       11,075,680 100%   17.40MB/s    0:00:00 (xfr#52, to-chk=56/108)
Site1y_S120_L001_R1_001.fastq.gz
         32,768   0%   52.72kB/s    0:02:43        8,648,625 100%   12.93MB/s    0:00:00 (xfr#53, to-chk=55/108)
Site1y_S120_L001_R2_001.fastq.gz
         32,768   0%   50.16kB/s    0:03:13        9,735,967 100%   14.13MB/s    0:00:00 (xfr#54, to-chk=54/108)
Site1z_S129_L001_R1_001.fastq.gz
         32,768   0%   48.71kB/s    0:02:21        6,906,228 100%    9.82MB/s    0:00:00 (xfr#55, to-chk=53/108)
Site1z_S129_L001_R2_001.fastq.gz
         32,768   0%   47.69kB/s    0:02:38        7,594,675 100%   10.57MB/s    0:00:00 (xfr#56, to-chk=52/108)
Site5J_S107_L001_R1_001.fastq.gz
          3,152 100%    4.49kB/s    0:00:00            3,152 100%    4.49kB/s    0:00:00 (xfr#57, to-chk=51/108)
Site5J_S107_L001_R2_001.fastq.gz
          3,267 100%    4.66kB/s    0:00:00            3,267 100%    4.66kB/s    0:00:00 (xfr#58, to-chk=50/108)
Site5L_S109_L001_R1_001.fastq.gz
         32,768   0%   46.72kB/s    0:03:46       10,591,641 100%   14.43MB/s    0:00:00 (xfr#59, to-chk=49/108)
Site5L_S109_L001_R2_001.fastq.gz
         32,768   0%   45.71kB/s    0:04:29       12,335,179 100%   16.25MB/s    0:00:00 (xfr#60, to-chk=48/108)
Site5a_S87_L001_R1_001.fastq.gz
         32,768   0%   44.20kB/s    0:02:09        5,736,372 100%    7.41MB/s    0:00:00 (xfr#61, to-chk=47/108)
Site5a_S87_L001_R2_001.fastq.gz
         32,768   0%   43.36kB/s    0:02:35        6,756,033 100%    8.63MB/s    0:00:00 (xfr#62, to-chk=46/108)
Site5aa_S132_L001_R1_001.fastq.gz
         32,768   0%   42.78kB/s    0:06:43       17,275,560 100%   21.07MB/s    0:00:00 (xfr#63, to-chk=45/108)
Site5aa_S132_L001_R2_001.fastq.gz
         32,768   0%   40.92kB/s    0:07:38       18,803,462 100%   21.66MB/s    0:00:00 (xfr#64, to-chk=44/108)
Site5b_S88_L001_R1_001.fastq.gz
         32,768   0%   38.60kB/s    0:03:54        9,073,525 100%   10.25MB/s    0:00:00 (xfr#65, to-chk=43/108)
Site5b_S88_L001_R2_001.fastq.gz
         32,768   0%   37.91kB/s    0:04:42       10,726,254 100%   11.88MB/s    0:00:00 (xfr#66, to-chk=42/108)
Site5bb_S133_L001_R1_001.fastq.gz
         32,768   0%   37.12kB/s    0:09:28       21,123,332 100%   22.36MB/s    0:00:00 (xfr#67, to-chk=41/108)
Site5bb_S133_L001_R2_001.fastq.gz
         32,768   0%   35.52kB/s    0:11:02       23,551,318 100%   23.67MB/s    0:00:00 (xfr#68, to-chk=40/108)
Site5c_S89_L001_R1_001.fastq.gz
         32,768   0%   33.72kB/s    0:01:52        3,809,576 100%    3.76MB/s    0:00:00 (xfr#69, to-chk=39/108)
Site5c_S89_L001_R2_001.fastq.gz
         32,768   0%   33.09kB/s    0:02:01        4,045,650 100%    3.91MB/s    0:00:00 (xfr#70, to-chk=38/108)
Site5cc_S134_L001_R1_001.fastq.gz
         32,768   0%   32.39kB/s    0:11:49        2,195,456   9%    2.09MB/s    0:00:09       23,024,076 100%   20.81MB/s    0:00:01 (xfr#71, to-chk=37/108)
Site5cc_S134_L001_R2_001.fastq.gz
         32,768   0%  581.82kB/s    0:00:47       27,433,098 100%  261.62MB/s    0:00:00 (xfr#72, to-chk=36/108)
Site5e_S95_L001_R1_001.fastq.gz
         32,768   0%  316.83kB/s    0:00:39       12,623,394 100%   91.90MB/s    0:00:00 (xfr#73, to-chk=35/108)
Site5e_S95_L001_R2_001.fastq.gz
         32,768   0%  244.27kB/s    0:00:55       13,593,516 100%   83.64MB/s    0:00:00 (xfr#74, to-chk=34/108)
Site5ee_S136_L001_R1_001.fastq.gz
         32,768   0%  205.13kB/s    0:01:10       14,409,417 100%   73.49MB/s    0:00:00 (xfr#75, to-chk=33/108)
Site5ee_S136_L001_R2_001.fastq.gz
         32,768   0%  171.12kB/s    0:01:34       16,177,439 100%   67.08MB/s    0:00:00 (xfr#76, to-chk=32/108)
Site5f_S96_L001_R1_001.fastq.gz
         32,768   4%  139.13kB/s    0:00:04          687,567 100%    2.83MB/s    0:00:00 (xfr#77, to-chk=31/108)
Site5f_S96_L001_R2_001.fastq.gz
         32,768   5%  137.93kB/s    0:00:04          586,382 100%    2.39MB/s    0:00:00 (xfr#78, to-chk=30/108)
Site5ff_S137_L001_R1_001.fastq.gz
         32,768   0%  136.75kB/s    0:03:08       25,791,900 100%   84.53MB/s    0:00:00 (xfr#79, to-chk=29/108)
Site5ff_S137_L001_R2_001.fastq.gz
         32,768   0%  109.97kB/s    0:04:22       28,940,772 100%   76.67MB/s    0:00:00 (xfr#80, to-chk=28/108)
Site5g_S97_L001_R1_001.fastq.gz
         32,768   0%   88.89kB/s    0:01:43        9,192,625 100%   23.01MB/s    0:00:00 (xfr#81, to-chk=27/108)
Site5g_S97_L001_R2_001.fastq.gz
         32,768   0%   83.99kB/s    0:02:01       10,215,095 100%   24.35MB/s    0:00:00 (xfr#82, to-chk=26/108)
Site5h_S98_L001_R1_001.fastq.gz
         32,768   0%   80.00kB/s    0:02:36       12,583,460 100%   27.97MB/s    0:00:00 (xfr#83, to-chk=25/108)
Site5h_S98_L001_R2_001.fastq.gz
         32,768   0%   74.59kB/s    0:03:00       13,492,257 100%   28.22MB/s    0:00:00 (xfr#84, to-chk=24/108)
Site5k_S108_L001_R1_001.fastq.gz
         32,768   0%   70.18kB/s    0:00:49        3,490,162 100%    7.19MB/s    0:00:00 (xfr#85, to-chk=23/108)
Site5k_S108_L001_R2_001.fastq.gz
         32,768   0%   68.97kB/s    0:01:03        4,381,885 100%    8.85MB/s    0:00:00 (xfr#86, to-chk=22/108)
Site5m_S110_L001_R1_001.fastq.gz
         32,768   0%   67.80kB/s    0:01:33        6,394,434 100%   12.60MB/s    0:00:00 (xfr#87, to-chk=21/108)
Site5m_S110_L001_R2_001.fastq.gz
         32,768   0%   65.98kB/s    0:01:50        7,315,926 100%   13.95MB/s    0:00:00 (xfr#88, to-chk=20/108)
Site5n_S115_L001_R1_001.fastq.gz
         32,768   0%   64.00kB/s    0:01:38        6,337,613 100%   11.80MB/s    0:00:00 (xfr#89, to-chk=19/108)
Site5n_S115_L001_R2_001.fastq.gz
         32,768   0%   62.50kB/s    0:01:49        6,878,872 100%   12.42MB/s    0:00:00 (xfr#90, to-chk=18/108)
Site5o_S116_L001_R1_001.fastq.gz
         32,768   0%   60.61kB/s    0:01:39        6,076,898 100%   10.71MB/s    0:00:00 (xfr#91, to-chk=17/108)
Site5o_S116_L001_R2_001.fastq.gz
         32,768   0%   59.15kB/s    0:01:51        6,648,513 100%   11.40MB/s    0:00:00 (xfr#92, to-chk=16/108)
Site5p_S121_L001_R1_001.fastq.gz
         32,768   0%   57.45kB/s    0:02:19        8,050,112 100%   13.24MB/s    0:00:00 (xfr#93, to-chk=15/108)
Site5p_S121_L001_R2_001.fastq.gz
         32,768   0%   55.08kB/s    0:02:45        9,138,963 100%   14.41MB/s    0:00:00 (xfr#94, to-chk=14/108)
Site5q_S122_L001_R1_001.fastq.gz
         32,768   0%   52.89kB/s    0:04:56       15,700,699 100%   23.25MB/s    0:00:00 (xfr#95, to-chk=13/108)
Site5q_S122_L001_R2_001.fastq.gz
         32,768   0%   49.69kB/s    0:05:23       16,118,120 100%   22.77MB/s    0:00:00 (xfr#96, to-chk=12/108)
Site5r_S123_L001_R1_001.fastq.gz
         32,768   0%   47.41kB/s    0:03:24        9,712,950 100%   13.33MB/s    0:00:00 (xfr#97, to-chk=11/108)
Site5r_S123_L001_R2_001.fastq.gz
         32,768   0%   46.04kB/s    0:03:55       10,897,039 100%   14.51MB/s    0:00:00 (xfr#98, to-chk=10/108)
Site5s_S124_L001_R1_001.fastq.gz
         32,768   0%   44.69kB/s    0:02:43        7,340,512 100%    9.56MB/s    0:00:00 (xfr#99, to-chk=9/108)
Site5s_S124_L001_R2_001.fastq.gz
         32,768   0%   43.72kB/s    0:03:19        8,732,252 100%   11.12MB/s    0:00:00 (xfr#100, to-chk=8/108)
Site5u_S126_L001_R1_001.fastq.gz
         32,768   0%   42.72kB/s    0:01:57        5,033,056 100%    6.32MB/s    0:00:00 (xfr#101, to-chk=7/108)
Site5u_S126_L001_R2_001.fastq.gz
         32,768   0%   42.16kB/s    0:02:11        5,584,122 100%    6.90MB/s    0:00:00 (xfr#102, to-chk=6/108)
Site5v_S127_L001_R1_001.fastq.gz
         32,768   0%   41.45kB/s    0:01:43        4,309,446 100%    5.26MB/s    0:00:00 (xfr#103, to-chk=5/108)
Site5v_S127_L001_R2_001.fastq.gz
         32,768   0%   40.97kB/s    0:01:56        4,795,986 100%    5.78MB/s    0:00:00 (xfr#104, to-chk=4/108)
Site5w_S128_L001_R1_001.fastq.gz
          1,181 100%    1.46kB/s    0:00:00            1,181 100%    1.46kB/s    0:00:00 (xfr#105, to-chk=3/108)
Site5w_S128_L001_R2_001.fastq.gz
          1,265 100%    1.56kB/s    0:00:00            1,265 100%    1.56kB/s    0:00:00 (xfr#106, to-chk=2/108)
Site5z_S131_L001_R1_001.fastq.gz
         32,768   0%   40.46kB/s    0:14:08       34,377,141 100%   37.64MB/s    0:00:00 (xfr#107, to-chk=1/108)
Site5z_S131_L001_R2_001.fastq.gz
         32,768   0%   36.74kB/s    0:17:11       37,920,700 100%   37.55MB/s    0:00:00 (xfr#108, to-chk=0/108)

sent 1,198,592,058 bytes  received 2,068 bytes  342,455,464.57 bytes/sec
total size is 1,198,291,484  speedup is 1.00
total 2.7G
-rw-rw-r-- 1 sam sam 406M Apr 24 07:43 demux-paired-end-filtered.qza
-rw-rw-r-- 1 sam sam  17K Apr 24 07:43 demux-paired-end-filter-stats.qza
-rw-rw-r-- 1 sam sam 1.2G Apr 24 07:23 demux-paired-end.qza
-rw-rw-r-- 1 sam sam 9.7M Apr 22 16:49 Site1a_S83_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  11M Apr 22 16:49 Site1a_S83_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 7.0M Apr 22 16:49 Site1bb_S139_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 7.5M Apr 22 16:49 Site1bb_S139_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 9.2M Apr 22 16:50 Site1b_S84_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  10M Apr 22 16:50 Site1b_S84_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  17M Apr 22 16:50 Site1cc_S140_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  19M Apr 22 16:50 Site1cc_S140_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 6.8M Apr 22 16:51 Site1c_S85_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 7.4M Apr 22 16:51 Site1c_S85_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  19M Apr 22 16:51 Site1dd_S141_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  19M Apr 22 16:51 Site1dd_S141_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 9.5M Apr 22 16:52 Site1d_S86_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  11M Apr 22 16:52 Site1d_S86_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  12M Apr 22 16:52 Site1ee_S142_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  11M Apr 22 16:52 Site1ee_S142_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 1.6M Apr 22 16:52 Site1e_S91_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 1.7M Apr 22 16:52 Site1e_S91_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  11M Apr 22 16:53 Site1f_S92_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  12M Apr 22 16:53 Site1f_S92_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 9.7M Apr 22 16:53 Site1g_S93_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  11M Apr 22 16:53 Site1g_S93_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 3.5M Apr 22 16:54 Site1h_S94_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 3.8M Apr 22 16:54 Site1h_S94_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 1.1M Apr 22 16:54 Site1J_S99_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 851K Apr 22 16:54 Site1J_S99_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 7.1M Apr 22 16:54 Site1k_S100_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 7.0M Apr 22 16:54 Site1k_S100_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 2.6M Apr 22 16:54 Site1L_S101_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 3.0M Apr 22 16:54 Site1L_S101_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 9.9M Apr 22 16:54 Site1m_S102_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  11M Apr 22 16:54 Site1m_S102_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  14M Apr 22 16:55 Site1n_S103_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  15M Apr 22 16:55 Site1n_S103_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  12M Apr 22 16:56 Site1o_S104_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  13M Apr 22 16:56 Site1o_S104_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  55M Apr 22 16:56 Site1p_S105_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  56M Apr 22 16:56 Site1p_S105_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 5.8M Apr 22 16:57 Site1q_S106_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 6.2M Apr 22 16:57 Site1q_S106_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 8.5M Apr 22 16:57 Site1r_S111_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 9.6M Apr 22 16:57 Site1r_S111_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  12M Apr 22 16:58 Site1s_S112_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  13M Apr 22 16:58 Site1s_S112_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  12M Apr 22 16:58 Site1t_S113_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  13M Apr 22 16:58 Site1t_S113_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 3.2M Apr 22 16:58 Site1U_S114_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 3.6M Apr 22 16:58 Site1U_S114_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 9.0M Apr 22 16:59 Site1v_S117_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 9.9M Apr 22 16:59 Site1v_S117_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 9.6M Apr 22 16:59 Site1w_S118_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  11M Apr 22 16:59 Site1w_S118_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 8.3M Apr 22 17:00 Site1y_S120_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 9.3M Apr 22 17:00 Site1y_S120_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 6.6M Apr 22 17:00 Site1z_S129_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 7.3M Apr 22 17:00 Site1z_S129_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  17M Apr 22 17:01 Site5aa_S132_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  18M Apr 22 17:01 Site5aa_S132_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 5.5M Apr 22 17:01 Site5a_S87_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 6.5M Apr 22 17:01 Site5a_S87_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  21M Apr 22 17:02 Site5bb_S133_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  23M Apr 22 17:02 Site5bb_S133_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 8.7M Apr 22 17:02 Site5b_S88_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  11M Apr 22 17:02 Site5b_S88_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  22M Apr 22 17:03 Site5cc_S134_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  27M Apr 22 17:03 Site5cc_S134_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 3.7M Apr 22 17:03 Site5c_S89_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 3.9M Apr 22 17:03 Site5c_S89_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  14M Apr 22 17:03 Site5ee_S136_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  16M Apr 22 17:03 Site5ee_S136_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  13M Apr 22 17:04 Site5e_S95_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  13M Apr 22 17:04 Site5e_S95_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  25M Apr 22 17:04 Site5ff_S137_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  28M Apr 22 17:04 Site5ff_S137_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 672K Apr 22 17:04 Site5f_S96_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 573K Apr 22 17:04 Site5f_S96_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 8.8M Apr 22 17:05 Site5g_S97_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 9.8M Apr 22 17:05 Site5g_S97_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  13M Apr 22 17:05 Site5h_S98_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  13M Apr 22 17:05 Site5h_S98_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 3.1K Apr 22 17:05 Site5J_S107_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 3.2K Apr 22 17:05 Site5J_S107_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 3.4M Apr 22 17:05 Site5k_S108_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 4.2M Apr 22 17:05 Site5k_S108_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  11M Apr 22 17:06 Site5L_S109_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  12M Apr 22 17:06 Site5L_S109_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 6.1M Apr 22 17:06 Site5m_S110_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 7.0M Apr 22 17:06 Site5m_S110_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 6.1M Apr 22 17:06 Site5n_S115_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 6.6M Apr 22 17:06 Site5n_S115_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 5.8M Apr 22 17:07 Site5o_S116_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 6.4M Apr 22 17:07 Site5o_S116_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 7.7M Apr 22 17:07 Site5p_S121_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 8.8M Apr 22 17:07 Site5p_S121_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  15M Apr 22 17:08 Site5q_S122_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  16M Apr 22 17:08 Site5q_S122_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 9.3M Apr 22 17:08 Site5r_S123_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  11M Apr 22 17:08 Site5r_S123_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 7.1M Apr 22 17:08 Site5s_S124_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 8.4M Apr 22 17:08 Site5s_S124_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 4.8M Apr 22 17:09 Site5u_S126_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 5.4M Apr 22 17:09 Site5u_S126_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 4.2M Apr 22 17:09 Site5v_S127_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 4.6M Apr 22 17:09 Site5v_S127_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam 1.2K Apr 22 17:09 Site5w_S128_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam 1.3K Apr 22 17:09 Site5w_S128_L001_R2_001.fastq.gz
-rw-rw-r-- 1 sam sam  33M Apr 22 17:09 Site5z_S131_L001_R1_001.fastq.gz
-rw-rw-r-- 1 sam sam  37M Apr 22 17:09 Site5z_S131_L001_R2_001.fastq.gz</code></pre>
</section>
<section id="import-to-qiime2-qza-format" class="level2">
<h2 class="anchored" data-anchor-id="import-to-qiime2-qza-format">4.2 Import to QIIME2 QZA format</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools import <span class="dt">\</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--type</span> <span class="st">'SampleData[PairedEndSequencesWithQuality]'</span> <span class="dt">\</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--input-path</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--input-format</span> CasavaOneEightSingleLanePerSampleDirFmt <span class="dt">\</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output-path</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/demux-paired-end.qza</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove trimmed reads</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/<span class="pp">*</span>.fastq.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>There was a problem importing /home/sam/gitrepos/RobertsLab/project-eDNA-yellow/output/02.00-qiime2:

  Unrecognized file (/home/sam/gitrepos/RobertsLab/project-eDNA-yellow/output/02.00-qiime2/demux-paired-end-filtered.qza) for CasavaOneEightSingleLanePerSampleDirFmt.</code></pre>
</section>
</section>
<section id="deblur" class="level1">
<h1>5 Deblur</h1>
<section id="quality-filter" class="level2">
<h2 class="anchored" data-anchor-id="quality-filter">5.1 Quality filter</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> quality-filter q-score <span class="dt">\</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a> <span class="at">--i-demux</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/demux-paired-end.qza <span class="dt">\</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a> <span class="at">--o-filtered-sequences</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/demux-paired-end-filtered.qza <span class="dt">\</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a> <span class="at">--o-filter-stats</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/demux-paired-end-filter-stats.qza</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Saved SampleData[SequencesWithQuality] to: /home/sam/gitrepos/RobertsLab/project-eDNA-yellow/output/02.00-qiime2/demux-paired-end-filtered.qza
Saved QualityFilterStats to: /home/sam/gitrepos/RobertsLab/project-eDNA-yellow/output/02.00-qiime2/demux-paired-end-filter-stats.qza</code></pre>
</section>
<section id="denoise" class="level2">
<h2 class="anchored" data-anchor-id="denoise">5.2 Denoise</h2>
<p>Trim length is based on FastQC results, where most samples show sharp drop in quality scores.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> deblur denoise-other <span class="dt">\</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-demultiplexed-seqs</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/demux-paired-end-filtered.qza <span class="dt">\</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-reference-seqs</span> <span class="st">"</span><span class="va">${silva_ref_seqs}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--p-trim-length</span> <span class="st">"</span><span class="va">${deblur_ptrim_length}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-representative-sequences</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/rep-seqs-deblur.qza <span class="dt">\</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-table</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/table-deblur.qza <span class="dt">\</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">--p-sample-stats</span> <span class="dt">\</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">--p-jobs-to-start</span> <span class="st">"</span><span class="va">${threads}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-stats</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/deblur-stats.qza</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Saved FeatureTable[Frequency] to: /home/sam/gitrepos/RobertsLab/project-eDNA-yellow/output/02.00-qiime2/table-deblur.qza
Saved FeatureData[Sequence] to: /home/sam/gitrepos/RobertsLab/project-eDNA-yellow/output/02.00-qiime2/rep-seqs-deblur.qza
Saved DeblurStats to: /home/sam/gitrepos/RobertsLab/project-eDNA-yellow/output/02.00-qiime2/deblur-stats.qza</code></pre>
</section>
<section id="deblur-visualization" class="level2">
<h2 class="anchored" data-anchor-id="deblur-visualization">5.3 Deblur visualization</h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> deblur visualize-stats <span class="dt">\</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-deblur-stats</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/deblur-stats.qza <span class="dt">\</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/deblur-stats.qzv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Saved Visualization to: /home/sam/gitrepos/RobertsLab/project-eDNA-yellow/output/02.00-qiime2/deblur-stats.qzv</code></pre>
</section>
</section>
<section id="feature-table" class="level1">
<h1>6 Feature Table</h1>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-table summarize <span class="dt">\</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-table</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/table-deblur.qza <span class="dt">\</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/table.qzv <span class="dt">\</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--m-sample-metadata-file</span> <span class="st">"</span><span class="va">${metadata}</span><span class="st">"</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-table tabulate-seqs <span class="dt">\</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-data</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/rep-seqs-deblur.qza <span class="dt">\</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/rep-seqs-deblur.qzv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Saved Visualization to: /home/sam/gitrepos/RobertsLab/project-eDNA-yellow/output/02.00-qiime2/table.qzv
Saved Visualization to: /home/sam/gitrepos/RobertsLab/project-eDNA-yellow/output/02.00-qiime2/rep-seqs-deblur.qzv</code></pre>
</section>
<section id="taxonomy" class="level1">
<h1>7 Taxonomy</h1>
<section id="classification" class="level2">
<h2 class="anchored" data-anchor-id="classification">7.1 Classification</h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier classify-sklearn <span class="dt">\</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-classifier</span> <span class="st">"</span><span class="va">${silva_classifier}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-reads</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/rep-seqs-deblur.qza <span class="dt">\</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-classification</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/taxonomy.qza</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> metadata tabulate <span class="dt">\</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">--m-input-file</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/taxonomy.qza <span class="dt">\</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/taxonomy.qzv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Saved FeatureData[Taxonomy] to: /home/sam/gitrepos/RobertsLab/project-eDNA-yellow/output/02.00-qiime2/taxonomy.qza
Saved Visualization to: /home/sam/gitrepos/RobertsLab/project-eDNA-yellow/output/02.00-qiime2/taxonomy.qzv</code></pre>
</section>
<section id="bar-plots" class="level2">
<h2 class="anchored" data-anchor-id="bar-plots">7.2 Bar plots</h2>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> taxa barplot <span class="dt">\</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-table</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/table-deblur.qza <span class="dt">\</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--i-taxonomy</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/taxonomy.qza <span class="dt">\</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> <span class="st">"</span><span class="va">${metadata}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--o-visualization</span> <span class="st">"</span><span class="va">${output_dir}</span><span class="st">"</span>/taxa-bar-plots.qzv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Saved Visualization to: /home/sam/gitrepos/RobertsLab/project-eDNA-yellow/output/02.00-qiime2/taxa-bar-plots.qzv</code></pre>
</section>
</section>
<section id="results" class="level1">
<h1>RESULTS</h1>
<p>Taxonomic assignments are available in this QIIME 2 visualization file:</p>
<ul>
<li><a href="https://github.com/RobertsLab/project-eDNA-yellow/raw/516fd3772577550c1d8f991b116fe1f1081a0cec/output/02.00-qiime2/taxa-bar-plots.qzv"><code>taxa-bar-plots.qzv</code></a></li>
</ul>
<p>The resulting taxonomic output file is a QIIME 2 bar plot visualization. This can be downloaded (either by using the link below or by cloning this project’s repo) and then uploaded to the <a href="https://view.qiime2.org/">QIIME 2 Viewer website</a>. This provides interactive viewing/filtering/sorting of taxonomic assignments.</p>
<p>If one wants the actual tables used to generate the interactive visualization, take the following steps:</p>
<ol type="1">
<li><p>Rename <code>taxa-bar-plots.qzv</code> to <code>taxa-bar-plots.zip</code> (QZV files are simply a ZIP file).</p></li>
<li><p>Navigate to the <code>66fd6b7b-8685-46ee-995c-3cd1e139f288/data</code> directory.</p></li>
<li><p>Open any of the <code>level-*.csv</code> files. Each level corresponds to a taxonomic level. E.g. Level 1 corresponds to Kingdom, Level 2 corresponds to Phylum, etc down to Level 7, which is Species.</p></li>
</ol>
<p>These CSVs contain counts assigned at each specific level of taxonomy for each Yellow Island site.</p>
<p>When viewed at the Family level of classifcation, the bulk of the classifications fall into two groups/levels:</p>
<ul>
<li><p>Kingdom: Eukaryotes (light purple bars in plot below)</p></li>
<li><p>Family: Capitellida (light green bars in plot below).</p></li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./20250424-eDNA-taxonomic-bar-plots-01.png" class="img-fluid figure-img" alt="Screenshot showing taxonomic distribution across sites. The bulk of the classifications fall into two groups: Kingdom Eukaryotes (light purple bars) and Family Capitellida (light green bars)."></p>
<figcaption>Screenshot showing taxonomic distribution across sites. The bulk of the classifications fall into two groups: Kingdom Eukaryotes (light purple bars) and Family Capitellida (light green bars).</figcaption>
</figure>
</div>
<p>Additionaly analyses is needed to determine if the counts differ significantly between sites. Also, it would be interesting to somehow overlay these with data on currents flowing around Yellow Island.</p>
</section>
<section id="references" class="level1">
<h1>REFERENCES</h1>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-bolyen2019" class="csl-entry" role="listitem">
Bolyen, Evan, Jai Ram Rideout, Matthew R. Dillon, Nicholas A. Bokulich, Christian C. Abnet, Gabriel A. Al-Ghalith, Harriet Alexander, et al. 2019. “Reproducible, Interactive, Scalable and Extensible Microbiome Data Science Using QIIME 2.” <em>Nature Biotechnology</em> 37 (8): 852–57. <a href="https://doi.org/10.1038/s41587-019-0209-9" class="uri">https://doi.org/10.1038/s41587-019-0209-9</a>.
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/robertslab\.github\.io\/sams-notebook");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>