<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sam White">
<meta name="dcterms.date" content="2024-12-20">

<title>qPCR Analysis - C.gigas Lifestages Carryover – Sam’s Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark-b53751a350365c71b6c909e95f209ed1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-e518e50a9807db09b51ea894c4c266f8.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 2,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 100,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-7483T61T11"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-7483T61T11', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Sam’s Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" target="_blank">
 <span class="dropdown-text">Rstudio Server - Raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu/Atumefaciens/" target="_blank">
 <span class="dropdown-text">My Gannet Directory</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" target="_blank">
 <span class="dropdown-text">Steven’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shedurkin.github.io/Roberts-LabNotebook/" target="_blank">
 <span class="dropdown-text">Kathleen’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://meganewing.github.io/mewing-notebook/" target="_blank">
 <span class="dropdown-text">Megan’s Notebook</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/sams-notebook" target="_blank">
 <span class="dropdown-text">Notebook repo</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" target="_blank">
 <span class="dropdown-text">Lab Organization (GitHub)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" target="_blank">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" target="_blank"> <i class="bi bi-calendar-week" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-right">
      <h1 class="title">qPCR Analysis - C.gigas Lifestages Carryover</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2024</div>
                <div class="quarto-category">Crassostrea gigas</div>
                <div class="quarto-category">Pacific oyster</div>
                <div class="quarto-category">qPCR</div>
                <div class="quarto-category">project-gigas-carryover</div>
                <div class="quarto-category">lifestage_carryover</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-right">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Sam White </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">December 20, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#intro" id="toc-intro" class="nav-link active" data-scroll-target="#intro">INTRO</a></li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary">SUMMARY</a>
  <ul>
  <li><a href="#t-tests-delta-cq" id="toc-t-tests-delta-cq" class="nav-link" data-scroll-target="#t-tests-delta-cq">t-tests (Delta Cq)</a>
  <ul>
  <li><a href="#seed-vs.-adult" id="toc-seed-vs.-adult" class="nav-link" data-scroll-target="#seed-vs.-adult">Seed vs.&nbsp;Adult</a></li>
  <li><a href="#seed-vs.-juvenile" id="toc-seed-vs.-juvenile" class="nav-link" data-scroll-target="#seed-vs.-juvenile">Seed vs.&nbsp;Juvenile</a></li>
  <li><a href="#seed-vs.-spat" id="toc-seed-vs.-spat" class="nav-link" data-scroll-target="#seed-vs.-spat">Seed vs.&nbsp;Spat</a></li>
  <li><a href="#adult-vs.-juvenile" id="toc-adult-vs.-juvenile" class="nav-link" data-scroll-target="#adult-vs.-juvenile">Adult vs.&nbsp;Juvenile</a></li>
  <li><a href="#adult-vs.-spat" id="toc-adult-vs.-spat" class="nav-link" data-scroll-target="#adult-vs.-spat">Adult vs.&nbsp;Spat</a></li>
  <li><a href="#juvenile-vs.-spat" id="toc-juvenile-vs.-spat" class="nav-link" data-scroll-target="#juvenile-vs.-spat">Juvenile vs.&nbsp;Spat</a></li>
  <li><a href="#seed---conditioning-treated-vs.-control" id="toc-seed---conditioning-treated-vs.-control" class="nav-link" data-scroll-target="#seed---conditioning-treated-vs.-control">Seed - Conditioning Treated vs.&nbsp;Control</a></li>
  <li><a href="#adult---conditioning-treated-vs.-control" id="toc-adult---conditioning-treated-vs.-control" class="nav-link" data-scroll-target="#adult---conditioning-treated-vs.-control">Adult - Conditioning Treated vs.&nbsp;Control</a></li>
  <li><a href="#juvenile---conditioning-treated-vs.-control" id="toc-juvenile---conditioning-treated-vs.-control" class="nav-link" data-scroll-target="#juvenile---conditioning-treated-vs.-control">Juvenile - Conditioning Treated vs.&nbsp;Control</a></li>
  <li><a href="#spat---conditioning-treated-vs.-control" id="toc-spat---conditioning-treated-vs.-control" class="nav-link" data-scroll-target="#spat---conditioning-treated-vs.-control">Spat - Conditioning Treated vs.&nbsp;Control</a></li>
  <li><a href="#adult---acute-ambient-vs.-high" id="toc-adult---acute-ambient-vs.-high" class="nav-link" data-scroll-target="#adult---acute-ambient-vs.-high">Adult - Acute Ambient vs.&nbsp;High</a></li>
  <li><a href="#juvenile---acute-ambient-vs.-high" id="toc-juvenile---acute-ambient-vs.-high" class="nav-link" data-scroll-target="#juvenile---acute-ambient-vs.-high">Juvenile - Acute Ambient vs.&nbsp;High</a></li>
  <li><a href="#adult---conditioning-treated-acute-ambient-vs.-high" id="toc-adult---conditioning-treated-acute-ambient-vs.-high" class="nav-link" data-scroll-target="#adult---conditioning-treated-acute-ambient-vs.-high">Adult - Conditioning Treated: Acute Ambient vs.&nbsp;High</a></li>
  <li><a href="#adult---conditioning-control-acute-ambient-vs.-high" id="toc-adult---conditioning-control-acute-ambient-vs.-high" class="nav-link" data-scroll-target="#adult---conditioning-control-acute-ambient-vs.-high">Adult - Conditioning Control: Acute Ambient vs.&nbsp;High</a></li>
  <li><a href="#juvenile---conditioning-treated-acute-ambient-vs.-high" id="toc-juvenile---conditioning-treated-acute-ambient-vs.-high" class="nav-link" data-scroll-target="#juvenile---conditioning-treated-acute-ambient-vs.-high">Juvenile - Conditioning Treated: Acute Ambient vs.&nbsp;High</a></li>
  <li><a href="#juvenile---conditioning-control-acute-ambient-vs.-high" id="toc-juvenile---conditioning-control-acute-ambient-vs.-high" class="nav-link" data-scroll-target="#juvenile---conditioning-control-acute-ambient-vs.-high">Juvenile - Conditioning Control: Acute Ambient vs.&nbsp;High</a></li>
  </ul></li>
  <li><a href="#gene-expression-delta-delta-cq-or-fold-change" id="toc-gene-expression-delta-delta-cq-or-fold-change" class="nav-link" data-scroll-target="#gene-expression-delta-delta-cq-or-fold-change">Gene Expression (delta delta Cq or fold change)</a>
  <ul>
  <li><a href="#adult---conditioning-treated-acute-high-vs.-ambient" id="toc-adult---conditioning-treated-acute-high-vs.-ambient" class="nav-link" data-scroll-target="#adult---conditioning-treated-acute-high-vs.-ambient">Adult - Conditioning Treated: Acute High vs.&nbsp;Ambient</a></li>
  <li><a href="#juvenile---conditioning-treated-acute-high-vs.-ambient" id="toc-juvenile---conditioning-treated-acute-high-vs.-ambient" class="nav-link" data-scroll-target="#juvenile---conditioning-treated-acute-high-vs.-ambient">Juvenile - Conditioning Treated: Acute High vs.&nbsp;Ambient</a></li>
  <li><a href="#adult---adult-vs.-seed" id="toc-adult---adult-vs.-seed" class="nav-link" data-scroll-target="#adult---adult-vs.-seed">Adult - Adult vs.&nbsp;Seed</a></li>
  <li><a href="#adult---adult-vs.-spat" id="toc-adult---adult-vs.-spat" class="nav-link" data-scroll-target="#adult---adult-vs.-spat">Adult - Adult vs.&nbsp;Spat</a></li>
  <li><a href="#adult---spat-vs.-seed" id="toc-adult---spat-vs.-seed" class="nav-link" data-scroll-target="#adult---spat-vs.-seed">Adult - Spat vs.&nbsp;Seed</a></li>
  <li><a href="#conditioning-by-target-and-lifestage" id="toc-conditioning-by-target-and-lifestage" class="nav-link" data-scroll-target="#conditioning-by-target-and-lifestage">Conditioning by Target and Lifestage</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#description" id="toc-description" class="nav-link" data-scroll-target="#description">1 Description</a></li>
  <li><a href="#set-variables" id="toc-set-variables" class="nav-link" data-scroll-target="#set-variables">2 Set variables</a>
  <ul>
  <li><a href="#set-sample-groups" id="toc-set-sample-groups" class="nav-link" data-scroll-target="#set-sample-groups">2.1 Set sample groups</a></li>
  <li><a href="#assign-groups-to-list" id="toc-assign-groups-to-list" class="nav-link" data-scroll-target="#assign-groups-to-list">2.2 Assign groups to list</a></li>
  </ul></li>
  <li><a href="#functions" id="toc-functions" class="nav-link" data-scroll-target="#functions">3 Functions</a>
  <ul>
  <li><a href="#calculate-delta-cq" id="toc-calculate-delta-cq" class="nav-link" data-scroll-target="#calculate-delta-cq">3.1 Calculate delta Cq</a></li>
  <li><a href="#create-delta-cq-boxplots" id="toc-create-delta-cq-boxplots" class="nav-link" data-scroll-target="#create-delta-cq-boxplots">3.2 Create delta Cq boxplots</a>
  <ul>
  <li><a href="#lifestage-comparison" id="toc-lifestage-comparison" class="nav-link" data-scroll-target="#lifestage-comparison">3.2.1 Lifestage comparison</a></li>
  <li><a href="#conditioning-comparisons" id="toc-conditioning-comparisons" class="nav-link" data-scroll-target="#conditioning-comparisons">3.2.2 Conditioning comparisons</a></li>
  <li><a href="#acute-comparisons" id="toc-acute-comparisons" class="nav-link" data-scroll-target="#acute-comparisons">3.2.3 Acute comparisons</a></li>
  <li><a href="#acute-treatements-within-life-stage-conditioning" id="toc-acute-treatements-within-life-stage-conditioning" class="nav-link" data-scroll-target="#acute-treatements-within-life-stage-conditioning">3.2.4 Acute treatements within life stage conditioning</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#read-in-files" id="toc-read-in-files" class="nav-link" data-scroll-target="#read-in-files">4 Read in files</a></li>
  <li><a href="#clean-data" id="toc-clean-data" class="nav-link" data-scroll-target="#clean-data">5 Clean data</a>
  <ul>
  <li><a href="#replace-target-names" id="toc-replace-target-names" class="nav-link" data-scroll-target="#replace-target-names">5.1 Replace target names</a></li>
  <li><a href="#identify-samples-with-cq.std..dev-0.5" id="toc-identify-samples-with-cq.std..dev-0.5" class="nav-link" data-scroll-target="#identify-samples-with-cq.std..dev-0.5">5.2 Identify Samples with Cq.Std..Dev &gt; 0.5</a></li>
  <li><a href="#remove-bad-technical-reps" id="toc-remove-bad-technical-reps" class="nav-link" data-scroll-target="#remove-bad-technical-reps">5.3 Remove bad technical reps</a></li>
  </ul></li>
  <li><a href="#group-samples-by-target" id="toc-group-samples-by-target" class="nav-link" data-scroll-target="#group-samples-by-target">6 Group samples by target</a></li>
  <li><a href="#add-life-stage-and-treatment-cols" id="toc-add-life-stage-and-treatment-cols" class="nav-link" data-scroll-target="#add-life-stage-and-treatment-cols">7 Add life stage and treatment cols</a></li>
  <li><a href="#delta-cq-to-normalizing-gene" id="toc-delta-cq-to-normalizing-gene" class="nav-link" data-scroll-target="#delta-cq-to-normalizing-gene">8 Delta Cq to Normalizing Gene</a>
  <ul>
  <li><a href="#t-tests" id="toc-t-tests" class="nav-link" data-scroll-target="#t-tests">8.1 t-tests</a>
  <ul>
  <li><a href="#life-stages" id="toc-life-stages" class="nav-link" data-scroll-target="#life-stages">8.1.1 Life Stages</a></li>
  <li><a href="#conditioning-treatments" id="toc-conditioning-treatments" class="nav-link" data-scroll-target="#conditioning-treatments">8.1.2 Conditioning treatments</a></li>
  <li><a href="#acute-treatments" id="toc-acute-treatments" class="nav-link" data-scroll-target="#acute-treatments">8.1.3 Acute treatments</a></li>
  <li><a href="#acute-within-life-stage-and-conditioning" id="toc-acute-within-life-stage-and-conditioning" class="nav-link" data-scroll-target="#acute-within-life-stage-and-conditioning">8.1.4 Acute within life stage and conditioning</a></li>
  </ul></li>
  <li><a href="#plotting" id="toc-plotting" class="nav-link" data-scroll-target="#plotting">8.2 Plotting</a>
  <ul>
  <li><a href="#delta-cq-boxplots" id="toc-delta-cq-boxplots" class="nav-link" data-scroll-target="#delta-cq-boxplots">8.2.1 Delta Cq boxplots</a>
  <ul class="collapse">
  <li><a href="#lifestage-comparisons" id="toc-lifestage-comparisons" class="nav-link" data-scroll-target="#lifestage-comparisons">8.2.1.1 Lifestage comparisons</a></li>
  </ul></li>
  <li><a href="#conditioning-comparisons-1" id="toc-conditioning-comparisons-1" class="nav-link" data-scroll-target="#conditioning-comparisons-1">8.2.2 Conditioning comparisons</a></li>
  <li><a href="#acute-treatment-comparisons" id="toc-acute-treatment-comparisons" class="nav-link" data-scroll-target="#acute-treatment-comparisons">8.2.3 Acute treatment comparisons</a></li>
  <li><a href="#acute-within-life-stage-conditioning" id="toc-acute-within-life-stage-conditioning" class="nav-link" data-scroll-target="#acute-within-life-stage-conditioning">8.2.4 Acute within life stage conditioning</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#delta-delta-cq" id="toc-delta-delta-cq" class="nav-link" data-scroll-target="#delta-delta-cq">9 Delta delta Cq</a>
  <ul>
  <li><a href="#calculations" id="toc-calculations" class="nav-link" data-scroll-target="#calculations">9.1 Calculations</a>
  <ul>
  <li><a href="#conditioning" id="toc-conditioning" class="nav-link" data-scroll-target="#conditioning">9.1.1 Conditioning</a></li>
  <li><a href="#acute-treatment" id="toc-acute-treatment" class="nav-link" data-scroll-target="#acute-treatment">9.1.2 Acute treatment</a></li>
  <li><a href="#life-stage" id="toc-life-stage" class="nav-link" data-scroll-target="#life-stage">9.1.3 Life stage</a></li>
  <li><a href="#calculate-delta-delta-acute-treatments-within-lifestage-and-conditioning" id="toc-calculate-delta-delta-acute-treatments-within-lifestage-and-conditioning" class="nav-link" data-scroll-target="#calculate-delta-delta-acute-treatments-within-lifestage-and-conditioning">9.1.4 Calculate delta delta acute treatments within lifestage and conditioning</a></li>
  <li><a href="#calculate-the-fold-change-life-stage-comparison" id="toc-calculate-the-fold-change-life-stage-comparison" class="nav-link" data-scroll-target="#calculate-the-fold-change-life-stage-comparison">9.1.5 Calculate the fold change life stage comparison</a></li>
  <li><a href="#calculate-the-fold-change-conditioning-comparison" id="toc-calculate-the-fold-change-conditioning-comparison" class="nav-link" data-scroll-target="#calculate-the-fold-change-conditioning-comparison">9.1.6 Calculate the fold change conditioning comparison</a></li>
  <li><a href="#calculate-the-fold-change-acute-comparison" id="toc-calculate-the-fold-change-acute-comparison" class="nav-link" data-scroll-target="#calculate-the-fold-change-acute-comparison">9.1.7 Calculate the fold change acute comparison</a></li>
  <li><a href="#calculate-fold-change-acute-treatments-within-lifestage-and-conditioning" id="toc-calculate-fold-change-acute-treatments-within-lifestage-and-conditioning" class="nav-link" data-scroll-target="#calculate-fold-change-acute-treatments-within-lifestage-and-conditioning">9.1.8 Calculate fold change acute treatments within lifestage and conditioning</a></li>
  </ul></li>
  <li><a href="#plotting-fold-changes" id="toc-plotting-fold-changes" class="nav-link" data-scroll-target="#plotting-fold-changes">9.2 Plotting fold changes</a>
  <ul>
  <li><a href="#acute-comparisons-within-lifestage-and-conditioning" id="toc-acute-comparisons-within-lifestage-and-conditioning" class="nav-link" data-scroll-target="#acute-comparisons-within-lifestage-and-conditioning">9.2.1 Acute comparisons within lifestage and conditioning</a></li>
  <li><a href="#life-stage-comparisons" id="toc-life-stage-comparisons" class="nav-link" data-scroll-target="#life-stage-comparisons">9.2.2 Life stage comparisons</a></li>
  <li><a href="#conditioning-comparisons-2" id="toc-conditioning-comparisons-2" class="nav-link" data-scroll-target="#conditioning-comparisons-2">9.2.3 Conditioning comparisons</a></li>
  <li><a href="#line-plot-conditioning-comparisons-across-lifestages" id="toc-line-plot-conditioning-comparisons-across-lifestages" class="nav-link" data-scroll-target="#line-plot-conditioning-comparisons-across-lifestages">9.2.4 Line plot conditioning comparisons across lifestages</a></li>
  <li><a href="#acute-treatment-comparison" id="toc-acute-treatment-comparison" class="nav-link" data-scroll-target="#acute-treatment-comparison">9.2.5 Acute treatment comparison</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#running-linear-models" id="toc-running-linear-models" class="nav-link" data-scroll-target="#running-linear-models">10 Running linear models</a>
  <ul>
  <li><a href="#anova-models" id="toc-anova-models" class="nav-link" data-scroll-target="#anova-models">10.1 ANOVA models</a>
  <ul>
  <li><a href="#atp-synthase" id="toc-atp-synthase" class="nav-link" data-scroll-target="#atp-synthase">10.1.1 ATP synthase</a></li>
  <li><a href="#dnmt1" id="toc-dnmt1" class="nav-link" data-scroll-target="#dnmt1">10.1.2 DNMT1</a></li>
  <li><a href="#hsp70" id="toc-hsp70" class="nav-link" data-scroll-target="#hsp70">10.1.3 HSP70</a></li>
  <li><a href="#hsp90" id="toc-hsp90" class="nav-link" data-scroll-target="#hsp90">10.1.4 HSP90</a></li>
  <li><a href="#cgas" id="toc-cgas" class="nav-link" data-scroll-target="#cgas">10.1.5 cGAS</a></li>
  <li><a href="#citrate-synthase" id="toc-citrate-synthase" class="nav-link" data-scroll-target="#citrate-synthase">10.1.6 Citrate synthase</a></li>
  <li><a href="#viperin" id="toc-viperin" class="nav-link" data-scroll-target="#viperin">10.1.7 Viperin</a></li>
  </ul></li>
  <li><a href="#plotting-1" id="toc-plotting-1" class="nav-link" data-scroll-target="#plotting-1">10.2 Plotting</a>
  <ul>
  <li><a href="#atp-synthase-no-effects" id="toc-atp-synthase-no-effects" class="nav-link" data-scroll-target="#atp-synthase-no-effects">10.2.1 ATP synthase: No effects</a></li>
  <li><a href="#dnmt1-no-effects" id="toc-dnmt1-no-effects" class="nav-link" data-scroll-target="#dnmt1-no-effects">10.2.2 DNMT1: No effects</a></li>
  <li><a href="#hsp70-significant-effect-of-conditioning-x-acute-treatment." id="toc-hsp70-significant-effect-of-conditioning-x-acute-treatment." class="nav-link" data-scroll-target="#hsp70-significant-effect-of-conditioning-x-acute-treatment.">10.2.3 HSP70: Significant effect of conditioning x acute treatment.</a></li>
  <li><a href="#hsp90-significant-effect-of-lifestage-x-acute-treatment-lifestage-x-conditioning-treatment-acute-treatment-and-lifestage." id="toc-hsp90-significant-effect-of-lifestage-x-acute-treatment-lifestage-x-conditioning-treatment-acute-treatment-and-lifestage." class="nav-link" data-scroll-target="#hsp90-significant-effect-of-lifestage-x-acute-treatment-lifestage-x-conditioning-treatment-acute-treatment-and-lifestage.">10.2.4 HSP90: Significant effect of lifestage x acute treatment, lifestage x conditioning treatment, acute treatment, and lifestage.</a></li>
  <li><a href="#cgas-significant-effect-of-lifestage-x-acute-treatment-and-lifestage." id="toc-cgas-significant-effect-of-lifestage-x-acute-treatment-and-lifestage." class="nav-link" data-scroll-target="#cgas-significant-effect-of-lifestage-x-acute-treatment-and-lifestage.">10.2.5 cGAS: Significant effect of lifestage x acute treatment and lifestage.</a></li>
  <li><a href="#citrate-synthase-significant-effect-of-acute-treatment-and-lifestage-x-acute-treatment." id="toc-citrate-synthase-significant-effect-of-acute-treatment-and-lifestage-x-acute-treatment." class="nav-link" data-scroll-target="#citrate-synthase-significant-effect-of-acute-treatment-and-lifestage-x-acute-treatment.">10.2.6 Citrate synthase: Significant effect of acute treatment and lifestage x acute treatment.</a></li>
  <li><a href="#viperin-significant-effect-of-lifestage-and-lifestage-x-acute-treatment." id="toc-viperin-significant-effect-of-lifestage-and-lifestage-x-acute-treatment." class="nav-link" data-scroll-target="#viperin-significant-effect-of-lifestage-and-lifestage-x-acute-treatment.">10.2.7 Viperin: Significant effect of lifestage and lifestage x acute treatment.</a></li>
  </ul></li>
  <li><a href="#conclusions" id="toc-conclusions" class="nav-link" data-scroll-target="#conclusions">10.3 Conclusions</a>
  <ul>
  <li><a href="#effects-of-conditioning-on-acute-stress-response" id="toc-effects-of-conditioning-on-acute-stress-response" class="nav-link" data-scroll-target="#effects-of-conditioning-on-acute-stress-response">10.3.1 Effects of conditioning on acute stress response</a></li>
  <li><a href="#effects-of-acute-stress-and-lifestage" id="toc-effects-of-acute-stress-and-lifestage" class="nav-link" data-scroll-target="#effects-of-acute-stress-and-lifestage">10.3.2 Effects of acute stress and lifestage</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-right" id="quarto-document-content">





<section id="intro" class="level1">
<h1>INTRO</h1>
<p>This notebook performs pair-wise comparisons of qPCR gene expression, normalized to <code>GAPDH</code> expression. It calculates delta Cq, delta delta Cq, and fold changes in expression. Additionally, it generates box plots (delta Cq), and bar plots (fold change expression).</p>
<p>I’ve provided a summary of the various pair-wise comparisons immediately below, but you might want to just <a href="#plotting">skip to the plots</a>, as the summary <em>and</em> the notebook content is <em>very</em> lengthy.</p>
</section>
<section id="summary" class="level1">
<h1>SUMMARY</h1>
<section id="t-tests-delta-cq" class="level2">
<h2 class="anchored" data-anchor-id="t-tests-delta-cq">t-tests (Delta Cq)</h2>
<section id="seed-vs.-adult" class="level3">
<h3 class="anchored" data-anchor-id="seed-vs.-adult">Seed vs.&nbsp;Adult</h3>
<p>These genes had p-values &lt;= 0.05:</p>
<ul>
<li>VIPERIN</li>
</ul>
</section>
<section id="seed-vs.-juvenile" class="level3">
<h3 class="anchored" data-anchor-id="seed-vs.-juvenile">Seed vs.&nbsp;Juvenile</h3>
<p>These genes had p-values &lt;= 0.05:</p>
<ul>
<li><p>citrate synthase</p></li>
<li><p>HSP90</p></li>
</ul>
</section>
<section id="seed-vs.-spat" class="level3">
<h3 class="anchored" data-anchor-id="seed-vs.-spat">Seed vs.&nbsp;Spat</h3>
<p>These genes had p-values &lt;= 0.05:</p>
<ul>
<li>VIPERIN</li>
</ul>
</section>
<section id="adult-vs.-juvenile" class="level3">
<h3 class="anchored" data-anchor-id="adult-vs.-juvenile">Adult vs.&nbsp;Juvenile</h3>
<p>The genes had p-values &lt;= 0.05:</p>
<ul>
<li><p>citrate synthase</p></li>
<li><p>VIPERIN</p></li>
</ul>
</section>
<section id="adult-vs.-spat" class="level3">
<h3 class="anchored" data-anchor-id="adult-vs.-spat">Adult vs.&nbsp;Spat</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="juvenile-vs.-spat" class="level3">
<h3 class="anchored" data-anchor-id="juvenile-vs.-spat">Juvenile vs.&nbsp;Spat</h3>
<p>The genes had p-values &lt;= 0.05:</p>
<ul>
<li><p>citrate synthase</p></li>
<li><p>VIPERIN</p></li>
</ul>
</section>
<section id="seed---conditioning-treated-vs.-control" class="level3">
<h3 class="anchored" data-anchor-id="seed---conditioning-treated-vs.-control">Seed - Conditioning Treated vs.&nbsp;Control</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="adult---conditioning-treated-vs.-control" class="level3">
<h3 class="anchored" data-anchor-id="adult---conditioning-treated-vs.-control">Adult - Conditioning Treated vs.&nbsp;Control</h3>
<p>The genes had p-values &lt;= 0.05:</p>
<ul>
<li>HSP90</li>
</ul>
</section>
<section id="juvenile---conditioning-treated-vs.-control" class="level3">
<h3 class="anchored" data-anchor-id="juvenile---conditioning-treated-vs.-control">Juvenile - Conditioning Treated vs.&nbsp;Control</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="spat---conditioning-treated-vs.-control" class="level3">
<h3 class="anchored" data-anchor-id="spat---conditioning-treated-vs.-control">Spat - Conditioning Treated vs.&nbsp;Control</h3>
<p>The genes had p-values &lt;= 0.05:</p>
<ul>
<li><p>citrate synthase</p></li>
<li><p>DNMT1</p></li>
</ul>
</section>
<section id="adult---acute-ambient-vs.-high" class="level3">
<h3 class="anchored" data-anchor-id="adult---acute-ambient-vs.-high">Adult - Acute Ambient vs.&nbsp;High</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="juvenile---acute-ambient-vs.-high" class="level3">
<h3 class="anchored" data-anchor-id="juvenile---acute-ambient-vs.-high">Juvenile - Acute Ambient vs.&nbsp;High</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="adult---conditioning-treated-acute-ambient-vs.-high" class="level3">
<h3 class="anchored" data-anchor-id="adult---conditioning-treated-acute-ambient-vs.-high">Adult - Conditioning Treated: Acute Ambient vs.&nbsp;High</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="adult---conditioning-control-acute-ambient-vs.-high" class="level3">
<h3 class="anchored" data-anchor-id="adult---conditioning-control-acute-ambient-vs.-high">Adult - Conditioning Control: Acute Ambient vs.&nbsp;High</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="juvenile---conditioning-treated-acute-ambient-vs.-high" class="level3">
<h3 class="anchored" data-anchor-id="juvenile---conditioning-treated-acute-ambient-vs.-high">Juvenile - Conditioning Treated: Acute Ambient vs.&nbsp;High</h3>
<p>The genes had p-values &lt;= 0.05:</p>
<ul>
<li>HSP70</li>
</ul>
</section>
<section id="juvenile---conditioning-control-acute-ambient-vs.-high" class="level3">
<h3 class="anchored" data-anchor-id="juvenile---conditioning-control-acute-ambient-vs.-high">Juvenile - Conditioning Control: Acute Ambient vs.&nbsp;High</h3>
<p>The genes had p-values &lt;= 0.05:</p>
<ul>
<li>VIPERIN</li>
</ul>
</section>
</section>
<section id="gene-expression-delta-delta-cq-or-fold-change" class="level2">
<h2 class="anchored" data-anchor-id="gene-expression-delta-delta-cq-or-fold-change">Gene Expression (delta delta Cq or fold change)</h2>
<section id="adult---conditioning-treated-acute-high-vs.-ambient" class="level3">
<h3 class="anchored" data-anchor-id="adult---conditioning-treated-acute-high-vs.-ambient">Adult - Conditioning Treated: Acute High vs.&nbsp;Ambient</h3>
<ul>
<li><p>All genes show elevated expression relative to Ambient.</p></li>
<li><p>Fold change is at similar levels across all genes, with DNMT1 and HSP70 being the highest.</p></li>
</ul>
</section>
<section id="juvenile---conditioning-treated-acute-high-vs.-ambient" class="level3">
<h3 class="anchored" data-anchor-id="juvenile---conditioning-treated-acute-high-vs.-ambient">Juvenile - Conditioning Treated: Acute High vs.&nbsp;Ambient</h3>
<ul>
<li><p>All genes show elevated expression relative to Ambient.</p></li>
<li><p>HSP70 shows ~4-fold higher fold change in expression compared to most of the other genes.</p></li>
<li><p>DNMT1 exhibits fold change expression ~1/2 that of most of the other genes.</p></li>
</ul>
</section>
<section id="adult---adult-vs.-seed" class="level3">
<h3 class="anchored" data-anchor-id="adult---adult-vs.-seed">Adult - Adult vs.&nbsp;Seed</h3>
<ul>
<li><p>All genes show elevated expression relative to Seed.</p></li>
<li><p>VIPERIN exhibits fold change in expression &gt;2-fold higher than the other genes.</p></li>
</ul>
</section>
<section id="adult---adult-vs.-spat" class="level3">
<h3 class="anchored" data-anchor-id="adult---adult-vs.-spat">Adult - Adult vs.&nbsp;Spat</h3>
<ul>
<li><p>All genes show elevated expression relative to Spat.</p></li>
<li><p>All genes have similar fold changes in expression, with ATP synthase and DNMT1 showing the highest levels of fold change in expression.</p></li>
</ul>
</section>
<section id="adult---spat-vs.-seed" class="level3">
<h3 class="anchored" data-anchor-id="adult---spat-vs.-seed">Adult - Spat vs.&nbsp;Seed</h3>
<ul>
<li><p>All genes show elevated expression relative to Seed.</p></li>
<li><p>Most genes have similar fold changes in expression.</p></li>
<li><p>VIPERIN exhibits fold change in a expression ~2-fold higher than the other genes.</p></li>
</ul>
</section>
<section id="conditioning-by-target-and-lifestage" class="level3">
<h3 class="anchored" data-anchor-id="conditioning-by-target-and-lifestage">Conditioning by Target and Lifestage</h3>
<ul>
<li><p>All genes show elevated expression relative to Control.</p></li>
<li><p>VIPERIN, citrate synthase, and ATP synthase show an initial increase in fold change expression from seed to spat life stages, declining to juvenile and adult lifestages.</p></li>
<li><p>HSP90 shows level fold changes in expression across seed, spat, and juvenile lifestages, followed by a decrease in the adult stage.</p></li>
<li><p>DNMT1 exhibits a drastic decrease in fold change expression from seed to spat, followed by a substantial increase from spat to adult.</p></li>
<li><p>cGAS and HSP70 exhibit decreases from seed to spat, followed by a sharp increase from spat to juvenile. This is followed by a moderate decrease from juvenile to adult by cGAS and a sharp decrease in HSP70.</p></li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The post below is markdown generated from <a href="https://github.com/RobertsLab/project-gigas-carryover/blob/main/lifestage_carryover/scripts/01.01-qPCR.Rmd">01.01-qPCR.Rmd</a> (commit <code>71b99c4</code>).</p>
<p>Ariana added everything <em>after</em> <a href="#acute-treatment-comparison"><code>### 9.2.5 Acute treatment comparison</code></a>.</p>
</div>
</div>
</section>
</section>
</section>
<section id="description" class="level1">
<h1>1 Description</h1>
<p>This notebook performs pair-wise comparisons of qPCR gene expression, normalized to <code>GAPDH</code> expression. It calculates delta Cq, delta delta Cq, and fold changes in expression. Additionally, it generates box plots (delta Cq), and bar plots (fold change expression).</p>
</section>
<section id="set-variables" class="level1">
<h1>2 Set variables</h1>
<section id="set-sample-groups" class="level2">
<h2 class="anchored" data-anchor-id="set-sample-groups">2.1 Set sample groups</h2>
<p>Groups are named in the following fashion:</p>
<p><code>&lt;life.stage&gt;.&lt;conditioning.treatment&gt;.&lt;acute.treatment&gt;</code></p>
<p>This allows for parsing downstream.</p>
<p>NOTE: Below is the full set of groups for the <em>entire experiment</em>. For the current qPCR analysis, <code>seed</code> and <code>spat</code> do <em>not</em> have acute treatments; just conditioning treatments.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>seed.control.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"29"</span>, <span class="st">"40"</span>, <span class="st">"55"</span>, <span class="st">"63"</span>, <span class="st">"69"</span>, <span class="st">"101"</span>, <span class="st">"119"</span>, <span class="st">"122"</span>, <span class="st">"155"</span>, <span class="st">"164"</span>, <span class="st">"187"</span>, <span class="st">"202"</span>, <span class="st">"209"</span>, <span class="st">"214"</span>, <span class="st">"233"</span>, <span class="st">"236"</span>, <span class="st">"275"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>seed.control.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"42"</span>, <span class="st">"59"</span>, <span class="st">"60"</span>, <span class="st">"62"</span>, <span class="st">"86"</span>, <span class="st">"102"</span>, <span class="st">"140"</span>, <span class="st">"176"</span>, <span class="st">"177"</span>, <span class="st">"184"</span>, <span class="st">"192"</span>, <span class="st">"223"</span>, <span class="st">"234"</span>, <span class="st">"243"</span>, <span class="st">"244"</span>, <span class="st">"254"</span>, <span class="st">"264"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>seed.treated.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"14"</span>, <span class="st">"48"</span>, <span class="st">"66"</span>, <span class="st">"72"</span>, <span class="st">"89"</span>, <span class="st">"115"</span>, <span class="st">"129"</span>, <span class="st">"138"</span>, <span class="st">"156"</span>, <span class="st">"182"</span>, <span class="st">"191"</span>, <span class="st">"201"</span>, <span class="st">"227"</span>, <span class="st">"239"</span>, <span class="st">"270"</span>, <span class="st">"277"</span>, <span class="st">"280"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>seed.treated.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"15"</span>, <span class="st">"19"</span>, <span class="st">"24"</span>, <span class="st">"88"</span>, <span class="st">"92"</span>, <span class="st">"105"</span>, <span class="st">"111"</span>, <span class="st">"113"</span>, <span class="st">"120"</span>, <span class="st">"128"</span>, <span class="st">"161"</span>, <span class="st">"200"</span>, <span class="st">"211"</span>, <span class="st">"256"</span>, <span class="st">"257"</span>, <span class="st">"266"</span>, <span class="st">"285"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>spat.control.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"11"</span>, <span class="st">"30"</span>, <span class="st">"36"</span>, <span class="st">"52"</span>, <span class="st">"77"</span>, <span class="st">"114"</span>, <span class="st">"134"</span>, <span class="st">"142"</span>, <span class="st">"144"</span>, <span class="st">"183"</span>, <span class="st">"193"</span>, <span class="st">"229"</span>, <span class="st">"230"</span>, <span class="st">"231"</span>, <span class="st">"240"</span>, <span class="st">"272"</span>, <span class="st">"287"</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>spat.control.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"27"</span>, <span class="st">"74"</span>, <span class="st">"93"</span>, <span class="st">"96"</span>, <span class="st">"97"</span>, <span class="st">"137"</span>, <span class="st">"143"</span>, <span class="st">"153"</span>, <span class="st">"168"</span>, <span class="st">"178"</span>, <span class="st">"189"</span>, <span class="st">"206"</span>, <span class="st">"262"</span>, <span class="st">"274"</span>, <span class="st">"282"</span>, <span class="st">"284"</span>, <span class="st">"289"</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>spat.treated.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"9"</span>, <span class="st">"13"</span>, <span class="st">"38"</span>, <span class="st">"46"</span>, <span class="st">"47"</span>, <span class="st">"121"</span>, <span class="st">"145"</span>, <span class="st">"151"</span>, <span class="st">"174"</span>, <span class="st">"194"</span>, <span class="st">"197"</span>, <span class="st">"198"</span>, <span class="st">"216"</span>, <span class="st">"235"</span>, <span class="st">"241"</span>, <span class="st">"252"</span>, <span class="st">"291"</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>spat.treated.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"6"</span>, <span class="st">"25"</span>, <span class="st">"50"</span>, <span class="st">"78"</span>, <span class="st">"124"</span>, <span class="st">"126"</span>, <span class="st">"131"</span>, <span class="st">"160"</span>, <span class="st">"163"</span>, <span class="st">"172"</span>, <span class="st">"220"</span>, <span class="st">"226"</span>, <span class="st">"242"</span>, <span class="st">"253"</span>, <span class="st">"296"</span>, <span class="st">"298"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>juvenile.control.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"18"</span>, <span class="st">"57"</span>, <span class="st">"65"</span>, <span class="st">"75"</span>, <span class="st">"79"</span>, <span class="st">"104"</span>, <span class="st">"110"</span>, <span class="st">"123"</span>, <span class="st">"125"</span>, <span class="st">"171"</span>, <span class="st">"175"</span>, <span class="st">"205"</span>, <span class="st">"238"</span>, <span class="st">"273"</span>, <span class="st">"279"</span>, <span class="st">"293"</span>, <span class="st">"317"</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>juvenile.control.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"12"</span>, <span class="st">"39"</span>, <span class="st">"43"</span>, <span class="st">"49"</span>, <span class="st">"71"</span>, <span class="st">"130"</span>, <span class="st">"141"</span>, <span class="st">"146"</span>, <span class="st">"150"</span>, <span class="st">"170"</span>, <span class="st">"195"</span>, <span class="st">"297"</span>, <span class="st">"301"</span>, <span class="st">"324"</span>, <span class="st">"351"</span>, <span class="st">"355"</span>, <span class="st">"371"</span>)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>juvenile.treated.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"34"</span>, <span class="st">"64"</span>, <span class="st">"83"</span>, <span class="st">"98"</span>, <span class="st">"147"</span>, <span class="st">"152"</span>, <span class="st">"158"</span>, <span class="st">"162"</span>, <span class="st">"169"</span>, <span class="st">"188"</span>, <span class="st">"271"</span>, <span class="st">"295"</span>, <span class="st">"310"</span>, <span class="st">"357"</span>, <span class="st">"361"</span>, <span class="st">"381"</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>juvenile.treated.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"28"</span>, <span class="st">"53"</span>, <span class="st">"61"</span>, <span class="st">"73"</span>, <span class="st">"81"</span>, <span class="st">"106"</span>, <span class="st">"109"</span>, <span class="st">"139"</span>, <span class="st">"149"</span>, <span class="st">"173"</span>, <span class="st">"181"</span>, <span class="st">"213"</span>, <span class="st">"290"</span>, <span class="st">"302"</span>, <span class="st">"311"</span>, <span class="st">"364"</span>, <span class="st">"392"</span>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>adult.control.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"3"</span>, <span class="st">"5"</span>, <span class="st">"13*"</span>, <span class="st">"16"</span>, <span class="st">"17"</span>, <span class="st">"80"</span>, <span class="st">"87"</span>, <span class="st">"94"</span>, <span class="st">"148"</span>, <span class="st">"159"</span>, <span class="st">"179"</span>, <span class="st">"180"</span>, <span class="st">"250"</span>, <span class="st">"258"</span>, <span class="st">"268"</span>, <span class="st">"312"</span>, <span class="st">"326"</span>, <span class="st">"330"</span>, <span class="st">"334"</span>, <span class="st">"346"</span>, <span class="st">"360"</span>, <span class="st">"377"</span>, <span class="st">"379"</span>, <span class="st">"386"</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>adult.control.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"20"</span>, <span class="st">"23"</span>, <span class="st">"26"</span>, <span class="st">"32"</span>, <span class="st">"33"</span>, <span class="st">"67"</span>, <span class="st">"70"</span>, <span class="st">"90"</span>, <span class="st">"107"</span>, <span class="st">"132"</span>, <span class="st">"135"</span>, <span class="st">"157"</span>, <span class="st">"166"</span>, <span class="st">"186"</span>, <span class="st">"207"</span>, <span class="st">"215"</span>, <span class="st">"248"</span>, <span class="st">"316"</span>, <span class="st">"341"</span>, <span class="st">"344"</span>, <span class="st">"349"</span>, <span class="st">"382"</span>, <span class="st">"394"</span>, <span class="st">"395"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>adult.treated.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"7"</span>, <span class="st">"31"</span>, <span class="st">"35"</span>, <span class="st">"37"</span>, <span class="st">"41"</span>, <span class="st">"54"</span>, <span class="st">"84"</span>, <span class="st">"100"</span>, <span class="st">"112"</span>, <span class="st">"116"</span>, <span class="st">"118"</span>, <span class="st">"133"</span>, <span class="st">"154"</span>, <span class="st">"199"</span>, <span class="st">"203"</span>, <span class="st">"204"</span>, <span class="st">"208"</span>, <span class="st">"219"</span>, <span class="st">"294"</span>, <span class="st">"318"</span>, <span class="st">"339"</span>, <span class="st">"353"</span>, <span class="st">"363"</span>, <span class="st">"378"</span>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>adult.treated.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"21"</span>, <span class="st">"22"</span>, <span class="st">"45"</span>, <span class="st">"82"</span>, <span class="st">"85"</span>, <span class="st">"91"</span>, <span class="st">"95"</span>, <span class="st">"99"</span>, <span class="st">"103"</span>, <span class="st">"108"</span>, <span class="st">"117"</span>, <span class="st">"127"</span>, <span class="st">"165"</span>, <span class="st">"185"</span>, <span class="st">"190"</span>, <span class="st">"196"</span>, <span class="st">"232"</span>, <span class="st">"237"</span>, <span class="st">"245"</span>, <span class="st">"263"</span>, <span class="st">"276"</span>, <span class="st">"306"</span>, <span class="st">"343"</span>, <span class="st">"374"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="assign-groups-to-list" class="level2">
<h2 class="anchored" data-anchor-id="assign-groups-to-list">2.2 Assign groups to list</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine vectors into lists</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Used for adding treatment info and/or subsetting downstream</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>groups_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">juvenile.control.ambient =</span> juvenile.control.ambient,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">juvenile.control.high =</span> juvenile.control.high,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">juvenile.treated.ambient =</span> juvenile.treated.ambient,</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">juvenile.treated.high =</span> juvenile.treated.high,</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">adult.control.ambient =</span> adult.control.ambient,</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">adult.control.high =</span> adult.control.high,</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">adult.treated.ambient =</span> adult.treated.ambient,</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">adult.treated.high =</span> adult.treated.high,</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">seed.control.ambient =</span> seed.control.ambient,</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">seed.control.high =</span> seed.control.high,</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">seed.treated.ambient =</span> seed.treated.ambient,</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">seed.treated.high =</span> seed.treated.high,</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">spat.control.ambient =</span> spat.control.ambient,</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">spat.control.high =</span> spat.control.high,</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">spat.treated.ambient =</span> spat.treated.ambient,</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">spat.treated.high =</span> spat.treated.high)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="functions" class="level1">
<h1>3 Functions</h1>
<section id="calculate-delta-cq" class="level2">
<h2 class="anchored" data-anchor-id="calculate-delta-cq">3.1 Calculate delta Cq</h2>
<p>Normalized to designated normalizing gene</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>calculate_delta_Cq <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Sample) <span class="sc">%&gt;%</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">delta_Cq =</span> Cq.Mean <span class="sc">-</span> Cq.Mean[Target <span class="sc">==</span> <span class="st">"GAPDH"</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>()</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="create-delta-cq-boxplots" class="level2">
<h2 class="anchored" data-anchor-id="create-delta-cq-boxplots">3.2 Create delta Cq boxplots</h2>
<section id="lifestage-comparison" class="level3">
<h3 class="anchored" data-anchor-id="lifestage-comparison">3.2.1 Lifestage comparison</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create box plots for each comparison</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>create_boxplot_delta_Cq <span class="ot">&lt;-</span> <span class="cf">function</span>(data, comparison, t_test_results) {</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract life stages from comparison</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  life_stages <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(comparison, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>))</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print life stages</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(paste("Life stages for comparison:", comparison))</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(life_stages)</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter data for the relevant life stages</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  filtered_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(life.stage <span class="sc">%in%</span> life_stages)</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print filtered data</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Filtered data:")</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(filtered_data)</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both life stages are included</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(life_stages <span class="sc">%in%</span> <span class="fu">unique</span>(filtered_data<span class="sc">$</span>life.stage))) {</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Not all life stages are included in the filtered data"</span>)</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>  y_limits <span class="ot">&lt;-</span> <span class="fu">range</span>(filtered_data<span class="sc">$</span>delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print y_limits</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Y limits:")</span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(y_limits)</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for the current comparison</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>  t_test_results_filtered <span class="ot">&lt;-</span> t_test_results <span class="sc">%&gt;%</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(comparison <span class="sc">==</span> <span class="sc">!!</span>comparison)</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print filtered t_test_results</span></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Filtered t_test_results:")</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(t_test_results_filtered)</span></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for asterisks</span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>  t_test_results_with_asterisks <span class="ot">&lt;-</span> t_test_results_filtered <span class="sc">%&gt;%</span></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(asterisk <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print t_test_results_with_asterisks</span></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("t_test_results_with_asterisks:")</span></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(t_test_results_with_asterisks)</span></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>  formatted_title <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">toupper</span>(<span class="fu">substring</span>(life_stages[<span class="dv">1</span>], <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(life_stages[<span class="dv">1</span>], <span class="dv">2</span>), </span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" vs. "</span>, </span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(life_stages[<span class="dv">2</span>], <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(life_stages[<span class="dv">2</span>], <span class="dv">2</span>))</span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>  boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(filtered_data, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> life.stage)) <span class="sc">+</span></span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"salmon"</span>, <span class="st">"lightblue"</span>, <span class="st">"lightgreen"</span>)) <span class="sc">+</span></span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(y_limits) <span class="sc">+</span></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Delta Cq"</span>, <span class="at">title =</span> formatted_title) <span class="sc">+</span></span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Highlighted section: Adds asterisks</span></span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> t_test_results_with_asterisks, </span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> y_limits[<span class="dv">2</span>] <span class="sc">-</span> <span class="dv">1</span>, <span class="at">label =</span> asterisk), </span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a>              <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(boxplot)</span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="conditioning-comparisons" class="level3">
<h3 class="anchored" data-anchor-id="conditioning-comparisons">3.2.2 Conditioning comparisons</h3>
<ol type="1">
<li>Extract Life Stage and Conditioning Treatments:</li>
</ol>
<ul>
<li>The comparison string is split into its components (<code>life_stage</code>, <code>treatment1</code>, and <code>treatment2</code>).</li>
</ul>
<ol start="2" type="1">
<li>Filter Data:</li>
</ol>
<ul>
<li>The filtered_data data frame is filtered to include only the rows with the relevant life stage and conditioning treatments.</li>
</ul>
<ol start="3" type="1">
<li>Check for Both Treatments:</li>
</ol>
<ul>
<li>Ensure that both treatments are included in the <code>filtered_data</code>.</li>
</ul>
<ol start="4" type="1">
<li>Filter T-Test Results:</li>
</ol>
<ul>
<li><p>The <code>t_test_results_filtered</code> data frame is filtered for the specific comparison.</p></li>
<li><p>The <code>t_test_results_with_asterisks</code> data frame is created to include only the rows with asterisks.</p></li>
</ul>
<ol start="5" type="1">
<li>Format the Title:</li>
</ol>
<ul>
<li><p>The <code>formatted_title</code> variable is created by capitalizing the first letter of each component and concatenating them with ” - ” and ” vs.&nbsp;” in between.</p></li>
<li><p>This should create box plots comparing conditioning treatments within each life stage, with titles formatted as <code>&lt;life.stage&gt; - Treated vs. Control</code>.</p></li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create box plots for each comparison of conditioning treatments within life stages</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>create_boxplot_conditioning <span class="ot">&lt;-</span> <span class="cf">function</span>(data, comparison, t_test_results) {</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract life stage and conditioning treatments from comparison</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  comparison_parts <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(comparison, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>))</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  life_stage <span class="ot">&lt;-</span> comparison_parts[<span class="dv">1</span>]</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  treatment1 <span class="ot">&lt;-</span> comparison_parts[<span class="dv">2</span>]</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  treatment2 <span class="ot">&lt;-</span> comparison_parts[<span class="dv">3</span>]</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print life stage and treatments</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(paste("Life stage and treatments for comparison:", comparison))</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(c(life_stage, treatment1, treatment2))</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter data for the relevant life stage and conditioning treatments</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  filtered_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(life.stage <span class="sc">==</span> life_stage, conditioning.treatment <span class="sc">%in%</span> <span class="fu">c</span>(treatment1, treatment2))</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print filtered data</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Filtered data:")</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(filtered_data)</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both treatments are included</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">c</span>(treatment1, treatment2) <span class="sc">%in%</span> <span class="fu">unique</span>(filtered_data<span class="sc">$</span>conditioning.treatment))) {</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Not all treatments are included in the filtered data"</span>)</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>  y_limits <span class="ot">&lt;-</span> <span class="fu">range</span>(filtered_data<span class="sc">$</span>delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print y_limits</span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Y limits:")</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(y_limits)</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for the current comparison</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>  t_test_results_filtered <span class="ot">&lt;-</span> t_test_results <span class="sc">%&gt;%</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(comparison <span class="sc">==</span> <span class="sc">!!</span>comparison)</span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print filtered t_test_results</span></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Filtered t_test_results:")</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(t_test_results_filtered)</span></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for asterisks</span></span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>  t_test_results_with_asterisks <span class="ot">&lt;-</span> t_test_results_filtered <span class="sc">%&gt;%</span></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(asterisk <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print t_test_results_with_asterisks</span></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("t_test_results_with_asterisks:")</span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(t_test_results_with_asterisks)</span></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Format the title</span></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>  formatted_title <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">toupper</span>(<span class="fu">substring</span>(life_stage, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(life_stage, <span class="dv">2</span>), </span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" - "</span>, </span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(treatment1, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(treatment1, <span class="dv">2</span>), </span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" vs. "</span>, </span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(treatment2, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(treatment2, <span class="dv">2</span>))</span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>  boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(filtered_data, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> conditioning.treatment)) <span class="sc">+</span></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"salmon"</span>)) <span class="sc">+</span></span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(y_limits) <span class="sc">+</span></span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Delta Cq"</span>, <span class="at">title =</span> formatted_title) <span class="sc">+</span></span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Highlighted section: Adds asterisks</span></span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> t_test_results_with_asterisks, </span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> y_limits[<span class="dv">2</span>] <span class="sc">-</span> <span class="dv">1</span>, <span class="at">label =</span> asterisk), </span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a>              <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(boxplot)</span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="acute-comparisons" class="level3">
<h3 class="anchored" data-anchor-id="acute-comparisons">3.2.3 Acute comparisons</h3>
<ol type="1">
<li>Extract Life Stage and Acute Treatments:</li>
</ol>
<ul>
<li>The comparison string is split into its components (<code>life_stage</code>, <code>treatment1</code>, and <code>treatment2</code>).</li>
</ul>
<ol start="2" type="1">
<li>Filter Data:</li>
</ol>
<ul>
<li>The <code>filtered_data</code> data frame is filtered to include only the rows with the relevant life stage and acute treatments.</li>
</ul>
<ol start="3" type="1">
<li>Check for Both Treatments:</li>
</ol>
<ul>
<li>Ensure that both treatments are included in the <code>filtered_data</code>.</li>
</ul>
<ol start="4" type="1">
<li>Filter T-Test Results:</li>
</ol>
<ul>
<li><p>The <code>t_test_results_filtered</code> data frame is filtered for the specific comparison.</p></li>
<li><p>The <code>t_test_results_with_asterisks</code> data frame is created to include only the rows with asterisks. Format the Title:</p></li>
</ul>
<ol start="5" type="1">
<li>The formatted_title variable is created by capitalizing the first letter of each component and concatenating them with ” - ” and ” vs.&nbsp;” in between.</li>
</ol>
<ul>
<li>This should create box plots comparing acute treatments within each life stage, with titles formatted as <code>&lt;life.stage&gt; - Ambient vs. High</code>.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create box plots for each comparison of acute treatments within life stages</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>create_boxplot_acute <span class="ot">&lt;-</span> <span class="cf">function</span>(data, comparison, t_test_results) {</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract life stage and acute treatments from comparison</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  comparison_parts <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(comparison, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>))</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  life_stage <span class="ot">&lt;-</span> comparison_parts[<span class="dv">1</span>]</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  treatment1 <span class="ot">&lt;-</span> comparison_parts[<span class="dv">2</span>]</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  treatment2 <span class="ot">&lt;-</span> comparison_parts[<span class="dv">3</span>]</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print life stage and treatments</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(paste("Life stage and treatments for comparison:", comparison))</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(c(life_stage, treatment1, treatment2))</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter data for the relevant life stage and acute treatments</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  filtered_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(life.stage <span class="sc">==</span> life_stage, acute.treatment <span class="sc">%in%</span> <span class="fu">c</span>(treatment1, treatment2))</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print filtered data</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Filtered data:")</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(filtered_data)</span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both treatments are included</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">c</span>(treatment1, treatment2) <span class="sc">%in%</span> <span class="fu">unique</span>(filtered_data<span class="sc">$</span>acute.treatment))) {</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Not all treatments are included in the filtered data"</span>)</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>  y_limits <span class="ot">&lt;-</span> <span class="fu">range</span>(filtered_data<span class="sc">$</span>delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print y_limits</span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Y limits:")</span></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(y_limits)</span></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for the current comparison</span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>  t_test_results_filtered <span class="ot">&lt;-</span> t_test_results <span class="sc">%&gt;%</span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(comparison <span class="sc">==</span> <span class="sc">!!</span>comparison)</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print filtered t_test_results</span></span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Filtered t_test_results:")</span></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(t_test_results_filtered)</span></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for asterisks</span></span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>  t_test_results_with_asterisks <span class="ot">&lt;-</span> t_test_results_filtered <span class="sc">%&gt;%</span></span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(asterisk <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print t_test_results_with_asterisks</span></span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("t_test_results_with_asterisks:")</span></span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(t_test_results_with_asterisks)</span></span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Format the title</span></span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a>  formatted_title <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">toupper</span>(<span class="fu">substring</span>(life_stage, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(life_stage, <span class="dv">2</span>), </span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" - "</span>, </span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(treatment1, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(treatment1, <span class="dv">2</span>), </span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" vs. "</span>, </span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(treatment2, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(treatment2, <span class="dv">2</span>))</span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a>  boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(filtered_data, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> acute.treatment)) <span class="sc">+</span></span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"salmon"</span>)) <span class="sc">+</span></span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(y_limits) <span class="sc">+</span></span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Delta Cq"</span>, <span class="at">title =</span> formatted_title) <span class="sc">+</span></span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Highlighted section: Adds asterisks</span></span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> t_test_results_with_asterisks, </span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> y_limits[<span class="dv">2</span>] <span class="sc">-</span> <span class="dv">1</span>, <span class="at">label =</span> asterisk), </span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a>              <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"magenta"</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(boxplot)</span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="acute-treatements-within-life-stage-conditioning" class="level3">
<h3 class="anchored" data-anchor-id="acute-treatements-within-life-stage-conditioning">3.2.4 Acute treatements within life stage conditioning</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create box plots for each comparison of acute treatments within life stages and conditioning treatments</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>create_boxplot_acute_conditioning <span class="ot">&lt;-</span> <span class="cf">function</span>(data, comparison, t_test_results) {</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract life stage, conditioning treatment, and acute treatments from comparison</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  comparison_parts <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(comparison, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>))</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  life_stage <span class="ot">&lt;-</span> comparison_parts[<span class="dv">1</span>]</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  conditioning_treatment <span class="ot">&lt;-</span> comparison_parts[<span class="dv">2</span>]</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  treatment1 <span class="ot">&lt;-</span> comparison_parts[<span class="dv">3</span>]</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  treatment2 <span class="ot">&lt;-</span> comparison_parts[<span class="dv">5</span>]</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter data for the relevant life stage, conditioning treatment, and acute treatments</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  filtered_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(life.stage <span class="sc">==</span> life_stage, conditioning.treatment <span class="sc">==</span> conditioning_treatment, acute.treatment <span class="sc">%in%</span> <span class="fu">c</span>(treatment1, treatment2))</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both treatments are included</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">c</span>(treatment1, treatment2) <span class="sc">%in%</span> <span class="fu">unique</span>(filtered_data<span class="sc">$</span>acute.treatment))) {</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Not all treatments are included in the filtered data"</span>)</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>  y_limits <span class="ot">&lt;-</span> <span class="fu">range</span>(filtered_data<span class="sc">$</span>delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for the current comparison</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>  t_test_results_filtered <span class="ot">&lt;-</span> t_test_results <span class="sc">%&gt;%</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(comparison <span class="sc">==</span> <span class="sc">!!</span>comparison)</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for asterisks</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>  t_test_results_with_asterisks <span class="ot">&lt;-</span> t_test_results_filtered <span class="sc">%&gt;%</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(asterisk <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Format the title</span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>  formatted_title <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">toupper</span>(<span class="fu">substring</span>(life_stage, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(life_stage, <span class="dv">2</span>), </span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" - "</span>, </span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(conditioning_treatment, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(conditioning_treatment, <span class="dv">2</span>), </span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" - "</span>, </span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(treatment1, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(treatment1, <span class="dv">2</span>), </span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" vs. "</span>, </span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(treatment2, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(treatment2, <span class="dv">2</span>))</span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>  boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(filtered_data, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> acute.treatment)) <span class="sc">+</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"salmon"</span>)) <span class="sc">+</span></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(y_limits) <span class="sc">+</span></span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Delta Cq"</span>, <span class="at">title =</span> formatted_title) <span class="sc">+</span></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Adds asterisks</span></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> t_test_results_with_asterisks, </span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> y_limits[<span class="dv">2</span>] <span class="sc">-</span> <span class="dv">1</span>, <span class="at">label =</span> asterisk), </span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a>              <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>)</span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(boxplot)</span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
</section>
<section id="read-in-files" class="level1">
<h1>4 Read in files</h1>
<div class="sourceCode" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get a list of all CSV files in the directory with the naming structure "*Cq-Results.csv"</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>cq_file_list <span class="ot">&lt;-</span> <span class="fu">list</span>() <span class="co"># Initialize list</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>cq_file_list <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> cqs_directory, <span class="at">pattern =</span> <span class="st">"Cq-Results</span><span class="sc">\\</span><span class="st">.csv$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize an empty list to store the data frames</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>data_frames_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each file and read it into a data frame, then add it to the list</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (file <span class="cf">in</span> cq_file_list) {</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(file, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  data<span class="sc">$</span>Sample <span class="ot">&lt;-</span> <span class="fu">as.character</span>(data<span class="sc">$</span>Sample)  <span class="co"># Convert Sample column to character type</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  data_frames_list[[file]] <span class="ot">&lt;-</span> data</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine all data frames into a single data frame</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(data_frames_list, <span class="at">.id =</span> <span class="st">"data_frame_id"</span>)</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(combined_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>'data.frame':   2816 obs. of  17 variables:
 $ data_frame_id         : chr  "../data/qPCR/Cq/sam_2024-03-25_06-10-54_Connect-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-03-25_06-10-54_Connect-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-03-25_06-10-54_Connect-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-03-25_06-10-54_Connect-Quantification-Cq-Results.csv" ...
 $ X                     : logi  NA NA NA NA NA NA ...
 $ Well                  : chr  "A01" "A02" "A03" "A04" ...
 $ Fluor                 : chr  "SYBR" "SYBR" "SYBR" "SYBR" ...
 $ Target                : chr  "ATPsynthase" "ATPsynthase" "ATPsynthase" "ATPsynthase" ...
 $ Content               : chr  "Unkn-01" "Unkn-01" "Unkn-02" "Unkn-02" ...
 $ Sample                : chr  "206" "206" "220" "220" ...
 $ Biological.Set.Name   : logi  NA NA NA NA NA NA ...
 $ Cq                    : num  26.7 26.7 25.8 25.9 25.1 ...
 $ Cq.Mean               : num  26.7 26.7 25.9 25.9 25.1 ...
 $ Cq.Std..Dev           : num  0.0455 0.0455 0.0239 0.0239 0.0813 ...
 $ Starting.Quantity..SQ.: num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Log.Starting.Quantity : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Mean               : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Std..Dev           : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Set.Point             : int  60 60 60 60 60 60 60 60 60 60 ...
 $ Well.Note             : logi  NA NA NA NA NA NA ...</code></pre>
</section>
<section id="clean-data" class="level1">
<h1>5 Clean data</h1>
<section id="replace-target-names" class="level2">
<h2 class="anchored" data-anchor-id="replace-target-names">5.1 Replace target names</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with Sample name "NTC"</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> combined_df[combined_df<span class="sc">$</span>Sample <span class="sc">!=</span> <span class="st">"NTC"</span>, ]</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace values in the Target column</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_GAPDH_205_F-355_R </span><span class="sc">\\</span><span class="st">(SR IDs: 1172/3</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"GAPDH"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_ATPsynthase_F/R </span><span class="sc">\\</span><span class="st">(SR IDs: 1385/6</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"ATPsynthase"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_cGAS </span><span class="sc">\\</span><span class="st">(SR IDs: 1826/7</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"cGAS"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_citrate_synthase </span><span class="sc">\\</span><span class="st">(SR IDs: 1383/4</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"citrate synthase"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_DNMT1_F </span><span class="sc">\\</span><span class="st">(SR IDs: 1510/1</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"DNMT1"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_HSP70_F/R </span><span class="sc">\\</span><span class="st">(SR IDs: 598/9</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"HSP70"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_Hsp90_F/R </span><span class="sc">\\</span><span class="st">(SR IDs: 1532/3</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"HSP90"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_VIPERIN_F/R </span><span class="sc">\\</span><span class="st">(SR IDs: 1828/9</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"viperin"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(combined_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>'data.frame':   2801 obs. of  17 variables:
 $ data_frame_id         : chr  "../data/qPCR/Cq/sam_2024-03-25_06-10-54_Connect-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-03-25_06-10-54_Connect-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-03-25_06-10-54_Connect-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-03-25_06-10-54_Connect-Quantification-Cq-Results.csv" ...
 $ X                     : logi  NA NA NA NA NA NA ...
 $ Well                  : chr  "A01" "A02" "A03" "A04" ...
 $ Fluor                 : chr  "SYBR" "SYBR" "SYBR" "SYBR" ...
 $ Target                : chr  "ATPsynthase" "ATPsynthase" "ATPsynthase" "ATPsynthase" ...
 $ Content               : chr  "Unkn-01" "Unkn-01" "Unkn-02" "Unkn-02" ...
 $ Sample                : chr  "206" "206" "220" "220" ...
 $ Biological.Set.Name   : logi  NA NA NA NA NA NA ...
 $ Cq                    : num  26.7 26.7 25.8 25.9 25.1 ...
 $ Cq.Mean               : num  26.7 26.7 25.9 25.9 25.1 ...
 $ Cq.Std..Dev           : num  0.0455 0.0455 0.0239 0.0239 0.0813 ...
 $ Starting.Quantity..SQ.: num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Log.Starting.Quantity : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Mean               : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Std..Dev           : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Set.Point             : int  60 60 60 60 60 60 60 60 60 60 ...
 $ Well.Note             : logi  NA NA NA NA NA NA ...</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(<span class="fu">as.factor</span>(combined_df<span class="sc">$</span>Target))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>[1] "ATPsynthase"      "cGAS"             "citrate synthase" "DNMT1"           
[5] "GAPDH"            "HSP70"            "HSP90"            "viperin"         </code></pre>
</section>
<section id="identify-samples-with-cq.std..dev-0.5" class="level2">
<h2 class="anchored" data-anchor-id="identify-samples-with-cq.std..dev-0.5">5.2 Identify Samples with Cq.Std..Dev &gt; 0.5</h2>
<div class="sourceCode" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out rows where Cq.Std..Dev is NA</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> combined_df[<span class="sc">!</span><span class="fu">is.na</span>(combined_df<span class="sc">$</span>Cq.Std..Dev), ]</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter rows where Cq.Std..Dev is greater than 0.5</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>high_cq_std_dev <span class="ot">&lt;-</span> combined_df[combined_df<span class="sc">$</span>Cq.Std..Dev <span class="sc">&gt;</span> <span class="fl">0.5</span>, ]</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the filtered rows with specified columns, without row names</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(high_cq_std_dev[, <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"Sample"</span>, <span class="st">"Cq"</span>, <span class="st">"Cq.Std..Dev"</span>)], <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>           Target Sample       Cq Cq.Std..Dev
            HSP70    244 33.10339   4.0838809
            HSP70    244 27.32791   4.0838809
            HSP90    223 24.85319   0.7548714
            HSP90    223 25.92074   0.7548714
          viperin    223 30.30089   0.6058663
          viperin    223 31.15772   0.6058663
          viperin    243 32.57817   0.5527617
          viperin    243 33.35989   0.5527617
            DNMT1    296 31.21374   0.6417578
            DNMT1    296 30.30616   0.6417578
            DNMT1    298 35.68716   0.5406704
            DNMT1    298 34.92253   0.5406704
            DNMT1    223 32.24089   0.6214201
            DNMT1    223 33.11971   0.6214201
            DNMT1    243 36.63921   0.5125743
            DNMT1    243 35.91432   0.5125743
            DNMT1    285 33.63443   0.7036122
            DNMT1    285 34.62949   0.7036122
            GAPDH    316 23.94926   8.5684728
            GAPDH    316 24.14183   8.5684728
            GAPDH    316 38.88564   8.5684728
            GAPDH    213 26.98012   2.2910353
            GAPDH    213 23.00009   2.2910353
            GAPDH    213 26.95634   2.2910353
            GAPDH    263 22.42154   0.8731474
            GAPDH    263 23.77008   0.8731474
            GAPDH    263 24.05667   0.8731474
 citrate synthase    230 24.44066   4.4783429
 citrate synthase    230 24.40421   4.4783429
 citrate synthase    230 32.17909   4.4783429
          viperin    227 30.47773   3.5152533
          viperin    227 30.37738   3.5152533
          viperin    227 36.51553   3.5152533
          viperin    245 26.05748   5.1635899
          viperin    245 34.98192   5.1635899
          viperin    245 26.01928   5.1635899
          viperin    341 26.48675   2.9838590
          viperin    341 31.67235   2.9838590
          viperin    341 26.52174   2.9838590
          viperin    344 29.98184   2.3712440
          viperin    344 25.90358   2.3712440
          viperin    344 25.84648   2.3712440
          viperin    355 28.79712   0.5821437
          viperin    355 29.57428   0.5821437
          viperin    355 28.43490   0.5821437
          viperin     18 30.22286   2.5813451
          viperin     18 31.97799   2.5813451
          viperin     18 35.30515   2.5813451
          viperin     25 36.49472   0.6540634
          viperin     25 35.62766   0.6540634
          viperin     25 35.21293   0.6540634
          viperin     62 32.44608   0.6249794
          viperin     62 31.31507   0.6249794
          viperin     62 31.41970   0.6249794
          viperin     75 33.47202   1.6346910
          viperin     75 32.63655   1.6346910
          viperin     75 30.31693   1.6346910
             cGAS     14 38.23519   0.7502227
             cGAS     14 38.02258   0.7502227
             cGAS     14 39.41520   0.7502227
             cGAS     29 37.25712   1.2060821
             cGAS     29 39.31941   1.2060821
             cGAS     29 37.20469   1.2060821
             cGAS     40 36.28250   0.5469250
             cGAS     40 36.05499   0.5469250
             cGAS     40 35.24216   0.5469250
      ATPsynthase     15 22.46831   1.6297928
      ATPsynthase     15 25.31150   1.6297928
      ATPsynthase     15 22.50938   1.6297928
      ATPsynthase     18 24.52172   0.7966632
      ATPsynthase     18 25.64075   0.7966632
      ATPsynthase     18 24.09896   0.7966632
            HSP70     12 27.55137   3.6393890
            HSP70     12 33.62834   3.6393890
            HSP70     12 27.12023   3.6393890
            HSP70     29 35.56445   1.1431868
            HSP70     29 37.62398   1.1431868
            HSP70     29 35.73434   1.1431868
            HSP70     53 27.67834   0.6164802
            HSP70     53 26.50222   0.6164802
            HSP70     53 26.76980   0.6164802
            HSP90     14 36.83671   1.7075603
            HSP90     14 39.25156   1.7075603
            HSP90     18 25.15642   0.9422544
            HSP90     18 24.41329   0.9422544
            HSP90     18 23.28508   0.9422544
            HSP90     29 38.90450   1.0457320
            HSP90     29 37.42561   1.0457320
            GAPDH     24 25.78634   0.6778352
            GAPDH     24 24.82774   0.6778352
            GAPDH     29 37.22294   1.3777016
            GAPDH     29 39.17130   1.3777016</code></pre>
</section>
<section id="remove-bad-technical-reps" class="level2">
<h2 class="anchored" data-anchor-id="remove-bad-technical-reps">5.3 Remove bad technical reps</h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by Sample and Target, then filter out the outlier replicate</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>combined.fitered_df<span class="ot">&lt;-</span> combined_df <span class="sc">%&gt;%</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Sample, Target) <span class="sc">%&gt;%</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">abs</span>(Cq <span class="sc">-</span> <span class="fu">mean</span>(Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">&lt;=</span> Cq.Std..Dev)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the filtered data frame</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(combined.fitered_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>gropd_df [1,905 × 17] (S3: grouped_df/tbl_df/tbl/data.frame)
 $ data_frame_id         : chr [1:1905] "../data/qPCR/Cq/sam_2024-03-25_06-10-54_Connect-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-03-25_06-10-54_Connect-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-03-25_06-10-54_Connect-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-03-25_06-10-54_Connect-Quantification-Cq-Results.csv" ...
 $ X                     : logi [1:1905] NA NA NA NA NA NA ...
 $ Well                  : chr [1:1905] "A01" "A02" "A03" "A04" ...
 $ Fluor                 : chr [1:1905] "SYBR" "SYBR" "SYBR" "SYBR" ...
 $ Target                : chr [1:1905] "ATPsynthase" "ATPsynthase" "ATPsynthase" "ATPsynthase" ...
 $ Content               : chr [1:1905] "Unkn-01" "Unkn-01" "Unkn-02" "Unkn-02" ...
 $ Sample                : chr [1:1905] "206" "206" "220" "220" ...
 $ Biological.Set.Name   : logi [1:1905] NA NA NA NA NA NA ...
 $ Cq                    : num [1:1905] 26.7 26.7 25.8 25.9 25.1 ...
 $ Cq.Mean               : num [1:1905] 26.7 26.7 25.9 25.9 25.1 ...
 $ Cq.Std..Dev           : num [1:1905] 0.0455 0.0455 0.0239 0.0239 0.0813 ...
 $ Starting.Quantity..SQ.: num [1:1905] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Log.Starting.Quantity : num [1:1905] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Mean               : num [1:1905] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Std..Dev           : num [1:1905] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Set.Point             : int [1:1905] 60 60 60 60 60 60 60 60 60 60 ...
 $ Well.Note             : logi [1:1905] NA NA NA NA NA NA ...
 - attr(*, "groups")= tibble [956 × 3] (S3: tbl_df/tbl/data.frame)
  ..$ Sample: chr [1:956] "12" "12" "12" "12" ...
  ..$ Target: chr [1:956] "ATPsynthase" "DNMT1" "GAPDH" "HSP70" ...
  ..$ .rows : list&lt;int&gt; [1:956] 
  .. ..$ : int [1:2] 1616 1617
  .. ..$ : int [1:2] 1712 1713
  .. ..$ : int [1:2] 1857 1858
  .. ..$ : int [1:2] 1758 1759
  .. ..$ : int [1:2] 1807 1808
  .. ..$ : int [1:2] 1566 1567
  .. ..$ : int [1:2] 1664 1665
  .. ..$ : int [1:2] 1521 1522
  .. ..$ : int 1618
  .. ..$ : int 1859
  .. ..$ : int 1760
  .. ..$ : int [1:2] 1809 1810
  .. ..$ : int [1:2] 1568 1569
  .. ..$ : int 1666
  .. ..$ : int [1:2] 1619 1620
  .. ..$ : int [1:2] 1714 1715
  .. ..$ : int [1:2] 1860 1861
  .. ..$ : int [1:2] 1761 1762
  .. ..$ : int [1:2] 1811 1812
  .. ..$ : int [1:2] 1570 1571
  .. ..$ : int [1:2] 1667 1668
  .. ..$ : int [1:2] 1523 1524
  .. ..$ : int [1:2] 1621 1622
  .. ..$ : int [1:2] 1716 1717
  .. ..$ : int [1:2] 1862 1863
  .. ..$ : int [1:2] 1763 1764
  .. ..$ : int [1:2] 1813 1814
  .. ..$ : int [1:2] 1572 1573
  .. ..$ : int [1:2] 1669 1670
  .. ..$ : int [1:2] 1525 1526
  .. ..$ : int [1:2] 1623 1624
  .. ..$ : int [1:2] 1718 1719
  .. ..$ : int [1:2] 1864 1865
  .. ..$ : int [1:2] 1765 1766
  .. ..$ : int [1:2] 1815 1816
  .. ..$ : int [1:2] 1574 1575
  .. ..$ : int [1:2] 1671 1672
  .. ..$ : int [1:2] 1527 1528
  .. ..$ : int [1:2] 21 22
  .. ..$ : int [1:2] 245 246
  .. ..$ : int [1:2] 85 86
  .. ..$ : int [1:2] 53 54
  .. ..$ : int [1:2] 117 118
  .. ..$ : int [1:2] 149 150
  .. ..$ : int [1:2] 213 214
  .. ..$ : int [1:2] 181 182
  .. ..$ : int [1:2] 321 322
  .. ..$ : int [1:2] 889 890
  .. ..$ : int [1:2] 637 638
  .. ..$ : int [1:2] 1047 1048
  .. ..$ : int [1:2] 1205 1206
  .. ..$ : int [1:2] 479 480
  .. ..$ : int [1:2] 731 732
  .. ..$ : int [1:2] 1363 1364
  .. ..$ : int [1:2] 323 324
  .. ..$ : int [1:2] 891 892
  .. ..$ : int [1:2] 639 640
  .. ..$ : int [1:2] 1049 1050
  .. ..$ : int [1:2] 1207 1208
  .. ..$ : int [1:2] 481 482
  .. ..$ : int [1:2] 733 734
  .. ..$ : int [1:2] 1365 1366
  .. ..$ : int [1:2] 325 326
  .. ..$ : int [1:2] 893 894
  .. ..$ : int [1:2] 641 642
  .. ..$ : int [1:2] 1051 1052
  .. ..$ : int [1:2] 1209 1210
  .. ..$ : int [1:2] 483 484
  .. ..$ : int [1:2] 735 736
  .. ..$ : int [1:2] 1367 1368
  .. ..$ : int [1:2] 327 328
  .. ..$ : int [1:2] 895 896
  .. ..$ : int [1:2] 643 644
  .. ..$ : int [1:2] 1053 1054
  .. ..$ : int [1:2] 1211 1212
  .. ..$ : int [1:2] 485 486
  .. ..$ : int [1:2] 737 738
  .. ..$ : int [1:2] 1369 1370
  .. ..$ : int [1:2] 329 330
  .. ..$ : int [1:2] 897 898
  .. ..$ : int [1:2] 645 646
  .. ..$ : int [1:2] 1055 1056
  .. ..$ : int [1:2] 1213 1214
  .. ..$ : int [1:2] 487 488
  .. ..$ : int [1:2] 739 740
  .. ..$ : int [1:2] 1371 1372
  .. ..$ : int [1:2] 1 2
  .. ..$ : int [1:2] 225 226
  .. ..$ : int [1:2] 65 66
  .. ..$ : int [1:2] 33 34
  .. ..$ : int [1:2] 97 98
  .. ..$ : int [1:2] 129 130
  .. ..$ : int [1:2] 193 194
  .. ..$ : int [1:2] 161 162
  .. ..$ : int [1:2] 331 332
  .. ..$ : int [1:2] 899 900
  .. ..$ : int [1:2] 647 648
  .. ..$ : int [1:2] 1057 1058
  .. ..$ : int [1:2] 1215 1216
  .. .. [list output truncated]
  .. ..@ ptype: int(0) 
  ..- attr(*, ".drop")= logi TRUE</code></pre>
</section>
</section>
<section id="group-samples-by-target" class="level1">
<h1>6 Group samples by target</h1>
<div class="sourceCode" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by Sample and Target, then summarize to get unique rows for each sample</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>grouped_df <span class="ot">&lt;-</span> combined.fitered_df<span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Sample, Target) <span class="sc">%&gt;%</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Cq.Mean =</span> <span class="fu">mean</span>(Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(grouped_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [956 × 3] (S3: tbl_df/tbl/data.frame)
 $ Sample : chr [1:956] "12" "12" "12" "12" ...
 $ Target : chr [1:956] "ATPsynthase" "DNMT1" "GAPDH" "HSP70" ...
 $ Cq.Mean: num [1:956] 27.6 32.5 26 27.3 26.8 ...</code></pre>
</section>
<section id="add-life-stage-and-treatment-cols" class="level1">
<h1>7 Add life stage and treatment cols</h1>
<div class="sourceCode" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize new columns</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>grouped_df <span class="ot">&lt;-</span> grouped_df <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">life.stage =</span> <span class="cn">NA_character_</span>,</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">conditioning.treatment =</span> <span class="cn">NA_character_</span>,</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">acute.treatment =</span> <span class="cn">NA_character_</span>)</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each vector</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (vec_name <span class="cf">in</span> <span class="fu">names</span>(groups_list)) {</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>  vec <span class="ot">&lt;-</span> groups_list[[vec_name]]</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>  stage <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(vec_name, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>  conditioning_treatment <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(vec_name, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>  acute_treatment <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(vec_name, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>)[[<span class="dv">1</span>]][<span class="dv">3</span>]</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Loop through each row in grouped_df</span></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(grouped_df)) {</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>    sample <span class="ot">&lt;-</span> grouped_df<span class="sc">$</span>Sample[i]</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if sample is in the vector</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (sample <span class="sc">%in%</span> vec) {</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Update life.stage and treatment columns</span></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>      grouped_df<span class="sc">$</span>life.stage[i] <span class="ot">&lt;-</span> stage</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>      grouped_df<span class="sc">$</span>conditioning.treatment[i] <span class="ot">&lt;-</span> conditioning_treatment</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>      grouped_df<span class="sc">$</span>acute.treatment[i] <span class="ot">&lt;-</span>acute_treatment</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(grouped_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [956 × 6] (S3: tbl_df/tbl/data.frame)
 $ Sample                : chr [1:956] "12" "12" "12" "12" ...
 $ Target                : chr [1:956] "ATPsynthase" "DNMT1" "GAPDH" "HSP70" ...
 $ Cq.Mean               : num [1:956] 27.6 32.5 26 27.3 26.8 ...
 $ life.stage            : chr [1:956] "juvenile" "juvenile" "juvenile" "juvenile" ...
 $ conditioning.treatment: chr [1:956] "control" "control" "control" "control" ...
 $ acute.treatment       : chr [1:956] "high" "high" "high" "high" ...</code></pre>
</section>
<section id="delta-cq-to-normalizing-gene" class="level1">
<h1>8 Delta Cq to Normalizing Gene</h1>
<div class="sourceCode" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta Cq by subtracting GAPDH Cq.Mean from each corresponding Sample Cq.Mean</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>delta_Cq_df <span class="ot">&lt;-</span> <span class="fu">calculate_delta_Cq</span>(grouped_df)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filters out normalizing gene, since no need to compare normalizing gene to itself.</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>delta_Cq_df <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(life.stage), <span class="sc">!</span><span class="fu">is.na</span>(Target), Target <span class="sc">!=</span> <span class="st">"GAPDH"</span>)</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_Cq_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [836 × 7] (S3: tbl_df/tbl/data.frame)
 $ Sample                : chr [1:836] "12" "12" "12" "12" ...
 $ Target                : chr [1:836] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
 $ Cq.Mean               : num [1:836] 27.6 32.5 27.3 26.8 30.7 ...
 $ life.stage            : chr [1:836] "juvenile" "juvenile" "juvenile" "juvenile" ...
 $ conditioning.treatment: chr [1:836] "control" "control" "control" "control" ...
 $ acute.treatment       : chr [1:836] "high" "high" "high" "high" ...
 $ delta_Cq              : num [1:836] 1.567 6.499 1.306 0.756 4.692 ...</code></pre>
<section id="t-tests" class="level2">
<h2 class="anchored" data-anchor-id="t-tests">8.1 t-tests</h2>
<section id="life-stages" class="level3">
<h3 class="anchored" data-anchor-id="life-stages">8.1.1 Life Stages</h3>
<p>This code does the following:</p>
<ol type="1">
<li>Extracts the unique life.stage levels from the data frame.</li>
<li>Generates all possible pairs of life.stage levels using the combn function.</li>
<li>Iterates over each pair and performs the t-test for each Target. Adds an asterisk column and an asterisk if the p-value is &lt;= 0.05. Useful for downstream parsing.</li>
<li>Stores the results in a list and combines them into a single data frame.</li>
<li>Adds a comparison column to indicate which life.stage levels were compared.</li>
</ol>
<div class="sourceCode" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract unique life.stage levels</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>unique_life_stages <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df<span class="sc">$</span>life.stage)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate all possible pairs of life.stage levels</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>life_stage_pairs <span class="ot">&lt;-</span> <span class="fu">combn</span>(unique_life_stages, <span class="dv">2</span>, <span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store results</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>life_stage_t_test_results_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pair <span class="cf">in</span> life_stage_pairs) {</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  stage1 <span class="ot">&lt;-</span> pair[<span class="dv">1</span>]</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  stage2 <span class="ot">&lt;-</span> pair[<span class="dv">2</span>]</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Perform t-test for each Target comparing the two life.stage levels</span></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  t_test_results <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(life.stage <span class="sc">%in%</span> <span class="fu">c</span>(stage1, stage2)) <span class="sc">%&gt;%</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>      <span class="at">t_test_result =</span> <span class="fu">list</span>(<span class="fu">t.test</span>(delta_Cq <span class="sc">~</span> life.stage))</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>      <span class="at">estimate_diff =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>estimate[<span class="dv">1</span>] <span class="sc">-</span> x<span class="sc">$</span>estimate[<span class="dv">2</span>]),</span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>      <span class="at">p_value =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>p.value),</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>      <span class="at">asterisk =</span> <span class="fu">ifelse</span>(p_value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="st">""</span>), <span class="co"># Adds asterisk column and asterisk for p-value.</span></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>      <span class="at">comparison =</span> <span class="fu">paste</span>(stage1, <span class="st">"vs"</span>, stage2, <span class="at">sep =</span> <span class="st">"."</span>)</span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">!</span>t_test_result)</span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>  life_stage_t_test_results_list[[<span class="fu">paste</span>(stage1, stage2, <span class="at">sep =</span> <span class="st">"."</span>)]] <span class="ot">&lt;-</span> t_test_results</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results into a single data frame</span></span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>life_stage_t_test_results_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(life_stage_t_test_results_list, <span class="at">.id =</span> <span class="st">"comparison"</span>)</span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a><span class="co"># View the results</span></span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(life_stage_t_test_results_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code># A tibble: 42 × 5
   Target           estimate_diff  p_value asterisk comparison    
   &lt;chr&gt;                    &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;         
 1 ATPsynthase            0.292   0.0995   ""       juvenile.seed 
 2 DNMT1                 -0.0616  0.846    ""       juvenile.seed 
 3 HSP70                  0.976   0.0324   "*"      juvenile.seed 
 4 HSP90                  0.802   0.000644 "*"      juvenile.seed 
 5 cGAS                  -0.0369  0.924    ""       juvenile.seed 
 6 citrate synthase       0.158   0.502    ""       juvenile.seed 
 7 viperin                0.00222 0.996    ""       juvenile.seed 
 8 ATPsynthase           -0.252   0.124    ""       juvenile.adult
 9 DNMT1                 -0.152   0.562    ""       juvenile.adult
10 HSP70                 -0.0139  0.976    ""       juvenile.adult
# ℹ 32 more rows</code></pre>
</section>
<section id="conditioning-treatments" class="level3">
<h3 class="anchored" data-anchor-id="conditioning-treatments">8.1.2 Conditioning treatments</h3>
<p>This code does the following:</p>
<ol type="1">
<li>Extracts the unique life.stage levels from the data frame.</li>
<li>For each life.stage, extracts the unique conditioning.treatment levels.</li>
<li>Generates all possible pairs of conditioning.treatment levels within each life.stage.</li>
<li>Iterates over each pair and performs the t-test for each Target. Adds an asterisk column and an asterisk if the p-value is &lt;= 0.05. Useful for downstream parsing.</li>
<li>Stores the results in a list and combines them into a single data frame.</li>
<li>Adds a comparison column to indicate which life.stage and conditioning.treatment levels were compared.</li>
</ol>
<div class="sourceCode" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract unique life.stage levels</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>unique_life_stages <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df<span class="sc">$</span>life.stage)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store results</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>conditioning_treatment_t_test_results_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (stage <span class="cf">in</span> unique_life_stages) {</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract unique conditioning.treatment levels within the current life.stage</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  unique_treatments <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(life.stage <span class="sc">==</span> stage) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(conditioning.treatment))</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Generate all possible pairs of conditioning.treatment levels</span></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>  treatment_pairs <span class="ot">&lt;-</span> <span class="fu">combn</span>(unique_treatments, <span class="dv">2</span>, <span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (pair <span class="cf">in</span> treatment_pairs) {</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>    treatment1 <span class="ot">&lt;-</span> pair[<span class="dv">1</span>]</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>    treatment2 <span class="ot">&lt;-</span> pair[<span class="dv">2</span>]</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Perform t-test for each Target comparing the two conditioning.treatment levels within the current life.stage</span></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>    t_test_results <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(life.stage <span class="sc">==</span> stage, conditioning.treatment <span class="sc">%in%</span> <span class="fu">c</span>(treatment1, treatment2)) <span class="sc">%&gt;%</span></span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">t_test_result =</span> <span class="fu">list</span>(<span class="fu">t.test</span>(delta_Cq <span class="sc">~</span> conditioning.treatment))</span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>      ) <span class="sc">%&gt;%</span></span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(</span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">estimate_diff =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>estimate[<span class="dv">1</span>] <span class="sc">-</span> x<span class="sc">$</span>estimate[<span class="dv">2</span>]),</span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">p_value =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>p.value),</span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a>        <span class="at">asterisk =</span> <span class="fu">ifelse</span>(p_value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="st">""</span>), <span class="co"># Adds asterisk column and asterisk for p-value.</span></span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>        <span class="at">comparison =</span> <span class="fu">paste</span>(stage, treatment1, <span class="st">"vs"</span>, treatment2, <span class="at">sep =</span> <span class="st">"."</span>)</span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>      ) <span class="sc">%&gt;%</span></span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">!</span>t_test_result)</span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a>    conditioning_treatment_t_test_results_list[[<span class="fu">paste</span>(stage, treatment1, treatment2, <span class="at">sep =</span> <span class="st">"."</span>)]] <span class="ot">&lt;-</span> t_test_results</span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results into a single data frame</span></span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a>conditioning_treatment_t_test_results_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(conditioning_treatment_t_test_results_list, <span class="at">.id =</span> <span class="st">"comparison"</span>)</span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a><span class="co"># View the results</span></span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(conditioning_treatment_t_test_results_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code># A tibble: 28 × 5
   Target           estimate_diff p_value asterisk comparison              
   &lt;chr&gt;                    &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;                   
 1 ATPsynthase             0.427    0.103 ""       juvenile.control.treated
 2 DNMT1                   0.578    0.178 ""       juvenile.control.treated
 3 HSP70                  -0.587    0.351 ""       juvenile.control.treated
 4 HSP90                   0.0666   0.842 ""       juvenile.control.treated
 5 cGAS                    0.249    0.336 ""       juvenile.control.treated
 6 citrate synthase        0.384    0.209 ""       juvenile.control.treated
 7 viperin                 1.02     0.109 ""       juvenile.control.treated
 8 ATPsynthase             0.110    0.592 ""       seed.treated.control    
 9 DNMT1                   0.213    0.646 ""       seed.treated.control    
10 HSP70                  -0.509    0.432 ""       seed.treated.control    
# ℹ 18 more rows</code></pre>
</section>
<section id="acute-treatments" class="level3">
<h3 class="anchored" data-anchor-id="acute-treatments">8.1.3 Acute treatments</h3>
<p>This code does the following:</p>
<ol type="1">
<li>Extracts the unique life.stage levels from the data frame.</li>
<li>For each life.stage, extracts the unique acute.treatment levels.</li>
<li>Generates all possible pairs of acute.treatment levels within each life.stage.</li>
<li>Iterates over each pair and performs the t-test for each Target. Adds an asterisk column and an asterisk if the p-value is &lt;= 0.05. Useful for downstream parsing.</li>
<li>Stores the results in a list and combines them into a single data frame.</li>
<li>Adds a comparison column to indicate which life.stage and acute.treatment levels were compared.</li>
</ol>
<p>Excludes <code>seed</code> and <code>spat</code>, as these were only held at <code>ambient</code> for the acute treatment.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract unique life.stage levels, excluding 'seed' and 'spat'</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>unique_life_stages <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df<span class="sc">$</span>life.stage)</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>unique_life_stages <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(unique_life_stages, <span class="fu">c</span>(<span class="st">"seed"</span>, <span class="st">"spat"</span>))</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store results</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>acute_treatment_t_test_results_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (stage <span class="cf">in</span> unique_life_stages) {</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract unique acute.treatment levels within the current life.stage</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>  unique_treatments <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(life.stage <span class="sc">==</span> stage) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(acute.treatment))</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if there are at least 2 unique treatments</span></span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(unique_treatments) <span class="sc">&gt;=</span> <span class="dv">2</span>) {</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Generate all possible pairs of acute.treatment levels</span></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>    treatment_pairs <span class="ot">&lt;-</span> <span class="fu">combn</span>(unique_treatments, <span class="dv">2</span>, <span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (pair <span class="cf">in</span> treatment_pairs) {</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>      treatment1 <span class="ot">&lt;-</span> pair[<span class="dv">1</span>]</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>      treatment2 <span class="ot">&lt;-</span> pair[<span class="dv">2</span>]</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Perform t-test for each Target comparing the two acute.treatment levels within the current life.stage</span></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>      t_test_results <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(life.stage <span class="sc">==</span> stage, acute.treatment <span class="sc">%in%</span> <span class="fu">c</span>(treatment1, treatment2)) <span class="sc">%&gt;%</span></span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>        <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>        <span class="fu">summarise</span>(</span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>          <span class="at">t_test_result =</span> <span class="fu">list</span>(<span class="fu">t.test</span>(delta_Cq <span class="sc">~</span> acute.treatment))</span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(</span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a>          <span class="at">estimate_diff =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>estimate[<span class="dv">1</span>] <span class="sc">-</span> x<span class="sc">$</span>estimate[<span class="dv">2</span>]),</span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a>          <span class="at">p_value =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>p.value),</span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a>          <span class="at">asterisk =</span> <span class="fu">ifelse</span>(p_value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="st">""</span>), <span class="co"># Adds asterisk column and asterisk for p-value.</span></span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a>          <span class="at">comparison =</span> <span class="fu">paste</span>(stage, treatment1, <span class="st">"vs"</span>, treatment2, <span class="at">sep =</span> <span class="st">"."</span>)</span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a>        <span class="fu">select</span>(<span class="sc">!</span>t_test_result)</span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a>      acute_treatment_t_test_results_list[[<span class="fu">paste</span>(stage, treatment1, treatment2, <span class="at">sep =</span> <span class="st">"."</span>)]] <span class="ot">&lt;-</span> t_test_results</span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results into a single data frame</span></span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a>acute_treatment_t_test_results_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(acute_treatment_t_test_results_list, <span class="at">.id =</span> <span class="st">"comparison"</span>)</span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a><span class="co"># View the results</span></span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(acute_treatment_t_test_results_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code># A tibble: 14 × 5
   Target           estimate_diff p_value asterisk comparison           
   &lt;chr&gt;                    &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;                
 1 ATPsynthase             0.174   0.602  ""       juvenile.high.ambient
 2 DNMT1                  -0.552   0.250  ""       juvenile.high.ambient
 3 HSP70                   0.886   0.170  ""       juvenile.high.ambient
 4 HSP90                   0.790   0.0261 "*"      juvenile.high.ambient
 5 cGAS                   -0.191   0.448  ""       juvenile.high.ambient
 6 citrate synthase       -0.115   0.716  ""       juvenile.high.ambient
 7 viperin                 0.319   0.690  ""       juvenile.high.ambient
 8 ATPsynthase             0.0605  0.687  ""       adult.ambient.high   
 9 DNMT1                   0.314   0.275  ""       adult.ambient.high   
10 HSP70                   0.276   0.696  ""       adult.ambient.high   
11 HSP90                   0.499   0.149  ""       adult.ambient.high   
12 cGAS                    0.329   0.200  ""       adult.ambient.high   
13 citrate synthase        0.0668  0.622  ""       adult.ambient.high   
14 viperin                 0.323   0.251  ""       adult.ambient.high   </code></pre>
</section>
<section id="acute-within-life-stage-and-conditioning" class="level3">
<h3 class="anchored" data-anchor-id="acute-within-life-stage-and-conditioning">8.1.4 Acute within life stage and conditioning</h3>
<div class="sourceCode" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract unique life.stage levels, excluding 'seed' and 'spat'</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>unique_life_stages <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df<span class="sc">$</span>life.stage)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co">#unique_life_stages &lt;- setdiff(unique_life_stages, c("seed", "spat"))</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract unique conditioning.treatment levels</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>unique_conditioning_treatments <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df<span class="sc">$</span>conditioning.treatment)</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store results</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>acute_treatment_within_life.stages_conditioning_t_test_results_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (stage <span class="cf">in</span> unique_life_stages) {</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (conditioning <span class="cf">in</span> unique_conditioning_treatments) {</span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Extract unique acute.treatment levels within the current life.stage and conditioning.treatment</span></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>    unique_treatments <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(life.stage <span class="sc">==</span> stage, conditioning.treatment <span class="sc">==</span> conditioning) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(acute.treatment))</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if there are at least 2 unique treatments</span></span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(unique_treatments) <span class="sc">&gt;=</span> <span class="dv">2</span>) {</span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Generate all possible pairs of acute.treatment levels</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>      treatment_pairs <span class="ot">&lt;-</span> <span class="fu">combn</span>(unique_treatments, <span class="dv">2</span>, <span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span> (pair <span class="cf">in</span> treatment_pairs) {</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>        treatment1 <span class="ot">&lt;-</span> pair[<span class="dv">1</span>]</span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>        treatment2 <span class="ot">&lt;-</span> pair[<span class="dv">2</span>]</span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Perform t-test for each Target comparing the two acute.treatment levels within the current life.stage and conditioning.treatment</span></span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>        t_test_results <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>          <span class="fu">filter</span>(life.stage <span class="sc">==</span> stage, conditioning.treatment <span class="sc">==</span> conditioning, acute.treatment <span class="sc">%in%</span> <span class="fu">c</span>(treatment1, treatment2)) <span class="sc">%&gt;%</span></span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a>          <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a>          <span class="fu">summarise</span>(</span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>            <span class="at">t_test_result =</span> <span class="fu">list</span>(<span class="fu">t.test</span>(delta_Cq <span class="sc">~</span> acute.treatment))</span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>          ) <span class="sc">%&gt;%</span></span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>          <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a>          <span class="fu">mutate</span>(</span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a>            <span class="at">estimate_diff =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>estimate[<span class="dv">1</span>] <span class="sc">-</span> x<span class="sc">$</span>estimate[<span class="dv">2</span>]),</span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_value =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>p.value),</span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a>            <span class="at">asterisk =</span> <span class="fu">ifelse</span>(p_value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="st">""</span>), <span class="co"># Adds asterisk column and asterisk for p-value.</span></span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a>            <span class="at">comparison =</span> <span class="fu">paste</span>(stage, conditioning, treatment1, <span class="st">"vs"</span>, treatment2, <span class="at">sep =</span> <span class="st">"."</span>)</span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a>          ) <span class="sc">%&gt;%</span></span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a>          <span class="fu">select</span>(<span class="sc">!</span>t_test_result)</span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a>        acute_treatment_within_life.stages_conditioning_t_test_results_list[[<span class="fu">paste</span>(stage, conditioning, treatment1, treatment2, <span class="at">sep =</span> <span class="st">"."</span>)]] <span class="ot">&lt;-</span> t_test_results</span>
<span id="cb30-42"><a href="#cb30-42" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb30-43"><a href="#cb30-43" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb30-44"><a href="#cb30-44" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb30-45"><a href="#cb30-45" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-46"><a href="#cb30-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-47"><a href="#cb30-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results into a single data frame</span></span>
<span id="cb30-48"><a href="#cb30-48" aria-hidden="true" tabindex="-1"></a>acute_treatment_within_life.stages_conditioning_t_test_results_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(acute_treatment_within_life.stages_conditioning_t_test_results_list, <span class="at">.id =</span> <span class="st">"comparison_id"</span>)</span>
<span id="cb30-49"><a href="#cb30-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-50"><a href="#cb30-50" aria-hidden="true" tabindex="-1"></a><span class="co"># View the results</span></span>
<span id="cb30-51"><a href="#cb30-51" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(acute_treatment_within_life.stages_conditioning_t_test_results_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code># A tibble: 56 × 6
   comparison_id                Target estimate_diff p_value asterisk comparison
   &lt;chr&gt;                        &lt;chr&gt;          &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;     
 1 juvenile.control.high.ambie… ATPsy…        0.369   0.500  ""       juvenile.…
 2 juvenile.control.high.ambie… DNMT1        -0.170   0.811  ""       juvenile.…
 3 juvenile.control.high.ambie… HSP70         0.187   0.837  ""       juvenile.…
 4 juvenile.control.high.ambie… HSP90         0.995   0.0661 ""       juvenile.…
 5 juvenile.control.high.ambie… cGAS         -0.144   0.632  ""       juvenile.…
 6 juvenile.control.high.ambie… citra…        0.0872  0.844  ""       juvenile.…
 7 juvenile.control.high.ambie… viper…        0.680   0.590  ""       juvenile.…
 8 juvenile.treated.high.ambie… ATPsy…       -0.143   0.421  ""       juvenile.…
 9 juvenile.treated.high.ambie… DNMT1        -1.15    0.0274 "*"      juvenile.…
10 juvenile.treated.high.ambie… HSP70         1.92    0.0306 "*"      juvenile.…
# ℹ 46 more rows</code></pre>
</section>
</section>
<section id="plotting" class="level2">
<h2 class="anchored" data-anchor-id="plotting">8.2 Plotting</h2>
<section id="delta-cq-boxplots" class="level3">
<h3 class="anchored" data-anchor-id="delta-cq-boxplots">8.2.1 Delta Cq boxplots</h3>
<section id="lifestage-comparisons" class="level4">
<h4 class="anchored" data-anchor-id="lifestage-comparisons">8.2.1.1 Lifestage comparisons</h4>
<div class="sourceCode" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create box plots for each comparison</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>unique_comparisons <span class="ot">&lt;-</span> <span class="fu">unique</span>(life_stage_t_test_results_df<span class="sc">$</span>comparison)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (comparison <span class="cf">in</span> unique_comparisons) {</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_boxplot_delta_Cq</span>(delta_Cq_df, comparison, life_stage_t_test_results_df)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-lifestage-1.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-lifestage-2.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-lifestage-3.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-lifestage-4.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-lifestage-5.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-lifestage-6.png" class="img-fluid"><!-- --></p>
</section>
</section>
<section id="conditioning-comparisons-1" class="level3">
<h3 class="anchored" data-anchor-id="conditioning-comparisons-1">8.2.2 Conditioning comparisons</h3>
<div class="sourceCode" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create box plots for each comparison</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>unique_comparisons <span class="ot">&lt;-</span> <span class="fu">unique</span>(conditioning_treatment_t_test_results_df<span class="sc">$</span>comparison)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (comparison <span class="cf">in</span> unique_comparisons) {</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_boxplot_conditioning</span>(delta_Cq_df, comparison, conditioning_treatment_t_test_results_df)</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-conditioning-1.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-conditioning-2.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-conditioning-3.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-conditioning-4.png" class="img-fluid"><!-- --></p>
</section>
<section id="acute-treatment-comparisons" class="level3">
<h3 class="anchored" data-anchor-id="acute-treatment-comparisons">8.2.3 Acute treatment comparisons</h3>
<div class="sourceCode" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create box plots for each comparison</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>unique_comparisons <span class="ot">&lt;-</span> <span class="fu">unique</span>(acute_treatment_t_test_results_df<span class="sc">$</span>comparison)</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (comparison <span class="cf">in</span> unique_comparisons) {</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_boxplot_acute</span>(delta_Cq_df, comparison, acute_treatment_t_test_results_df)</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-acute-treatment-1.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-acute-treatment-2.png" class="img-fluid"><!-- --></p>
</section>
<section id="acute-within-life-stage-conditioning" class="level3">
<h3 class="anchored" data-anchor-id="acute-within-life-stage-conditioning">8.2.4 Acute within life stage conditioning</h3>
<div class="sourceCode" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each comparison in the t-test results and create box plots</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (comparison <span class="cf">in</span> <span class="fu">unique</span>(acute_treatment_within_life.stages_conditioning_t_test_results_df<span class="sc">$</span>comparison)) {</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_boxplot_acute_conditioning</span>(delta_Cq_df, comparison, acute_treatment_within_life.stages_conditioning_t_test_results_df)</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/boxplots-acute-treatments-within-life-stages-conditioning-1.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/boxplots-acute-treatments-within-life-stages-conditioning-2.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/boxplots-acute-treatments-within-life-stages-conditioning-3.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/boxplots-acute-treatments-within-life-stages-conditioning-4.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/boxplots-acute-treatments-within-life-stages-conditioning-5.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/boxplots-acute-treatments-within-life-stages-conditioning-6.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/boxplots-acute-treatments-within-life-stages-conditioning-7.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/boxplots-acute-treatments-within-life-stages-conditioning-8.png" class="img-fluid"><!-- --></p>
</section>
</section>
</section>
<section id="delta-delta-cq" class="level1">
<h1>9 Delta delta Cq</h1>
<section id="calculations" class="level2">
<h2 class="anchored" data-anchor-id="calculations">9.1 Calculations</h2>
<section id="conditioning" class="level3">
<h3 class="anchored" data-anchor-id="conditioning">9.1.1 Conditioning</h3>
<div class="sourceCode" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta_delta_Cq</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>delta_delta_conditioning_fold_change <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, Target) <span class="sc">%&gt;%</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">treated_delta_Cq =</span> <span class="fu">mean</span>(delta_Cq[conditioning.treatment <span class="sc">==</span> <span class="st">"treated"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">control_delta_Cq =</span> <span class="fu">mean</span>(delta_Cq[conditioning.treatment <span class="sc">==</span> <span class="st">"control"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">delta_delta_Cq =</span> treated_delta_Cq <span class="sc">-</span> control_delta_Cq) <span class="sc">%&gt;%</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(life.stage, Target, delta_delta_Cq)</span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_conditioning_fold_change)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>gropd_df [28 × 3] (S3: grouped_df/tbl_df/tbl/data.frame)
 $ life.stage    : chr [1:28] "adult" "adult" "adult" "adult" ...
 $ Target        : chr [1:28] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
 $ delta_delta_Cq: num [1:28] -0.0779 -0.3116 0.941 0.7639 0.1955 ...
 - attr(*, "groups")= tibble [4 × 2] (S3: tbl_df/tbl/data.frame)
  ..$ life.stage: chr [1:4] "adult" "juvenile" "seed" "spat"
  ..$ .rows     : list&lt;int&gt; [1:4] 
  .. ..$ : int [1:7] 1 2 3 4 5 6 7
  .. ..$ : int [1:7] 8 9 10 11 12 13 14
  .. ..$ : int [1:7] 15 16 17 18 19 20 21
  .. ..$ : int [1:7] 22 23 24 25 26 27 28
  .. ..@ ptype: int(0) 
  ..- attr(*, ".drop")= logi TRUE</code></pre>
</section>
<section id="acute-treatment" class="level3">
<h3 class="anchored" data-anchor-id="acute-treatment">9.1.2 Acute treatment</h3>
<div class="sourceCode" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta_delta_Cq for acute treatment</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>delta_delta_Cq_acute_df <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, Target, acute.treatment) <span class="sc">%&gt;%</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">treated_delta_Cq =</span> <span class="fu">mean</span>(delta_Cq[conditioning.treatment <span class="sc">==</span> <span class="st">"treated"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">control_delta_Cq =</span> <span class="fu">mean</span>(delta_Cq[conditioning.treatment <span class="sc">==</span> <span class="st">"control"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">delta_delta_Cq =</span> treated_delta_Cq <span class="sc">-</span> control_delta_Cq) <span class="sc">%&gt;%</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(life.stage, Target, acute.treatment, delta_delta_Cq)</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_Cq_acute_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>gropd_df [56 × 4] (S3: grouped_df/tbl_df/tbl/data.frame)
 $ life.stage     : chr [1:56] "adult" "adult" "adult" "adult" ...
 $ Target         : chr [1:56] "ATPsynthase" "ATPsynthase" "DNMT1" "DNMT1" ...
 $ acute.treatment: chr [1:56] "ambient" "high" "ambient" "high" ...
 $ delta_delta_Cq : num [1:56] -0.112 -0.0438 -0.2467 -0.3765 0.9455 ...
 - attr(*, "groups")= tibble [28 × 3] (S3: tbl_df/tbl/data.frame)
  ..$ life.stage: chr [1:28] "adult" "adult" "adult" "adult" ...
  ..$ Target    : chr [1:28] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
  ..$ .rows     : list&lt;int&gt; [1:28] 
  .. ..$ : int [1:2] 1 2
  .. ..$ : int [1:2] 3 4
  .. ..$ : int [1:2] 5 6
  .. ..$ : int [1:2] 7 8
  .. ..$ : int [1:2] 9 10
  .. ..$ : int [1:2] 11 12
  .. ..$ : int [1:2] 13 14
  .. ..$ : int [1:2] 15 16
  .. ..$ : int [1:2] 17 18
  .. ..$ : int [1:2] 19 20
  .. ..$ : int [1:2] 21 22
  .. ..$ : int [1:2] 23 24
  .. ..$ : int [1:2] 25 26
  .. ..$ : int [1:2] 27 28
  .. ..$ : int [1:2] 29 30
  .. ..$ : int [1:2] 31 32
  .. ..$ : int [1:2] 33 34
  .. ..$ : int [1:2] 35 36
  .. ..$ : int [1:2] 37 38
  .. ..$ : int [1:2] 39 40
  .. ..$ : int [1:2] 41 42
  .. ..$ : int [1:2] 43 44
  .. ..$ : int [1:2] 45 46
  .. ..$ : int [1:2] 47 48
  .. ..$ : int [1:2] 49 50
  .. ..$ : int [1:2] 51 52
  .. ..$ : int [1:2] 53 54
  .. ..$ : int [1:2] 55 56
  .. ..@ ptype: int(0) 
  ..- attr(*, ".drop")= logi TRUE</code></pre>
</section>
<section id="life-stage" class="level3">
<h3 class="anchored" data-anchor-id="life-stage">9.1.3 Life stage</h3>
<div class="sourceCode" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta_delta_Cq for life stage comparisons</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>delta_delta_Cq_life_stage_df <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target, life.stage) <span class="sc">%&gt;%</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean_delta_Cq =</span> <span class="fu">mean</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> life.stage, <span class="at">values_from =</span> mean_delta_Cq) <span class="sc">%&gt;%</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">delta_delta_Cq_adult_vs_seed =</span> adult <span class="sc">-</span> seed,</span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">delta_delta_Cq_spat_vs_seed =</span> spat <span class="sc">-</span> seed,</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">delta_delta_Cq_adult_vs_spat =</span> adult <span class="sc">-</span> spat</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"delta_delta_Cq_"</span>), <span class="at">names_to =</span> <span class="st">"comparison"</span>, <span class="at">values_to =</span> <span class="st">"delta_delta_Cq"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(delta_delta_Cq))</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the structure of the resulting data frame</span></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_Cq_life_stage_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [21 × 7] (S3: tbl_df/tbl/data.frame)
 $ Target        : chr [1:21] "ATPsynthase" "ATPsynthase" "ATPsynthase" "DNMT1" ...
 $ adult         : num [1:21] 0.48 0.48 0.48 6.17 6.17 ...
 $ juvenile      : num [1:21] 0.732 0.732 0.732 6.32 6.32 ...
 $ seed          : num [1:21] 0.44 0.44 0.44 6.38 6.38 ...
 $ spat          : num [1:21] 0.324 0.324 0.324 6.488 6.488 ...
 $ comparison    : chr [1:21] "delta_delta_Cq_adult_vs_seed" "delta_delta_Cq_spat_vs_seed" "delta_delta_Cq_adult_vs_spat" "delta_delta_Cq_adult_vs_seed" ...
 $ delta_delta_Cq: num [1:21] 0.0405 -0.1157 0.1561 -0.2139 0.1065 ...</code></pre>
</section>
<section id="calculate-delta-delta-acute-treatments-within-lifestage-and-conditioning" class="level3">
<h3 class="anchored" data-anchor-id="calculate-delta-delta-acute-treatments-within-lifestage-and-conditioning">9.1.4 Calculate delta delta acute treatments within lifestage and conditioning</h3>
<div class="sourceCode" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta_delta_Cq for acute treatment comparisons within each life stage and conditioning treatment</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>delta_delta_Cq_acute_within_life_stage_conditioning_df <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, conditioning.treatment, Target, acute.treatment) <span class="sc">%&gt;%</span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean_delta_Cq =</span> <span class="fu">mean</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> acute.treatment, <span class="at">values_from =</span> mean_delta_Cq) <span class="sc">%&gt;%</span></span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">delta_delta_Cq_high_vs_ambient =</span> high <span class="sc">-</span> ambient) <span class="sc">%&gt;%</span></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"delta_delta_Cq_"</span>), <span class="at">names_to =</span> <span class="st">"comparison"</span>, <span class="at">values_to =</span> <span class="st">"delta_delta_Cq"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(delta_delta_Cq))</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the structure of the resulting data frame</span></span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_Cq_acute_within_life_stage_conditioning_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [56 × 7] (S3: tbl_df/tbl/data.frame)
 $ life.stage            : chr [1:56] "adult" "adult" "adult" "adult" ...
 $ conditioning.treatment: chr [1:56] "control" "control" "control" "control" ...
 $ Target                : chr [1:56] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
 $ ambient               : num [1:56] 0.566 6.448 3.944 1.259 5.207 ...
 $ high                  : num [1:56] 0.472 6.199 3.673 0.29 4.609 ...
 $ comparison            : chr [1:56] "delta_delta_Cq_high_vs_ambient" "delta_delta_Cq_high_vs_ambient" "delta_delta_Cq_high_vs_ambient" "delta_delta_Cq_high_vs_ambient" ...
 $ delta_delta_Cq        : num [1:56] -0.0946 -0.2492 -0.2715 -0.969 -0.5983 ...</code></pre>
</section>
<section id="calculate-the-fold-change-life-stage-comparison" class="level3">
<h3 class="anchored" data-anchor-id="calculate-the-fold-change-life-stage-comparison">9.1.5 Calculate the fold change life stage comparison</h3>
<div class="sourceCode" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate fold change and output to a new data frame</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>fold_change_life_stage_df <span class="ot">&lt;-</span> delta_delta_Cq_life_stage_df <span class="sc">%&gt;%</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq))</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the structure of the resulting data frame</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(fold_change_life_stage_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [21 × 8] (S3: tbl_df/tbl/data.frame)
 $ Target        : chr [1:21] "ATPsynthase" "ATPsynthase" "ATPsynthase" "DNMT1" ...
 $ adult         : num [1:21] 0.48 0.48 0.48 6.17 6.17 ...
 $ juvenile      : num [1:21] 0.732 0.732 0.732 6.32 6.32 ...
 $ seed          : num [1:21] 0.44 0.44 0.44 6.38 6.38 ...
 $ spat          : num [1:21] 0.324 0.324 0.324 6.488 6.488 ...
 $ comparison    : chr [1:21] "delta_delta_Cq_adult_vs_seed" "delta_delta_Cq_spat_vs_seed" "delta_delta_Cq_adult_vs_spat" "delta_delta_Cq_adult_vs_seed" ...
 $ delta_delta_Cq: num [1:21] 0.0405 -0.1157 0.1561 -0.2139 0.1065 ...
 $ fold_change   : num [1:21] 0.972 1.083 0.897 1.16 0.929 ...</code></pre>
</section>
<section id="calculate-the-fold-change-conditioning-comparison" class="level3">
<h3 class="anchored" data-anchor-id="calculate-the-fold-change-conditioning-comparison">9.1.6 Calculate the fold change conditioning comparison</h3>
<div class="sourceCode" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>delta_delta_conditioning_fold_change <span class="ot">&lt;-</span> delta_delta_conditioning_fold_change <span class="sc">%&gt;%</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Target, fold_change)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_conditioning_fold_change)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>gropd_df [28 × 3] (S3: grouped_df/tbl_df/tbl/data.frame)
 $ life.stage : chr [1:28] "adult" "adult" "adult" "adult" ...
 $ Target     : chr [1:28] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
 $ fold_change: num [1:28] 1.055 1.241 0.521 0.589 0.873 ...
 - attr(*, "groups")= tibble [4 × 2] (S3: tbl_df/tbl/data.frame)
  ..$ life.stage: chr [1:4] "adult" "juvenile" "seed" "spat"
  ..$ .rows     : list&lt;int&gt; [1:4] 
  .. ..$ : int [1:7] 1 2 3 4 5 6 7
  .. ..$ : int [1:7] 8 9 10 11 12 13 14
  .. ..$ : int [1:7] 15 16 17 18 19 20 21
  .. ..$ : int [1:7] 22 23 24 25 26 27 28
  .. ..@ ptype: int(0) 
  ..- attr(*, ".drop")= logi TRUE</code></pre>
</section>
<section id="calculate-the-fold-change-acute-comparison" class="level3">
<h3 class="anchored" data-anchor-id="calculate-the-fold-change-acute-comparison">9.1.7 Calculate the fold change acute comparison</h3>
<div class="sourceCode" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate fold change for acute treatment</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>delta_delta_acute_fold_change <span class="ot">&lt;-</span> delta_delta_Cq_acute_df <span class="sc">%&gt;%</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq)) <span class="sc">%&gt;%</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(life.stage, Target, acute.treatment, fold_change)</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the structure of the resulting data frame</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_acute_fold_change)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>gropd_df [56 × 4] (S3: grouped_df/tbl_df/tbl/data.frame)
 $ life.stage     : chr [1:56] "adult" "adult" "adult" "adult" ...
 $ Target         : chr [1:56] "ATPsynthase" "ATPsynthase" "DNMT1" "DNMT1" ...
 $ acute.treatment: chr [1:56] "ambient" "high" "ambient" "high" ...
 $ fold_change    : num [1:56] 1.081 1.031 1.186 1.298 0.519 ...
 - attr(*, "groups")= tibble [28 × 3] (S3: tbl_df/tbl/data.frame)
  ..$ life.stage: chr [1:28] "adult" "adult" "adult" "adult" ...
  ..$ Target    : chr [1:28] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
  ..$ .rows     : list&lt;int&gt; [1:28] 
  .. ..$ : int [1:2] 1 2
  .. ..$ : int [1:2] 3 4
  .. ..$ : int [1:2] 5 6
  .. ..$ : int [1:2] 7 8
  .. ..$ : int [1:2] 9 10
  .. ..$ : int [1:2] 11 12
  .. ..$ : int [1:2] 13 14
  .. ..$ : int [1:2] 15 16
  .. ..$ : int [1:2] 17 18
  .. ..$ : int [1:2] 19 20
  .. ..$ : int [1:2] 21 22
  .. ..$ : int [1:2] 23 24
  .. ..$ : int [1:2] 25 26
  .. ..$ : int [1:2] 27 28
  .. ..$ : int [1:2] 29 30
  .. ..$ : int [1:2] 31 32
  .. ..$ : int [1:2] 33 34
  .. ..$ : int [1:2] 35 36
  .. ..$ : int [1:2] 37 38
  .. ..$ : int [1:2] 39 40
  .. ..$ : int [1:2] 41 42
  .. ..$ : int [1:2] 43 44
  .. ..$ : int [1:2] 45 46
  .. ..$ : int [1:2] 47 48
  .. ..$ : int [1:2] 49 50
  .. ..$ : int [1:2] 51 52
  .. ..$ : int [1:2] 53 54
  .. ..$ : int [1:2] 55 56
  .. ..@ ptype: int(0) 
  ..- attr(*, ".drop")= logi TRUE</code></pre>
</section>
<section id="calculate-fold-change-acute-treatments-within-lifestage-and-conditioning" class="level3">
<h3 class="anchored" data-anchor-id="calculate-fold-change-acute-treatments-within-lifestage-and-conditioning">9.1.8 Calculate fold change acute treatments within lifestage and conditioning</h3>
<div class="sourceCode" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate fold change for acute treatment comparisons within each life stage and conditioning treatment</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>fold_change_acute_within_life_stage_conditioning_df <span class="ot">&lt;-</span> delta_delta_Cq_acute_within_life_stage_conditioning_df <span class="sc">%&gt;%</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq))</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the structure of the resulting data frame</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(fold_change_acute_within_life_stage_conditioning_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [56 × 8] (S3: tbl_df/tbl/data.frame)
 $ life.stage            : chr [1:56] "adult" "adult" "adult" "adult" ...
 $ conditioning.treatment: chr [1:56] "control" "control" "control" "control" ...
 $ Target                : chr [1:56] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
 $ ambient               : num [1:56] 0.566 6.448 3.944 1.259 5.207 ...
 $ high                  : num [1:56] 0.472 6.199 3.673 0.29 4.609 ...
 $ comparison            : chr [1:56] "delta_delta_Cq_high_vs_ambient" "delta_delta_Cq_high_vs_ambient" "delta_delta_Cq_high_vs_ambient" "delta_delta_Cq_high_vs_ambient" ...
 $ delta_delta_Cq        : num [1:56] -0.0946 -0.2492 -0.2715 -0.969 -0.5983 ...
 $ fold_change           : num [1:56] 1.07 1.19 1.21 1.96 1.51 ...</code></pre>
</section>
</section>
<section id="plotting-fold-changes" class="level2">
<h2 class="anchored" data-anchor-id="plotting-fold-changes">9.2 Plotting fold changes</h2>
<section id="acute-comparisons-within-lifestage-and-conditioning" class="level3">
<h3 class="anchored" data-anchor-id="acute-comparisons-within-lifestage-and-conditioning">9.2.1 Acute comparisons within lifestage and conditioning</h3>
<div class="sourceCode" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate bar plots for each group of comparison within each life stage and conditioning treatment</span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> fold_change_acute_within_life_stage_conditioning_df <span class="sc">%&gt;%</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">split</span>(<span class="fu">list</span>(.<span class="sc">$</span>life.stage, .<span class="sc">$</span>conditioning.treatment, .<span class="sc">$</span>comparison)) <span class="sc">%&gt;%</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(<span class="cf">function</span>(df) {</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>    life_stage <span class="ot">&lt;-</span> <span class="fu">unique</span>(df<span class="sc">$</span>life.stage)</span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>    conditioning_treatment <span class="ot">&lt;-</span> <span class="fu">unique</span>(df<span class="sc">$</span>conditioning.treatment)</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>    comparison_title <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta_delta_Cq_"</span>, <span class="st">""</span>, <span class="fu">unique</span>(df<span class="sc">$</span>comparison))</span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a>    comparison_title <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_vs_"</span>, <span class="st">" vs. "</span>, comparison_title)</span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> fold_change)) <span class="sc">+</span></span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Gene Expression -"</span>, life_stage, <span class="st">"-"</span>, conditioning_treatment, <span class="st">"-"</span>, comparison_title), </span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a>           <span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the plots</span></span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (plot <span class="cf">in</span> plot_list) {</span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-acute-within-lifestage-conditioning-1.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-acute-within-lifestage-conditioning-2.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-acute-within-lifestage-conditioning-3.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-acute-within-lifestage-conditioning-4.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-acute-within-lifestage-conditioning-5.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-acute-within-lifestage-conditioning-6.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-acute-within-lifestage-conditioning-7.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-acute-within-lifestage-conditioning-8.png" class="img-fluid"><!-- --></p>
</section>
<section id="life-stage-comparisons" class="level3">
<h3 class="anchored" data-anchor-id="life-stage-comparisons">9.2.2 Life stage comparisons</h3>
<div class="sourceCode" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate bar plots for each group of comparison</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> fold_change_life_stage_df <span class="sc">%&gt;%</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">split</span>(.<span class="sc">$</span>comparison) <span class="sc">%&gt;%</span></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(<span class="cf">function</span>(df) {</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>    comparison_title <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta_delta_Cq_"</span>, <span class="st">""</span>, <span class="fu">unique</span>(df<span class="sc">$</span>comparison))</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>    comparison_title <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_vs_"</span>, <span class="st">" vs. "</span>, comparison_title)</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> fold_change)) <span class="sc">+</span></span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Gene Expression -"</span>, comparison_title), <span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the plots</span></span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (plot <span class="cf">in</span> plot_list) {</span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-lifestages-1.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-lifestages-2.png" class="img-fluid"><!-- --><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-lifestages-3.png" class="img-fluid"><!-- --></p>
</section>
<section id="conditioning-comparisons-2" class="level3">
<h3 class="anchored" data-anchor-id="conditioning-comparisons-2">9.2.3 Conditioning comparisons</h3>
<p><img src="01.01-qPCR_files/figure-gfm/barplot-fold-change-conditioning-1.png" class="img-fluid"><!-- --></p>
</section>
<section id="line-plot-conditioning-comparisons-across-lifestages" class="level3">
<h3 class="anchored" data-anchor-id="line-plot-conditioning-comparisons-across-lifestages">9.2.4 Line plot conditioning comparisons across lifestages</h3>
<p><img src="01.01-qPCR_files/figure-gfm/lineplot-fold-change-conditioning-1.png" class="img-fluid"><!-- --></p>
</section>
<section id="acute-treatment-comparison" class="level3">
<h3 class="anchored" data-anchor-id="acute-treatment-comparison">9.2.5 Acute treatment comparison</h3>
<p><img src="01.01-qPCR_files/figure-gfm/barplot-fold-change-acute-1.png" class="img-fluid"><!-- --></p>
</section>
</section>
</section>
<section id="running-linear-models" class="level1">
<h1>10 Running linear models</h1>
<section id="anova-models" class="level2">
<h2 class="anchored" data-anchor-id="anova-models">10.1 ANOVA models</h2>
<div class="sourceCode" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(delta_Cq_df<span class="sc">$</span>delta_Cq)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-1-1.png" class="img-fluid"><!-- --></p>
<p>Run an anova model to test for effects of lifestage, conditioning, and acute treatment on delta Cq values for each target.</p>
<section id="atp-synthase" class="level3">
<h3 class="anchored" data-anchor-id="atp-synthase">10.1.1 ATP synthase</h3>
<p>ATP synthase is an enzyme complex that functions to synthesize adenosine triphosphate (ATP) from adenosine diphosphate (ADP) and inorganic phosphate (Pi), essentially generating the cell’s primary energy currency by harnessing the energy from a proton gradient across a membrane.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(car)</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(emmeans)</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>model<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"ATPsynthase"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> life.stage <span class="sc">*</span> conditioning.treatment <span class="sc">*</span> acute.treatment, <span class="at">data=</span>.)</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>                                                   Df Sum Sq Mean Sq F value
life.stage                                          3   2.66  0.8874   2.360
conditioning.treatment                              1   1.33  1.3317   3.542
acute.treatment                                     1   0.61  0.6073   1.615
life.stage:conditioning.treatment                   3   0.60  0.2012   0.535
life.stage:acute.treatment                          3   0.37  0.1221   0.325
conditioning.treatment:acute.treatment              1   0.11  0.1095   0.291
life.stage:conditioning.treatment:acute.treatment   3   0.48  0.1591   0.423
Residuals                                         104  39.10  0.3760        
                                                  Pr(&gt;F)  
life.stage                                        0.0757 .
conditioning.treatment                            0.0626 .
acute.treatment                                   0.2066  
life.stage:conditioning.treatment                 0.6592  
life.stage:acute.treatment                        0.8074  
conditioning.treatment:acute.treatment            0.5906  
life.stage:conditioning.treatment:acute.treatment 0.7367  
Residuals                                                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<div class="sourceCode" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-2-1.png" class="img-fluid"><!-- --></p>
<pre><code>[1] 117   4</code></pre>
<div class="sourceCode" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model<span class="sc">$</span>residuals <span class="sc">~</span> life.stage<span class="sc">*</span>conditioning.treatment<span class="sc">*</span>acute.treatment, <span class="at">data=</span>delta_Cq_df<span class="sc">%&gt;%</span><span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"ATPsynthase"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
       Df F value Pr(&gt;F)
group  15  1.2733 0.2324
      104               </code></pre>
<p>No significant effects.</p>
</section>
<section id="dnmt1" class="level3">
<h3 class="anchored" data-anchor-id="dnmt1">10.1.2 DNMT1</h3>
<p>The DNMT1 gene provides instructions for making an enzyme called DNA methyltransferase 1. This enzyme is involved in DNA methylation, which is the addition of methyl groups, consisting of one carbon atom and three hydrogen atoms, to DNA molecules.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>model<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"DNMT1"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> life.stage <span class="sc">*</span> conditioning.treatment <span class="sc">*</span> acute.treatment, <span class="at">data=</span>.)</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>                                                   Df Sum Sq Mean Sq F value
life.stage                                          3   1.47   0.490   0.347
conditioning.treatment                              1   0.25   0.252   0.178
acute.treatment                                     1   2.88   2.885   2.041
life.stage:conditioning.treatment                   3  10.35   3.451   2.442
life.stage:acute.treatment                          3   4.53   1.511   1.069
conditioning.treatment:acute.treatment              1   0.01   0.011   0.008
life.stage:conditioning.treatment:acute.treatment   3   9.86   3.288   2.326
Residuals                                         102 144.16   1.413        
                                                  Pr(&gt;F)  
life.stage                                        0.7915  
conditioning.treatment                            0.6736  
acute.treatment                                   0.1561  
life.stage:conditioning.treatment                 0.0685 .
life.stage:acute.treatment                        0.3657  
conditioning.treatment:acute.treatment            0.9288  
life.stage:conditioning.treatment:acute.treatment 0.0791 .
Residuals                                                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<div class="sourceCode" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-3-1.png" class="img-fluid"><!-- --></p>
<pre><code>[1] 115  10</code></pre>
<div class="sourceCode" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model<span class="sc">$</span>residuals <span class="sc">~</span> life.stage<span class="sc">*</span>conditioning.treatment<span class="sc">*</span>acute.treatment, <span class="at">data=</span>delta_Cq_df<span class="sc">%&gt;%</span><span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"DNMT1"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
       Df F value Pr(&gt;F)
group  15  1.0926 0.3727
      102               </code></pre>
<p>No significant effects.</p>
</section>
<section id="hsp70" class="level3">
<h3 class="anchored" data-anchor-id="hsp70">10.1.3 HSP70</h3>
<p>Heat Shock Protein 70 (Hsp70) is a molecular chaperone that plays crucial roles in maintaining cellular protein homeostasis and protecting cells from stress.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>model<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"HSP70"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> life.stage <span class="sc">*</span> conditioning.treatment <span class="sc">*</span> acute.treatment, <span class="at">data=</span>.)</span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>                                                   Df Sum Sq Mean Sq F value
life.stage                                          3   23.4   7.814   2.464
conditioning.treatment                              1    3.7   3.729   1.176
acute.treatment                                     1    2.1   2.136   0.674
life.stage:conditioning.treatment                   3   14.2   4.735   1.493
life.stage:acute.treatment                          3   14.9   4.966   1.566
conditioning.treatment:acute.treatment              1   19.0  18.966   5.980
life.stage:conditioning.treatment:acute.treatment   3   17.2   5.749   1.813
Residuals                                         104  329.8   3.172        
                                                  Pr(&gt;F)  
life.stage                                        0.0666 .
conditioning.treatment                            0.2807  
acute.treatment                                   0.4137  
life.stage:conditioning.treatment                 0.2208  
life.stage:acute.treatment                        0.2022  
conditioning.treatment:acute.treatment            0.0161 *
life.stage:conditioning.treatment:acute.treatment 0.1494  
Residuals                                                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<div class="sourceCode" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-4-1.png" class="img-fluid"><!-- --></p>
<pre><code>[1] 103  64</code></pre>
<div class="sourceCode" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model<span class="sc">$</span>residuals <span class="sc">~</span> life.stage<span class="sc">*</span>conditioning.treatment<span class="sc">*</span>acute.treatment, <span class="at">data=</span>delta_Cq_df<span class="sc">%&gt;%</span><span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"HSP70"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
       Df F value Pr(&gt;F)
group  15  0.5223 0.9231
      104               </code></pre>
<div class="sourceCode" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>emm<span class="ot">&lt;-</span><span class="fu">emmeans</span>(model, <span class="sc">~</span> conditioning.treatment<span class="sc">:</span>acute.treatment <span class="sc">|</span> life.stage)</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(emm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>life.stage = adult:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient  -0.9455 0.890 104  -1.062  0.7134
 control ambient - control high      0.2715 0.890 104   0.305  0.9901
 control ambient - treated high     -0.6649 0.890 104  -0.747  0.8779
 treated ambient - control high      1.2170 0.890 104   1.367  0.5230
 treated ambient - treated high      0.2806 0.890 104   0.315  0.9891
 control high - treated high        -0.9364 0.890 104  -1.052  0.7195

life.stage = juvenile:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient  -1.6377 0.939 104  -1.745  0.3061
 control ambient - control high      0.1868 0.800 104   0.233  0.9955
 control ambient - treated high      0.2839 0.840 104   0.338  0.9866
 treated ambient - control high      1.8245 0.904 104   2.019  0.1878
 treated ambient - treated high      1.9216 0.939 104   2.047  0.1776
 control high - treated high         0.0971 0.800 104   0.121  0.9994

life.stage = seed:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient  -0.9181 0.818 104  -1.122  0.6769
 control ambient - control high     -1.4793 0.920 104  -1.608  0.3784
 control ambient - treated high     -1.2825 0.920 104  -1.395  0.5056
 treated ambient - control high     -0.5612 0.939 104  -0.598  0.9325
 treated ambient - treated high     -0.3644 0.939 104  -0.388  0.9800
 control high - treated high         0.1967 1.030 104   0.191  0.9975

life.stage = spat:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient  -1.2358 1.080 104  -1.146  0.6620
 control ambient - control high     -1.8735 1.150 104  -1.630  0.3666
 control ambient - treated high      1.4741 0.991 104   1.488  0.4485
 treated ambient - control high     -0.6376 1.190 104  -0.534  0.9506
 treated ambient - treated high      2.7099 1.040 104   2.599  0.0516
 control high - treated high         3.3475 1.120 104   2.999  0.0176

P value adjustment: tukey method for comparing a family of 4 estimates </code></pre>
<p>Significant effect of conditioning x acute treatment.</p>
</section>
<section id="hsp90" class="level3">
<h3 class="anchored" data-anchor-id="hsp90">10.1.4 HSP90</h3>
<p>Heat shock protein 90 (Hsp90) is a molecular chaperone that helps proteins fold, mature, and remain active. Hsp90 also helps regulate signaling networks and is involved in many cellular processes.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>model<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"HSP90"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> life.stage <span class="sc">*</span> conditioning.treatment <span class="sc">*</span> acute.treatment, <span class="at">data=</span>.)</span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>                                                   Df Sum Sq Mean Sq F value
life.stage                                          3  27.25   9.083  13.042
conditioning.treatment                              1   0.66   0.662   0.950
acute.treatment                                     1  17.57  17.570  25.230
life.stage:conditioning.treatment                   3   5.61   1.870   2.686
life.stage:acute.treatment                          3   3.89   1.298   1.864
conditioning.treatment:acute.treatment              1   1.09   1.088   1.562
life.stage:conditioning.treatment:acute.treatment   3   3.52   1.173   1.684
Residuals                                         104  72.43   0.696        
                                                    Pr(&gt;F)    
life.stage                                        2.70e-07 ***
conditioning.treatment                              0.3319    
acute.treatment                                   2.12e-06 ***
life.stage:conditioning.treatment                   0.0504 .  
life.stage:acute.treatment                          0.1402    
conditioning.treatment:acute.treatment              0.2142    
life.stage:conditioning.treatment:acute.treatment   0.1749    
Residuals                                                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<div class="sourceCode" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-5-1.png" class="img-fluid"><!-- --></p>
<pre><code>[1]   4 103</code></pre>
<div class="sourceCode" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model<span class="sc">$</span>residuals <span class="sc">~</span> life.stage<span class="sc">*</span>conditioning.treatment<span class="sc">*</span>acute.treatment, <span class="at">data=</span>delta_Cq_df<span class="sc">%&gt;%</span><span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"HSP90"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
       Df F value Pr(&gt;F)
group  15  1.1241 0.3447
      104               </code></pre>
<div class="sourceCode" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>emm<span class="ot">&lt;-</span><span class="fu">emmeans</span>(model, <span class="sc">~</span> conditioning.treatment<span class="sc">:</span>acute.treatment <span class="sc">|</span> life.stage)</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(emm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>life.stage = adult:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient  -0.2939 0.417 104  -0.704  0.8951
 control ambient - control high      0.9690 0.417 104   2.322  0.0995
 control ambient - treated high     -0.2648 0.417 104  -0.635  0.9206
 treated ambient - control high      1.2630 0.417 104   3.027  0.0162
 treated ambient - treated high      0.0292 0.417 104   0.070  0.9999
 control high - treated high        -1.2338 0.417 104  -2.957  0.0198

life.stage = juvenile:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient   0.3117 0.440 104   0.709  0.8935
 control ambient - control high      0.9955 0.375 104   2.654  0.0448
 control ambient - treated high      0.8158 0.393 104   2.074  0.1686
 treated ambient - control high      0.6838 0.424 104   1.614  0.3750
 treated ambient - treated high      0.5041 0.440 104   1.146  0.6619
 control high - treated high        -0.1797 0.375 104  -0.479  0.9636

life.stage = seed:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient  -0.0352 0.383 104  -0.092  0.9997
 control ambient - control high      0.7954 0.431 104   1.846  0.2581
 control ambient - treated high      0.0628 0.431 104   0.146  0.9989
 treated ambient - control high      0.8307 0.440 104   1.889  0.2392
 treated ambient - treated high      0.0981 0.440 104   0.223  0.9961
 control high - treated high        -0.7326 0.482 104  -1.520  0.4290

life.stage = spat:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient  -0.1549 0.505 104  -0.306  0.9900
 control ambient - control high      0.8926 0.539 104   1.657  0.3516
 control ambient - treated high      1.7849 0.464 104   3.845  0.0012
 treated ambient - control high      1.0475 0.560 104   1.871  0.2468
 treated ambient - treated high      1.9398 0.489 104   3.970  0.0008
 control high - treated high         0.8923 0.523 104   1.706  0.3258

P value adjustment: tukey method for comparing a family of 4 estimates </code></pre>
<p>Significant effect of lifestage x acute treatment, lifestage x conditioning treatment, acute treatment, and lifestage.</p>
</section>
<section id="cgas" class="level3">
<h3 class="anchored" data-anchor-id="cgas">10.1.5 cGAS</h3>
<p>The cGAS gene is involved in several processes, including cellular response to exogenous dsRNA, positive regulation of intracellular signal transduction, and regulation of defense response.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>model<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"cGAS"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> life.stage <span class="sc">*</span> conditioning.treatment <span class="sc">*</span> acute.treatment, <span class="at">data=</span>.)</span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>                                                   Df Sum Sq Mean Sq F value
life.stage                                          3   5.64   1.881   1.302
conditioning.treatment                              1   0.55   0.545   0.377
acute.treatment                                     1   2.43   2.429   1.681
life.stage:conditioning.treatment                   3   1.32   0.440   0.305
life.stage:acute.treatment                          3  12.86   4.286   2.965
conditioning.treatment:acute.treatment              1   0.31   0.311   0.215
life.stage:conditioning.treatment:acute.treatment   3   4.65   1.550   1.073
Residuals                                         104 150.32   1.445        
                                                  Pr(&gt;F)  
life.stage                                        0.2779  
conditioning.treatment                            0.5404  
acute.treatment                                   0.1977  
life.stage:conditioning.treatment                 0.8220  
life.stage:acute.treatment                        0.0355 *
conditioning.treatment:acute.treatment            0.6436  
life.stage:conditioning.treatment:acute.treatment 0.3641  
Residuals                                                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<div class="sourceCode" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-6-1.png" class="img-fluid"><!-- --></p>
<pre><code>[1]  2 64</code></pre>
<div class="sourceCode" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model<span class="sc">$</span>residuals <span class="sc">~</span> life.stage<span class="sc">*</span>conditioning.treatment<span class="sc">*</span>acute.treatment, <span class="at">data=</span>delta_Cq_df<span class="sc">%&gt;%</span><span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"cGAS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
       Df F value  Pr(&gt;F)  
group  15  1.5918 0.08854 .
      104                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<div class="sourceCode" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>emm<span class="ot">&lt;-</span><span class="fu">emmeans</span>(model, <span class="sc">~</span> conditioning.treatment<span class="sc">:</span>acute.treatment <span class="sc">|</span> life.stage)</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(emm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>life.stage = adult:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient   0.0739 0.601 104   0.123  0.9993
 control ambient - control high      0.5983 0.601 104   0.995  0.7524
 control ambient - treated high      0.1334 0.601 104   0.222  0.9961
 treated ambient - control high      0.5244 0.601 104   0.872  0.8191
 treated ambient - treated high      0.0595 0.601 104   0.099  0.9996
 control high - treated high        -0.4649 0.601 104  -0.773  0.8663

life.stage = juvenile:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient   0.3421 0.634 104   0.540  0.9490
 control ambient - control high     -0.1441 0.540 104  -0.267  0.9933
 control ambient - treated high      0.0553 0.567 104   0.098  0.9997
 treated ambient - control high     -0.4862 0.610 104  -0.797  0.8557
 treated ambient - treated high     -0.2869 0.634 104  -0.453  0.9690
 control high - treated high         0.1994 0.540 104   0.369  0.9828

life.stage = seed:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient   0.1936 0.552 104   0.350  0.9852
 control ambient - control high     -1.5259 0.621 104  -2.458  0.0728
 control ambient - treated high     -1.0451 0.621 104  -1.683  0.3376
 treated ambient - control high     -1.7195 0.634 104  -2.714  0.0384
 treated ambient - treated high     -1.2387 0.634 104  -1.955  0.2120
 control high - treated high         0.4808 0.694 104   0.693  0.8997

life.stage = spat:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient  -0.6338 0.728 104  -0.871  0.8200
 control ambient - control high     -0.8533 0.776 104  -1.100  0.6906
 control ambient - treated high      0.2963 0.669 104   0.443  0.9708
 treated ambient - control high     -0.2195 0.806 104  -0.272  0.9929
 treated ambient - treated high      0.9301 0.704 104   1.321  0.5516
 control high - treated high         1.1496 0.754 104   1.526  0.4260

P value adjustment: tukey method for comparing a family of 4 estimates </code></pre>
<p>Significant effect of lifestage x acute treatment and lifestage.</p>
</section>
<section id="citrate-synthase" class="level3">
<h3 class="anchored" data-anchor-id="citrate-synthase">10.1.6 Citrate synthase</h3>
<p>Citrate synthase is important for energy production in the TCA cycle and is linked to the electron transport chain. It is also used as an enzyme marker for intact mitochondria.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>model<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"citrate synthase"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> life.stage <span class="sc">*</span> conditioning.treatment <span class="sc">*</span> acute.treatment, <span class="at">data=</span>.)</span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>                                                   Df Sum Sq Mean Sq F value
life.stage                                          3  10.15   3.383   6.975
conditioning.treatment                              1   1.31   1.305   2.692
acute.treatment                                     1   6.78   6.777  13.975
life.stage:conditioning.treatment                   3   0.47   0.158   0.325
life.stage:acute.treatment                          3   7.74   2.580   5.320
conditioning.treatment:acute.treatment              1   1.10   1.104   2.276
life.stage:conditioning.treatment:acute.treatment   3   0.13   0.045   0.092
Residuals                                         104  50.43   0.485        
                                                    Pr(&gt;F)    
life.stage                                        0.000254 ***
conditioning.treatment                            0.103872    
acute.treatment                                   0.000303 ***
life.stage:conditioning.treatment                 0.807157    
life.stage:acute.treatment                        0.001888 ** 
conditioning.treatment:acute.treatment            0.134455    
life.stage:conditioning.treatment:acute.treatment 0.964379    
Residuals                                                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<div class="sourceCode" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-7-1.png" class="img-fluid"><!-- --></p>
<pre><code>[1] 117  15</code></pre>
<div class="sourceCode" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model<span class="sc">$</span>residuals <span class="sc">~</span> life.stage<span class="sc">*</span>conditioning.treatment<span class="sc">*</span>acute.treatment, <span class="at">data=</span>delta_Cq_df<span class="sc">%&gt;%</span><span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"citrate synthase"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
       Df F value  Pr(&gt;F)  
group  15  2.0302 0.01964 *
      104                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<div class="sourceCode" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>emm<span class="ot">&lt;-</span><span class="fu">emmeans</span>(model, <span class="sc">~</span> conditioning.treatment<span class="sc">:</span>acute.treatment <span class="sc">|</span> life.stage)</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(emm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>life.stage = adult:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient  0.31754 0.348 104   0.912  0.7985
 control ambient - control high     0.30936 0.348 104   0.888  0.8108
 control ambient - treated high     0.14178 0.348 104   0.407  0.9771
 treated ambient - control high    -0.00818 0.348 104  -0.023  1.0000
 treated ambient - treated high    -0.17576 0.348 104  -0.505  0.9578
 control high - treated high       -0.16758 0.348 104  -0.481  0.9631

life.stage = juvenile:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient  0.69618 0.367 104   1.897  0.2357
 control ambient - control high     0.08720 0.313 104   0.279  0.9924
 control ambient - treated high     0.25576 0.328 104   0.779  0.8638
 treated ambient - control high    -0.60898 0.353 104  -1.723  0.3170
 treated ambient - treated high    -0.44042 0.367 104  -1.200  0.6282
 control high - treated high        0.16856 0.313 104   0.539  0.9494

life.stage = seed:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient  0.36475 0.320 104   1.140  0.6657
 control ambient - control high    -0.97515 0.360 104  -2.712  0.0386
 control ambient - treated high    -0.88377 0.360 104  -2.458  0.0728
 treated ambient - control high    -1.33990 0.367 104  -3.651  0.0023
 treated ambient - treated high    -1.24852 0.367 104  -3.402  0.0052
 control high - treated high        0.09138 0.402 104   0.227  0.9958

life.stage = spat:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient  0.37761 0.422 104   0.895  0.8072
 control ambient - control high    -0.89118 0.450 104  -1.983  0.2012
 control ambient - treated high    -0.70657 0.387 104  -1.824  0.2681
 treated ambient - control high    -1.26879 0.467 104  -2.716  0.0382
 treated ambient - treated high    -1.08418 0.408 104  -2.659  0.0443
 control high - treated high        0.18460 0.436 104   0.423  0.9744

P value adjustment: tukey method for comparing a family of 4 estimates </code></pre>
<p>Significant effect of acute treatment and lifestage x acute treatment.</p>
</section>
<section id="viperin" class="level3">
<h3 class="anchored" data-anchor-id="viperin">10.1.7 Viperin</h3>
<p>Viperin limits replication of DNA and RNA viruses.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>model<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"viperin"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> life.stage <span class="sc">*</span> conditioning.treatment <span class="sc">*</span> acute.treatment, <span class="at">data=</span>.)</span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>                                                   Df Sum Sq Mean Sq F value
life.stage                                          3  91.90  30.635  15.749
conditioning.treatment                              1   2.08   2.080   1.069
acute.treatment                                     1   0.76   0.760   0.391
life.stage:conditioning.treatment                   3   9.03   3.011   1.548
life.stage:acute.treatment                          3   8.58   2.861   1.471
conditioning.treatment:acute.treatment              1   2.01   2.010   1.033
life.stage:conditioning.treatment:acute.treatment   3   0.94   0.313   0.161
Residuals                                         102 198.41   1.945        
                                                    Pr(&gt;F)    
life.stage                                        1.73e-08 ***
conditioning.treatment                               0.304    
acute.treatment                                      0.533    
life.stage:conditioning.treatment                    0.207    
life.stage:acute.treatment                           0.227    
conditioning.treatment:acute.treatment               0.312    
life.stage:conditioning.treatment:acute.treatment    0.922    
Residuals                                                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<div class="sourceCode" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-8-1.png" class="img-fluid"><!-- --></p>
<pre><code>[1] 115   3</code></pre>
<div class="sourceCode" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model<span class="sc">$</span>residuals <span class="sc">~</span> life.stage<span class="sc">*</span>conditioning.treatment<span class="sc">*</span>acute.treatment, <span class="at">data=</span>delta_Cq_df<span class="sc">%&gt;%</span><span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"viperin"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
       Df F value Pr(&gt;F)
group  15  1.4856 0.1245
      102               </code></pre>
<div class="sourceCode" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>emm<span class="ot">&lt;-</span><span class="fu">emmeans</span>(model, <span class="sc">~</span> conditioning.treatment<span class="sc">:</span>acute.treatment <span class="sc">|</span> life.stage)</span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(emm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>life.stage = adult:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient   0.1683 0.697 102   0.241  0.9950
 control ambient - control high      0.5763 0.697 102   0.826  0.8418
 control ambient - treated high      0.2376 0.697 102   0.341  0.9863
 treated ambient - control high      0.4081 0.697 102   0.585  0.9363
 treated ambient - treated high      0.0693 0.697 102   0.099  0.9996
 control high - treated high        -0.3388 0.697 102  -0.486  0.9621

life.stage = juvenile:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient   1.5727 0.735 102   2.140  0.1477
 control ambient - control high      0.6795 0.627 102   1.084  0.7001
 control ambient - treated high      1.2753 0.657 102   1.940  0.2181
 treated ambient - control high     -0.8932 0.708 102  -1.262  0.5892
 treated ambient - treated high     -0.2974 0.735 102  -0.405  0.9775
 control high - treated high         0.5958 0.627 102   0.950  0.7777

life.stage = seed:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient  -0.1371 0.678 102  -0.202  0.9971
 control ambient - control high     -0.5242 0.735 102  -0.713  0.8917
 control ambient - treated high     -1.1583 0.735 102  -1.576  0.3970
 treated ambient - control high     -0.3872 0.753 102  -0.514  0.9556
 treated ambient - treated high     -1.0212 0.753 102  -1.356  0.5298
 control high - treated high        -0.6341 0.805 102  -0.787  0.8600

life.stage = spat:
 contrast                          estimate    SE  df t.ratio p.value
 control ambient - treated ambient   0.5456 0.845 102   0.646  0.9167
 control ambient - control high     -0.9091 0.900 102  -1.010  0.7440
 control ambient - treated high     -0.2650 0.776 102  -0.341  0.9862
 treated ambient - control high     -1.4547 0.936 102  -1.555  0.4090
 treated ambient - treated high     -0.8105 0.817 102  -0.993  0.7540
 control high - treated high         0.6442 0.874 102   0.737  0.8820

P value adjustment: tukey method for comparing a family of 4 estimates </code></pre>
<p>Significant effect of acute treatment and lifestage x acute treatment.</p>
<p>All tests pass normality and homogeneity of variance. ANOVA tests are appropriate.</p>
</section>
</section>
<section id="plotting-1" class="level2">
<h2 class="anchored" data-anchor-id="plotting-1">10.2 Plotting</h2>
<p>Display plot of acute x conditioning treatment faceted for each lifestage for each target.</p>
<p>Show sample size for each group.</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, acute.treatment, conditioning.treatment)<span class="sc">%&gt;%</span></span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">length</span>(<span class="fu">unique</span>(Sample)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code># A tibble: 16 × 4
# Groups:   life.stage, acute.treatment [8]
   life.stage acute.treatment conditioning.treatment     n
   &lt;chr&gt;      &lt;chr&gt;           &lt;chr&gt;                  &lt;int&gt;
 1 adult      ambient         control                    8
 2 adult      ambient         treated                    8
 3 adult      high            control                    8
 4 adult      high            treated                    8
 5 juvenile   ambient         control                    9
 6 juvenile   ambient         treated                    6
 7 juvenile   high            control                   11
 8 juvenile   high            treated                    9
 9 seed       ambient         control                   10
10 seed       ambient         treated                    9
11 seed       high            control                    6
12 seed       high            treated                    6
13 spat       ambient         control                    6
14 spat       ambient         treated                    5
15 spat       high            control                    4
16 spat       high            treated                    7</code></pre>
<p>Note that sample sizes for spat, seed, or juveniles that are &lt;6 are due to lack of RNA isolated from the samples.</p>
<p>Adults were all sampled from one family (Pink). Previous flow cytometry was unclear as to diploid or triploid classification. We have extracted DNA that we can use to verify ploidy.</p>
<section id="atp-synthase-no-effects" class="level3">
<h3 class="anchored" data-anchor-id="atp-synthase-no-effects">10.2.1 ATP synthase: No effects</h3>
<div class="sourceCode" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>plot1<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"ATPsynthase"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target, life.stage, acute.treatment, conditioning.treatment)<span class="sc">%&gt;%</span></span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="at">se=</span><span class="fu">sd</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>)<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">length</span>(delta_Cq)))<span class="sc">%&gt;%</span></span>
<span id="cb109-6"><a href="#cb109-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb109-7"><a href="#cb109-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>acute.treatment, <span class="at">y=</span>mean, <span class="at">colour=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb109-8"><a href="#cb109-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>life.stage)<span class="sc">+</span></span>
<span id="cb109-9"><a href="#cb109-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"orange"</span>))<span class="sc">+</span></span>
<span id="cb109-10"><a href="#cb109-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb109-11"><a href="#cb109-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb109-12"><a href="#cb109-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean<span class="sc">-</span>se, <span class="at">ymax=</span>mean<span class="sc">+</span>se), <span class="at">width=</span><span class="fl">0.1</span>)<span class="sc">+</span></span>
<span id="cb109-13"><a href="#cb109-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"ATP synthase (no sign. effects)"</span>)<span class="sc">+</span></span>
<span id="cb109-14"><a href="#cb109-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Delta Cq"</span>)<span class="sc">+</span></span>
<span id="cb109-15"><a href="#cb109-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.5</span>,<span class="fl">1.5</span>)<span class="sc">+</span></span>
<span id="cb109-16"><a href="#cb109-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>)<span class="sc">+</span></span>
<span id="cb109-17"><a href="#cb109-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>();plot1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-10-1.png" class="img-fluid"><!-- --></p>
</section>
<section id="dnmt1-no-effects" class="level3">
<h3 class="anchored" data-anchor-id="dnmt1-no-effects">10.2.2 DNMT1: No effects</h3>
<div class="sourceCode" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>plot2<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"DNMT1"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target, life.stage, acute.treatment, conditioning.treatment)<span class="sc">%&gt;%</span></span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="at">se=</span><span class="fu">sd</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>)<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">length</span>(delta_Cq)))<span class="sc">%&gt;%</span></span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>acute.treatment, <span class="at">y=</span>mean, <span class="at">colour=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>life.stage)<span class="sc">+</span></span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb110-10"><a href="#cb110-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"orange"</span>))<span class="sc">+</span></span>
<span id="cb110-11"><a href="#cb110-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb110-12"><a href="#cb110-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean<span class="sc">-</span>se, <span class="at">ymax=</span>mean<span class="sc">+</span>se), <span class="at">width=</span><span class="fl">0.1</span>)<span class="sc">+</span></span>
<span id="cb110-13"><a href="#cb110-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"DMNT1 (no sign. effects)"</span>)<span class="sc">+</span></span>
<span id="cb110-14"><a href="#cb110-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Delta Cq"</span>)<span class="sc">+</span></span>
<span id="cb110-15"><a href="#cb110-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>)<span class="sc">+</span></span>
<span id="cb110-16"><a href="#cb110-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">9</span>)<span class="sc">+</span></span>
<span id="cb110-17"><a href="#cb110-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>();plot2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-11-1.png" class="img-fluid"><!-- --></p>
</section>
<section id="hsp70-significant-effect-of-conditioning-x-acute-treatment." class="level3">
<h3 class="anchored" data-anchor-id="hsp70-significant-effect-of-conditioning-x-acute-treatment.">10.2.3 HSP70: Significant effect of conditioning x acute treatment.</h3>
<div class="sourceCode" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>plot3<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"HSP70"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target, life.stage, acute.treatment, conditioning.treatment)<span class="sc">%&gt;%</span></span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="at">se=</span><span class="fu">sd</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>)<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">length</span>(delta_Cq)))<span class="sc">%&gt;%</span></span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>acute.treatment, <span class="at">y=</span>mean, <span class="at">colour=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb111-8"><a href="#cb111-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>life.stage)<span class="sc">+</span></span>
<span id="cb111-9"><a href="#cb111-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"orange"</span>))<span class="sc">+</span></span>
<span id="cb111-10"><a href="#cb111-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb111-11"><a href="#cb111-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb111-12"><a href="#cb111-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean<span class="sc">-</span>se, <span class="at">ymax=</span>mean<span class="sc">+</span>se), <span class="at">width=</span><span class="fl">0.1</span>)<span class="sc">+</span></span>
<span id="cb111-13"><a href="#cb111-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"HSP70  (sign. conditioning x acute treatment)"</span>)<span class="sc">+</span></span>
<span id="cb111-14"><a href="#cb111-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Delta Cq"</span>)<span class="sc">+</span></span>
<span id="cb111-15"><a href="#cb111-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb111-16"><a href="#cb111-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>)<span class="sc">+</span></span>
<span id="cb111-17"><a href="#cb111-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>();plot3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-12-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a>plot3a<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"HSP70"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target, acute.treatment, conditioning.treatment)<span class="sc">%&gt;%</span></span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="at">se=</span><span class="fu">sd</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>)<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">length</span>(delta_Cq)))<span class="sc">%&gt;%</span></span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>acute.treatment, <span class="at">y=</span>mean, <span class="at">colour=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"orange"</span>))<span class="sc">+</span></span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean<span class="sc">-</span>se, <span class="at">ymax=</span>mean<span class="sc">+</span>se), <span class="at">width=</span><span class="fl">0.1</span>)<span class="sc">+</span></span>
<span id="cb112-12"><a href="#cb112-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"HSP70  (sign. conditioning x acute treatment)"</span>)<span class="sc">+</span></span>
<span id="cb112-13"><a href="#cb112-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Delta Cq"</span>)<span class="sc">+</span></span>
<span id="cb112-14"><a href="#cb112-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>)<span class="sc">+</span></span>
<span id="cb112-15"><a href="#cb112-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb112-16"><a href="#cb112-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>();plot3a</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-12-2.png" class="img-fluid"><!-- --></p>
</section>
<section id="hsp90-significant-effect-of-lifestage-x-acute-treatment-lifestage-x-conditioning-treatment-acute-treatment-and-lifestage." class="level3">
<h3 class="anchored" data-anchor-id="hsp90-significant-effect-of-lifestage-x-acute-treatment-lifestage-x-conditioning-treatment-acute-treatment-and-lifestage.">10.2.4 HSP90: Significant effect of lifestage x acute treatment, lifestage x conditioning treatment, acute treatment, and lifestage.</h3>
<div class="sourceCode" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>plot4<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"HSP90"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target, life.stage, acute.treatment, conditioning.treatment)<span class="sc">%&gt;%</span></span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="at">se=</span><span class="fu">sd</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>)<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">length</span>(delta_Cq)))<span class="sc">%&gt;%</span></span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>acute.treatment, <span class="at">y=</span>mean, <span class="at">colour=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>life.stage)<span class="sc">+</span></span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"orange"</span>))<span class="sc">+</span></span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean<span class="sc">-</span>se, <span class="at">ymax=</span>mean<span class="sc">+</span>se), <span class="at">width=</span><span class="fl">0.1</span>)<span class="sc">+</span></span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"HSP90  (sign. lifestage x acute &amp; lifestage x conditioning treatment)"</span>)<span class="sc">+</span></span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Delta Cq"</span>)<span class="sc">+</span></span>
<span id="cb113-15"><a href="#cb113-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="fl">2.5</span>)<span class="sc">+</span></span>
<span id="cb113-16"><a href="#cb113-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>)<span class="sc">+</span></span>
<span id="cb113-17"><a href="#cb113-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>();plot4</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-13-1.png" class="img-fluid"><!-- --></p>
</section>
<section id="cgas-significant-effect-of-lifestage-x-acute-treatment-and-lifestage." class="level3">
<h3 class="anchored" data-anchor-id="cgas-significant-effect-of-lifestage-x-acute-treatment-and-lifestage.">10.2.5 cGAS: Significant effect of lifestage x acute treatment and lifestage.</h3>
<div class="sourceCode" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>plot5<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"cGAS"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target, life.stage, acute.treatment, conditioning.treatment)<span class="sc">%&gt;%</span></span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="at">se=</span><span class="fu">sd</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>)<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">length</span>(delta_Cq)))<span class="sc">%&gt;%</span></span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>acute.treatment, <span class="at">y=</span>mean, <span class="at">colour=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>life.stage)<span class="sc">+</span></span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"orange"</span>))<span class="sc">+</span></span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb114-12"><a href="#cb114-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean<span class="sc">-</span>se, <span class="at">ymax=</span>mean<span class="sc">+</span>se), <span class="at">width=</span><span class="fl">0.1</span>)<span class="sc">+</span></span>
<span id="cb114-13"><a href="#cb114-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"cGAS  (sign. lifestage x acute treatment)"</span>)<span class="sc">+</span></span>
<span id="cb114-14"><a href="#cb114-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Delta Cq"</span>)<span class="sc">+</span></span>
<span id="cb114-15"><a href="#cb114-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">2</span>,<span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb114-16"><a href="#cb114-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>)<span class="sc">+</span></span>
<span id="cb114-17"><a href="#cb114-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>();plot5</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-14-1.png" class="img-fluid"><!-- --></p>
</section>
<section id="citrate-synthase-significant-effect-of-acute-treatment-and-lifestage-x-acute-treatment." class="level3">
<h3 class="anchored" data-anchor-id="citrate-synthase-significant-effect-of-acute-treatment-and-lifestage-x-acute-treatment.">10.2.6 Citrate synthase: Significant effect of acute treatment and lifestage x acute treatment.</h3>
<div class="sourceCode" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>plot6<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"citrate synthase"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target, life.stage, acute.treatment, conditioning.treatment)<span class="sc">%&gt;%</span></span>
<span id="cb115-5"><a href="#cb115-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="at">se=</span><span class="fu">sd</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>)<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">length</span>(delta_Cq)))<span class="sc">%&gt;%</span></span>
<span id="cb115-6"><a href="#cb115-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-7"><a href="#cb115-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>acute.treatment, <span class="at">y=</span>mean, <span class="at">colour=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb115-8"><a href="#cb115-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>life.stage)<span class="sc">+</span></span>
<span id="cb115-9"><a href="#cb115-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"orange"</span>))<span class="sc">+</span></span>
<span id="cb115-10"><a href="#cb115-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb115-11"><a href="#cb115-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb115-12"><a href="#cb115-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean<span class="sc">-</span>se, <span class="at">ymax=</span>mean<span class="sc">+</span>se), <span class="at">width=</span><span class="fl">0.1</span>)<span class="sc">+</span></span>
<span id="cb115-13"><a href="#cb115-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"cGAS  (sign. lifestage x acute treatment)"</span>)<span class="sc">+</span></span>
<span id="cb115-14"><a href="#cb115-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Delta Cq"</span>)<span class="sc">+</span></span>
<span id="cb115-15"><a href="#cb115-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">3</span>)<span class="sc">+</span></span>
<span id="cb115-16"><a href="#cb115-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>)<span class="sc">+</span></span>
<span id="cb115-17"><a href="#cb115-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>();plot6</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-15-1.png" class="img-fluid"><!-- --></p>
</section>
<section id="viperin-significant-effect-of-lifestage-and-lifestage-x-acute-treatment." class="level3">
<h3 class="anchored" data-anchor-id="viperin-significant-effect-of-lifestage-and-lifestage-x-acute-treatment.">10.2.7 Viperin: Significant effect of lifestage and lifestage x acute treatment.</h3>
<div class="sourceCode" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>plot7<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"viperin"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target, life.stage, acute.treatment, conditioning.treatment)<span class="sc">%&gt;%</span></span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="at">se=</span><span class="fu">sd</span>(delta_Cq, <span class="at">na.rm=</span><span class="cn">TRUE</span>)<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">length</span>(delta_Cq)))<span class="sc">%&gt;%</span></span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>acute.treatment, <span class="at">y=</span>mean, <span class="at">colour=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>life.stage)<span class="sc">+</span></span>
<span id="cb116-9"><a href="#cb116-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"orange"</span>))<span class="sc">+</span></span>
<span id="cb116-10"><a href="#cb116-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb116-11"><a href="#cb116-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb116-12"><a href="#cb116-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean<span class="sc">-</span>se, <span class="at">ymax=</span>mean<span class="sc">+</span>se), <span class="at">width=</span><span class="fl">0.1</span>)<span class="sc">+</span></span>
<span id="cb116-13"><a href="#cb116-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Viperin  (sign. lifestage x acute treatment)"</span>)<span class="sc">+</span></span>
<span id="cb116-14"><a href="#cb116-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Delta Cq"</span>)<span class="sc">+</span></span>
<span id="cb116-15"><a href="#cb116-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">6</span>)<span class="sc">+</span></span>
<span id="cb116-16"><a href="#cb116-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>)<span class="sc">+</span></span>
<span id="cb116-17"><a href="#cb116-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>();plot7</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-16-1.png" class="img-fluid"><!-- --></p>
<p>Plot spat HSP 70 and HSP 90</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a>plot8<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"HSP90"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(life.stage<span class="sc">==</span><span class="st">"spat"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>acute.treatment, <span class="at">y=</span>delta_Cq, <span class="at">colour=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Target, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span></span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"orange"</span>), <span class="at">name=</span><span class="st">"Priming"</span>)<span class="sc">+</span></span>
<span id="cb117-8"><a href="#cb117-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outliers=</span><span class="cn">FALSE</span>)<span class="sc">+</span></span>
<span id="cb117-9"><a href="#cb117-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position=</span><span class="fu">position_jitterdodge</span>(<span class="fl">0.1</span>))<span class="sc">+</span></span>
<span id="cb117-10"><a href="#cb117-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"HSP90"</span>)<span class="sc">+</span></span>
<span id="cb117-11"><a href="#cb117-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">x=</span><span class="dv">2</span>, <span class="at">y=</span><span class="dv">1</span>, <span class="at">label=</span><span class="st">"*"</span>, <span class="at">size=</span><span class="dv">12</span>, <span class="at">color=</span><span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb117-12"><a href="#cb117-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Expression"</span>)<span class="sc">+</span></span>
<span id="cb117-13"><a href="#cb117-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Temperature"</span>)<span class="sc">+</span></span>
<span id="cb117-14"><a href="#cb117-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">3</span>)<span class="sc">+</span></span>
<span id="cb117-15"><a href="#cb117-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()<span class="sc">+</span></span>
<span id="cb117-16"><a href="#cb117-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">"none"</span>);plot8</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-17-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a>plot8a<span class="ot">&lt;-</span>delta_Cq_df<span class="sc">%&gt;%</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Target<span class="sc">==</span><span class="st">"HSP70"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(life.stage<span class="sc">==</span><span class="st">"spat"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>acute.treatment, <span class="at">y=</span>delta_Cq, <span class="at">colour=</span>conditioning.treatment))<span class="sc">+</span></span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Target, <span class="at">scales=</span><span class="st">"free"</span>)<span class="sc">+</span></span>
<span id="cb118-7"><a href="#cb118-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"orange"</span>), <span class="at">name=</span><span class="st">"Priming"</span>)<span class="sc">+</span></span>
<span id="cb118-8"><a href="#cb118-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outliers=</span><span class="cn">FALSE</span>)<span class="sc">+</span></span>
<span id="cb118-9"><a href="#cb118-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position=</span><span class="fu">position_jitterdodge</span>(<span class="fl">0.1</span>))<span class="sc">+</span></span>
<span id="cb118-10"><a href="#cb118-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"HSP70"</span>)<span class="sc">+</span></span>
<span id="cb118-11"><a href="#cb118-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">x=</span><span class="dv">2</span>, <span class="at">y=</span><span class="dv">9</span>, <span class="at">label=</span><span class="st">"*"</span>, <span class="at">size=</span><span class="dv">12</span>, <span class="at">color=</span><span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb118-12"><a href="#cb118-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Expression"</span>)<span class="sc">+</span></span>
<span id="cb118-13"><a href="#cb118-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Temperature"</span>)<span class="sc">+</span></span>
<span id="cb118-14"><a href="#cb118-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">10</span>)<span class="sc">+</span></span>
<span id="cb118-15"><a href="#cb118-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>();plot8a</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/unnamed-chunk-17-2.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a>plots<span class="ot">&lt;-</span><span class="fu">plot_grid</span>(plot8, plot8a, <span class="at">nrow=</span><span class="dv">1</span>, <span class="at">rel_widths=</span><span class="fu">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>))</span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(plots, <span class="at">filename=</span><span class="st">"../output/qPCR/spat-heat-shock.png"</span>, <span class="at">width=</span><span class="dv">8</span>, <span class="at">height=</span><span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="conclusions" class="level2">
<h2 class="anchored" data-anchor-id="conclusions">10.3 Conclusions</h2>
<section id="effects-of-conditioning-on-acute-stress-response" class="level3">
<h3 class="anchored" data-anchor-id="effects-of-conditioning-on-acute-stress-response">10.3.1 Effects of conditioning on acute stress response</h3>
<ul>
<li>HSP70 and 90 genes are affected by conditioning.</li>
<li>HSP90 has significantly higher expression in treated adults than control adults at elevated temperature. No other lifestages have significant differences in expression between conditioning treatments at high temperature.<br>
</li>
<li>HSP70 has significantly lower expression in treated spat than control spat at elevated temperature. No other lifestages have significant differences between conditioning treatments at high temperature.</li>
</ul>
</section>
<section id="effects-of-acute-stress-and-lifestage" class="level3">
<h3 class="anchored" data-anchor-id="effects-of-acute-stress-and-lifestage">10.3.2 Effects of acute stress and lifestage</h3>
<ul>
<li>Across lifestages (except for juveniles) HSP90 decreases in expression under elevated temperature.</li>
<li>Expression of cGAS increases most dramatically in seed under elevated temperature.</li>
<li>ATP synthase and DMNT1 are not affected by elevated temperature.</li>
<li>Citrate synthase increases in expression in seed and spat under elevated temperature, but does not change in adult or juvenile stages.</li>
<li>Effects of acute treatment and life stage are just barely not significant for ATPsynthase and DMNT1. It is particularly interesting to see the trend for higher DMNT1 expression in spat at ambient temperature. There is also a trend for lower expression of ATPsynthase in seed at high temperature, but not technically significant.</li>
</ul>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/robertslab\.github\.io\/sams-notebook");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>