<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sam White">
<meta name="dcterms.date" content="2024-12-20">

<title>qPCR Analysis - C.gigas Lifestages Carryover – Sam’s Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 2,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 100,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-7483T61T11"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-7483T61T11', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Sam’s Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" target="_blank">
 <span class="dropdown-text">Rstudio Server - Raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu/Atumefaciens/" target="_blank">
 <span class="dropdown-text">My Gannet Directory</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" target="_blank">
 <span class="dropdown-text">Steven’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shedurkin.github.io/Roberts-LabNotebook/" target="_blank">
 <span class="dropdown-text">Kathleen’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://meganewing.github.io/mewing-notebook/" target="_blank">
 <span class="dropdown-text">Megan’s Notebook</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/sams-notebook" target="_blank">
 <span class="dropdown-text">Notebook repo</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" target="_blank">
 <span class="dropdown-text">Lab Organization (GitHub)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" target="_blank">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" target="_blank"> <i class="bi bi-calendar-week" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-right">
      <h1 class="title">qPCR Analysis - C.gigas Lifestages Carryover</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2024</div>
                <div class="quarto-category">Crassostrea gigas</div>
                <div class="quarto-category">Pacific oyster</div>
                <div class="quarto-category">qPCR</div>
                <div class="quarto-category">project-gigas-carryover</div>
                <div class="quarto-category">lifestage_carryover</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-right">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Sam White </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">December 20, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#intro" id="toc-intro" class="nav-link active" data-scroll-target="#intro">INTRO</a></li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary">SUMMARY</a>
  <ul>
  <li><a href="#t-tests-delta-cq" id="toc-t-tests-delta-cq" class="nav-link" data-scroll-target="#t-tests-delta-cq">t-tests (Delta Cq)</a>
  <ul>
  <li><a href="#seed-vs.-adult" id="toc-seed-vs.-adult" class="nav-link" data-scroll-target="#seed-vs.-adult">Seed vs.&nbsp;Adult</a></li>
  <li><a href="#seed-vs.-juvenile" id="toc-seed-vs.-juvenile" class="nav-link" data-scroll-target="#seed-vs.-juvenile">Seed vs.&nbsp;Juvenile</a></li>
  <li><a href="#seed-vs.-spat" id="toc-seed-vs.-spat" class="nav-link" data-scroll-target="#seed-vs.-spat">Seed vs.&nbsp;Spat</a></li>
  <li><a href="#adult-vs.-juvenile" id="toc-adult-vs.-juvenile" class="nav-link" data-scroll-target="#adult-vs.-juvenile">Adult vs.&nbsp;Juvenile</a></li>
  <li><a href="#adult-vs.-spat" id="toc-adult-vs.-spat" class="nav-link" data-scroll-target="#adult-vs.-spat">Adult vs.&nbsp;Spat</a></li>
  <li><a href="#juvenile-vs.-spat" id="toc-juvenile-vs.-spat" class="nav-link" data-scroll-target="#juvenile-vs.-spat">Juvenile vs.&nbsp;Spat</a></li>
  <li><a href="#seed---conditioning-treated-vs.-control" id="toc-seed---conditioning-treated-vs.-control" class="nav-link" data-scroll-target="#seed---conditioning-treated-vs.-control">Seed - Conditioning Treated vs.&nbsp;Control</a></li>
  <li><a href="#adult---conditioning-treated-vs.-control" id="toc-adult---conditioning-treated-vs.-control" class="nav-link" data-scroll-target="#adult---conditioning-treated-vs.-control">Adult - Conditioning Treated vs.&nbsp;Control</a></li>
  <li><a href="#juvenile---conditioning-treated-vs.-control" id="toc-juvenile---conditioning-treated-vs.-control" class="nav-link" data-scroll-target="#juvenile---conditioning-treated-vs.-control">Juvenile - Conditioning Treated vs.&nbsp;Control</a></li>
  <li><a href="#spat---conditioning-treated-vs.-control" id="toc-spat---conditioning-treated-vs.-control" class="nav-link" data-scroll-target="#spat---conditioning-treated-vs.-control">Spat - Conditioning Treated vs.&nbsp;Control</a></li>
  <li><a href="#adult---acute-ambient-vs.-high" id="toc-adult---acute-ambient-vs.-high" class="nav-link" data-scroll-target="#adult---acute-ambient-vs.-high">Adult - Acute Ambient vs.&nbsp;High</a></li>
  <li><a href="#juvenile---acute-ambient-vs.-high" id="toc-juvenile---acute-ambient-vs.-high" class="nav-link" data-scroll-target="#juvenile---acute-ambient-vs.-high">Juvenile - Acute Ambient vs.&nbsp;High</a></li>
  <li><a href="#adult---conditioning-treated-acute-ambient-vs.-high" id="toc-adult---conditioning-treated-acute-ambient-vs.-high" class="nav-link" data-scroll-target="#adult---conditioning-treated-acute-ambient-vs.-high">Adult - Conditioning Treated: Acute Ambient vs.&nbsp;High</a></li>
  <li><a href="#adult---conditioning-control-acute-ambient-vs.-high" id="toc-adult---conditioning-control-acute-ambient-vs.-high" class="nav-link" data-scroll-target="#adult---conditioning-control-acute-ambient-vs.-high">Adult - Conditioning Control: Acute Ambient vs.&nbsp;High</a></li>
  <li><a href="#juvenile---conditioning-treated-acute-ambient-vs.-high" id="toc-juvenile---conditioning-treated-acute-ambient-vs.-high" class="nav-link" data-scroll-target="#juvenile---conditioning-treated-acute-ambient-vs.-high">Juvenile - Conditioning Treated: Acute Ambient vs.&nbsp;High</a></li>
  <li><a href="#juvenile---conditioning-control-acute-ambient-vs.-high" id="toc-juvenile---conditioning-control-acute-ambient-vs.-high" class="nav-link" data-scroll-target="#juvenile---conditioning-control-acute-ambient-vs.-high">Juvenile - Conditioning Control: Acute Ambient vs.&nbsp;High</a></li>
  </ul></li>
  <li><a href="#gene-expression-delta-delta-cq-or-fold-change" id="toc-gene-expression-delta-delta-cq-or-fold-change" class="nav-link" data-scroll-target="#gene-expression-delta-delta-cq-or-fold-change">Gene Expression (delta delta Cq or fold change)</a>
  <ul>
  <li><a href="#adult---conditioning-treated-acute-high-vs.-ambient" id="toc-adult---conditioning-treated-acute-high-vs.-ambient" class="nav-link" data-scroll-target="#adult---conditioning-treated-acute-high-vs.-ambient">Adult - Conditioning Treated: Acute High vs.&nbsp;Ambient</a></li>
  <li><a href="#juvenile---conditioning-treated-acute-high-vs.-ambient" id="toc-juvenile---conditioning-treated-acute-high-vs.-ambient" class="nav-link" data-scroll-target="#juvenile---conditioning-treated-acute-high-vs.-ambient">Juvenile - Conditioning Treated: Acute High vs.&nbsp;Ambient</a></li>
  <li><a href="#adult---adult-vs.-seed" id="toc-adult---adult-vs.-seed" class="nav-link" data-scroll-target="#adult---adult-vs.-seed">Adult - Adult vs.&nbsp;Seed</a></li>
  <li><a href="#adult---adult-vs.-spat" id="toc-adult---adult-vs.-spat" class="nav-link" data-scroll-target="#adult---adult-vs.-spat">Adult - Adult vs.&nbsp;Spat</a></li>
  <li><a href="#adult---spat-vs.-seed" id="toc-adult---spat-vs.-seed" class="nav-link" data-scroll-target="#adult---spat-vs.-seed">Adult - Spat vs.&nbsp;Seed</a></li>
  <li><a href="#conditioning-by-target-and-lifestage" id="toc-conditioning-by-target-and-lifestage" class="nav-link" data-scroll-target="#conditioning-by-target-and-lifestage">Conditioning by Target and Lifestage</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#description" id="toc-description" class="nav-link" data-scroll-target="#description">1 Description</a></li>
  <li><a href="#set-variables" id="toc-set-variables" class="nav-link" data-scroll-target="#set-variables">2 Set variables</a>
  <ul>
  <li><a href="#set-sample-groups" id="toc-set-sample-groups" class="nav-link" data-scroll-target="#set-sample-groups">2.1 Set sample groups</a></li>
  <li><a href="#assign-groups-to-list" id="toc-assign-groups-to-list" class="nav-link" data-scroll-target="#assign-groups-to-list">2.2 Assign groups to list</a></li>
  </ul></li>
  <li><a href="#functions" id="toc-functions" class="nav-link" data-scroll-target="#functions">3 Functions</a>
  <ul>
  <li><a href="#calculate-delta-cq" id="toc-calculate-delta-cq" class="nav-link" data-scroll-target="#calculate-delta-cq">3.1 Calculate delta Cq</a></li>
  <li><a href="#create-delta-cq-boxplots" id="toc-create-delta-cq-boxplots" class="nav-link" data-scroll-target="#create-delta-cq-boxplots">3.2 Create delta Cq boxplots</a>
  <ul>
  <li><a href="#lifestage-comparison" id="toc-lifestage-comparison" class="nav-link" data-scroll-target="#lifestage-comparison">3.2.1 Lifestage comparison</a></li>
  <li><a href="#conditioning-comparisons" id="toc-conditioning-comparisons" class="nav-link" data-scroll-target="#conditioning-comparisons">3.2.2 Conditioning comparisons</a></li>
  <li><a href="#acute-comparisons" id="toc-acute-comparisons" class="nav-link" data-scroll-target="#acute-comparisons">3.2.3 Acute comparisons</a></li>
  <li><a href="#acute-treatements-within-life-stage-conditioning" id="toc-acute-treatements-within-life-stage-conditioning" class="nav-link" data-scroll-target="#acute-treatements-within-life-stage-conditioning">3.2.4 Acute treatements within life stage conditioning</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#read-in-files" id="toc-read-in-files" class="nav-link" data-scroll-target="#read-in-files">4 Read in files</a></li>
  <li><a href="#clean-data" id="toc-clean-data" class="nav-link" data-scroll-target="#clean-data">5 Clean data</a>
  <ul>
  <li><a href="#replace-target-names" id="toc-replace-target-names" class="nav-link" data-scroll-target="#replace-target-names">5.1 Replace target names</a></li>
  <li><a href="#identify-samples-with-cq.std..dev-0.5" id="toc-identify-samples-with-cq.std..dev-0.5" class="nav-link" data-scroll-target="#identify-samples-with-cq.std..dev-0.5">5.2 Identify Samples with Cq.Std..Dev &gt; 0.5</a></li>
  <li><a href="#remove-bad-technical-reps" id="toc-remove-bad-technical-reps" class="nav-link" data-scroll-target="#remove-bad-technical-reps">5.3 Remove bad technical reps</a></li>
  </ul></li>
  <li><a href="#group-samples-by-target" id="toc-group-samples-by-target" class="nav-link" data-scroll-target="#group-samples-by-target">6 Group samples by target</a></li>
  <li><a href="#add-life-stage-and-treatment-cols" id="toc-add-life-stage-and-treatment-cols" class="nav-link" data-scroll-target="#add-life-stage-and-treatment-cols">7 Add life stage and treatment cols</a></li>
  <li><a href="#delta-cq-to-normalizing-gene" id="toc-delta-cq-to-normalizing-gene" class="nav-link" data-scroll-target="#delta-cq-to-normalizing-gene">8 Delta Cq to Normalizing Gene</a>
  <ul>
  <li><a href="#t-tests" id="toc-t-tests" class="nav-link" data-scroll-target="#t-tests">8.1 t-tests</a>
  <ul>
  <li><a href="#life-stages" id="toc-life-stages" class="nav-link" data-scroll-target="#life-stages">8.1.1 Life Stages</a></li>
  <li><a href="#conditioning-treatments" id="toc-conditioning-treatments" class="nav-link" data-scroll-target="#conditioning-treatments">8.1.2 Conditioning treatments</a></li>
  <li><a href="#acute-treatments" id="toc-acute-treatments" class="nav-link" data-scroll-target="#acute-treatments">8.1.3 Acute treatments</a></li>
  <li><a href="#acute-within-life-stage-and-conditioning" id="toc-acute-within-life-stage-and-conditioning" class="nav-link" data-scroll-target="#acute-within-life-stage-and-conditioning">8.1.4 Acute within life stage and conditioning</a></li>
  </ul></li>
  <li><a href="#plotting" id="toc-plotting" class="nav-link" data-scroll-target="#plotting">8.2 Plotting</a>
  <ul>
  <li><a href="#delta-cq-boxplots" id="toc-delta-cq-boxplots" class="nav-link" data-scroll-target="#delta-cq-boxplots">8.2.1 Delta Cq boxplots</a>
  <ul class="collapse">
  <li><a href="#lifestage-comparisons" id="toc-lifestage-comparisons" class="nav-link" data-scroll-target="#lifestage-comparisons">8.2.1.1 Lifestage comparisons</a></li>
  </ul></li>
  <li><a href="#conditioning-comparisons-1" id="toc-conditioning-comparisons-1" class="nav-link" data-scroll-target="#conditioning-comparisons-1">8.2.2 Conditioning comparisons</a></li>
  <li><a href="#acute-treatment-comparisons" id="toc-acute-treatment-comparisons" class="nav-link" data-scroll-target="#acute-treatment-comparisons">8.2.3 Acute treatment comparisons</a></li>
  <li><a href="#acute-within-life-stage-conditioning" id="toc-acute-within-life-stage-conditioning" class="nav-link" data-scroll-target="#acute-within-life-stage-conditioning">8.2.4 Acute within life stage conditioning</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#delta-delta-cq" id="toc-delta-delta-cq" class="nav-link" data-scroll-target="#delta-delta-cq">9 Delta delta Cq</a>
  <ul>
  <li><a href="#calculations" id="toc-calculations" class="nav-link" data-scroll-target="#calculations">9.1 Calculations</a>
  <ul>
  <li><a href="#conditioning" id="toc-conditioning" class="nav-link" data-scroll-target="#conditioning">9.1.1 Conditioning</a></li>
  <li><a href="#acute-treatment" id="toc-acute-treatment" class="nav-link" data-scroll-target="#acute-treatment">9.1.2 Acute treatment</a></li>
  <li><a href="#life-stage" id="toc-life-stage" class="nav-link" data-scroll-target="#life-stage">9.1.3 Life stage</a></li>
  <li><a href="#calculate-delta-delta-acute-treatments-within-lifestage-and-conditioning" id="toc-calculate-delta-delta-acute-treatments-within-lifestage-and-conditioning" class="nav-link" data-scroll-target="#calculate-delta-delta-acute-treatments-within-lifestage-and-conditioning">9.1.4 Calculate delta delta acute treatments within lifestage and conditioning</a></li>
  <li><a href="#calculate-the-fold-change-life-stage-comparison" id="toc-calculate-the-fold-change-life-stage-comparison" class="nav-link" data-scroll-target="#calculate-the-fold-change-life-stage-comparison">9.1.5 Calculate the fold change life stage comparison</a></li>
  <li><a href="#calculate-the-fold-change-conditioning-comparison" id="toc-calculate-the-fold-change-conditioning-comparison" class="nav-link" data-scroll-target="#calculate-the-fold-change-conditioning-comparison">9.1.6 Calculate the fold change conditioning comparison</a></li>
  <li><a href="#calculate-the-fold-change-acute-comparison" id="toc-calculate-the-fold-change-acute-comparison" class="nav-link" data-scroll-target="#calculate-the-fold-change-acute-comparison">9.1.7 Calculate the fold change acute comparison</a></li>
  <li><a href="#calculate-fold-change-acute-treatments-within-lifestage-and-conditioning" id="toc-calculate-fold-change-acute-treatments-within-lifestage-and-conditioning" class="nav-link" data-scroll-target="#calculate-fold-change-acute-treatments-within-lifestage-and-conditioning">9.1.8 Calculate fold change acute treatments within lifestage and conditioning</a></li>
  </ul></li>
  <li><a href="#plotting-fold-changes" id="toc-plotting-fold-changes" class="nav-link" data-scroll-target="#plotting-fold-changes">9.2 Plotting fold changes</a>
  <ul>
  <li><a href="#acute-comparisons-within-lifestage-and-conditioning" id="toc-acute-comparisons-within-lifestage-and-conditioning" class="nav-link" data-scroll-target="#acute-comparisons-within-lifestage-and-conditioning">9.2.1 Acute comparisons within lifestage and conditioning</a></li>
  <li><a href="#life-stage-comparisons" id="toc-life-stage-comparisons" class="nav-link" data-scroll-target="#life-stage-comparisons">9.2.2 Life stage comparisons</a></li>
  <li><a href="#conditioning-comparisons-2" id="toc-conditioning-comparisons-2" class="nav-link" data-scroll-target="#conditioning-comparisons-2">9.2.3 Conditioning comparisons</a></li>
  <li><a href="#line-plot-conditioning-comparisons-across-lifestages" id="toc-line-plot-conditioning-comparisons-across-lifestages" class="nav-link" data-scroll-target="#line-plot-conditioning-comparisons-across-lifestages">9.2.4 Line plot conditioning comparisons across lifestages</a></li>
  <li><a href="#acute-treatment-comparison" id="toc-acute-treatment-comparison" class="nav-link" data-scroll-target="#acute-treatment-comparison">9.2.5 Acute treatment comparison</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-right" id="quarto-document-content">





<section id="intro" class="level1">
<h1>INTRO</h1>
<p>This notebook performs pair-wise comparisons of qPCR gene expression, normalized to <code>GAPDH</code> expression. It calculates delta Cq, delta delta Cq, and fold changes in expression. Additionally, it generates box plots (delta Cq), and bar plots (fold change expression).</p>
<p>I’ve provided a summary of the various pair-wise comparisons immediately below, but you might want to just <a href="#plotting">skip to the plots</a>, as the summary <em>and</em> the notebook content is <em>very</em> lengthy.</p>
</section>
<section id="summary" class="level1">
<h1>SUMMARY</h1>
<section id="t-tests-delta-cq" class="level2">
<h2 class="anchored" data-anchor-id="t-tests-delta-cq">t-tests (Delta Cq)</h2>
<section id="seed-vs.-adult" class="level3">
<h3 class="anchored" data-anchor-id="seed-vs.-adult">Seed vs.&nbsp;Adult</h3>
<p>These genes had p-values &lt;= 0.05:</p>
<ul>
<li>VIPERIN</li>
</ul>
</section>
<section id="seed-vs.-juvenile" class="level3">
<h3 class="anchored" data-anchor-id="seed-vs.-juvenile">Seed vs.&nbsp;Juvenile</h3>
<p>These genes had p-values &lt;= 0.05:</p>
<ul>
<li><p>citrate synthase</p></li>
<li><p>HSP90</p></li>
</ul>
</section>
<section id="seed-vs.-spat" class="level3">
<h3 class="anchored" data-anchor-id="seed-vs.-spat">Seed vs.&nbsp;Spat</h3>
<p>These genes had p-values &lt;= 0.05:</p>
<ul>
<li>VIPERIN</li>
</ul>
</section>
<section id="adult-vs.-juvenile" class="level3">
<h3 class="anchored" data-anchor-id="adult-vs.-juvenile">Adult vs.&nbsp;Juvenile</h3>
<p>The genes had p-values &lt;= 0.05:</p>
<ul>
<li><p>citrate synthase</p></li>
<li><p>VIPERIN</p></li>
</ul>
</section>
<section id="adult-vs.-spat" class="level3">
<h3 class="anchored" data-anchor-id="adult-vs.-spat">Adult vs.&nbsp;Spat</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="juvenile-vs.-spat" class="level3">
<h3 class="anchored" data-anchor-id="juvenile-vs.-spat">Juvenile vs.&nbsp;Spat</h3>
<p>The genes had p-values &lt;= 0.05:</p>
<ul>
<li><p>citrate synthase</p></li>
<li><p>VIPERIN</p></li>
</ul>
</section>
<section id="seed---conditioning-treated-vs.-control" class="level3">
<h3 class="anchored" data-anchor-id="seed---conditioning-treated-vs.-control">Seed - Conditioning Treated vs.&nbsp;Control</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="adult---conditioning-treated-vs.-control" class="level3">
<h3 class="anchored" data-anchor-id="adult---conditioning-treated-vs.-control">Adult - Conditioning Treated vs.&nbsp;Control</h3>
<p>The genes had p-values &lt;= 0.05:</p>
<ul>
<li>HSP90</li>
</ul>
</section>
<section id="juvenile---conditioning-treated-vs.-control" class="level3">
<h3 class="anchored" data-anchor-id="juvenile---conditioning-treated-vs.-control">Juvenile - Conditioning Treated vs.&nbsp;Control</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="spat---conditioning-treated-vs.-control" class="level3">
<h3 class="anchored" data-anchor-id="spat---conditioning-treated-vs.-control">Spat - Conditioning Treated vs.&nbsp;Control</h3>
<p>The genes had p-values &lt;= 0.05:</p>
<ul>
<li><p>citrate synthase</p></li>
<li><p>DNMT1</p></li>
</ul>
</section>
<section id="adult---acute-ambient-vs.-high" class="level3">
<h3 class="anchored" data-anchor-id="adult---acute-ambient-vs.-high">Adult - Acute Ambient vs.&nbsp;High</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="juvenile---acute-ambient-vs.-high" class="level3">
<h3 class="anchored" data-anchor-id="juvenile---acute-ambient-vs.-high">Juvenile - Acute Ambient vs.&nbsp;High</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="adult---conditioning-treated-acute-ambient-vs.-high" class="level3">
<h3 class="anchored" data-anchor-id="adult---conditioning-treated-acute-ambient-vs.-high">Adult - Conditioning Treated: Acute Ambient vs.&nbsp;High</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="adult---conditioning-control-acute-ambient-vs.-high" class="level3">
<h3 class="anchored" data-anchor-id="adult---conditioning-control-acute-ambient-vs.-high">Adult - Conditioning Control: Acute Ambient vs.&nbsp;High</h3>
<p>No genes had p-values &lt;= 0.05</p>
</section>
<section id="juvenile---conditioning-treated-acute-ambient-vs.-high" class="level3">
<h3 class="anchored" data-anchor-id="juvenile---conditioning-treated-acute-ambient-vs.-high">Juvenile - Conditioning Treated: Acute Ambient vs.&nbsp;High</h3>
<p>The genes had p-values &lt;= 0.05:</p>
<ul>
<li>HSP70</li>
</ul>
</section>
<section id="juvenile---conditioning-control-acute-ambient-vs.-high" class="level3">
<h3 class="anchored" data-anchor-id="juvenile---conditioning-control-acute-ambient-vs.-high">Juvenile - Conditioning Control: Acute Ambient vs.&nbsp;High</h3>
<p>The genes had p-values &lt;= 0.05:</p>
<ul>
<li>VIPERIN</li>
</ul>
</section>
</section>
<section id="gene-expression-delta-delta-cq-or-fold-change" class="level2">
<h2 class="anchored" data-anchor-id="gene-expression-delta-delta-cq-or-fold-change">Gene Expression (delta delta Cq or fold change)</h2>
<section id="adult---conditioning-treated-acute-high-vs.-ambient" class="level3">
<h3 class="anchored" data-anchor-id="adult---conditioning-treated-acute-high-vs.-ambient">Adult - Conditioning Treated: Acute High vs.&nbsp;Ambient</h3>
<ul>
<li><p>All genes show elevated expression relative to Ambient.</p></li>
<li><p>Fold change is at similar levels across all genes, with DNMT1 and HSP70 being the highest.</p></li>
</ul>
</section>
<section id="juvenile---conditioning-treated-acute-high-vs.-ambient" class="level3">
<h3 class="anchored" data-anchor-id="juvenile---conditioning-treated-acute-high-vs.-ambient">Juvenile - Conditioning Treated: Acute High vs.&nbsp;Ambient</h3>
<ul>
<li><p>All genes show elevated expression relative to Ambient.</p></li>
<li><p>HSP70 shows ~4-fold higher fold change in expression compared to most of the other genes.</p></li>
<li><p>DNMT1 exhibits fold change expression ~1/2 that of most of the other genes.</p></li>
</ul>
</section>
<section id="adult---adult-vs.-seed" class="level3">
<h3 class="anchored" data-anchor-id="adult---adult-vs.-seed">Adult - Adult vs.&nbsp;Seed</h3>
<ul>
<li><p>All genes show elevated expression relative to Seed.</p></li>
<li><p>VIPERIN exhibits fold change in expression &gt;2-fold higher than the other genes.</p></li>
</ul>
</section>
<section id="adult---adult-vs.-spat" class="level3">
<h3 class="anchored" data-anchor-id="adult---adult-vs.-spat">Adult - Adult vs.&nbsp;Spat</h3>
<ul>
<li><p>All genes show elevated expression relative to Spat.</p></li>
<li><p>All genes have similar fold changes in expression, with ATP synthase and DNMT1 showing the highest levels of fold change in expression.</p></li>
</ul>
</section>
<section id="adult---spat-vs.-seed" class="level3">
<h3 class="anchored" data-anchor-id="adult---spat-vs.-seed">Adult - Spat vs.&nbsp;Seed</h3>
<ul>
<li><p>All genes show elevated expression relative to Seed.</p></li>
<li><p>Most genes have similar fold changes in expression.</p></li>
<li><p>VIPERIN exhibits fold change in a expression ~2-fold higher than the other genes.</p></li>
</ul>
</section>
<section id="conditioning-by-target-and-lifestage" class="level3">
<h3 class="anchored" data-anchor-id="conditioning-by-target-and-lifestage">Conditioning by Target and Lifestage</h3>
<ul>
<li><p>All genes show elevated expression relative to Control.</p></li>
<li><p>VIPERIN, citrate synthase, and ATP synthase show an initial increase in fold change expression from seed to spat life stages, declining to juvenile and adult lifestages.</p></li>
<li><p>HSP90 shows level fold changes in expression across seed, spat, and juvenile lifestages, followed by a decrease in the adult stage.</p></li>
<li><p>DNMT1 exhibits a drastic decrease in fold change expression from seed to spat, followed by a substantial increase from spat to adult.</p></li>
<li><p>cGAS and HSP70 exhibit decreases from seed to spat, followed by a sharp increase from spat to juvenile. This is followed by a moderate decrease from juvenile to adult by cGAS and a sharp decrease in HSP70.</p></li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The post below is markdown generated from <a href="https://github.com/RobertsLab/project-gigas-carryover/blob/9a32e40ee9739803a3f5b523c40ffc925d9f7472/lifestage_carryover/scripts/01.01-qPCR.Rmd">01.01-qPCR.Rmd</a> (commit <code>9a32e40</code>).</p>
</div>
</div>
</section>
</section>
</section>
<section id="description" class="level1">
<h1>1 Description</h1>
<p>This notebook performs pair-wise comparisons of qPCR gene expression, normalized to <code>GAPDH</code> expression. It calculates delta Cq, delta delta Cq, and fold changes in expression. Additionally, it generates box plots (delta Cq), and bar plots (fold change expression).</p>
</section>
<section id="set-variables" class="level1">
<h1>2 Set variables</h1>
<section id="set-sample-groups" class="level2">
<h2 class="anchored" data-anchor-id="set-sample-groups">2.1 Set sample groups</h2>
<p>Groups are named in the following fashion:</p>
<p><code>&lt;life.stage&gt;.&lt;conditioning.treatment&gt;.&lt;acute.treatment&gt;</code></p>
<p>This allows for parsing downstream.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<p>Below is the full set of groups for the <em>entire experiment</em>. For the current qPCR analysis, <code>seed</code> and <code>spat</code> do <em>not</em> have acute treatments; just conditioning treatments.</p>
</div>
</div>
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>seed.control.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"29"</span>, <span class="st">"40"</span>, <span class="st">"55"</span>, <span class="st">"63"</span>, <span class="st">"69"</span>, <span class="st">"101"</span>, <span class="st">"119"</span>, <span class="st">"122"</span>, <span class="st">"155"</span>, <span class="st">"164"</span>, <span class="st">"187"</span>, <span class="st">"202"</span>, <span class="st">"209"</span>, <span class="st">"214"</span>, <span class="st">"233"</span>, <span class="st">"236"</span>, <span class="st">"275"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>seed.control.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"42"</span>, <span class="st">"59"</span>, <span class="st">"60"</span>, <span class="st">"62"</span>, <span class="st">"86"</span>, <span class="st">"102"</span>, <span class="st">"140"</span>, <span class="st">"176"</span>, <span class="st">"177"</span>, <span class="st">"184"</span>, <span class="st">"192"</span>, <span class="st">"223"</span>, <span class="st">"234"</span>, <span class="st">"243"</span>, <span class="st">"244"</span>, <span class="st">"254"</span>, <span class="st">"264"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>seed.treated.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"14"</span>, <span class="st">"48"</span>, <span class="st">"66"</span>, <span class="st">"72"</span>, <span class="st">"89"</span>, <span class="st">"115"</span>, <span class="st">"129"</span>, <span class="st">"138"</span>, <span class="st">"156"</span>, <span class="st">"182"</span>, <span class="st">"191"</span>, <span class="st">"201"</span>, <span class="st">"227"</span>, <span class="st">"239"</span>, <span class="st">"270"</span>, <span class="st">"277"</span>, <span class="st">"280"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>seed.treated.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"15"</span>, <span class="st">"19"</span>, <span class="st">"24"</span>, <span class="st">"88"</span>, <span class="st">"92"</span>, <span class="st">"105"</span>, <span class="st">"111"</span>, <span class="st">"113"</span>, <span class="st">"120"</span>, <span class="st">"128"</span>, <span class="st">"161"</span>, <span class="st">"200"</span>, <span class="st">"211"</span>, <span class="st">"256"</span>, <span class="st">"257"</span>, <span class="st">"266"</span>, <span class="st">"285"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>spat.control.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"11"</span>, <span class="st">"30"</span>, <span class="st">"36"</span>, <span class="st">"52"</span>, <span class="st">"77"</span>, <span class="st">"114"</span>, <span class="st">"134"</span>, <span class="st">"142"</span>, <span class="st">"144"</span>, <span class="st">"183"</span>, <span class="st">"193"</span>, <span class="st">"229"</span>, <span class="st">"230"</span>, <span class="st">"231"</span>, <span class="st">"240"</span>, <span class="st">"272"</span>, <span class="st">"287"</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>spat.control.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"27"</span>, <span class="st">"74"</span>, <span class="st">"93"</span>, <span class="st">"96"</span>, <span class="st">"97"</span>, <span class="st">"137"</span>, <span class="st">"143"</span>, <span class="st">"153"</span>, <span class="st">"168"</span>, <span class="st">"178"</span>, <span class="st">"189"</span>, <span class="st">"206"</span>, <span class="st">"262"</span>, <span class="st">"274"</span>, <span class="st">"282"</span>, <span class="st">"284"</span>, <span class="st">"289"</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>spat.treated.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"9"</span>, <span class="st">"13"</span>, <span class="st">"38"</span>, <span class="st">"46"</span>, <span class="st">"47"</span>, <span class="st">"121"</span>, <span class="st">"145"</span>, <span class="st">"151"</span>, <span class="st">"174"</span>, <span class="st">"194"</span>, <span class="st">"197"</span>, <span class="st">"198"</span>, <span class="st">"216"</span>, <span class="st">"235"</span>, <span class="st">"241"</span>, <span class="st">"252"</span>, <span class="st">"291"</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>spat.treated.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"6"</span>, <span class="st">"25"</span>, <span class="st">"50"</span>, <span class="st">"78"</span>, <span class="st">"124"</span>, <span class="st">"126"</span>, <span class="st">"131"</span>, <span class="st">"160"</span>, <span class="st">"163"</span>, <span class="st">"172"</span>, <span class="st">"220"</span>, <span class="st">"226"</span>, <span class="st">"242"</span>, <span class="st">"253"</span>, <span class="st">"296"</span>, <span class="st">"298"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>juvenile.control.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"18"</span>, <span class="st">"57"</span>, <span class="st">"65"</span>, <span class="st">"75"</span>, <span class="st">"79"</span>, <span class="st">"104"</span>, <span class="st">"110"</span>, <span class="st">"123"</span>, <span class="st">"125"</span>, <span class="st">"171"</span>, <span class="st">"175"</span>, <span class="st">"205"</span>, <span class="st">"238"</span>, <span class="st">"273"</span>, <span class="st">"279"</span>, <span class="st">"293"</span>, <span class="st">"317"</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>juvenile.control.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"12"</span>, <span class="st">"39"</span>, <span class="st">"43"</span>, <span class="st">"49"</span>, <span class="st">"71"</span>, <span class="st">"130"</span>, <span class="st">"141"</span>, <span class="st">"146"</span>, <span class="st">"150"</span>, <span class="st">"170"</span>, <span class="st">"195"</span>, <span class="st">"297"</span>, <span class="st">"301"</span>, <span class="st">"324"</span>, <span class="st">"351"</span>, <span class="st">"355"</span>, <span class="st">"371"</span>)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>juvenile.treated.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"34"</span>, <span class="st">"64"</span>, <span class="st">"83"</span>, <span class="st">"98"</span>, <span class="st">"147"</span>, <span class="st">"152"</span>, <span class="st">"158"</span>, <span class="st">"162"</span>, <span class="st">"169"</span>, <span class="st">"188"</span>, <span class="st">"271"</span>, <span class="st">"295"</span>, <span class="st">"310"</span>, <span class="st">"357"</span>, <span class="st">"361"</span>, <span class="st">"381"</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>juvenile.treated.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"28"</span>, <span class="st">"53"</span>, <span class="st">"61"</span>, <span class="st">"73"</span>, <span class="st">"81"</span>, <span class="st">"106"</span>, <span class="st">"109"</span>, <span class="st">"139"</span>, <span class="st">"149"</span>, <span class="st">"173"</span>, <span class="st">"181"</span>, <span class="st">"213"</span>, <span class="st">"290"</span>, <span class="st">"302"</span>, <span class="st">"311"</span>, <span class="st">"364"</span>, <span class="st">"392"</span>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>adult.control.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"3"</span>, <span class="st">"5"</span>, <span class="st">"13*"</span>, <span class="st">"16"</span>, <span class="st">"17"</span>, <span class="st">"80"</span>, <span class="st">"87"</span>, <span class="st">"94"</span>, <span class="st">"148"</span>, <span class="st">"159"</span>, <span class="st">"179"</span>, <span class="st">"180"</span>, <span class="st">"250"</span>, <span class="st">"258"</span>, <span class="st">"268"</span>, <span class="st">"312"</span>, <span class="st">"326"</span>, <span class="st">"330"</span>, <span class="st">"334"</span>, <span class="st">"346"</span>, <span class="st">"360"</span>, <span class="st">"377"</span>, <span class="st">"379"</span>, <span class="st">"386"</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>adult.control.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"20"</span>, <span class="st">"23"</span>, <span class="st">"26"</span>, <span class="st">"32"</span>, <span class="st">"33"</span>, <span class="st">"67"</span>, <span class="st">"70"</span>, <span class="st">"90"</span>, <span class="st">"107"</span>, <span class="st">"132"</span>, <span class="st">"135"</span>, <span class="st">"157"</span>, <span class="st">"166"</span>, <span class="st">"186"</span>, <span class="st">"207"</span>, <span class="st">"215"</span>, <span class="st">"248"</span>, <span class="st">"316"</span>, <span class="st">"341"</span>, <span class="st">"344"</span>, <span class="st">"349"</span>, <span class="st">"382"</span>, <span class="st">"394"</span>, <span class="st">"395"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>adult.treated.ambient<span class="ot">=</span><span class="fu">c</span>(<span class="st">"7"</span>, <span class="st">"31"</span>, <span class="st">"35"</span>, <span class="st">"37"</span>, <span class="st">"41"</span>, <span class="st">"54"</span>, <span class="st">"84"</span>, <span class="st">"100"</span>, <span class="st">"112"</span>, <span class="st">"116"</span>, <span class="st">"118"</span>, <span class="st">"133"</span>, <span class="st">"154"</span>, <span class="st">"199"</span>, <span class="st">"203"</span>, <span class="st">"204"</span>, <span class="st">"208"</span>, <span class="st">"219"</span>, <span class="st">"294"</span>, <span class="st">"318"</span>, <span class="st">"339"</span>, <span class="st">"353"</span>, <span class="st">"363"</span>, <span class="st">"378"</span>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>adult.treated.high<span class="ot">=</span><span class="fu">c</span>(<span class="st">"21"</span>, <span class="st">"22"</span>, <span class="st">"45"</span>, <span class="st">"82"</span>, <span class="st">"85"</span>, <span class="st">"91"</span>, <span class="st">"95"</span>, <span class="st">"99"</span>, <span class="st">"103"</span>, <span class="st">"108"</span>, <span class="st">"117"</span>, <span class="st">"127"</span>, <span class="st">"165"</span>, <span class="st">"185"</span>, <span class="st">"190"</span>, <span class="st">"196"</span>, <span class="st">"232"</span>, <span class="st">"237"</span>, <span class="st">"245"</span>, <span class="st">"263"</span>, <span class="st">"276"</span>, <span class="st">"306"</span>, <span class="st">"343"</span>, <span class="st">"374"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="assign-groups-to-list" class="level2">
<h2 class="anchored" data-anchor-id="assign-groups-to-list">2.2 Assign groups to list</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine vectors into lists</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Used for adding treatment info and/or subsetting downstream</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>groups_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">juvenile.control.ambient =</span> juvenile.control.ambient,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">juvenile.control.high =</span> juvenile.control.high,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">juvenile.treated.ambient =</span> juvenile.treated.ambient,</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">juvenile.treated.high =</span> juvenile.treated.high,</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">adult.control.ambient =</span> adult.control.ambient,</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">adult.control.high =</span> adult.control.high,</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">adult.treated.ambient =</span> adult.treated.ambient,</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">adult.treated.high =</span> adult.treated.high,</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">seed.control.ambient =</span> seed.control.ambient,</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">seed.control.high =</span> seed.control.high,</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">seed.treated.ambient =</span> seed.treated.ambient,</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">seed.treated.high =</span> seed.treated.high,</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">spat.control.ambient =</span> spat.control.ambient,</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">spat.control.high =</span> spat.control.high,</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">spat.treated.ambient =</span> spat.treated.ambient,</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">spat.treated.high =</span> spat.treated.high)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="functions" class="level1">
<h1>3 Functions</h1>
<section id="calculate-delta-cq" class="level2">
<h2 class="anchored" data-anchor-id="calculate-delta-cq">3.1 Calculate delta Cq</h2>
<p>Normalized to designated normalizing gene</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>calculate_delta_Cq <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Sample) <span class="sc">%&gt;%</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">delta_Cq =</span> Cq.Mean <span class="sc">-</span> Cq.Mean[Target <span class="sc">==</span> <span class="st">"GAPDH"</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>()</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="create-delta-cq-boxplots" class="level2">
<h2 class="anchored" data-anchor-id="create-delta-cq-boxplots">3.2 Create delta Cq boxplots</h2>
<section id="lifestage-comparison" class="level3">
<h3 class="anchored" data-anchor-id="lifestage-comparison">3.2.1 Lifestage comparison</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create box plots for each comparison</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>create_boxplot_delta_Cq <span class="ot">&lt;-</span> <span class="cf">function</span>(data, comparison, t_test_results) {</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract life stages from comparison</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  life_stages <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(comparison, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>))</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print life stages</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(paste("Life stages for comparison:", comparison))</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(life_stages)</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter data for the relevant life stages</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  filtered_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(life.stage <span class="sc">%in%</span> life_stages)</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print filtered data</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Filtered data:")</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(filtered_data)</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both life stages are included</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(life_stages <span class="sc">%in%</span> <span class="fu">unique</span>(filtered_data<span class="sc">$</span>life.stage))) {</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Not all life stages are included in the filtered data"</span>)</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>  y_limits <span class="ot">&lt;-</span> <span class="fu">range</span>(filtered_data<span class="sc">$</span>delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print y_limits</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Y limits:")</span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(y_limits)</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for the current comparison</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>  t_test_results_filtered <span class="ot">&lt;-</span> t_test_results <span class="sc">%&gt;%</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(comparison <span class="sc">==</span> <span class="sc">!!</span>comparison)</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print filtered t_test_results</span></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Filtered t_test_results:")</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(t_test_results_filtered)</span></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for asterisks</span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>  t_test_results_with_asterisks <span class="ot">&lt;-</span> t_test_results_filtered <span class="sc">%&gt;%</span></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(asterisk <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print t_test_results_with_asterisks</span></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("t_test_results_with_asterisks:")</span></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(t_test_results_with_asterisks)</span></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>  formatted_title <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">toupper</span>(<span class="fu">substring</span>(life_stages[<span class="dv">1</span>], <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(life_stages[<span class="dv">1</span>], <span class="dv">2</span>), </span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" vs. "</span>, </span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(life_stages[<span class="dv">2</span>], <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(life_stages[<span class="dv">2</span>], <span class="dv">2</span>))</span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>  boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(filtered_data, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> life.stage)) <span class="sc">+</span></span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"salmon"</span>, <span class="st">"lightblue"</span>, <span class="st">"lightgreen"</span>)) <span class="sc">+</span></span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(y_limits) <span class="sc">+</span></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Delta Cq"</span>, <span class="at">title =</span> formatted_title) <span class="sc">+</span></span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Highlighted section: Adds asterisks</span></span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> t_test_results_with_asterisks, </span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> y_limits[<span class="dv">2</span>] <span class="sc">-</span> <span class="dv">1</span>, <span class="at">label =</span> asterisk), </span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a>              <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(boxplot)</span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="conditioning-comparisons" class="level3">
<h3 class="anchored" data-anchor-id="conditioning-comparisons">3.2.2 Conditioning comparisons</h3>
<ol type="1">
<li>Extract Life Stage and Conditioning Treatments:</li>
</ol>
<ul>
<li>The comparison string is split into its components (<code>life_stage</code>, <code>treatment1</code>, and <code>treatment2</code>).</li>
</ul>
<ol start="2" type="1">
<li>Filter Data:</li>
</ol>
<ul>
<li>The filtered_data data frame is filtered to include only the rows with the relevant life stage and conditioning treatments.</li>
</ul>
<ol start="3" type="1">
<li>Check for Both Treatments:</li>
</ol>
<ul>
<li>Ensure that both treatments are included in the <code>filtered_data</code>.</li>
</ul>
<ol start="4" type="1">
<li>Filter T-Test Results:</li>
</ol>
<ul>
<li><p>The <code>t_test_results_filtered</code> data frame is filtered for the specific comparison.</p></li>
<li><p>The <code>t_test_results_with_asterisks</code> data frame is created to include only the rows with asterisks.</p></li>
</ul>
<ol start="5" type="1">
<li>Format the Title:</li>
</ol>
<ul>
<li><p>The <code>formatted_title</code> variable is created by capitalizing the first letter of each component and concatenating them with ” - ” and ” vs.&nbsp;” in between.</p></li>
<li><p>This should create box plots comparing conditioning treatments within each life stage, with titles formatted as <code>&lt;life.stage&gt; - Treated vs. Control</code>.</p></li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create box plots for each comparison of conditioning treatments within life stages</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>create_boxplot_conditioning <span class="ot">&lt;-</span> <span class="cf">function</span>(data, comparison, t_test_results) {</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract life stage and conditioning treatments from comparison</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  comparison_parts <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(comparison, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>))</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  life_stage <span class="ot">&lt;-</span> comparison_parts[<span class="dv">1</span>]</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  treatment1 <span class="ot">&lt;-</span> comparison_parts[<span class="dv">2</span>]</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  treatment2 <span class="ot">&lt;-</span> comparison_parts[<span class="dv">3</span>]</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print life stage and treatments</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(paste("Life stage and treatments for comparison:", comparison))</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(c(life_stage, treatment1, treatment2))</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter data for the relevant life stage and conditioning treatments</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  filtered_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(life.stage <span class="sc">==</span> life_stage, conditioning.treatment <span class="sc">%in%</span> <span class="fu">c</span>(treatment1, treatment2))</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print filtered data</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Filtered data:")</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(filtered_data)</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both treatments are included</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">c</span>(treatment1, treatment2) <span class="sc">%in%</span> <span class="fu">unique</span>(filtered_data<span class="sc">$</span>conditioning.treatment))) {</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Not all treatments are included in the filtered data"</span>)</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>  y_limits <span class="ot">&lt;-</span> <span class="fu">range</span>(filtered_data<span class="sc">$</span>delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print y_limits</span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Y limits:")</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(y_limits)</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for the current comparison</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>  t_test_results_filtered <span class="ot">&lt;-</span> t_test_results <span class="sc">%&gt;%</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(comparison <span class="sc">==</span> <span class="sc">!!</span>comparison)</span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print filtered t_test_results</span></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Filtered t_test_results:")</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(t_test_results_filtered)</span></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for asterisks</span></span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>  t_test_results_with_asterisks <span class="ot">&lt;-</span> t_test_results_filtered <span class="sc">%&gt;%</span></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(asterisk <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print t_test_results_with_asterisks</span></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("t_test_results_with_asterisks:")</span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(t_test_results_with_asterisks)</span></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Format the title</span></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>  formatted_title <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">toupper</span>(<span class="fu">substring</span>(life_stage, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(life_stage, <span class="dv">2</span>), </span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" - "</span>, </span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(treatment1, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(treatment1, <span class="dv">2</span>), </span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" vs. "</span>, </span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(treatment2, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(treatment2, <span class="dv">2</span>))</span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>  boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(filtered_data, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> conditioning.treatment)) <span class="sc">+</span></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"salmon"</span>)) <span class="sc">+</span></span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(y_limits) <span class="sc">+</span></span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Delta Cq"</span>, <span class="at">title =</span> formatted_title) <span class="sc">+</span></span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Highlighted section: Adds asterisks</span></span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> t_test_results_with_asterisks, </span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> y_limits[<span class="dv">2</span>] <span class="sc">-</span> <span class="dv">1</span>, <span class="at">label =</span> asterisk), </span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a>              <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(boxplot)</span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="acute-comparisons" class="level3">
<h3 class="anchored" data-anchor-id="acute-comparisons">3.2.3 Acute comparisons</h3>
<ol type="1">
<li>Extract Life Stage and Acute Treatments:</li>
</ol>
<ul>
<li>The comparison string is split into its components (<code>life_stage</code>, <code>treatment1</code>, and <code>treatment2</code>).</li>
</ul>
<ol start="2" type="1">
<li>Filter Data:</li>
</ol>
<ul>
<li>The <code>filtered_data</code> data frame is filtered to include only the rows with the relevant life stage and acute treatments.</li>
</ul>
<ol start="3" type="1">
<li>Check for Both Treatments:</li>
</ol>
<ul>
<li>Ensure that both treatments are included in the <code>filtered_data</code>.</li>
</ul>
<ol start="4" type="1">
<li>Filter T-Test Results:</li>
</ol>
<ul>
<li><p>The <code>t_test_results_filtered</code> data frame is filtered for the specific comparison.</p></li>
<li><p>The <code>t_test_results_with_asterisks</code> data frame is created to include only the rows with asterisks. Format the Title:</p></li>
</ul>
<ol start="5" type="1">
<li>The formatted_title variable is created by capitalizing the first letter of each component and concatenating them with ” - ” and ” vs.&nbsp;” in between.</li>
</ol>
<ul>
<li>This should create box plots comparing acute treatments within each life stage, with titles formatted as <code>&lt;life.stage&gt; - Ambient vs. High</code>.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create box plots for each comparison of acute treatments within life stages</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>create_boxplot_acute <span class="ot">&lt;-</span> <span class="cf">function</span>(data, comparison, t_test_results) {</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract life stage and acute treatments from comparison</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  comparison_parts <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(comparison, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>))</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  life_stage <span class="ot">&lt;-</span> comparison_parts[<span class="dv">1</span>]</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  treatment1 <span class="ot">&lt;-</span> comparison_parts[<span class="dv">2</span>]</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  treatment2 <span class="ot">&lt;-</span> comparison_parts[<span class="dv">3</span>]</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print life stage and treatments</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(paste("Life stage and treatments for comparison:", comparison))</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(c(life_stage, treatment1, treatment2))</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter data for the relevant life stage and acute treatments</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  filtered_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(life.stage <span class="sc">==</span> life_stage, acute.treatment <span class="sc">%in%</span> <span class="fu">c</span>(treatment1, treatment2))</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print filtered data</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Filtered data:")</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(filtered_data)</span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both treatments are included</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">c</span>(treatment1, treatment2) <span class="sc">%in%</span> <span class="fu">unique</span>(filtered_data<span class="sc">$</span>acute.treatment))) {</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Not all treatments are included in the filtered data"</span>)</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>  y_limits <span class="ot">&lt;-</span> <span class="fu">range</span>(filtered_data<span class="sc">$</span>delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print y_limits</span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Y limits:")</span></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(y_limits)</span></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for the current comparison</span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>  t_test_results_filtered <span class="ot">&lt;-</span> t_test_results <span class="sc">%&gt;%</span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(comparison <span class="sc">==</span> <span class="sc">!!</span>comparison)</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print filtered t_test_results</span></span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("Filtered t_test_results:")</span></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(t_test_results_filtered)</span></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for asterisks</span></span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>  t_test_results_with_asterisks <span class="ot">&lt;-</span> t_test_results_filtered <span class="sc">%&gt;%</span></span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(asterisk <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Debugging: Print t_test_results_with_asterisks</span></span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print("t_test_results_with_asterisks:")</span></span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(t_test_results_with_asterisks)</span></span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Format the title</span></span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a>  formatted_title <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">toupper</span>(<span class="fu">substring</span>(life_stage, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(life_stage, <span class="dv">2</span>), </span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" - "</span>, </span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(treatment1, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(treatment1, <span class="dv">2</span>), </span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" vs. "</span>, </span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(treatment2, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(treatment2, <span class="dv">2</span>))</span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a>  boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(filtered_data, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> acute.treatment)) <span class="sc">+</span></span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"salmon"</span>)) <span class="sc">+</span></span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(y_limits) <span class="sc">+</span></span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Delta Cq"</span>, <span class="at">title =</span> formatted_title) <span class="sc">+</span></span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Highlighted section: Adds asterisks</span></span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> t_test_results_with_asterisks, </span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> y_limits[<span class="dv">2</span>] <span class="sc">-</span> <span class="dv">1</span>, <span class="at">label =</span> asterisk), </span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a>              <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"magenta"</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(boxplot)</span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="acute-treatements-within-life-stage-conditioning" class="level3">
<h3 class="anchored" data-anchor-id="acute-treatements-within-life-stage-conditioning">3.2.4 Acute treatements within life stage conditioning</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create box plots for each comparison of acute treatments within life stages and conditioning treatments</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>create_boxplot_acute_conditioning <span class="ot">&lt;-</span> <span class="cf">function</span>(data, comparison, t_test_results) {</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract life stage, conditioning treatment, and acute treatments from comparison</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  comparison_parts <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(comparison, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>))</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  life_stage <span class="ot">&lt;-</span> comparison_parts[<span class="dv">1</span>]</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  conditioning_treatment <span class="ot">&lt;-</span> comparison_parts[<span class="dv">2</span>]</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  treatment1 <span class="ot">&lt;-</span> comparison_parts[<span class="dv">3</span>]</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  treatment2 <span class="ot">&lt;-</span> comparison_parts[<span class="dv">5</span>]</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter data for the relevant life stage, conditioning treatment, and acute treatments</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  filtered_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(life.stage <span class="sc">==</span> life_stage, conditioning.treatment <span class="sc">==</span> conditioning_treatment, acute.treatment <span class="sc">%in%</span> <span class="fu">c</span>(treatment1, treatment2))</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if both treatments are included</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">c</span>(treatment1, treatment2) <span class="sc">%in%</span> <span class="fu">unique</span>(filtered_data<span class="sc">$</span>acute.treatment))) {</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Not all treatments are included in the filtered data"</span>)</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>  y_limits <span class="ot">&lt;-</span> <span class="fu">range</span>(filtered_data<span class="sc">$</span>delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for the current comparison</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>  t_test_results_filtered <span class="ot">&lt;-</span> t_test_results <span class="sc">%&gt;%</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(comparison <span class="sc">==</span> <span class="sc">!!</span>comparison)</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter t_test_results for asterisks</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>  t_test_results_with_asterisks <span class="ot">&lt;-</span> t_test_results_filtered <span class="sc">%&gt;%</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(asterisk <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Format the title</span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>  formatted_title <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">toupper</span>(<span class="fu">substring</span>(life_stage, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(life_stage, <span class="dv">2</span>), </span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" - "</span>, </span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(conditioning_treatment, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(conditioning_treatment, <span class="dv">2</span>), </span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" - "</span>, </span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(treatment1, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(treatment1, <span class="dv">2</span>), </span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>                            <span class="st">" vs. "</span>, </span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(<span class="fu">substring</span>(treatment2, <span class="dv">1</span>, <span class="dv">1</span>)), <span class="fu">substring</span>(treatment2, <span class="dv">2</span>))</span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>  boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(filtered_data, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> acute.treatment)) <span class="sc">+</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"darkgray"</span>, <span class="st">"salmon"</span>)) <span class="sc">+</span></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(y_limits) <span class="sc">+</span></span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Delta Cq"</span>, <span class="at">title =</span> formatted_title) <span class="sc">+</span></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Adds asterisks</span></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> t_test_results_with_asterisks, </span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> y_limits[<span class="dv">2</span>] <span class="sc">-</span> <span class="dv">1</span>, <span class="at">label =</span> asterisk), </span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a>              <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>)</span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(boxplot)</span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
</section>
<section id="read-in-files" class="level1">
<h1>4 Read in files</h1>
<div class="sourceCode" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get a list of all CSV files in the directory with the naming structure "*Cq-Results.csv"</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>cq_file_list <span class="ot">&lt;-</span> <span class="fu">list</span>() <span class="co"># Initialize list</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>cq_file_list <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> cqs_directory, <span class="at">pattern =</span> <span class="st">"Cq-Results</span><span class="sc">\\</span><span class="st">.csv$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize an empty list to store the data frames</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>data_frames_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each file and read it into a data frame, then add it to the list</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (file <span class="cf">in</span> cq_file_list) {</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(file, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  data<span class="sc">$</span>Sample <span class="ot">&lt;-</span> <span class="fu">as.character</span>(data<span class="sc">$</span>Sample)  <span class="co"># Convert Sample column to character type</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  data_frames_list[[file]] <span class="ot">&lt;-</span> data</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine all data frames into a single data frame</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(data_frames_list, <span class="at">.id =</span> <span class="st">"data_frame_id"</span>)</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(combined_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>'data.frame':   1920 obs. of  17 variables:
 $ data_frame_id         : chr  "../data/qPCR/Cq/sam_2024-12-10_11-47-51_CFX96_GAPDH-02-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-12-10_11-47-51_CFX96_GAPDH-02-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-12-10_11-47-51_CFX96_GAPDH-02-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-12-10_11-47-51_CFX96_GAPDH-02-Quantification-Cq-Results.csv" ...
 $ X                     : logi  NA NA NA NA NA NA ...
 $ Well                  : chr  "A01" "A02" "A03" "A04" ...
 $ Fluor                 : chr  "SYBR" "SYBR" "SYBR" "SYBR" ...
 $ Target                : chr  "Cg_GAPDH_205_F-355_R (SR IDs: 1172/3)" "Cg_GAPDH_205_F-355_R (SR IDs: 1172/3)" "Cg_GAPDH_205_F-355_R (SR IDs: 1172/3)" "Cg_GAPDH_205_F-355_R (SR IDs: 1172/3)" ...
 $ Content               : chr  "Unkn-01" "Unkn-01" "Unkn-01" "Unkn-02" ...
 $ Sample                : chr  "270" "270" "270" "271" ...
 $ Biological.Set.Name   : logi  NA NA NA NA NA NA ...
 $ Cq                    : num  24.8 24.8 25 24.4 24.3 ...
 $ Cq.Mean               : num  24.9 24.9 24.9 24.4 24.4 ...
 $ Cq.Std..Dev           : num  0.1005 0.1005 0.1005 0.0914 0.0914 ...
 $ Starting.Quantity..SQ.: num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Log.Starting.Quantity : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Mean               : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Std..Dev           : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Set.Point             : int  60 60 60 60 60 60 60 60 60 60 ...
 $ Well.Note             : logi  NA NA NA NA NA NA ...</code></pre>
</section>
<section id="clean-data" class="level1">
<h1>5 Clean data</h1>
<section id="replace-target-names" class="level2">
<h2 class="anchored" data-anchor-id="replace-target-names">5.1 Replace target names</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with Sample name "NTC"</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> combined_df[combined_df<span class="sc">$</span>Sample <span class="sc">!=</span> <span class="st">"NTC"</span>, ]</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace values in the Target column</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_GAPDH_205_F-355_R </span><span class="sc">\\</span><span class="st">(SR IDs: 1172/3</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"GAPDH"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_ATPsynthase_F/R </span><span class="sc">\\</span><span class="st">(SR IDs: 1385/6</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"ATPsynthase"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_cGAS </span><span class="sc">\\</span><span class="st">(SR IDs: 1826/7</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"cGAS"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_citrate_synthase </span><span class="sc">\\</span><span class="st">(SR IDs: 1383/4</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"citrate.sythase"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_DNMT1_F </span><span class="sc">\\</span><span class="st">(SR IDs: 1510/1</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"DNMT1"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_HSP70_F/R </span><span class="sc">\\</span><span class="st">(SR IDs: 598/9</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"HSP70"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_Hsp90_F/R </span><span class="sc">\\</span><span class="st">(SR IDs: 1532/3</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"HSP90"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>combined_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Cg_VIPERIN_F/R </span><span class="sc">\\</span><span class="st">(SR IDs: 1828/9</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">"VIPERIN"</span>, combined_df<span class="sc">$</span>Target)</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(combined_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>'data.frame':   1908 obs. of  17 variables:
 $ data_frame_id         : chr  "../data/qPCR/Cq/sam_2024-12-10_11-47-51_CFX96_GAPDH-02-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-12-10_11-47-51_CFX96_GAPDH-02-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-12-10_11-47-51_CFX96_GAPDH-02-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-12-10_11-47-51_CFX96_GAPDH-02-Quantification-Cq-Results.csv" ...
 $ X                     : logi  NA NA NA NA NA NA ...
 $ Well                  : chr  "A01" "A02" "A03" "A04" ...
 $ Fluor                 : chr  "SYBR" "SYBR" "SYBR" "SYBR" ...
 $ Target                : chr  "GAPDH" "GAPDH" "GAPDH" "GAPDH" ...
 $ Content               : chr  "Unkn-01" "Unkn-01" "Unkn-01" "Unkn-02" ...
 $ Sample                : chr  "270" "270" "270" "271" ...
 $ Biological.Set.Name   : logi  NA NA NA NA NA NA ...
 $ Cq                    : num  24.8 24.8 25 24.4 24.3 ...
 $ Cq.Mean               : num  24.9 24.9 24.9 24.4 24.4 ...
 $ Cq.Std..Dev           : num  0.1005 0.1005 0.1005 0.0914 0.0914 ...
 $ Starting.Quantity..SQ.: num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Log.Starting.Quantity : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Mean               : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Std..Dev           : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Set.Point             : int  60 60 60 60 60 60 60 60 60 60 ...
 $ Well.Note             : logi  NA NA NA NA NA NA ...</code></pre>
</section>
<section id="identify-samples-with-cq.std..dev-0.5" class="level2">
<h2 class="anchored" data-anchor-id="identify-samples-with-cq.std..dev-0.5">5.2 Identify Samples with Cq.Std..Dev &gt; 0.5</h2>
<div class="sourceCode" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out rows where Cq.Std..Dev is NA</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> combined_df[<span class="sc">!</span><span class="fu">is.na</span>(combined_df<span class="sc">$</span>Cq.Std..Dev), ]</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter rows where Cq.Std..Dev is greater than 0.5</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>high_cq_std_dev <span class="ot">&lt;-</span> combined_df[combined_df<span class="sc">$</span>Cq.Std..Dev <span class="sc">&gt;</span> <span class="fl">0.5</span>, ]</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the filtered rows with specified columns, without row names</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(high_cq_std_dev[, <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"Sample"</span>, <span class="st">"Cq"</span>, <span class="st">"Cq.Std..Dev"</span>)], <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>          Target Sample       Cq Cq.Std..Dev
           GAPDH    316 23.94926   8.5684728
           GAPDH    316 24.14183   8.5684728
           GAPDH    316 38.88564   8.5684728
           GAPDH    213 26.98012   2.2910353
           GAPDH    213 23.00009   2.2910353
           GAPDH    213 26.95634   2.2910353
           GAPDH    263 22.42154   0.8731474
           GAPDH    263 23.77008   0.8731474
           GAPDH    263 24.05667   0.8731474
 citrate.sythase    230 24.44066   4.4783429
 citrate.sythase    230 24.40421   4.4783429
 citrate.sythase    230 32.17909   4.4783429
         VIPERIN    227 30.47773   3.5152533
         VIPERIN    227 30.37738   3.5152533
         VIPERIN    227 36.51553   3.5152533
         VIPERIN    245 26.05748   5.1635899
         VIPERIN    245 34.98192   5.1635899
         VIPERIN    245 26.01928   5.1635899
         VIPERIN    341 26.48675   2.9838590
         VIPERIN    341 31.67235   2.9838590
         VIPERIN    341 26.52174   2.9838590
         VIPERIN    344 29.98184   2.3712440
         VIPERIN    344 25.90358   2.3712440
         VIPERIN    344 25.84648   2.3712440
         VIPERIN    355 28.79712   0.5821437
         VIPERIN    355 29.57428   0.5821437
         VIPERIN    355 28.43490   0.5821437</code></pre>
</section>
<section id="remove-bad-technical-reps" class="level2">
<h2 class="anchored" data-anchor-id="remove-bad-technical-reps">5.3 Remove bad technical reps</h2>
<div class="sourceCode" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by Sample and Target, then filter out the outlier replicate</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>combined.fitered_df<span class="ot">&lt;-</span> combined_df <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Sample, Target) <span class="sc">%&gt;%</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">abs</span>(Cq <span class="sc">-</span> <span class="fu">mean</span>(Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">&lt;=</span> Cq.Std..Dev)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the filtered data frame</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(combined.fitered_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>gropd_df [1,264 × 17] (S3: grouped_df/tbl_df/tbl/data.frame)
 $ data_frame_id         : chr [1:1264] "../data/qPCR/Cq/sam_2024-12-10_11-47-51_CFX96_GAPDH-02-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-12-10_11-47-51_CFX96_GAPDH-02-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-12-10_11-47-51_CFX96_GAPDH-02-Quantification-Cq-Results.csv" "../data/qPCR/Cq/sam_2024-12-10_11-47-51_CFX96_GAPDH-02-Quantification-Cq-Results.csv" ...
 $ X                     : logi [1:1264] NA NA NA NA NA NA ...
 $ Well                  : chr [1:1264] "A01" "A02" "A04" "A06" ...
 $ Fluor                 : chr [1:1264] "SYBR" "SYBR" "SYBR" "SYBR" ...
 $ Target                : chr [1:1264] "GAPDH" "GAPDH" "GAPDH" "GAPDH" ...
 $ Content               : chr [1:1264] "Unkn-01" "Unkn-01" "Unkn-02" "Unkn-02" ...
 $ Sample                : chr [1:1264] "270" "270" "271" "271" ...
 $ Biological.Set.Name   : logi [1:1264] NA NA NA NA NA NA ...
 $ Cq                    : num [1:1264] 24.8 24.8 24.4 24.4 24.2 ...
 $ Cq.Mean               : num [1:1264] 24.9 24.9 24.4 24.4 23.9 ...
 $ Cq.Std..Dev           : num [1:1264] 0.1005 0.1005 0.0914 0.0914 0.2866 ...
 $ Starting.Quantity..SQ.: num [1:1264] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Log.Starting.Quantity : num [1:1264] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Mean               : num [1:1264] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Std..Dev           : num [1:1264] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Set.Point             : int [1:1264] 60 60 60 60 60 60 60 60 60 60 ...
 $ Well.Note             : logi [1:1264] NA NA NA NA NA NA ...
 - attr(*, "groups")= tibble [632 × 3] (S3: tbl_df/tbl/data.frame)
  ..$ Sample: chr [1:632] "201" "201" "201" "201" ...
  ..$ Target: chr [1:632] "ATPsynthase" "DNMT1" "GAPDH" "HSP70" ...
  ..$ .rows : list&lt;int&gt; [1:632] 
  .. ..$ : int [1:2] 65 66
  .. ..$ : int [1:2] 633 634
  .. ..$ : int [1:2] 381 382
  .. ..$ : int [1:2] 791 792
  .. ..$ : int [1:2] 949 950
  .. ..$ : int [1:2] 1107 1108
  .. ..$ : int [1:2] 223 224
  .. ..$ : int [1:2] 475 476
  .. ..$ : int [1:2] 67 68
  .. ..$ : int [1:2] 635 636
  .. ..$ : int [1:2] 383 384
  .. ..$ : int [1:2] 793 794
  .. ..$ : int [1:2] 951 952
  .. ..$ : int [1:2] 1109 1110
  .. ..$ : int [1:2] 225 226
  .. ..$ : int [1:2] 477 478
  .. ..$ : int [1:2] 69 70
  .. ..$ : int [1:2] 637 638
  .. ..$ : int [1:2] 385 386
  .. ..$ : int [1:2] 795 796
  .. ..$ : int [1:2] 953 954
  .. ..$ : int [1:2] 1111 1112
  .. ..$ : int [1:2] 227 228
  .. ..$ : int [1:2] 479 480
  .. ..$ : int [1:2] 71 72
  .. ..$ : int [1:2] 639 640
  .. ..$ : int [1:2] 387 388
  .. ..$ : int [1:2] 797 798
  .. ..$ : int [1:2] 955 956
  .. ..$ : int [1:2] 1113 1114
  .. ..$ : int [1:2] 229 230
  .. ..$ : int [1:2] 481 482
  .. ..$ : int [1:2] 73 74
  .. ..$ : int [1:2] 641 642
  .. ..$ : int [1:2] 389 390
  .. ..$ : int [1:2] 799 800
  .. ..$ : int [1:2] 957 958
  .. ..$ : int [1:2] 1115 1116
  .. ..$ : int [1:2] 231 232
  .. ..$ : int [1:2] 483 484
  .. ..$ : int [1:2] 75 76
  .. ..$ : int [1:2] 643 644
  .. ..$ : int [1:2] 391 392
  .. ..$ : int [1:2] 801 802
  .. ..$ : int [1:2] 959 960
  .. ..$ : int [1:2] 1117 1118
  .. ..$ : int [1:2] 233 234
  .. ..$ : int [1:2] 485 486
  .. ..$ : int [1:2] 77 78
  .. ..$ : int [1:2] 645 646
  .. ..$ : int [1:2] 393 394
  .. ..$ : int [1:2] 803 804
  .. ..$ : int [1:2] 961 962
  .. ..$ : int [1:2] 1119 1120
  .. ..$ : int [1:2] 235 236
  .. ..$ : int [1:2] 487 488
  .. ..$ : int [1:2] 79 80
  .. ..$ : int [1:2] 647 648
  .. ..$ : int [1:2] 395 396
  .. ..$ : int [1:2] 805 806
  .. ..$ : int [1:2] 963 964
  .. ..$ : int [1:2] 1121 1122
  .. ..$ : int [1:2] 237 238
  .. ..$ : int [1:2] 489 490
  .. ..$ : int [1:2] 81 82
  .. ..$ : int [1:2] 649 650
  .. ..$ : int [1:2] 397 398
  .. ..$ : int [1:2] 807 808
  .. ..$ : int [1:2] 965 966
  .. ..$ : int [1:2] 1123 1124
  .. ..$ : int [1:2] 239 240
  .. ..$ : int [1:2] 491 492
  .. ..$ : int [1:2] 83 84
  .. ..$ : int [1:2] 651 652
  .. ..$ : int [1:2] 399 400
  .. ..$ : int [1:2] 809 810
  .. ..$ : int [1:2] 967 968
  .. ..$ : int [1:2] 1125 1126
  .. ..$ : int [1:2] 241 242
  .. ..$ : int [1:2] 493 494
  .. ..$ : int [1:2] 85 86
  .. ..$ : int [1:2] 653 654
  .. ..$ : int [1:2] 401 402
  .. ..$ : int [1:2] 811 812
  .. ..$ : int [1:2] 969 970
  .. ..$ : int [1:2] 1127 1128
  .. ..$ : int [1:2] 243 244
  .. ..$ : int [1:2] 495 496
  .. ..$ : int [1:2] 87 88
  .. ..$ : int [1:2] 655 656
  .. ..$ : int [1:2] 403 404
  .. ..$ : int [1:2] 813 814
  .. ..$ : int [1:2] 971 972
  .. ..$ : int [1:2] 1129 1130
  .. ..$ : int [1:2] 245 246
  .. ..$ : int [1:2] 497 498
  .. ..$ : int [1:2] 89 90
  .. ..$ : int [1:2] 657 658
  .. ..$ : int [1:2] 405 406
  .. .. [list output truncated]
  .. ..@ ptype: int(0) 
  ..- attr(*, ".drop")= logi TRUE</code></pre>
</section>
</section>
<section id="group-samples-by-target" class="level1">
<h1>6 Group samples by target</h1>
<div class="sourceCode" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by Sample and Target, then summarize to get unique rows for each sample</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>grouped_df <span class="ot">&lt;-</span> combined.fitered_df<span class="sc">%&gt;%</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Sample, Target) <span class="sc">%&gt;%</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Cq.Mean =</span> <span class="fu">mean</span>(Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(grouped_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [632 × 3] (S3: tbl_df/tbl/data.frame)
 $ Sample : chr [1:632] "201" "201" "201" "201" ...
 $ Target : chr [1:632] "ATPsynthase" "DNMT1" "GAPDH" "HSP70" ...
 $ Cq.Mean: num [1:632] 24 28.7 23.8 29.3 25.5 ...</code></pre>
</section>
<section id="add-life-stage-and-treatment-cols" class="level1">
<h1>7 Add life stage and treatment cols</h1>
<div class="sourceCode" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize new columns</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>grouped_df <span class="ot">&lt;-</span> grouped_df <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">life.stage =</span> <span class="cn">NA_character_</span>,</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">conditioning.treatment =</span> <span class="cn">NA_character_</span>,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">acute.treatment =</span> <span class="cn">NA_character_</span>)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each vector</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (vec_name <span class="cf">in</span> <span class="fu">names</span>(groups_list)) {</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  vec <span class="ot">&lt;-</span> groups_list[[vec_name]]</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  stage <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(vec_name, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  conditioning_treatment <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(vec_name, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  acute_treatment <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(vec_name, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>)[[<span class="dv">1</span>]][<span class="dv">3</span>]</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Loop through each row in grouped_df</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(grouped_df)) {</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>    sample <span class="ot">&lt;-</span> grouped_df<span class="sc">$</span>Sample[i]</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if sample is in the vector</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (sample <span class="sc">%in%</span> vec) {</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Update life.stage and treatment columns</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>      grouped_df<span class="sc">$</span>life.stage[i] <span class="ot">&lt;-</span> stage</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>      grouped_df<span class="sc">$</span>conditioning.treatment[i] <span class="ot">&lt;-</span> conditioning_treatment</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>      grouped_df<span class="sc">$</span>acute.treatment[i] <span class="ot">&lt;-</span>acute_treatment</span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(grouped_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [632 × 6] (S3: tbl_df/tbl/data.frame)
 $ Sample                : chr [1:632] "201" "201" "201" "201" ...
 $ Target                : chr [1:632] "ATPsynthase" "DNMT1" "GAPDH" "HSP70" ...
 $ Cq.Mean               : num [1:632] 24 28.7 23.8 29.3 25.5 ...
 $ life.stage            : chr [1:632] "seed" "seed" "seed" "seed" ...
 $ conditioning.treatment: chr [1:632] "treated" "treated" "treated" "treated" ...
 $ acute.treatment       : chr [1:632] "ambient" "ambient" "ambient" "ambient" ...</code></pre>
</section>
<section id="delta-cq-to-normalizing-gene" class="level1">
<h1>8 Delta Cq to Normalizing Gene</h1>
<div class="sourceCode" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta Cq by subtracting GAPDH Cq.Mean from each corresponding Sample Cq.Mean</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>delta_Cq_df <span class="ot">&lt;-</span> <span class="fu">calculate_delta_Cq</span>(grouped_df)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filters out normalizing gene, since no need to compare normalizing gene to itself.</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>delta_Cq_df <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(life.stage), <span class="sc">!</span><span class="fu">is.na</span>(Target), Target <span class="sc">!=</span> <span class="st">"GAPDH"</span>)</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_Cq_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [553 × 7] (S3: tbl_df/tbl/data.frame)
 $ Sample                : chr [1:553] "201" "201" "201" "201" ...
 $ Target                : chr [1:553] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
 $ Cq.Mean               : num [1:553] 24 28.7 29.3 25.5 28.9 ...
 $ life.stage            : chr [1:553] "seed" "seed" "seed" "seed" ...
 $ conditioning.treatment: chr [1:553] "treated" "treated" "treated" "treated" ...
 $ acute.treatment       : chr [1:553] "ambient" "ambient" "ambient" "ambient" ...
 $ delta_Cq              : num [1:553] 0.128 4.882 5.476 1.707 5.02 ...</code></pre>
<section id="t-tests" class="level2">
<h2 class="anchored" data-anchor-id="t-tests">8.1 t-tests</h2>
<section id="life-stages" class="level3">
<h3 class="anchored" data-anchor-id="life-stages">8.1.1 Life Stages</h3>
<p>This code does the following:</p>
<ol type="1">
<li>Extracts the unique life.stage levels from the data frame.</li>
<li>Generates all possible pairs of life.stage levels using the combn function.</li>
<li>Iterates over each pair and performs the t-test for each Target. Adds an asterisk column and an asterisk if the p-value is &lt;= 0.05. Useful for downstream parsing.</li>
<li>Stores the results in a list and combines them into a single data frame.</li>
<li>Adds a comparison column to indicate which life.stage levels were compared.</li>
</ol>
<div class="sourceCode" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract unique life.stage levels</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>unique_life_stages <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df<span class="sc">$</span>life.stage)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate all possible pairs of life.stage levels</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>life_stage_pairs <span class="ot">&lt;-</span> <span class="fu">combn</span>(unique_life_stages, <span class="dv">2</span>, <span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store results</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>life_stage_t_test_results_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pair <span class="cf">in</span> life_stage_pairs) {</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>  stage1 <span class="ot">&lt;-</span> pair[<span class="dv">1</span>]</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>  stage2 <span class="ot">&lt;-</span> pair[<span class="dv">2</span>]</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Perform t-test for each Target comparing the two life.stage levels</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>  t_test_results <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(life.stage <span class="sc">%in%</span> <span class="fu">c</span>(stage1, stage2)) <span class="sc">%&gt;%</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>      <span class="at">t_test_result =</span> <span class="fu">list</span>(<span class="fu">t.test</span>(delta_Cq <span class="sc">~</span> life.stage))</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>      <span class="at">estimate_diff =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>estimate[<span class="dv">1</span>] <span class="sc">-</span> x<span class="sc">$</span>estimate[<span class="dv">2</span>]),</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>      <span class="at">p_value =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>p.value),</span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>      <span class="at">asterisk =</span> <span class="fu">ifelse</span>(p_value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="st">""</span>), <span class="co"># Adds asterisk column and asterisk for p-value.</span></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>      <span class="at">comparison =</span> <span class="fu">paste</span>(stage1, <span class="st">"vs"</span>, stage2, <span class="at">sep =</span> <span class="st">"."</span>)</span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">!</span>t_test_result)</span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a>  life_stage_t_test_results_list[[<span class="fu">paste</span>(stage1, stage2, <span class="at">sep =</span> <span class="st">"."</span>)]] <span class="ot">&lt;-</span> t_test_results</span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results into a single data frame</span></span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a>life_stage_t_test_results_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(life_stage_t_test_results_list, <span class="at">.id =</span> <span class="st">"comparison"</span>)</span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a><span class="co"># View the results</span></span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(life_stage_t_test_results_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code># A tibble: 42 × 5
   Target          estimate_diff p_value asterisk comparison   
   &lt;chr&gt;                   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;        
 1 ATPsynthase            0.251  0.107   ""       seed.adult   
 2 DNMT1                  0.442  0.147   ""       seed.adult   
 3 HSP70                  0.466  0.365   ""       seed.adult   
 4 HSP90                  0.329  0.163   ""       seed.adult   
 5 VIPERIN               -0.771  0.00645 "*"      seed.adult   
 6 cGAS                   0.393  0.271   ""       seed.adult   
 7 citrate.sythase        0.328  0.132   ""       seed.adult   
 8 ATPsynthase            0.148  0.304   ""       seed.juvenile
 9 DNMT1                  0.0824 0.807   ""       seed.juvenile
10 HSP70                  0.944  0.0803  ""       seed.juvenile
# ℹ 32 more rows</code></pre>
</section>
<section id="conditioning-treatments" class="level3">
<h3 class="anchored" data-anchor-id="conditioning-treatments">8.1.2 Conditioning treatments</h3>
<p>This code does the following:</p>
<ol type="1">
<li>Extracts the unique life.stage levels from the data frame.</li>
<li>For each life.stage, extracts the unique conditioning.treatment levels.</li>
<li>Generates all possible pairs of conditioning.treatment levels within each life.stage.</li>
<li>Iterates over each pair and performs the t-test for each Target. Adds an asterisk column and an asterisk if the p-value is &lt;= 0.05. Useful for downstream parsing.</li>
<li>Stores the results in a list and combines them into a single data frame.</li>
<li>Adds a comparison column to indicate which life.stage and conditioning.treatment levels were compared.</li>
</ol>
<div class="sourceCode" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract unique life.stage levels</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>unique_life_stages <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df<span class="sc">$</span>life.stage)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store results</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>conditioning_treatment_t_test_results_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (stage <span class="cf">in</span> unique_life_stages) {</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract unique conditioning.treatment levels within the current life.stage</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  unique_treatments <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(life.stage <span class="sc">==</span> stage) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(conditioning.treatment))</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Generate all possible pairs of conditioning.treatment levels</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  treatment_pairs <span class="ot">&lt;-</span> <span class="fu">combn</span>(unique_treatments, <span class="dv">2</span>, <span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (pair <span class="cf">in</span> treatment_pairs) {</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>    treatment1 <span class="ot">&lt;-</span> pair[<span class="dv">1</span>]</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>    treatment2 <span class="ot">&lt;-</span> pair[<span class="dv">2</span>]</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Perform t-test for each Target comparing the two conditioning.treatment levels within the current life.stage</span></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>    t_test_results <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(life.stage <span class="sc">==</span> stage, conditioning.treatment <span class="sc">%in%</span> <span class="fu">c</span>(treatment1, treatment2)) <span class="sc">%&gt;%</span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(</span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">t_test_result =</span> <span class="fu">list</span>(<span class="fu">t.test</span>(delta_Cq <span class="sc">~</span> conditioning.treatment))</span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>      ) <span class="sc">%&gt;%</span></span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(</span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">estimate_diff =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>estimate[<span class="dv">1</span>] <span class="sc">-</span> x<span class="sc">$</span>estimate[<span class="dv">2</span>]),</span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">p_value =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>p.value),</span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>        <span class="at">asterisk =</span> <span class="fu">ifelse</span>(p_value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="st">""</span>), <span class="co"># Adds asterisk column and asterisk for p-value.</span></span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>        <span class="at">comparison =</span> <span class="fu">paste</span>(stage, treatment1, <span class="st">"vs"</span>, treatment2, <span class="at">sep =</span> <span class="st">"."</span>)</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>      ) <span class="sc">%&gt;%</span></span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">!</span>t_test_result)</span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>    conditioning_treatment_t_test_results_list[[<span class="fu">paste</span>(stage, treatment1, treatment2, <span class="at">sep =</span> <span class="st">"."</span>)]] <span class="ot">&lt;-</span> t_test_results</span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results into a single data frame</span></span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a>conditioning_treatment_t_test_results_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(conditioning_treatment_t_test_results_list, <span class="at">.id =</span> <span class="st">"comparison"</span>)</span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a><span class="co"># View the results</span></span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(conditioning_treatment_t_test_results_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code># A tibble: 28 × 5
   Target          estimate_diff p_value asterisk comparison           
   &lt;chr&gt;                   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;                
 1 ATPsynthase           -0.345    0.193 ""       seed.treated.control 
 2 DNMT1                  0.211    0.701 ""       seed.treated.control 
 3 HSP70                 -0.608    0.444 ""       seed.treated.control 
 4 HSP90                 -0.150    0.652 ""       seed.treated.control 
 5 VIPERIN                0.0507   0.912 ""       seed.treated.control 
 6 cGAS                  -0.341    0.623 ""       seed.treated.control 
 7 citrate.sythase       -0.335    0.416 ""       seed.treated.control 
 8 ATPsynthase            0.0779   0.603 ""       adult.treated.control
 9 DNMT1                  0.312    0.278 ""       adult.treated.control
10 HSP70                 -0.941    0.177 ""       adult.treated.control
# ℹ 18 more rows</code></pre>
</section>
<section id="acute-treatments" class="level3">
<h3 class="anchored" data-anchor-id="acute-treatments">8.1.3 Acute treatments</h3>
<p>This code does the following:</p>
<ol type="1">
<li>Extracts the unique life.stage levels from the data frame.</li>
<li>For each life.stage, extracts the unique acute.treatment levels.</li>
<li>Generates all possible pairs of acute.treatment levels within each life.stage.</li>
<li>Iterates over each pair and performs the t-test for each Target. Adds an asterisk column and an asterisk if the p-value is &lt;= 0.05. Useful for downstream parsing.</li>
<li>Stores the results in a list and combines them into a single data frame.</li>
<li>Adds a comparison column to indicate which life.stage and acute.treatment levels were compared.</li>
</ol>
<p>Excludes <code>seed</code> and <code>spat</code>, as these were only held at <code>ambient</code> for the acute treatment.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract unique life.stage levels, excluding 'seed' and 'spat'</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>unique_life_stages <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df<span class="sc">$</span>life.stage)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>unique_life_stages <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(unique_life_stages, <span class="fu">c</span>(<span class="st">"seed"</span>, <span class="st">"spat"</span>))</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store results</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>acute_treatment_t_test_results_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (stage <span class="cf">in</span> unique_life_stages) {</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract unique acute.treatment levels within the current life.stage</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>  unique_treatments <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(life.stage <span class="sc">==</span> stage) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(acute.treatment))</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if there are at least 2 unique treatments</span></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(unique_treatments) <span class="sc">&gt;=</span> <span class="dv">2</span>) {</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Generate all possible pairs of acute.treatment levels</span></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>    treatment_pairs <span class="ot">&lt;-</span> <span class="fu">combn</span>(unique_treatments, <span class="dv">2</span>, <span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (pair <span class="cf">in</span> treatment_pairs) {</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>      treatment1 <span class="ot">&lt;-</span> pair[<span class="dv">1</span>]</span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>      treatment2 <span class="ot">&lt;-</span> pair[<span class="dv">2</span>]</span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Perform t-test for each Target comparing the two acute.treatment levels within the current life.stage</span></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>      t_test_results <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(life.stage <span class="sc">==</span> stage, acute.treatment <span class="sc">%in%</span> <span class="fu">c</span>(treatment1, treatment2)) <span class="sc">%&gt;%</span></span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>        <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>        <span class="fu">summarise</span>(</span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>          <span class="at">t_test_result =</span> <span class="fu">list</span>(<span class="fu">t.test</span>(delta_Cq <span class="sc">~</span> acute.treatment))</span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(</span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>          <span class="at">estimate_diff =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>estimate[<span class="dv">1</span>] <span class="sc">-</span> x<span class="sc">$</span>estimate[<span class="dv">2</span>]),</span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>          <span class="at">p_value =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>p.value),</span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a>          <span class="at">asterisk =</span> <span class="fu">ifelse</span>(p_value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="st">""</span>), <span class="co"># Adds asterisk column and asterisk for p-value.</span></span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>          <span class="at">comparison =</span> <span class="fu">paste</span>(stage, treatment1, <span class="st">"vs"</span>, treatment2, <span class="at">sep =</span> <span class="st">"."</span>)</span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>        <span class="fu">select</span>(<span class="sc">!</span>t_test_result)</span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a>      acute_treatment_t_test_results_list[[<span class="fu">paste</span>(stage, treatment1, treatment2, <span class="at">sep =</span> <span class="st">"."</span>)]] <span class="ot">&lt;-</span> t_test_results</span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results into a single data frame</span></span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a>acute_treatment_t_test_results_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(acute_treatment_t_test_results_list, <span class="at">.id =</span> <span class="st">"comparison"</span>)</span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a><span class="co"># View the results</span></span>
<span id="cb26-46"><a href="#cb26-46" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(acute_treatment_t_test_results_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code># A tibble: 14 × 5
   Target          estimate_diff p_value asterisk comparison           
   &lt;chr&gt;                   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;                
 1 ATPsynthase            0.0605   0.687 ""       adult.ambient.high   
 2 DNMT1                  0.314    0.275 ""       adult.ambient.high   
 3 HSP70                  0.276    0.696 ""       adult.ambient.high   
 4 HSP90                  0.499    0.149 ""       adult.ambient.high   
 5 VIPERIN                0.323    0.251 ""       adult.ambient.high   
 6 cGAS                   0.329    0.200 ""       adult.ambient.high   
 7 citrate.sythase        0.0668   0.622 ""       adult.ambient.high   
 8 ATPsynthase           -0.0370   0.738 ""       juvenile.ambient.high
 9 DNMT1                 -0.672    0.121 ""       juvenile.ambient.high
10 HSP70                  0.745    0.319 ""       juvenile.ambient.high
11 HSP90                  0.0450   0.859 ""       juvenile.ambient.high
12 VIPERIN               -0.424    0.304 ""       juvenile.ambient.high
13 cGAS                  -0.203    0.474 ""       juvenile.ambient.high
14 citrate.sythase        0.0399   0.870 ""       juvenile.ambient.high</code></pre>
</section>
<section id="acute-within-life-stage-and-conditioning" class="level3">
<h3 class="anchored" data-anchor-id="acute-within-life-stage-and-conditioning">8.1.4 Acute within life stage and conditioning</h3>
<div class="sourceCode" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract unique life.stage levels, excluding 'seed' and 'spat'</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>unique_life_stages <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df<span class="sc">$</span>life.stage)</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>unique_life_stages <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(unique_life_stages, <span class="fu">c</span>(<span class="st">"seed"</span>, <span class="st">"spat"</span>))</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract unique conditioning.treatment levels</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>unique_conditioning_treatments <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df<span class="sc">$</span>conditioning.treatment)</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store results</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>acute_treatment_within_life.stages_conditioning_t_test_results_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (stage <span class="cf">in</span> unique_life_stages) {</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (conditioning <span class="cf">in</span> unique_conditioning_treatments) {</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Extract unique acute.treatment levels within the current life.stage and conditioning.treatment</span></span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>    unique_treatments <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(life.stage <span class="sc">==</span> stage, conditioning.treatment <span class="sc">==</span> conditioning) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(acute.treatment))</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if there are at least 2 unique treatments</span></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(unique_treatments) <span class="sc">&gt;=</span> <span class="dv">2</span>) {</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Generate all possible pairs of acute.treatment levels</span></span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>      treatment_pairs <span class="ot">&lt;-</span> <span class="fu">combn</span>(unique_treatments, <span class="dv">2</span>, <span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span> (pair <span class="cf">in</span> treatment_pairs) {</span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>        treatment1 <span class="ot">&lt;-</span> pair[<span class="dv">1</span>]</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>        treatment2 <span class="ot">&lt;-</span> pair[<span class="dv">2</span>]</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Perform t-test for each Target comparing the two acute.treatment levels within the current life.stage and conditioning.treatment</span></span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>        t_test_results <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a>          <span class="fu">filter</span>(life.stage <span class="sc">==</span> stage, conditioning.treatment <span class="sc">==</span> conditioning, acute.treatment <span class="sc">%in%</span> <span class="fu">c</span>(treatment1, treatment2)) <span class="sc">%&gt;%</span></span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a>          <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a>          <span class="fu">summarise</span>(</span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a>            <span class="at">t_test_result =</span> <span class="fu">list</span>(<span class="fu">t.test</span>(delta_Cq <span class="sc">~</span> acute.treatment))</span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a>          ) <span class="sc">%&gt;%</span></span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a>          <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a>          <span class="fu">mutate</span>(</span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a>            <span class="at">estimate_diff =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>estimate[<span class="dv">1</span>] <span class="sc">-</span> x<span class="sc">$</span>estimate[<span class="dv">2</span>]),</span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_value =</span> <span class="fu">sapply</span>(t_test_result, <span class="cf">function</span>(x) x<span class="sc">$</span>p.value),</span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a>            <span class="at">asterisk =</span> <span class="fu">ifelse</span>(p_value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="st">""</span>), <span class="co"># Adds asterisk column and asterisk for p-value.</span></span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a>            <span class="at">comparison =</span> <span class="fu">paste</span>(stage, conditioning, treatment1, <span class="st">"vs"</span>, treatment2, <span class="at">sep =</span> <span class="st">"."</span>)</span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a>          ) <span class="sc">%&gt;%</span></span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a>          <span class="fu">select</span>(<span class="sc">!</span>t_test_result)</span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a>        acute_treatment_within_life.stages_conditioning_t_test_results_list[[<span class="fu">paste</span>(stage, conditioning, treatment1, treatment2, <span class="at">sep =</span> <span class="st">"."</span>)]] <span class="ot">&lt;-</span> t_test_results</span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results into a single data frame</span></span>
<span id="cb28-48"><a href="#cb28-48" aria-hidden="true" tabindex="-1"></a>acute_treatment_within_life.stages_conditioning_t_test_results_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(acute_treatment_within_life.stages_conditioning_t_test_results_list, <span class="at">.id =</span> <span class="st">"comparison_id"</span>)</span>
<span id="cb28-49"><a href="#cb28-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-50"><a href="#cb28-50" aria-hidden="true" tabindex="-1"></a><span class="co"># View the results</span></span>
<span id="cb28-51"><a href="#cb28-51" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(acute_treatment_within_life.stages_conditioning_t_test_results_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code># A tibble: 28 × 6
   comparison_id              Target   estimate_diff p_value asterisk comparison
   &lt;chr&gt;                      &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;     
 1 adult.treated.ambient.high ATPsynt…        0.0264   0.893 ""       adult.tre…
 2 adult.treated.ambient.high DNMT1           0.379    0.372 ""       adult.tre…
 3 adult.treated.ambient.high HSP70           0.281    0.739 ""       adult.tre…
 4 adult.treated.ambient.high HSP90           0.0292   0.925 ""       adult.tre…
 5 adult.treated.ambient.high VIPERIN         0.0693   0.872 ""       adult.tre…
 6 adult.treated.ambient.high cGAS            0.0595   0.828 ""       adult.tre…
 7 adult.treated.ambient.high citrate…       -0.176    0.336 ""       adult.tre…
 8 adult.control.high.ambient ATPsynt…        0.0946   0.698 ""       adult.con…
 9 adult.control.high.ambient DNMT1           0.249    0.542 ""       adult.con…
10 adult.control.high.ambient HSP70           0.271    0.815 ""       adult.con…
# ℹ 18 more rows</code></pre>
</section>
</section>
<section id="plotting" class="level2">
<h2 class="anchored" data-anchor-id="plotting">8.2 Plotting</h2>
<section id="delta-cq-boxplots" class="level3">
<h3 class="anchored" data-anchor-id="delta-cq-boxplots">8.2.1 Delta Cq boxplots</h3>
<section id="lifestage-comparisons" class="level4">
<h4 class="anchored" data-anchor-id="lifestage-comparisons">8.2.1.1 Lifestage comparisons</h4>
<div class="sourceCode" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create box plots for each comparison</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>unique_comparisons <span class="ot">&lt;-</span> <span class="fu">unique</span>(life_stage_t_test_results_df<span class="sc">$</span>comparison)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (comparison <span class="cf">in</span> unique_comparisons) {</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_boxplot_delta_Cq</span>(delta_Cq_df, comparison, life_stage_t_test_results_df)</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-lifestage-1.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-lifestage-2.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-lifestage-3.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-lifestage-4.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-lifestage-5.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-lifestage-6.png" class="img-fluid"></p>
</section>
</section>
<section id="conditioning-comparisons-1" class="level3">
<h3 class="anchored" data-anchor-id="conditioning-comparisons-1">8.2.2 Conditioning comparisons</h3>
<div class="sourceCode" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create box plots for each comparison</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>unique_comparisons <span class="ot">&lt;-</span> <span class="fu">unique</span>(conditioning_treatment_t_test_results_df<span class="sc">$</span>comparison)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (comparison <span class="cf">in</span> unique_comparisons) {</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_boxplot_conditioning</span>(delta_Cq_df, comparison, conditioning_treatment_t_test_results_df)</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-conditioning-1.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-conditioning-2.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-conditioning-3.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-conditioning-4.png" class="img-fluid"></p>
</section>
<section id="acute-treatment-comparisons" class="level3">
<h3 class="anchored" data-anchor-id="acute-treatment-comparisons">8.2.3 Acute treatment comparisons</h3>
<div class="sourceCode" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create box plots for each comparison</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>unique_comparisons <span class="ot">&lt;-</span> <span class="fu">unique</span>(acute_treatment_t_test_results_df<span class="sc">$</span>comparison)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (comparison <span class="cf">in</span> unique_comparisons) {</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_boxplot_acute</span>(delta_Cq_df, comparison, acute_treatment_t_test_results_df)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-acute-treatment-1.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/delta-Cq-boxplots-acute-treatment-2.png" class="img-fluid"></p>
</section>
<section id="acute-within-life-stage-conditioning" class="level3">
<h3 class="anchored" data-anchor-id="acute-within-life-stage-conditioning">8.2.4 Acute within life stage conditioning</h3>
<div class="sourceCode" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each comparison in the t-test results and create box plots</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (comparison <span class="cf">in</span> <span class="fu">unique</span>(acute_treatment_within_life.stages_conditioning_t_test_results_df<span class="sc">$</span>comparison)) {</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_boxplot_acute_conditioning</span>(delta_Cq_df, comparison, acute_treatment_within_life.stages_conditioning_t_test_results_df)</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/boxplots-acute-treatments-within-life-stages-conditioning-1.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/boxplots-acute-treatments-within-life-stages-conditioning-2.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/boxplots-acute-treatments-within-life-stages-conditioning-3.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/boxplots-acute-treatments-within-life-stages-conditioning-4.png" class="img-fluid"></p>
</section>
</section>
</section>
<section id="delta-delta-cq" class="level1">
<h1>9 Delta delta Cq</h1>
<section id="calculations" class="level2">
<h2 class="anchored" data-anchor-id="calculations">9.1 Calculations</h2>
<section id="conditioning" class="level3">
<h3 class="anchored" data-anchor-id="conditioning">9.1.1 Conditioning</h3>
<div class="sourceCode" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta_delta_Cq</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>delta_delta_conditioning_fold_change <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, Target) <span class="sc">%&gt;%</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">treated_delta_Cq =</span> <span class="fu">mean</span>(delta_Cq[conditioning.treatment <span class="sc">==</span> <span class="st">"treated"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">control_delta_Cq =</span> <span class="fu">mean</span>(delta_Cq[conditioning.treatment <span class="sc">==</span> <span class="st">"control"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">delta_delta_Cq =</span> treated_delta_Cq <span class="sc">-</span> control_delta_Cq) <span class="sc">%&gt;%</span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(life.stage, Target, delta_delta_Cq)</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_conditioning_fold_change)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>gropd_df [28 × 3] (S3: grouped_df/tbl_df/tbl/data.frame)
 $ life.stage    : chr [1:28] "adult" "adult" "adult" "adult" ...
 $ Target        : chr [1:28] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
 $ delta_delta_Cq: num [1:28] -0.0779 -0.3116 0.941 0.7639 0.0852 ...
 - attr(*, "groups")= tibble [4 × 2] (S3: tbl_df/tbl/data.frame)
  ..$ life.stage: chr [1:4] "adult" "juvenile" "seed" "spat"
  ..$ .rows     : list&lt;int&gt; [1:4] 
  .. ..$ : int [1:7] 1 2 3 4 5 6 7
  .. ..$ : int [1:7] 8 9 10 11 12 13 14
  .. ..$ : int [1:7] 15 16 17 18 19 20 21
  .. ..$ : int [1:7] 22 23 24 25 26 27 28
  .. ..@ ptype: int(0) 
  ..- attr(*, ".drop")= logi TRUE</code></pre>
</section>
<section id="acute-treatment" class="level3">
<h3 class="anchored" data-anchor-id="acute-treatment">9.1.2 Acute treatment</h3>
<div class="sourceCode" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta_delta_Cq for acute treatment</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>delta_delta_Cq_acute_df <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, Target, acute.treatment) <span class="sc">%&gt;%</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">treated_delta_Cq =</span> <span class="fu">mean</span>(delta_Cq[conditioning.treatment <span class="sc">==</span> <span class="st">"treated"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">control_delta_Cq =</span> <span class="fu">mean</span>(delta_Cq[conditioning.treatment <span class="sc">==</span> <span class="st">"control"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">delta_delta_Cq =</span> treated_delta_Cq <span class="sc">-</span> control_delta_Cq) <span class="sc">%&gt;%</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(life.stage, Target, acute.treatment, delta_delta_Cq)</span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_Cq_acute_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>gropd_df [42 × 4] (S3: grouped_df/tbl_df/tbl/data.frame)
 $ life.stage     : chr [1:42] "adult" "adult" "adult" "adult" ...
 $ Target         : chr [1:42] "ATPsynthase" "ATPsynthase" "DNMT1" "DNMT1" ...
 $ acute.treatment: chr [1:42] "ambient" "high" "ambient" "high" ...
 $ delta_delta_Cq : num [1:42] -0.112 -0.0438 -0.2467 -0.3765 0.9455 ...
 - attr(*, "groups")= tibble [28 × 3] (S3: tbl_df/tbl/data.frame)
  ..$ life.stage: chr [1:28] "adult" "adult" "adult" "adult" ...
  ..$ Target    : chr [1:28] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
  ..$ .rows     : list&lt;int&gt; [1:28] 
  .. ..$ : int [1:2] 1 2
  .. ..$ : int [1:2] 3 4
  .. ..$ : int [1:2] 5 6
  .. ..$ : int [1:2] 7 8
  .. ..$ : int [1:2] 9 10
  .. ..$ : int [1:2] 11 12
  .. ..$ : int [1:2] 13 14
  .. ..$ : int [1:2] 15 16
  .. ..$ : int [1:2] 17 18
  .. ..$ : int [1:2] 19 20
  .. ..$ : int [1:2] 21 22
  .. ..$ : int [1:2] 23 24
  .. ..$ : int [1:2] 25 26
  .. ..$ : int [1:2] 27 28
  .. ..$ : int 29
  .. ..$ : int 30
  .. ..$ : int 31
  .. ..$ : int 32
  .. ..$ : int 33
  .. ..$ : int 34
  .. ..$ : int 35
  .. ..$ : int 36
  .. ..$ : int 37
  .. ..$ : int 38
  .. ..$ : int 39
  .. ..$ : int 40
  .. ..$ : int 41
  .. ..$ : int 42
  .. ..@ ptype: int(0) 
  ..- attr(*, ".drop")= logi TRUE</code></pre>
</section>
<section id="life-stage" class="level3">
<h3 class="anchored" data-anchor-id="life-stage">9.1.3 Life stage</h3>
<div class="sourceCode" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta_delta_Cq for life stage comparisons</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>delta_delta_Cq_life_stage_df <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target, life.stage) <span class="sc">%&gt;%</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean_delta_Cq =</span> <span class="fu">mean</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> life.stage, <span class="at">values_from =</span> mean_delta_Cq) <span class="sc">%&gt;%</span></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">delta_delta_Cq_adult_vs_seed =</span> adult <span class="sc">-</span> seed,</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">delta_delta_Cq_spat_vs_seed =</span> spat <span class="sc">-</span> seed,</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">delta_delta_Cq_adult_vs_spat =</span> adult <span class="sc">-</span> spat</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"delta_delta_Cq_"</span>), <span class="at">names_to =</span> <span class="st">"comparison"</span>, <span class="at">values_to =</span> <span class="st">"delta_delta_Cq"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(delta_delta_Cq))</span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the structure of the resulting data frame</span></span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_Cq_life_stage_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [21 × 7] (S3: tbl_df/tbl/data.frame)
 $ Target        : chr [1:21] "ATPsynthase" "ATPsynthase" "ATPsynthase" "DNMT1" ...
 $ adult         : num [1:21] 0.48 0.48 0.48 6.17 6.17 ...
 $ juvenile      : num [1:21] 0.378 0.378 0.378 5.808 5.808 ...
 $ seed          : num [1:21] 0.229 0.229 0.229 5.725 5.725 ...
 $ spat          : num [1:21] 0.519 0.519 0.519 6.277 6.277 ...
 $ comparison    : chr [1:21] "delta_delta_Cq_adult_vs_seed" "delta_delta_Cq_spat_vs_seed" "delta_delta_Cq_adult_vs_spat" "delta_delta_Cq_adult_vs_seed" ...
 $ delta_delta_Cq: num [1:21] 0.2508 0.2894 -0.0386 0.4424 0.5511 ...</code></pre>
</section>
<section id="calculate-delta-delta-acute-treatments-within-lifestage-and-conditioning" class="level3">
<h3 class="anchored" data-anchor-id="calculate-delta-delta-acute-treatments-within-lifestage-and-conditioning">9.1.4 Calculate delta delta acute treatments within lifestage and conditioning</h3>
<div class="sourceCode" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta_delta_Cq for acute treatment comparisons within each life stage and conditioning treatment</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>delta_delta_Cq_acute_within_life_stage_conditioning_df <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(life.stage, conditioning.treatment, Target, acute.treatment) <span class="sc">%&gt;%</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean_delta_Cq =</span> <span class="fu">mean</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> acute.treatment, <span class="at">values_from =</span> mean_delta_Cq) <span class="sc">%&gt;%</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">delta_delta_Cq_high_vs_ambient =</span> high <span class="sc">-</span> ambient) <span class="sc">%&gt;%</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"delta_delta_Cq_"</span>), <span class="at">names_to =</span> <span class="st">"comparison"</span>, <span class="at">values_to =</span> <span class="st">"delta_delta_Cq"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(delta_delta_Cq))</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the structure of the resulting data frame</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_Cq_acute_within_life_stage_conditioning_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [28 × 7] (S3: tbl_df/tbl/data.frame)
 $ life.stage            : chr [1:28] "adult" "adult" "adult" "adult" ...
 $ conditioning.treatment: chr [1:28] "control" "control" "control" "control" ...
 $ Target                : chr [1:28] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
 $ ambient               : num [1:28] 0.566 6.448 3.944 1.259 3.758 ...
 $ high                  : num [1:28] 0.472 6.199 3.673 0.29 3.181 ...
 $ comparison            : chr [1:28] "delta_delta_Cq_high_vs_ambient" "delta_delta_Cq_high_vs_ambient" "delta_delta_Cq_high_vs_ambient" "delta_delta_Cq_high_vs_ambient" ...
 $ delta_delta_Cq        : num [1:28] -0.0946 -0.2492 -0.2715 -0.969 -0.5763 ...</code></pre>
</section>
<section id="calculate-the-fold-change-life-stage-comparison" class="level3">
<h3 class="anchored" data-anchor-id="calculate-the-fold-change-life-stage-comparison">9.1.5 Calculate the fold change life stage comparison</h3>
<div class="sourceCode" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate fold change and output to a new data frame</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>fold_change_life_stage_df <span class="ot">&lt;-</span> delta_delta_Cq_life_stage_df <span class="sc">%&gt;%</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq))</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the structure of the resulting data frame</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(fold_change_life_stage_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [21 × 8] (S3: tbl_df/tbl/data.frame)
 $ Target        : chr [1:21] "ATPsynthase" "ATPsynthase" "ATPsynthase" "DNMT1" ...
 $ adult         : num [1:21] 0.48 0.48 0.48 6.17 6.17 ...
 $ juvenile      : num [1:21] 0.378 0.378 0.378 5.808 5.808 ...
 $ seed          : num [1:21] 0.229 0.229 0.229 5.725 5.725 ...
 $ spat          : num [1:21] 0.519 0.519 0.519 6.277 6.277 ...
 $ comparison    : chr [1:21] "delta_delta_Cq_adult_vs_seed" "delta_delta_Cq_spat_vs_seed" "delta_delta_Cq_adult_vs_spat" "delta_delta_Cq_adult_vs_seed" ...
 $ delta_delta_Cq: num [1:21] 0.2508 0.2894 -0.0386 0.4424 0.5511 ...
 $ fold_change   : num [1:21] 0.84 0.818 1.027 0.736 0.683 ...</code></pre>
</section>
<section id="calculate-the-fold-change-conditioning-comparison" class="level3">
<h3 class="anchored" data-anchor-id="calculate-the-fold-change-conditioning-comparison">9.1.6 Calculate the fold change conditioning comparison</h3>
<div class="sourceCode" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>delta_delta_conditioning_fold_change <span class="ot">&lt;-</span> delta_delta_conditioning_fold_change <span class="sc">%&gt;%</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq)) <span class="sc">%&gt;%</span> </span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Target, fold_change)</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_conditioning_fold_change)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>gropd_df [28 × 3] (S3: grouped_df/tbl_df/tbl/data.frame)
 $ life.stage : chr [1:28] "adult" "adult" "adult" "adult" ...
 $ Target     : chr [1:28] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
 $ fold_change: num [1:28] 1.055 1.241 0.521 0.589 0.943 ...
 - attr(*, "groups")= tibble [4 × 2] (S3: tbl_df/tbl/data.frame)
  ..$ life.stage: chr [1:4] "adult" "juvenile" "seed" "spat"
  ..$ .rows     : list&lt;int&gt; [1:4] 
  .. ..$ : int [1:7] 1 2 3 4 5 6 7
  .. ..$ : int [1:7] 8 9 10 11 12 13 14
  .. ..$ : int [1:7] 15 16 17 18 19 20 21
  .. ..$ : int [1:7] 22 23 24 25 26 27 28
  .. ..@ ptype: int(0) 
  ..- attr(*, ".drop")= logi TRUE</code></pre>
</section>
<section id="calculate-the-fold-change-acute-comparison" class="level3">
<h3 class="anchored" data-anchor-id="calculate-the-fold-change-acute-comparison">9.1.7 Calculate the fold change acute comparison</h3>
<div class="sourceCode" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate fold change for acute treatment</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>delta_delta_acute_fold_change <span class="ot">&lt;-</span> delta_delta_Cq_acute_df <span class="sc">%&gt;%</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq)) <span class="sc">%&gt;%</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(life.stage, Target, acute.treatment, fold_change)</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the structure of the resulting data frame</span></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_delta_acute_fold_change)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>gropd_df [42 × 4] (S3: grouped_df/tbl_df/tbl/data.frame)
 $ life.stage     : chr [1:42] "adult" "adult" "adult" "adult" ...
 $ Target         : chr [1:42] "ATPsynthase" "ATPsynthase" "DNMT1" "DNMT1" ...
 $ acute.treatment: chr [1:42] "ambient" "high" "ambient" "high" ...
 $ fold_change    : num [1:42] 1.081 1.031 1.186 1.298 0.519 ...
 - attr(*, "groups")= tibble [28 × 3] (S3: tbl_df/tbl/data.frame)
  ..$ life.stage: chr [1:28] "adult" "adult" "adult" "adult" ...
  ..$ Target    : chr [1:28] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
  ..$ .rows     : list&lt;int&gt; [1:28] 
  .. ..$ : int [1:2] 1 2
  .. ..$ : int [1:2] 3 4
  .. ..$ : int [1:2] 5 6
  .. ..$ : int [1:2] 7 8
  .. ..$ : int [1:2] 9 10
  .. ..$ : int [1:2] 11 12
  .. ..$ : int [1:2] 13 14
  .. ..$ : int [1:2] 15 16
  .. ..$ : int [1:2] 17 18
  .. ..$ : int [1:2] 19 20
  .. ..$ : int [1:2] 21 22
  .. ..$ : int [1:2] 23 24
  .. ..$ : int [1:2] 25 26
  .. ..$ : int [1:2] 27 28
  .. ..$ : int 29
  .. ..$ : int 30
  .. ..$ : int 31
  .. ..$ : int 32
  .. ..$ : int 33
  .. ..$ : int 34
  .. ..$ : int 35
  .. ..$ : int 36
  .. ..$ : int 37
  .. ..$ : int 38
  .. ..$ : int 39
  .. ..$ : int 40
  .. ..$ : int 41
  .. ..$ : int 42
  .. ..@ ptype: int(0) 
  ..- attr(*, ".drop")= logi TRUE</code></pre>
</section>
<section id="calculate-fold-change-acute-treatments-within-lifestage-and-conditioning" class="level3">
<h3 class="anchored" data-anchor-id="calculate-fold-change-acute-treatments-within-lifestage-and-conditioning">9.1.8 Calculate fold change acute treatments within lifestage and conditioning</h3>
<div class="sourceCode" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate fold change for acute treatment comparisons within each life stage and conditioning treatment</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>fold_change_acute_within_life_stage_conditioning_df <span class="ot">&lt;-</span> delta_delta_Cq_acute_within_life_stage_conditioning_df <span class="sc">%&gt;%</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq))</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the structure of the resulting data frame</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(fold_change_acute_within_life_stage_conditioning_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>tibble [28 × 8] (S3: tbl_df/tbl/data.frame)
 $ life.stage            : chr [1:28] "adult" "adult" "adult" "adult" ...
 $ conditioning.treatment: chr [1:28] "control" "control" "control" "control" ...
 $ Target                : chr [1:28] "ATPsynthase" "DNMT1" "HSP70" "HSP90" ...
 $ ambient               : num [1:28] 0.566 6.448 3.944 1.259 3.758 ...
 $ high                  : num [1:28] 0.472 6.199 3.673 0.29 3.181 ...
 $ comparison            : chr [1:28] "delta_delta_Cq_high_vs_ambient" "delta_delta_Cq_high_vs_ambient" "delta_delta_Cq_high_vs_ambient" "delta_delta_Cq_high_vs_ambient" ...
 $ delta_delta_Cq        : num [1:28] -0.0946 -0.2492 -0.2715 -0.969 -0.5763 ...
 $ fold_change           : num [1:28] 1.07 1.19 1.21 1.96 1.49 ...</code></pre>
</section>
</section>
<section id="plotting-fold-changes" class="level2">
<h2 class="anchored" data-anchor-id="plotting-fold-changes">9.2 Plotting fold changes</h2>
<section id="acute-comparisons-within-lifestage-and-conditioning" class="level3">
<h3 class="anchored" data-anchor-id="acute-comparisons-within-lifestage-and-conditioning">9.2.1 Acute comparisons within lifestage and conditioning</h3>
<div class="sourceCode" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate bar plots for each group of comparison within each life stage and conditioning treatment</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> fold_change_acute_within_life_stage_conditioning_df <span class="sc">%&gt;%</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">split</span>(<span class="fu">list</span>(.<span class="sc">$</span>life.stage, .<span class="sc">$</span>conditioning.treatment, .<span class="sc">$</span>comparison)) <span class="sc">%&gt;%</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(<span class="cf">function</span>(df) {</span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>    life_stage <span class="ot">&lt;-</span> <span class="fu">unique</span>(df<span class="sc">$</span>life.stage)</span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>    conditioning_treatment <span class="ot">&lt;-</span> <span class="fu">unique</span>(df<span class="sc">$</span>conditioning.treatment)</span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>    comparison_title <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta_delta_Cq_"</span>, <span class="st">""</span>, <span class="fu">unique</span>(df<span class="sc">$</span>comparison))</span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>    comparison_title <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_vs_"</span>, <span class="st">" vs. "</span>, comparison_title)</span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> fold_change)) <span class="sc">+</span></span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Gene Expression -"</span>, life_stage, <span class="st">"-"</span>, conditioning_treatment, <span class="st">"-"</span>, comparison_title), </span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a>           <span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the plots</span></span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (plot <span class="cf">in</span> plot_list) {</span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-acute-within-lifestage-conditioning-1.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-acute-within-lifestage-conditioning-2.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-acute-within-lifestage-conditioning-3.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-acute-within-lifestage-conditioning-4.png" class="img-fluid"></p>
</section>
<section id="life-stage-comparisons" class="level3">
<h3 class="anchored" data-anchor-id="life-stage-comparisons">9.2.2 Life stage comparisons</h3>
<div class="sourceCode" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate bar plots for each group of comparison</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> fold_change_life_stage_df <span class="sc">%&gt;%</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">split</span>(.<span class="sc">$</span>comparison) <span class="sc">%&gt;%</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(<span class="cf">function</span>(df) {</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>    comparison_title <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"delta_delta_Cq_"</span>, <span class="st">""</span>, <span class="fu">unique</span>(df<span class="sc">$</span>comparison))</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>    comparison_title <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_vs_"</span>, <span class="st">" vs. "</span>, comparison_title)</span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> fold_change)) <span class="sc">+</span></span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Gene Expression -"</span>, comparison_title), <span class="at">x =</span> <span class="st">"Target"</span>, <span class="at">y =</span> <span class="st">"Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the plots</span></span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (plot <span class="cf">in</span> plot_list) {</span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-lifestages-1.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-lifestages-2.png" class="img-fluid"></p>
<p><img src="01.01-qPCR_files/figure-gfm/bar-plots-fold-change-lifestages-3.png" class="img-fluid"></p>
</section>
<section id="conditioning-comparisons-2" class="level3">
<h3 class="anchored" data-anchor-id="conditioning-comparisons-2">9.2.3 Conditioning comparisons</h3>
<p><img src="01.01-qPCR_files/figure-gfm/barplot-fold-change-conditioning-1.png" class="img-fluid"></p>
</section>
<section id="line-plot-conditioning-comparisons-across-lifestages" class="level3">
<h3 class="anchored" data-anchor-id="line-plot-conditioning-comparisons-across-lifestages">9.2.4 Line plot conditioning comparisons across lifestages</h3>
<p><img src="01.01-qPCR_files/figure-gfm/lineplot-fold-change-conditioning-1.png" class="img-fluid"></p>
</section>
<section id="acute-treatment-comparison" class="level3">
<h3 class="anchored" data-anchor-id="acute-treatment-comparison">9.2.5 Acute treatment comparison</h3>
<p><img src="01.01-qPCR_files/figure-gfm/barplot-fold-change-acute-1.png" class="img-fluid"></p>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/robertslab\.github\.io\/sams-notebook");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>