<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.5">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sam White">
<meta name="dcterms.date" content="2025-01-17">

<title>Primer Design - C.gigas Alternative Oxidase AOX Variants Using Primer3 – Sam’s Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark-dec4957386d03255e1655963b31d62d3.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-d24cc6793ea89e8aa59926ce2f45f698.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 2,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 100,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-7483T61T11"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-7483T61T11', { 'anonymize_ip': true});
</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Sam’s Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" target="_blank">
 <span class="dropdown-text">Rstudio Server - Raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu/Atumefaciens/" target="_blank">
 <span class="dropdown-text">My Gannet Directory</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" target="_blank">
 <span class="dropdown-text">Steven’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shedurkin.github.io/Roberts-LabNotebook/" target="_blank">
 <span class="dropdown-text">Kathleen’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://meganewing.github.io/mewing-notebook/" target="_blank">
 <span class="dropdown-text">Megan’s Notebook</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/sams-notebook" target="_blank">
 <span class="dropdown-text">Notebook repo</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" target="_blank">
 <span class="dropdown-text">Lab Organization (GitHub)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" target="_blank">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" target="_blank"> <i class="bi bi-calendar-week" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-right">
      <h1 class="title">Primer Design - C.gigas Alternative Oxidase AOX Variants Using Primer3</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2025</div>
                <div class="quarto-category">Crassotrea gigas</div>
                <div class="quarto-category">Pacific oyster</div>
                <div class="quarto-category">Primer3</div>
                <div class="quarto-category">EMBOSS</div>
                <div class="quarto-category">PrimerSearch</div>
                <div class="quarto-category">Magallana gigas</div>
                <div class="quarto-category">primers</div>
                <div class="quarto-category">AOX</div>
                <div class="quarto-category">alternative oxidase</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-right">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Sam White </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">January 17, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul>
  <li><a href="#objectives" id="toc-objectives" class="nav-link" data-scroll-target="#objectives">Objectives</a></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods">Methods</a></li>
  </ul></li>
  <li><a href="#background" id="toc-background" class="nav-link" data-scroll-target="#background">1 Background</a>
  <ul>
  <li><a href="#input-files" id="toc-input-files" class="nav-link" data-scroll-target="#input-files">1.1 Input files</a></li>
  <li><a href="#output-files" id="toc-output-files" class="nav-link" data-scroll-target="#output-files">1.2 Output files</a></li>
  <li><a href="#required-software" id="toc-required-software" class="nav-link" data-scroll-target="#required-software">1.3 Required software</a></li>
  </ul></li>
  <li><a href="#create-bash-variables-file" id="toc-create-bash-variables-file" class="nav-link" data-scroll-target="#create-bash-variables-file">2 CREATE BASH VARIABLES FILE</a></li>
  <li><a href="#download-ncbi-genome-files" id="toc-download-ncbi-genome-files" class="nav-link" data-scroll-target="#download-ncbi-genome-files">3 DOWNLOAD NCBI GENOME FILES</a>
  <ul>
  <li><a href="#download-the-files" id="toc-download-the-files" class="nav-link" data-scroll-target="#download-the-files">3.1 Download the files</a></li>
  <li><a href="#check-md5-checkums" id="toc-check-md5-checkums" class="nav-link" data-scroll-target="#check-md5-checkums">3.2 Check MD5 Checkums</a></li>
  <li><a href="#decompress-ncbi-files" id="toc-decompress-ncbi-files" class="nav-link" data-scroll-target="#decompress-ncbi-files">3.3 Decompress NCBI files</a></li>
  </ul></li>
  <li><a href="#retrieve-gene-sequences" id="toc-retrieve-gene-sequences" class="nav-link" data-scroll-target="#retrieve-gene-sequences">4 RETRIEVE GENE SEQUENCES</a></li>
  <li><a href="#primer-design-using-primer3" id="toc-primer-design-using-primer3" class="nav-link" data-scroll-target="#primer-design-using-primer3">5 PRIMER DESIGN USING PRIMER3</a>
  <ul>
  <li><a href="#design-primers" id="toc-design-primers" class="nav-link" data-scroll-target="#design-primers">5.1 Design primers</a>
  <ul>
  <li><a href="#ncbi-blastn-alignments" id="toc-ncbi-blastn-alignments" class="nav-link" data-scroll-target="#ncbi-blastn-alignments">5.1.0.1 NCBI BLASTn alignments</a>
  <ul class="collapse">
  <li><a href="#nm_001305360.1" id="toc-nm_001305360.1" class="nav-link" data-scroll-target="#nm_001305360.1">5.1.0.1.1 NM_001305360.1</a></li>
  <li><a href="#xm_034462847.2" id="toc-xm_034462847.2" class="nav-link" data-scroll-target="#xm_034462847.2">5.1.0.1.2 XM_034462847.2</a></li>
  <li><a href="#xm_066083499.1" id="toc-xm_066083499.1" class="nav-link" data-scroll-target="#xm_066083499.1">5.1.0.1.3 XM_066083499.1</a></li>
  </ul></li>
  <li><a href="#nm_001305360.1-1" id="toc-nm_001305360.1-1" class="nav-link" data-scroll-target="#nm_001305360.1-1">5.1.1 NM_001305360.1</a>
  <ul class="collapse">
  <li><a href="#review-primers" id="toc-review-primers" class="nav-link" data-scroll-target="#review-primers">5.1.1.1 Review primers</a>
  <ul class="collapse">
  <li><a href="#easy-to-read" id="toc-easy-to-read" class="nav-link" data-scroll-target="#easy-to-read">5.1.1.1.1 Easy-to-read</a></li>
  <li><a href="#print-the-full-output-file" id="toc-print-the-full-output-file" class="nav-link" data-scroll-target="#print-the-full-output-file">5.1.1.1.2 Print the full output file</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#xm_034462847.2-1" id="toc-xm_034462847.2-1" class="nav-link" data-scroll-target="#xm_034462847.2-1">5.1.2 XM_034462847.2</a>
  <ul class="collapse">
  <li><a href="#review-primers-1" id="toc-review-primers-1" class="nav-link" data-scroll-target="#review-primers-1">5.1.2.1 Review primers</a>
  <ul class="collapse">
  <li><a href="#easy-to-read-1" id="toc-easy-to-read-1" class="nav-link" data-scroll-target="#easy-to-read-1">5.1.2.1.1 Easy-to-read</a></li>
  <li><a href="#print-the-full-output-file-1" id="toc-print-the-full-output-file-1" class="nav-link" data-scroll-target="#print-the-full-output-file-1">5.1.2.1.2 Print the full output file</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#xm_066083499.1-1" id="toc-xm_066083499.1-1" class="nav-link" data-scroll-target="#xm_066083499.1-1">5.1.3 XM_066083499.1</a>
  <ul class="collapse">
  <li><a href="#review-primers-2" id="toc-review-primers-2" class="nav-link" data-scroll-target="#review-primers-2">5.1.3.1 Review primers</a>
  <ul class="collapse">
  <li><a href="#easy-to-read-2" id="toc-easy-to-read-2" class="nav-link" data-scroll-target="#easy-to-read-2">5.1.3.1.1 Easy-to-read</a></li>
  <li><a href="#print-the-full-output-file-2" id="toc-print-the-full-output-file-2" class="nav-link" data-scroll-target="#print-the-full-output-file-2">5.1.3.1.2 Print the full output file</a></li>
  </ul></li>
  </ul></li>
  </ul></li>
  </ul></li>
  <li><a href="#split-genome-fasta" id="toc-split-genome-fasta" class="nav-link" data-scroll-target="#split-genome-fasta">6 SPLIT GENOME FASTA</a>
  <ul>
  <li><a href="#split-mulit-fasta-file-in-to-individual-fasta-files-with-pyfaidx" id="toc-split-mulit-fasta-file-in-to-individual-fasta-files-with-pyfaidx" class="nav-link" data-scroll-target="#split-mulit-fasta-file-in-to-individual-fasta-files-with-pyfaidx">6.1 Split mulit-FastA file in to individual FastA files with PyFaidx</a></li>
  </ul></li>
  <li><a href="#primer-search-with-emboss-primersearch" id="toc-primer-search-with-emboss-primersearch" class="nav-link" data-scroll-target="#primer-search-with-emboss-primersearch">7 PRIMER SEARCH WITH EMBOSS PRIMERSEARCH</a>
  <ul>
  <li><a href="#create-emboss-primersearch-primers-file" id="toc-create-emboss-primersearch-primers-file" class="nav-link" data-scroll-target="#create-emboss-primersearch-primers-file">7.1 Create EMBOSS PrimerSearch Primers File</a></li>
  <li><a href="#run-emboss-primersearch" id="toc-run-emboss-primersearch" class="nav-link" data-scroll-target="#run-emboss-primersearch">7.2 Run EMBOSS PrimerSearch</a></li>
  <li><a href="#check-primer-matches" id="toc-check-primer-matches" class="nav-link" data-scroll-target="#check-primer-matches">7.3 Check primer matches</a></li>
  </ul></li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary">8 SUMMARY</a></li>
  <li><a href="#citations" id="toc-citations" class="nav-link" data-scroll-target="#citations">9 CITATIONS</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-right" id="quarto-document-content">





<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>This document outlines the process of designing primers for the alternative oxidase (AOX) gene variants in <em>Crassostrea gigas</em> (Pacific oyster) using Primer3. The goal is to develop specific primers for qPCR to study the expression of AOX isoforms under different experimental conditions.</p>
<p>The AOX gene is involved in the mitochondrial electron transport chain and is known to be responsive to oxidative stress. By designing specific primers for different AOX isoforms, we can investigate their expression patterns and potential roles in oyster physiology and stress responses.</p>
<p>The following sections detail the steps taken to identify AOX isoforms, retrieve their sequences, design primers using Primer3, and validate the primers using EMBOSS PrimerSearch.</p>
<section id="objectives" class="level2">
<h2 class="anchored" data-anchor-id="objectives">Objectives</h2>
<ol type="1">
<li>Identify AOX isoforms in <em>Crassostrea gigas</em>.</li>
<li>Retrieve nucleotide sequences for the identified isoforms.</li>
<li>Design specific primers for each isoform using Primer3.</li>
<li>Validate primer specificity using EMBOSS PrimerSearch.</li>
</ol>
</section>
<section id="methods" class="level2">
<h2 class="anchored" data-anchor-id="methods">Methods</h2>
<p>The methods section includes the following steps:</p>
<ol type="1">
<li><strong>Identification of AOX Isoforms</strong>: Using BLASTx to identify AOX isoforms in the <em>Crassostrea gigas</em> genome.</li>
<li><strong>Sequence Retrieval</strong>: Fetching nucleotide sequences for the identified isoforms using NCBI’s e-utilities.</li>
<li><strong>Primer Design</strong>: Using Primer3 to design specific primers for each AOX isoform.</li>
<li><strong>Primer Validation</strong>: Using EMBOSS PrimerSearch to validate the specificity of the designed primers against the <em>Crassostrea gigas</em> genome.</li>
</ol>
<p>The detailed procedures and scripts used for each step are provided in the subsequent sections.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The contents below are from markdown knitted from <a href="https://github.com/RobertsLab/code/blob/75f0197d210d79570ff60e356c70cf8232502b5b/r_projects/sam/20250117-cgig-aox-primer-design/code/20250117-cgig-aox-primer-design.Rmd"><code>20250117-cgig-aox-primer-design.Rmd</code></a> (commit <code>75f0197</code>).</p>
</div>
</div>
</section>
</section>
<section id="background" class="level1">
<h1>1 Background</h1>
<p>This notebook describes using <a href="https://github.com/primer3-org/primer3">Primer3</a> (<a href="#ref-untergasser2012">Untergasser et al. 2012</a>; <a href="#ref-koressaar2007">Koressaar and Remm 2007</a>) to reproducibly design qPCR primers for <em>Crassostrea gigas</em> (<em>Magallana gigas</em>) <a href="https://en.wikipedia.org/wiki/Alternative_oxidase">alternative oxidase (AOX)</a> (Wikipedia) isoforms. Alternative oxidase is a mitochondrial gene/enzyme known to be responsive to oxidative stress, so this may be a useful biomarker that we haven’t previously considered.</p>
<p>Isoforms of AOX in <em>Crassostrea gigas</em> (<em>Magallana gigas</em>) were identified by running BLASTx using the <a href="https://www.ncbi.nlm.nih.gov/sites/entrez?cmd=Search&amp;db=nucleotide&amp;term=NM_113135.4&amp;dopt=GenBank"><em>Arabidopsis</em> AOX1a nucleotide sequence (NM_113135.4)</a> as a query against the <em>Crassostrea gigas</em> (<em>Magallana gigas</em>) genome using NCBI’s web-based BLAST interface.</p>
<p>BLASTx returned three matches:</p>
<ul>
<li><p><a href="https://www.ncbi.nlm.nih.gov/protein/NP_001292289.1?report=genbank&amp;log$=protalign&amp;blast_rank=1&amp;RID=SMV9RMT6013">NP_001292289.1</a>: alternative oxidase, mitochondrial-like <span class="math display">\[Magallana gigas\]</span></p>
<ul>
<li><p>Nucleotide: <a href="https://www.ncbi.nlm.nih.gov/nuccore/NM_001305360.1"><code>NM_001305360.1</code></a></p></li>
<li><p>Gene: <code>LOC105322704</code></p></li>
</ul></li>
<li><p><a href="https://www.ncbi.nlm.nih.gov/protein/XP_034318738.2?report=genbank&amp;log$=protalign&amp;blast_rank=2&amp;RID=SMV9RMT6013"><code>XP_034318738.2</code></a>: alternative oxidase, mitochondrial-like isoform X2 <span class="math display">\[Magallana
gigas\]</span></p>
<ul>
<li><p>Nucleotide: <a href="https://www.ncbi.nlm.nih.gov/nuccore/XM_034462847.2"><code>XM_034462847.2</code></a></p></li>
<li><p>Gene: <code>LOC105322704</code></p></li>
</ul></li>
<li><p><a href="https://www.ncbi.nlm.nih.gov/protein/XP_065939571.1?report=genbank&amp;log$=protalign&amp;blast_rank=3&amp;RID=SMV9RMT6013"><code>XP_065939571.1</code></a>: alternative oxidase, mitochondrial-like isoform X1 <span class="math display">\[Magallana
gigas\]</span></p>
<ul>
<li><p>Nucleotide: <a href="https://www.ncbi.nlm.nih.gov/nuccore/XM_066083499.1"><code>XM_066083499.1</code></a></p></li>
<li><p>Gene: <code>LOC105322704</code></p></li>
</ul></li>
</ul>
<section id="input-files" class="level2">
<h2 class="anchored" data-anchor-id="input-files">1.1 Input files</h2>
<ul>
<li><p>Target FastA sequences</p></li>
<li><p>Genome FastA</p></li>
</ul>
</section>
<section id="output-files" class="level2">
<h2 class="anchored" data-anchor-id="output-files">1.2 Output files</h2>
<ul>
<li><p><code>emboss-primers.txt</code>: Output from EMBOSS PrimerSearch containing primer pair matches within the genome.</p></li>
<li><p><code>*primer3-primers-default-format.txt</code>: The default output format containing Primer3 primer identification info. Machine-readable format.</p></li>
<li><p><code>*primer3-primers.txt</code>: Primer3 output file containing primer identification info. An easy-to-read format.</p></li>
</ul>
</section>
<section id="required-software" class="level2">
<h2 class="anchored" data-anchor-id="required-software">1.3 Required software</h2>
<p>Utilizes NCBI’s <a href="https://www.ncbi.nlm.nih.gov/books/NBK179288/#chapter6.Getting_Started">e-utilities</a> (<a href="#ref-kans2013entrez">Kans 2013</a>) for fetching FastA sequences.</p>
<p>This process also utilizes <a href="https://github.com/mdshw5/pyfaidx">pyfaidx</a> (<a href="#ref-shirley2015">Shirley et al.&nbsp;2015</a>).</p>
<p><a href="https://emboss.sourceforge.net/apps/cvs/emboss/apps/primersearch.html">EMBOSS PrimerSearch</a> (<a href="#ref-rice2000">Rice, Longden, and Bleasby 2000</a>) will be utilized to assess primer specificity across the genome.</p>
</section>
</section>
<section id="create-bash-variables-file" class="level1">
<h1>2 CREATE BASH VARIABLES FILE</h1>
<p>This allows usage of Bash variables across R Markdown chunks.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"#### Assign Variables ####"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# DATA DIRECTORIES"</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export data_dir="../data"'</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export output_top="../output"'</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export genome_fasta_splits_dir="${data_dir}/fasta_splits"'</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# SEQUENCES"</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export sequence_IDs=(XM_066083499.1 XM_034462847.2 NM_001305360.1)'</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# SEQUENCE REGIONS"</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export left_buffer="500"'</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export right_buffer="500"'</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# INPUT FILES"</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export genome_fasta="GCF_963853765.1_xbMagGiga1.1_genomic.fna"'</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export genome_gff="GCF_963853765.1_xbMagGiga1.1_genomic.gff"'</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export ncbi_gff_gz="GCF_963853765.1_xbMagGiga1.1_genomic.gff.gz"'</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export ncbi_fasta_gz="GCF_963853765.1_xbMagGiga1.1_genomic.fna.gz"'</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export ncbi_md5sums="md5checksums.txt"'</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export ncbi_url="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/963/853/765/GCF_963853765.1_xbMagGiga1.1/"'</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export NM_001305360_fasta="NM_001305360.1.fasta"'</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export XM_034462847_fasta="XM_034462847.2.fasta"'</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export XM_066083499_fasta="XM_066083499.1.fasta"'</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# OUTPUT FILES"</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# SET CPUS"</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export threads=40'</span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# PROGRAMS"</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'esearch="/home/sam/edirect/esearch"'</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'efetch="/home/sam/edirect/efetch"'</span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export pyfaidx=/home/shared/pyfaidx-0.8.1.1'</span></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export primer3_dir="/home/shared/primer3-2.6.1/src"'</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export primer3="${primer3_dir}/primer3_core"'</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export primer3_config="${primer3_dir}/primer3_config"'</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export primersearch="/home/shared/EMBOSS-6.6.0/emboss/primersearch"'</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="kw">}</span> <span class="op">&gt;</span> .bashvars</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> .bashvars</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>#### Assign Variables ####

# DATA DIRECTORIES
export data_dir="../data"
export output_top="../output"
export genome_fasta_splits_dir="${data_dir}/fasta_splits"

# SEQUENCES
export sequence_IDs=(XM_066083499.1 XM_034462847.2 NM_001305360.1)

# SEQUENCE REGIONS
export left_buffer="500"
export right_buffer="500"

# INPUT FILES
export genome_fasta="GCF_963853765.1_xbMagGiga1.1_genomic.fna"
export genome_gff="GCF_963853765.1_xbMagGiga1.1_genomic.gff"
export ncbi_gff_gz="GCF_963853765.1_xbMagGiga1.1_genomic.gff.gz"
export ncbi_fasta_gz="GCF_963853765.1_xbMagGiga1.1_genomic.fna.gz"
export ncbi_md5sums="md5checksums.txt"
export ncbi_url="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/963/853/765/GCF_963853765.1_xbMagGiga1.1/"
export NM_001305360_fasta="NM_001305360.1.fasta"
export XM_034462847_fasta="XM_034462847.2.fasta"
export XM_066083499_fasta="XM_066083499.1.fasta"
# OUTPUT FILES

# SET CPUS
export threads=40

# PROGRAMS
esearch="/home/sam/edirect/esearch"
efetch="/home/sam/edirect/efetch"
export pyfaidx=/home/shared/pyfaidx-0.8.1.1
export primer3_dir="/home/shared/primer3-2.6.1/src"
export primer3="${primer3_dir}/primer3_core"
export primer3_config="${primer3_dir}/primer3_config"
export primersearch="/home/shared/EMBOSS-6.6.0/emboss/primersearch"</code></pre>
</section>
<section id="download-ncbi-genome-files" class="level1">
<h1>3 DOWNLOAD NCBI GENOME FILES</h1>
<section id="download-the-files" class="level2">
<h2 class="anchored" data-anchor-id="download-the-files">3.1 Download the files</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="va">${ncbi_gff_gz}</span> <span class="va">${ncbi_fasta_gz}</span> <span class="va">${ncbi_md5sums}</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">wget</span> <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">--no-check-certificate</span> <span class="dt">\</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">--continue</span> <span class="dt">\</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">--quiet</span> <span class="dt">\</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">--directory-prefix</span><span class="op">=</span><span class="va">${data_dir}</span> <span class="dt">\</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="va">${ncbi_url}${file}</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-lh</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>total 1.1G
drwxr-xr-x 2 sam sam 4.0K Jan 21 16:59 fasta_splits
-rw-r--r-- 1 sam sam 545M Jun 28  2024 GCF_963853765.1_xbMagGiga1.1_genomic.fna
-rw-r--r-- 1 sam sam 1.1K Jan 21 10:28 GCF_963853765.1_xbMagGiga1.1_genomic.fna.fai
-rw-r--r-- 1 sam sam 167M Jun 28  2024 GCF_963853765.1_xbMagGiga1.1_genomic.fna.gz
-rw-r--r-- 1 sam sam 373M Jun 28  2024 GCF_963853765.1_xbMagGiga1.1_genomic.gff
-rw-r--r-- 1 sam sam  15M Jun 28  2024 GCF_963853765.1_xbMagGiga1.1_genomic.gff.gz
-rw-r--r-- 1 sam sam  20K Oct 27 16:10 md5checksums.txt
-rw-r--r-- 1 sam sam 1.4K Jan 18 23:18 NM_001305360.1.fasta
-rw-rw-r-- 1 sam sam 104K Jan 21 13:50 NM_001305360.1-NCBI-BLASTn-alignment.png
-rw-rw-r-- 1 sam sam 1.1K Jan 18 23:18 XM_034462847.2.fasta
-rw-rw-r-- 1 sam sam 119K Jan 21 13:50 XM_034462847.2-NCBI-BLASTn-alignment.png
-rw-rw-r-- 1 sam sam 1.5K Jan 18 23:18 XM_066083499.1.fasta
-rw-rw-r-- 1 sam sam 147K Jan 21 13:50 XM_066083499.1-NCBI-BLASTn-alignment.png</code></pre>
</section>
<section id="check-md5-checkums" class="level2">
<h2 class="anchored" data-anchor-id="check-md5-checkums">3.2 Check MD5 Checkums</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">"</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.gz</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span> <span class="st">"</span><span class="va">${file}</span><span class="st">"</span> <span class="va">${ncbi_md5sums}</span> <span class="kw">|</span> <span class="fu">md5sum</span> <span class="at">-c</span> <span class="at">-</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>./GCF_963853765.1_xbMagGiga1.1_genomic.fna.gz: OK
./GCF_963853765.1_xbMagGiga1.1_genomic.gff.gz: OK</code></pre>
</section>
<section id="decompress-ncbi-files" class="level2">
<h2 class="anchored" data-anchor-id="decompress-ncbi-files">3.3 Decompress NCBI files</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">"</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.gz</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gunzip</span> <span class="st">"</span><span class="va">${file}</span><span class="st">"</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-lh</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>gzip: GCF_963853765.1_xbMagGiga1.1_genomic.fna already exists;  not overwritten
gzip: GCF_963853765.1_xbMagGiga1.1_genomic.gff already exists;  not overwritten
total 1.1G
drwxr-xr-x 2 sam sam 4.0K Jan 21 16:59 fasta_splits
-rw-r--r-- 1 sam sam 545M Jun 28  2024 GCF_963853765.1_xbMagGiga1.1_genomic.fna
-rw-r--r-- 1 sam sam 1.1K Jan 21 10:28 GCF_963853765.1_xbMagGiga1.1_genomic.fna.fai
-rw-r--r-- 1 sam sam 167M Jun 28  2024 GCF_963853765.1_xbMagGiga1.1_genomic.fna.gz
-rw-r--r-- 1 sam sam 373M Jun 28  2024 GCF_963853765.1_xbMagGiga1.1_genomic.gff
-rw-r--r-- 1 sam sam  15M Jun 28  2024 GCF_963853765.1_xbMagGiga1.1_genomic.gff.gz
-rw-r--r-- 1 sam sam  20K Oct 27 16:10 md5checksums.txt
-rw-r--r-- 1 sam sam 1.4K Jan 18 23:18 NM_001305360.1.fasta
-rw-rw-r-- 1 sam sam 104K Jan 21 13:50 NM_001305360.1-NCBI-BLASTn-alignment.png
-rw-rw-r-- 1 sam sam 1.1K Jan 18 23:18 XM_034462847.2.fasta
-rw-rw-r-- 1 sam sam 119K Jan 21 13:50 XM_034462847.2-NCBI-BLASTn-alignment.png
-rw-rw-r-- 1 sam sam 1.5K Jan 18 23:18 XM_066083499.1.fasta
-rw-rw-r-- 1 sam sam 147K Jan 21 13:50 XM_066083499.1-NCBI-BLASTn-alignment.png</code></pre>
</section>
</section>
<section id="retrieve-gene-sequences" class="level1">
<h1>4 RETRIEVE GENE SEQUENCES</h1>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The below would <em>not</em> execute in Rstudio/Rstudio Server. I tried on two different computers and the chunk would just “run” indefinitly. Code executes without issue in a terminal.</p>
</div>
</div>
<pre><code>for ID in "${sequence_IDs[@]}"
do
  ${esearch} -db nuccore -query "${ID}" \
  | ${efetch} -format fasta \
  &gt; "${data_dir}"/"${ID}".fasta
done</code></pre>
</section>
<section id="primer-design-using-primer3" class="level1">
<h1>5 PRIMER DESIGN USING <a href="https://github.com/primer3-org/primer3">PRIMER3</a></h1>
<section id="design-primers" class="level2">
<h2 class="anchored" data-anchor-id="design-primers">5.1 Design primers</h2>
<p>Quick explanation: <a href="https://github.com/primer3-org/primer3">Primer3</a> requires a specially formatted input file. The file must be formatted similarly to this:</p>
<pre><code>SEQUENCE_ID=${seq_id}
SEQUENCE_TEMPLATE=${sequence}
PRIMER_TASK=generic
PRIMER_PICK_LEFT_PRIMER=3
PRIMER_PICK_RIGHT_PRIMER=3
PRIMER_OPT_SIZE=18
PRIMER_MIN_SIZE=15
PRIMER_MAX_SIZE=21
PRIMER_MAX_NS_ACCEPTED=1
PRIMER_PRODUCT_SIZE_RANGE=75-150
P3_FILE_FLAG=1
PRIMER_EXPLAIN_FLAG=1
=</code></pre>
<p>Values after the <code>=</code> on each line can be changed to whatever values the user decides. The ${sequence} must be a nucleotide sequence on a single line, with no line breaks.</p>
<p>The code in the chunk below uses a heredoc to write this information to a file. Use of a heredoc allows the variables specified in the Primer3 config to expand to their actual values. Everything <em>between</em> the following two lines gets printed (via cat) as shown and then redirected to the indicated file (<code>primer3-params.txt</code>):</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="op">&lt;&lt; EOF</span> <span class="op">&gt;</span> <span class="va">${output_top}</span>/primer3-params.txt</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="st">This text will end up in the file.</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="st">So will this.</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="st">And this.</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="op">EOF</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><a href="https://github.com/primer3-org/primer3">Primer3</a> is run with the <code>--format_output</code> to make a nice, human-readable output format.</p>
<p>I’ve also set <a href="https://github.com/primer3-org/primer3">Primer3</a> to look for sequencing primers and have defined the <code>SEQUENCE_TARGET</code>.</p>
<section id="ncbi-blastn-alignments" class="level4">
<h4 class="anchored" data-anchor-id="ncbi-blastn-alignments">5.1.0.1 NCBI BLASTn alignments</h4>
<p>Each sequence was BLASTn’d against <em>C.gigas</em> nucleotides to identify unique regions in each of the three variants to improve our ability to detect each variant’s expression level.</p>
<section id="nm_001305360.1" class="level5">
<h5 class="anchored" data-anchor-id="nm_001305360.1">5.1.0.1.1 NM_001305360.1</h5>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./NM_001305360.1-NCBI-BLASTn-alignment.png" class="img-fluid figure-img"></p>
<figcaption>NM_001305360.1 BLASTn alignment</figcaption>
</figure>
</div>
</section>
<section id="xm_034462847.2" class="level5">
<h5 class="anchored" data-anchor-id="xm_034462847.2">5.1.0.1.2 XM_034462847.2</h5>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./XM_034462847.2-NCBI-BLASTn-alignment.png" class="img-fluid figure-img"></p>
<figcaption>XM_034462847.2 BLASTn alignment</figcaption>
</figure>
</div>
</section>
<section id="xm_066083499.1" class="level5">
<h5 class="anchored" data-anchor-id="xm_066083499.1">5.1.0.1.3 XM_066083499.1</h5>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./XM_066083499.1-NCBI-BLASTn-alignment.png" class="img-fluid figure-img"></p>
<figcaption>XM_066083499.1 BLASTn alignment</figcaption>
</figure>
</div>
</section>
</section>
<section id="nm_001305360.1-1" class="level3">
<h3 class="anchored" data-anchor-id="nm_001305360.1-1">5.1.1 NM_001305360.1</h3>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get sequence only, by skipping the first record</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove newlines so sequence is on a single line</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="va">sequence</span><span class="op">=</span><span class="va">$(</span><span class="fu">awk</span> <span class="st">'NR &gt; 1'</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">/</span><span class="va">${NM_001305360_fasta}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="at">-d</span> <span class="st">'\n'</span><span class="va">)</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="va">sequence_ID</span><span class="op">=</span><span class="st">"NM_001305360.1"</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Use heredoc to create Primer3 parameters file</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="co"># PRIMER_TASK=generic is used with PRIMER_PICK settings to be PCR primers.</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="op">&lt;&lt; EOF</span> <span class="op">&gt;</span> <span class="va">${output_top}</span>/NM_001305360.1-primer3-params.txt</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a><span class="st">SEQUENCE_ID=</span><span class="va">${sequence_ID}</span></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a><span class="st">SEQUENCE_TEMPLATE=</span><span class="va">${sequence}</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_TASK=generic</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MIN_TM=50</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_OPT_TM=60</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a><span class="st">SEQUENCE_PRIMER_PAIR_OK_REGION_LIST=1109,50,,</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PICK_LEFT_PRIMER=1</span></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PICK_INTERNAL_OLIGO=0</span></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PICK_RIGHT_PRIMER=1</span></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_OPT_SIZE=20</span></span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MIN_SIZE=18</span></span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MAX_SIZE=25</span></span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PRODUCT_SIZE_RANGE=75-150</span></span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MAX_NS_ACCEPTED=0</span></span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a><span class="st">P3_FILE_FLAG=1</span></span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_EXPLAIN_FLAG=1</span></span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_THERMODYNAMIC_PARAMETERS_PATH=</span><span class="va">${primer3_config}</span></span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_NUM_RETURN=3</span></span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PAIR_EXPLAIN=considered 0, ok 0</span></span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a><span class="st">=</span></span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a><span class="op">EOF</span></span>
<span id="cb12-35"><a href="#cb12-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-36"><a href="#cb12-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-37"><a href="#cb12-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Run Primer3</span></span>
<span id="cb12-38"><a href="#cb12-38" aria-hidden="true" tabindex="-1"></a><span class="va">${primer3}</span> <span class="dt">\</span></span>
<span id="cb12-39"><a href="#cb12-39" aria-hidden="true" tabindex="-1"></a>--format_output <span class="dt">\</span></span>
<span id="cb12-40"><a href="#cb12-40" aria-hidden="true" tabindex="-1"></a>--output=<span class="st">"</span><span class="va">${output_top}</span><span class="st">/NM_001305360.1-primer3-primers.txt"</span> <span class="dt">\</span></span>
<span id="cb12-41"><a href="#cb12-41" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${output_top}</span><span class="st">/NM_001305360.1-primer3-params.txt"</span></span>
<span id="cb12-42"><a href="#cb12-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-43"><a href="#cb12-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Run Primer3 with default output for parsable results</span></span>
<span id="cb12-44"><a href="#cb12-44" aria-hidden="true" tabindex="-1"></a><span class="va">${primer3}</span> <span class="dt">\</span></span>
<span id="cb12-45"><a href="#cb12-45" aria-hidden="true" tabindex="-1"></a>--output=<span class="st">"</span><span class="va">${output_top}</span><span class="st">/NM_001305360.1-primer3-primers-default-format.txt"</span> <span class="dt">\</span></span>
<span id="cb12-46"><a href="#cb12-46" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${output_top}</span><span class="st">/NM_001305360.1-primer3-params.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="review-primers" class="level4">
<h4 class="anchored" data-anchor-id="review-primers">5.1.1.1 Review primers</h4>
<section id="easy-to-read" class="level5">
<h5 class="anchored" data-anchor-id="easy-to-read">5.1.1.1.1 Easy-to-read</h5>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/NM_001305360.1-primer3-primers.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>PRIMER PICKING RESULTS FOR NM_001305360.1

No mispriming library specified
Using 0-based sequence positions
OLIGO            start  len      tm     gc%  any_th  3'_th hairpin seq
LEFT PRIMER       1112   20   57.36   45.00    8.11   7.53    0.00 TCTCACTGTGGAAATGCACA
RIGHT PRIMER      1194   21   56.16   42.86    0.00   0.00    0.00 TGGACATTCATCTCTACTGCT
SEQUENCE SIZE: 1214
INCLUDED REGION SIZE: 1214

PRODUCT SIZE: 83, PAIR ANY_TH COMPL: 0.00, PAIR 3'_TH COMPL: 0.00
PAIR_OK_REGIONS (left_start, left_len, right_start, right_len)*: 1109,50,1158,57

    0 GGAATTTTCAGATTTGAGAAGTGAAGAATATATCTGAGGTACTTTAAATATACGATGCGC
                                                                  

   60 TATGGGAAGTTTGCGACAAATAACGAAATTGAGTGAAAATGGTGTCCGGATTTTCTGTTC
                                                                  

  120 CCAACTAAAGAATCTAGAAAATAATTCAATTCTACTGAGGGTCAGTGGAATTCGCACCAG
                                                                  

  180 CAATGGGCTTCGCAATGCAGGAACAAAGGCTGACGTTGATGAAAATATAAAGAAATTTAA
                                                                  

  240 AGAGGAAAATTTTGAAAAAATTCCAGACCCCGAACAGCTCGATCACTTCAGGAAAACACA
                                                                  

  300 GAGCACAGATCAACTGGTGGAGTCAATGAAGAACCCTCCCCCAATGGGCACGCACACACT
                                                                  

  360 GCCCCACCCAATATGGTCTGAAGAAGAACTTCATAGTGTGAAGGTCACACATAAACCGCC
                                                                  

  420 CGAGGGATTCGTAGACAAGCTTGCCTTTCGATCAGTGAAGCTGCTACGTTCAACTTTTGA
                                                                  

  480 CCTTTTAACCGGGTTCAACTGGGGAGAAAGAACGGAGAAGAAATGGGTCTTACGGATTTG
                                                                  

  540 CTTCCTAGAGACTGTCGCTGGAGTTCCAGGAATGGTGGCAGCGATGACGCGACATTTACA
                                                                  

  600 CTCTTTGCGCCGACTCAAGCGGGATCACGGCTGGATTCATACCCTTTTAGAGGAAGCAGA
                                                                  

  660 AAACGAGAGAATGCATCTGATGACAGCATTGCAGCTTCGACAACCATCTTGGTTATTTAG
                                                                  

  720 AAGTGGCGTCATCGTGTCTCAGGGTGCTTTCGTCACGATGTTCAGCATTGCCTATATGCT
                                                                  

  780 AAGTCCTCGGTTTTGTCATCGTTTCGTTGGCTATCTAGAGGAGGAAGCAGTGTTTACTTA
                                                                  

  840 TTCAAAATGTTTGAAGGATATAGAGTCAGGGTCCCTTAAACACTGGCAAACAAAAGCAGC
                                                                  

  900 ACCCGACGTGGCCATCCGCTATTGGAAACTCCCAGAAACTGCATCCATGAAGGATGTAGT
                                                                  

  960 GTTGGCAATTCGAGCAGATGAGGCTCACCATAGGGTCGTGAACCATACTCTGGCTTCCAT
                                                                  

 1020 GAAAGAAGACGAATACAATCCTTATGAGCCAGGGAAGTGATATGGAAGCAGAGAAAACAG
                                                                  

 1080 TGTTAACATTCTGTAGTTATACTGTGAATACATCTCACTGTGGAAATGCACATTACTTGC
                                      &gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;        

 1140 CTTTGATACAATGTACATGTACATGTAGTAAGTAAGCAGTAGAGATGAATGTCCATTGTA
                                        &lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;     

 1200 TAATTCAGGCCATG
                    

KEYS (in order of precedence):
&gt;&gt;&gt;&gt;&gt;&gt; left primer
&lt;&lt;&lt;&lt;&lt;&lt; right primer

ADDITIONAL OLIGOS
                    start  len      tm     gc%  any_th  3'_th hairpin seq

 1 LEFT PRIMER       1110   22   59.18   45.45    8.11   7.53    0.00 CATCTCACTGTGGAAATGCACA
   RIGHT PRIMER      1194   21   56.16   42.86    0.00   0.00    0.00 TGGACATTCATCTCTACTGCT
   PRODUCT SIZE: 85, PAIR ANY_TH COMPL: 0.00, PAIR 3'_TH COMPL: 0.00

 2 LEFT PRIMER       1109   22   59.18   45.45    8.11   0.00    0.00 ACATCTCACTGTGGAAATGCAC
   RIGHT PRIMER      1194   21   56.16   42.86    0.00   0.00    0.00 TGGACATTCATCTCTACTGCT
   PRODUCT SIZE: 86, PAIR ANY_TH COMPL: 0.00, PAIR 3'_TH COMPL: 0.00

Statistics
         con   too    in    in   not          no    tm    tm   high  high  high        high      
         sid  many   tar  excl    ok   bad    GC   too   too any_th 3'_th hair-  poly   end      
        ered    Ns   get   reg   reg   GC% clamp   low  high  compl compl   pin     X  stab    ok
Left    9120     0     0     0  8887     0     0    22     0      0     0     0     0     0   211
Right   9120     0     0     0  8836     0     0    90     0      0     0     0     0     0   194
Pair Stats:
considered 3, ok 3
libprimer3 release 2.6.1</code></pre>
</section>
<section id="print-the-full-output-file" class="level5">
<h5 class="anchored" data-anchor-id="print-the-full-output-file">5.1.1.1.2 Print the full output file</h5>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/NM_001305360.1-primer3-primers-default-format.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>SEQUENCE_ID=NM_001305360.1
SEQUENCE_TEMPLATE=GGAATTTTCAGATTTGAGAAGTGAAGAATATATCTGAGGTACTTTAAATATACGATGCGCTATGGGAAGTTTGCGACAAATAACGAAATTGAGTGAAAATGGTGTCCGGATTTTCTGTTCCCAACTAAAGAATCTAGAAAATAATTCAATTCTACTGAGGGTCAGTGGAATTCGCACCAGCAATGGGCTTCGCAATGCAGGAACAAAGGCTGACGTTGATGAAAATATAAAGAAATTTAAAGAGGAAAATTTTGAAAAAATTCCAGACCCCGAACAGCTCGATCACTTCAGGAAAACACAGAGCACAGATCAACTGGTGGAGTCAATGAAGAACCCTCCCCCAATGGGCACGCACACACTGCCCCACCCAATATGGTCTGAAGAAGAACTTCATAGTGTGAAGGTCACACATAAACCGCCCGAGGGATTCGTAGACAAGCTTGCCTTTCGATCAGTGAAGCTGCTACGTTCAACTTTTGACCTTTTAACCGGGTTCAACTGGGGAGAAAGAACGGAGAAGAAATGGGTCTTACGGATTTGCTTCCTAGAGACTGTCGCTGGAGTTCCAGGAATGGTGGCAGCGATGACGCGACATTTACACTCTTTGCGCCGACTCAAGCGGGATCACGGCTGGATTCATACCCTTTTAGAGGAAGCAGAAAACGAGAGAATGCATCTGATGACAGCATTGCAGCTTCGACAACCATCTTGGTTATTTAGAAGTGGCGTCATCGTGTCTCAGGGTGCTTTCGTCACGATGTTCAGCATTGCCTATATGCTAAGTCCTCGGTTTTGTCATCGTTTCGTTGGCTATCTAGAGGAGGAAGCAGTGTTTACTTATTCAAAATGTTTGAAGGATATAGAGTCAGGGTCCCTTAAACACTGGCAAACAAAAGCAGCACCCGACGTGGCCATCCGCTATTGGAAACTCCCAGAAACTGCATCCATGAAGGATGTAGTGTTGGCAATTCGAGCAGATGAGGCTCACCATAGGGTCGTGAACCATACTCTGGCTTCCATGAAAGAAGACGAATACAATCCTTATGAGCCAGGGAAGTGATATGGAAGCAGAGAAAACAGTGTTAACATTCTGTAGTTATACTGTGAATACATCTCACTGTGGAAATGCACATTACTTGCCTTTGATACAATGTACATGTACATGTAGTAAGTAAGCAGTAGAGATGAATGTCCATTGTATAATTCAGGCCATG
PRIMER_TASK=generic
PRIMER_MIN_TM=50
PRIMER_OPT_TM=60
SEQUENCE_PRIMER_PAIR_OK_REGION_LIST=1109,50,,
PRIMER_PICK_LEFT_PRIMER=1
PRIMER_PICK_INTERNAL_OLIGO=0
PRIMER_PICK_RIGHT_PRIMER=1
PRIMER_OPT_SIZE=20
PRIMER_MIN_SIZE=18
PRIMER_MAX_SIZE=25
PRIMER_PRODUCT_SIZE_RANGE=75-150
PRIMER_MAX_NS_ACCEPTED=0
P3_FILE_FLAG=1
PRIMER_EXPLAIN_FLAG=1
PRIMER_THERMODYNAMIC_PARAMETERS_PATH=/home/shared/primer3-2.6.1/src/primer3_config
PRIMER_NUM_RETURN=3
PRIMER_PAIR_EXPLAIN=considered 0, ok 0
PRIMER_LEFT_EXPLAIN=considered 9120, low tm 22, not in any ok left region 8887, ok 211
PRIMER_RIGHT_EXPLAIN=considered 9120, low tm 90, not in any ok right region 8836, ok 194
PRIMER_PAIR_EXPLAIN=considered 3, ok 3
PRIMER_LEFT_NUM_RETURNED=3
PRIMER_RIGHT_NUM_RETURNED=3
PRIMER_INTERNAL_NUM_RETURNED=0
PRIMER_PAIR_NUM_RETURNED=3
PRIMER_PAIR_0_PENALTY=7.475896
PRIMER_LEFT_0_PENALTY=2.636647
PRIMER_RIGHT_0_PENALTY=4.839250
PRIMER_LEFT_0_SEQUENCE=TCTCACTGTGGAAATGCACA
PRIMER_RIGHT_0_SEQUENCE=TGGACATTCATCTCTACTGCT
PRIMER_LEFT_0=1112,20
PRIMER_RIGHT_0=1194,21
PRIMER_LEFT_0_TM=57.363
PRIMER_RIGHT_0_TM=56.161
PRIMER_LEFT_0_GC_PERCENT=45.000
PRIMER_RIGHT_0_GC_PERCENT=42.857
PRIMER_LEFT_0_SELF_ANY_TH=8.11
PRIMER_RIGHT_0_SELF_ANY_TH=0.00
PRIMER_LEFT_0_SELF_END_TH=7.53
PRIMER_RIGHT_0_SELF_END_TH=0.00
PRIMER_LEFT_0_HAIRPIN_TH=0.00
PRIMER_RIGHT_0_HAIRPIN_TH=0.00
PRIMER_LEFT_0_END_STABILITY=4.5700
PRIMER_RIGHT_0_END_STABILITY=4.2400
PRIMER_PAIR_0_COMPL_ANY_TH=0.00
PRIMER_PAIR_0_COMPL_END_TH=0.00
PRIMER_PAIR_0_PRODUCT_SIZE=83
PRIMER_PAIR_0_PRODUCT_TM=76.5
PRIMER_PAIR_1_PENALTY=7.654753
PRIMER_LEFT_1_PENALTY=2.815503
PRIMER_RIGHT_1_PENALTY=4.839250
PRIMER_LEFT_1_SEQUENCE=CATCTCACTGTGGAAATGCACA
PRIMER_RIGHT_1_SEQUENCE=TGGACATTCATCTCTACTGCT
PRIMER_LEFT_1=1110,22
PRIMER_RIGHT_1=1194,21
PRIMER_LEFT_1_TM=59.184
PRIMER_RIGHT_1_TM=56.161
PRIMER_LEFT_1_GC_PERCENT=45.455
PRIMER_RIGHT_1_GC_PERCENT=42.857
PRIMER_LEFT_1_SELF_ANY_TH=8.11
PRIMER_RIGHT_1_SELF_ANY_TH=0.00
PRIMER_LEFT_1_SELF_END_TH=7.53
PRIMER_RIGHT_1_SELF_END_TH=0.00
PRIMER_LEFT_1_HAIRPIN_TH=0.00
PRIMER_RIGHT_1_HAIRPIN_TH=0.00
PRIMER_LEFT_1_END_STABILITY=4.5700
PRIMER_RIGHT_1_END_STABILITY=4.2400
PRIMER_PAIR_1_COMPL_ANY_TH=0.00
PRIMER_PAIR_1_COMPL_END_TH=0.00
PRIMER_PAIR_1_PRODUCT_SIZE=85
PRIMER_PAIR_1_PRODUCT_TM=76.8
PRIMER_PAIR_2_PENALTY=7.655337
PRIMER_LEFT_2_PENALTY=2.816087
PRIMER_RIGHT_2_PENALTY=4.839250
PRIMER_LEFT_2_SEQUENCE=ACATCTCACTGTGGAAATGCAC
PRIMER_RIGHT_2_SEQUENCE=TGGACATTCATCTCTACTGCT
PRIMER_LEFT_2=1109,22
PRIMER_RIGHT_2=1194,21
PRIMER_LEFT_2_TM=59.184
PRIMER_RIGHT_2_TM=56.161
PRIMER_LEFT_2_GC_PERCENT=45.455
PRIMER_RIGHT_2_GC_PERCENT=42.857
PRIMER_LEFT_2_SELF_ANY_TH=8.11
PRIMER_RIGHT_2_SELF_ANY_TH=0.00
PRIMER_LEFT_2_SELF_END_TH=0.00
PRIMER_RIGHT_2_SELF_END_TH=0.00
PRIMER_LEFT_2_HAIRPIN_TH=0.00
PRIMER_RIGHT_2_HAIRPIN_TH=0.00
PRIMER_LEFT_2_END_STABILITY=4.5700
PRIMER_RIGHT_2_END_STABILITY=4.2400
PRIMER_PAIR_2_COMPL_ANY_TH=0.00
PRIMER_PAIR_2_COMPL_END_TH=0.00
PRIMER_PAIR_2_PRODUCT_SIZE=86
PRIMER_PAIR_2_PRODUCT_TM=76.7
=</code></pre>
</section>
</section>
</section>
<section id="xm_034462847.2-1" class="level3">
<h3 class="anchored" data-anchor-id="xm_034462847.2-1">5.1.2 XM_034462847.2</h3>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get sequence only, by skipping the first record</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove newlines so sequence is on a single line</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="va">sequence</span><span class="op">=</span><span class="va">$(</span><span class="fu">awk</span> <span class="st">'NR &gt; 1'</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">/</span><span class="va">${XM_034462847_fasta}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="at">-d</span> <span class="st">'\n'</span><span class="va">)</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="va">sequence_ID</span><span class="op">=</span><span class="st">"XM_034462847.2"</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Use heredoc to create Primer3 parameters file</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="co"># PRIMER_TASK=generic is used with PRIMER_PICK settings to be PCR primers.</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="op">&lt;&lt; EOF</span> <span class="op">&gt;</span> <span class="va">${output_top}</span>/XM_034462847.2-primer3-params.txt</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="st">SEQUENCE_ID=</span><span class="va">${sequence_ID}</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a><span class="st">SEQUENCE_TEMPLATE=</span><span class="va">${sequence}</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_TASK=generic</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MIN_TM=50</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_OPT_TM=60</span></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a><span class="st">SEQUENCE_PRIMER_PAIR_OK_REGION_LIST=90,25,,</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PICK_LEFT_PRIMER=1</span></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PICK_INTERNAL_OLIGO=0</span></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PICK_RIGHT_PRIMER=1</span></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_OPT_SIZE=20</span></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MIN_SIZE=18</span></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MAX_SIZE=25</span></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PRODUCT_SIZE_RANGE=75-150</span></span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MAX_NS_ACCEPTED=0</span></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a><span class="st">P3_FILE_FLAG=1</span></span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_EXPLAIN_FLAG=1</span></span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_THERMODYNAMIC_PARAMETERS_PATH=</span><span class="va">${primer3_config}</span></span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_NUM_RETURN=3</span></span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PAIR_EXPLAIN=considered 0, ok 0</span></span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a><span class="st">=</span></span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a><span class="op">EOF</span></span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Run Primer3</span></span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a><span class="va">${primer3}</span> <span class="dt">\</span></span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>--format_output <span class="dt">\</span></span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>--output=<span class="st">"</span><span class="va">${output_top}</span><span class="st">/XM_034462847.2-primer3-primers.txt"</span> <span class="dt">\</span></span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${output_top}</span><span class="st">/XM_034462847.2-primer3-params.txt"</span></span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Run Primer3 with default output for parsable results</span></span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a><span class="va">${primer3}</span> <span class="dt">\</span></span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a>--output=<span class="st">"</span><span class="va">${output_top}</span><span class="st">/XM_034462847.2-primer3-primers-default-format.txt"</span> <span class="dt">\</span></span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${output_top}</span><span class="st">/XM_034462847.2-primer3-params.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="review-primers-1" class="level4">
<h4 class="anchored" data-anchor-id="review-primers-1">5.1.2.1 Review primers</h4>
<section id="easy-to-read-1" class="level5">
<h5 class="anchored" data-anchor-id="easy-to-read-1">5.1.2.1.1 Easy-to-read</h5>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/XM_034462847.2-primer3-primers.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>PRIMER PICKING RESULTS FOR XM_034462847.2

No mispriming library specified
Using 0-based sequence positions
OLIGO            start  len      tm     gc%  any_th  3'_th hairpin seq
LEFT PRIMER         91   24   59.07   41.67    0.00   0.00    0.00 TCGATCACTTCAGAATTTCACAGG
RIGHT PRIMER       169   20   60.03   60.00    0.00   0.00    0.00 CCATTGGGGGAGGGTTCTTC
SEQUENCE SIZE: 931
INCLUDED REGION SIZE: 931

PRODUCT SIZE: 79, PAIR ANY_TH COMPL: 0.00, PAIR 3'_TH COMPL: 0.00
PAIR_OK_REGIONS (left_start, left_len, right_start, right_len)*: 90,25,139,109

    0 CTTCGCAATGCAGGAACAAAGGCTGACGTTGATGAAAATATAAAGAAATTTAAAGAGGAA
                                                                  

   60 AATTTTGAAAAAATTCCAGACCCCGAACAGCTCGATCACTTCAGAATTTCACAGGAAAAC
                                     &gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;     

  120 ACAGAGCACAGATCAACTGGTGGAGTCAATGAAGAACCCTCCCCCAATGGGCACGCACAC
                                    &lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;          

  180 ACTGCCCCACCCAATATGGTCTGAAGAAGAACTTCATAGTGTGAAGGTCACACATAAACC
                                                                  

  240 GCCCGAGGGATTCGTAGACAAGCTTGCCTTTCGATCAGTGAAGCTGCTACGTTCAACTTT
                                                                  

  300 TGACCTTTTAACCGGGTTCAACTGGGGAGAAAGAACGGAGAAGAAATGGGTGTTACGGAT
                                                                  

  360 TTGCTTCCTAGAGACTGTCGCTGGAGTTCCAGGGATGGTGGCAGCGATGACGCGACATTT
                                                                  

  420 ACACTCTTTGCGCCGACTCAAGCGGGATCACGGCTGGATTCATACCCTTTTAGAGGAAGC
                                                                  

  480 AGAAAACGAAAGAATGCATCTGATGACAGCATTGCAGCTTCGACAACCATCTTGGTTATT
                                                                  

  540 TAGAAGTGGCGTCATCGTGTCTCAGGGTGCTTTCGTCACGATGTTCAGCATTGCCTATAT
                                                                  

  600 GCTAAGCCCTCGGTTTTGTCATCGGTTCGTTGGCTATCTAGAGGAGGAAGCAGTGTTTAC
                                                                  

  660 TTACTCAAAATGTTTGAAGGATATAGAGTCAGGGCCCCTTAAACACTGGCAAACAAAAGC
                                                                  

  720 AGCACCCGACGTGGCTATCCGCTATTGGAAACTCCCAGAAACTGCATCCATGAAGGATGT
                                                                  

  780 AGTGTTGGCAATTCGAGCAGATGAGGCTCACCATAGGGTCGTGAACCATACTCTGGCTTC
                                                                  

  840 CATGAAAGAAGACGAATACAATCCTTATGAGCCAGGGAAGTGATATGGAAGCAGAGAAAA
                                                                  

  900 CAGCGTTAACATTCTGTAGTTATACTGTGAA
                                     

KEYS (in order of precedence):
&gt;&gt;&gt;&gt;&gt;&gt; left primer
&lt;&lt;&lt;&lt;&lt;&lt; right primer

ADDITIONAL OLIGOS
                    start  len      tm     gc%  any_th  3'_th hairpin seq

 1 LEFT PRIMER         91   24   59.07   41.67    0.00   0.00    0.00 TCGATCACTTCAGAATTTCACAGG
   RIGHT PRIMER       196   20   60.03   55.00    0.00   0.00    0.00 ATATTGGGTGGGGCAGTGTG
   PRODUCT SIZE: 106, PAIR ANY_TH COMPL: 0.00, PAIR 3'_TH COMPL: 0.00

 2 LEFT PRIMER         91   24   59.07   41.67    0.00   0.00    0.00 TCGATCACTTCAGAATTTCACAGG
   RIGHT PRIMER       197   20   60.03   55.00    0.00   0.00    0.00 CATATTGGGTGGGGCAGTGT
   PRODUCT SIZE: 107, PAIR ANY_TH COMPL: 0.00, PAIR 3'_TH COMPL: 0.00

Statistics
         con   too    in    in   not          no    tm    tm   high  high  high        high      
         sid  many   tar  excl    ok   bad    GC   too   too any_th 3'_th hair-  poly   end      
        ered    Ns   get   reg   reg   GC% clamp   low  high  compl compl   pin     X  stab    ok
Left    6856     0     0     0  6820     0     0    10     0      0     0     0     0     0    26
Right   6856     0     0     0  6148     0     0    42   255      0     0   185     0     0   226
Pair Stats:
considered 4, ok 4
libprimer3 release 2.6.1</code></pre>
</section>
<section id="print-the-full-output-file-1" class="level5">
<h5 class="anchored" data-anchor-id="print-the-full-output-file-1">5.1.2.1.2 Print the full output file</h5>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/XM_034462847.2-primer3-primers-default-format.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>SEQUENCE_ID=XM_034462847.2
SEQUENCE_TEMPLATE=CTTCGCAATGCAGGAACAAAGGCTGACGTTGATGAAAATATAAAGAAATTTAAAGAGGAAAATTTTGAAAAAATTCCAGACCCCGAACAGCTCGATCACTTCAGAATTTCACAGGAAAACACAGAGCACAGATCAACTGGTGGAGTCAATGAAGAACCCTCCCCCAATGGGCACGCACACACTGCCCCACCCAATATGGTCTGAAGAAGAACTTCATAGTGTGAAGGTCACACATAAACCGCCCGAGGGATTCGTAGACAAGCTTGCCTTTCGATCAGTGAAGCTGCTACGTTCAACTTTTGACCTTTTAACCGGGTTCAACTGGGGAGAAAGAACGGAGAAGAAATGGGTGTTACGGATTTGCTTCCTAGAGACTGTCGCTGGAGTTCCAGGGATGGTGGCAGCGATGACGCGACATTTACACTCTTTGCGCCGACTCAAGCGGGATCACGGCTGGATTCATACCCTTTTAGAGGAAGCAGAAAACGAAAGAATGCATCTGATGACAGCATTGCAGCTTCGACAACCATCTTGGTTATTTAGAAGTGGCGTCATCGTGTCTCAGGGTGCTTTCGTCACGATGTTCAGCATTGCCTATATGCTAAGCCCTCGGTTTTGTCATCGGTTCGTTGGCTATCTAGAGGAGGAAGCAGTGTTTACTTACTCAAAATGTTTGAAGGATATAGAGTCAGGGCCCCTTAAACACTGGCAAACAAAAGCAGCACCCGACGTGGCTATCCGCTATTGGAAACTCCCAGAAACTGCATCCATGAAGGATGTAGTGTTGGCAATTCGAGCAGATGAGGCTCACCATAGGGTCGTGAACCATACTCTGGCTTCCATGAAAGAAGACGAATACAATCCTTATGAGCCAGGGAAGTGATATGGAAGCAGAGAAAACAGCGTTAACATTCTGTAGTTATACTGTGAA
PRIMER_TASK=generic
PRIMER_MIN_TM=50
PRIMER_OPT_TM=60
SEQUENCE_PRIMER_PAIR_OK_REGION_LIST=90,25,,
PRIMER_PICK_LEFT_PRIMER=1
PRIMER_PICK_INTERNAL_OLIGO=0
PRIMER_PICK_RIGHT_PRIMER=1
PRIMER_OPT_SIZE=20
PRIMER_MIN_SIZE=18
PRIMER_MAX_SIZE=25
PRIMER_PRODUCT_SIZE_RANGE=75-150
PRIMER_MAX_NS_ACCEPTED=0
P3_FILE_FLAG=1
PRIMER_EXPLAIN_FLAG=1
PRIMER_THERMODYNAMIC_PARAMETERS_PATH=/home/shared/primer3-2.6.1/src/primer3_config
PRIMER_NUM_RETURN=3
PRIMER_PAIR_EXPLAIN=considered 0, ok 0
PRIMER_LEFT_EXPLAIN=considered 6856, low tm 10, not in any ok left region 6820, ok 26
PRIMER_RIGHT_EXPLAIN=considered 6856, low tm 42, high tm 255, high hairpin stability 185, not in any ok right region 6148, ok 226
PRIMER_PAIR_EXPLAIN=considered 4, ok 4
PRIMER_LEFT_NUM_RETURNED=3
PRIMER_RIGHT_NUM_RETURNED=3
PRIMER_INTERNAL_NUM_RETURNED=0
PRIMER_PAIR_NUM_RETURNED=3
PRIMER_PAIR_0_PENALTY=4.964710
PRIMER_LEFT_0_PENALTY=4.931601
PRIMER_RIGHT_0_PENALTY=0.033109
PRIMER_LEFT_0_SEQUENCE=TCGATCACTTCAGAATTTCACAGG
PRIMER_RIGHT_0_SEQUENCE=CCATTGGGGGAGGGTTCTTC
PRIMER_LEFT_0=91,24
PRIMER_RIGHT_0=169,20
PRIMER_LEFT_0_TM=59.068
PRIMER_RIGHT_0_TM=60.033
PRIMER_LEFT_0_GC_PERCENT=41.667
PRIMER_RIGHT_0_GC_PERCENT=60.000
PRIMER_LEFT_0_SELF_ANY_TH=0.00
PRIMER_RIGHT_0_SELF_ANY_TH=0.00
PRIMER_LEFT_0_SELF_END_TH=0.00
PRIMER_RIGHT_0_SELF_END_TH=0.00
PRIMER_LEFT_0_HAIRPIN_TH=0.00
PRIMER_RIGHT_0_HAIRPIN_TH=0.00
PRIMER_LEFT_0_END_STABILITY=4.0000
PRIMER_RIGHT_0_END_STABILITY=2.8700
PRIMER_PAIR_0_COMPL_ANY_TH=0.00
PRIMER_PAIR_0_COMPL_END_TH=0.00
PRIMER_PAIR_0_PRODUCT_SIZE=79
PRIMER_PAIR_0_PRODUCT_TM=80.1
PRIMER_PAIR_1_PENALTY=4.964809
PRIMER_LEFT_1_PENALTY=4.931601
PRIMER_RIGHT_1_PENALTY=0.033208
PRIMER_LEFT_1_SEQUENCE=TCGATCACTTCAGAATTTCACAGG
PRIMER_RIGHT_1_SEQUENCE=ATATTGGGTGGGGCAGTGTG
PRIMER_LEFT_1=91,24
PRIMER_RIGHT_1=196,20
PRIMER_LEFT_1_TM=59.068
PRIMER_RIGHT_1_TM=60.033
PRIMER_LEFT_1_GC_PERCENT=41.667
PRIMER_RIGHT_1_GC_PERCENT=55.000
PRIMER_LEFT_1_SELF_ANY_TH=0.00
PRIMER_RIGHT_1_SELF_ANY_TH=0.00
PRIMER_LEFT_1_SELF_END_TH=0.00
PRIMER_RIGHT_1_SELF_END_TH=0.00
PRIMER_LEFT_1_HAIRPIN_TH=0.00
PRIMER_RIGHT_1_HAIRPIN_TH=0.00
PRIMER_LEFT_1_END_STABILITY=4.0000
PRIMER_RIGHT_1_END_STABILITY=3.8200
PRIMER_PAIR_1_COMPL_ANY_TH=0.00
PRIMER_PAIR_1_COMPL_END_TH=0.00
PRIMER_PAIR_1_PRODUCT_SIZE=106
PRIMER_PAIR_1_PRODUCT_TM=83.3
PRIMER_PAIR_2_PENALTY=4.964809
PRIMER_LEFT_2_PENALTY=4.931601
PRIMER_RIGHT_2_PENALTY=0.033208
PRIMER_LEFT_2_SEQUENCE=TCGATCACTTCAGAATTTCACAGG
PRIMER_RIGHT_2_SEQUENCE=CATATTGGGTGGGGCAGTGT
PRIMER_LEFT_2=91,24
PRIMER_RIGHT_2=197,20
PRIMER_LEFT_2_TM=59.068
PRIMER_RIGHT_2_TM=60.033
PRIMER_LEFT_2_GC_PERCENT=41.667
PRIMER_RIGHT_2_GC_PERCENT=55.000
PRIMER_LEFT_2_SELF_ANY_TH=0.00
PRIMER_RIGHT_2_SELF_ANY_TH=0.00
PRIMER_LEFT_2_SELF_END_TH=0.00
PRIMER_RIGHT_2_SELF_END_TH=0.00
PRIMER_LEFT_2_HAIRPIN_TH=0.00
PRIMER_RIGHT_2_HAIRPIN_TH=0.00
PRIMER_LEFT_2_END_STABILITY=4.0000
PRIMER_RIGHT_2_END_STABILITY=3.5500
PRIMER_PAIR_2_COMPL_ANY_TH=0.00
PRIMER_PAIR_2_COMPL_END_TH=0.00
PRIMER_PAIR_2_PRODUCT_SIZE=107
PRIMER_PAIR_2_PRODUCT_TM=83.5
=</code></pre>
</section>
</section>
</section>
<section id="xm_066083499.1-1" class="level3">
<h3 class="anchored" data-anchor-id="xm_066083499.1-1">5.1.3 XM_066083499.1</h3>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get sequence only, by skipping the first record</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove newlines so sequence is on a single line</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="va">sequence</span><span class="op">=</span><span class="va">$(</span><span class="fu">awk</span> <span class="st">'NR &gt; 1'</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">/</span><span class="va">${XM_066083499_fasta}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="at">-d</span> <span class="st">'\n'</span><span class="va">)</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="va">sequence_ID</span><span class="op">=</span><span class="st">"XM_066083499.1"</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Use heredoc to create Primer3 parameters file</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="co"># PRIMER_TASK=generic is used with PRIMER_PICK settings to be PCR primers.</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="op">&lt;&lt; EOF</span> <span class="op">&gt;</span> <span class="va">${output_top}</span>/XM_066083499.1-primer3-params.txt</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="st">SEQUENCE_ID=</span><span class="va">${sequence_ID}</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a><span class="st">SEQUENCE_TEMPLATE=</span><span class="va">${sequence}</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_TASK=generic</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MIN_TM=50</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_OPT_TM=60</span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a><span class="st">SEQUENCE_PRIMER_PAIR_OK_REGION_LIST=,,1100,50</span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PICK_LEFT_PRIMER=1</span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PICK_INTERNAL_OLIGO=0</span></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PICK_RIGHT_PRIMER=1</span></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_OPT_SIZE=20</span></span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MIN_SIZE=18</span></span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MAX_SIZE=25</span></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PRODUCT_SIZE_RANGE=75-150</span></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MAX_NS_ACCEPTED=0</span></span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a><span class="st">P3_FILE_FLAG=1</span></span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_EXPLAIN_FLAG=1</span></span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_THERMODYNAMIC_PARAMETERS_PATH=</span><span class="va">${primer3_config}</span></span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_NUM_RETURN=3</span></span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PAIR_EXPLAIN=considered 0, ok 0</span></span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a><span class="st">=</span></span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a><span class="op">EOF</span></span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Run Primer3</span></span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a><span class="va">${primer3}</span> <span class="dt">\</span></span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a>--format_output <span class="dt">\</span></span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a>--output=<span class="st">"</span><span class="va">${output_top}</span><span class="st">/XM_066083499.1-primer3-primers.txt"</span> <span class="dt">\</span></span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${output_top}</span><span class="st">/XM_066083499.1-primer3-params.txt"</span></span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Run Primer3 with default output for parsable results</span></span>
<span id="cb22-44"><a href="#cb22-44" aria-hidden="true" tabindex="-1"></a><span class="va">${primer3}</span> <span class="dt">\</span></span>
<span id="cb22-45"><a href="#cb22-45" aria-hidden="true" tabindex="-1"></a>--output=<span class="st">"</span><span class="va">${output_top}</span><span class="st">/XM_066083499.1-primer3-primers-default-format.txt"</span> <span class="dt">\</span></span>
<span id="cb22-46"><a href="#cb22-46" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${output_top}</span><span class="st">/XM_066083499.1-primer3-params.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="review-primers-2" class="level4">
<h4 class="anchored" data-anchor-id="review-primers-2">5.1.3.1 Review primers</h4>
<section id="easy-to-read-2" class="level5">
<h5 class="anchored" data-anchor-id="easy-to-read-2">5.1.3.1.1 Easy-to-read</h5>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/XM_066083499.1-primer3-primers.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>PRIMER PICKING RESULTS FOR XM_066083499.1

No mispriming library specified
Using 0-based sequence positions
OLIGO            start  len      tm     gc%  any_th  3'_th hairpin seq
LEFT PRIMER       1028   20   60.25   60.00    0.00   0.00    0.00 CAGACGAGGCTAATCACCGG
RIGHT PRIMER      1141   21   58.73   47.62    0.00   0.00   34.07 TGAGGCACATTAAGGGTCTGA
SEQUENCE SIZE: 1329
INCLUDED REGION SIZE: 1329

PRODUCT SIZE: 114, PAIR ANY_TH COMPL: 0.00, PAIR 3'_TH COMPL: 0.00
PAIR_OK_REGIONS (left_start, left_len, right_start, right_len)*: 967,134,1100,50

    0 ATATATTGAGACCGTCTTGCATATTTCAGAAAATAAGAAAAAAAGTCAGTGTAAGGTATT
                                                                  

   60 TTCAGATTTGAGAAGTGAAGAATATATCTGAGGTATTTTAAATATACGAATCGCTATGGG
                                                                  

  120 AAGTTTACGACAAATAACGAAATTGAGTGAAAATGGTGTCCGGATTTTCTGTTCCCAACT
                                                                  

  180 AAAGAATCTAGAAAATAATTCAATTCTACTGAGGGTCAGTGGAATTCGCACCAGCAATGG
                                                                  

  240 GCTTCGCAATGCAGGAACAAAGGCTGACGTTGATGAAAATATAAAGAAATTTAAAGAGGA
                                                                  

  300 AAATTTTGAAAAAATTCCAGACCCCGAACAGCTCGATCACTTCAGGAAAACACAGAGCAC
                                                                  

  360 AGATCAACTGGTGGAGTCAATGAAGAACCCTCCCCCAATGGGCACGCACACACTGCCCCA
                                                                  

  420 CCCAATATGGTCTGAAGAAGAACTTCATAGTGTGAAGGTCACACATAAACCGCCCGAGGG
                                                                  

  480 ATTCGTAGACAAGCTTGCCTTTCGATCAGTGAAGCTGCTACGTTCAACTTTTGACCTTTT
                                                                  

  540 AACCGGGTTCAACTGGGGAGAAAGAACGGAGAAGAAATGGGTGTTACGGATTTGCTTCCT
                                                                  

  600 AGAGACTGTCGCTGGAGTTCCAGGGATGGTGGCAGCGATGACGCGACATTTACACTCTTT
                                                                  

  660 GCGCCGACTCAAGCGGGATCACGGCTGGATTCATACCCTTTTAGAGGAAGCAGAAAACGA
                                                                  

  720 AAGAATGCATCTGATGACAGCATTGCAGCTTCGACAACCATCTTGGTTATTTAGAAGTGG
                                                                  

  780 CGTCATCGTGTCTCAGGGTGCTTTCGTCACGATGTTCAGCATTGCCTATATGCTAAGCCC
                                                                  

  840 TCGGTTTTGTCATCGGTTCGTTGGCTATCTAGAGGAGGAAGCAGTGTTTACTTACTCAAA
                                                                  

  900 ATGTTTGAAGGATATAGAGTCAGGGCCCCTTAAACACTGGCAAACAAAAGCAGCACCCGA
                                                                  

  960 CGTGGCTATCCGCTATTGGAAACTCCCAAATAACGGAACAATGAAAGACGTCATATTTAA
                                                                  

 1020 CATCCGAGCAGACGAGGCTAATCACCGGAAAGTGAACCATGTTTTGGCATCCATGAAAAC
              &gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;                                

 1080 CGACCAGTACAACCCATTTCTACCCGGTCAGTAGTTTTAAATCAGACCCTTAATGTGCCT
                                               &lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;

 1140 CATCTCTGTTTACAGGAAACTGCATCCATGAAGGATGTAGTGTTGGCAATTCGAGCAGAT
      &lt;&lt;                                                          

 1200 GAGGCTCACCATAGGGTCGTGAACCATACTCTGGCTTCCATGAAAGAAGACGAATACAAT
                                                                  

 1260 CCTTATGAGCCAGGGAAGTGATATGGAAGCAGAGAAAACAGCGTTAACATTCTGTAGTTA
                                                                  

 1320 TACTGTGAA
               

KEYS (in order of precedence):
&gt;&gt;&gt;&gt;&gt;&gt; left primer
&lt;&lt;&lt;&lt;&lt;&lt; right primer

ADDITIONAL OLIGOS
                    start  len      tm     gc%  any_th  3'_th hairpin seq

 1 LEFT PRIMER       1023   20   60.32   60.00    0.00   0.00   42.78 CCGAGCAGACGAGGCTAATC
   RIGHT PRIMER      1141   21   58.73   47.62    0.00   0.00   34.07 TGAGGCACATTAAGGGTCTGA
   PRODUCT SIZE: 119, PAIR ANY_TH COMPL: 0.07, PAIR 3'_TH COMPL: 0.00

 2 LEFT PRIMER       1024   20   59.62   55.00    0.00   0.00   42.78 CGAGCAGACGAGGCTAATCA
   RIGHT PRIMER      1141   21   58.73   47.62    0.00   0.00   34.07 TGAGGCACATTAAGGGTCTGA
   PRODUCT SIZE: 118, PAIR ANY_TH COMPL: 0.07, PAIR 3'_TH COMPL: 0.00

Statistics
         con   too    in    in   not          no    tm    tm   high  high  high        high      
         sid  many   tar  excl    ok   bad    GC   too   too any_th 3'_th hair-  poly   end      
        ered    Ns   get   reg   reg   GC% clamp   low  high  compl compl   pin     X  stab    ok
Left   10040     0     0     0  9132     0     0    66    87      0     0   179     0     0   576
Right  10040     0     0     0  9804     0     0    49     0      0     0     0     0     0   187
Pair Stats:
considered 3, ok 3
libprimer3 release 2.6.1</code></pre>
</section>
<section id="print-the-full-output-file-2" class="level5">
<h5 class="anchored" data-anchor-id="print-the-full-output-file-2">5.1.3.1.2 Print the full output file</h5>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/XM_066083499.1-primer3-primers-default-format.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>SEQUENCE_ID=XM_066083499.1
SEQUENCE_TEMPLATE=ATATATTGAGACCGTCTTGCATATTTCAGAAAATAAGAAAAAAAGTCAGTGTAAGGTATTTTCAGATTTGAGAAGTGAAGAATATATCTGAGGTATTTTAAATATACGAATCGCTATGGGAAGTTTACGACAAATAACGAAATTGAGTGAAAATGGTGTCCGGATTTTCTGTTCCCAACTAAAGAATCTAGAAAATAATTCAATTCTACTGAGGGTCAGTGGAATTCGCACCAGCAATGGGCTTCGCAATGCAGGAACAAAGGCTGACGTTGATGAAAATATAAAGAAATTTAAAGAGGAAAATTTTGAAAAAATTCCAGACCCCGAACAGCTCGATCACTTCAGGAAAACACAGAGCACAGATCAACTGGTGGAGTCAATGAAGAACCCTCCCCCAATGGGCACGCACACACTGCCCCACCCAATATGGTCTGAAGAAGAACTTCATAGTGTGAAGGTCACACATAAACCGCCCGAGGGATTCGTAGACAAGCTTGCCTTTCGATCAGTGAAGCTGCTACGTTCAACTTTTGACCTTTTAACCGGGTTCAACTGGGGAGAAAGAACGGAGAAGAAATGGGTGTTACGGATTTGCTTCCTAGAGACTGTCGCTGGAGTTCCAGGGATGGTGGCAGCGATGACGCGACATTTACACTCTTTGCGCCGACTCAAGCGGGATCACGGCTGGATTCATACCCTTTTAGAGGAAGCAGAAAACGAAAGAATGCATCTGATGACAGCATTGCAGCTTCGACAACCATCTTGGTTATTTAGAAGTGGCGTCATCGTGTCTCAGGGTGCTTTCGTCACGATGTTCAGCATTGCCTATATGCTAAGCCCTCGGTTTTGTCATCGGTTCGTTGGCTATCTAGAGGAGGAAGCAGTGTTTACTTACTCAAAATGTTTGAAGGATATAGAGTCAGGGCCCCTTAAACACTGGCAAACAAAAGCAGCACCCGACGTGGCTATCCGCTATTGGAAACTCCCAAATAACGGAACAATGAAAGACGTCATATTTAACATCCGAGCAGACGAGGCTAATCACCGGAAAGTGAACCATGTTTTGGCATCCATGAAAACCGACCAGTACAACCCATTTCTACCCGGTCAGTAGTTTTAAATCAGACCCTTAATGTGCCTCATCTCTGTTTACAGGAAACTGCATCCATGAAGGATGTAGTGTTGGCAATTCGAGCAGATGAGGCTCACCATAGGGTCGTGAACCATACTCTGGCTTCCATGAAAGAAGACGAATACAATCCTTATGAGCCAGGGAAGTGATATGGAAGCAGAGAAAACAGCGTTAACATTCTGTAGTTATACTGTGAA
PRIMER_TASK=generic
PRIMER_MIN_TM=50
PRIMER_OPT_TM=60
SEQUENCE_PRIMER_PAIR_OK_REGION_LIST=,,1100,50
PRIMER_PICK_LEFT_PRIMER=1
PRIMER_PICK_INTERNAL_OLIGO=0
PRIMER_PICK_RIGHT_PRIMER=1
PRIMER_OPT_SIZE=20
PRIMER_MIN_SIZE=18
PRIMER_MAX_SIZE=25
PRIMER_PRODUCT_SIZE_RANGE=75-150
PRIMER_MAX_NS_ACCEPTED=0
P3_FILE_FLAG=1
PRIMER_EXPLAIN_FLAG=1
PRIMER_THERMODYNAMIC_PARAMETERS_PATH=/home/shared/primer3-2.6.1/src/primer3_config
PRIMER_NUM_RETURN=3
PRIMER_PAIR_EXPLAIN=considered 0, ok 0
PRIMER_LEFT_EXPLAIN=considered 10040, low tm 66, high tm 87, high hairpin stability 179, not in any ok left region 9132, ok 576
PRIMER_RIGHT_EXPLAIN=considered 10040, low tm 49, not in any ok right region 9804, ok 187
PRIMER_PAIR_EXPLAIN=considered 3, ok 3
PRIMER_LEFT_NUM_RETURNED=3
PRIMER_RIGHT_NUM_RETURNED=3
PRIMER_INTERNAL_NUM_RETURNED=0
PRIMER_PAIR_NUM_RETURNED=3
PRIMER_PAIR_0_PENALTY=2.514439
PRIMER_LEFT_0_PENALTY=0.249073
PRIMER_RIGHT_0_PENALTY=2.265367
PRIMER_LEFT_0_SEQUENCE=CAGACGAGGCTAATCACCGG
PRIMER_RIGHT_0_SEQUENCE=TGAGGCACATTAAGGGTCTGA
PRIMER_LEFT_0=1028,20
PRIMER_RIGHT_0=1141,21
PRIMER_LEFT_0_TM=60.249
PRIMER_RIGHT_0_TM=58.735
PRIMER_LEFT_0_GC_PERCENT=60.000
PRIMER_RIGHT_0_GC_PERCENT=47.619
PRIMER_LEFT_0_SELF_ANY_TH=0.00
PRIMER_RIGHT_0_SELF_ANY_TH=0.00
PRIMER_LEFT_0_SELF_END_TH=0.00
PRIMER_RIGHT_0_SELF_END_TH=0.00
PRIMER_LEFT_0_HAIRPIN_TH=0.00
PRIMER_RIGHT_0_HAIRPIN_TH=34.07
PRIMER_LEFT_0_END_STABILITY=5.2800
PRIMER_RIGHT_0_END_STABILITY=3.2700
PRIMER_PAIR_0_COMPL_ANY_TH=0.00
PRIMER_PAIR_0_COMPL_END_TH=0.00
PRIMER_PAIR_0_PRODUCT_SIZE=114
PRIMER_PAIR_0_PRODUCT_TM=81.9
PRIMER_PAIR_1_PENALTY=2.583642
PRIMER_LEFT_1_PENALTY=0.318275
PRIMER_RIGHT_1_PENALTY=2.265367
PRIMER_LEFT_1_SEQUENCE=CCGAGCAGACGAGGCTAATC
PRIMER_RIGHT_1_SEQUENCE=TGAGGCACATTAAGGGTCTGA
PRIMER_LEFT_1=1023,20
PRIMER_RIGHT_1=1141,21
PRIMER_LEFT_1_TM=60.318
PRIMER_RIGHT_1_TM=58.735
PRIMER_LEFT_1_GC_PERCENT=60.000
PRIMER_RIGHT_1_GC_PERCENT=47.619
PRIMER_LEFT_1_SELF_ANY_TH=0.00
PRIMER_RIGHT_1_SELF_ANY_TH=0.00
PRIMER_LEFT_1_SELF_END_TH=0.00
PRIMER_RIGHT_1_SELF_END_TH=0.00
PRIMER_LEFT_1_HAIRPIN_TH=42.78
PRIMER_RIGHT_1_HAIRPIN_TH=34.07
PRIMER_LEFT_1_END_STABILITY=1.7500
PRIMER_RIGHT_1_END_STABILITY=3.2700
PRIMER_PAIR_1_COMPL_ANY_TH=0.07
PRIMER_PAIR_1_COMPL_END_TH=0.00
PRIMER_PAIR_1_PRODUCT_SIZE=119
PRIMER_PAIR_1_PRODUCT_TM=82.7
PRIMER_PAIR_2_PENALTY=2.647525
PRIMER_LEFT_2_PENALTY=0.382158
PRIMER_RIGHT_2_PENALTY=2.265367
PRIMER_LEFT_2_SEQUENCE=CGAGCAGACGAGGCTAATCA
PRIMER_RIGHT_2_SEQUENCE=TGAGGCACATTAAGGGTCTGA
PRIMER_LEFT_2=1024,20
PRIMER_RIGHT_2=1141,21
PRIMER_LEFT_2_TM=59.618
PRIMER_RIGHT_2_TM=58.735
PRIMER_LEFT_2_GC_PERCENT=55.000
PRIMER_RIGHT_2_GC_PERCENT=47.619
PRIMER_LEFT_2_SELF_ANY_TH=0.00
PRIMER_RIGHT_2_SELF_ANY_TH=0.00
PRIMER_LEFT_2_SELF_END_TH=0.00
PRIMER_RIGHT_2_SELF_END_TH=0.00
PRIMER_LEFT_2_HAIRPIN_TH=42.78
PRIMER_RIGHT_2_HAIRPIN_TH=34.07
PRIMER_LEFT_2_END_STABILITY=2.5700
PRIMER_RIGHT_2_END_STABILITY=3.2700
PRIMER_PAIR_2_COMPL_ANY_TH=0.07
PRIMER_PAIR_2_COMPL_END_TH=0.00
PRIMER_PAIR_2_PRODUCT_SIZE=118
PRIMER_PAIR_2_PRODUCT_TM=82.5
=</code></pre>
<p>Next, we’ll test them against the entire genome to assess specificity.</p>
</section>
</section>
</section>
</section>
</section>
<section id="split-genome-fasta" class="level1">
<h1>6 SPLIT GENOME FASTA</h1>
<p>For some reason, I feel like the EMBOSS PrimerSearch tool will <em>not</em> work on mulit-FastA files (or, it makes the results more difficult to decipher?), but I’m just going off of memory and repeating a process I’ve done previously.</p>
<p>This will split the genome multi-FastA file into separate FastA files.</p>
<section id="split-mulit-fasta-file-in-to-individual-fasta-files-with-pyfaidx" class="level2">
<h2 class="anchored" data-anchor-id="split-mulit-fasta-file-in-to-individual-fasta-files-with-pyfaidx">6.1 Split mulit-FastA file in to individual FastA files with PyFaidx</h2>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Make directory if it doesn't exist</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">--parents</span> <span class="va">${genome_fasta_splits_dir}</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${genome_fasta_splits_dir}</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Count sequences in FastA</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------------------------------------------"</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"NUMBER OF SEQUENCES IN ORIGINAL FASTA"</span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="st">"^&gt;"</span> ../<span class="va">${genome_fasta}</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------------------------------------------"</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Split FastA</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a><span class="va">${pyfaidx}</span> <span class="dt">\</span></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>--split-files <span class="dt">\</span></span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>../<span class="va">${genome_fasta}</span></span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Count number of individual FastA files</span></span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------------------------------------------"</span></span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"NUMBER OF INDIVIDUAL FASTA FILES"</span></span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> <span class="pp">*</span>.fna <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------------------------------------------"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>-------------------------------------------------------------------
NUMBER OF SEQUENCES IN ORIGINAL FASTA
29
-------------------------------------------------------------------


-------------------------------------------------------------------
NUMBER OF INDIVIDUAL FASTA FILES
29
-------------------------------------------------------------------</code></pre>
</section>
</section>
<section id="primer-search-with-emboss-primersearch" class="level1">
<h1>7 PRIMER SEARCH WITH <a href="https://emboss.sourceforge.net/apps/cvs/emboss/apps/primersearch.html">EMBOSS PRIMERSEARCH</a></h1>
<p>This will run <a href="https://emboss.sourceforge.net/apps/cvs/emboss/apps/primersearch.html">EMBOSS PrimerSearch</a> against the genome to assess primer specificity.</p>
<section id="create-emboss-primersearch-primers-file" class="level2">
<h2 class="anchored" data-anchor-id="create-emboss-primersearch-primers-file">7.1 Create EMBOSS PrimerSearch Primers File</h2>
<p>Create a tab-delimited file to use with EMBOSS PrimerSearch.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove if exists.</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Prevents appending to output file repeatedly</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="bu">[</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/emboss-primers.txt"</span> <span class="bu">]</span><span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/emboss-primers.txt"</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="cf">fi</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> primer_file <span class="kw">in</span> <span class="va">${output_top}</span>/<span class="pp">*</span>primer3-primers-default-format.txt</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  <span class="va">seq_id</span><span class="op">=</span><span class="va">$(</span><span class="fu">grep</span> <span class="st">"SEQUENCE_ID="</span> <span class="st">"</span><span class="va">${primer_file}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/SEQUENCE_ID=//'</span><span class="va">)</span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>  <span class="va">left_primer</span><span class="op">=</span><span class="va">$(</span><span class="fu">grep</span> <span class="st">"PRIMER_LEFT_0_SEQUENCE="</span> <span class="st">"</span><span class="va">${primer_file}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/PRIMER_LEFT_0_SEQUENCE=//'</span> <span class="dt">\</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">tr</span> <span class="st">'[:lower:]'</span> <span class="st">'[:upper:]'</span><span class="va">)</span></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>  <span class="va">right_primer</span><span class="op">=</span><span class="va">$(</span><span class="fu">grep</span> <span class="st">"PRIMER_RIGHT_0_SEQUENCE="</span> <span class="st">"</span><span class="va">${primer_file}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/PRIMER_RIGHT_0_SEQUENCE=//'</span> <span class="dt">\</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">tr</span> <span class="st">'[:lower:]'</span> <span class="st">'[:upper:]'</span><span class="va">)</span></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create EMBOSS primer file</span></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>  <span class="bu">printf</span> <span class="st">"%s\t%s\t%s\t\n"</span> <span class="st">"</span><span class="va">${seq_id}</span><span class="st">"</span> <span class="st">"</span><span class="va">${left_primer}</span><span class="st">"</span> <span class="st">"</span><span class="va">${right_primer}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">tee</span> <span class="at">--append</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/emboss-primers.txt"</span></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>NM_001305360.1  TCTCACTGTGGAAATGCACA    TGGACATTCATCTCTACTGCT   
XM_034462847.2  TCGATCACTTCAGAATTTCACAGG    CCATTGGGGGAGGGTTCTTC    
XM_066083499.1  CAGACGAGGCTAATCACCGG    TGAGGCACATTAAGGGTCTGA   </code></pre>
</section>
<section id="run-emboss-primersearch" class="level2">
<h2 class="anchored" data-anchor-id="run-emboss-primersearch">7.2 Run EMBOSS PrimerSearch</h2>
<p>This will run EMBOSS PrimerSearch and allow for a 10% mismatch in primer annealing sites (<code>${primersearch} -auto ${fasta} ${primers} 10</code>).</p>
<p>Afterwards, the resulting output files (<code>*.primersearch</code>) will be searched for the term <code>Amplimer</code>, indicating a PCR product would be produced. If no amplimer is identified in a <code>*.primersearch</code> file, then that file is deleted. This should leave just the results in which primer matches were identified.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${genome_fasta_splits_dir}</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="va">primers</span><span class="op">=</span><span class="st">"../../output/emboss-primers.txt"</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a><span class="bu">time</span> <span class="dt">\</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>for fasta in <span class="pp">*</span>.fna</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">do</span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove path from FastA filename</span></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>  <span class="va">fasta_no_path</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">${fasta</span><span class="op">##</span><span class="pp">*</span>/<span class="va">})</span></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove file extension from FastA filename</span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>  <span class="va">fasta_no_ext</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">${fasta_no_path</span><span class="op">%%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert filename to lowercase</span></span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Will be used for output from EMBOSS PrimerSearch</span></span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>  <span class="va">fasta_no_ext_lower</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">${fasta_no_ext}</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">'[:upper:]'</span> <span class="st">'[:lower:]'</span><span class="va">)</span></span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a>  <span class="co">###### Run EMBOSS PrimerSearch on all FastA files ########</span></span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Allows for a 10% mismatch</span></span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a>  <span class="va">${primersearch}</span> <span class="at">-auto</span> <span class="va">${fasta}</span> <span class="va">${primers}</span> 10</span>
<span id="cb31-25"><a href="#cb31-25" aria-hidden="true" tabindex="-1"></a>  <span class="co">##### </span><span class="re">END</span><span class="co"> EMBOSS ##########</span></span>
<span id="cb31-26"><a href="#cb31-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-27"><a href="#cb31-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find EMBOSS PrimerSearch output files with primer matches</span></span>
<span id="cb31-28"><a href="#cb31-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove those without a match</span></span>
<span id="cb31-29"><a href="#cb31-29" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> <span class="ot">! </span><span class="fu">grep</span> <span class="at">--quiet</span> <span class="st">"Amplimer"</span> <span class="st">"</span><span class="va">${fasta_no_ext_lower}</span><span class="st">.primersearch"</span></span>
<span id="cb31-30"><a href="#cb31-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">then</span> <span class="fu">rm</span> <span class="va">${fasta_no_ext_lower}</span>.primersearch</span>
<span id="cb31-31"><a href="#cb31-31" aria-hidden="true" tabindex="-1"></a>  <span class="cf">fi</span></span>
<span id="cb31-32"><a href="#cb31-32" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>real    1m8.469s
user    1m5.269s
sys 0m3.411s</code></pre>
</section>
<section id="check-primer-matches" class="level2">
<h2 class="anchored" data-anchor-id="check-primer-matches">7.3 Check primer matches</h2>
<p>This will print the contents of any remaining <code>*.primersearch</code> output files (i.e.&nbsp;sequences with primer matches)</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${genome_fasta_splits_dir}</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Check contents of files with matches</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.primersearch</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">do</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"FILE: </span><span class="va">${file}</span><span class="st">"</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span> <span class="va">${file}</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"----------------------------------"</span></span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>FILE: nc_088857.primersearch


Primer name NM_001305360.1
Amplimer 1
    Sequence: NC_088857.1  
    
    TGGACATTCATCTCTACTGCT hits forward strand at 40310307 with 0 mismatches
    TCTCACTGTGGAAATGCACA hits reverse strand at [16964538] with 0 mismatches
    Amplimer length: 83 bp

Primer name XM_034462847.2

Primer name XM_066083499.1
Amplimer 1
    Sequence: NC_088857.1  
    
    TGAGGCACATTAAGGGTCTGA hits forward strand at 40310581 with 0 mismatches
    CAGACGAGGCTAATCACCGG hits reverse strand at [16964233] with 0 mismatches
    Amplimer length: 114 bp
----------------------------------</code></pre>
</section>
</section>
<section id="summary" class="level1">
<h1>8 SUMMARY</h1>
<p>EMBOSS PrimerSearch has identified a single amplicon on chromosome <code>NC_088857.1</code> for <code>NM_001305360.1</code> and <code>XM_066083499.1</code>, and each predicted amplicon matches the lengths generated by Primer3. However, no amplicon is predicted for <code>XM_034462847.2</code>. This could possibly be explained by what appears to be an insertion of ~10bp in this variant.</p>
<p>Anyway, I’ll get these ordered and tested out!</p>
</section>
<section id="citations" class="level1">
<h1>9 CITATIONS</h1>
<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-kans2013entrez" class="csl-entry" role="listitem">
Kans, James. 2013. <em>Entrez Direct: E-Utilities on the Unix Command Line</em>. Bethesda (MD): National Center for Biotechnology Information (US). <a href="https://www.ncbi.nlm.nih.gov/books/NBK179288/" class="uri">https://www.ncbi.nlm.nih.gov/books/NBK179288/</a>.
</div>
<div id="ref-koressaar2007" class="csl-entry" role="listitem">
Koressaar, Triinu, and Maido Remm. 2007. “Enhancements and Modifications of Primer Design Program Primer3.” <em>Bioinformatics</em> 23 (10): 1289–91. <a href="https://doi.org/10.1093/bioinformatics/btm091" class="uri">https://doi.org/10.1093/bioinformatics/btm091</a>.
</div>
<div id="ref-rice2000" class="csl-entry" role="listitem">
Rice, Peter, Ian Longden, and Alan Bleasby. 2000. “EMBOSS: The European Molecular Biology Open Software Suite.” <em>Trends in Genetics</em> 16 (6): 276–77. <a href="https://doi.org/10.1016/s0168-9525(00)02024-2" class="uri">https://doi.org/10.1016/s0168-9525(00)02024-2</a>.
</div>
<div id="ref-shirley2015" class="csl-entry" role="listitem">
Shirley, Matthew D, Zhaorong Ma, Brent S Pedersen, and Sarah J Wheelan. 2015. “Efficient”Pythonic” Access to FASTA Files Using Pyfaidx.” <a href="http://dx.doi.org/10.7287/peerj.preprints.970v1" class="uri">http://dx.doi.org/10.7287/peerj.preprints.970v1</a>.
</div>
<div id="ref-untergasser2012" class="csl-entry" role="listitem">
Untergasser, Andreas, Ioana Cutcutache, Triinu Koressaar, Jian Ye, Brant C. Faircloth, Maido Remm, and Steven G. Rozen. 2012. “Primer3new Capabilities and Interfaces.” <em>Nucleic Acids Research</em> 40 (15): e115–15. <a href="https://doi.org/10.1093/nar/gks596" class="uri">https://doi.org/10.1093/nar/gks596</a>.
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/robertslab\.github\.io\/sams-notebook");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>