---
author: Sam White
toc-title: Contents
toc-depth: 5
toc-location: left
title: Daily Bits - February 2024
date: '2024-02-01'
draft: false
engine: knitr
categories: 
  - 2024
  - Daily Bits
bibliography: references.bib
---

## 20240227

### Miscellaneous

- [Setup account on Owl for Ariana.](https://github.com/RobertsLab/resources/issues/1807) GitHub Issue

- [Set up access to nightingales Google Sheet for Ariana.](https://github.com/RobertsLab/resources/issues/1808) GitHub Issue

### [CEABIGR](https://github.com/sr320/ceabigr) (GitHub repo)

- Generated table with mean FPKM values for genes in female/male control/exposed.

## 20240226

### In lab

- Selected and prepared primers for Eric's qPCRs.

- Helped Eric conduct qPCRs.

- Helped Katherine gather sampling supplies for hatchery.

- Ran all over the place in Bagley Hall to get LN<sub>2</sub>! Chemistry Stores was out and so was secondary location in Rm. 305A. Eventually managed to get LN<sub>2</sub> in 405A...

### [CEABIGR](https://github.com/sr320/ceabigr) (GitHub repo)

- Generated canonical predominant isoform table.

## 20240223

### [Coral E5](https://github.com/urol-e5/deep-dive) (GitHub repo)

- All hands meeting.

- Read original proposal.

- Read paper about plant epigenetic stress responses.

### Science Hour

### [CEABIGR](https://github.com/sr320/ceabigr) (GitHub repo)

- Created canonical max transcripts per gene table for females controls/exposed and males controls/exposed.

## 20240222

### Lab Meeting

- Discussed Pub-a-thon 2024

### [CEABIGR](https://github.com/sr320/ceabigr) (GitHub repo)

- Reviewed recent GitHub Issues

- Worked on creating canonical gene/transcript counts table.

### Miscellaneous

- Updated [`Nightingales`](https://b.link/nightingales) (Google Sheet) spreadsheet with Cod RNA-seq info.

- Raven was bogged down by some weird rogue `git` command, so had to reboot.

## 20240221

### Miscellaneous

- [Set Ariana up with Gannet access](https://github.com/RobertsLab/resources/issues/1806) (GitHub Issue)

- Received [Pacific cod RNA-seq data](../2024-02-21-Data-Received---Pacific-cod-RNA-seq-Azenta-Project-30-943133806/index.qmd)

### [CEABIGR](https://github.com/sr320/ceabigr) (GitHub repo)

- Generated volcano plots

::: {layout-nrow="2"}
![Volcano plot of gene expression in females. No genes are differentially expressed. Grey points are genes with p-values <=0.05. Black points are genes with p/q-values > 0.05.](https://raw.githubusercontent.com/sr320/ceabigr/main/output/18-differential-gene-expression-ballgown/figures/volcano_plot-DEGs-females-CvE.png){fig-alt="Volcano plot of gene expression in females. No genes are differentially expressed. Grey points are genes with p-values <=0.05. Black points are genes with p/q-values > 0.05."}

![Volcano plot of transcript expression in females. A single transcript is differentially expressed and is rendered as a blue point and labelled with its associated geneID. Grey points are transcripts with p-values <=0.05. Black points are transcripts with p/q-values > 0.05.](https://raw.githubusercontent.com/sr320/ceabigr/main/output/18-differential-gene-expression-ballgown/figures/volcano_plot-DETs-females-CvE.png){fig-alt="Volcano plot of transcript expression in females. A single transcript is differentially expressed and is rendered as a blue point and labelled with its associated geneID. Grey points are transcripts with p-values <=0.05. Black points are transcripts with p/q-values > 0.05."}

![Volcano plot of gene expression in males. No genes are differentially expressed. Grey points are genes with p-values <=0.05. Black points are genes with p/q-values > 0.05.](https://raw.githubusercontent.com/sr320/ceabigr/main/output/18-differential-gene-expression-ballgown/figures/volcano_plot-DEGs-males-CvE.png){fig-alt="Volcano plot of gene expression in males. No genes are differentially expressed. Grey points are genes with p-values <=0.05. Black points are genes with p/q-values > 0.05."}

![Volcano plot of transcript expression in males. No transcripts are differentially expressed. Grey points are transcripts with p-values <=0.05. Black points are transcripts with p/q-values > 0.05](https://raw.githubusercontent.com/sr320/ceabigr/main/output/18-differential-gene-expression-ballgown/figures/volcano_plot-DETs-males-CvE.png){fig-alt="Volcano plot of transcript expression in males. No transcripts are differentially expressed. Grey points are transcripts with p-values <=0.05. Black points are transcripts with p/q-values > 0.05."}
:::
## 20240220

### [CEABIGR](https://github.com/sr320/ceabigr) (GitHub repo)

- Started working on generating volcano plots.

## 20240216

### [Coral E5](https://github.com/urol-e5/deep-dive) (GitHub repo)

- Finished running [`ShortStack`](https://github.com/MikeAxtell/ShortStack) variations with newly trimmed _P.meandrina_ sRNA-seq.

## 20240215

### [Coral E5](https://github.com/urol-e5/deep-dive) (GitHub repo)

- Ran [`ShortStack`](https://github.com/MikeAxtell/ShortStack) variations with newly trimmed _P.meandrina_ sRNA-seq.

## 20240214

### [Coral E5](https://github.com/urol-e5/deep-dive) (GitHub repo)

- Ran [`ShortStack`](https://github.com/MikeAxtell/ShortStack) variations with newly trimmed [_Acropora pulchra_](http://www.coralsoftheworld.org/species_factsheets/species_factsheet_summary/acropora-pulchra/) sRNA-seq.

## 20240213

### [Coral E5](https://github.com/urol-e5/deep-dive) (GitHub repo)

- Cranked out new trimming on _P.meandrina_ sRNA-seq samples.

## 20240212

## In lab

- Reviewed reverse transcription and qPCR with Eric.

### [Coral E5](https://github.com/urol-e5/deep-dive) (GitHub repo)

- Started running new trimming on _A.pulchra_ sRNA-seq samples (preivously, had just tried with _P.evermanni_).

## 20240209

## Science Hour

- Steven and I looked at and discussed the exon expression patterns of individual genes and compared those patterns to the WGCNA outputs which Ariana generated at yesterday's meeting. Definitely a few individuals whose exon usage seems to deviate a great deal from that WGCNA module assignment...

### [Coral E5](https://github.com/urol-e5/deep-dive) (GitHub repo)

- Continued to work on implementing new trimming/merging parameters, but [ShortStack](https://github.com/MikeAxtell/ShortStack/tree/master) keeps encountering an [error](https://github.com/MikeAxtell/ShortStack/issues/146) (GitHub Issue). I'm hoping the developer can lend some insight. This is weird, because these samples, albeit trimmed with a different program and used as paired-end reads (only trying R1 or merged reads at this point), were successfully analyzed previously...

## 20240208

### Lab meeting

### [CEABIGR](https://github.com/sr320/ceabigr) (GitHub repo)

- Special, extended session with Arianna in attendance. She ran some [WGCNA](https://cran.r-project.org/web/packages/WGCNA/WGCNA.pdf) (PDF) analysis to try to see how exon expression patterns varied between control/exposed samples. Very cool!

## 20240207

### [Coral E5](https://github.com/urol-e5/deep-dive) (GitHub repo)

- Finished messing around with trimming/merging stuff. Here's the [notebook entry](https://robertslab.github.io/sams-notebook/posts/2024/2024-02-02-sRNA-seq-Paired-Read-Merging-Exploratios---Coral-E5-Deep-Dive/).

- Began implementing new trimming/merging.

## 20240206

### [Coral E5](https://github.com/urol-e5/deep-dive) (GitHub repo)

- Continued to explore trimming and merging strategies for sRNA-seq data, using [`fastp`](https://github.com/OpenGene/fastp) alone, BBMerge alone, and/or combining the two approaches.

### [CEABIGR](https://github.com/sr320/ceabigr) (GitHub repo)

- Spent a few hours on a video chat with Steven and Yaamini going through exon fold-change code/analysis.

## 20240205

### [Coral E5](https://github.com/urol-e5/deep-dive) (GitHub repo)

- Continued to explore trimming and merging strategies for sRNA-seq data, using [`fastp`](https://github.com/OpenGene/fastp) alone, BBMerge alone, and/or combining the two approaches.

## 20240202

### [Coral E5](https://github.com/urol-e5/deep-dive) (GitHub repo)

### Science Hour

- Steven and I played around with exon fold-change data.

## 20240201

### In Lab

-   Lab meeting.

-   Helped Eric get started with RNA quantification using Qubit.

### [Coral E5](https://github.com/urol-e5/deep-dive) (GitHub repo)

- Tested out more R1/R2 merging/trimming using [`fastp`](https://github.com/OpenGene/fastp).

  - Merging is similar to BBMerge results - i.e. a (relatively) small number of reads are actually merging, despite the fact that these should be short insert sizes and R1/R2 reads should overlap...

-   Installed [XICRA](https://github.com/HCGB-IGTP/XICRA) [@sanchezherrero2021] on Raven to identify isomiRs in sRNA-seq data. Possibly to get around the merging "issues" experienced with BBMerge and [`fastp`](https://github.com/OpenGene/fastp).

    -   Installation was a bit bumpy, due to some weird Conda/Mamba "flexible solve" problem, but got that straightened out with `conda config --set channel_priority flexible`.

    -   Installation test(s) had issues. I created the following GitHub Issues on the developer's site in hopes of getting help:

      - [`installer.sh error`](https://github.com/HCGB-IGTP/XICRA/issues/47)

      - ["No samples were retrieved" in test data](https://github.com/HCGB-IGTP/XICRA/issues/42)


### [CEABIGR](https://github.com/sr320/ceabigr) (GitHub repo)

- Need to generate R<sup>2</sup> values and slope of exon expression for all genes, Exons 1 - 6.