<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.554">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sam White">
<meta name="dcterms.date" content="2024-07-29">

<title>Sam’s Notebook - Primer Design - Lake Trout C1q Gene Bisulfite Sequencing Primers Using Primer3</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 2,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 100,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-7483T61T11"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-7483T61T11', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Sam’s Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" target="_blank">
 <span class="dropdown-text">Rstudio Server - Raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu/Atumefaciens/" target="_blank">
 <span class="dropdown-text">My Gannet Directory</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" target="_blank">
 <span class="dropdown-text">Steven’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shedurkin.github.io/Roberts-LabNotebook/" target="_blank">
 <span class="dropdown-text">Kathleen’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://meganewing.github.io/mewing-notebook/" target="_blank">
 <span class="dropdown-text">Megan’s Notebook</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/sams-notebook" target="_blank">
 <span class="dropdown-text">Notebook repo</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" target="_blank">
 <span class="dropdown-text">Lab Organization (GitHub)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" target="_blank">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" target="_blank"> <i class="bi bi-calendar-week" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-right">
      <h1 class="title">Primer Design - Lake Trout C1q Gene Bisulfite Sequencing Primers Using Primer3</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2024</div>
                <div class="quarto-category">C1q</div>
                <div class="quarto-category">lake trout</div>
                <div class="quarto-category">Salvelinus namaycush</div>
                <div class="quarto-category">Primer3</div>
                <div class="quarto-category">primer design</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-right">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Sam White </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">July 29, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#create-bash-variables-file" id="toc-create-bash-variables-file" class="nav-link active" data-scroll-target="#create-bash-variables-file">1 CREATE BASH VARIABLES FILE</a></li>
  <li><a href="#download-ncbi-genome-files" id="toc-download-ncbi-genome-files" class="nav-link" data-scroll-target="#download-ncbi-genome-files">2 DOWNLOAD NCBI GENOME FILES</a>
  <ul>
  <li><a href="#download-the-actual-files" id="toc-download-the-actual-files" class="nav-link" data-scroll-target="#download-the-actual-files">2.1 Download the actual files</a></li>
  <li><a href="#check-md5-checkums" id="toc-check-md5-checkums" class="nav-link" data-scroll-target="#check-md5-checkums">2.2 Check MD5 Checkums</a></li>
  <li><a href="#decompress-ncbi-files" id="toc-decompress-ncbi-files" class="nav-link" data-scroll-target="#decompress-ncbi-files">2.3 Decompress NCBI files</a></li>
  </ul></li>
  <li><a href="#extract-c1q-gene-sequence" id="toc-extract-c1q-gene-sequence" class="nav-link" data-scroll-target="#extract-c1q-gene-sequence">3 EXTRACT C1Q GENE SEQUENCE</a>
  <ul>
  <li><a href="#peek-at-gff" id="toc-peek-at-gff" class="nav-link" data-scroll-target="#peek-at-gff">3.1 Peek at GFF</a></li>
  <li><a href="#format-region-and-extract-c1q-sequence-as-fasta." id="toc-format-region-and-extract-c1q-sequence-as-fasta." class="nav-link" data-scroll-target="#format-region-and-extract-c1q-sequence-as-fasta.">3.2 Format region and extract C1q sequence as FastA.</a></li>
  </ul></li>
  <li><a href="#extract-c1q-with-53-buffer-regions" id="toc-extract-c1q-with-53-buffer-regions" class="nav-link" data-scroll-target="#extract-c1q-with-53-buffer-regions">4 EXTRACT C1Q WITH 5’/3’ BUFFER REGIONS</a></li>
  <li><a href="#primer-design-using-primer3" id="toc-primer-design-using-primer3" class="nav-link" data-scroll-target="#primer-design-using-primer3">5 PRIMER DESIGN USING PRIMER3</a>
  <ul>
  <li><a href="#bisulfite-conversions" id="toc-bisulfite-conversions" class="nav-link" data-scroll-target="#bisulfite-conversions">5.1 Bisulfite conversions</a></li>
  <li><a href="#design-primers" id="toc-design-primers" class="nav-link" data-scroll-target="#design-primers">5.2 Design primers</a>
  <ul>
  <li><a href="#bisulfite-lowercase-t" id="toc-bisulfite-lowercase-t" class="nav-link" data-scroll-target="#bisulfite-lowercase-t">5.2.1 Bisulfite lowercase <code>t</code></a></li>
  </ul></li>
  </ul></li>
  <li><a href="#genome-bisulfite-conversion" id="toc-genome-bisulfite-conversion" class="nav-link" data-scroll-target="#genome-bisulfite-conversion">6 GENOME BISULFITE CONVERSION</a></li>
  <li><a href="#split-genome" id="toc-split-genome" class="nav-link" data-scroll-target="#split-genome">7 SPLIT GENOME</a>
  <ul>
  <li><a href="#split-mulit-fasta-file-in-to-individual-fasta-files-with-pyfaidx" id="toc-split-mulit-fasta-file-in-to-individual-fasta-files-with-pyfaidx" class="nav-link" data-scroll-target="#split-mulit-fasta-file-in-to-individual-fasta-files-with-pyfaidx">7.1 Split mulit-FastA file in to individual FastA files with PyFaidx</a></li>
  </ul></li>
  <li><a href="#primer-search-with-emboss-primersearch" id="toc-primer-search-with-emboss-primersearch" class="nav-link" data-scroll-target="#primer-search-with-emboss-primersearch">8 PRIMER SEARCH WITH EMBOSS PRIMERSEARCH</a>
  <ul>
  <li><a href="#create-emboss-primersearch-primers-file" id="toc-create-emboss-primersearch-primers-file" class="nav-link" data-scroll-target="#create-emboss-primersearch-primers-file">8.1 Create EMBOSS PrimerSearch Primers File</a></li>
  <li><a href="#run-emboss-primersearch" id="toc-run-emboss-primersearch" class="nav-link" data-scroll-target="#run-emboss-primersearch">8.2 Run EMBOSS PrimerSearch</a></li>
  <li><a href="#check-primer-matches" id="toc-check-primer-matches" class="nav-link" data-scroll-target="#check-primer-matches">8.3 Check primer matches</a></li>
  </ul></li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary">9 SUMMARY</a></li>
  <li><a href="#citations" id="toc-citations" class="nav-link" data-scroll-target="#citations">10 CITATIONS</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-right" id="quarto-document-content">





<p>This notebook describes using <a href="https://github.com/primer3-org/primer3">Primer3</a> (<a href="#ref-untergasser2012">Untergasser et al. 2012</a>; <a href="#ref-koressaar2007">Koressaar and Remm 2007</a>) to reproducibly design primers for sequencing of the <a href="https://en.wikipedia.org/wiki/Lake_trout">lake trout (<em>S.namaycush</em>)</a> (Wikipedia) C1q gene (<a href="https://github.com/RobertsLab/resources/issues/1941#issuecomment-2246548960"><code>LOC120027825</code></a> (GitHub Issue)), after bisulfite treatment.</p>
<p>This process also utilizes <a href="https://github.com/mdshw5/pyfaidx">pyfaidx</a> (<a href="#ref-shirley2015">Shirley et al.&nbsp;2015</a>).</p>
<p><a href="https://emboss.sourceforge.net/apps/cvs/emboss/apps/primersearch.html">EMBOSS PrimerSearch</a> (<a href="#ref-rice2000">Rice, Longden, and Bleasby 2000</a>) will be utilized to assess primer specificity across the genome.</p>
<p>Markdown from this post was rendered from the R Markdown file, <a href="https://github.com/RobertsLab/code/blob/739ea5953e7aef1453c8071a06fe05ce5d824ee0/r_projects/sam/20240726-snam-c1q-bs-primer-design/code/20240726-snam-c1q-bs-primer-design.Rmd">20240726-snam-c1q-bs-primer-design.Rmd</a> (GitHub), commit <code>739ea59</code>.</p>
<ul>
<li><a href="#1-create-bash-variables-file" id="toc-1-create-bash-variables-file">1 CREATE BASH VARIABLES FILE</a></li>
<li><a href="#2-download-ncbi-genome-files" id="toc-2-download-ncbi-genome-files">2 DOWNLOAD NCBI GENOME FILES</a>
<ul>
<li><a href="#21-download-the-actual-files" id="toc-21-download-the-actual-files">2.1 Download the actual files</a></li>
<li><a href="#22-check-md5-checkums" id="toc-22-check-md5-checkums">2.2 Check MD5 Checkums</a></li>
<li><a href="#23-decompress-ncbi-files" id="toc-23-decompress-ncbi-files">2.3 Decompress NCBI files</a></li>
</ul></li>
<li><a href="#3-extract-c1q-gene-sequence" id="toc-3-extract-c1q-gene-sequence">3 EXTRACT C1Q GENE SEQUENCE</a>
<ul>
<li><a href="#31-peek-at-gff" id="toc-31-peek-at-gff">3.1 Peek at GFF</a></li>
<li><a href="#32-format-region-and-extract-c1q-sequence-as-fasta" id="toc-32-format-region-and-extract-c1q-sequence-as-fasta">3.2 Format region and extract C1q sequence as FastA.</a></li>
</ul></li>
<li><a href="#4-extract-c1q-with-53-buffer-regions" id="toc-4-extract-c1q-with-53-buffer-regions">4 EXTRACT C1Q WITH 5’/3’ BUFFER REGIONS</a></li>
<li><a href="#5-primer-design-using-primer3" id="toc-5-primer-design-using-primer3">5 PRIMER DESIGN USING PRIMER3</a>
<ul>
<li><a href="#51-bisulfite-conversions" id="toc-51-bisulfite-conversions">5.1 Bisulfite conversions</a></li>
<li><a href="#52-design-primers" id="toc-52-design-primers">5.2 Design primers</a>
<ul>
<li><a href="#521-bisulfite-lowercase-t" id="toc-521-bisulfite-lowercase-t">5.2.1 Bisulfite lowercase <code>t</code></a></li>
</ul></li>
</ul></li>
<li><a href="#6-genome-bisulfite-conversion" id="toc-6-genome-bisulfite-conversion">6 GENOME BISULFITE CONVERSION</a></li>
<li><a href="#7-split-genome" id="toc-7-split-genome">7 SPLIT GENOME</a>
<ul>
<li><a href="#71-split-mulit-fasta-file-in-to-individual-fasta-files-with-pyfaidx" id="toc-71-split-mulit-fasta-file-in-to-individual-fasta-files-with-pyfaidx">7.1 Split mulit-FastA file in to individual FastA files with PyFaidx</a></li>
</ul></li>
<li><a href="#8-primer-search-with-emboss-primersearch" id="toc-8-primer-search-with-emboss-primersearch">8 PRIMER SEARCH WITH EMBOSS PRIMERSEARCH</a>
<ul>
<li><a href="#81-create-emboss-primersearch-primers-file" id="toc-81-create-emboss-primersearch-primers-file">8.1 Create EMBOSS PrimerSearch Primers File</a></li>
<li><a href="#82-run-emboss-primersearch" id="toc-82-run-emboss-primersearch">8.2 Run EMBOSS PrimerSearch</a></li>
<li><a href="#83-check-primer-matches" id="toc-83-check-primer-matches">8.3 Check primer matches</a></li>
</ul></li>
<li><a href="#9-summary" id="toc-9-summary">9 SUMMARY</a></li>
<li><a href="#10-citations" id="toc-10-citations">10 CITATIONS</a></li>
</ul>
<section id="create-bash-variables-file" class="level1">
<h1>1 CREATE BASH VARIABLES FILE</h1>
<p>This allows usage of Bash variables across R Markdown chunks.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"#### Assign Variables ####"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# DATA DIRECTORIES"</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export data_dir="../data/S_namaycush/genomes"'</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export output_top=../output'</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export genome_fasta_splits_dir="${data_dir}/fasta_splits"'</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# SEQUENCE"</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export sequence_ID="LOC120027825"'</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# SEQUENCE REGIONS"</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export left_buffer="3500"'</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export right_buffer="3500"'</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# INPUT FILES"</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export genome_fasta="GCF_016432855.1_SaNama_1.0_genomic.fna"'</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export genome_gff="GCF_016432855.1_SaNama_1.0_genomic.gff"'</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export ncbi_gff_gz="GCF_016432855.1_SaNama_1.0_genomic.gff.gz"'</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export ncbi_fasta_gz="GCF_016432855.1_SaNama_1.0_genomic.fna.gz"'</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export ncbi_md5sums="md5checksums.txt"'</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export ncbi_url="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/016/432/855/GCF_016432855.1_SaNama_1.0/"'</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# OUTPUT FILES"</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export bisulfite_genome_fasta="bisulfite.GCF_016432855.1_SaNama_1.0_genomic.fna"'</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export c1q_fasta="${sequence_ID}.fasta"'</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export c1q_buffer_fasta="${sequence_ID}-${left_buffer}bp.left-${right_buffer}bp.right.fasta"'</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export c1q_faidx_region_file="${sequence_ID}"-region.txt'</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export c1q_buffer_faidx_region_file="${sequence_ID}-${left_buffer}bp.left-${right_buffer}bp.right-region.txt"'</span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export c1q_bisulfite_t_fasta="${sequence_ID}-BS_conversion-t-${left_buffer}bp.left-${right_buffer}bp.right.fasta"'</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export c1q_bisulfite_N_fasta="${sequence_ID}-BS_conversion-N-${left_buffer}bp.left-${right_buffer}bp.right.fasta"'</span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# SET CPUS"</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export threads=40'</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# PROGRAMS"</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export pyfaidx=/home/shared/pyfaidx-0.8.1.1'</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export primer3_dir="/home/shared/primer3-2.6.1/src"'</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export primer3="${primer3_dir}/primer3_core"'</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export primer3_config="${primer3_dir}/primer3_config"'</span></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export primersearch="/home/shared/EMBOSS-6.6.0/emboss/primersearch"'</span></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="kw">}</span> <span class="op">&gt;</span> .bashvars</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> .bashvars</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>#### Assign Variables ####

# DATA DIRECTORIES
export data_dir="../data/S_namaycush/genomes"
export output_top=../output
export genome_fasta_splits_dir="${data_dir}/fasta_splits"

# SEQUENCE
export sequence_ID="LOC120027825"

# SEQUENCE REGIONS
export left_buffer="3500"
export right_buffer="3500"

# INPUT FILES
export genome_fasta="GCF_016432855.1_SaNama_1.0_genomic.fna"
export genome_gff="GCF_016432855.1_SaNama_1.0_genomic.gff"
export ncbi_gff_gz="GCF_016432855.1_SaNama_1.0_genomic.gff.gz"
export ncbi_fasta_gz="GCF_016432855.1_SaNama_1.0_genomic.fna.gz"
export ncbi_md5sums="md5checksums.txt"
export ncbi_url="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/016/432/855/GCF_016432855.1_SaNama_1.0/"

# OUTPUT FILES
export bisulfite_genome_fasta="bisulfite.GCF_016432855.1_SaNama_1.0_genomic.fna"
export c1q_fasta="${sequence_ID}.fasta"
export c1q_buffer_fasta="${sequence_ID}-${left_buffer}bp.left-${right_buffer}bp.right.fasta"
export c1q_faidx_region_file="${sequence_ID}"-region.txt
export c1q_buffer_faidx_region_file="${sequence_ID}-${left_buffer}bp.left-${right_buffer}bp.right-region.txt"
export c1q_bisulfite_t_fasta="${sequence_ID}-BS_conversion-t-${left_buffer}bp.left-${right_buffer}bp.right.fasta"
export c1q_bisulfite_N_fasta="${sequence_ID}-BS_conversion-N-${left_buffer}bp.left-${right_buffer}bp.right.fasta"

# SET CPUS
export threads=40

# PROGRAMS
export pyfaidx=/home/shared/pyfaidx-0.8.1.1
export primer3_dir="/home/shared/primer3-2.6.1/src"
export primer3="${primer3_dir}/primer3_core"
export primer3_config="${primer3_dir}/primer3_config"
export primersearch="/home/shared/EMBOSS-6.6.0/emboss/primersearch"</code></pre>
</section>
<section id="download-ncbi-genome-files" class="level1">
<h1>2 DOWNLOAD NCBI GENOME FILES</h1>
<section id="download-the-actual-files" class="level2">
<h2 class="anchored" data-anchor-id="download-the-actual-files">2.1 Download the actual files</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="va">${ncbi_gff_gz}</span> <span class="va">${ncbi_fasta_gz}</span> <span class="va">${ncbi_md5sums}</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">wget</span> <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">--no-check-certificate</span> <span class="dt">\</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">--continue</span> <span class="dt">\</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">--quiet</span> <span class="dt">\</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">--directory-prefix</span><span class="op">=</span><span class="va">${data_dir}</span> <span class="dt">\</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="va">${ncbi_url}${file}</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-lh</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>total 660M
-rw-r--r-- 1 sam sam 642M Jan 13  2021 GCF_016432855.1_SaNama_1.0_genomic.fna.gz
-rw-r--r-- 1 sam sam  18M Sep 28  2022 GCF_016432855.1_SaNama_1.0_genomic.gff.gz
-rw-r--r-- 1 sam sam  33K Jul 26 06:11 md5checksums.txt</code></pre>
</section>
<section id="check-md5-checkums" class="level2">
<h2 class="anchored" data-anchor-id="check-md5-checkums">2.2 Check MD5 Checkums</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">"</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.gz</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span> <span class="st">"</span><span class="va">${file}</span><span class="st">"</span> <span class="va">${ncbi_md5sums}</span> <span class="kw">|</span> <span class="fu">md5sum</span> <span class="at">-c</span> <span class="at">-</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>./GCF_016432855.1_SaNama_1.0_genomic.fna.gz: OK
./GCF_016432855.1_SaNama_1.0_genomic.gff.gz: OK</code></pre>
</section>
<section id="decompress-ncbi-files" class="level2">
<h2 class="anchored" data-anchor-id="decompress-ncbi-files">2.3 Decompress NCBI files</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">"</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.gz</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gunzip</span> <span class="st">"</span><span class="va">${file}</span><span class="st">"</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-lh</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>total 2.6G
-rw-r--r-- 1 sam sam 2.3G Jan 13  2021 GCF_016432855.1_SaNama_1.0_genomic.fna
-rw-r--r-- 1 sam sam 373M Sep 28  2022 GCF_016432855.1_SaNama_1.0_genomic.gff
-rw-r--r-- 1 sam sam  33K Jul 26 06:11 md5checksums.txt</code></pre>
</section>
</section>
<section id="extract-c1q-gene-sequence" class="level1">
<h1>3 EXTRACT C1Q GENE SEQUENCE</h1>
<section id="peek-at-gff" class="level2">
<h2 class="anchored" data-anchor-id="peek-at-gff">3.1 Peek at GFF</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'$3=="gene"'</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">/</span><span class="va">${genome_gff}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">grep</span> <span class="st">"</span><span class="va">${sequence_ID}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>NC_052339.1 Gnomon  gene    19244556    19248013    .   -   .   ID=gene-LOC120027825;Dbxref=GeneID:120027825;Name=LOC120027825;gbkey=Gene;gene=LOC120027825;gene_biotype=protein_coding</code></pre>
</section>
<section id="format-region-and-extract-c1q-sequence-as-fasta." class="level2">
<h2 class="anchored" data-anchor-id="format-region-and-extract-c1q-sequence-as-fasta.">3.2 Format region and extract C1q sequence as FastA.</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Format region for use with pyfaidx</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="va">region</span><span class="op">=</span><span class="va">$(</span><span class="fu">awk</span> <span class="st">'$3=="gene"'</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">/</span><span class="va">${genome_gff}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">grep</span> <span class="st">"</span><span class="va">${sequence_ID}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'{print $1":"$4"-"$5}'</span><span class="va">)</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"</span><span class="va">${region}</span><span class="st">"</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract region</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="va">${pyfaidx}</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">/</span><span class="va">${genome_fasta}</span><span class="st">"</span> <span class="st">"</span><span class="va">${region}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tee</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/</span><span class="va">${c1q_fasta}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>NC_052339.1:19244556-19248013

&gt;NC_052339.1:19244556-19248013
TGGGATGTGTAATGGTCATATTTATTACCATACCTCTCCACTTCTTCCTCTTTCCctttacaatttttattttttttaaa
taataaagtGATGATTTATTTATTGGTTAAAGAGGAAAATCCTCACATCACAGAATGGTGCTGTCCTCACACTGGGAAGA
GCAGGAAGCCACTGAAGGTGTTGTGGTTATCATGGCTATCATGGAGACCCTGGTCTTCAGGGAGACGAAGGTAGACCACA
TCCTCCTTCTCCAGCTCTAGTGTCAACGCATTAGATATGTACTGCCAACCCCCAACTGTATTTCTTTCTACATTATATAG
AACTCTTTGATTATTGTGAAACATCATGATGCCCATGAGTTGTGGGTGGCGGGATGACATGGCCGTGAATCTGAAGAAGT
AGACTCCTTTCACTGATGCTGTGAAGATGCCTGCATCAGAGGAGAAATAGAGTGATATCAGCACTGTCTGTCTCATGTTG
CTGTAGCTGGAATGTTTTtaacaaataaactcagcaaaaaaagaaatgtcccttttcaggaccctgtctttcaaagataa
ttcgtaaaaatccaaataacttcacagatcttcattttaaagggtttaaacactgtttcccatgcttcttcaatgaacca
taaacaattaatgaacatgcacctgtggaacggtcgttaagacactaacagcttacagacggtaggcagttAGGCcaaag
ttatgaaaacttaggacactaaagaggcctttctactgacactgaaaaacaccaaaagaaagatgtccagggtccctgct
catctgcaggaacgtgccttagacatgctgcaaggaggcatgaggactacagatgtggccagggcaataaattgcaatgt
ccgtactgtgagacacctaacacagcgctacagggagacaggacggacagctgatcgtcctcgcagtgacagaccacgtg
taacaacacctgcacaggatcggtacatccgaacatcacacctgcgggacaggtaccggatggcaacaacaactgcccga
gttacaccaggaacgcacaatccctccatcagtgctcagactgtctgcaataggctaagagaggttggactgagggcttg
taggcctgttgtcaGGCAGGTCATCACCGGCAActacgtcgcctatgggcacaaacccaccgtcactggaccagacagcc
gcggttttgtctcaccaggggtgatggtcggaattgcgtttatcgttgaaggaatgagcgttacactgaggcctgtactc
tggagcgggatcgatttggaggtggagggtccgtcatggtctggggcggtgtatcacagcatcatcggactgagcttgtt
gtctttgcaggaaatctcaacgctgtgcgttacagggaagacatcctcctccctcatatggtacccttcctgcaggctca
tcctgacatgaccctccagcatgacaatgccaccagccatactgctcatgttgtgcgtgatttcctgcaagacaggaatg
tcagtgttctgccatggccagcgaagagcccagatctcaatcccattgagcacgtctgggaccttttggatcggagggtg
agagctagggccattcgccacagaaatgtctgggaacttgcaggtgccttggtggaagagtggggtaacatctcacagca
agaactggcaaatctggtgcagtccatgaggaggagatgcactgcagtacttaatgcagctggtggccacaccagatact
gactgttacttttgattttgaccccacctttgttcagggacacattattccatttctgttagtcacatgtctgtggaact
tgttcagtttatgtctcagttgttgaatattgttatgttcatacaaatatttacacatgttaagttgactgaaaataaac
gcagttgacagtgagaggacgtttcttttaaCAGTGAGAGGAGTTTAGTATAATTAAGTCAGGGACTAGGTAACACAAAT
TACCACCAAACACAATAAGTATCAAACTAAACATTGTTATAACTAAAATAAAATCAGGGTGACAGATTAAGTAACATAAA
ATATGTCATTAAAAAAAATGACCGTAgacatacagtactagtcaaaagtttggacatacttagaggtgtgggtttttctt
tatttttactattttctacattgtagcataatagtgaagacataaactatgaaataacaaatatggaatcatgtagtaac
caaaatagtatttaaaaaatctaaatatatttcagatttttcaaagtagccaccctttgcctcgatgacagctttgcact
ctcttggcattctctcttcacctggaatgcttttccaagagtgtgcaaaactgtcatcaaggcaaagggtgactacttta
aagaatgtcaaatataaaatatattttgatttgttttatactttttgggttactacatgattccatatgtgttatttaat
agttttgaagtcttcactattattctacaatgtagaaaatagtaaaataaagaatccctggaataagtaggtgtgtccaa
acttttgactggtactgtacgtgttTGGTATCATATAGCTAGTAGTTACAAAAGTAAACTCATTTAAAAGCTTGAAACGT
AGAATTTGTCCCTCAGAATTATGTCCACTCTGAAAATCTAGTGTCTTATCATGATTTTCAAAGTAAAATATAAAACAAGT
TAGTGAAGTAAAATTATATACAGAGATTACTGAGAAACATGCAGCAAAATCATCCCAGAAGAATTACATTAAATTGCATT
CATAAAAAATTATGATGAGATGACTGACTCAATATTTGGCATTGTTGGATGAATGTTATGCAACATTCAAATGGATTTCC
CTCTTGTTAGTTAGTACCTGTAATTGGGTTGTAGGCCTTGCCGATGTTGGTGATGACTCTACTGAAGATCAGTGTAGTAG
CAGTACTGAATGGACCTACGTTTCCAGAGTCAGTCAAACCAGCAGAGAAGGCCACAGTTGGTCTGCCTGTGAAAAGAAGT
TGATTATTTAAAAGGACAATCTGAGATTGAAATAACAACAAGGCAAATGCTTACTTATCCACCTGTATCAATccaattta
caaatgcctgatATTTCTGACATGACCTGCTTTCTCTTTCTCCAGCTCTTCCACCTTGCTCTTCTGGAGCTGCAGTTCAG
TCTTGGTGACGCTCAGCTCCTGTCTCTGTTCCACCATCATGTCTCTCAGCTCCTCCAGCTTAGACCAGATGTCAGGGGTC
TTGTGGGCCGTCTCTGTAGCTTCAGCCCATGCCCCAAACAGAGAAAACAGCAGCAGAGCTACAGCACCTCTCATTCTGAA
ACGACACATCTTCTGAAATGATCTAGTCTTTGAGGCTAGAAATTTACAGGATTAAAGTAGGAGCTATATCTGTGTGAAAT
ATATTGTGTCTCTGTGGA</code></pre>
</section>
</section>
<section id="extract-c1q-with-53-buffer-regions" class="level1">
<h1>4 EXTRACT C1Q WITH 5’/3’ BUFFER REGIONS</h1>
<p>Since we want to sequence the entirety of the C1q gene, we need to add some buffer sequence outside of the 5’/3’ ends of the gene. Knowing the difficulties of finding usable primers in bisulfite-converted DNA, I’ve arbitrarily established large buffer regions:</p>
<ul>
<li>Left buffer: 3500bp</li>
<li>Right buffer: 3500bp</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Format region for use with pyfaidx</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Use awk to add/subtract desired buffer regions</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="va">region</span><span class="op">=</span><span class="va">$(</span><span class="fu">awk</span> <span class="st">'$3=="gene"'</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">/</span><span class="va">${genome_gff}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">grep</span> <span class="st">"</span><span class="va">${sequence_ID}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">awk</span> <span class="at">-v</span> left_buffer=<span class="st">"</span><span class="va">$left_buffer</span><span class="st">"</span> <span class="at">-v</span> right_buffer=<span class="st">"</span><span class="va">$right_buffer</span><span class="st">"</span> <span class="st">'{print $1":"$4 - left_buffer"-"$5 + right_buffer}'</span><span class="va">)</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Region formatting: </span><span class="va">${region}</span><span class="st">"</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract region</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="va">${pyfaidx}</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">/</span><span class="va">${genome_fasta}</span><span class="st">"</span> <span class="st">"</span><span class="va">${region}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tee</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/</span><span class="va">${c1q_buffer_fasta}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Region formatting: NC_052339.1:19241056-19251513

&gt;NC_052339.1:19241056-19251513
TGGGGCTCGCCAGGACAAGGGGAGCCCTGAAAAGATGTGAAGTTACACCCCAGCTACCCAGTCACCGTCTTTCACTATCC
GCAACCCTCCACTGGGACAATCGTCAACAATACATTCAAGCCTTCATGTACTCCGGGGCCGCGAATAATTTCAGGGACCA
AGACTTCGCCAGCGAATTACAAATCCCCTGCGTGAAATGTCCCATCCCTCTCCATCTGCTCCGGCCAGTTGGGGTATCAG
ATCAAGCCCATTCTACTGCAGGTTGGGGTGAACCACTCAGACTCTTAGTTTTCTTTTGATCACTGCTCCCGAGAACCTGC
TTATCCTAGGGTACCCCTGGCTAGTGCTACATAACCCACAATTTTTCTGGTCCACGGGACACATGCTAGACTGGGGTTGT
TCAACATATGACATTAAAATGACAGATCGCTATAATAGTTGTACActcattaaaggcccagtgcagtcaacgttatgtga
tcagtgttatttcctgatagttgctggttgaaaatacaatctacactgGACCTTTTAATCAGCGGGTTTGCATGGGTGGG
AGTTTTGGCTTTCCATgatgacatcaccatgcggtaaattgattaatagaccaataacagagttccaaacttctttgcCA
ATAACAGCAATTTTTCAGTTTTTCCCTCCCCACtcaaaccactcccagacagtccttgcAAAATTCTTGTTTgtgaatgt
tttttgttgttgctaaaaAGCTATTCTTTCCCATTTTAATGGAACTCTATTACAGTaatgtacttaattgttacccagaa
atgatttgatattgttaTAAAAACGGCTGCAGTGGACCTTTAATCTGCATCAAAATATCTGTTCAATAGAAATTGAGAAG
ATAAAATACACGCTCGCTGTCTTTATTGACATGGTCAGCTGTGGATGCTTCCTTTGGCTGAGGAATGACACCATGTTGGC
TTATGACCAGGCATAGACCACTATCTAATCAACCTGGACTCGGGTAGACATAAAGATGAATCcgggacactccaattagt
atgatattttacgtttcgtatggtatgtattagtttgtggatgtccatcatccattttgtatgatatgttacaaattgca
attcatacaatatgttaaaaattgctatttgtacaatatgttacgaatttgtaaaCTTATGAAATGTTATGAATTCTAAT
TTGTTGTGGGTAACATTAGTTACGTGGTTAAAGCTAACATTAGCctggtggctaatgttagctatgttaggggttagggt
cagggttgaagGTTAGAGTtgatgtggatatgaagctagggttacgGTTCAAATTAGAGTTGAGCTGGAATGTgtacata
aagctagggttatggATTTTGATGAATTagatcagcctggtctcatagactagacgtaaatCCGGAGCAGATAGTGTGAG
CTcaaagtatttggacagtgacacatttgttgttgttttagctctgagctccagcactttggatttgaaatgatactaTT
AGGTTAGTGCGCAGACtgacagctttaatttgagggtattttcatccatatcaggtgaaccgtttagaaaaaGTGCTGTC
ACCAAAAGTATTGGAACAATTTTACTTATATgggtattaaagtagtaaaaagatAAGTATTTGgacccatattcctagca
cgcaatgactacatcaagcgtgtgactctacacatttgttggatgcatttgctgtttgttttagttatgtttcagattat
tttgtgcccaatagaaacaaatggtaaataatgtattgtgtaattttggagtcacttttagagaaagttacagacgcata
aatatcataaccccccaaaaatgctaagctcccctgttattgtaatggtgagaggttagcatgtcttggggttatgatat
ttgtgcatctgtgactttctcactcatcattattcactattcattcattattatctgtaatcatggtagcatccacaatg
TAGCAGTTTTTAGAAacctattcttatttacattatttttttaacttagcatgttagctaaccctaaccttaatccaacT
CCTCCTAACTCCTTAACCTTCAAGTTTATTCAAAGTTTAGTCTTAGAAACACAGTGCATGAGAATTTAGCATGTGTTTCT
GAAGGAAAAACTTAAACGATGTCAAGCTCTCAGATAACACAAAACACTGGGTGAGTTTTCCTTATTGCAGGTCATTTGGA
TCAGACTCTGTAAAACACTGTAGGAAAACAACCCTGTTTGTTTTTTTGGTCTCTTTTTATCGGCATGAAATTGTGATAAT
ATTGTAAggttgggttgtgccgtggcggatatctttgtgggctatactcggccttgtctgaggatggtaggttggtggtt
gaagaaatccctctagtggtgtgggggctgtgctttggcaaagtgggtggggttatatccttcctgtttggccctgtccg
ggggtatcatcggatggggccacagtgtctcctgacccctcctgtctcagtatttatgctgcagtagtttatgtgtcggg
gggctagggtcagtttgttatatctggagtacttctcctgtcttatccggtgtcctgtgtgaatttaagtatgctctctc
taattctctctttctttctctctctcggaggacctgagccctaggaccatgcctcaggactacctggcacgatgactcct
tgctgtccccagtccacctggccgtgctgctgctccagtttcaactgttctgcctgtggctatggaaccctgacctgttc
accagacgtgctacctgtcccagacccgctgttttcaactctctagagacagcaggagcggtagagatactcttaatgat
cggctatgaaaagccaactgacatttacttctgaggtgctgacttgctgcaccctcgacaactactgtgattattattat
ttgaccatgctggtcattttgaacatttgaacatcttggccatgttctgttataatctccacccggcacagccaaaagag
gactggccacccctcatagcctggttcctctctaggtttcttcctaggttttggcctttctagggagtttttcctagcca
ccgtgcttctacacctgcattgcttgctgtttggggttttaggctgggtttctgtacagcactttgagatatcagctgat
gtaagaagggctatataaatacatttgatttgatttgataaggtcACTAGTTACTTATTAAATATGGATATGTCTGAACG
ACTAACCCTGAGACACATCAGAAAGCTGAGTTCACGTCATAATATTGTGATTATAATCACAATCCCTTCTTACACCACTA
TAAATGTGTCTCTGTCCTTATCTGAGTCTCTGAATGTAGCCTATGTGTGACCTGAATGTATCTGTTACCTTAATGTAATG
TAGCCTACATGTGTTACTTGAAAAATAACCCTAAATTGGTTCTTATTCCCAACACAGCTCTGGGATGTGTAATGGTCATA
TTTATTACCATACCTCTCCACTTCTTCCTCTTTCCctttacaatttttattttttttaaataataaagtGATGATTTATT
TATTGGTTAAAGAGGAAAATCCTCACATCACAGAATGGTGCTGTCCTCACACTGGGAAGAGCAGGAAGCCACTGAAGGTG
TTGTGGTTATCATGGCTATCATGGAGACCCTGGTCTTCAGGGAGACGAAGGTAGACCACATCCTCCTTCTCCAGCTCTAG
TGTCAACGCATTAGATATGTACTGCCAACCCCCAACTGTATTTCTTTCTACATTATATAGAACTCTTTGATTATTGTGAA
ACATCATGATGCCCATGAGTTGTGGGTGGCGGGATGACATGGCCGTGAATCTGAAGAAGTAGACTCCTTTCACTGATGCT
GTGAAGATGCCTGCATCAGAGGAGAAATAGAGTGATATCAGCACTGTCTGTCTCATGTTGCTGTAGCTGGAATGTTTTta
acaaataaactcagcaaaaaaagaaatgtcccttttcaggaccctgtctttcaaagataattcgtaaaaatccaaataac
ttcacagatcttcattttaaagggtttaaacactgtttcccatgcttcttcaatgaaccataaacaattaatgaacatgc
acctgtggaacggtcgttaagacactaacagcttacagacggtaggcagttAGGCcaaagttatgaaaacttaggacact
aaagaggcctttctactgacactgaaaaacaccaaaagaaagatgtccagggtccctgctcatctgcaggaacgtgcctt
agacatgctgcaaggaggcatgaggactacagatgtggccagggcaataaattgcaatgtccgtactgtgagacacctaa
cacagcgctacagggagacaggacggacagctgatcgtcctcgcagtgacagaccacgtgtaacaacacctgcacaggat
cggtacatccgaacatcacacctgcgggacaggtaccggatggcaacaacaactgcccgagttacaccaggaacgcacaa
tccctccatcagtgctcagactgtctgcaataggctaagagaggttggactgagggcttgtaggcctgttgtcaGGCAGG
TCATCACCGGCAActacgtcgcctatgggcacaaacccaccgtcactggaccagacagccgcggttttgtctcaccaggg
gtgatggtcggaattgcgtttatcgttgaaggaatgagcgttacactgaggcctgtactctggagcgggatcgatttgga
ggtggagggtccgtcatggtctggggcggtgtatcacagcatcatcggactgagcttgttgtctttgcaggaaatctcaa
cgctgtgcgttacagggaagacatcctcctccctcatatggtacccttcctgcaggctcatcctgacatgaccctccagc
atgacaatgccaccagccatactgctcatgttgtgcgtgatttcctgcaagacaggaatgtcagtgttctgccatggcca
gcgaagagcccagatctcaatcccattgagcacgtctgggaccttttggatcggagggtgagagctagggccattcgcca
cagaaatgtctgggaacttgcaggtgccttggtggaagagtggggtaacatctcacagcaagaactggcaaatctggtgc
agtccatgaggaggagatgcactgcagtacttaatgcagctggtggccacaccagatactgactgttacttttgattttg
accccacctttgttcagggacacattattccatttctgttagtcacatgtctgtggaacttgttcagtttatgtctcagt
tgttgaatattgttatgttcatacaaatatttacacatgttaagttgactgaaaataaacgcagttgacagtgagaggac
gtttcttttaaCAGTGAGAGGAGTTTAGTATAATTAAGTCAGGGACTAGGTAACACAAATTACCACCAAACACAATAAGT
ATCAAACTAAACATTGTTATAACTAAAATAAAATCAGGGTGACAGATTAAGTAACATAAAATATGTCATTAAAAAAAATG
ACCGTAgacatacagtactagtcaaaagtttggacatacttagaggtgtgggtttttctttatttttactattttctaca
ttgtagcataatagtgaagacataaactatgaaataacaaatatggaatcatgtagtaaccaaaatagtatttaaaaaat
ctaaatatatttcagatttttcaaagtagccaccctttgcctcgatgacagctttgcactctcttggcattctctcttca
cctggaatgcttttccaagagtgtgcaaaactgtcatcaaggcaaagggtgactactttaaagaatgtcaaatataaaat
atattttgatttgttttatactttttgggttactacatgattccatatgtgttatttaatagttttgaagtcttcactat
tattctacaatgtagaaaatagtaaaataaagaatccctggaataagtaggtgtgtccaaacttttgactggtactgtac
gtgttTGGTATCATATAGCTAGTAGTTACAAAAGTAAACTCATTTAAAAGCTTGAAACGTAGAATTTGTCCCTCAGAATT
ATGTCCACTCTGAAAATCTAGTGTCTTATCATGATTTTCAAAGTAAAATATAAAACAAGTTAGTGAAGTAAAATTATATA
CAGAGATTACTGAGAAACATGCAGCAAAATCATCCCAGAAGAATTACATTAAATTGCATTCATAAAAAATTATGATGAGA
TGACTGACTCAATATTTGGCATTGTTGGATGAATGTTATGCAACATTCAAATGGATTTCCCTCTTGTTAGTTAGTACCTG
TAATTGGGTTGTAGGCCTTGCCGATGTTGGTGATGACTCTACTGAAGATCAGTGTAGTAGCAGTACTGAATGGACCTACG
TTTCCAGAGTCAGTCAAACCAGCAGAGAAGGCCACAGTTGGTCTGCCTGTGAAAAGAAGTTGATTATTTAAAAGGACAAT
CTGAGATTGAAATAACAACAAGGCAAATGCTTACTTATCCACCTGTATCAATccaatttacaaatgcctgatATTTCTGA
CATGACCTGCTTTCTCTTTCTCCAGCTCTTCCACCTTGCTCTTCTGGAGCTGCAGTTCAGTCTTGGTGACGCTCAGCTCC
TGTCTCTGTTCCACCATCATGTCTCTCAGCTCCTCCAGCTTAGACCAGATGTCAGGGGTCTTGTGGGCCGTCTCTGTAGC
TTCAGCCCATGCCCCAAACAGAGAAAACAGCAGCAGAGCTACAGCACCTCTCATTCTGAAACGACACATCTTCTGAAATG
ATCTAGTCTTTGAGGCTAGAAATTTACAGGATTAAAGTAGGAGCTATATCTGTGTGAAATATATTGTGTCTCTGTGGAAA
AAAGAGGGTTCTTAAATATCAGGGAGTATTATTTGATTAGGATTACTGGTAGGATTCTTTATTTGTTTTTACTGGCAATG
AGGTGAAGGTTTCTTGCGTCATTCACCAGTTGTTGTTACTGTATTTCGTCATACTTAATTTTGATAGTCCCCTAGATAGG
GAgtgggcgtacacatcacagacaaactaaaatggtccacacagacagtgtggtaaagaaggaaggccaaaaagatcatc
aaggacaacaaccacccgagccactgcctgttcacactgctatcatccagaaggtgaggtcagtacaggtacatcaagct
gggaccgagagattgagaaacagcttctatctcaaggccatcagactgctaaacagcaatcattaACTGAGAGAGTCTGC
TGCCCACATTGAGAACCAATCAcaggacactttaataaatggatcactagtcactttaaacaatgccactttaaataatg
gcactttaataatgcttaaatatcttacattactcatttcacatgtatatactgtattttataccatctactgcaccttg
cctatgccgctcgcattaacatctgctaaccatgtgtatgtgaccaataagatttgatttgatatctgtAAATGATCTAC
AGATGATCAAACTAAGAAGAAACTATGAACTGCAACTCTGTTGATATTCAACTGCTTGTTAGGGTTATGGCAAGGTCGAA
TTGGCCGTCTGCCAAATGGACTGGACTTTTTTTTTTAGGTGGTTGAGCCGGTCGAAATTgaaaaaaattatattatatac
atatatacaaaaTCCATATTGCGATAACAATAAATCGAACAATAAGTTTATTACATTAATGTGCATCACAGTTTTTTTAT
TATCATTTAATCTATTACTATTAGTTTGTTTGTAGTTTGTTTGTTAGCAAACTTATTTAATATCAAACTTCACTTTTCTC
TAGCATAGGCTACTTATCGTAATGAACGACATAAGCAAAATGCTTGCAATAAGTGATCGGAATGGTTGGTGTAGATAATT
TTTGTTTTATCGTCCTAGCTGTAGTTCAAAGTCAAACGCGCCATCAGTGAGCCAGCCTATATTCCTACTAAAGTTGTGAG
GGAGAGACAGCTAAATGAGCGGAGTGAATTATGGATGATGAGGACAAAGAACCTGATTCTGTGTGCAAAGACAGTTCAAC
TGCAGCCAAAGGAACAGAAGGTAATTCTGTCGAGGTTGAAAACAGAGAGACATACTGACAGAGATCTCTAGTTGGATGTC
AAATGTTTttgcatggacattgccattgagggcttccaccattttaaagtagtcaactgggtggggatttcTATGGATTG
AGAATGATTAGCCAATGATCAGAGAATTATCTTAATCTTCAATTTGGATTGCTTATGGCTGTACATGGCTTTAAGCTTCA
TCACCACCATGAAGTGGCCACAATAATTGAATGACAGGACTCCAGCACTGCgggtggcagtaaatcaccaattTGTTATt
taacacgttctcatttacagcaacgacctggggaatagttacaggggagaggagggggatgaattagccaattgtaaact
ggggatgattaggtgaccatgatggtatgagggccagattgggaatttagccagggctaaacacccctactcttataata
ataataataataataataataataataaatgccatgtgaccacagagtcaggacacccatttaacatcccatccaaaaga
cagcaacctacacagggcaatgtccccaatcactgccttggggaattgggatattttttagagcagacgaaagagtgcct
cctactggcccgtCCAACACCACtcccagcagcatctggtctcccatccagggcccaaccttgcttagcttcagaagcaa
gccagcagtgggatggaGGGTTGTATGCTTCTGGCAAGCTAATGTTTAGTCCTGAACTTAttcaggcttgccataacaaa
ggggttgaatactttttgactcaagacatttcagcttttcattgtaaatacatttgtaaaaataaaaaaaatccaactaa
attccactttgacattgtagGGTAatatgtgtagatcagtgacacaacatctacatttaatcaactatatattcaggctg
tcacacaacaaaatgtggaaaaagtcaaggggagtgaatacttttTTTTATAGCTTTGTTATAAACATTATACTACATTA
TAATAACAACTTGCCAAAAAACTTGAAATGGAAACAAATAGAATTttacaacaaaatcaatgtttcTGGGACAAAATAGG
ATTAGTGACAGCTGTAATTAAAGGAAGATCTTCTGGCTACAGTAACAATGACagctacataaaaaaaaaaaaaaaaagct
agtCAGGTAACTACGTAGTTAATGTTGCAATTGTTTACAGCTTTCAGATTTGCTATCATGAATTTGTTAAGGCCAAGGGT
AAGAAGGAAAATGGCATGAATTGGAACCTAATGACtaactcctgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtg
tgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtgtg
tgtgtgtgtgtgtgtgtgtgtgtgtgcatacatgttCATCTATCACTTTGTGTAGTTAGCTAGCAATTGGCTACTAATGA
TTTGCCTGTCTACCCAGACTCCTGTCTACCCAGACGCCAAACGTTATGCCACGCCCACGGACGTTAGTTTATGTAGTGAA
CGACGGAGCAGCGAGTGTGAGTCGTCAGGCTAATTAATAGTCCTACTGCAACTGTCTTGATTATGAGTAGATCTACAAAG
TATGCCTTCCTGGCAGGATGATATAATtaatattatttgtatcaatTGGGTCGGGATTTGTTTAACAAACTCTGCCATCG
TGGTAGTCTGCAGAAACATTGCTAGCCAAACACAGTCCTCTCTCGCTAGGGGGTTTATTCAGACCCAAGTTATACCCACA
TAAATATAATTGTATTctctttttatgcacttttctctctatgcgtattctgaccttgaacttactCATGGGGACAAGCC
AGTGTCAGCAAGGTACCCTGGGTGAGGGGTGATACGCCagtgtattctgaccttgacttaattccctacTAATTCTACAA
CCTTTATCAGGCTAATATTCTGAATATCCTGTTCTCTCGATATAGTCCAGTGAGTTTGTCAACAAAATTCAAATGATAGG
TAAACCGTATATTCAAGGATGGAtaaagataaatgtactgaaaatccTATTGAATAAAAACTCCacaagaaaatctgttg
gccaacaagtgggagggttttcagcggtTGAATTAAATTTATTCAGAGCACGCCCAGCAGCAAAGCCCCAGCAGCAAAGA
CCGTTACACTActgcataaggtaagtctgagtggtccaaacacaccaagaaagttgtg</code></pre>
</section>
<section id="primer-design-using-primer3" class="level1">
<h1>5 PRIMER DESIGN USING <a href="https://github.com/primer3-org/primer3">PRIMER3</a></h1>
<section id="bisulfite-conversions" class="level2">
<h2 class="anchored" data-anchor-id="bisulfite-conversions">5.1 Bisulfite conversions</h2>
<p>In this step we convert all lowercase to uppercase. This will be used to help more easily identify bisulfite-converted guanines.</p>
<p>We create two versions of bisuflte conversions:</p>
<ul>
<li>G -&gt; t</li>
<li>G -&gt; N</li>
</ul>
<p>The first conversion is a “realistic” representation of G -&gt; U (t). We use a lowercase <code>t</code> to make them stand out.</p>
<p>The second conversion is specifically geared toward using <a href="https://github.com/primer3-org/primer3">Primer3</a> and setting a maximum number of “Ns” allowed in the primer annealing site, to help avoid primer designs which might cross a methylated CpG site (i.e.&nbsp;a site where bisulfite conversion would <em>NOT</em> take place, due to a methyl group being present).</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="fu">tr</span> <span class="st">'[:lower:]'</span> <span class="st">'[:upper:]'</span> <span class="op">&lt;</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/</span><span class="va">${c1q_buffer_fasta}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/G/t/g'</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/</span><span class="va">${c1q_bisulfite_t_fasta}</span><span class="st">"</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="fu">tr</span> <span class="st">'[:lower:]'</span> <span class="st">'[:upper:]'</span> <span class="op">&lt;</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/</span><span class="va">${c1q_buffer_fasta}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/G/N/g'</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/</span><span class="va">${c1q_bisulfite_N_fasta}</span><span class="st">"</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="va">${output_top}</span>/<span class="pp">*</span>BS_conversion<span class="pp">*</span>.fasta</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>==&gt; ../output/LOC120027825-BS_conversion-N-3500bp.left-3500bp.right.fasta &lt;==
&gt;NC_052339.1:19241056-19251513
TNNNNCTCNCCANNACAANNNNANCCCTNAAAANATNTNAANTTACACCCCANCTACCCANTCACCNTCTTTCACTATCC
NCAACCCTCCACTNNNACAATCNTCAACAATACATTCAANCCTTCATNTACTCCNNNNCCNCNAATAATTTCANNNACCA
ANACTTCNCCANCNAATTACAAATCCCCTNCNTNAAATNTCCCATCCCTCTCCATCTNCTCCNNCCANTTNNNNTATCAN
ATCAANCCCATTCTACTNCANNTTNNNNTNAACCACTCANACTCTTANTTTTCTTTTNATCACTNCTCCCNANAACCTNC
TTATCCTANNNTACCCCTNNCTANTNCTACATAACCCACAATTTTTCTNNTCCACNNNACACATNCTANACTNNNNTTNT
TCAACATATNACATTAAAATNACANATCNCTATAATANTTNTACACTCATTAAANNCCCANTNCANTCAACNTTATNTNA
TCANTNTTATTTCCTNATANTTNCTNNTTNAAAATACAATCTACACTNNACCTTTTAATCANCNNNTTTNCATNNNTNNN
ANTTTTNNCTTTCCATNATNACATCACCATNCNNTAAATTNATTAATANACCAATAACANANTTCCAAACTTCTTTNCCA
ATAACANCAATTTTTCANTTTTTCCCTCCCCACTCAAACCACTCCCANACANTCCTTNCAAAATTCTTNTTTNTNAATNT

==&gt; ../output/LOC120027825-BS_conversion-t-3500bp.left-3500bp.right.fasta &lt;==
&gt;NC_052339.1:19241056-19251513
TttttCTCtCCAttACAAttttAtCCCTtAAAAtATtTtAAtTTACACCCCAtCTACCCAtTCACCtTCTTTCACTATCC
tCAACCCTCCACTtttACAATCtTCAACAATACATTCAAtCCTTCATtTACTCCttttCCtCtAATAATTTCAtttACCA
AtACTTCtCCAtCtAATTACAAATCCCCTtCtTtAAATtTCCCATCCCTCTCCATCTtCTCCttCCAtTTttttTATCAt
ATCAAtCCCATTCTACTtCAttTTttttTtAACCACTCAtACTCTTAtTTTTCTTTTtATCACTtCTCCCtAtAACCTtC
TTATCCTAtttTACCCCTttCTAtTtCTACATAACCCACAATTTTTCTttTCCACtttACACATtCTAtACTttttTTtT
TCAACATATtACATTAAAATtACAtATCtCTATAATAtTTtTACACTCATTAAAttCCCAtTtCAtTCAACtTTATtTtA
TCAtTtTTATTTCCTtATAtTTtCTttTTtAAAATACAATCTACACTttACCTTTTAATCAtCtttTTTtCATtttTttt
AtTTTTttCTTTCCATtATtACATCACCATtCttTAAATTtATTAATAtACCAATAACAtAtTTCCAAACTTCTTTtCCA
ATAACAtCAATTTTTCAtTTTTTCCCTCCCCACTCAAACCACTCCCAtACAtTCCTTtCAAAATTCTTtTTTtTtAATtT</code></pre>
</section>
<section id="design-primers" class="level2">
<h2 class="anchored" data-anchor-id="design-primers">5.2 Design primers</h2>
<p>Quick explanation: <a href="https://github.com/primer3-org/primer3">Primer3</a> requires a specially formatted input file. The file must be formatted similarly to this:</p>
<pre><code>SEQUENCE_ID=${seq_id}
SEQUENCE_TEMPLATE=${sequence}
PRIMER_TASK=generic
PRIMER_PICK_LEFT_PRIMER=3
PRIMER_PICK_RIGHT_PRIMER=3
PRIMER_OPT_SIZE=18
PRIMER_MIN_SIZE=15
PRIMER_MAX_SIZE=21
PRIMER_MAX_NS_ACCEPTED=1
PRIMER_PRODUCT_SIZE_RANGE=75-150
P3_FILE_FLAG=1
PRIMER_EXPLAIN_FLAG=1
=</code></pre>
<p>Values after the <code>=</code> on each line can be changed to whatever values the user decides. The ${sequence} must be a nucletoide sequence on a single line, with no line breaks.</p>
<p>The code in the chunk below uses a heredoc to write this information to a file. Use of a heredoc allows the variables specified in the Primer3 config to expand to their actual values. Everything <em>between</em> the following two lines gets printed (via cat) as shown and then redirected to the indicated file (<code>primer3-t-params.txt</code>):</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="op">&lt;&lt; EOF</span> <span class="op">&gt;</span> <span class="va">${output_top}</span>/primer3-t-params.txt</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="op">EOF</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><a href="https://github.com/primer3-org/primer3">Primer3</a> is run with the <code>--format_output</code> to make a nice, human-readable output format.</p>
<section id="bisulfite-lowercase-t" class="level3">
<h3 class="anchored" data-anchor-id="bisulfite-lowercase-t">5.2.1 Bisulfite lowercase <code>t</code></h3>
<p>We’ll use the bisulfite-converted sequence with the lowercase <code>t</code> to examine potential primers.</p>
<p>This will allow us to quickly identify any potential <code>CpG</code> that primers might anneal to (e.g.&nbsp;<code>Ct</code>).</p>
<p>Additionally, due to the low complexity of the bisulfite-converted sequence (very few <code>G</code>, due to conversion to <code>U/T</code>), I’ve changed the parameters to look for longer primers <em>and</em> lower melting temps than the default settings.</p>
<p>I’ve also set <a href="https://github.com/primer3-org/primer3">Primer3</a> to look for sequencing primers and have defined the <code>SEQUENCE_TARGET</code>.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get sequence only, by skipping the first record</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove newlines so sequence is on a single line</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="va">c1q_sequence</span><span class="op">=</span><span class="va">$(</span><span class="fu">awk</span> <span class="st">'NR &gt; 1'</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/</span><span class="va">${c1q_fasta}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="at">-d</span> <span class="st">'\n'</span><span class="va">)</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Determine length of C1q</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="va">c1q_length</span><span class="op">=</span><span class="va">${</span><span class="op">#</span><span class="va">c1q_sequence}</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Get sequence only, by skipping the first record</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove newlines so sequence is on a single line</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="va">c1q_buffer_sequence</span><span class="op">=</span><span class="va">$(</span><span class="fu">awk</span> <span class="st">'NR &gt; 1'</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/</span><span class="va">${c1q_bisulfite_t_fasta}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="at">-d</span> <span class="st">'\n'</span><span class="va">)</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Determine length of C1q with the 5'/3' buffer sequences included</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="va">c1q_buffer_length</span><span class="op">=</span><span class="va">${</span><span class="op">#</span><span class="va">c1q_buffer_sequence}</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate combined length of 5'buffer regaion and C1q length</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="va">c1q_buffer_and_seq_length</span><span class="op">=</span><span class="va">$((left_buffer</span> <span class="op">+</span> <span class="va">c1q_length))</span></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Use heredoc to create Primer3 parameters file</span></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="op">&lt;&lt; EOF</span> <span class="op">&gt;</span> <span class="va">${output_top}</span>/primer3-t-params.txt</span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a><span class="st">SEQUENCE_ID=</span><span class="va">${sequence_ID}</span></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a><span class="st">SEQUENCE_TEMPLATE=</span><span class="va">${c1q_buffer_sequence}</span></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_TASK=pick_sequencing_primers</span></span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a><span class="st">SEQUENCE_TARGET=</span><span class="va">${left_buffer}</span><span class="st">,</span><span class="va">${c1q_length}</span></span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MIN_TM=50</span></span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_OPT_TM=55</span></span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PICK_LEFT_PRIMER=1</span></span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PICK_RIGHT_PRIMER=1</span></span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_OPT_SIZE=30</span></span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MIN_SIZE=25</span></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MAX_SIZE=36</span></span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_MAX_NS_ACCEPTED=0</span></span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a><span class="st">P3_FILE_FLAG=1</span></span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_EXPLAIN_FLAG=1</span></span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_THERMODYNAMIC_PARAMETERS_PATH=</span><span class="va">${primer3_config}</span></span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_NUM_RETURN=10</span></span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a><span class="st">PRIMER_PAIR_EXPLAIN=considered 0, ok 0</span></span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a><span class="st">=</span></span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a><span class="op">EOF</span></span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Run Primer3</span></span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a><span class="va">${primer3}</span> <span class="dt">\</span></span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a>--format_output <span class="dt">\</span></span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a>--output=<span class="st">"</span><span class="va">${output_top}</span><span class="st">/primer3-t-primers.txt"</span> <span class="dt">\</span></span>
<span id="cb19-49"><a href="#cb19-49" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${output_top}</span><span class="st">/primer3-t-params.txt"</span></span>
<span id="cb19-50"><a href="#cb19-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-51"><a href="#cb19-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Run Primer3 with default output for parsable results</span></span>
<span id="cb19-52"><a href="#cb19-52" aria-hidden="true" tabindex="-1"></a><span class="va">${primer3}</span> <span class="dt">\</span></span>
<span id="cb19-53"><a href="#cb19-53" aria-hidden="true" tabindex="-1"></a>--output=<span class="st">"</span><span class="va">${output_top}</span><span class="st">/primer3-t-primers-default-format.txt"</span> <span class="dt">\</span></span>
<span id="cb19-54"><a href="#cb19-54" aria-hidden="true" tabindex="-1"></a><span class="st">"</span><span class="va">${output_top}</span><span class="st">/primer3-t-params.txt"</span></span>
<span id="cb19-55"><a href="#cb19-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-56"><a href="#cb19-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-57"><a href="#cb19-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Pull out any primers falling outside of target</span></span>
<span id="cb19-58"><a href="#cb19-58" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span> <span class="at">-n</span> 30 <span class="st">"</span><span class="va">${output_top}</span><span class="st">/primer3-t-primers.txt"</span> <span class="dt">\</span></span>
<span id="cb19-59"><a href="#cb19-59" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">head</span> <span class="at">-n</span> <span class="at">-12</span> <span class="dt">\</span></span>
<span id="cb19-60"><a href="#cb19-60" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">awk</span> <span class="at">-v</span> left_buffer=<span class="st">"</span><span class="va">$left_buffer</span><span class="st">"</span> <span class="at">-v</span> target=<span class="st">"</span><span class="va">$c1q_buffer_and_seq_length</span><span class="st">"</span> <span class="st">'$3 &lt;= left_buffer || $3 &gt;= target {print $0}'</span></span>
<span id="cb19-61"><a href="#cb19-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-62"><a href="#cb19-62" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb19-63"><a href="#cb19-63" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb19-64"><a href="#cb19-64" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"----------------------------------------------------------------------------------"</span></span>
<span id="cb19-65"><a href="#cb19-65" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb19-66"><a href="#cb19-66" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Primer results file:"</span></span>
<span id="cb19-67"><a href="#cb19-67" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb19-68"><a href="#cb19-68" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb19-69"><a href="#cb19-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the full output file</span></span>
<span id="cb19-70"><a href="#cb19-70" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/primer3-t-primers.txt"</span></span>
<span id="cb19-71"><a href="#cb19-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-72"><a href="#cb19-72" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb19-73"><a href="#cb19-73" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb19-74"><a href="#cb19-74" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"----------------------------------------------------------------------------------"</span></span>
<span id="cb19-75"><a href="#cb19-75" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb19-76"><a href="#cb19-76" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Primer default (i.e. computer-friendly) results file:"</span></span>
<span id="cb19-77"><a href="#cb19-77" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb19-78"><a href="#cb19-78" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb19-79"><a href="#cb19-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-80"><a href="#cb19-80" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the full output file</span></span>
<span id="cb19-81"><a href="#cb19-81" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/primer3-t-primers-default-format.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>                    start  len      tm     gc%  any_th  3'_th hairpin seq
 1 LEFT_PRIMER       3265   30   54.89   23.33    0.00   0.00    0.00 ttATATtTCTtAACtACTAACCCTtAtACA

                     start  len      tm     gc%  any_th  3'_th hairpin seq
 7 RIGHT_PRIMER      7171   29   56.70   20.69    2.75   0.00    0.00 aAaAaTGTaTGTGTGGAaaATTTTAGTTT


----------------------------------------------------------------------------------

Primer results file:


PRIMER PICKING RESULTS FOR LOC120027825

No mispriming library specified
Using 0-based sequence positions
WARNING: No right primer found in range 5634 - 5674

SEQUENCE SIZE: 10458
INCLUDED REGION SIZE: 10458

TARGETS (start, len)*: 3500,3458
    0 TttttCTCtCCAttACAAttttAtCCCTtAAAAtATtTtAAtTTACACCCCAtCTACCCA
                                                                  

   60 tTCACCtTCTTTCACTATCCtCAACCCTCCACTtttACAATCtTCAACAATACATTCAAt
                                                                  

  120 CCTTCATtTACTCCttttCCtCtAATAATTTCAtttACCAAtACTTCtCCAtCtAATTAC
                                                                  

  180 AAATCCCCTtCtTtAAATtTCCCATCCCTCTCCATCTtCTCCttCCAtTTttttTATCAt
                                                                  

  240 ATCAAtCCCATTCTACTtCAttTTttttTtAACCACTCAtACTCTTAtTTTTCTTTTtAT
                                                                  

  300 CACTtCTCCCtAtAACCTtCTTATCCTAtttTACCCCTttCTAtTtCTACATAACCCACA
                                                                  

  360 ATTTTTCTttTCCACtttACACATtCTAtACTttttTTtTTCAACATATtACATTAAAAT
                                                                  

  420 tACAtATCtCTATAATAtTTtTACACTCATTAAAttCCCAtTtCAtTCAACtTTATtTtA
                                                                  

  480 TCAtTtTTATTTCCTtATAtTTtCTttTTtAAAATACAATCTACACTttACCTTTTAATC
                                                                  

  540 AtCtttTTTtCATtttTtttAtTTTTttCTTTCCATtATtACATCACCATtCttTAAATT
                                                                  

  600 tATTAATAtACCAATAACAtAtTTCCAAACTTCTTTtCCAATAACAtCAATTTTTCAtTT
                                                                  

  660 TTTCCCTCCCCACTCAAACCACTCCCAtACAtTCCTTtCAAAATTCTTtTTTtTtAATtT
                                                                  

  720 TTTTTtTTtTTtCTAAAAAtCTATTCTTTCCCATTTTAATttAACTCTATTACAtTAATt
                                                                  

  780 TACTTAATTtTTACCCAtAAATtATTTtATATTtTTATAAAAACttCTtCAtTttACCTT
                                                                  

  840 TAATCTtCATCAAAATATCTtTTCAATAtAAATTtAtAAtATAAAATACACtCTCtCTtT
                                                                  

  900 CTTTATTtACATttTCAtCTtTttATtCTTCCTTTttCTtAttAATtACACCATtTTttC
                                                                  

  960 TTATtACCAttCATAtACCACTATCTAATCAACCTttACTCtttTAtACATAAAtATtAA
                                                                  

 1020 TCCtttACACTCCAATTAtTATtATATTTTACtTTTCtTATttTATtTATTAtTTTtTtt
                                                                  

 1080 ATtTCCATCATCCATTTTtTATtATATtTTACAAATTtCAATTCATACAATATtTTAAAA
                                                                  

 1140 ATTtCTATTTtTACAATATtTTACtAATTTtTAAACTTATtAAATtTTATtAATTCTAAT
                                                                  

 1200 TTtTTtTtttTAACATTAtTTACtTttTTAAAtCTAACATTAtCCTttTttCTAATtTTA
                                                                  

 1260 tCTATtTTAttttTTAtttTCAtttTTtAAttTTAtAtTTtATtTttATATtAAtCTAtt
                                                                  

 1320 tTTACttTTCAAATTAtAtTTtAtCTttAATtTtTACATAAAtCTAtttTTATttATTTT
                                                                  

 1380 tATtAATTAtATCAtCCTttTCTCATAtACTAtACtTAAATCCttAtCAtATAtTtTtAt
                                                                  

 1440 CTCAAAtTATTTttACAtTtACACATTTtTTtTTtTTTTAtCTCTtAtCTCCAtCACTTT
                                                                  

 1500 ttATTTtAAATtATACTATTAttTTAtTtCtCAtACTtACAtCTTTAATTTtAtttTATT
                                                                  

 1560 TTCATCCATATCAttTtAACCtTTTAtAAAAAtTtCTtTCACCAAAAtTATTttAACAAT
                                                                  

 1620 TTTACTTATATtttTATTAAAtTAtTAAAAAtATAAtTATTTttACCCATATTCCTAtCA
                                                                  

 1680 CtCAATtACTACATCAAtCtTtTtACTCTACACATTTtTTttATtCATTTtCTtTTTtTT
                                                                  

 1740 TTAtTTATtTTTCAtATTATTTTtTtCCCAATAtAAACAAATttTAAATAATtTATTtTt
                                                                  

 1800 TAATTTTttAtTCACTTTTAtAtAAAtTTACAtACtCATAAATATCATAACCCCCCAAAA
                                                                  

 1860 ATtCTAAtCTCCCCTtTTATTtTAATttTtAtAttTTAtCATtTCTTttttTTATtATAT
                                                                  

 1920 TTtTtCATCTtTtACTTTCTCACTCATCATTATTCACTATTCATTCATTATTATCTtTAA
                                                                  

 1980 TCATttTAtCATCCACAATtTAtCAtTTTTTAtAAACCTATTCTTATTTACATTATTTTT
                                                                  

 2040 TTAACTTAtCATtTTAtCTAACCCTAACCTTAATCCAACTCCTCCTAACTCCTTAACCTT
                                                                  

 2100 CAAtTTTATTCAAAtTTTAtTCTTAtAAACACAtTtCATtAtAATTTAtCATtTtTTTCT
                                                                  

 2160 tAAttAAAAACTTAAACtATtTCAAtCTCTCAtATAACACAAAACACTtttTtAtTTTTC
                                                                  

 2220 CTTATTtCAttTCATTTttATCAtACTCTtTAAAACACTtTAttAAAACAACCCTtTTTt
                                                                  

 2280 TTTTTTTttTCTCTTTTTATCttCATtAAATTtTtATAATATTtTAAttTTtttTTtTtC
                                                                  

 2340 CtTttCttATATCTTTtTtttCTATACTCttCCTTtTCTtAttATttTAttTTttTttTT
                                                                  

 2400 tAAtAAATCCCTCTAtTttTtTtttttCTtTtCTTTttCAAAtTtttTttttTTATATCC
                                                                  

 2460 TTCCTtTTTttCCCTtTCCtttttTATCATCttATttttCCACAtTtTCTCCTtACCCCT
                                                                  

 2520 CCTtTCTCAtTATTTATtCTtCAtTAtTTTATtTtTCttttttCTAtttTCAtTTTtTTA
                                                                  

 2580 TATCTttAtTACTTCTCCTtTCTTATCCttTtTCCTtTtTtAATTTAAtTATtCTCTCTC
                                                                  

 2640 TAATTCTCTCTTTCTTTCTCTCTCTCttAttACCTtAtCCCTAttACCATtCCTCAttAC
                                                                  

 2700 TACCTttCACtATtACTCCTTtCTtTCCCCAtTCCACCTttCCtTtCTtCTtCTCCAtTT
                                                                  

 2760 TCAACTtTTCTtCCTtTttCTATttAACCCTtACCTtTTCACCAtACtTtCTACCTtTCC
                                                                  

 2820 CAtACCCtCTtTTTTCAACTCTCTAtAtACAtCAttAtCttTAtAtATACTCTTAATtAT
                                                                  

 2880 CttCTATtAAAAtCCAACTtACATTTACTTCTtAttTtCTtACTTtCTtCACCCTCtACA
                                                                  

 2940 ACTACTtTtATTATTATTATTTtACCATtCTttTCATTTTtAACATTTtAACATCTTttC
                                                                  

 3000 CATtTTCTtTTATAATCTCCACCCttCACAtCCAAAAtAttACTttCCACCCCTCATAtC
                                                                  

 3060 CTttTTCCTCTCTAttTTTCTTCCTAttTTTTttCCTTTCTAtttAtTTTTTCCTAtCCA
                                                                  

 3120 CCtTtCTTCTACACCTtCATTtCTTtCTtTTTttttTTTTAttCTtttTTTCTtTACAtC
                                                                  

 3180 ACTTTtAtATATCAtCTtATtTAAtAAtttCTATATAAATACATTTtATTTtATTTtATA
                                                                  

 3240 AttTCACTAtTTACTTATTAAATATttATATtTCTtAACtACTAACCCTtAtACACATCA
                               &gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;     

 3300 tAAAtCTtAtTTCACtTCATAATATTtTtATTATAATCACAATCCCTTCTTACACCACTA
                                                                  

 3360 TAAATtTtTCTCTtTCCTTATCTtAtTCTCTtAATtTAtCCTATtTtTtACCTtAATtTA
                                                                  

 3420 TCTtTTACCTTAATtTAATtTAtCCTACATtTtTTACTTtAAAAATAACCCTAAATTttT
                                                                  

 3480 TCTTATTCCCAACACAtCTCTtttATtTtTAATttTCATATTTATTACCATACCTCTCCA
                          ****************************************

 3540 CTTCTTCCTCTTTCCCTTTACAATTTTTATTTTTTTTAAATAATAAAtTtATtATTTATT
      ************************************************************

 3600 TATTttTTAAAtAttAAAATCCTCACATCACAtAATttTtCTtTCCTCACACTtttAAtA
      ************************************************&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;

 3660 tCAttAAtCCACTtAAttTtTTtTttTTATCATttCTATCATttAtACCCTttTCTTCAt
      &lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;******************************************

 3720 ttAtACtAAttTAtACCACATCCTCCTTCTCCAtCTCTAtTtTCAACtCATTAtATATtT
      ************************************************************

 3780 ACTtCCAACCCCCAACTtTATTTCTTTCTACATTATATAtAACTCTTTtATTATTtTtAA
      **********&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;********************

 3840 ACATCATtATtCCCATtAtTTtTtttTttCtttATtACATttCCtTtAATCTtAAtAAtT
      ************************************************************

 3900 AtACTCCTTTCACTtATtCTtTtAAtATtCCTtCATCAtAttAtAAATAtAtTtATATCA
      ************************************************************

 3960 tCACTtTCTtTCTCATtTTtCTtTAtCTttAATtTTTTTAACAAATAAACTCAtCAAAAA
      ************************************************************

 4020 AAtAAATtTCCCTTTTCAttACCCTtTCTTTCAAAtATAATTCtTAAAAATCCAAATAAC
      ************************************************************

 4080 TTCACAtATCTTCATTTTAAAtttTTTAAACACTtTTTCCCATtCTTCTTCAATtAACCA
      ************************************************************

 4140 TAAACAATTAATtAACATtCACCTtTttAACttTCtTTAAtACACTAACAtCTTACAtAC
      *************************&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;*****

 4200 ttTAttCAtTTAttCCAAAtTTATtAAAACTTAttACACTAAAtAttCCTTTCTACTtAC
      *******************************************************&gt;&gt;&gt;&gt;&gt;

 4260 ACTtAAAAACACCAAAAtAAAtATtTCCAtttTCCCTtCTCATCTtCAttAACtTtCCTT
      &gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;***********************************

 4320 AtACATtCTtCAAttAttCATtAttACTACAtATtTttCCAtttCAATAAATTtCAATtT
      ************************************************************

 4380 CCtTACTtTtAtACACCTAACACAtCtCTACAtttAtACAttACttACAtCTtATCtTCC
      ************************************************************

 4440 TCtCAtTtACAtACCACtTtTAACAACACCTtCACAttATCttTACATCCtAACATCACA
      ************************************************************

 4500 CCTtCtttACAttTACCttATttCAACAACAACTtCCCtAtTTACACCAttAACtCACAA
      ************************************************************

 4560 TCCCTCCATCAtTtCTCAtACTtTCTtCAATAttCTAAtAtAttTTttACTtAtttCTTt
      ************************************************************

 4620 TAttCCTtTTtTCAttCAttTCATCACCttCAACTACtTCtCCTATtttCACAAACCCAC
      ***************************&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;****

 4680 CtTCACTttACCAtACAtCCtCttTTTTtTCTCACCAttttTtATttTCttAATTtCtTT
      ************************************************************

 4740 TATCtTTtAAttAATtAtCtTTACACTtAttCCTtTACTCTttAtCtttATCtATTTttA
      ************************&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;******

 4800 ttTttAtttTCCtTCATttTCTttttCttTtTATCACAtCATCATCttACTtAtCTTtTT
      ************************************************************

 4860 tTCTTTtCAttAAATCTCAACtCTtTtCtTTACAtttAAtACATCCTCCTCCCTCATATt
      ************************************************************

 4920 tTACCCTTCCTtCAttCTCATCCTtACATtACCCTCCAtCATtACAATtCCACCAtCCAT
      ************************************************************

 4980 ACTtCTCATtTTtTtCtTtATTTCCTtCAAtACAttAATtTCAtTtTTCTtCCATttCCA
      ************************************************************

 5040 tCtAAtAtCCCAtATCTCAATCCCATTtAtCACtTCTtttACCTTTTttATCttAtttTt
      ************************************************************

 5100 AtAtCTAtttCCATTCtCCACAtAAATtTCTtttAACTTtCAttTtCCTTttTttAAtAt
      ************************************************************

 5160 TttttTAACATCTCACAtCAAtAACTttCAAATCTttTtCAtTCCATtAttAttAtATtC
      *************&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;*****************

 5220 ACTtCAtTACTTAATtCAtCTttTttCCACACCAtATACTtACTtTTACTTTTtATTTTt
      ************************************************************

 5280 ACCCCACCTTTtTTCAtttACACATTATTCCATTTCTtTTAtTCACATtTCTtTttAACT
      **************&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;****************

 5340 TtTTCAtTTTATtTCTCAtTTtTTtAATATTtTTATtTTCATACAAATATTTACACATtT
      ************************************************************

 5400 TAAtTTtACTtAAAATAAACtCAtTTtACAtTtAtAttACtTTTCTTTTAACAtTtAtAt
      ************************************************************

 5460 tAtTTTAtTATAATTAAtTCAtttACTAttTAACACAAATTACCACCAAACACAATAAtT
      ************************************************************

 5520 ATCAAACTAAACATTtTTATAACTAAAATAAAATCAtttTtACAtATTAAtTAACATAAA
      ************************************************************

 5580 ATATtTCATTAAAAAAAATtACCtTAtACATACAtTACTAtTCAAAAtTTTttACATACT
      ************************************************************

 5640 TAtAttTtTtttTTTTTCTTTATTTTTACTATTTTCTACATTtTAtCATAATAtTtAAtA
      ************************************************************

 5700 CATAAACTATtAAATAACAAATATttAATCATtTAtTAACCAAAATAtTATTTAAAAAAT
      ************************************************************

 5760 CTAAATATATTTCAtATTTTTCAAAtTAtCCACCCTTTtCCTCtATtACAtCTTTtCACT
      ****&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;**************************

 5820 CTCTTttCATTCTCTCTTCACCTttAATtCTTTTCCAAtAtTtTtCAAAACTtTCATCAA
      ************************************************************

 5880 ttCAAAtttTtACTACTTTAAAtAATtTCAAATATAAAATATATTTTtATTTtTTTTATA
      ************************************************************

 5940 CTTTTTtttTTACTACATtATTCCATATtTtTTATTTAATAtTTTTtAAtTCTTCACTAT
      ************************************************************

 6000 TATTCTACAATtTAtAAAATAtTAAAATAAAtAATCCCTttAATAAtTAttTtTtTCCAA
      ************************************************************

 6060 ACTTTTtACTttTACTtTACtTtTTTttTATCATATAtCTAtTAtTTACAAAAtTAAACT
      ************************************************************

 6120 CATTTAAAAtCTTtAAACtTAtAATTTtTCCCTCAtAATTATtTCCACTCTtAAAATCTA
      *******************************&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;

 6180 tTtTCTTATCATtATTTTCAAAtTAAAATATAAAACAAtTTAtTtAAtTAAAATTATATA
      &lt;***********************************************************

 6240 CAtAtATTACTtAtAAACATtCAtCAAAATCATCCCAtAAtAATTACATTAAATTtCATT
      *****************************&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;*

 6300 CATAAAAAATTATtATtAtATtACTtACTCAATATTTttCATTtTTttATtAATtTTATt
      ************************************************************

 6360 CAACATTCAAATttATTTCCCTCTTtTTAtTTAtTACCTtTAATTtttTTtTAttCCTTt
      ************************************************************

 6420 CCtATtTTttTtATtACTCTACTtAAtATCAtTtTAtTAtCAtTACTtAATttACCTACt
      ************************************************************

 6480 TTTCCAtAtTCAtTCAAACCAtCAtAtAAttCCACAtTTttTCTtCCTtTtAAAAtAAtT
      ************************************************************

 6540 TtATTATTTAAAAttACAATCTtAtATTtAAATAACAACAAttCAAATtCTTACTTATCC
      ************************************************************

 6600 ACCTtTATCAATCCAATTTACAAATtCCTtATATTTCTtACATtACCTtCTTTCTCTTTC
      ************************************************************

 6660 TCCAtCTCTTCCACCTTtCTCTTCTttAtCTtCAtTTCAtTCTTttTtACtCTCAtCTCC
      *************&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;******************

 6720 TtTCTCTtTTCCACCATCATtTCTCTCAtCTCCTCCAtCTTAtACCAtATtTCAttttTC
      ***********************************&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;

 6780 TTtTtttCCtTCTCTtTAtCTTCAtCCCATtCCCCAAACAtAtAAAACAtCAtCAtAtCT
      &gt;&gt;&gt;&gt;&gt;*******************************************************

 6840 ACAtCACCTCTCATTCTtAAACtACACATCTTCTtAAATtATCTAtTCTTTtAttCTAtA
      ************************************************************

 6900 AATTTACAttATTAAAtTAttAtCTATATCTtTtTtAAATATATTtTtTCTCTtTttAAA
      **********************************************************  

 6960 AAAtAtttTTCTTAAATATCAtttAtTATTATTTtATTAttATTACTttTAttATTCTTT
                                                                  

 7020 ATTTtTTTTTACTttCAATtAttTtAAttTTTCTTtCtTCATTCACCAtTTtTTtTTACT
                                                                  

 7080 tTATTTCtTCATACTTAATTTTtATAtTCCCCTAtATAtttAtTtttCtTACACATCACA
                                                                  

 7140 tACAAACTAAAATttTCCACACAtACAtTtTttTAAAtAAttAAttCCAAAAAtATCATC
         &lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;                            

 7200 AAttACAACAACCACCCtAtCCACTtCCTtTTCACACTtCTATCATCCAtAAttTtAttT
                                                                  

 7260 CAtTACAttTACATCAAtCTtttACCtAtAtATTtAtAAACAtCTTCTATCTCAAttCCA
                                                                  

 7320 TCAtACTtCTAAACAtCAATCATTAACTtAtAtAtTCTtCTtCCCACATTtAtAACCAAT
                                                                  

 7380 CACAttACACTTTAATAAATttATCACTAtTCACTTTAAACAATtCCACTTTAAATAATt
                                                                  

 7440 tCACTTTAATAATtCTTAAATATCTTACATTACTCATTTCACATtTATATACTtTATTTT
                                                                  

 7500 ATACCATCTACTtCACCTTtCCTATtCCtCTCtCATTAACATCTtCTAACCATtTtTATt
                                                                  

 7560 TtACCAATAAtATTTtATTTtATATCTtTAAATtATCTACAtATtATCAAACTAAtAAtA
                                                                  

 7620 AACTATtAACTtCAACTCTtTTtATATTCAACTtCTTtTTAtttTTATttCAAttTCtAA
                                                                  

 7680 TTttCCtTCTtCCAAATttACTttACTTTTTTTTTTAttTttTTtAtCCttTCtAAATTt
                                                                  

 7740 AAAAAAATTATATTATATACATATATACAAAATCCATATTtCtATAACAATAAATCtAAC
                                                                  

 7800 AATAAtTTTATTACATTAATtTtCATCACAtTTTTTTTATTATCATTTAATCTATTACTA
                                                                  

 7860 TTAtTTTtTTTtTAtTTTtTTTtTTAtCAAACTTATTTAATATCAAACTTCACTTTTCTC
                                                                  

 7920 TAtCATAttCTACTTATCtTAATtAACtACATAAtCAAAATtCTTtCAATAAtTtATCtt
                                                                  

 7980 AATttTTttTtTAtATAATTTTTtTTTTATCtTCCTAtCTtTAtTTCAAAtTCAAACtCt
                                                                  

 8040 CCATCAtTtAtCCAtCCTATATTCCTACTAAAtTTtTtAtttAtAtACAtCTAAATtAtC
                                                                  

 8100 ttAtTtAATTATttATtATtAttACAAAtAACCTtATTCTtTtTtCAAAtACAtTTCAAC
                                                                  

 8160 TtCAtCCAAAttAACAtAAttTAATTCTtTCtAttTTtAAAACAtAtAtACATACTtACA
                                                                  

 8220 tAtATCTCTAtTTttATtTCAAATtTTTTTtCATttACATTtCCATTtAtttCTTCCACC
                                                                  

 8280 ATTTTAAAtTAtTCAACTtttTttttATTTCTATttATTtAtAATtATTAtCCAATtATC
                                                                  

 8340 AtAtAATTATCTTAATCTTCAATTTttATTtCTTATttCTtTACATttCTTTAAtCTTCA
                                                                  

 8400 TCACCACCATtAAtTttCCACAATAATTtAATtACAttACTCCAtCACTtCtttTttCAt
                                                                  

 8460 TAAATCACCAATTTtTTATTTAACACtTTCTCATTTACAtCAACtACCTttttAATAtTT
                                                                  

 8520 ACAttttAtAttAtttttATtAATTAtCCAATTtTAAACTttttATtATTAttTtACCAT
                                                                  

 8580 tATttTATtAtttCCAtATTtttAATTTAtCCAtttCTAAACACCCCTACTCTTATAATA
                                                                  

 8640 ATAATAATAATAATAATAATAATAATAAATtCCATtTtACCACAtAtTCAttACACCCAT
                                                                  

 8700 TTAACATCCCATCCAAAAtACAtCAACCTACACAtttCAATtTCCCCAATCACTtCCTTt
                                                                  

 8760 tttAATTtttATATTTTTTAtAtCAtACtAAAtAtTtCCTCCTACTttCCCtTCCAACAC
                                                                  

 8820 CACTCCCAtCAtCATCTttTCTCCCATCCAtttCCCAACCTTtCTTAtCTTCAtAAtCAA
                                                                  

 8880 tCCAtCAtTtttATttAtttTTtTATtCTTCTttCAAtCTAATtTTTAtTCCTtAACTTA
                                                                  

 8940 TTCAttCTTtCCATAACAAAttttTTtAATACTTTTTtACTCAAtACATTTCAtCTTTTC
                                                                  

 9000 ATTtTAAATACATTTtTAAAAATAAAAAAAATCCAACTAAATTCCACTTTtACATTtTAt
                                                                  

 9060 ttTAATATtTtTAtATCAtTtACACAACATCTACATTTAATCAACTATATATTCAttCTt
                                                                  

 9120 TCACACAACAAAATtTttAAAAAtTCAAttttAtTtAATACTTTTTTTTATAtCTTTtTT
                                                                  

 9180 ATAAACATTATACTACATTATAATAACAACTTtCCAAAAAACTTtAAATttAAACAAATA
                                                                  

 9240 tAATTTTACAACAAAATCAATtTTTCTtttACAAAATAttATTAtTtACAtCTtTAATTA
                                                                  

 9300 AAttAAtATCTTCTttCTACAtTAACAATtACAtCTACATAAAAAAAAAAAAAAAAAtCT
                                                                  

 9360 AtTCAttTAACTACtTAtTTAATtTTtCAATTtTTTACAtCTTTCAtATTTtCTATCATt
                                                                  

 9420 AATTTtTTAAttCCAAtttTAAtAAttAAAATttCATtAATTttAACCTAATtACTAACT
                                                                  

 9480 CCTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTt
                                                                  

 9540 TtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTt
                                                                  

 9600 TtTtTtTtTtTtTtTtTtTtTtTtTtCATACATtTTCATCTATCACTTTtTtTAtTTAtC
                                                                  

 9660 TAtCAATTttCTACTAATtATTTtCCTtTCTACCCAtACTCCTtTCTACCCAtACtCCAA
                                                                  

 9720 ACtTTATtCCACtCCCACttACtTTAtTTTATtTAtTtAACtACttAtCAtCtAtTtTtA
                                                                  

 9780 tTCtTCAttCTAATTAATAtTCCTACTtCAACTtTCTTtATTATtAtTAtATCTACAAAt
                                                                  

 9840 TATtCCTTCCTttCAttATtATATAATTAATATTATTTtTATCAATTtttTCtttATTTt
                                                                  

 9900 TTTAACAAACTCTtCCATCtTttTAtTCTtCAtAAACATTtCTAtCCAAACACAtTCCTC
                                                                  

 9960 TCTCtCTAtttttTTTATTCAtACCCAAtTTATACCCACATAAATATAATTtTATTCTCT
                                                                  

10020 TTTTATtCACTTTTCTCTCTATtCtTATTCTtACCTTtAACTTACTCATttttACAAtCC
                                                                  

10080 AtTtTCAtCAAttTACCCTtttTtAttttTtATACtCCAtTtTATTCTtACCTTtACTTA
                                                                  

10140 ATTCCCTACTAATTCTACAACCTTTATCAttCTAATATTCTtAATATCCTtTTCTCTCtA
                                                                  

10200 TATAtTCCAtTtAtTTTtTCAACAAAATTCAAATtATAttTAAACCtTATATTCAAttAT
                                                                  

10260 ttATAAAtATAAATtTACTtAAAATCCTATTtAATAAAAACTCCACAAtAAAATCTtTTt
                                                                  

10320 tCCAACAAtTtttAtttTTTTCAtCttTTtAATTAAATTTATTCAtAtCACtCCCAtCAt
                                                                  

10380 CAAAtCCCCAtCAtCAAAtACCtTTACACTACTtCATAAttTAAtTCTtAtTttTCCAAA
                                                                  

10440 CACACCAAtAAAtTTtTt
                        

KEYS (in order of precedence):
****** target
&gt;&gt;&gt;&gt;&gt;&gt; left primer
&lt;&lt;&lt;&lt;&lt;&lt; right primer
^^^^^^ left primer / right primer overlap

                    start  len      tm     gc%  any_th  3'_th hairpin seq
 1 LEFT_PRIMER       3265   30   54.89   23.33    0.00   0.00    0.00 ttATATtTCTtAACtACTAACCCTtAtACA
 2 LEFT_PRIMER       3790   30   55.04   23.33    0.00   0.00    0.00 CCCAACTtTATTTCTTTCTACATTATATAt
 3 LEFT_PRIMER       4255   30   55.30   20.00    0.00   0.00    0.00 CTtACACTtAAAAACACCAAAAtAAAtATt
 4 LEFT_PRIMER       4764   30   55.09   23.33    0.00   0.00    0.00 ACTtAttCCTtTACTCTttAtCtttATCtA
 5 LEFT_PRIMER       5294   30   55.72   23.33    0.00   0.00    0.00 CAtttACACATTATTCCATTTCTtTTAtTC
 6 LEFT_PRIMER       5764   30   54.46   20.00    0.00   0.00    0.00 ATATATTTCAtATTTTTCAAAtTAtCCACC
 7 LEFT_PRIMER       6269   30   55.20   20.00    0.00   0.00   31.63 TCATCCCAtAAtAATTACATTAAATTtCAT
 8 LEFT_PRIMER       6755   30   54.98   20.00    0.00   0.00    0.00 CAtCTTAtACCAtATtTCAttttTCTTtTt

                     start  len      tm     gc%  any_th  3'_th hairpin seq
 1 RIGHT_PRIMER      3677   30   55.32   20.00    0.00   0.00   42.14 aaTTaAGTGGaTTaaTGaTaTTaaaAGTGT
 2 RIGHT_PRIMER      4194   30   54.81   20.00    0.00   0.00    0.00 TAAGaTGTTAGTGTaTTAAaGAaaGTTaaA
 3 RIGHT_PRIMER      4675   29   59.24   34.48    0.00   0.00    0.00 GTTTGTGaaaATAGGaGAaGTAGTTGaaG
 4 RIGHT_PRIMER      5202   30   57.40   23.33    0.00   0.00    0.00 AaTGaAaaAGATTTGaaAGTTaTTGaTGTG
 5 RIGHT_PRIMER      6180   30   55.05   23.33    0.00   0.00    0.00 aTAGATTTTaAGAGTGGAaATAATTaTGAG
 6 RIGHT_PRIMER      6701   29   57.64   31.03    0.00   0.00    0.00 GAaTGAAaTGaAGaTaaAGAAGAGaAAGG
 7 RIGHT_PRIMER      7171   29   56.70   20.69    2.75   0.00    0.00 aAaAaTGTaTGTGTGGAaaATTTTAGTTT

Statistics
         con   too    in    in   not          no    tm    tm   high  high  high        high      
         sid  many   tar  excl    ok   bad    GC   too   too any_th 3'_th hair-  poly   end      
        ered    Ns   get   reg   reg   GC% clamp   low  high  compl compl   pin     X  stab    ok
Left    3303     0     0     0     0   974     0     9   185      0     0     0    61     0     8
Right   3607     0     0     0     0  1486     0     3   460      0     0     0    34     0     7
Pair Stats:
considered 0, ok 0
libprimer3 release 2.6.1




----------------------------------------------------------------------------------

Primer default (i.e. computer-friendly) results file:


SEQUENCE_ID=LOC120027825
SEQUENCE_TEMPLATE=TttttCTCtCCAttACAAttttAtCCCTtAAAAtATtTtAAtTTACACCCCAtCTACCCAtTCACCtTCTTTCACTATCCtCAACCCTCCACTtttACAATCtTCAACAATACATTCAAtCCTTCATtTACTCCttttCCtCtAATAATTTCAtttACCAAtACTTCtCCAtCtAATTACAAATCCCCTtCtTtAAATtTCCCATCCCTCTCCATCTtCTCCttCCAtTTttttTATCAtATCAAtCCCATTCTACTtCAttTTttttTtAACCACTCAtACTCTTAtTTTTCTTTTtATCACTtCTCCCtAtAACCTtCTTATCCTAtttTACCCCTttCTAtTtCTACATAACCCACAATTTTTCTttTCCACtttACACATtCTAtACTttttTTtTTCAACATATtACATTAAAATtACAtATCtCTATAATAtTTtTACACTCATTAAAttCCCAtTtCAtTCAACtTTATtTtATCAtTtTTATTTCCTtATAtTTtCTttTTtAAAATACAATCTACACTttACCTTTTAATCAtCtttTTTtCATtttTtttAtTTTTttCTTTCCATtATtACATCACCATtCttTAAATTtATTAATAtACCAATAACAtAtTTCCAAACTTCTTTtCCAATAACAtCAATTTTTCAtTTTTTCCCTCCCCACTCAAACCACTCCCAtACAtTCCTTtCAAAATTCTTtTTTtTtAATtTTTTTTtTTtTTtCTAAAAAtCTATTCTTTCCCATTTTAATttAACTCTATTACAtTAATtTACTTAATTtTTACCCAtAAATtATTTtATATTtTTATAAAAACttCTtCAtTttACCTTTAATCTtCATCAAAATATCTtTTCAATAtAAATTtAtAAtATAAAATACACtCTCtCTtTCTTTATTtACATttTCAtCTtTttATtCTTCCTTTttCTtAttAATtACACCATtTTttCTTATtACCAttCATAtACCACTATCTAATCAACCTttACTCtttTAtACATAAAtATtAATCCtttACACTCCAATTAtTATtATATTTTACtTTTCtTATttTATtTATTAtTTTtTttATtTCCATCATCCATTTTtTATtATATtTTACAAATTtCAATTCATACAATATtTTAAAAATTtCTATTTtTACAATATtTTACtAATTTtTAAACTTATtAAATtTTATtAATTCTAATTTtTTtTtttTAACATTAtTTACtTttTTAAAtCTAACATTAtCCTttTttCTAATtTTAtCTATtTTAttttTTAtttTCAtttTTtAAttTTAtAtTTtATtTttATATtAAtCTAtttTTACttTTCAAATTAtAtTTtAtCTttAATtTtTACATAAAtCTAtttTTATttATTTTtATtAATTAtATCAtCCTttTCTCATAtACTAtACtTAAATCCttAtCAtATAtTtTtAtCTCAAAtTATTTttACAtTtACACATTTtTTtTTtTTTTAtCTCTtAtCTCCAtCACTTTttATTTtAAATtATACTATTAttTTAtTtCtCAtACTtACAtCTTTAATTTtAtttTATTTTCATCCATATCAttTtAACCtTTTAtAAAAAtTtCTtTCACCAAAAtTATTttAACAATTTTACTTATATtttTATTAAAtTAtTAAAAAtATAAtTATTTttACCCATATTCCTAtCACtCAATtACTACATCAAtCtTtTtACTCTACACATTTtTTttATtCATTTtCTtTTTtTTTTAtTTATtTTTCAtATTATTTTtTtCCCAATAtAAACAAATttTAAATAATtTATTtTtTAATTTTttAtTCACTTTTAtAtAAAtTTACAtACtCATAAATATCATAACCCCCCAAAAATtCTAAtCTCCCCTtTTATTtTAATttTtAtAttTTAtCATtTCTTttttTTATtATATTTtTtCATCTtTtACTTTCTCACTCATCATTATTCACTATTCATTCATTATTATCTtTAATCATttTAtCATCCACAATtTAtCAtTTTTTAtAAACCTATTCTTATTTACATTATTTTTTTAACTTAtCATtTTAtCTAACCCTAACCTTAATCCAACTCCTCCTAACTCCTTAACCTTCAAtTTTATTCAAAtTTTAtTCTTAtAAACACAtTtCATtAtAATTTAtCATtTtTTTCTtAAttAAAAACTTAAACtATtTCAAtCTCTCAtATAACACAAAACACTtttTtAtTTTTCCTTATTtCAttTCATTTttATCAtACTCTtTAAAACACTtTAttAAAACAACCCTtTTTtTTTTTTTttTCTCTTTTTATCttCATtAAATTtTtATAATATTtTAAttTTtttTTtTtCCtTttCttATATCTTTtTtttCTATACTCttCCTTtTCTtAttATttTAttTTttTttTTtAAtAAATCCCTCTAtTttTtTtttttCTtTtCTTTttCAAAtTtttTttttTTATATCCTTCCTtTTTttCCCTtTCCtttttTATCATCttATttttCCACAtTtTCTCCTtACCCCTCCTtTCTCAtTATTTATtCTtCAtTAtTTTATtTtTCttttttCTAtttTCAtTTTtTTATATCTttAtTACTTCTCCTtTCTTATCCttTtTCCTtTtTtAATTTAAtTATtCTCTCTCTAATTCTCTCTTTCTTTCTCTCTCTCttAttACCTtAtCCCTAttACCATtCCTCAttACTACCTttCACtATtACTCCTTtCTtTCCCCAtTCCACCTttCCtTtCTtCTtCTCCAtTTTCAACTtTTCTtCCTtTttCTATttAACCCTtACCTtTTCACCAtACtTtCTACCTtTCCCAtACCCtCTtTTTTCAACTCTCTAtAtACAtCAttAtCttTAtAtATACTCTTAATtATCttCTATtAAAAtCCAACTtACATTTACTTCTtAttTtCTtACTTtCTtCACCCTCtACAACTACTtTtATTATTATTATTTtACCATtCTttTCATTTTtAACATTTtAACATCTTttCCATtTTCTtTTATAATCTCCACCCttCACAtCCAAAAtAttACTttCCACCCCTCATAtCCTttTTCCTCTCTAttTTTCTTCCTAttTTTTttCCTTTCTAtttAtTTTTTCCTAtCCACCtTtCTTCTACACCTtCATTtCTTtCTtTTTttttTTTTAttCTtttTTTCTtTACAtCACTTTtAtATATCAtCTtATtTAAtAAtttCTATATAAATACATTTtATTTtATTTtATAAttTCACTAtTTACTTATTAAATATttATATtTCTtAACtACTAACCCTtAtACACATCAtAAAtCTtAtTTCACtTCATAATATTtTtATTATAATCACAATCCCTTCTTACACCACTATAAATtTtTCTCTtTCCTTATCTtAtTCTCTtAATtTAtCCTATtTtTtACCTtAATtTATCTtTTACCTTAATtTAATtTAtCCTACATtTtTTACTTtAAAAATAACCCTAAATTttTTCTTATTCCCAACACAtCTCTtttATtTtTAATttTCATATTTATTACCATACCTCTCCACTTCTTCCTCTTTCCCTTTACAATTTTTATTTTTTTTAAATAATAAAtTtATtATTTATTTATTttTTAAAtAttAAAATCCTCACATCACAtAATttTtCTtTCCTCACACTtttAAtAtCAttAAtCCACTtAAttTtTTtTttTTATCATttCTATCATttAtACCCTttTCTTCAtttAtACtAAttTAtACCACATCCTCCTTCTCCAtCTCTAtTtTCAACtCATTAtATATtTACTtCCAACCCCCAACTtTATTTCTTTCTACATTATATAtAACTCTTTtATTATTtTtAAACATCATtATtCCCATtAtTTtTtttTttCtttATtACATttCCtTtAATCTtAAtAAtTAtACTCCTTTCACTtATtCTtTtAAtATtCCTtCATCAtAttAtAAATAtAtTtATATCAtCACTtTCTtTCTCATtTTtCTtTAtCTttAATtTTTTTAACAAATAAACTCAtCAAAAAAAtAAATtTCCCTTTTCAttACCCTtTCTTTCAAAtATAATTCtTAAAAATCCAAATAACTTCACAtATCTTCATTTTAAAtttTTTAAACACTtTTTCCCATtCTTCTTCAATtAACCATAAACAATTAATtAACATtCACCTtTttAACttTCtTTAAtACACTAACAtCTTACAtACttTAttCAtTTAttCCAAAtTTATtAAAACTTAttACACTAAAtAttCCTTTCTACTtACACTtAAAAACACCAAAAtAAAtATtTCCAtttTCCCTtCTCATCTtCAttAACtTtCCTTAtACATtCTtCAAttAttCATtAttACTACAtATtTttCCAtttCAATAAATTtCAATtTCCtTACTtTtAtACACCTAACACAtCtCTACAtttAtACAttACttACAtCTtATCtTCCTCtCAtTtACAtACCACtTtTAACAACACCTtCACAttATCttTACATCCtAACATCACACCTtCtttACAttTACCttATttCAACAACAACTtCCCtAtTTACACCAttAACtCACAATCCCTCCATCAtTtCTCAtACTtTCTtCAATAttCTAAtAtAttTTttACTtAtttCTTtTAttCCTtTTtTCAttCAttTCATCACCttCAACTACtTCtCCTATtttCACAAACCCACCtTCACTttACCAtACAtCCtCttTTTTtTCTCACCAttttTtATttTCttAATTtCtTTTATCtTTtAAttAATtAtCtTTACACTtAttCCTtTACTCTttAtCtttATCtATTTttAttTttAtttTCCtTCATttTCTttttCttTtTATCACAtCATCATCttACTtAtCTTtTTtTCTTTtCAttAAATCTCAACtCTtTtCtTTACAtttAAtACATCCTCCTCCCTCATATttTACCCTTCCTtCAttCTCATCCTtACATtACCCTCCAtCATtACAATtCCACCAtCCATACTtCTCATtTTtTtCtTtATTTCCTtCAAtACAttAATtTCAtTtTTCTtCCATttCCAtCtAAtAtCCCAtATCTCAATCCCATTtAtCACtTCTtttACCTTTTttATCttAtttTtAtAtCTAtttCCATTCtCCACAtAAATtTCTtttAACTTtCAttTtCCTTttTttAAtAtTttttTAACATCTCACAtCAAtAACTttCAAATCTttTtCAtTCCATtAttAttAtATtCACTtCAtTACTTAATtCAtCTttTttCCACACCAtATACTtACTtTTACTTTTtATTTTtACCCCACCTTTtTTCAtttACACATTATTCCATTTCTtTTAtTCACATtTCTtTttAACTTtTTCAtTTTATtTCTCAtTTtTTtAATATTtTTATtTTCATACAAATATTTACACATtTTAAtTTtACTtAAAATAAACtCAtTTtACAtTtAtAttACtTTTCTTTTAACAtTtAtAttAtTTTAtTATAATTAAtTCAtttACTAttTAACACAAATTACCACCAAACACAATAAtTATCAAACTAAACATTtTTATAACTAAAATAAAATCAtttTtACAtATTAAtTAACATAAAATATtTCATTAAAAAAAATtACCtTAtACATACAtTACTAtTCAAAAtTTTttACATACTTAtAttTtTtttTTTTTCTTTATTTTTACTATTTTCTACATTtTAtCATAATAtTtAAtACATAAACTATtAAATAACAAATATttAATCATtTAtTAACCAAAATAtTATTTAAAAAATCTAAATATATTTCAtATTTTTCAAAtTAtCCACCCTTTtCCTCtATtACAtCTTTtCACTCTCTTttCATTCTCTCTTCACCTttAATtCTTTTCCAAtAtTtTtCAAAACTtTCATCAAttCAAAtttTtACTACTTTAAAtAATtTCAAATATAAAATATATTTTtATTTtTTTTATACTTTTTtttTTACTACATtATTCCATATtTtTTATTTAATAtTTTTtAAtTCTTCACTATTATTCTACAATtTAtAAAATAtTAAAATAAAtAATCCCTttAATAAtTAttTtTtTCCAAACTTTTtACTttTACTtTACtTtTTTttTATCATATAtCTAtTAtTTACAAAAtTAAACTCATTTAAAAtCTTtAAACtTAtAATTTtTCCCTCAtAATTATtTCCACTCTtAAAATCTAtTtTCTTATCATtATTTTCAAAtTAAAATATAAAACAAtTTAtTtAAtTAAAATTATATACAtAtATTACTtAtAAACATtCAtCAAAATCATCCCAtAAtAATTACATTAAATTtCATTCATAAAAAATTATtATtAtATtACTtACTCAATATTTttCATTtTTttATtAATtTTATtCAACATTCAAATttATTTCCCTCTTtTTAtTTAtTACCTtTAATTtttTTtTAttCCTTtCCtATtTTttTtATtACTCTACTtAAtATCAtTtTAtTAtCAtTACTtAATttACCTACtTTTCCAtAtTCAtTCAAACCAtCAtAtAAttCCACAtTTttTCTtCCTtTtAAAAtAAtTTtATTATTTAAAAttACAATCTtAtATTtAAATAACAACAAttCAAATtCTTACTTATCCACCTtTATCAATCCAATTTACAAATtCCTtATATTTCTtACATtACCTtCTTTCTCTTTCTCCAtCTCTTCCACCTTtCTCTTCTttAtCTtCAtTTCAtTCTTttTtACtCTCAtCTCCTtTCTCTtTTCCACCATCATtTCTCTCAtCTCCTCCAtCTTAtACCAtATtTCAttttTCTTtTtttCCtTCTCTtTAtCTTCAtCCCATtCCCCAAACAtAtAAAACAtCAtCAtAtCTACAtCACCTCTCATTCTtAAACtACACATCTTCTtAAATtATCTAtTCTTTtAttCTAtAAATTTACAttATTAAAtTAttAtCTATATCTtTtTtAAATATATTtTtTCTCTtTttAAAAAAtAtttTTCTTAAATATCAtttAtTATTATTTtATTAttATTACTttTAttATTCTTTATTTtTTTTTACTttCAATtAttTtAAttTTTCTTtCtTCATTCACCAtTTtTTtTTACTtTATTTCtTCATACTTAATTTTtATAtTCCCCTAtATAtttAtTtttCtTACACATCACAtACAAACTAAAATttTCCACACAtACAtTtTttTAAAtAAttAAttCCAAAAAtATCATCAAttACAACAACCACCCtAtCCACTtCCTtTTCACACTtCTATCATCCAtAAttTtAttTCAtTACAttTACATCAAtCTtttACCtAtAtATTtAtAAACAtCTTCTATCTCAAttCCATCAtACTtCTAAACAtCAATCATTAACTtAtAtAtTCTtCTtCCCACATTtAtAACCAATCACAttACACTTTAATAAATttATCACTAtTCACTTTAAACAATtCCACTTTAAATAATttCACTTTAATAATtCTTAAATATCTTACATTACTCATTTCACATtTATATACTtTATTTTATACCATCTACTtCACCTTtCCTATtCCtCTCtCATTAACATCTtCTAACCATtTtTATtTtACCAATAAtATTTtATTTtATATCTtTAAATtATCTACAtATtATCAAACTAAtAAtAAACTATtAACTtCAACTCTtTTtATATTCAACTtCTTtTTAtttTTATttCAAttTCtAATTttCCtTCTtCCAAATttACTttACTTTTTTTTTTAttTttTTtAtCCttTCtAAATTtAAAAAAATTATATTATATACATATATACAAAATCCATATTtCtATAACAATAAATCtAACAATAAtTTTATTACATTAATtTtCATCACAtTTTTTTTATTATCATTTAATCTATTACTATTAtTTTtTTTtTAtTTTtTTTtTTAtCAAACTTATTTAATATCAAACTTCACTTTTCTCTAtCATAttCTACTTATCtTAATtAACtACATAAtCAAAATtCTTtCAATAAtTtATCttAATttTTttTtTAtATAATTTTTtTTTTATCtTCCTAtCTtTAtTTCAAAtTCAAACtCtCCATCAtTtAtCCAtCCTATATTCCTACTAAAtTTtTtAtttAtAtACAtCTAAATtAtCttAtTtAATTATttATtATtAttACAAAtAACCTtATTCTtTtTtCAAAtACAtTTCAACTtCAtCCAAAttAACAtAAttTAATTCTtTCtAttTTtAAAACAtAtAtACATACTtACAtAtATCTCTAtTTttATtTCAAATtTTTTTtCATttACATTtCCATTtAtttCTTCCACCATTTTAAAtTAtTCAACTtttTttttATTTCTATttATTtAtAATtATTAtCCAATtATCAtAtAATTATCTTAATCTTCAATTTttATTtCTTATttCTtTACATttCTTTAAtCTTCATCACCACCATtAAtTttCCACAATAATTtAATtACAttACTCCAtCACTtCtttTttCAtTAAATCACCAATTTtTTATTTAACACtTTCTCATTTACAtCAACtACCTttttAATAtTTACAttttAtAttAtttttATtAATTAtCCAATTtTAAACTttttATtATTAttTtACCATtATttTATtAtttCCAtATTtttAATTTAtCCAtttCTAAACACCCCTACTCTTATAATAATAATAATAATAATAATAATAATAATAAATtCCATtTtACCACAtAtTCAttACACCCATTTAACATCCCATCCAAAAtACAtCAACCTACACAtttCAATtTCCCCAATCACTtCCTTttttAATTtttATATTTTTTAtAtCAtACtAAAtAtTtCCTCCTACTttCCCtTCCAACACCACTCCCAtCAtCATCTttTCTCCCATCCAtttCCCAACCTTtCTTAtCTTCAtAAtCAAtCCAtCAtTtttATttAtttTTtTATtCTTCTttCAAtCTAATtTTTAtTCCTtAACTTATTCAttCTTtCCATAACAAAttttTTtAATACTTTTTtACTCAAtACATTTCAtCTTTTCATTtTAAATACATTTtTAAAAATAAAAAAAATCCAACTAAATTCCACTTTtACATTtTAtttTAATATtTtTAtATCAtTtACACAACATCTACATTTAATCAACTATATATTCAttCTtTCACACAACAAAATtTttAAAAAtTCAAttttAtTtAATACTTTTTTTTATAtCTTTtTTATAAACATTATACTACATTATAATAACAACTTtCCAAAAAACTTtAAATttAAACAAATAtAATTTTACAACAAAATCAATtTTTCTtttACAAAATAttATTAtTtACAtCTtTAATTAAAttAAtATCTTCTttCTACAtTAACAATtACAtCTACATAAAAAAAAAAAAAAAAAtCTAtTCAttTAACTACtTAtTTAATtTTtCAATTtTTTACAtCTTTCAtATTTtCTATCATtAATTTtTTAAttCCAAtttTAAtAAttAAAATttCATtAATTttAACCTAATtACTAACTCCTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtTtCATACATtTTCATCTATCACTTTtTtTAtTTAtCTAtCAATTttCTACTAATtATTTtCCTtTCTACCCAtACTCCTtTCTACCCAtACtCCAAACtTTATtCCACtCCCACttACtTTAtTTTATtTAtTtAACtACttAtCAtCtAtTtTtAtTCtTCAttCTAATTAATAtTCCTACTtCAACTtTCTTtATTATtAtTAtATCTACAAAtTATtCCTTCCTttCAttATtATATAATTAATATTATTTtTATCAATTtttTCtttATTTtTTTAACAAACTCTtCCATCtTttTAtTCTtCAtAAACATTtCTAtCCAAACACAtTCCTCTCTCtCTAtttttTTTATTCAtACCCAAtTTATACCCACATAAATATAATTtTATTCTCTTTTTATtCACTTTTCTCTCTATtCtTATTCTtACCTTtAACTTACTCATttttACAAtCCAtTtTCAtCAAttTACCCTtttTtAttttTtATACtCCAtTtTATTCTtACCTTtACTTAATTCCCTACTAATTCTACAACCTTTATCAttCTAATATTCTtAATATCCTtTTCTCTCtATATAtTCCAtTtAtTTTtTCAACAAAATTCAAATtATAttTAAACCtTATATTCAAttATttATAAAtATAAATtTACTtAAAATCCTATTtAATAAAAACTCCACAAtAAAATCTtTTttCCAACAAtTtttAtttTTTTCAtCttTTtAATTAAATTTATTCAtAtCACtCCCAtCAtCAAAtCCCCAtCAtCAAAtACCtTTACACTACTtCATAAttTAAtTCTtAtTttTCCAAACACACCAAtAAAtTTtTt
PRIMER_TASK=pick_sequencing_primers
SEQUENCE_TARGET=3500,3458
PRIMER_MIN_TM=50
PRIMER_OPT_TM=55
PRIMER_PICK_LEFT_PRIMER=1
PRIMER_PICK_RIGHT_PRIMER=1
PRIMER_OPT_SIZE=30
PRIMER_MIN_SIZE=25
PRIMER_MAX_SIZE=36
PRIMER_MAX_NS_ACCEPTED=0
P3_FILE_FLAG=1
PRIMER_EXPLAIN_FLAG=1
PRIMER_THERMODYNAMIC_PARAMETERS_PATH=/home/shared/primer3-2.6.1/src/primer3_config
PRIMER_NUM_RETURN=10
PRIMER_PAIR_EXPLAIN=considered 0, ok 0
PRIMER_WARNING=No right primer found in range 5634 - 5674
PRIMER_LEFT_EXPLAIN=sequencing locations 8, considered 3303, GC content failed 974, low tm 9, high tm 185, long poly-x seq 61, ok 8
PRIMER_RIGHT_EXPLAIN=sequencing locations 8, considered 3607, GC content failed 1486, low tm 3, high tm 460, long poly-x seq 34, ok 7
PRIMER_PAIR_EXPLAIN=considered 0, ok 0
PRIMER_LEFT_NUM_RETURNED=8
PRIMER_RIGHT_NUM_RETURNED=7
PRIMER_INTERNAL_NUM_RETURNED=0
PRIMER_PAIR_NUM_RETURNED=0
PRIMER_LEFT_0_PENALTY=0.111457
PRIMER_RIGHT_0_PENALTY=0.319138
PRIMER_LEFT_0_SEQUENCE=ttATATtTCTtAACtACTAACCCTtAtACA
PRIMER_RIGHT_0_SEQUENCE=aaTTaAGTGGaTTaaTGaTaTTaaaAGTGT
PRIMER_LEFT_0=3265,30
PRIMER_RIGHT_0=3677,30
PRIMER_LEFT_0_TM=54.889
PRIMER_RIGHT_0_TM=55.319
PRIMER_LEFT_0_GC_PERCENT=23.333
PRIMER_RIGHT_0_GC_PERCENT=20.000
PRIMER_LEFT_0_SELF_ANY_TH=0.00
PRIMER_RIGHT_0_SELF_ANY_TH=0.00
PRIMER_LEFT_0_SELF_END_TH=0.00
PRIMER_RIGHT_0_SELF_END_TH=0.00
PRIMER_LEFT_0_HAIRPIN_TH=0.00
PRIMER_RIGHT_0_HAIRPIN_TH=42.14
PRIMER_LEFT_0_END_STABILITY=2.2900
PRIMER_RIGHT_0_END_STABILITY=3.5500
PRIMER_LEFT_1_PENALTY=0.036926
PRIMER_RIGHT_1_PENALTY=0.190778
PRIMER_LEFT_1_SEQUENCE=CCCAACTtTATTTCTTTCTACATTATATAt
PRIMER_RIGHT_1_SEQUENCE=TAAGaTGTTAGTGTaTTAAaGAaaGTTaaA
PRIMER_LEFT_1=3790,30
PRIMER_RIGHT_1=4194,30
PRIMER_LEFT_1_TM=55.037
PRIMER_RIGHT_1_TM=54.809
PRIMER_LEFT_1_GC_PERCENT=23.333
PRIMER_RIGHT_1_GC_PERCENT=20.000
PRIMER_LEFT_1_SELF_ANY_TH=0.00
PRIMER_RIGHT_1_SELF_ANY_TH=0.00
PRIMER_LEFT_1_SELF_END_TH=0.00
PRIMER_RIGHT_1_SELF_END_TH=0.00
PRIMER_LEFT_1_HAIRPIN_TH=0.00
PRIMER_RIGHT_1_HAIRPIN_TH=0.00
PRIMER_LEFT_1_END_STABILITY=0.8600
PRIMER_RIGHT_1_END_STABILITY=1.5200
PRIMER_LEFT_2_PENALTY=0.297844
PRIMER_RIGHT_2_PENALTY=5.242954
PRIMER_LEFT_2_SEQUENCE=CTtACACTtAAAAACACCAAAAtAAAtATt
PRIMER_RIGHT_2_SEQUENCE=GTTTGTGaaaATAGGaGAaGTAGTTGaaG
PRIMER_LEFT_2=4255,30
PRIMER_RIGHT_2=4675,29
PRIMER_LEFT_2_TM=55.298
PRIMER_RIGHT_2_TM=59.243
PRIMER_LEFT_2_GC_PERCENT=20.000
PRIMER_RIGHT_2_GC_PERCENT=34.483
PRIMER_LEFT_2_SELF_ANY_TH=0.00
PRIMER_RIGHT_2_SELF_ANY_TH=0.00
PRIMER_LEFT_2_SELF_END_TH=0.00
PRIMER_RIGHT_2_SELF_END_TH=0.00
PRIMER_LEFT_2_HAIRPIN_TH=0.00
PRIMER_RIGHT_2_HAIRPIN_TH=0.00
PRIMER_LEFT_2_END_STABILITY=1.2800
PRIMER_RIGHT_2_END_STABILITY=3.0200
PRIMER_LEFT_3_PENALTY=0.093240
PRIMER_RIGHT_3_PENALTY=2.400939
PRIMER_LEFT_3_SEQUENCE=ACTtAttCCTtTACTCTttAtCtttATCtA
PRIMER_RIGHT_3_SEQUENCE=AaTGaAaaAGATTTGaaAGTTaTTGaTGTG
PRIMER_LEFT_3=4764,30
PRIMER_RIGHT_3=5202,30
PRIMER_LEFT_3_TM=55.093
PRIMER_RIGHT_3_TM=57.401
PRIMER_LEFT_3_GC_PERCENT=23.333
PRIMER_RIGHT_3_GC_PERCENT=23.333
PRIMER_LEFT_3_SELF_ANY_TH=0.00
PRIMER_RIGHT_3_SELF_ANY_TH=0.00
PRIMER_LEFT_3_SELF_END_TH=0.00
PRIMER_RIGHT_3_SELF_END_TH=0.00
PRIMER_LEFT_3_HAIRPIN_TH=0.00
PRIMER_RIGHT_3_HAIRPIN_TH=0.00
PRIMER_LEFT_3_END_STABILITY=1.9800
PRIMER_RIGHT_3_END_STABILITY=3.2100
PRIMER_LEFT_4_PENALTY=0.721992
PRIMER_RIGHT_4_PENALTY=0.050560
PRIMER_LEFT_4_SEQUENCE=CAtttACACATTATTCCATTTCTtTTAtTC
PRIMER_RIGHT_4_SEQUENCE=aTAGATTTTaAGAGTGGAaATAATTaTGAG
PRIMER_LEFT_4=5294,30
PRIMER_RIGHT_4=6180,30
PRIMER_LEFT_4_TM=55.722
PRIMER_RIGHT_4_TM=55.051
PRIMER_LEFT_4_GC_PERCENT=23.333
PRIMER_RIGHT_4_GC_PERCENT=23.333
PRIMER_LEFT_4_SELF_ANY_TH=0.00
PRIMER_RIGHT_4_SELF_ANY_TH=0.00
PRIMER_LEFT_4_SELF_END_TH=0.00
PRIMER_RIGHT_4_SELF_END_TH=0.00
PRIMER_LEFT_4_HAIRPIN_TH=0.00
PRIMER_RIGHT_4_HAIRPIN_TH=0.00
PRIMER_LEFT_4_END_STABILITY=1.7500
PRIMER_RIGHT_4_END_STABILITY=2.9000
PRIMER_LEFT_5_PENALTY=0.543198
PRIMER_RIGHT_5_PENALTY=3.641468
PRIMER_LEFT_5_SEQUENCE=ATATATTTCAtATTTTTCAAAtTAtCCACC
PRIMER_RIGHT_5_SEQUENCE=GAaTGAAaTGaAGaTaaAGAAGAGaAAGG
PRIMER_LEFT_5=5764,30
PRIMER_RIGHT_5=6701,29
PRIMER_LEFT_5_TM=54.457
PRIMER_RIGHT_5_TM=57.641
PRIMER_LEFT_5_GC_PERCENT=20.000
PRIMER_RIGHT_5_GC_PERCENT=31.034
PRIMER_LEFT_5_SELF_ANY_TH=0.00
PRIMER_RIGHT_5_SELF_ANY_TH=0.00
PRIMER_LEFT_5_SELF_END_TH=0.00
PRIMER_RIGHT_5_SELF_END_TH=0.00
PRIMER_LEFT_5_HAIRPIN_TH=0.00
PRIMER_RIGHT_5_HAIRPIN_TH=0.00
PRIMER_LEFT_5_END_STABILITY=4.6100
PRIMER_RIGHT_5_END_STABILITY=3.1100
PRIMER_LEFT_6_PENALTY=0.195406
PRIMER_RIGHT_6_PENALTY=2.701604
PRIMER_LEFT_6_SEQUENCE=TCATCCCAtAAtAATTACATTAAATTtCAT
PRIMER_RIGHT_6_SEQUENCE=aAaAaTGTaTGTGTGGAaaATTTTAGTTT
PRIMER_LEFT_6=6269,30
PRIMER_RIGHT_6=7171,29
PRIMER_LEFT_6_TM=55.195
PRIMER_RIGHT_6_TM=56.702
PRIMER_LEFT_6_GC_PERCENT=20.000
PRIMER_RIGHT_6_GC_PERCENT=20.690
PRIMER_LEFT_6_SELF_ANY_TH=0.00
PRIMER_RIGHT_6_SELF_ANY_TH=2.75
PRIMER_LEFT_6_SELF_END_TH=0.00
PRIMER_RIGHT_6_SELF_END_TH=0.00
PRIMER_LEFT_6_HAIRPIN_TH=31.63
PRIMER_RIGHT_6_HAIRPIN_TH=0.00
PRIMER_LEFT_6_END_STABILITY=2.5700
PRIMER_RIGHT_6_END_STABILITY=2.6600
PRIMER_LEFT_7_PENALTY=0.018465
PRIMER_LEFT_7_SEQUENCE=CAtCTTAtACCAtATtTCAttttTCTTtTt
PRIMER_LEFT_7=6755,30
PRIMER_LEFT_7_TM=54.982
PRIMER_LEFT_7_GC_PERCENT=20.000
PRIMER_LEFT_7_SELF_ANY_TH=0.00
PRIMER_LEFT_7_SELF_END_TH=0.00
PRIMER_LEFT_7_HAIRPIN_TH=0.00
PRIMER_LEFT_7_END_STABILITY=1.9400
=</code></pre>
<p>Looks like we have two primers we can use for sequencing. Neither overlaps a potential <code>CpG</code>, and both are outside of the C1q gene sequence.</p>
<p>Next, we need to test them against the entire bisulfite-converted genome to assess specificity.</p>
</section>
</section>
</section>
<section id="genome-bisulfite-conversion" class="level1">
<h1>6 GENOME BISULFITE CONVERSION</h1>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="fu">tr</span> <span class="st">'[:lower:]'</span> <span class="st">'[:upper:]'</span> <span class="op">&lt;</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">/</span><span class="va">${genome_fasta}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'/^&gt;/!s/G/T/g'</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">/</span><span class="va">${bisulfite_genome_fasta}</span><span class="st">"</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="st">"</span><span class="va">${data_dir}</span><span class="st">/</span><span class="va">${bisulfite_genome_fasta}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>&gt;NC_052307.1 SALVELINUS NAMAYCUSH ISOLATE SENECA CHROMOSOME 1, SANAMA_1.0, WHOLE GENOME SHOTGUN SEQUENCE
TTTTTAATTAATTATTTTACAATTTTTTTAAAACATAATTACTTTTTATTTTTATTCCACAATATAATTTAAATTACAAA
TCTTTACTCCTATTATATTCTATATATTACCACATTATTATTCATAAAACTTATCTTTCAAACAAAAATCAACTTTTAAT
ATACATTTCACATCTCATATATATCAATTCCATACATTTTTTTTTCTACTTCCTCATTTTTTTTTTTCTACTTCCTCATT
TTTTTTTTTCTACTTCCTCATTTTTTTTTTTCTACTTCCTCATTTTTTTTTTTCTACTTCCTCATTTTTTTTTTTCTACT
TCCTCATTTTTTTTTTTCTACTTCCTCATTTTTTTTTTTCTACTTCCTCATTTTCTACTTCCTCATTTTTTTTTTTCTAC
ATCCTCATTTTTTTTTTTCTACTTCTTTTTTTTCTACTTCCTCATTTTTTTTTTTCTACATCCTCATTTTTTTTTTTCTA
CTTCCTCATTTTTTTTTTTCTACTTCATCATTTTTTTTTTTCTACTTCTTTTTTTTCTACATCCTCATTTTTTTTTTTCT
ACTTCCTCATTTTTTTTTTTCTACTTCCTCATTTTTTTTTTTCTACTTCCTCATTTTTTTTTTTCTACTTCCTCATTTTT
TTTTTTCTACTTCTTTTTTTTCTACTTCCTCATTTTTTTTTTTCTACTTCCTCATTTTTTTTTTTCTACTTCTTTTTTTT</code></pre>
</section>
<section id="split-genome" class="level1">
<h1>7 SPLIT GENOME</h1>
<p>For some reason, I feel like the EMBOSS PrimerSearch tool will <em>not</em> work on mulit-FastA files (or, it makes the results more difficult to decipher?), but I’m just going off of memory and repeating a process I’ve done previously.</p>
<p>This will split the bisulfite-converted genome multi-FastA file into separate FastA files.</p>
<section id="split-mulit-fasta-file-in-to-individual-fasta-files-with-pyfaidx" class="level2">
<h2 class="anchored" data-anchor-id="split-mulit-fasta-file-in-to-individual-fasta-files-with-pyfaidx">7.1 Split mulit-FastA file in to individual FastA files with PyFaidx</h2>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Make directory if it doesn't exist</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">--parents</span> <span class="va">${genome_fasta_splits_dir}</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${genome_fasta_splits_dir}</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Count sequences in FastA</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------------------------------------------"</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"NUMBER OF SEQUENCES IN ORIGINAL FASTA"</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="st">"^&gt;"</span> ../<span class="va">${bisulfite_genome_fasta}</span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------------------------------------------"</span></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Split FastA</span></span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a><span class="va">${pyfaidx}</span> <span class="dt">\</span></span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>--split-files <span class="dt">\</span></span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>../<span class="va">${bisulfite_genome_fasta}</span></span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Count number of individual FastA files</span></span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------------------------------------------"</span></span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"NUMBER OF INDIVIDUAL FASTA FILES"</span></span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span>
<span id="cb23-28"><a href="#cb23-28" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------------------------------------------"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>-------------------------------------------------------------------
NUMBER OF SEQUENCES IN ORIGINAL FASTA
4121
-------------------------------------------------------------------


-------------------------------------------------------------------
NUMBER OF INDIVIDUAL FASTA FILES
4121
-------------------------------------------------------------------</code></pre>
</section>
</section>
<section id="primer-search-with-emboss-primersearch" class="level1">
<h1>8 PRIMER SEARCH WITH <a href="https://emboss.sourceforge.net/apps/cvs/emboss/apps/primersearch.html">EMBOSS PRIMERSEARCH</a></h1>
<p>This will run <a href="https://emboss.sourceforge.net/apps/cvs/emboss/apps/primersearch.html">EMBOSS PrimerSearch</a> against the genome to assess primer specificity.</p>
<section id="create-emboss-primersearch-primers-file" class="level2">
<h2 class="anchored" data-anchor-id="create-emboss-primersearch-primers-file">8.1 Create EMBOSS PrimerSearch Primers File</h2>
<p>Create a tab-delimited file to use with EMBOSS PrimerSearch.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get primer info from Primer3 default format output</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="va">seq_id</span><span class="op">=</span><span class="va">$(</span><span class="fu">grep</span> <span class="st">"SEQUENCE_ID="</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/primer3-t-primers-default-format.txt"</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/SEQUENCE_ID=//'</span><span class="va">)</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="va">left_primer</span><span class="op">=</span><span class="va">$(</span><span class="fu">grep</span> <span class="st">"PRIMER_LEFT_0_SEQUENCE="</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/primer3-t-primers-default-format.txt"</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/PRIMER_LEFT_0_SEQUENCE=//'</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">'[:lower:]'</span> <span class="st">'[:upper:]'</span><span class="va">)</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="va">right_primer</span><span class="op">=</span><span class="va">$(</span><span class="fu">grep</span> <span class="st">"PRIMER_RIGHT_6_SEQUENCE="</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/primer3-t-primers-default-format.txt"</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/PRIMER_RIGHT_6_SEQUENCE=//'</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">'[:lower:]'</span> <span class="st">'[:upper:]'</span><span class="va">)</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Create EMBOSS primer file</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="bu">printf</span> <span class="st">"%s\t%s\t%s\t\n"</span> <span class="st">"</span><span class="va">${seq_id}</span><span class="st">"</span> <span class="st">"</span><span class="va">${left_primer}</span><span class="st">"</span> <span class="st">"</span><span class="va">${right_primer}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tee</span> <span class="st">"</span><span class="va">${output_top}</span><span class="st">/emboss-primers.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>LOC120027825    TTATATTTCTTAACTACTAACCCTTATACA  AAAAATGTATGTGTGGAAAATTTTAGTTT   </code></pre>
</section>
<section id="run-emboss-primersearch" class="level2">
<h2 class="anchored" data-anchor-id="run-emboss-primersearch">8.2 Run EMBOSS PrimerSearch</h2>
<p>This will run EMBOSS PrimerSearch and allow for a 10% mismatch in primer annealing sites (<code>${primersearch} -auto ${fasta} ${primers} 10</code>).</p>
<p>Afterwards, the resulting output files (<code>*.primersearch</code>) will be searched for the term <code>Amplimer</code>, indicating a PCR product would be produced. If no amplimer is identified in a <code>*.primersearch</code> file, then that file is deleted. This should leave just the results in which primer matches were identified.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${genome_fasta_splits_dir}</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="va">primers</span><span class="op">=</span><span class="st">"../../../../output/emboss-primers.txt"</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a><span class="bu">time</span> <span class="dt">\</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>for fasta in <span class="pp">*</span>.fna</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">do</span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove path from FastA filename</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>  <span class="va">fasta_no_path</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">${fasta</span><span class="op">##</span><span class="pp">*</span>/<span class="va">})</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove file extension from FastA filename</span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>  <span class="va">fasta_no_ext</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">${fasta_no_path</span><span class="op">%%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert filename to lowercase</span></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Will be used for output from EMBOSS PrimerSearch</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>  <span class="va">fasta_no_ext_lower</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">${fasta_no_ext}</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">'[:upper:]'</span> <span class="st">'[:lower:]'</span><span class="va">)</span></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>  <span class="co">###### Run EMBOSS PrimerSearch on all FastA files ########</span></span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Allows for a 10% mismatch</span></span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>  <span class="va">${primersearch}</span> <span class="at">-auto</span> <span class="va">${fasta}</span> <span class="va">${primers}</span> 10</span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a>  <span class="co">##### </span><span class="re">END</span><span class="co"> EMBOSS ##########</span></span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find EMBOSS PrimerSearch output files with primer matches</span></span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove those without a match</span></span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> <span class="ot">! </span><span class="fu">grep</span> <span class="at">--quiet</span> <span class="st">"Amplimer"</span> <span class="st">"</span><span class="va">${fasta_no_ext_lower}</span><span class="st">.primersearch"</span></span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">then</span> <span class="fu">rm</span> <span class="va">${fasta_no_ext_lower}</span>.primersearch</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a>  <span class="cf">fi</span></span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>real    5m31.589s
user    4m35.689s
sys 1m14.992s</code></pre>
</section>
<section id="check-primer-matches" class="level2">
<h2 class="anchored" data-anchor-id="check-primer-matches">8.3 Check primer matches</h2>
<p>This will print the contents of any remaining <code>*.primersearch</code> output files (i.e.&nbsp;sequences with primer matches)</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${genome_fasta_splits_dir}</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Check contents of files with matches</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.primersearch</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">do</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"FILE: </span><span class="va">${file}</span><span class="st">"</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span> <span class="va">${file}</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"----------------------------------"</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>FILE: nc_052339.primersearch


Primer name LOC120027825
Amplimer 1
    Sequence: NC_052339.1  
    
    TTATATTTCTTAACTACTAACCCTTATACA hits forward strand at 19244321 with 0 mismatches
    AAAAATGTATGTGTGGAAAATTTTAGTTT hits reverse strand at [18091692] with 0 mismatches
    Amplimer length: 3907 bp
----------------------------------</code></pre>
</section>
</section>
<section id="summary" class="level1">
<h1>9 SUMMARY</h1>
<p>EMBOSS PrimerSearch has identified only a single location in the genome where the forward and reverse primers match, as well as in the expected location on chromosome <code>NC_052339</code>.</p>
</section>
<section id="citations" class="level1">
<h1>10 CITATIONS</h1>
<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-koressaar2007" class="csl-entry" role="listitem">
Koressaar, Triinu, and Maido Remm. 2007. “Enhancements and Modifications of Primer Design Program Primer3.” <em>Bioinformatics</em> 23 (10): 1289–91. <a href="https://doi.org/10.1093/bioinformatics/btm091" class="uri">https://doi.org/10.1093/bioinformatics/btm091</a>.
</div>
<div id="ref-rice2000" class="csl-entry" role="listitem">
Rice, Peter, Ian Longden, and Alan Bleasby. 2000. “EMBOSS: The European Molecular Biology Open Software Suite.” <em>Trends in Genetics</em> 16 (6): 276–77. <a href="https://doi.org/10.1016/s0168-9525(00)02024-2" class="uri">https://doi.org/10.1016/s0168-9525(00)02024-2</a>.
</div>
<div id="ref-shirley2015" class="csl-entry" role="listitem">
Shirley, Matthew D, Zhaorong Ma, Brent S Pedersen, and Sarah J Wheelan. 2015. “Efficient”Pythonic” Access to FASTA Files Using Pyfaidx.” <a href="http://dx.doi.org/10.7287/peerj.preprints.970v1" class="uri">http://dx.doi.org/10.7287/peerj.preprints.970v1</a>.
</div>
<div id="ref-untergasser2012" class="csl-entry" role="listitem">
Untergasser, Andreas, Ioana Cutcutache, Triinu Koressaar, Jian Ye, Brant C. Faircloth, Maido Remm, and Steven G. Rozen. 2012. “Primer3new Capabilities and Interfaces.” <em>Nucleic Acids Research</em> 40 (15): e115–15. <a href="https://doi.org/10.1093/nar/gks596" class="uri">https://doi.org/10.1093/nar/gks596</a>.
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/robertslab\.github\.io\/sams-notebook");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>