---
author: Sam White
toc-title: Contents
toc-depth: 5
toc-location: left
title: PCR - Lake Trout C1q Primers with Bisulfite-treated gDNA
date: '2024-10-09'
draft: false
engine: knitr
categories: 
  - "2024"
  - "C1q"
  - "Lake trout"
  - "Salvelinus namaycush"
  - "PCR"
  - "bisulfite"
  - "gDNA"
  - "liver"
  - "gel"
bibliography: references.bib
---

## INTRO

The original \_[*Salvelinus namaycush*](https://en.wikipedia.org/wiki/Lake_trout) [C1q bisulfite PCR on 20240911](../2024-09-11-Bisulfite-PCR---Second-Primer-Annealing-Gradient-Test-with-Lake-Trout-Bisulfite-treated-DNA/index.qmd) (Notebook) using [bisulfite primers designed on 20240729](../2024-07-29-Primer-Design---Lake-Trout-C1q-Gene-Bisulfite-Sequencing-Primers-Using-Primer3/index.qmd) (Notebook) failed to produce any amplification. In response, we opted to just confirm that we could actually amplify the C1q gene using [regular primers, designed on 20240916](../2024-09-16-Primer-Design---Lake-Trout-C1q-Gene-Sequencing-Primers-Using-Primer3/index.qmd) (Notebook) and the [original liver gDNA from 20240712](../2024-07-12-DNA-Isolation-and-Quantification---Lake-Trout-Liver/index.qmd). This latter PCR was successful. So, just for kicks, we decided to try running a PCR with the regular C1q primers and the bisulfite-treated gDNA.

## MATERIALS & METHODS

### DNA Pooling

Used 0.5uL of the pooled gDNA in each reaction.

### PCR Reactions

PCR reactions were run using [EpiMark Hot Start Taq DNA Polymerase](https://www.neb.com/en-us/products/m0490-epimark-hot-start-taq-dna-polymerase) (New England Biolabs), according to the [manufacturer's protocol](https://www.neb.com/en-us/protocols/2012/09/25/epimark-hot-start-taq-dna-polymerase-guidelines-for-pcr-reactions-m0490), in a total reaction volume of 25uL.

I ran the reaction in duplicate, using 0.5uL of the pooled liver gDNA for template. Each reaction was prepared *individually* (i.e. no master mix was created).

Primers used:

-   `Sn-C1q-F/R` (SRIDs: 1842/3)

### PCR Cycling

Annealing temperature gradient PCR was performed in a CFX Connect (BioRad) thermalcycler.

Cycling params:

![Screenshot of cycling program, also sdhowing temperature gradient.](./202401009-snam-c1q-BS-DNA-pcr.png){#fig-PCR-parameters fig-alt="PCR cycling parameters."}

1.  95<sup>o</sup>C - 30s
2.  95<sup>o</sup>C - 15s
3.  55<sup>o</sup>C - 30s
4.  68<sup>o</sup>C - 4m
5.  Go to Step 2 x 39.
6.  68<sup>o</sup>C - 5m
7.  4<sup>o</sup>C - FOREVER

::: callout-tip
Extension step was set to 4m due to the [expected amplicon size of C1q using the Roberts Lab primers](../2024-09-16-Primer-Design---Lake-Trout-C1q-Gene-Sequencing-Primers-Using-Primer3/index.qmd) (Notebook) of 3192bp. Most DNA polymerases (including [EpiMark Hot Start Taq DNA Polymerase](https://www.neb.com/en-us/products/m0490-epimark-hot-start-taq-dna-polymerase)) process 1Kbp/min, so setting to four mins is a tad long, but will certainly ensure the full gene length is amplified.
:::

PCR reactions were loaded on a low-TAE, 1.0% agarose gel, with ethidium bromide. Gel was run at 107V for \~45mins and then imaged.

O'GeneRuler (ThermoFisher) Ladder was used (5uL) for size reference:

![Ladder](../../../images/ogene-ruler.jpeg){#fig-ladder .lightbox fig-alt="O'GeneRuler"}

## RESULTS

Unsurprisingly, we do not see any amplification.

![Gel image showing only molecular weight marker and no visible amplicons in the other lanes.](./20241009-snam-c1q-PCR-gel.jpg){#fig-gel-image .lightbox fig-alt="Gel image showing only molecular weight marker and no visible amplicons in the other lanes."}

## DISCUSSION

Based on the failure of our previous [bisulfite PCR using bisulfite primers](../2024-09-11-Bisulfite-PCR---Second-Primer-Annealing-Gradient-Test-with-Lake-Trout-Bisulfite-treated-DNA/index.qmd) (Notebook), which utilized primers specifically designed to amplify bisulfite-treated DNA, I had little expectation this would work. It did not.

In retrospect, we probably shouldn't have had *any* expectations this would work. The reasons for this are related to the amount of DNA fragmentation/degradation that takes place during bisulfite treatment of gDNA. I was certainly aware of the fact that bisulfite treatement was considered "harsh" and resulted in single-stranded gDNA, as well as some degradation. However, after some investigation into this (spurred by the results of this project), it's clear that the degradation that occurs is high! High enough, in fact, one is unlikely to ever expect any successful PCR of products larger than \~2,000bp.

As an example, [ZymoResearch's "Bisulfite Beginner Guide"](https://www.zymoresearch.com/pages/bisulfite-beginner-guide) (ZymoResearch website), provides an example gel to display the level of degradation expected from bisulfite treatment of gDNA. Keep in mind, companies usually display their *best* examples:

![Screencapture from ZymoResearch website of a two gel images showing bisulfite gDNA. The gel image on the left has been run under normal circumstances. The gel on the right shows the same gel, but after cooling in an ice bath; allowing the DNA to form enough base pairing to allow the ethidium bromide to intercalate the DNA.](./zymo-bs-dna-gel.png){.lightbox fig-alt="Screencapture from ZymoResearch website of a two gel images showing bisulfite gDNA. The gel image on the left has been run under normal circumstances. The gel on the right shows the same gel, but after cooling in an ice bath; allowing the DNA to form enough base pairing to allow the ethidium bromide to intercalate the DNA."}

Notice that the molecular weight marker is a 100bp marker. This means the highest molecular weight band in the marker corresponds to 1500bp! That means our attempts to potentially amplify a \~3,200bp from bisulfite-treated gDNA were virtually impossible to achieve.

Also, as part of my research into the impacts of bisulfite degradation on gDNA, I came acroos this excellent paper (["Evaluation of bisulfite kits for DNA methylation profiling in terms of DNA fragmentation and DNA recovery using digital PCR "](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0199091)) [@kint2018] which compares the performance of multiple bisulfite treatment kits, including:

-   recovery (yield)
-   recovery integrity
-   processing time

The combination of information from that ZymoResearch site and the paper make it pretty clear that PCR of a product size of ~3,200bp would not happen. Have discussed with Steven and we'll try to come up with some other ideas on how to approach the issue of determining gene-level DNA methylation of C1q in individuals.

## CITATIONS
