---
layout: post
title: Daily Bits - July 2022
date: '2022-07-05 09:49'
tags: 
  - daily bits
  - July
categories: 
  - Daily Bits
---

20220709

- Nextflow epidiverse/snp: Let it run overnight and woke up to the same timeout error message and kernel memory warnings, despite changes in config file.. Although, I had been running it with the `-resume` flag... I'll remove all previous `snps/` and `work/` dirs and run with the following command:

  ```shell
  NXF_VER=20.07.1 /home/shared/nextflow run epidiverse/snp \
  -c nextflow-docker_permissions.config \
  -profile docker \
  --input /home/shared/8TB_HDD_01/sam/data/O_lurida/BSseq/ \
  --reference /home/shared/8TB_HDD_01/sam/data/O_lurida/genomes/Olurida_v081.fa
  ```

  - Well that didn't work... I'm going to try again with a limited data set, as the `Olurida_v081.fa` has a TON of contigs, which has caused issues with other bisfulfite analysis software.

---

20220708

- Spent time at Science Hour with Steven visualizing some of the [_Ostrea lurida_ (Olympia oyster)](http://en.wikipedia.org/wiki/Pacific_oyster) transcript counts stats in R - I mostly just watched and nodded. :)

- Nextflow epidiverse/snp problems. Got to deal with these shenanigans:

  ```shell
  ================================================
  E P I D I V E R S E - S N P    P I P E L I N E
  ================================================
  ~ version 1.0

  input dir     : /home/shared/8TB_HDD_01/sam/data/O_lurida/BSseq/
  reference     : /home/shared/8TB_HDD_01/sam/data/O_lurida/genomes/Olurida_v081.fa
  output dir    : snps
  variant calls : enabled
  clustering    : enabled

  ================================================
  RUN NAME: nostalgic_bhabha

  executor >  local (34)
  [d7/390a8d] process > SNPS:preprocessing (zr1394_all_s456_trimmed_bismark_bt2.deduplicated.sorted)     [100%] 10 of 10 âœ”
  [49/25512a] process > SNPS:masking (zr1394_11_s456_trimmed_bismark_bt2.deduplicated.sorted - variants) [ 60%] 24 of 40, failed: 24, retries: 20
  [-        ] process > SNPS:extracting                                                                  -
  [-        ] process > SNPS:khmer                                                                       -
  [-        ] process > SNPS:kwip                                                                        -
  [-        ] process > SNPS:clustering                                                                  -
  [-        ] process > SNPS:sorting                                                                     -
  [-        ] process > SNPS:freebayes                                                                   -
  [-        ] process > SNPS:bcftools                                                                    -
  [-        ] process > SNPS:plot_vcfstats                                                               -
  Error executing process > 'SNPS:masking (zr1394_17_s456_trimmed_bismark_bt2.deduplicated.sorted - clustering)'

  Caused by:
    Process exceeded running time limit (4h)

  Command executed:

    change_sam_queries.py -Q -T 16 -t . -G calmd.bam clustering.bam || exit $?
    find -mindepth 1 -maxdepth 1 -type d -exec rm -r {} \;

  Command exit status:
    -

  Command output:
    (empty)

  Command error:
    WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.

  Work dir:
    /home/shared/8TB_HDD_01/sam/analyses/20220707-olur-epidiverse/work/8b/636e263c8924aa275cd5f1fc7787e8

  Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`
  ```

  - Updated custom config file (which was originally used solely to address permissions issues on the outptut files generated by the Docker container) to include improved time limits:


    ```
      docker {
      fixOwnership = true
    }


    process {

        // Process-specific resource requirements
        withLabel:process_low {
            time   = 30.d
        }
        withLabel:process_medium {
            time   = 30.d
        }
        withLabel:process_high {
            time   = 30.d
        }
        withLabel:process_long {
            time   = 30.d
        }
    }
    ```

  - Seemed to fix timeout issue as this ran for nearly 6hrs without timing out error, but still getting the kernel memory warning in log files and constant failures at the masking step. Added this line to config file:

    ```
    docker.runOptions = "--memory-swap '-1'"
    ```

     Config file now looks like this ([grabbed idea to use this line to resolve kernel issue from here](https://github.com/nf-core/tools/pull/351)):

     ```
          docker {
              docker.runOptions = "--memory-swap '-1'"
              fixOwnership = true
      }


      process {

          // Process-specific resource requirements
          withLabel:process_low {
              time   = 30.d
          }
          withLabel:process_medium {
              time   = 30.d
          }
          withLabel:process_high {
              time   = 30.d
          }
          withLabel:process_long {
              time   = 30.d
          }
      }
     ```
---

20220707

- In order to start Nextflow epidiverse/snp, genome FastA file requires FastA index file; this is not documented, but pipeline throws an error when a FastA index is not found.

- started Nextflow epidiverse/snp on raven

---

20220706

- fixed [Nextflow epidiverse/snp problem](https://github.com/RobertsLab/resources/issues/1488) (GitHub Issue)

- dealt with [lake trout BioProject PRJNA316738 data retrieval and QC](https://github.com/RobertsLab/resources/issues/1482) (GitHub Issue)

---

20220705

- CEABIGR meeting:

  - added gene names to transcript counts files 
  - added code to cover all desired iterations of output files (i.e. missing controls males files)

