<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sam White">
<meta name="dcterms.date" content="2026-02-02">

<title>qPCR Analysis - M.gigas PolyIC Data from Valentinas Project – Sam’s Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark-b53751a350365c71b6c909e95f209ed1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-e518e50a9807db09b51ea894c4c266f8.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 2,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 100,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-7483T61T11"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-7483T61T11', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Sam’s Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" target="_blank">
 <span class="dropdown-text">Rstudio Server - Raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu/Atumefaciens/" target="_blank">
 <span class="dropdown-text">My Gannet Directory</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" target="_blank">
 <span class="dropdown-text">Steven’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shedurkin.github.io/Roberts-LabNotebook/" target="_blank">
 <span class="dropdown-text">Kathleen’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://meganewing.github.io/mewing-notebook/" target="_blank">
 <span class="dropdown-text">Megan’s Notebook</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/sams-notebook" target="_blank">
 <span class="dropdown-text">Notebook repo</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" target="_blank">
 <span class="dropdown-text">Lab Organization (GitHub)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" target="_blank">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" target="_blank"> <i class="bi bi-calendar-week" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-right">
      <h1 class="title">qPCR Analysis - M.gigas PolyIC Data from Valentinas Project</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2026</div>
                <div class="quarto-category">qPCR</div>
                <div class="quarto-category">Pacific oyster</div>
                <div class="quarto-category">Magallana gigas</div>
                <div class="quarto-category">Crassostrea gigas</div>
                <div class="quarto-category">polyIC</div>
                <div class="quarto-category">HSP70</div>
                <div class="quarto-category">HSP90</div>
                <div class="quarto-category">cGAS</div>
                <div class="quarto-category">VIPERIN</div>
                <div class="quarto-category">ATP Synthase</div>
                <div class="quarto-category">Citrate Synthase</div>
                <div class="quarto-category">DNMT1</div>
                <div class="quarto-category">GAPDH</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-right">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Sam White </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 2, 2026</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#intro" id="toc-intro" class="nav-link active" data-scroll-target="#intro">INTRO</a></li>
  <li><a href="#load-data" id="toc-load-data" class="nav-link" data-scroll-target="#load-data">Load Data</a></li>
  <li><a href="#functions" id="toc-functions" class="nav-link" data-scroll-target="#functions">Functions</a>
  <ul>
  <li><a href="#calculate-delta-cq" id="toc-calculate-delta-cq" class="nav-link" data-scroll-target="#calculate-delta-cq">Calculate delta Cq</a></li>
  </ul></li>
  <li><a href="#data-cleaning" id="toc-data-cleaning" class="nav-link" data-scroll-target="#data-cleaning">Data Cleaning</a>
  <ul>
  <li><a href="#remove-ntc-samples" id="toc-remove-ntc-samples" class="nav-link" data-scroll-target="#remove-ntc-samples">Remove NTC samples</a></li>
  <li><a href="#check-for-high-standard-deviations" id="toc-check-for-high-standard-deviations" class="nav-link" data-scroll-target="#check-for-high-standard-deviations">Check for high standard deviations</a></li>
  <li><a href="#remove-outlier-technical-replicates" id="toc-remove-outlier-technical-replicates" class="nav-link" data-scroll-target="#remove-outlier-technical-replicates">Remove outlier technical replicates</a>
  <ul>
  <li><a href="#samples-excluded-due-to-quality-control" id="toc-samples-excluded-due-to-quality-control" class="nav-link" data-scroll-target="#samples-excluded-due-to-quality-control">Samples excluded due to quality control</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#group-and-summarize-data" id="toc-group-and-summarize-data" class="nav-link" data-scroll-target="#group-and-summarize-data">Group and Summarize Data</a></li>
  <li><a href="#parse-sample-names-and-add-treatment-columns" id="toc-parse-sample-names-and-add-treatment-columns" class="nav-link" data-scroll-target="#parse-sample-names-and-add-treatment-columns">Parse Sample Names and Add Treatment Columns</a>
  <ul>
  <li><a href="#view-treatment-groups" id="toc-view-treatment-groups" class="nav-link" data-scroll-target="#view-treatment-groups">View treatment groups</a></li>
  </ul></li>
  <li><a href="#calculate-delta-cq-1" id="toc-calculate-delta-cq-1" class="nav-link" data-scroll-target="#calculate-delta-cq-1">Calculate Delta Cq</a>
  <ul>
  <li><a href="#missing-samples-analysis" id="toc-missing-samples-analysis" class="nav-link" data-scroll-target="#missing-samples-analysis">Missing samples analysis</a></li>
  <li><a href="#gapdh-impact-analysis" id="toc-gapdh-impact-analysis" class="nav-link" data-scroll-target="#gapdh-impact-analysis">GAPDH Impact Analysis</a></li>
  </ul></li>
  <li><a href="#statistical-comparisons" id="toc-statistical-comparisons" class="nav-link" data-scroll-target="#statistical-comparisons">Statistical Comparisons</a>
  <ul>
  <li><a href="#t-tests-for-treatment-comparisons" id="toc-t-tests-for-treatment-comparisons" class="nav-link" data-scroll-target="#t-tests-for-treatment-comparisons">T-tests for treatment comparisons</a>
  <ul>
  <li><a href="#comparison-1-basic-control-vs-stressors-no-polyic" id="toc-comparison-1-basic-control-vs-stressors-no-polyic" class="nav-link" data-scroll-target="#comparison-1-basic-control-vs-stressors-no-polyic">Comparison 1: Basic Control vs Stressors (No PolyIC)</a>
  <ul class="collapse">
  <li><a href="#significant-genes-p-0.05" id="toc-significant-genes-p-0.05" class="nav-link" data-scroll-target="#significant-genes-p-0.05">Significant genes (p ≤ 0.05)</a></li>
  </ul></li>
  <li><a href="#comparison-2-basic-control-vs-polyic-control" id="toc-comparison-2-basic-control-vs-polyic-control" class="nav-link" data-scroll-target="#comparison-2-basic-control-vs-polyic-control">Comparison 2: Basic Control vs PolyIC Control</a>
  <ul class="collapse">
  <li><a href="#significant-genes-p-0.05-1" id="toc-significant-genes-p-0.05-1" class="nav-link" data-scroll-target="#significant-genes-p-0.05-1">Significant genes (p ≤ 0.05)</a></li>
  </ul></li>
  <li><a href="#comparison-3-polyic-control-vs-polyic-stressors" id="toc-comparison-3-polyic-control-vs-polyic-stressors" class="nav-link" data-scroll-target="#comparison-3-polyic-control-vs-polyic-stressors">Comparison 3: PolyIC Control vs PolyIC Stressors</a>
  <ul class="collapse">
  <li><a href="#significant-genes-p-0.05-2" id="toc-significant-genes-p-0.05-2" class="nav-link" data-scroll-target="#significant-genes-p-0.05-2">Significant genes (p ≤ 0.05)</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#create-anova-models" id="toc-create-anova-models" class="nav-link" data-scroll-target="#create-anova-models">Create ANOVA Models</a></li>
  </ul></li>
  <li><a href="#visualizations" id="toc-visualizations" class="nav-link" data-scroll-target="#visualizations">Visualizations</a>
  <ul>
  <li><a href="#delta-cq-boxplots" id="toc-delta-cq-boxplots" class="nav-link" data-scroll-target="#delta-cq-boxplots">Delta Cq Boxplots</a>
  <ul>
  <li><a href="#comparison-1-control-vs-stressors-no-polyic" id="toc-comparison-1-control-vs-stressors-no-polyic" class="nav-link" data-scroll-target="#comparison-1-control-vs-stressors-no-polyic">Comparison 1: Control vs Stressors (No PolyIC)</a></li>
  <li><a href="#comparison-2-control-vs-polyic-control" id="toc-comparison-2-control-vs-polyic-control" class="nav-link" data-scroll-target="#comparison-2-control-vs-polyic-control">Comparison 2: Control vs PolyIC Control</a></li>
  <li><a href="#comparison-3-polyic-control-vs-polyic-stressors-1" id="toc-comparison-3-polyic-control-vs-polyic-stressors-1" class="nav-link" data-scroll-target="#comparison-3-polyic-control-vs-polyic-stressors-1">Comparison 3: PolyIC Control vs PolyIC Stressors</a></li>
  </ul></li>
  <li><a href="#all-treatments-comparison" id="toc-all-treatments-comparison" class="nav-link" data-scroll-target="#all-treatments-comparison">All treatments comparison</a></li>
  </ul></li>
  <li><a href="#delta-delta-cq-and-fold-change-analysis" id="toc-delta-delta-cq-and-fold-change-analysis" class="nav-link" data-scroll-target="#delta-delta-cq-and-fold-change-analysis">Delta Delta Cq and Fold Change Analysis</a>
  <ul>
  <li><a href="#calculate-delta-delta-cq-for-each-comparison" id="toc-calculate-delta-delta-cq-for-each-comparison" class="nav-link" data-scroll-target="#calculate-delta-delta-cq-for-each-comparison">Calculate delta delta Cq for each comparison</a>
  <ul>
  <li><a href="#comparison-1-stressors-vs-control-no-polyic" id="toc-comparison-1-stressors-vs-control-no-polyic" class="nav-link" data-scroll-target="#comparison-1-stressors-vs-control-no-polyic">Comparison 1: Stressors vs Control (No PolyIC)</a></li>
  <li><a href="#comparison-2-polyic-control-vs-basic-control" id="toc-comparison-2-polyic-control-vs-basic-control" class="nav-link" data-scroll-target="#comparison-2-polyic-control-vs-basic-control">Comparison 2: PolyIC Control vs Basic Control</a></li>
  <li><a href="#comparison-3-polyic-stressors-vs-polyic-control" id="toc-comparison-3-polyic-stressors-vs-polyic-control" class="nav-link" data-scroll-target="#comparison-3-polyic-stressors-vs-polyic-control">Comparison 3: PolyIC Stressors vs PolyIC Control</a></li>
  </ul></li>
  <li><a href="#fold-change-bar-plots" id="toc-fold-change-bar-plots" class="nav-link" data-scroll-target="#fold-change-bar-plots">Fold Change Bar Plots</a>
  <ul>
  <li><a href="#no-polyic-stressors-vs-control" id="toc-no-polyic-stressors-vs-control" class="nav-link" data-scroll-target="#no-polyic-stressors-vs-control">No PolyIC: Stressors vs Control</a></li>
  <li><a href="#polyic-control-vs-basic-control" id="toc-polyic-control-vs-basic-control" class="nav-link" data-scroll-target="#polyic-control-vs-basic-control">PolyIC Control vs Basic Control</a></li>
  <li><a href="#polyic-stressors-vs-control" id="toc-polyic-stressors-vs-control" class="nav-link" data-scroll-target="#polyic-stressors-vs-control">PolyIC: Stressors vs Control</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#anova-results-and-diagnostics" id="toc-anova-results-and-diagnostics" class="nav-link" data-scroll-target="#anova-results-and-diagnostics">ANOVA Results and Diagnostics</a>
  <ul>
  <li><a href="#data-distribution-check" id="toc-data-distribution-check" class="nav-link" data-scroll-target="#data-distribution-check">Data distribution check</a></li>
  <li><a href="#atp-synthase" id="toc-atp-synthase" class="nav-link" data-scroll-target="#atp-synthase">ATP Synthase</a>
  <ul>
  <li><a href="#assumption-checks-for-atp-synthase" id="toc-assumption-checks-for-atp-synthase" class="nav-link" data-scroll-target="#assumption-checks-for-atp-synthase">Assumption Checks for ATP Synthase</a></li>
  <li><a href="#atp-synthase-anova-interpretation" id="toc-atp-synthase-anova-interpretation" class="nav-link" data-scroll-target="#atp-synthase-anova-interpretation">ATP Synthase ANOVA Interpretation</a></li>
  </ul></li>
  <li><a href="#citrate-synthase" id="toc-citrate-synthase" class="nav-link" data-scroll-target="#citrate-synthase">Citrate Synthase</a>
  <ul>
  <li><a href="#assumption-checks-for-citrate-synthase" id="toc-assumption-checks-for-citrate-synthase" class="nav-link" data-scroll-target="#assumption-checks-for-citrate-synthase">Assumption Checks for Citrate Synthase</a></li>
  <li><a href="#citrate-synthase-anova-interpretation" id="toc-citrate-synthase-anova-interpretation" class="nav-link" data-scroll-target="#citrate-synthase-anova-interpretation">Citrate Synthase ANOVA Interpretation</a></li>
  </ul></li>
  <li><a href="#dnmt1" id="toc-dnmt1" class="nav-link" data-scroll-target="#dnmt1">DNMT1</a>
  <ul>
  <li><a href="#assumption-checks-for-dnmt1" id="toc-assumption-checks-for-dnmt1" class="nav-link" data-scroll-target="#assumption-checks-for-dnmt1">Assumption Checks for DNMT1</a></li>
  <li><a href="#dnmt1-anova-interpretation" id="toc-dnmt1-anova-interpretation" class="nav-link" data-scroll-target="#dnmt1-anova-interpretation">DNMT1 ANOVA Interpretation</a></li>
  </ul></li>
  <li><a href="#hsp70" id="toc-hsp70" class="nav-link" data-scroll-target="#hsp70">HSP70</a>
  <ul>
  <li><a href="#assumption-checks-for-hsp70" id="toc-assumption-checks-for-hsp70" class="nav-link" data-scroll-target="#assumption-checks-for-hsp70">Assumption Checks for HSP70</a></li>
  <li><a href="#hsp70-anova-interpretation" id="toc-hsp70-anova-interpretation" class="nav-link" data-scroll-target="#hsp70-anova-interpretation">HSP70 ANOVA Interpretation</a></li>
  </ul></li>
  <li><a href="#hsp90" id="toc-hsp90" class="nav-link" data-scroll-target="#hsp90">HSP90</a>
  <ul>
  <li><a href="#assumption-checks-for-hsp90" id="toc-assumption-checks-for-hsp90" class="nav-link" data-scroll-target="#assumption-checks-for-hsp90">Assumption Checks for HSP90</a></li>
  <li><a href="#hsp90-anova-interpretation" id="toc-hsp90-anova-interpretation" class="nav-link" data-scroll-target="#hsp90-anova-interpretation">HSP90 ANOVA Interpretation</a></li>
  </ul></li>
  <li><a href="#cgas" id="toc-cgas" class="nav-link" data-scroll-target="#cgas">cGAS</a>
  <ul>
  <li><a href="#assumption-checks-for-cgas" id="toc-assumption-checks-for-cgas" class="nav-link" data-scroll-target="#assumption-checks-for-cgas">Assumption Checks for cGAS</a></li>
  <li><a href="#cgas-anova-interpretation" id="toc-cgas-anova-interpretation" class="nav-link" data-scroll-target="#cgas-anova-interpretation">cGAS ANOVA Interpretation</a></li>
  </ul></li>
  <li><a href="#viperin" id="toc-viperin" class="nav-link" data-scroll-target="#viperin">VIPERIN</a>
  <ul>
  <li><a href="#assumption-checks-for-viperin" id="toc-assumption-checks-for-viperin" class="nav-link" data-scroll-target="#assumption-checks-for-viperin">Assumption Checks for VIPERIN</a></li>
  <li><a href="#viperin-anova-interpretation" id="toc-viperin-anova-interpretation" class="nav-link" data-scroll-target="#viperin-anova-interpretation">VIPERIN ANOVA Interpretation</a></li>
  </ul></li>
  <li><a href="#summary-plots" id="toc-summary-plots" class="nav-link" data-scroll-target="#summary-plots">Summary Plots</a></li>
  </ul></li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary">SUMMARY</a>
  <ul>
  <li><a href="#gene-specific-results" id="toc-gene-specific-results" class="nav-link" data-scroll-target="#gene-specific-results">Gene-Specific Results</a>
  <ul>
  <li><a href="#atp-synthase-1" id="toc-atp-synthase-1" class="nav-link" data-scroll-target="#atp-synthase-1">ATP Synthase</a></li>
  <li><a href="#citrate-synthase-1" id="toc-citrate-synthase-1" class="nav-link" data-scroll-target="#citrate-synthase-1">Citrate Synthase</a></li>
  <li><a href="#dnmt1-1" id="toc-dnmt1-1" class="nav-link" data-scroll-target="#dnmt1-1">DNMT1</a></li>
  <li><a href="#hsp70-1" id="toc-hsp70-1" class="nav-link" data-scroll-target="#hsp70-1">HSP70</a></li>
  <li><a href="#hsp90-1" id="toc-hsp90-1" class="nav-link" data-scroll-target="#hsp90-1">HSP90</a></li>
  <li><a href="#cgas-1" id="toc-cgas-1" class="nav-link" data-scroll-target="#cgas-1">cGAS</a></li>
  <li><a href="#viperin-1" id="toc-viperin-1" class="nav-link" data-scroll-target="#viperin-1">VIPERIN</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#r-session-info" id="toc-r-session-info" class="nav-link" data-scroll-target="#r-session-info">R SESSION INFO</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-right" id="quarto-document-content">





<section id="intro" class="level1">
<h1>INTRO</h1>
<p>This is an analysis of qPCR data from <em>Magallana gigas</em> (Pacific oyster) samples treated with PolyIC (a synthetic analog of double-stranded RNA that mimics viral infection) and subjected to different stress types (Temperature, Mechanical, Control). The goal was to assess gene expression changes in response to these treatments.</p>
<p>The qPCR data was generated by me on January 27-29, 2026, as part of Valentina’s project on immune responses in oysters. Eight genes were targeted: ATP Synthase, Citrate Synthase, DNMT1, HSP70, HSP90, cGAS, and VIPERIN. GAPDH was used as the normalizing gene. Notebook entries documenting the qPCR runs and data exports can be found in the following posts:</p>
<ul>
<li><a href="../2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/">2026-01-27-qPCRs - M.gigas Valentina PolyIC cGAS, citrate_synthase, and DNMT1</a></li>
<li><a href="../2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/">2026-01-28-qPCRs - M.gigas Valentina PolyIC ATP synthase, GAPDH, HSP70, and HSP90</a></li>
<li><a href="../2026-01-29-qPCR---M.gigas-Valentina-PolyIC-VIPERIN/">2026-01-29-qPCR - M.gigas Valentina PolyIC VIPERIN</a></li>
</ul>
<p>I also managed to track down the qPCRs run on the RNA I isolated on <a href="../2025-08-04-RNA-Isolation-and-Quantification---C.gigas-PolyIC-Mechanical-Trials-for-Valentina-Using-Quick-DNA-RNA-MiniPrep-Plus-Kit/">isolated on 20250804</a>. After RNA isolation, they were passed to Valentina for reverse transcription and qPCR. I believe those qPCRs were run/analyzed by Aakriti <a href="https://genefish.wordpress.com/2025/08/14/results-from-qpcr-with-valentina/">on 20250814</a> (Notebook entry). I found the files here:</p>
<ul>
<li><p><a href="https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/admin_2025-08-14%2011-11-18_Connect_ATPase_vv.pcrd">https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/admin_2025-08-14%2011-11-18_Connect_ATPase_vv.pcrd</a></p></li>
<li><p><a href="https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/admin_2025-08-12%2014-19-37_Connect_HSP70_vv.pcrd">https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/admin_2025-08-12%2014-19-37_Connect_HSP70_vv.pcrd</a></p></li>
<li><p><a href="https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/admin_2025-08-12%2011-18-15_Connect_citrate_VV.pcrd">https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/admin_2025-08-12%2011-18-15_Connect_citrate_VV.pcrd</a></p></li>
<li><p><a href="https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/admin_2025-08-08%2014-18-00_Connect_GAPDH.pcrd">https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/admin_2025-08-08%2014-18-00_Connect_GAPDH.pcrd</a></p></li>
</ul>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>I have not taken the time to evaluate the qPCRs run by Aakriti in detail. I’ve relied on the processing below to filter out samples with poor technical replicates. Maybe these can be revisited at a later date to see if any samples need to be re-run.</p>
</div>
</div>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">echo =</span> <span class="cn">TRUE</span>,         <span class="co"># Display code chunks</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">eval =</span> <span class="cn">TRUE</span>,        <span class="co"># Evaluate code chunks</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">warning =</span> <span class="cn">FALSE</span>,     <span class="co"># Hide warnings</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">message =</span> <span class="cn">FALSE</span>,     <span class="co"># Hide messages</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">comment =</span> <span class="st">""</span>         <span class="co"># Prevents appending '##' to beginning of lines in code output</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.2     ✔ tibble    3.2.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.4     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(car)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: carData

Attaching package: 'car'

The following object is masked from 'package:dplyr':

    recode

The following object is masked from 'package:purrr':

    some</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(emmeans)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to emmeans.
Caution: You lose important information if you filter this package's results.
See '? untidy'</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(multcomp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: mvtnorm
Loading required package: survival
Loading required package: TH.data
Loading required package: MASS

Attaching package: 'MASS'

The following object is masked from 'package:dplyr':

    select


Attaching package: 'TH.data'

The following object is masked from 'package:MASS':

    geyser</code></pre>
</div>
</div>
</section>
<section id="load-data" class="level1">
<h1>Load Data</h1>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load qPCR data</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Define file URLs and target names in a structured format</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>file_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># CITRATE SYNTHASE</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_15-32-08_Connect-citrate_synthase-01--Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"Citrate.Synthase"</span>, <span class="at">gene =</span> <span class="st">"citrate_synthase"</span>),</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_15-30-47_CFX96-citrate_synthase-02-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"Citrate.Synthase"</span>, <span class="at">gene =</span> <span class="st">"citrate_synthase"</span>),</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_16-31-12_Connect-citrate_synthase-03-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"Citrate.Synthase"</span>, <span class="at">gene =</span> <span class="st">"citrate_synthase"</span>),</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-02-02-qPCR-Analysis---M.gigas-PolyIC-Data-from-Valentinas-Project/admin_2025-08-12%2011-18-15_Connect_citrate_VV-SJW-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"Citrate.Synthase"</span>, <span class="at">gene =</span> <span class="st">"citrate_synthase"</span>),</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># DNMT1</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_13-34-02_Connect-DNMT1-01-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"DNMT1"</span>, <span class="at">gene =</span> <span class="st">"DNMT1"</span>),</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_13-29-34_CFX96-DNMT1-02%20Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"DNMT1"</span>, <span class="at">gene =</span> <span class="st">"DNMT1"</span>),</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_14-33-31_Connect-DNMT1-03-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"DNMT1"</span>, <span class="at">gene =</span> <span class="st">"DNMT1"</span>),</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># cGAS</span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_11-25-14_Connect-cGAS-01-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"cGAS"</span>, <span class="at">gene =</span> <span class="st">"cGAS"</span>),</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_11-49-08_CFX96-cGAS-02-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"cGAS"</span>, <span class="at">gene =</span> <span class="st">"cGAS"</span>),</span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-27-qPCRs---M.gigas-Valentina-PolyIC-cGAS-citrate_synthase-and-DNMT1/sam_2026-01-27_12-35-04_Connect-cGAS-03-QuantificationCq_Results.csv"</span>, </span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"cGAS"</span>, <span class="at">gene =</span> <span class="st">"cGAS"</span>),</span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ATP SYNTHASE</span></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_09-56-00_Connect-ATP_synthase-01-QuantificationCq_Results.csv"</span>, </span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"ATP.Synthase"</span>, <span class="at">gene =</span> <span class="st">"ATP_synthase"</span>),</span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_10-13-15_CFX96-ATP_synthase-02-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"ATP.Synthase"</span>, <span class="at">gene =</span> <span class="st">"ATP_synthase"</span>),</span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_10-55-49_Connect-ATP_synthase-03-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"ATP.Synthase"</span>, <span class="at">gene =</span> <span class="st">"ATP_synthase"</span>),</span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-02-02-qPCR-Analysis---M.gigas-PolyIC-Data-from-Valentinas-Project/admin_2025-08-14%2011-11-18_Connect_ATPase_vv-SJW-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"ATP.Synthase"</span>, <span class="at">gene =</span> <span class="st">"ATP_synthase"</span>),</span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># GAPDH</span></span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_12-06-16_Connect-GAPDH-01-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"GAPDH"</span>, <span class="at">gene =</span> <span class="st">"GAPDH"</span>),</span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_12-07-20_CFX96-GAPDH-02-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"GAPDH"</span>, <span class="at">gene =</span> <span class="st">"GAPDH"</span>),</span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_13-05-59_Connect-GAPDH-03-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"GAPDH"</span>, <span class="at">gene =</span> <span class="st">"GAPDH"</span>),</span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-02-02-qPCR-Analysis---M.gigas-PolyIC-Data-from-Valentinas-Project/admin_2025-08-08%2014-18-00_Connect_GAPD-SJW-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"GAPDH"</span>, <span class="at">gene =</span> <span class="st">"GAPDH"</span>),</span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>  <span class="co"># HSP70</span></span>
<span id="cb9-52"><a href="#cb9-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_14-03-35_Connect-HSP70-01-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-53"><a href="#cb9-53" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"HSP70"</span>, <span class="at">gene =</span> <span class="st">"HSP70"</span>),</span>
<span id="cb9-54"><a href="#cb9-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_14-25-31_CFX96-HSP70-02-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-55"><a href="#cb9-55" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"HSP70"</span>, <span class="at">gene =</span> <span class="st">"HSP70"</span>),</span>
<span id="cb9-56"><a href="#cb9-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_15-00-57_Connect-HSP70-03-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-57"><a href="#cb9-57" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"HSP70"</span>, <span class="at">gene =</span> <span class="st">"HSP70"</span>),</span>
<span id="cb9-58"><a href="#cb9-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-02-02-qPCR-Analysis---M.gigas-PolyIC-Data-from-Valentinas-Project/admin_2025-08-12%2014-19-37_Connect_HSP70_vv-SJW-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-59"><a href="#cb9-59" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"HSP70"</span>, <span class="at">gene =</span> <span class="st">"HSP70"</span>),</span>
<span id="cb9-60"><a href="#cb9-60" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-61"><a href="#cb9-61" aria-hidden="true" tabindex="-1"></a>  <span class="co"># HSP90</span></span>
<span id="cb9-62"><a href="#cb9-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28-16-06-43_Connect-HSP90-01-QuantificationCq_Results.csv"</span>, </span>
<span id="cb9-63"><a href="#cb9-63" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"HSP90"</span>, <span class="at">gene =</span> <span class="st">"HSP90"</span>),</span>
<span id="cb9-64"><a href="#cb9-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_16-07-47_CFX96-HSP90-02-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-65"><a href="#cb9-65" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"HSP90"</span>, <span class="at">gene =</span> <span class="st">"HSP90"</span>),</span>
<span id="cb9-66"><a href="#cb9-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-28-qPCRs---M.gigas-Valentina-PolyIC-ATP-synthase-GAPDH-HSP70-and-HSP90/sam_2026-01-28_17-07-09_Connect-HSP90-03-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-67"><a href="#cb9-67" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"HSP90"</span>, <span class="at">gene =</span> <span class="st">"HSP90"</span>),</span>
<span id="cb9-68"><a href="#cb9-68" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-69"><a href="#cb9-69" aria-hidden="true" tabindex="-1"></a>  <span class="co"># VIPERIN</span></span>
<span id="cb9-70"><a href="#cb9-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-29-qPCR---M.gigas-Valentina-PolyIC-VIPERIN/sam_2026-01-29_07-21-16_Connect-VIPERIN-01-QuantificationCq_Results.csv"</span>, </span>
<span id="cb9-71"><a href="#cb9-71" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"VIPERIN"</span>, <span class="at">gene =</span> <span class="st">"VIPERIN"</span>),</span>
<span id="cb9-72"><a href="#cb9-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-29-qPCR---M.gigas-Valentina-PolyIC-VIPERIN/sam_2026-01-29_07-39-06_CFX96-VIPERIN-02-Quantification-Cq_Results.csv"</span>, </span>
<span id="cb9-73"><a href="#cb9-73" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"VIPERIN"</span>, <span class="at">gene =</span> <span class="st">"VIPERIN"</span>),</span>
<span id="cb9-74"><a href="#cb9-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">url =</span> <span class="st">"https://raw.githubusercontent.com/RobertsLab/sams-notebook/refs/heads/master/posts/2026/2026-01-29-qPCR---M.gigas-Valentina-PolyIC-VIPERIN/sam_2026-01-29_08-24-59_Connect-VIPERIN-03-QuantificationCq_Results.csv"</span>, </span>
<span id="cb9-75"><a href="#cb9-75" aria-hidden="true" tabindex="-1"></a>       <span class="at">target =</span> <span class="st">"VIPERIN"</span>, <span class="at">gene =</span> <span class="st">"VIPERIN"</span>)</span>
<span id="cb9-76"><a href="#cb9-76" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb9-77"><a href="#cb9-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-78"><a href="#cb9-78" aria-hidden="true" tabindex="-1"></a><span class="co"># Load all files and combine automatically</span></span>
<span id="cb9-79"><a href="#cb9-79" aria-hidden="true" tabindex="-1"></a>qpcr_data <span class="ot">&lt;-</span> <span class="fu">map_df</span>(file_list, <span class="cf">function</span>(file_info) {</span>
<span id="cb9-80"><a href="#cb9-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read.csv</span>(file_info<span class="sc">$</span>url) <span class="sc">%&gt;%</span></span>
<span id="cb9-81"><a href="#cb9-81" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Target =</span> file_info<span class="sc">$</span>target)</span>
<span id="cb9-82"><a href="#cb9-82" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="functions" class="level1">
<h1>Functions</h1>
<section id="calculate-delta-cq" class="level2">
<h2 class="anchored" data-anchor-id="calculate-delta-cq">Calculate delta Cq</h2>
<p>Normalized to designated normalizing gene</p>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>calculate_delta_Cq <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Sample) <span class="sc">%&gt;%</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">delta_Cq =</span> Cq.Mean <span class="sc">-</span> Cq.Mean[Target <span class="sc">==</span> <span class="st">"GAPDH"</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>()</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="data-cleaning" class="level1">
<h1>Data Cleaning</h1>
<section id="remove-ntc-samples" class="level2">
<h2 class="anchored" data-anchor-id="remove-ntc-samples">Remove NTC samples</h2>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with Sample name "NTC" or empty/NA Sample names</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>qpcr_data <span class="ot">&lt;-</span> qpcr_data <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sample <span class="sc">!=</span> <span class="st">"NTC"</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(Sample) <span class="sc">&amp;</span> Sample <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(qpcr_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>'data.frame':   2463 obs. of  16 variables:
 $ X                     : logi  NA NA NA NA NA NA ...
 $ Well                  : chr  "A01" "A02" "A03" "A04" ...
 $ Fluor                 : chr  "SYBR" "SYBR" "SYBR" "SYBR" ...
 $ Target                : chr  "Citrate.Synthase" "Citrate.Synthase" "Citrate.Synthase" "Citrate.Synthase" ...
 $ Content               : chr  "Unkn-01" "Unkn-01" "Unkn-01" "Unkn-02" ...
 $ Sample                : chr  "A1C" "A1C" "A1C" "A2C" ...
 $ Biological.Set.Name   : logi  NA NA NA NA NA NA ...
 $ Cq                    : num  24.1 24 23.8 25.1 24.7 ...
 $ Cq.Mean               : num  23.9 23.9 23.9 24.9 24.9 ...
 $ Cq.Std..Dev           : num  0.147 0.147 0.147 0.208 0.208 ...
 $ Starting.Quantity..SQ.: num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Log.Starting.Quantity : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Mean               : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Std..Dev           : num  NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Set.Point             : int  60 60 60 60 60 60 60 60 60 60 ...
 $ Well.Note             : logi  NA NA NA NA NA NA ...</code></pre>
</div>
</div>
</section>
<section id="check-for-high-standard-deviations" class="level2">
<h2 class="anchored" data-anchor-id="check-for-high-standard-deviations">Check for high standard deviations</h2>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out rows where Cq.Std..Dev is NA</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>qpcr_data <span class="ot">&lt;-</span> qpcr_data[<span class="sc">!</span><span class="fu">is.na</span>(qpcr_data<span class="sc">$</span>Cq.Std..Dev), ]</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter rows where Cq.Std..Dev is greater than 0.5</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>high_cq_std_dev <span class="ot">&lt;-</span> qpcr_data[qpcr_data<span class="sc">$</span>Cq.Std..Dev <span class="sc">&gt;</span> <span class="fl">0.5</span>, ]</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the filtered rows with specified columns, without row names</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(high_cq_std_dev[, <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"Sample"</span>, <span class="st">"Cq"</span>, <span class="st">"Cq.Std..Dev"</span>)], <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>           Target Sample       Cq Cq.Std..Dev
 Citrate.Synthase   D4PC 23.58953   1.2262012
 Citrate.Synthase   D4PC 25.22609   1.2262012
 Citrate.Synthase   D4PC 25.98956   1.2262012
 Citrate.Synthase   C1PT 24.22698   0.6305806
 Citrate.Synthase   C1PT 24.09781   0.6305806
 Citrate.Synthase   C1PT 25.24885   0.6305806
 Citrate.Synthase   C5PT 24.30653   0.6151831
 Citrate.Synthase   C5PT 24.05928   0.6151831
 Citrate.Synthase   C5PT 25.22670   0.6151831
            DNMT1    A5M 30.84670   1.9690822
            DNMT1    A5M 27.36680   1.9690822
            DNMT1    A5M 30.70349   1.9690822
            DNMT1   D4PC 31.61104   0.6475757
            DNMT1   D4PC 31.51718   0.6475757
            DNMT1   D4PC 30.44542   0.6475757
            DNMT1   D5PC 31.81420   0.8774717
            DNMT1   D5PC 30.09977   0.8774717
            DNMT1   D5PC 31.28165   0.8774717
            DNMT1   D4PM 31.84097   0.6990197
            DNMT1   D4PM 31.56654   0.6990197
            DNMT1   D4PM 32.89094   0.6990197
            DNMT1   A3PT 31.37176   0.6647356
            DNMT1   A3PT 31.00085   0.6647356
            DNMT1   A3PT 32.29195   0.6647356
            DNMT1   C3PT 30.93391   1.9559044
            DNMT1   C3PT 34.32213   1.9559044
            DNMT1   C3PT 30.93490   1.9559044
            DNMT1   D4PT 28.51543   0.8274428
            DNMT1   D4PT 29.40357   0.8274428
            DNMT1   D4PT 30.16879   0.8274428
             cGAS    A1C 31.09889   0.5958513
             cGAS    A1C 30.51096   0.5958513
             cGAS    A1C 29.90723   0.5958513
             cGAS    A3C 30.71172   0.6725415
             cGAS    A3C 30.23246   0.6725415
             cGAS    A3C 29.38366   0.6725415
             cGAS    A4C 32.47341   0.5704754
             cGAS    A4C 31.67087   0.5704754
             cGAS    A4C 31.36981   0.5704754
             cGAS    B1C 28.69140   1.5243846
             cGAS    B1C 31.55769   1.5243846
             cGAS    B1C 29.22481   1.5243846
             cGAS    C1C 29.16134   0.7172895
             cGAS    C1C 29.31145   0.7172895
             cGAS    C1C 30.47195   0.7172895
             cGAS    C3M 29.88386   1.8065816
             cGAS    C3M 29.81184   1.8065816
             cGAS    C3M 32.97631   1.8065816
             cGAS    A5T 33.66309   2.4759798
             cGAS    A5T 29.55374   2.4759798
             cGAS    A5T 29.21543   2.4759798
             cGAS   D4PC 29.52347   0.7216823
             cGAS   D4PC 29.38534   0.7216823
             cGAS   D4PC 30.69866   0.7216823
             cGAS   B2PT 30.33579   2.9187950
             cGAS   B2PT 35.74373   2.9187950
             cGAS   B2PT 31.13614   2.9187950
             cGAS   C2PT 29.41279   0.8378923
             cGAS   C2PT 29.04691   0.8378923
             cGAS   C2PT 30.64611   0.8378923
             cGAS   D1PT 31.53902   1.1351264
             cGAS   D1PT 33.08089   1.1351264
             cGAS   D1PT 33.75305   1.1351264
     ATP.Synthase    B1M 21.94408   0.6345520
     ATP.Synthase    B1M 22.95451   0.6345520
     ATP.Synthase    B1M 23.11430   0.6345520
     ATP.Synthase   C3PC 24.15370   0.5694444
     ATP.Synthase   C3PC 23.19252   0.5694444
     ATP.Synthase   C3PC 23.14405   0.5694444
     ATP.Synthase   D3PT 24.70960   0.5708900
     ATP.Synthase   D3PT 24.60682   0.5708900
     ATP.Synthase   D3PT 25.64301   0.5708900
     ATP.Synthase   A1PC 29.49846   3.7832839
     ATP.Synthase   A1PC 30.53143   3.7832839
     ATP.Synthase   A1PC 23.52346   3.7832839
     ATP.Synthase   A2PC 29.79862   4.1202801
     ATP.Synthase   A2PC 30.34860   4.1202801
     ATP.Synthase   A2PC 22.95298   4.1202801
     ATP.Synthase   A4PC 31.51927   0.8092799
     ATP.Synthase   A4PC 32.72186   0.8092799
     ATP.Synthase   A4PC 33.05872   0.8092799
     ATP.Synthase   A2PC 29.22068   1.2680354
     ATP.Synthase   A2PC 31.04155   1.2680354
     ATP.Synthase   A2PC 31.65986   1.2680354
     ATP.Synthase   A1PM 25.88041   0.5968540
     ATP.Synthase   A1PM 25.38009   0.5968540
     ATP.Synthase   A1PM 26.56884   0.5968540
     ATP.Synthase   B1PC 28.20285   3.0574664
     ATP.Synthase   B1PC 33.21694   3.0574664
     ATP.Synthase   B1PC 33.74113   3.0574664
     ATP.Synthase   B2PC 31.61585   2.4554921
     ATP.Synthase   B2PC 26.77269   2.4554921
     ATP.Synthase   B2PC 28.48989   2.4554921
     ATP.Synthase   B5PC 33.25991   0.9734783
     ATP.Synthase   B5PC 33.02801   0.9734783
     ATP.Synthase   B5PC 34.81807   0.9734783
     ATP.Synthase   A2PM 31.87285   4.1105820
     ATP.Synthase   A2PM 25.74462   4.1105820
     ATP.Synthase   A2PM 24.06276   4.1105820
     ATP.Synthase   A3PM 37.05442   6.7379697
     ATP.Synthase   A3PM 23.69949   6.7379697
     ATP.Synthase   A3PM 31.93743   6.7379697
     ATP.Synthase   A4PM 30.86969   1.3433926
     ATP.Synthase   A4PM 31.10307   1.3433926
     ATP.Synthase   A4PM 33.30441   1.3433926
     ATP.Synthase   A5PM 23.07786   1.7068305
     ATP.Synthase   A5PM 23.92584   1.7068305
     ATP.Synthase   A5PM 26.36550   1.7068305
     ATP.Synthase   B1PM 31.02730   5.0891703
     ATP.Synthase   B1PM 23.83012   5.0891703
     ATP.Synthase   B2PM 30.27510   4.3654487
     ATP.Synthase   B2PM 24.10142   4.3654487
     ATP.Synthase   B3PM 38.17801   8.9249343
     ATP.Synthase   B3PM 22.96081   8.9249343
     ATP.Synthase   B3PM 22.48913   8.9249343
            GAPDH    A3T 22.96407   0.6473489
            GAPDH    A3T 23.45998   0.6473489
            GAPDH    A3T 24.24775   0.6473489
            GAPDH   B1PC 29.43218   1.2082882
            GAPDH   B1PC 29.11486   1.2082882
            GAPDH   B1PC 27.19883   1.2082882
            GAPDH   B5PC 27.69885   0.7892948
            GAPDH   B5PC 27.58553   0.7892948
            GAPDH   B5PC 29.00576   0.7892948
            GAPDH   B4PM 26.79585   1.2035459
            GAPDH   B4PM 28.79708   1.2035459
            GAPDH   B4PM 28.95486   1.2035459
            GAPDH   B5PM 29.58450   0.8701184
            GAPDH   B5PM 29.22245   0.8701184
            GAPDH   B5PM 27.92936   0.8701184
            HSP70    B5M 33.74524   0.8087996
            HSP70    B5M 32.37369   0.8087996
            HSP70    B5M 32.31676   0.8087996
            HSP70   B4PC 30.71240   0.6802331
            HSP70   B4PC 29.83791   0.6802331
            HSP70   B4PC 29.37261   0.6802331
            HSP70   A1PM 29.45236   3.1088225
            HSP70   A1PM 29.73179   3.1088225
            HSP70   A1PM 24.21288   3.1088225
            HSP70   A4PM 33.09655   0.6821469
            HSP70   A4PM 31.81483   0.6821469
            HSP70   A4PM 32.86049   0.6821469
            HSP70   B1PM 27.93827   0.5637303
            HSP70   B1PM 28.32120   0.5637303
            HSP70   B1PM 29.04810   0.5637303
            HSP90    B2C 24.79178   2.3098674
            HSP90    B2C 28.85831   2.3098674
            HSP90    B2C 24.92664   2.3098674</code></pre>
</div>
</div>
</section>
<section id="remove-outlier-technical-replicates" class="level2">
<h2 class="anchored" data-anchor-id="remove-outlier-technical-replicates">Remove outlier technical replicates</h2>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by Sample and Target, then filter out the outlier replicate</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>qpcr_filtered <span class="ot">&lt;-</span> qpcr_data <span class="sc">%&gt;%</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Sample, Target) <span class="sc">%&gt;%</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">n</span>() <span class="sc">==</span> <span class="dv">1</span> <span class="sc">|</span> <span class="fu">abs</span>(Cq <span class="sc">-</span> <span class="fu">median</span>(Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">&lt;=</span> <span class="fl">0.5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(qpcr_filtered)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [2,335 × 16] (S3: tbl_df/tbl/data.frame)
 $ X                     : logi [1:2335] NA NA NA NA NA NA ...
 $ Well                  : chr [1:2335] "A01" "A02" "A03" "A04" ...
 $ Fluor                 : chr [1:2335] "SYBR" "SYBR" "SYBR" "SYBR" ...
 $ Target                : chr [1:2335] "Citrate.Synthase" "Citrate.Synthase" "Citrate.Synthase" "Citrate.Synthase" ...
 $ Content               : chr [1:2335] "Unkn-01" "Unkn-01" "Unkn-01" "Unkn-02" ...
 $ Sample                : chr [1:2335] "A1C" "A1C" "A1C" "A2C" ...
 $ Biological.Set.Name   : logi [1:2335] NA NA NA NA NA NA ...
 $ Cq                    : num [1:2335] 24.1 24 23.8 25.1 24.7 ...
 $ Cq.Mean               : num [1:2335] 23.9 23.9 23.9 24.9 24.9 ...
 $ Cq.Std..Dev           : num [1:2335] 0.147 0.147 0.147 0.208 0.208 ...
 $ Starting.Quantity..SQ.: num [1:2335] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Log.Starting.Quantity : num [1:2335] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Mean               : num [1:2335] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ SQ.Std..Dev           : num [1:2335] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...
 $ Set.Point             : int [1:2335] 60 60 60 60 60 60 60 60 60 60 ...
 $ Well.Note             : logi [1:2335] NA NA NA NA NA NA ...</code></pre>
</div>
</div>
<section id="samples-excluded-due-to-quality-control" class="level3">
<h3 class="anchored" data-anchor-id="samples-excluded-due-to-quality-control">Samples excluded due to quality control</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify samples that were filtered out</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>all_samples <span class="ot">&lt;-</span> qpcr_data <span class="sc">%&gt;%</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Sample, Target) <span class="sc">%&gt;%</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>retained_samples <span class="ot">&lt;-</span> qpcr_filtered <span class="sc">%&gt;%</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Sample, Target) <span class="sc">%&gt;%</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>excluded_samples <span class="ot">&lt;-</span> all_samples <span class="sc">%&gt;%</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anti_join</span>(retained_samples, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Sample"</span>, <span class="st">"Target"</span>))</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(excluded_samples) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Samples excluded from analysis due to high standard deviation (&gt; 0.5):</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  excluded_samples <span class="sc">%&gt;%</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(Target, Sample) <span class="sc">%&gt;%</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>    knitr<span class="sc">::</span><span class="fu">kable</span>()</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"No samples were excluded from the analysis.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Samples excluded from analysis due to high standard deviation (&gt; 0.5):</code></pre>
</div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">Sample</th>
<th style="text-align: left;">Target</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B1PM</td>
<td style="text-align: left;">ATP.Synthase</td>
</tr>
<tr class="even">
<td style="text-align: left;">B2PM</td>
<td style="text-align: left;">ATP.Synthase</td>
</tr>
<tr class="odd">
<td style="text-align: left;">A2PM</td>
<td style="text-align: left;">Citrate.Synthase</td>
</tr>
<tr class="even">
<td style="text-align: left;">D4PM</td>
<td style="text-align: left;">DNMT1</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
</section>
</section>
<section id="group-and-summarize-data" class="level1">
<h1>Group and Summarize Data</h1>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by Sample and Target, then summarize to get unique rows for each sample</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>grouped_df <span class="ot">&lt;-</span> qpcr_filtered <span class="sc">%&gt;%</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Sample, Target) <span class="sc">%&gt;%</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">Cq.Mean =</span> <span class="fu">mean</span>(Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">Cq.Std..Dev =</span> <span class="fu">sd</span>(Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">'drop'</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(grouped_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [811 × 4] (S3: tbl_df/tbl/data.frame)
 $ Sample     : chr [1:811] "A1C" "A1C" "A1C" "A1C" ...
 $ Target     : chr [1:811] "ATP.Synthase" "Citrate.Synthase" "DNMT1" "GAPDH" ...
 $ Cq.Mean    : num [1:811] 23.6 23.9 29.7 23.6 30.4 ...
 $ Cq.Std..Dev: num [1:811] 0.0674 0.1467 0.1991 0.3435 0.2572 ...</code></pre>
</div>
</div>
</section>
<section id="parse-sample-names-and-add-treatment-columns" class="level1">
<h1>Parse Sample Names and Add Treatment Columns</h1>
<p>Sample naming convention: Letter + Number + P (optional) + Treatment</p>
<ul>
<li>Letter: Biological replicate</li>
<li>Number: Replicate number</li>
<li>P: PolyIC treatment (absent if no PolyIC)</li>
<li>C/T/M: Control/Temperature/Mechanical stress</li>
</ul>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Parse sample names to extract treatment information</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>grouped_df <span class="ot">&lt;-</span> grouped_df <span class="sc">%&gt;%</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Extract biological replicate (first letter)</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">bio_replicate =</span> <span class="fu">str_extract</span>(Sample, <span class="st">"^[A-Z]"</span>),</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Extract replicate number</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">replicate_num =</span> <span class="fu">str_extract</span>(Sample, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>),</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if PolyIC treatment (contains P)</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">polyIC =</span> <span class="fu">ifelse</span>(<span class="fu">str_detect</span>(Sample, <span class="st">"P"</span>), <span class="st">"PolyIC"</span>, <span class="st">"No_PolyIC"</span>),</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Extract stress type (last letter: C, T, or M)</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">stress_type =</span> <span class="fu">case_when</span>(</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_detect</span>(Sample, <span class="st">"C$"</span>) <span class="sc">~</span> <span class="st">"Control"</span>,</span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_detect</span>(Sample, <span class="st">"T$"</span>) <span class="sc">~</span> <span class="st">"Temperature"</span>,</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_detect</span>(Sample, <span class="st">"M$"</span>) <span class="sc">~</span> <span class="st">"Mechanical"</span>,</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA_character_</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create combined treatment group</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">treatment_group =</span> <span class="fu">paste</span>(polyIC, stress_type, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(grouped_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [811 × 9] (S3: tbl_df/tbl/data.frame)
 $ Sample         : chr [1:811] "A1C" "A1C" "A1C" "A1C" ...
 $ Target         : chr [1:811] "ATP.Synthase" "Citrate.Synthase" "DNMT1" "GAPDH" ...
 $ Cq.Mean        : num [1:811] 23.6 23.9 29.7 23.6 30.4 ...
 $ Cq.Std..Dev    : num [1:811] 0.0674 0.1467 0.1991 0.3435 0.2572 ...
 $ bio_replicate  : chr [1:811] "A" "A" "A" "A" ...
 $ replicate_num  : chr [1:811] "1" "1" "1" "1" ...
 $ polyIC         : chr [1:811] "No_PolyIC" "No_PolyIC" "No_PolyIC" "No_PolyIC" ...
 $ stress_type    : chr [1:811] "Control" "Control" "Control" "Control" ...
 $ treatment_group: chr [1:811] "No_PolyIC_Control" "No_PolyIC_Control" "No_PolyIC_Control" "No_PolyIC_Control" ...</code></pre>
</div>
</div>
<section id="view-treatment-groups" class="level2">
<h2 class="anchored" data-anchor-id="view-treatment-groups">View treatment groups</h2>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the unique treatment groups</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(grouped_df<span class="sc">$</span>treatment_group)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "No_PolyIC_Control"     "No_PolyIC_Mechanical"  "PolyIC_Control"       
[4] "PolyIC_Mechanical"     "PolyIC_Temperature"    "No_PolyIC_Temperature"</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Count samples per treatment group</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>grouped_df <span class="sc">%&gt;%</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(treatment_group, Target) <span class="sc">%&gt;%</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_samples =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">'drop'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Target, <span class="at">values_from =</span> n_samples) <span class="sc">%&gt;%</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 24%">
<col style="width: 14%">
<col style="width: 19%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 8%">
<col style="width: 5%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">treatment_group</th>
<th style="text-align: right;">ATP.Synthase</th>
<th style="text-align: right;">Citrate.Synthase</th>
<th style="text-align: right;">DNMT1</th>
<th style="text-align: right;">GAPDH</th>
<th style="text-align: right;">HSP70</th>
<th style="text-align: right;">HSP90</th>
<th style="text-align: right;">VIPERIN</th>
<th style="text-align: right;">cGAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">No_PolyIC_Control</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
</tr>
<tr class="even">
<td style="text-align: left;">No_PolyIC_Mechanical</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
</tr>
<tr class="odd">
<td style="text-align: left;">No_PolyIC_Temperature</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
</tr>
<tr class="even">
<td style="text-align: left;">PolyIC_Control</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PolyIC_Mechanical</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">PolyIC_Temperature</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
</section>
<section id="calculate-delta-cq-1" class="level1">
<h1>Calculate Delta Cq</h1>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta Cq by subtracting GAPDH Cq.Mean from each corresponding Sample Cq.Mean</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>delta_Cq_df <span class="ot">&lt;-</span> <span class="fu">calculate_delta_Cq</span>(grouped_df)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out normalizing gene (GAPDH), since no need to compare normalizing gene to itself</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>delta_Cq_df <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">!=</span> <span class="st">"GAPDH"</span>)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(delta_Cq_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [699 × 10] (S3: tbl_df/tbl/data.frame)
 $ Sample         : chr [1:699] "A1C" "A1C" "A1C" "A1C" ...
 $ Target         : chr [1:699] "ATP.Synthase" "Citrate.Synthase" "DNMT1" "HSP70" ...
 $ Cq.Mean        : num [1:699] 23.6 23.9 29.7 30.4 24.3 ...
 $ Cq.Std..Dev    : num [1:699] 0.0674 0.1467 0.1991 0.2572 0.0472 ...
 $ bio_replicate  : chr [1:699] "A" "A" "A" "A" ...
 $ replicate_num  : chr [1:699] "1" "1" "1" "1" ...
 $ polyIC         : chr [1:699] "No_PolyIC" "No_PolyIC" "No_PolyIC" "No_PolyIC" ...
 $ stress_type    : chr [1:699] "Control" "Control" "Control" "Control" ...
 $ treatment_group: chr [1:699] "No_PolyIC_Control" "No_PolyIC_Control" "No_PolyIC_Control" "No_PolyIC_Control" ...
 $ delta_Cq       : num [1:699] 0.0291 0.3858 6.1407 6.8687 0.7935 ...</code></pre>
</div>
</div>
<section id="missing-samples-analysis" class="level2">
<h2 class="anchored" data-anchor-id="missing-samples-analysis">Missing samples analysis</h2>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get all unique sample names from the data that made it to delta_Cq_df</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>available_samples <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Sample, Target, treatment_group, polyIC, stress_type) <span class="sc">%&gt;%</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Expected samples based on the naming pattern</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="co"># From the view-groups output, we know the expected samples</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>expected_no_polyIC <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="fu">rep</span>(LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">each=</span><span class="dv">5</span>), <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">4</span>), <span class="st">"C"</span>),</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">paste0</span>(<span class="fu">rep</span>(LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">each=</span><span class="dv">5</span>), <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">4</span>), <span class="st">"T"</span>),</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">paste0</span>(<span class="fu">rep</span>(LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">each=</span><span class="dv">5</span>), <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">4</span>), <span class="st">"M"</span>))</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>expected_polyIC <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="fu">rep</span>(LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">each=</span><span class="dv">5</span>), <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">4</span>), <span class="st">"PC"</span>),</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">paste0</span>(<span class="fu">rep</span>(LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">each=</span><span class="dv">5</span>), <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">4</span>), <span class="st">"PT"</span>),</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">paste0</span>(<span class="fu">rep</span>(LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">each=</span><span class="dv">5</span>), <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">4</span>), <span class="st">"PM"</span>))</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>all_expected_samples <span class="ot">&lt;-</span> <span class="fu">c</span>(expected_no_polyIC, expected_polyIC)</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Get all genes</span></span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(delta_Cq_df<span class="sc">$</span>Target)</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Create expected sample-gene combinations</span></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>expected_combinations <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">Sample =</span> all_expected_samples,</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">Target =</span> all_genes,</span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Find missing combinations</span></span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a>actual_combinations <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Sample, Target) <span class="sc">%&gt;%</span></span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a>missing_combinations <span class="ot">&lt;-</span> expected_combinations <span class="sc">%&gt;%</span></span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anti_join</span>(actual_combinations, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Sample"</span>, <span class="st">"Target"</span>))</span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize by treatment group and gene</span></span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Parse treatment info for missing samples</span></span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a>missing_combinations_parsed <span class="ot">&lt;-</span> missing_combinations <span class="sc">%&gt;%</span></span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">polyIC =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"P"</span>, <span class="fu">substr</span>(Sample, <span class="fu">nchar</span>(Sample)<span class="sc">-</span><span class="dv">1</span>, <span class="fu">nchar</span>(Sample))), <span class="st">"PolyIC"</span>, <span class="st">"No_PolyIC"</span>),</span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">stress_type =</span> <span class="fu">case_when</span>(</span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a>      <span class="fu">grepl</span>(<span class="st">"C$"</span>, Sample) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">"PC$"</span>, Sample) <span class="sc">~</span> <span class="st">"Control"</span>,</span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a>      <span class="fu">grepl</span>(<span class="st">"T$"</span>, Sample) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">"PT$"</span>, Sample) <span class="sc">~</span> <span class="st">"Temperature"</span>,</span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a>      <span class="fu">grepl</span>(<span class="st">"M$"</span>, Sample) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">"PM$"</span>, Sample) <span class="sc">~</span> <span class="st">"Mechanical"</span>,</span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Unknown"</span></span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">treatment_group =</span> <span class="fu">paste</span>(polyIC, stress_type, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb28-48"><a href="#cb28-48" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb28-49"><a href="#cb28-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-50"><a href="#cb28-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Summary table</span></span>
<span id="cb28-51"><a href="#cb28-51" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">=== MISSING SAMPLES SUMMARY BY TREATMENT GROUP ===</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb28-52"><a href="#cb28-52" aria-hidden="true" tabindex="-1"></a>missing_summary <span class="ot">&lt;-</span> missing_combinations_parsed <span class="sc">%&gt;%</span></span>
<span id="cb28-53"><a href="#cb28-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(treatment_group, Target) <span class="sc">%&gt;%</span></span>
<span id="cb28-54"><a href="#cb28-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_missing =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-55"><a href="#cb28-55" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Target, <span class="at">values_from =</span> n_missing, <span class="at">values_fill =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-56"><a href="#cb28-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(treatment_group)</span>
<span id="cb28-57"><a href="#cb28-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-58"><a href="#cb28-58" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(missing_summary)</span>
<span id="cb28-59"><a href="#cb28-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-60"><a href="#cb28-60" aria-hidden="true" tabindex="-1"></a><span class="co"># Detailed list of missing samples</span></span>
<span id="cb28-61"><a href="#cb28-61" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">=== DETAILED LIST OF MISSING SAMPLES ===</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb28-62"><a href="#cb28-62" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Total missing sample-gene combinations:"</span>, <span class="fu">nrow</span>(missing_combinations), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb28-63"><a href="#cb28-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-64"><a href="#cb28-64" aria-hidden="true" tabindex="-1"></a>missing_by_gene <span class="ot">&lt;-</span> missing_combinations_parsed <span class="sc">%&gt;%</span></span>
<span id="cb28-65"><a href="#cb28-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Target, treatment_group, Sample)</span>
<span id="cb28-66"><a href="#cb28-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-67"><a href="#cb28-67" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(gene <span class="cf">in</span> <span class="fu">sort</span>(<span class="fu">unique</span>(missing_by_gene<span class="sc">$</span>Target))) {</span>
<span id="cb28-68"><a href="#cb28-68" aria-hidden="true" tabindex="-1"></a>  gene_missing <span class="ot">&lt;-</span> missing_by_gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> gene)</span>
<span id="cb28-69"><a href="#cb28-69" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(gene_missing) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb28-70"><a href="#cb28-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**"</span>, gene, <span class="st">"** ("</span>, <span class="fu">nrow</span>(gene_missing), <span class="st">" missing samples):</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb28-71"><a href="#cb28-71" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(trt <span class="cf">in</span> <span class="fu">unique</span>(gene_missing<span class="sc">$</span>treatment_group)) {</span>
<span id="cb28-72"><a href="#cb28-72" aria-hidden="true" tabindex="-1"></a>      samples <span class="ot">&lt;-</span> gene_missing <span class="sc">%&gt;%</span> <span class="fu">filter</span>(treatment_group <span class="sc">==</span> trt) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Sample)</span>
<span id="cb28-73"><a href="#cb28-73" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">length</span>(samples) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb28-74"><a href="#cb28-74" aria-hidden="true" tabindex="-1"></a>        <span class="fu">cat</span>(<span class="st">"  "</span>, trt, <span class="st">": "</span>, <span class="fu">paste</span>(samples, <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb28-75"><a href="#cb28-75" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb28-76"><a href="#cb28-76" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb28-77"><a href="#cb28-77" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb28-78"><a href="#cb28-78" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-79"><a href="#cb28-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-80"><a href="#cb28-80" aria-hidden="true" tabindex="-1"></a><span class="co"># Comprehensive list of all problematic samples</span></span>
<span id="cb28-81"><a href="#cb28-81" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">=== ALL SAMPLES EXCLUDED OR MISSING FROM ANALYSIS ===</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb28-82"><a href="#cb28-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-83"><a href="#cb28-83" aria-hidden="true" tabindex="-1"></a><span class="co"># Get samples that failed QC by gene</span></span>
<span id="cb28-84"><a href="#cb28-84" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**Samples that Failed QC (high standard deviation &gt; 0.5) by Gene:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb28-85"><a href="#cb28-85" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(excluded_samples) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb28-86"><a href="#cb28-86" aria-hidden="true" tabindex="-1"></a>  failed_qc_by_gene <span class="ot">&lt;-</span> excluded_samples <span class="sc">%&gt;%</span></span>
<span id="cb28-87"><a href="#cb28-87" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(Target, Sample)</span>
<span id="cb28-88"><a href="#cb28-88" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-89"><a href="#cb28-89" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(gene <span class="cf">in</span> <span class="fu">unique</span>(failed_qc_by_gene<span class="sc">$</span>Target)) {</span>
<span id="cb28-90"><a href="#cb28-90" aria-hidden="true" tabindex="-1"></a>    gene_failed <span class="ot">&lt;-</span> failed_qc_by_gene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> gene) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Sample)</span>
<span id="cb28-91"><a href="#cb28-91" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"  "</span>, gene, <span class="st">" ("</span>, <span class="fu">length</span>(gene_failed), <span class="st">" samples): "</span>, <span class="fu">paste</span>(<span class="fu">sort</span>(gene_failed), <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb28-92"><a href="#cb28-92" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb28-93"><a href="#cb28-93" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-94"><a href="#cb28-94" aria-hidden="true" tabindex="-1"></a>  failed_qc_samples <span class="ot">&lt;-</span> <span class="fu">unique</span>(excluded_samples<span class="sc">$</span>Sample)</span>
<span id="cb28-95"><a href="#cb28-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">  Total unique samples that failed QC: "</span>, <span class="fu">length</span>(failed_qc_samples), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb28-96"><a href="#cb28-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  All failed QC samples: "</span>, <span class="fu">paste</span>(<span class="fu">sort</span>(failed_qc_samples), <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb28-97"><a href="#cb28-97" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb28-98"><a href="#cb28-98" aria-hidden="true" tabindex="-1"></a>  failed_qc_samples <span class="ot">&lt;-</span> <span class="fu">character</span>(<span class="dv">0</span>)</span>
<span id="cb28-99"><a href="#cb28-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  No samples failed QC.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb28-100"><a href="#cb28-100" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-101"><a href="#cb28-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-102"><a href="#cb28-102" aria-hidden="true" tabindex="-1"></a><span class="co"># Get unique samples that are missing</span></span>
<span id="cb28-103"><a href="#cb28-103" aria-hidden="true" tabindex="-1"></a>missing_sample_names <span class="ot">&lt;-</span> <span class="fu">unique</span>(missing_combinations<span class="sc">$</span>Sample)</span>
<span id="cb28-104"><a href="#cb28-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-105"><a href="#cb28-105" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**Missing from data (never run/loaded):** "</span>, <span class="fu">length</span>(missing_sample_names), <span class="st">" unique samples</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb28-106"><a href="#cb28-106" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">length</span>(missing_sample_names) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb28-107"><a href="#cb28-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  "</span>, <span class="fu">paste</span>(<span class="fu">sort</span>(missing_sample_names), <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb28-108"><a href="#cb28-108" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-109"><a href="#cb28-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-110"><a href="#cb28-110" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine and get all unique</span></span>
<span id="cb28-111"><a href="#cb28-111" aria-hidden="true" tabindex="-1"></a>all_problematic_samples <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(failed_qc_samples, missing_sample_names))</span>
<span id="cb28-112"><a href="#cb28-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-113"><a href="#cb28-113" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**Summary:**</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb28-114"><a href="#cb28-114" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"  Total unique samples with issues: "</span>, <span class="fu">length</span>(all_problematic_samples), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb28-115"><a href="#cb28-115" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"  All unique sample names (failed QC + missing): "</span>, <span class="fu">paste</span>(<span class="fu">sort</span>(all_problematic_samples), <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
=== MISSING SAMPLES SUMMARY BY TREATMENT GROUP ===

# A tibble: 2 × 8
  treatment_group  ATP.Synthase Citrate.Synthase DNMT1 HSP70 HSP90 VIPERIN  cGAS
  &lt;chr&gt;                   &lt;int&gt;            &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt;   &lt;int&gt; &lt;int&gt;
1 PolyIC_Control              3                2    12     2    12      12    12
2 PolyIC_Mechanic…            8                7    17     6    16      16    16


=== DETAILED LIST OF MISSING SAMPLES ===

Total missing sample-gene combinations: 141 


**ATP.Synthase** (11 missing samples):
  PolyIC_Control: A5PC, C4PC, C5PC
  PolyIC_Mechanical: B1PM, B2PM, C1PM, C2PM, C3PM, C4PM, C5PM, D5PM

**cGAS** (28 missing samples):
  PolyIC_Control: A1PC, A2PC, A3PC, A4PC, A5PC, B1PC, B2PC, B3PC, B4PC, B5PC, C4PC, C5PC
  PolyIC_Mechanical: A1PM, A2PM, A3PM, A4PM, A5PM, B1PM, B2PM, B3PM, B4PM, B5PM, C1PM, C2PM, C3PM, C4PM, C5PM, D5PM

**Citrate.Synthase** (9 missing samples):
  PolyIC_Control: C4PC, C5PC
  PolyIC_Mechanical: A2PM, C1PM, C2PM, C3PM, C4PM, C5PM, D5PM

**DNMT1** (29 missing samples):
  PolyIC_Control: A1PC, A2PC, A3PC, A4PC, A5PC, B1PC, B2PC, B3PC, B4PC, B5PC, C4PC, C5PC
  PolyIC_Mechanical: A1PM, A2PM, A3PM, A4PM, A5PM, B1PM, B2PM, B3PM, B4PM, B5PM, C1PM, C2PM, C3PM, C4PM, C5PM, D4PM, D5PM

**HSP70** (8 missing samples):
  PolyIC_Control: C4PC, C5PC
  PolyIC_Mechanical: C1PM, C2PM, C3PM, C4PM, C5PM, D5PM

**HSP90** (28 missing samples):
  PolyIC_Control: A1PC, A2PC, A3PC, A4PC, A5PC, B1PC, B2PC, B3PC, B4PC, B5PC, C4PC, C5PC
  PolyIC_Mechanical: A1PM, A2PM, A3PM, A4PM, A5PM, B1PM, B2PM, B3PM, B4PM, B5PM, C1PM, C2PM, C3PM, C4PM, C5PM, D5PM

**VIPERIN** (28 missing samples):
  PolyIC_Control: A1PC, A2PC, A3PC, A4PC, A5PC, B1PC, B2PC, B3PC, B4PC, B5PC, C4PC, C5PC
  PolyIC_Mechanical: A1PM, A2PM, A3PM, A4PM, A5PM, B1PM, B2PM, B3PM, B4PM, B5PM, C1PM, C2PM, C3PM, C4PM, C5PM, D5PM


=== ALL SAMPLES EXCLUDED OR MISSING FROM ANALYSIS ===

**Samples that Failed QC (high standard deviation &gt; 0.5) by Gene:**

  ATP.Synthase (2 samples): B1PM, B2PM
  Citrate.Synthase (1 samples): A2PM
  DNMT1 (1 samples): D4PM

  Total unique samples that failed QC: 4
  All failed QC samples: A2PM, B1PM, B2PM, D4PM

**Missing from data (never run/loaded):** 29 unique samples
   A1PC, A1PM, A2PC, A2PM, A3PC, A3PM, A4PC, A4PM, A5PC, A5PM, B1PC, B1PM, B2PC, B2PM, B3PC, B3PM, B4PC, B4PM, B5PC, B5PM, C1PM, C2PM, C3PM, C4PC, C4PM, C5PC, C5PM, D4PM, D5PM 

**Summary:**
  Total unique samples with issues: 29
  All unique sample names (failed QC + missing): A1PC, A1PM, A2PC, A2PM, A3PC, A3PM, A4PC, A4PM, A5PC, A5PM, B1PC, B1PM, B2PC, B2PM, B3PC, B3PM, B4PC, B4PM, B5PC, B5PM, C1PM, C2PM, C3PM, C4PC, C4PM, C5PC, C5PM, D4PM, D5PM</code></pre>
</div>
</div>
</section>
<section id="gapdh-impact-analysis" class="level2">
<h2 class="anchored" data-anchor-id="gapdh-impact-analysis">GAPDH Impact Analysis</h2>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">=== GAPDH QC FAILURES AND THEIR CASCADE EFFECTS ===</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Get samples where GAPDH passed QC</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>gapdh_passed <span class="ot">&lt;-</span> grouped_df <span class="sc">%&gt;%</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"GAPDH"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Sample) <span class="sc">%&gt;%</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(Sample)</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Get samples where genes of interest passed QC</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>genes_of_interest <span class="ot">&lt;-</span> grouped_df <span class="sc">%&gt;%</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">!=</span> <span class="st">"GAPDH"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Sample, Target) <span class="sc">%&gt;%</span></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Find samples that have gene data but missing GAPDH</span></span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>samples_with_gene_but_no_gapdh <span class="ot">&lt;-</span> genes_of_interest <span class="sc">%&gt;%</span></span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>Sample <span class="sc">%in%</span> gapdh_passed) <span class="sc">%&gt;%</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Sample, Target)</span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(samples_with_gene_but_no_gapdh) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"**Samples with good gene measurements but failed/missing GAPDH:**</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"(These samples CANNOT be normalized and are excluded from analysis)</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Count by sample</span></span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>  sample_counts <span class="ot">&lt;-</span> samples_with_gene_but_no_gapdh <span class="sc">%&gt;%</span></span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Sample) <span class="sc">%&gt;%</span></span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">n_genes_lost =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(n_genes_lost), Sample)</span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Total samples affected:"</span>, <span class="fu">nrow</span>(sample_counts), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Total gene measurements lost:"</span>, <span class="fu">nrow</span>(samples_with_gene_but_no_gapdh), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Detailed breakdown</span></span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(sample_counts)) {</span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a>    samp <span class="ot">&lt;-</span> sample_counts<span class="sc">$</span>Sample[i]</span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a>    genes_lost <span class="ot">&lt;-</span> samples_with_gene_but_no_gapdh <span class="sc">%&gt;%</span></span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(Sample <span class="sc">==</span> samp) <span class="sc">%&gt;%</span></span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pull</span>(Target)</span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"  "</span>, samp, <span class="st">" ("</span>, sample_counts<span class="sc">$</span>n_genes_lost[i], <span class="st">" genes lost): "</span>, </span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(genes_lost, <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb30-42"><a href="#cb30-42" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb30-43"><a href="#cb30-43" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb30-44"><a href="#cb30-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"**Good news!** All samples with gene measurements also have valid GAPDH.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-45"><a href="#cb30-45" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-46"><a href="#cb30-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-47"><a href="#cb30-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if GAPDH itself failed QC for any samples</span></span>
<span id="cb30-48"><a href="#cb30-48" aria-hidden="true" tabindex="-1"></a>gapdh_failed_qc <span class="ot">&lt;-</span> excluded_samples <span class="sc">%&gt;%</span></span>
<span id="cb30-49"><a href="#cb30-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"GAPDH"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-50"><a href="#cb30-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(Sample) <span class="sc">%&gt;%</span></span>
<span id="cb30-51"><a href="#cb30-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb30-52"><a href="#cb30-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-53"><a href="#cb30-53" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">length</span>(gapdh_failed_qc) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb30-54"><a href="#cb30-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">**GAPDH samples that failed QC (Std Dev &gt; 0.5):** "</span>, <span class="fu">length</span>(gapdh_failed_qc), <span class="st">" samples</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb30-55"><a href="#cb30-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  "</span>, <span class="fu">paste</span>(<span class="fu">sort</span>(gapdh_failed_qc), <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb30-56"><a href="#cb30-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">Note: These GAPDH failures prevent normalization for ALL genes in these samples,</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-57"><a href="#cb30-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"even if those genes had good quality measurements.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-58"><a href="#cb30-58" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb30-59"><a href="#cb30-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">**No GAPDH samples failed QC.**</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-60"><a href="#cb30-60" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-61"><a href="#cb30-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-62"><a href="#cb30-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify which "missing" samples in the final analysis are due to GAPDH issues</span></span>
<span id="cb30-63"><a href="#cb30-63" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">**Analysis of 'Missing' Samples:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb30-64"><a href="#cb30-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-65"><a href="#cb30-65" aria-hidden="true" tabindex="-1"></a><span class="co"># Get samples that made it to grouped_df (passed initial QC)</span></span>
<span id="cb30-66"><a href="#cb30-66" aria-hidden="true" tabindex="-1"></a>all_passed_qc <span class="ot">&lt;-</span> grouped_df <span class="sc">%&gt;%</span></span>
<span id="cb30-67"><a href="#cb30-67" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Sample, Target) <span class="sc">%&gt;%</span></span>
<span id="cb30-68"><a href="#cb30-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb30-69"><a href="#cb30-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-70"><a href="#cb30-70" aria-hidden="true" tabindex="-1"></a><span class="co"># For each "missing" sample in delta_Cq_df, check if the gene passed QC but GAPDH didn't</span></span>
<span id="cb30-71"><a href="#cb30-71" aria-hidden="true" tabindex="-1"></a>missing_due_to_gapdh <span class="ot">&lt;-</span> missing_combinations <span class="sc">%&gt;%</span></span>
<span id="cb30-72"><a href="#cb30-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(all_passed_qc, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Sample"</span>, <span class="st">"Target"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-73"><a href="#cb30-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>Sample <span class="sc">%in%</span> gapdh_passed)</span>
<span id="cb30-74"><a href="#cb30-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-75"><a href="#cb30-75" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(missing_due_to_gapdh) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb30-76"><a href="#cb30-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Gene measurements lost due to GAPDH failure/missing: "</span>, <span class="fu">nrow</span>(missing_due_to_gapdh), <span class="st">" sample-gene combinations</span><span class="sc">\n\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb30-77"><a href="#cb30-77" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-78"><a href="#cb30-78" aria-hidden="true" tabindex="-1"></a>  missing_gapdh_summary <span class="ot">&lt;-</span> missing_due_to_gapdh <span class="sc">%&gt;%</span></span>
<span id="cb30-79"><a href="#cb30-79" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb30-80"><a href="#cb30-80" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-81"><a href="#cb30-81" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(n))</span>
<span id="cb30-82"><a href="#cb30-82" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-83"><a href="#cb30-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Breakdown by gene:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-84"><a href="#cb30-84" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(missing_gapdh_summary)) {</span>
<span id="cb30-85"><a href="#cb30-85" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"  "</span>, missing_gapdh_summary<span class="sc">$</span>Target[i], <span class="st">": "</span>, missing_gapdh_summary<span class="sc">$</span>n[i], <span class="st">" samples</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb30-86"><a href="#cb30-86" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb30-87"><a href="#cb30-87" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb30-88"><a href="#cb30-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"None of the 'missing' samples are due to GAPDH QC failures.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-89"><a href="#cb30-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"(They were never run/loaded for those genes)</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-90"><a href="#cb30-90" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
=== GAPDH QC FAILURES AND THEIR CASCADE EFFECTS ===

**Good news!** All samples with gene measurements also have valid GAPDH.


**No GAPDH samples failed QC.**


**Analysis of 'Missing' Samples:**

None of the 'missing' samples are due to GAPDH QC failures.
(They were never run/loaded for those genes)</code></pre>
</div>
</div>
</section>
</section>
<section id="statistical-comparisons" class="level1">
<h1>Statistical Comparisons</h1>
<section id="t-tests-for-treatment-comparisons" class="level2">
<h2 class="anchored" data-anchor-id="t-tests-for-treatment-comparisons">T-tests for treatment comparisons</h2>
<section id="comparison-1-basic-control-vs-stressors-no-polyic" class="level3">
<h3 class="anchored" data-anchor-id="comparison-1-basic-control-vs-stressors-no-polyic">Comparison 1: Basic Control vs Stressors (No PolyIC)</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter data for samples without PolyIC treatment</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>no_polyIC_data <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(polyIC <span class="sc">==</span> <span class="st">"No_PolyIC"</span>)</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform t-tests for each target comparing Control vs each stressor</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>comparison1_results <span class="ot">&lt;-</span> no_polyIC_data <span class="sc">%&gt;%</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(stress_type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Control"</span>, <span class="st">"Temperature"</span>, <span class="st">"Mechanical"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Control vs Temperature</span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">temp_pvalue =</span> <span class="cf">if</span>(<span class="fu">sum</span>(stress_type <span class="sc">==</span> <span class="st">"Control"</span>) <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> <span class="fu">sum</span>(stress_type <span class="sc">==</span> <span class="st">"Temperature"</span>) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>      <span class="fu">t.test</span>(delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Control"</span>], </span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>             delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Temperature"</span>])<span class="sc">$</span>p.value</span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cn">NA_real_</span>,</span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Control vs Mechanical</span></span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">mech_pvalue =</span> <span class="cf">if</span>(<span class="fu">sum</span>(stress_type <span class="sc">==</span> <span class="st">"Control"</span>) <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> <span class="fu">sum</span>(stress_type <span class="sc">==</span> <span class="st">"Mechanical"</span>) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">t.test</span>(delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Control"</span>], </span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a>             delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Mechanical"</span>])<span class="sc">$</span>p.value</span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cn">NA_real_</span>,</span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">'drop'</span></span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(comparison1_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 7 × 3
  Target           temp_pvalue mech_pvalue
  &lt;chr&gt;                  &lt;dbl&gt;       &lt;dbl&gt;
1 ATP.Synthase        1.48e- 1      0.155 
2 Citrate.Synthase    3.24e- 1      0.0113
3 DNMT1               4.36e- 1      0.672 
4 HSP70               1.17e-12      0.994 
5 HSP90               8.39e- 1      0.447 
6 VIPERIN             6.37e- 1      0.262 
7 cGAS                5.51e- 1      0.105 </code></pre>
</div>
</div>
<section id="significant-genes-p-0.05" class="level4">
<h4 class="anchored" data-anchor-id="significant-genes-p-0.05">Significant genes (p ≤ 0.05)</h4>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify genes with significant p-values for Temperature comparison</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>sig_temp <span class="ot">&lt;-</span> comparison1_results <span class="sc">%&gt;%</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(temp_pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(temp_pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Target, <span class="at">pvalue =</span> temp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">comparison =</span> <span class="st">"Control vs Temperature"</span>)</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify genes with significant p-values for Mechanical comparison</span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>sig_mech <span class="ot">&lt;-</span> comparison1_results <span class="sc">%&gt;%</span></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(mech_pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(mech_pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Target, <span class="at">pvalue =</span> mech_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">comparison =</span> <span class="st">"Control vs Mechanical"</span>)</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine and display</span></span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a>sig_genes_comp1 <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>  sig_temp,</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>  sig_mech</span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(sig_genes_comp1) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">"Genes with p ≤ 0.05:"</span>)</span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a>  sig_genes_comp1 <span class="sc">%&gt;%</span></span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">pvalue =</span> <span class="fu">format</span>(pvalue, <span class="at">scientific =</span> <span class="cn">TRUE</span>, <span class="at">digits =</span> <span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a>    knitr<span class="sc">::</span><span class="fu">kable</span>()</span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">"No genes had p-values ≤ 0.05"</span>)</span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Genes with p ≤ 0.05:"</code></pre>
</div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">Target</th>
<th style="text-align: left;">pvalue</th>
<th style="text-align: left;">comparison</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">HSP70</td>
<td style="text-align: left;">1.17e-12</td>
<td style="text-align: left;">Control vs Temperature</td>
</tr>
<tr class="even">
<td style="text-align: left;">Citrate.Synthase</td>
<td style="text-align: left;">1.13e-02</td>
<td style="text-align: left;">Control vs Mechanical</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
</section>
<section id="comparison-2-basic-control-vs-polyic-control" class="level3">
<h3 class="anchored" data-anchor-id="comparison-2-basic-control-vs-polyic-control">Comparison 2: Basic Control vs PolyIC Control</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter data for control samples only (with and without PolyIC)</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>control_data <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(stress_type <span class="sc">==</span> <span class="st">"Control"</span>)</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform t-tests for each target comparing No_PolyIC vs PolyIC controls</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>comparison2_results <span class="ot">&lt;-</span> control_data <span class="sc">%&gt;%</span></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">n_no_polyIC =</span> <span class="fu">sum</span>(polyIC <span class="sc">==</span> <span class="st">"No_PolyIC"</span>),</span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">n_polyIC =</span> <span class="fu">sum</span>(polyIC <span class="sc">==</span> <span class="st">"PolyIC"</span>),</span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">pvalue =</span> <span class="cf">if</span>(<span class="fu">sum</span>(polyIC <span class="sc">==</span> <span class="st">"No_PolyIC"</span>) <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> <span class="fu">sum</span>(polyIC <span class="sc">==</span> <span class="st">"PolyIC"</span>) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>      <span class="fu">t.test</span>(delta_Cq[polyIC <span class="sc">==</span> <span class="st">"No_PolyIC"</span>], </span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>             delta_Cq[polyIC <span class="sc">==</span> <span class="st">"PolyIC"</span>])<span class="sc">$</span>p.value</span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cn">NA_real_</span>,</span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_no_polyIC =</span> <span class="fu">mean</span>(delta_Cq[polyIC <span class="sc">==</span> <span class="st">"No_PolyIC"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">sd_no_polyIC =</span> <span class="fu">sd</span>(delta_Cq[polyIC <span class="sc">==</span> <span class="st">"No_PolyIC"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_polyIC =</span> <span class="fu">mean</span>(delta_Cq[polyIC <span class="sc">==</span> <span class="st">"PolyIC"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">sd_polyIC =</span> <span class="fu">sd</span>(delta_Cq[polyIC <span class="sc">==</span> <span class="st">"PolyIC"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_diff =</span> mean_polyIC <span class="sc">-</span> mean_no_polyIC,</span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">'drop'</span></span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb36-25"><a href="#cb36-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-26"><a href="#cb36-26" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(comparison2_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 7 × 9
  Target    n_no_polyIC n_polyIC  pvalue mean_no_polyIC sd_no_polyIC mean_polyIC
  &lt;chr&gt;           &lt;int&gt;    &lt;int&gt;   &lt;dbl&gt;          &lt;dbl&gt;        &lt;dbl&gt;       &lt;dbl&gt;
1 ATP.Synt…          20       17 1.11e-3         -0.118        0.270       2.14 
2 Citrate.…          20       18 2.88e-4          0.390        0.308       2.41 
3 DNMT1              20        8 7.56e-1          6.26         0.904       6.42 
4 HSP70              20       18 4.73e-4          5.51         1.52        3.29 
5 HSP90              20        8 7.96e-3          0.457        0.405      -0.113
6 VIPERIN            20        8 1.01e-2          5.19         1.04        4.01 
7 cGAS               20        8 7.18e-1          6.39         0.873       6.22 
# ℹ 2 more variables: sd_polyIC &lt;dbl&gt;, mean_diff &lt;dbl&gt;</code></pre>
</div>
</div>
<section id="significant-genes-p-0.05-1" class="level4">
<h4 class="anchored" data-anchor-id="significant-genes-p-0.05-1">Significant genes (p ≤ 0.05)</h4>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify genes with significant p-values</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>sig_genes_comp2 <span class="ot">&lt;-</span> comparison2_results <span class="sc">%&gt;%</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Target, pvalue)</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(sig_genes_comp2) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">"Genes with p ≤ 0.05:"</span>)</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>  sig_genes_comp2 <span class="sc">%&gt;%</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">pvalue =</span> <span class="fu">format</span>(pvalue, <span class="at">scientific =</span> <span class="cn">TRUE</span>, <span class="at">digits =</span> <span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>    knitr<span class="sc">::</span><span class="fu">kable</span>()</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">"No genes had p-values ≤ 0.05"</span>)</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Genes with p ≤ 0.05:"</code></pre>
</div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">Target</th>
<th style="text-align: left;">pvalue</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ATP.Synthase</td>
<td style="text-align: left;">1.11e-03</td>
</tr>
<tr class="even">
<td style="text-align: left;">Citrate.Synthase</td>
<td style="text-align: left;">2.88e-04</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSP70</td>
<td style="text-align: left;">4.73e-04</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSP90</td>
<td style="text-align: left;">7.96e-03</td>
</tr>
<tr class="odd">
<td style="text-align: left;">VIPERIN</td>
<td style="text-align: left;">1.01e-02</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
</section>
<section id="comparison-3-polyic-control-vs-polyic-stressors" class="level3">
<h3 class="anchored" data-anchor-id="comparison-3-polyic-control-vs-polyic-stressors">Comparison 3: PolyIC Control vs PolyIC Stressors</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter data for PolyIC samples only</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>polyIC_data <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(polyIC <span class="sc">==</span> <span class="st">"PolyIC"</span>)</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform t-tests for each target comparing PolyIC Control vs each stressor</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>comparison3_results <span class="ot">&lt;-</span> polyIC_data <span class="sc">%&gt;%</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(stress_type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Control"</span>, <span class="st">"Temperature"</span>, <span class="st">"Mechanical"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># PolyIC Control vs Temperature</span></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">temp_pvalue =</span> <span class="cf">if</span>(<span class="fu">sum</span>(stress_type <span class="sc">==</span> <span class="st">"Control"</span>) <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> <span class="fu">sum</span>(stress_type <span class="sc">==</span> <span class="st">"Temperature"</span>) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>      <span class="fu">t.test</span>(delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Control"</span>], </span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>             delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Temperature"</span>])<span class="sc">$</span>p.value</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cn">NA_real_</span>,</span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># PolyIC Control vs Mechanical</span></span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">mech_pvalue =</span> <span class="cf">if</span>(<span class="fu">sum</span>(stress_type <span class="sc">==</span> <span class="st">"Control"</span>) <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> <span class="fu">sum</span>(stress_type <span class="sc">==</span> <span class="st">"Mechanical"</span>) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">t.test</span>(delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Control"</span>], </span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>             delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Mechanical"</span>])<span class="sc">$</span>p.value</span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cn">NA_real_</span>,</span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">'drop'</span></span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(comparison3_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 7 × 3
  Target            temp_pvalue mech_pvalue
  &lt;chr&gt;                   &lt;dbl&gt;       &lt;dbl&gt;
1 ATP.Synthase     0.00693           0.0766
2 Citrate.Synthase 0.000519          0.715 
3 DNMT1            0.124             0.845 
4 HSP70            0.0000000461      0.684 
5 HSP90            0.151             0.533 
6 VIPERIN          0.332             0.828 
7 cGAS             0.953             0.249 </code></pre>
</div>
</div>
<section id="significant-genes-p-0.05-2" class="level4">
<h4 class="anchored" data-anchor-id="significant-genes-p-0.05-2">Significant genes (p ≤ 0.05)</h4>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify genes with significant p-values for Temperature comparison</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>sig_temp_polyIC <span class="ot">&lt;-</span> comparison3_results <span class="sc">%&gt;%</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(temp_pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(temp_pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Target, <span class="at">pvalue =</span> temp_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">comparison =</span> <span class="st">"PolyIC Control vs Temperature"</span>)</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify genes with significant p-values for Mechanical comparison</span></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>sig_mech_polyIC <span class="ot">&lt;-</span> comparison3_results <span class="sc">%&gt;%</span></span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(mech_pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(mech_pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Target, <span class="at">pvalue =</span> mech_pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">comparison =</span> <span class="st">"PolyIC Control vs Mechanical"</span>)</span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine and display</span></span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>sig_genes_comp3 <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>  sig_temp_polyIC,</span>
<span id="cb42-16"><a href="#cb42-16" aria-hidden="true" tabindex="-1"></a>  sig_mech_polyIC</span>
<span id="cb42-17"><a href="#cb42-17" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb42-18"><a href="#cb42-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-19"><a href="#cb42-19" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(sig_genes_comp3) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb42-20"><a href="#cb42-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">"Genes with p ≤ 0.05:"</span>)</span>
<span id="cb42-21"><a href="#cb42-21" aria-hidden="true" tabindex="-1"></a>  sig_genes_comp3 <span class="sc">%&gt;%</span></span>
<span id="cb42-22"><a href="#cb42-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">pvalue =</span> <span class="fu">format</span>(pvalue, <span class="at">scientific =</span> <span class="cn">TRUE</span>, <span class="at">digits =</span> <span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb42-23"><a href="#cb42-23" aria-hidden="true" tabindex="-1"></a>    knitr<span class="sc">::</span><span class="fu">kable</span>()</span>
<span id="cb42-24"><a href="#cb42-24" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb42-25"><a href="#cb42-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">"No genes had p-values ≤ 0.05"</span>)</span>
<span id="cb42-26"><a href="#cb42-26" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Genes with p ≤ 0.05:"</code></pre>
</div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">Target</th>
<th style="text-align: left;">pvalue</th>
<th style="text-align: left;">comparison</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ATP.Synthase</td>
<td style="text-align: left;">6.93e-03</td>
<td style="text-align: left;">PolyIC Control vs Temperature</td>
</tr>
<tr class="even">
<td style="text-align: left;">Citrate.Synthase</td>
<td style="text-align: left;">5.19e-04</td>
<td style="text-align: left;">PolyIC Control vs Temperature</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSP70</td>
<td style="text-align: left;">4.61e-08</td>
<td style="text-align: left;">PolyIC Control vs Temperature</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
</section>
</section>
<section id="create-anova-models" class="level2">
<h2 class="anchored" data-anchor-id="create-anova-models">Create ANOVA Models</h2>
<p>Two-way ANOVA (PolyIC × Stress Type) for each gene to comprehensively test treatment effects and interactions.</p>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run ANOVA models for all genes and store emmeans objects for plotting</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Detailed assumption checks and interpretations are in the "ANOVA Results and Diagnostics" section below</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a><span class="co"># ATP Synthase</span></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>model_atp <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"ATP.Synthase"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_atp)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>  emm_atp <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_atp, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Citrate Synthase</span></span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>model_citrate <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"Citrate.Synthase"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_citrate)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a>  emm_citrate <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_citrate, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a><span class="co"># DNMT1</span></span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a>model_dnmt1 <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"DNMT1"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb44-24"><a href="#cb44-24" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_dnmt1)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb44-25"><a href="#cb44-25" aria-hidden="true" tabindex="-1"></a>  emm_dnmt1 <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_dnmt1, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb44-26"><a href="#cb44-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb44-27"><a href="#cb44-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-28"><a href="#cb44-28" aria-hidden="true" tabindex="-1"></a><span class="co"># HSP70</span></span>
<span id="cb44-29"><a href="#cb44-29" aria-hidden="true" tabindex="-1"></a>model_hsp70 <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb44-30"><a href="#cb44-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"HSP70"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-31"><a href="#cb44-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb44-32"><a href="#cb44-32" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_hsp70)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb44-33"><a href="#cb44-33" aria-hidden="true" tabindex="-1"></a>  emm_hsp70 <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_hsp70, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb44-34"><a href="#cb44-34" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb44-35"><a href="#cb44-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-36"><a href="#cb44-36" aria-hidden="true" tabindex="-1"></a><span class="co"># HSP90</span></span>
<span id="cb44-37"><a href="#cb44-37" aria-hidden="true" tabindex="-1"></a>model_hsp90 <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb44-38"><a href="#cb44-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"HSP90"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-39"><a href="#cb44-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb44-40"><a href="#cb44-40" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_hsp90)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb44-41"><a href="#cb44-41" aria-hidden="true" tabindex="-1"></a>  emm_hsp90 <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_hsp90, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb44-42"><a href="#cb44-42" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb44-43"><a href="#cb44-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-44"><a href="#cb44-44" aria-hidden="true" tabindex="-1"></a><span class="co"># cGAS</span></span>
<span id="cb44-45"><a href="#cb44-45" aria-hidden="true" tabindex="-1"></a>model_cgas <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb44-46"><a href="#cb44-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"cGAS"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-47"><a href="#cb44-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb44-48"><a href="#cb44-48" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_cgas)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb44-49"><a href="#cb44-49" aria-hidden="true" tabindex="-1"></a>  emm_cgas <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_cgas, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb44-50"><a href="#cb44-50" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb44-51"><a href="#cb44-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-52"><a href="#cb44-52" aria-hidden="true" tabindex="-1"></a><span class="co"># VIPERIN</span></span>
<span id="cb44-53"><a href="#cb44-53" aria-hidden="true" tabindex="-1"></a>model_viperin <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb44-54"><a href="#cb44-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"VIPERIN"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-55"><a href="#cb44-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb44-56"><a href="#cb44-56" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_viperin)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb44-57"><a href="#cb44-57" aria-hidden="true" tabindex="-1"></a>  emm_viperin <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_viperin, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb44-58"><a href="#cb44-58" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb44-59"><a href="#cb44-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-60"><a href="#cb44-60" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"ANOVA models created for all genes. Emmeans objects stored for post-hoc testing and plotting.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-61"><a href="#cb44-61" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"See 'ANOVA Results and Diagnostics' section below for detailed results, assumption checks, and interpretations.</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>ANOVA models created for all genes. Emmeans objects stored for post-hoc testing and plotting.
See 'ANOVA Results and Diagnostics' section below for detailed results, assumption checks, and interpretations.</code></pre>
</div>
</div>
</section>
</section>
<section id="visualizations" class="level1">
<h1>Visualizations</h1>
<section id="delta-cq-boxplots" class="level2">
<h2 class="anchored" data-anchor-id="delta-cq-boxplots">Delta Cq Boxplots</h2>
<section id="comparison-1-control-vs-stressors-no-polyic" class="level3">
<h3 class="anchored" data-anchor-id="comparison-1-control-vs-stressors-no-polyic">Comparison 1: Control vs Stressors (No PolyIC)</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare significance annotations from t-test results</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">exists</span>(<span class="st">"sig_genes_comp1"</span>) <span class="sc">&amp;&amp;</span> <span class="fu">nrow</span>(sig_genes_comp1) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract significant genes for Temperature</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>  sig_temp_genes <span class="ot">&lt;-</span> sig_genes_comp1 <span class="sc">%&gt;%</span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(comparison <span class="sc">==</span> <span class="st">"Control vs Temperature"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(Target)</span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract significant genes for Mechanical</span></span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>  sig_mech_genes <span class="ot">&lt;-</span> sig_genes_comp1 <span class="sc">%&gt;%</span></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(comparison <span class="sc">==</span> <span class="st">"Control vs Mechanical"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(Target)</span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create annotation data</span></span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>  sig_annotations_comp1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">Target =</span> <span class="fu">character</span>(),</span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">stress_type =</span> <span class="fu">character</span>(),</span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">character</span>(),</span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add Temperature annotations</span></span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(sig_temp_genes) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a>    sig_annotations_comp1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sig_annotations_comp1,</span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a>      <span class="fu">data.frame</span>(<span class="at">Target =</span> sig_temp_genes,</span>
<span id="cb46-25"><a href="#cb46-25" aria-hidden="true" tabindex="-1"></a>                 <span class="at">stress_type =</span> <span class="st">"Temperature"</span>,</span>
<span id="cb46-26"><a href="#cb46-26" aria-hidden="true" tabindex="-1"></a>                 <span class="at">label =</span> <span class="st">"*"</span>))</span>
<span id="cb46-27"><a href="#cb46-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb46-28"><a href="#cb46-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-29"><a href="#cb46-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add Mechanical annotations</span></span>
<span id="cb46-30"><a href="#cb46-30" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(sig_mech_genes) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb46-31"><a href="#cb46-31" aria-hidden="true" tabindex="-1"></a>    sig_annotations_comp1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sig_annotations_comp1,</span>
<span id="cb46-32"><a href="#cb46-32" aria-hidden="true" tabindex="-1"></a>      <span class="fu">data.frame</span>(<span class="at">Target =</span> sig_mech_genes,</span>
<span id="cb46-33"><a href="#cb46-33" aria-hidden="true" tabindex="-1"></a>                 <span class="at">stress_type =</span> <span class="st">"Mechanical"</span>,</span>
<span id="cb46-34"><a href="#cb46-34" aria-hidden="true" tabindex="-1"></a>                 <span class="at">label =</span> <span class="st">"*"</span>))</span>
<span id="cb46-35"><a href="#cb46-35" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb46-36"><a href="#cb46-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-37"><a href="#cb46-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get y positions for annotations</span></span>
<span id="cb46-38"><a href="#cb46-38" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(sig_annotations_comp1) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb46-39"><a href="#cb46-39" aria-hidden="true" tabindex="-1"></a>    y_pos_comp1 <span class="ot">&lt;-</span> no_polyIC_data <span class="sc">%&gt;%</span></span>
<span id="cb46-40"><a href="#cb46-40" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(Target, stress_type) <span class="sc">%&gt;%</span></span>
<span id="cb46-41"><a href="#cb46-41" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">max_y =</span> <span class="fu">max</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb46-42"><a href="#cb46-42" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb46-43"><a href="#cb46-43" aria-hidden="true" tabindex="-1"></a>    sig_annotations_comp1 <span class="ot">&lt;-</span> sig_annotations_comp1 <span class="sc">%&gt;%</span></span>
<span id="cb46-44"><a href="#cb46-44" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(y_pos_comp1, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"stress_type"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb46-45"><a href="#cb46-45" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">y_pos =</span> max_y <span class="sc">+</span> <span class="fl">0.5</span>)</span>
<span id="cb46-46"><a href="#cb46-46" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb46-47"><a href="#cb46-47" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb46-48"><a href="#cb46-48" aria-hidden="true" tabindex="-1"></a>  sig_annotations_comp1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb46-49"><a href="#cb46-49" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb46-50"><a href="#cb46-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-51"><a href="#cb46-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Create plot</span></span>
<span id="cb46-52"><a href="#cb46-52" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> no_polyIC_data <span class="sc">%&gt;%</span></span>
<span id="cb46-53"><a href="#cb46-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> stress_type, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> stress_type)) <span class="sc">+</span></span>
<span id="cb46-54"><a href="#cb46-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb46-55"><a href="#cb46-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Target, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb46-56"><a href="#cb46-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Control"</span> <span class="ot">=</span> <span class="st">"darkgray"</span>, <span class="st">"Temperature"</span> <span class="ot">=</span> <span class="st">"salmon"</span>, <span class="st">"Mechanical"</span> <span class="ot">=</span> <span class="st">"steelblue"</span>)) <span class="sc">+</span></span>
<span id="cb46-57"><a href="#cb46-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">clip =</span> <span class="st">"off"</span>) <span class="sc">+</span></span>
<span id="cb46-58"><a href="#cb46-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb46-59"><a href="#cb46-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb46-60"><a href="#cb46-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Delta Cq: Control vs Stressors (No PolyIC Treatment)"</span>,</span>
<span id="cb46-61"><a href="#cb46-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Stress Type"</span>,</span>
<span id="cb46-62"><a href="#cb46-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Delta Cq"</span>,</span>
<span id="cb46-63"><a href="#cb46-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Stress Type"</span></span>
<span id="cb46-64"><a href="#cb46-64" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb46-65"><a href="#cb46-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb46-66"><a href="#cb46-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-67"><a href="#cb46-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Add annotations if any exist</span></span>
<span id="cb46-68"><a href="#cb46-68" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(sig_annotations_comp1) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb46-69"><a href="#cb46-69" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> p <span class="sc">+</span> </span>
<span id="cb46-70"><a href="#cb46-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> sig_annotations_comp1, </span>
<span id="cb46-71"><a href="#cb46-71" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> stress_type, <span class="at">y =</span> y_pos, <span class="at">label =</span> label),</span>
<span id="cb46-72"><a href="#cb46-72" aria-hidden="true" tabindex="-1"></a>              <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>, <span class="at">size =</span> <span class="dv">6</span>, <span class="at">vjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"darkorange"</span>) <span class="sc">+</span></span>
<span id="cb46-73"><a href="#cb46-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">"Asterisks (*) indicate significant differences from Control (t-test p ≤ 0.05)"</span>,</span>
<span id="cb46-74"><a href="#cb46-74" aria-hidden="true" tabindex="-1"></a>         <span class="at">caption =</span> <span class="fu">paste</span>(<span class="st">"Significant comparisons:"</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(sig_genes_comp1<span class="sc">$</span>Target), <span class="at">collapse =</span> <span class="st">", "</span>)))</span>
<span id="cb46-75"><a href="#cb46-75" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb46-76"><a href="#cb46-76" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">"No significant differences detected in t-tests (all p &gt; 0.05)"</span>)</span>
<span id="cb46-77"><a href="#cb46-77" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb46-78"><a href="#cb46-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-79"><a href="#cb46-79" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/boxplot-no-polyIC-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="comparison-2-control-vs-polyic-control" class="level3">
<h3 class="anchored" data-anchor-id="comparison-2-control-vs-polyic-control">Comparison 2: Control vs PolyIC Control</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare significance annotations from t-test results</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">exists</span>(<span class="st">"sig_genes_comp2"</span>) <span class="sc">&amp;&amp;</span> <span class="fu">nrow</span>(sig_genes_comp2) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  sig_annotations_comp2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">Target =</span> sig_genes_comp2<span class="sc">$</span>Target,</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">polyIC =</span> <span class="st">"PolyIC"</span>,</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"*"</span>,</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get y positions for annotations</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>  y_pos_comp2 <span class="ot">&lt;-</span> control_data <span class="sc">%&gt;%</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Target, polyIC) <span class="sc">%&gt;%</span></span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">max_y =</span> <span class="fu">max</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>  sig_annotations_comp2 <span class="ot">&lt;-</span> sig_annotations_comp2 <span class="sc">%&gt;%</span></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(y_pos_comp2, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"polyIC"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">y_pos =</span> max_y <span class="sc">+</span> <span class="fl">0.5</span>)</span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>  sig_annotations_comp2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Create plot</span></span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> control_data <span class="sc">%&gt;%</span></span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> polyIC, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> polyIC)) <span class="sc">+</span></span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Target, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"No_PolyIC"</span> <span class="ot">=</span> <span class="st">"darkgray"</span>, <span class="st">"PolyIC"</span> <span class="ot">=</span> <span class="st">"salmon"</span>)) <span class="sc">+</span></span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">clip =</span> <span class="st">"off"</span>) <span class="sc">+</span></span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Delta Cq: Basic Control vs PolyIC Control"</span>,</span>
<span id="cb47-32"><a href="#cb47-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PolyIC Treatment"</span>,</span>
<span id="cb47-33"><a href="#cb47-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Delta Cq"</span>,</span>
<span id="cb47-34"><a href="#cb47-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Treatment"</span></span>
<span id="cb47-35"><a href="#cb47-35" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb47-36"><a href="#cb47-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb47-37"><a href="#cb47-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb47-38"><a href="#cb47-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">20</span>, <span class="at">r =</span> <span class="dv">10</span>, <span class="at">b =</span> <span class="dv">10</span>, <span class="at">l =</span> <span class="dv">10</span>)</span>
<span id="cb47-39"><a href="#cb47-39" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb47-40"><a href="#cb47-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-41"><a href="#cb47-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Add annotations if any exist</span></span>
<span id="cb47-42"><a href="#cb47-42" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(sig_annotations_comp2) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb47-43"><a href="#cb47-43" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Expand y-axis limits to ensure annotations are visible</span></span>
<span id="cb47-44"><a href="#cb47-44" aria-hidden="true" tabindex="-1"></a>  y_limits_comp2 <span class="ot">&lt;-</span> control_data <span class="sc">%&gt;%</span></span>
<span id="cb47-45"><a href="#cb47-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb47-46"><a href="#cb47-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(</span>
<span id="cb47-47"><a href="#cb47-47" aria-hidden="true" tabindex="-1"></a>      <span class="at">y_min =</span> <span class="fu">min</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">-</span> <span class="fl">0.5</span>,</span>
<span id="cb47-48"><a href="#cb47-48" aria-hidden="true" tabindex="-1"></a>      <span class="at">y_max =</span> <span class="fu">max</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="fl">1.5</span>,</span>
<span id="cb47-49"><a href="#cb47-49" aria-hidden="true" tabindex="-1"></a>      <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb47-50"><a href="#cb47-50" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb47-51"><a href="#cb47-51" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-52"><a href="#cb47-52" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> p <span class="sc">+</span> </span>
<span id="cb47-53"><a href="#cb47-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_blank</span>(<span class="at">data =</span> y_limits_comp2, <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span>, <span class="at">y =</span> y_max), <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb47-54"><a href="#cb47-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> sig_annotations_comp2, </span>
<span id="cb47-55"><a href="#cb47-55" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> polyIC, <span class="at">y =</span> y_pos, <span class="at">label =</span> label),</span>
<span id="cb47-56"><a href="#cb47-56" aria-hidden="true" tabindex="-1"></a>              <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>, <span class="at">size =</span> <span class="dv">6</span>, <span class="at">vjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"darkorange"</span>) <span class="sc">+</span></span>
<span id="cb47-57"><a href="#cb47-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">"Asterisks (*) indicate significant differences (t-test p ≤ 0.05)"</span>,</span>
<span id="cb47-58"><a href="#cb47-58" aria-hidden="true" tabindex="-1"></a>         <span class="at">caption =</span> <span class="fu">paste</span>(<span class="st">"Genes with significant PolyIC effect:"</span>, <span class="fu">paste</span>(sig_genes_comp2<span class="sc">$</span>Target, <span class="at">collapse =</span> <span class="st">", "</span>)))</span>
<span id="cb47-59"><a href="#cb47-59" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb47-60"><a href="#cb47-60" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">"No significant differences detected in t-tests (all p &gt; 0.05)"</span>)</span>
<span id="cb47-61"><a href="#cb47-61" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-62"><a href="#cb47-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-63"><a href="#cb47-63" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/boxplot-control-comparison-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="comparison-3-polyic-control-vs-polyic-stressors-1" class="level3">
<h3 class="anchored" data-anchor-id="comparison-3-polyic-control-vs-polyic-stressors-1">Comparison 3: PolyIC Control vs PolyIC Stressors</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare significance annotations from t-test results</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">exists</span>(<span class="st">"sig_genes_comp3"</span>) <span class="sc">&amp;&amp;</span> <span class="fu">nrow</span>(sig_genes_comp3) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract significant genes for Temperature</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>  sig_temp_genes <span class="ot">&lt;-</span> sig_genes_comp3 <span class="sc">%&gt;%</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(comparison <span class="sc">==</span> <span class="st">"PolyIC Control vs Temperature"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(Target)</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract significant genes for Mechanical</span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>  sig_mech_genes <span class="ot">&lt;-</span> sig_genes_comp3 <span class="sc">%&gt;%</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(comparison <span class="sc">==</span> <span class="st">"PolyIC Control vs Mechanical"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(Target)</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create annotation data</span></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>  sig_annotations_comp3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">Target =</span> <span class="fu">character</span>(),</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">stress_type =</span> <span class="fu">character</span>(),</span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">character</span>(),</span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add Temperature annotations</span></span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(sig_temp_genes) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a>    sig_annotations_comp3 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sig_annotations_comp3,</span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a>      <span class="fu">data.frame</span>(<span class="at">Target =</span> sig_temp_genes,</span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a>                 <span class="at">stress_type =</span> <span class="st">"Temperature"</span>,</span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a>                 <span class="at">label =</span> <span class="st">"*"</span>))</span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add Mechanical annotations</span></span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(sig_mech_genes) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb48-31"><a href="#cb48-31" aria-hidden="true" tabindex="-1"></a>    sig_annotations_comp3 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sig_annotations_comp3,</span>
<span id="cb48-32"><a href="#cb48-32" aria-hidden="true" tabindex="-1"></a>      <span class="fu">data.frame</span>(<span class="at">Target =</span> sig_mech_genes,</span>
<span id="cb48-33"><a href="#cb48-33" aria-hidden="true" tabindex="-1"></a>                 <span class="at">stress_type =</span> <span class="st">"Mechanical"</span>,</span>
<span id="cb48-34"><a href="#cb48-34" aria-hidden="true" tabindex="-1"></a>                 <span class="at">label =</span> <span class="st">"*"</span>))</span>
<span id="cb48-35"><a href="#cb48-35" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb48-36"><a href="#cb48-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb48-37"><a href="#cb48-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get y positions for annotations</span></span>
<span id="cb48-38"><a href="#cb48-38" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(sig_annotations_comp3) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb48-39"><a href="#cb48-39" aria-hidden="true" tabindex="-1"></a>    y_pos_comp3 <span class="ot">&lt;-</span> polyIC_data <span class="sc">%&gt;%</span></span>
<span id="cb48-40"><a href="#cb48-40" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(Target, stress_type) <span class="sc">%&gt;%</span></span>
<span id="cb48-41"><a href="#cb48-41" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">max_y =</span> <span class="fu">max</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb48-42"><a href="#cb48-42" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb48-43"><a href="#cb48-43" aria-hidden="true" tabindex="-1"></a>    sig_annotations_comp3 <span class="ot">&lt;-</span> sig_annotations_comp3 <span class="sc">%&gt;%</span></span>
<span id="cb48-44"><a href="#cb48-44" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(y_pos_comp3, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"stress_type"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb48-45"><a href="#cb48-45" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">y_pos =</span> max_y <span class="sc">+</span> <span class="fl">0.5</span>)</span>
<span id="cb48-46"><a href="#cb48-46" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb48-47"><a href="#cb48-47" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb48-48"><a href="#cb48-48" aria-hidden="true" tabindex="-1"></a>  sig_annotations_comp3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb48-49"><a href="#cb48-49" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb48-50"><a href="#cb48-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-51"><a href="#cb48-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Create plot</span></span>
<span id="cb48-52"><a href="#cb48-52" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> polyIC_data <span class="sc">%&gt;%</span></span>
<span id="cb48-53"><a href="#cb48-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> stress_type, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> stress_type)) <span class="sc">+</span></span>
<span id="cb48-54"><a href="#cb48-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb48-55"><a href="#cb48-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Target, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb48-56"><a href="#cb48-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Control"</span> <span class="ot">=</span> <span class="st">"darkgray"</span>, <span class="st">"Temperature"</span> <span class="ot">=</span> <span class="st">"salmon"</span>, <span class="st">"Mechanical"</span> <span class="ot">=</span> <span class="st">"steelblue"</span>)) <span class="sc">+</span></span>
<span id="cb48-57"><a href="#cb48-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb48-58"><a href="#cb48-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb48-59"><a href="#cb48-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Delta Cq: PolyIC Control vs PolyIC Stressors"</span>,</span>
<span id="cb48-60"><a href="#cb48-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Stress Type"</span>,</span>
<span id="cb48-61"><a href="#cb48-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Delta Cq"</span>,</span>
<span id="cb48-62"><a href="#cb48-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Stress Type"</span></span>
<span id="cb48-63"><a href="#cb48-63" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb48-64"><a href="#cb48-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb48-65"><a href="#cb48-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-66"><a href="#cb48-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Add annotations if any exist</span></span>
<span id="cb48-67"><a href="#cb48-67" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(sig_annotations_comp3) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb48-68"><a href="#cb48-68" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> p <span class="sc">+</span> </span>
<span id="cb48-69"><a href="#cb48-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> sig_annotations_comp3, </span>
<span id="cb48-70"><a href="#cb48-70" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> stress_type, <span class="at">y =</span> y_pos, <span class="at">label =</span> label),</span>
<span id="cb48-71"><a href="#cb48-71" aria-hidden="true" tabindex="-1"></a>              <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>, <span class="at">size =</span> <span class="dv">6</span>, <span class="at">vjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"darkorange"</span>) <span class="sc">+</span></span>
<span id="cb48-72"><a href="#cb48-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">"Asterisks (*) indicate significant differences from PolyIC Control (t-test p ≤ 0.05)"</span>,</span>
<span id="cb48-73"><a href="#cb48-73" aria-hidden="true" tabindex="-1"></a>         <span class="at">caption =</span> <span class="fu">paste</span>(<span class="st">"Significant comparisons:"</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(sig_genes_comp3<span class="sc">$</span>Target), <span class="at">collapse =</span> <span class="st">", "</span>)))</span>
<span id="cb48-74"><a href="#cb48-74" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb48-75"><a href="#cb48-75" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">"No significant differences detected in t-tests (all p &gt; 0.05)"</span>)</span>
<span id="cb48-76"><a href="#cb48-76" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb48-77"><a href="#cb48-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-78"><a href="#cb48-78" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/boxplot-polyIC-stress-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="all-treatments-comparison" class="level2">
<h2 class="anchored" data-anchor-id="all-treatments-comparison">All treatments comparison</h2>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use emmeans objects created in Statistical Comparisons section</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>all_treat_annotations <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a><span class="co"># ATP Synthase</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">exists</span>(<span class="st">"emm_atp"</span>)) {</span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>  cld_temp <span class="ot">&lt;-</span> <span class="fu">cld</span>(emm_atp, <span class="at">Letters =</span> letters, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>  cld_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(cld_temp)</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>  cld_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="st">"ATP.Synthase"</span></span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>  cld_df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">trimws</span>(cld_df<span class="sc">$</span>.group)</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>  cld_df<span class="sc">$</span>treatment_group <span class="ot">&lt;-</span> <span class="fu">paste</span>(cld_df<span class="sc">$</span>polyIC, cld_df<span class="sc">$</span>stress_type, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>  all_treat_annotations <span class="ot">&lt;-</span> <span class="fu">rbind</span>(all_treat_annotations, </span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>                                 cld_df[, <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"treatment_group"</span>, <span class="st">"label"</span>)])</span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a><span class="co"># HSP70</span></span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">exists</span>(<span class="st">"emm_hsp70"</span>)) {</span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>  cld_temp <span class="ot">&lt;-</span> <span class="fu">cld</span>(emm_hsp70, <span class="at">Letters =</span> letters, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a>  cld_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(cld_temp)</span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a>  cld_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="st">"HSP70"</span></span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a>  cld_df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">trimws</span>(cld_df<span class="sc">$</span>.group)</span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a>  cld_df<span class="sc">$</span>treatment_group <span class="ot">&lt;-</span> <span class="fu">paste</span>(cld_df<span class="sc">$</span>polyIC, cld_df<span class="sc">$</span>stress_type, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a>  all_treat_annotations <span class="ot">&lt;-</span> <span class="fu">rbind</span>(all_treat_annotations, </span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a>                                 cld_df[, <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"treatment_group"</span>, <span class="st">"label"</span>)])</span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a><span class="co"># HSP90</span></span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">exists</span>(<span class="st">"emm_hsp90"</span>)) {</span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a>  cld_temp <span class="ot">&lt;-</span> <span class="fu">cld</span>(emm_hsp90, <span class="at">Letters =</span> letters, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a>  cld_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(cld_temp)</span>
<span id="cb49-30"><a href="#cb49-30" aria-hidden="true" tabindex="-1"></a>  cld_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="st">"HSP90"</span></span>
<span id="cb49-31"><a href="#cb49-31" aria-hidden="true" tabindex="-1"></a>  cld_df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">trimws</span>(cld_df<span class="sc">$</span>.group)</span>
<span id="cb49-32"><a href="#cb49-32" aria-hidden="true" tabindex="-1"></a>  cld_df<span class="sc">$</span>treatment_group <span class="ot">&lt;-</span> <span class="fu">paste</span>(cld_df<span class="sc">$</span>polyIC, cld_df<span class="sc">$</span>stress_type, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb49-33"><a href="#cb49-33" aria-hidden="true" tabindex="-1"></a>  all_treat_annotations <span class="ot">&lt;-</span> <span class="fu">rbind</span>(all_treat_annotations, </span>
<span id="cb49-34"><a href="#cb49-34" aria-hidden="true" tabindex="-1"></a>                                 cld_df[, <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"treatment_group"</span>, <span class="st">"label"</span>)])</span>
<span id="cb49-35"><a href="#cb49-35" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-36"><a href="#cb49-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-37"><a href="#cb49-37" aria-hidden="true" tabindex="-1"></a><span class="co"># VIPERIN</span></span>
<span id="cb49-38"><a href="#cb49-38" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">exists</span>(<span class="st">"emm_viperin"</span>)) {</span>
<span id="cb49-39"><a href="#cb49-39" aria-hidden="true" tabindex="-1"></a>  cld_temp <span class="ot">&lt;-</span> <span class="fu">cld</span>(emm_viperin, <span class="at">Letters =</span> letters, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb49-40"><a href="#cb49-40" aria-hidden="true" tabindex="-1"></a>  cld_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(cld_temp)</span>
<span id="cb49-41"><a href="#cb49-41" aria-hidden="true" tabindex="-1"></a>  cld_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="st">"VIPERIN"</span></span>
<span id="cb49-42"><a href="#cb49-42" aria-hidden="true" tabindex="-1"></a>  cld_df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">trimws</span>(cld_df<span class="sc">$</span>.group)</span>
<span id="cb49-43"><a href="#cb49-43" aria-hidden="true" tabindex="-1"></a>  cld_df<span class="sc">$</span>treatment_group <span class="ot">&lt;-</span> <span class="fu">paste</span>(cld_df<span class="sc">$</span>polyIC, cld_df<span class="sc">$</span>stress_type, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb49-44"><a href="#cb49-44" aria-hidden="true" tabindex="-1"></a>  all_treat_annotations <span class="ot">&lt;-</span> <span class="fu">rbind</span>(all_treat_annotations, </span>
<span id="cb49-45"><a href="#cb49-45" aria-hidden="true" tabindex="-1"></a>                                 cld_df[, <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"treatment_group"</span>, <span class="st">"label"</span>)])</span>
<span id="cb49-46"><a href="#cb49-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-47"><a href="#cb49-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-48"><a href="#cb49-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate y positions if annotations exist</span></span>
<span id="cb49-49"><a href="#cb49-49" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(all_treat_annotations) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb49-50"><a href="#cb49-50" aria-hidden="true" tabindex="-1"></a>  y_pos_all <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb49-51"><a href="#cb49-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Target, treatment_group) <span class="sc">%&gt;%</span></span>
<span id="cb49-52"><a href="#cb49-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">max_y =</span> <span class="fu">max</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb49-53"><a href="#cb49-53" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-54"><a href="#cb49-54" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate data range for each gene to use proportional offset</span></span>
<span id="cb49-55"><a href="#cb49-55" aria-hidden="true" tabindex="-1"></a>  y_range_all <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb49-56"><a href="#cb49-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb49-57"><a href="#cb49-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">y_range =</span> <span class="fu">max</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">-</span> <span class="fu">min</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb49-58"><a href="#cb49-58" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-59"><a href="#cb49-59" aria-hidden="true" tabindex="-1"></a>  all_treat_annotations <span class="ot">&lt;-</span> all_treat_annotations <span class="sc">%&gt;%</span></span>
<span id="cb49-60"><a href="#cb49-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(y_pos_all, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"treatment_group"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-61"><a href="#cb49-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(y_range_all, <span class="at">by =</span> <span class="st">"Target"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-62"><a href="#cb49-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">y_pos =</span> max_y <span class="sc">+</span> y_range <span class="sc">*</span> <span class="fl">0.08</span>)  <span class="co"># 8% of data range above max</span></span>
<span id="cb49-63"><a href="#cb49-63" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-64"><a href="#cb49-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-65"><a href="#cb49-65" aria-hidden="true" tabindex="-1"></a><span class="co"># Create plot</span></span>
<span id="cb49-66"><a href="#cb49-66" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb49-67"><a href="#cb49-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> treatment_group, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> polyIC)) <span class="sc">+</span></span>
<span id="cb49-68"><a href="#cb49-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb49-69"><a href="#cb49-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Target, <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb49-70"><a href="#cb49-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"No_PolyIC"</span> <span class="ot">=</span> <span class="st">"darkgray"</span>, <span class="st">"PolyIC"</span> <span class="ot">=</span> <span class="st">"salmon"</span>)) <span class="sc">+</span></span>
<span id="cb49-71"><a href="#cb49-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.15</span>))) <span class="sc">+</span>  <span class="co"># Extra space at top for annotations</span></span>
<span id="cb49-72"><a href="#cb49-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb49-73"><a href="#cb49-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb49-74"><a href="#cb49-74" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Delta Cq Across All Treatment Groups"</span>,</span>
<span id="cb49-75"><a href="#cb49-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Treatment Group"</span>,</span>
<span id="cb49-76"><a href="#cb49-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Delta Cq"</span>,</span>
<span id="cb49-77"><a href="#cb49-77" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"PolyIC"</span></span>
<span id="cb49-78"><a href="#cb49-78" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb49-79"><a href="#cb49-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb49-80"><a href="#cb49-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-81"><a href="#cb49-81" aria-hidden="true" tabindex="-1"></a><span class="co"># Add annotations if they exist</span></span>
<span id="cb49-82"><a href="#cb49-82" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(all_treat_annotations) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb49-83"><a href="#cb49-83" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> p <span class="sc">+</span></span>
<span id="cb49-84"><a href="#cb49-84" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> all_treat_annotations,</span>
<span id="cb49-85"><a href="#cb49-85" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> treatment_group, <span class="at">y =</span> y_pos, <span class="at">label =</span> label),</span>
<span id="cb49-86"><a href="#cb49-86" aria-hidden="true" tabindex="-1"></a>              <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">vjust =</span> <span class="dv">0</span>, </span>
<span id="cb49-87"><a href="#cb49-87" aria-hidden="true" tabindex="-1"></a>              <span class="at">color =</span> <span class="st">"darkorange"</span>, <span class="at">fontface =</span> <span class="st">"bold"</span>) <span class="sc">+</span></span>
<span id="cb49-88"><a href="#cb49-88" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">"Letters show statistical groupings (p ≤ 0.05). Groups sharing a letter are NOT significantly different from each other"</span>,</span>
<span id="cb49-89"><a href="#cb49-89" aria-hidden="true" tabindex="-1"></a>         <span class="at">caption =</span> <span class="st">"Compact Letter Display shown for genes with significant ANOVA effects. See ANOVA Models section for detailed statistics."</span>)</span>
<span id="cb49-90"><a href="#cb49-90" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-91"><a href="#cb49-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-92"><a href="#cb49-92" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/boxplot-all-treatments-1.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="delta-delta-cq-and-fold-change-analysis" class="level1">
<h1>Delta Delta Cq and Fold Change Analysis</h1>
<section id="calculate-delta-delta-cq-for-each-comparison" class="level2">
<h2 class="anchored" data-anchor-id="calculate-delta-delta-cq-for-each-comparison">Calculate delta delta Cq for each comparison</h2>
<section id="comparison-1-stressors-vs-control-no-polyic" class="level3">
<h3 class="anchored" data-anchor-id="comparison-1-stressors-vs-control-no-polyic">Comparison 1: Stressors vs Control (No PolyIC)</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta delta Cq for Temperature vs Control (No PolyIC)</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>delta_delta_temp_no_polyIC <span class="ot">&lt;-</span> no_polyIC_data <span class="sc">%&gt;%</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_control =</span> <span class="fu">mean</span>(delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Control"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_temp =</span> <span class="fu">mean</span>(delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Temperature"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">delta_delta_Cq =</span> mean_temp <span class="sc">-</span> mean_control,</span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq),</span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">'drop'</span></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Temperature vs Control (No PolyIC):"</span>)</span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(delta_delta_temp_no_polyIC)</span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta delta Cq for Mechanical vs Control (No PolyIC)</span></span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a>delta_delta_mech_no_polyIC <span class="ot">&lt;-</span> no_polyIC_data <span class="sc">%&gt;%</span></span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_control =</span> <span class="fu">mean</span>(delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Control"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_mech =</span> <span class="fu">mean</span>(delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Mechanical"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">delta_delta_Cq =</span> mean_mech <span class="sc">-</span> mean_control,</span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq),</span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">'drop'</span></span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Mechanical vs Control (No PolyIC):"</span>)</span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(delta_delta_mech_no_polyIC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Temperature vs Control (No PolyIC):"
# A tibble: 7 × 5
  Target           mean_control mean_temp delta_delta_Cq fold_change
  &lt;chr&gt;                   &lt;dbl&gt;     &lt;dbl&gt;          &lt;dbl&gt;       &lt;dbl&gt;
1 ATP.Synthase           -0.118    0.0257         0.144        0.905
2 Citrate.Synthase        0.390    0.492          0.101        0.932
3 DNMT1                   6.26     6.04          -0.221        1.17 
4 HSP70                   5.51     0.493         -5.02        32.4  
5 HSP90                   0.457    0.427         -0.0306       1.02 
6 VIPERIN                 5.19     5.06          -0.136        1.10 
7 cGAS                    6.39     6.22          -0.166        1.12 
[1] "Mechanical vs Control (No PolyIC):"
# A tibble: 7 × 5
  Target           mean_control mean_mech delta_delta_Cq fold_change
  &lt;chr&gt;                   &lt;dbl&gt;     &lt;dbl&gt;          &lt;dbl&gt;       &lt;dbl&gt;
1 ATP.Synthase           -0.118    0.0333        0.151         0.900
2 Citrate.Synthase        0.390    0.685         0.295         0.815
3 DNMT1                   6.26     6.13         -0.128         1.09 
4 HSP70                   5.51     5.51          0.00376       0.997
5 HSP90                   0.457    0.346        -0.111         1.08 
6 VIPERIN                 5.19     4.88         -0.310         1.24 
7 cGAS                    6.39     5.94         -0.443         1.36 </code></pre>
</div>
</div>
</section>
<section id="comparison-2-polyic-control-vs-basic-control" class="level3">
<h3 class="anchored" data-anchor-id="comparison-2-polyic-control-vs-basic-control">Comparison 2: PolyIC Control vs Basic Control</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>delta_delta_polyIC_control <span class="ot">&lt;-</span> control_data <span class="sc">%&gt;%</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_no_polyIC =</span> <span class="fu">mean</span>(delta_Cq[polyIC <span class="sc">==</span> <span class="st">"No_PolyIC"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_polyIC =</span> <span class="fu">mean</span>(delta_Cq[polyIC <span class="sc">==</span> <span class="st">"PolyIC"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">delta_delta_Cq =</span> mean_polyIC <span class="sc">-</span> mean_no_polyIC,</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq),</span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">'drop'</span></span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"PolyIC Control vs Basic Control:"</span>)</span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(delta_delta_polyIC_control)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PolyIC Control vs Basic Control:"
# A tibble: 7 × 5
  Target           mean_no_polyIC mean_polyIC delta_delta_Cq fold_change
  &lt;chr&gt;                     &lt;dbl&gt;       &lt;dbl&gt;          &lt;dbl&gt;       &lt;dbl&gt;
1 ATP.Synthase             -0.118       2.14           2.26        0.209
2 Citrate.Synthase          0.390       2.41           2.02        0.247
3 DNMT1                     6.26        6.42           0.158       0.896
4 HSP70                     5.51        3.29          -2.22        4.66 
5 HSP90                     0.457      -0.113         -0.571       1.49 
6 VIPERIN                   5.19        4.01          -1.18        2.27 
7 cGAS                      6.39        6.22          -0.168       1.12 </code></pre>
</div>
</div>
</section>
<section id="comparison-3-polyic-stressors-vs-polyic-control" class="level3">
<h3 class="anchored" data-anchor-id="comparison-3-polyic-stressors-vs-polyic-control">Comparison 3: PolyIC Stressors vs PolyIC Control</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta delta Cq for Temperature vs Control (PolyIC)</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>delta_delta_temp_polyIC <span class="ot">&lt;-</span> polyIC_data <span class="sc">%&gt;%</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_control =</span> <span class="fu">mean</span>(delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Control"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_temp =</span> <span class="fu">mean</span>(delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Temperature"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">delta_delta_Cq =</span> mean_temp <span class="sc">-</span> mean_control,</span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq),</span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">'drop'</span></span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Temperature vs Control (PolyIC):"</span>)</span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(delta_delta_temp_polyIC)</span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate delta delta Cq for Mechanical vs Control (PolyIC)</span></span>
<span id="cb54-16"><a href="#cb54-16" aria-hidden="true" tabindex="-1"></a>delta_delta_mech_polyIC <span class="ot">&lt;-</span> polyIC_data <span class="sc">%&gt;%</span></span>
<span id="cb54-17"><a href="#cb54-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb54-18"><a href="#cb54-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb54-19"><a href="#cb54-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_control =</span> <span class="fu">mean</span>(delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Control"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb54-20"><a href="#cb54-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_mech =</span> <span class="fu">mean</span>(delta_Cq[stress_type <span class="sc">==</span> <span class="st">"Mechanical"</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb54-21"><a href="#cb54-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">delta_delta_Cq =</span> mean_mech <span class="sc">-</span> mean_control,</span>
<span id="cb54-22"><a href="#cb54-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">fold_change =</span> <span class="dv">2</span><span class="sc">^</span>(<span class="sc">-</span>delta_delta_Cq),</span>
<span id="cb54-23"><a href="#cb54-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">'drop'</span></span>
<span id="cb54-24"><a href="#cb54-24" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb54-25"><a href="#cb54-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-26"><a href="#cb54-26" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Mechanical vs Control (PolyIC):"</span>)</span>
<span id="cb54-27"><a href="#cb54-27" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(delta_delta_mech_polyIC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Temperature vs Control (PolyIC):"
# A tibble: 7 × 5
  Target           mean_control mean_temp delta_delta_Cq fold_change
  &lt;chr&gt;                   &lt;dbl&gt;     &lt;dbl&gt;          &lt;dbl&gt;       &lt;dbl&gt;
1 ATP.Synthase            2.14      0.376        -1.76         3.39 
2 Citrate.Synthase        2.41      0.508        -1.90         3.73 
3 DNMT1                   6.42      5.57         -0.854        1.81 
4 HSP70                   3.29     -0.692        -3.98        15.8  
5 HSP90                  -0.113    -0.395        -0.282        1.22 
6 VIPERIN                 4.01      4.42          0.408        0.754
7 cGAS                    6.22      6.19         -0.0287       1.02 
[1] "Mechanical vs Control (PolyIC):"
# A tibble: 7 × 5
  Target           mean_control mean_mech delta_delta_Cq fold_change
  &lt;chr&gt;                   &lt;dbl&gt;     &lt;dbl&gt;          &lt;dbl&gt;       &lt;dbl&gt;
1 ATP.Synthase            2.14      0.238        -1.90         3.73 
2 Citrate.Synthase        2.41      2.18         -0.224        1.17 
3 DNMT1                   6.42      6.30         -0.119        1.09 
4 HSP70                   3.29      3.02         -0.267        1.20 
5 HSP90                  -0.113    -0.284        -0.171        1.13 
6 VIPERIN                 4.01      4.11          0.0988       0.934
7 cGAS                    6.22      5.56         -0.655        1.57 </code></pre>
</div>
</div>
</section>
</section>
<section id="fold-change-bar-plots" class="level2">
<h2 class="anchored" data-anchor-id="fold-change-bar-plots">Fold Change Bar Plots</h2>
<section id="no-polyic-stressors-vs-control" class="level3">
<h3 class="anchored" data-anchor-id="no-polyic-stressors-vs-control">No PolyIC: Stressors vs Control</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine temperature and mechanical comparisons</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>fold_change_no_polyIC <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>  delta_delta_temp_no_polyIC <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">comparison =</span> <span class="st">"Temperature vs Control"</span>),</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>  delta_delta_mech_no_polyIC <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">comparison =</span> <span class="st">"Mechanical vs Control"</span>)</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>fold_change_no_polyIC <span class="sc">%&gt;%</span></span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> fold_change, <span class="at">fill =</span> comparison)) <span class="sc">+</span></span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">"No change (FC = 1)"</span>), <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_manual</span>(<span class="at">name =</span> <span class="st">""</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"No change (FC = 1)"</span> <span class="ot">=</span> <span class="st">"dashed"</span>)) <span class="sc">+</span></span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Fold Change: Stressors vs Control (No PolyIC)"</span>,</span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Target Gene"</span>,</span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Fold Change"</span>,</span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Comparison"</span></span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb56-19"><a href="#cb56-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/barplot-fold-change-no-polyIC-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="polyic-control-vs-basic-control" class="level3">
<h3 class="anchored" data-anchor-id="polyic-control-vs-basic-control">PolyIC Control vs Basic Control</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>delta_delta_polyIC_control <span class="sc">%&gt;%</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> fold_change, <span class="at">fill =</span> Target)) <span class="sc">+</span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">"No change (FC = 1)"</span>), <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_manual</span>(<span class="at">name =</span> <span class="st">""</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"No change (FC = 1)"</span> <span class="ot">=</span> <span class="st">"dashed"</span>)) <span class="sc">+</span></span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb57-8"><a href="#cb57-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Fold Change: PolyIC Control vs Basic Control"</span>,</span>
<span id="cb57-9"><a href="#cb57-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Target Gene"</span>,</span>
<span id="cb57-10"><a href="#cb57-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Fold Change"</span></span>
<span id="cb57-11"><a href="#cb57-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb57-12"><a href="#cb57-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/barplot-fold-change-polyIC-control-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="polyic-stressors-vs-control" class="level3">
<h3 class="anchored" data-anchor-id="polyic-stressors-vs-control">PolyIC: Stressors vs Control</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine temperature and mechanical comparisons for PolyIC</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>fold_change_polyIC <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>  delta_delta_temp_polyIC <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">comparison =</span> <span class="st">"Temperature vs Control"</span>),</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>  delta_delta_mech_polyIC <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">comparison =</span> <span class="st">"Mechanical vs Control"</span>)</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>fold_change_polyIC <span class="sc">%&gt;%</span></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Target, <span class="at">y =</span> fold_change, <span class="at">fill =</span> comparison)) <span class="sc">+</span></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">"No change (FC = 1)"</span>), <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_manual</span>(<span class="at">name =</span> <span class="st">""</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"No change (FC = 1)"</span> <span class="ot">=</span> <span class="st">"dashed"</span>)) <span class="sc">+</span></span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Fold Change: Stressors vs Control (PolyIC Treatment)"</span>,</span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Target Gene"</span>,</span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Fold Change"</span>,</span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Comparison"</span></span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/barplot-fold-change-polyIC-stress-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="anova-results-and-diagnostics" class="level1">
<h1>ANOVA Results and Diagnostics</h1>
<p>Run ANOVA models to test for effects of PolyIC treatment and stress type on delta Cq values for each target gene.</p>
<p><strong>Note:</strong> ANOVA models were created in the Statistical Comparisons section above. This section provides detailed assumption checks and interpretations for each gene.</p>
<section id="data-distribution-check" class="level2">
<h2 class="anchored" data-anchor-id="data-distribution-check">Data distribution check</h2>
<p>ANOVA assumes that data (or residuals) are approximately normally distributed. This histogram provides a quick visual check of the overall delta Cq distribution across all samples and targets.</p>
<p><strong>What to look for:</strong> - <strong>Bell-shaped curve</strong>: Indicates roughly normal distribution, supporting ANOVA use - <strong>Skewness</strong>: Strong left or right skew may violate normality assumptions - <strong>Outliers</strong>: Extreme values far from the main distribution - <strong>Multimodal</strong>: Multiple peaks may indicate distinct subpopulations</p>
<p>While this shows the overall distribution, we’ll check normality of residuals for each gene-specific model using QQ plots below.</p>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(delta_Cq_df<span class="sc">$</span>delta_Cq, <span class="at">main =</span> <span class="st">"Distribution of Delta Cq Values"</span>, <span class="at">xlab =</span> <span class="st">"Delta Cq"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/delta-cq-histogram-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="atp-synthase" class="level2">
<h2 class="anchored" data-anchor-id="atp-synthase">ATP Synthase</h2>
<p>ATP synthase is an enzyme complex that functions to synthesize adenosine triphosphate (ATP) from adenosine diphosphate (ADP) and inorganic phosphate (Pi), essentially generating the cell’s primary energy currency by harnessing the energy from a proton gradient across a membrane.</p>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>model_atp <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"ATP.Synthase"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model_atp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                    Df Sum Sq Mean Sq F value   Pr(&gt;F)    
polyIC               1  25.56  25.562  13.896 0.000316 ***
stress_type          2  13.54   6.771   3.681 0.028581 *  
polyIC:stress_type   2  23.45  11.723   6.373 0.002459 ** 
Residuals          103 189.48   1.840                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<section id="assumption-checks-for-atp-synthase" class="level3">
<h3 class="anchored" data-anchor-id="assumption-checks-for-atp-synthase">Assumption Checks for ATP Synthase</h3>
<p><strong>QQ Plot (Quantile-Quantile Plot)</strong>: Tests normality assumption of residuals.</p>
<ul>
<li><strong>What to look for</strong>: Points should fall along the diagonal reference line</li>
<li><strong>Good fit</strong>: Points closely follow the line, indicating normally distributed residuals</li>
<li><strong>Violations</strong>: Systematic deviations (S-curves, heavy tails) suggest non-normality</li>
<li><strong>Minor deviations</strong> at the ends are usually acceptable</li>
</ul>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model_atp<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/anova-atp-synthase-qq-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 16 43</code></pre>
</div>
</div>
<p><strong>Levene’s Test</strong>: Tests homogeneity of variance (equal variances across groups).</p>
<ul>
<li><strong>Null hypothesis</strong>: Variances are equal across all treatment groups</li>
<li><strong>p-value &gt; 0.05</strong>: Variances are homogeneous (assumption met) ✓</li>
<li><strong>p-value ≤ 0.05</strong>: Variances differ significantly (assumption violated)</li>
</ul>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model_atp<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"ATP.Synthase"</span>))</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a><span class="co"># If significant effects found, run post-hoc tests</span></span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_atp)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>  emm_atp <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_atp, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>  pairs_atp <span class="ot">&lt;-</span> <span class="fu">pairs</span>(emm_atp)</span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pairs_atp)</span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb64-10"><a href="#cb64-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify significant contrasts</span></span>
<span id="cb64-11"><a href="#cb64-11" aria-hidden="true" tabindex="-1"></a>  pairs_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pairs_atp)</span>
<span id="cb64-12"><a href="#cb64-12" aria-hidden="true" tabindex="-1"></a>  sig_contrasts <span class="ot">&lt;-</span> pairs_df[pairs_df<span class="sc">$</span>p.value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ]</span>
<span id="cb64-13"><a href="#cb64-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb64-14"><a href="#cb64-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**ATP Synthase - Pairwise Contrast Summary:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb64-15"><a href="#cb64-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(sig_contrasts) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb64-16"><a href="#cb64-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Significant contrasts (p ≤ 0.05):</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb64-17"><a href="#cb64-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(sig_contrasts)) {</span>
<span id="cb64-18"><a href="#cb64-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">"-"</span>, sig_contrasts<span class="sc">$</span>contrast[i], <span class="st">"(p ="</span>, <span class="fu">round</span>(sig_contrasts<span class="sc">$</span>p.value[i], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb64-19"><a href="#cb64-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb64-20"><a href="#cb64-20" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb64-21"><a href="#cb64-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"No significant pairwise contrasts found (all p &gt; 0.05).</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb64-22"><a href="#cb64-22" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb64-23"><a href="#cb64-23" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb64-24"><a href="#cb64-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"No significant effects found; post-hoc tests not performed.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb64-25"><a href="#cb64-25" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
       Df F value    Pr(&gt;F)    
group   5  18.846 2.899e-13 ***
      103                      
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
 contrast                                     estimate    SE  df t.ratio
 No_PolyIC Control - PolyIC Control           -2.25679 0.447 103  -5.044
 No_PolyIC Control - No_PolyIC Mechanical     -0.15123 0.429 103  -0.353
 No_PolyIC Control - PolyIC Mechanical        -0.35618 0.495 103  -0.719
 No_PolyIC Control - No_PolyIC Temperature    -0.14360 0.429 103  -0.335
 No_PolyIC Control - PolyIC Temperature       -0.49388 0.429 103  -1.151
 PolyIC Control - No_PolyIC Mechanical         2.10556 0.447 103   4.706
 PolyIC Control - PolyIC Mechanical            1.90061 0.511 103   3.717
 PolyIC Control - No_PolyIC Temperature        2.11319 0.447 103   4.723
 PolyIC Control - PolyIC Temperature           1.76291 0.447 103   3.940
 No_PolyIC Mechanical - PolyIC Mechanical     -0.20495 0.495 103  -0.414
 No_PolyIC Mechanical - No_PolyIC Temperature  0.00763 0.429 103   0.018
 No_PolyIC Mechanical - PolyIC Temperature    -0.34265 0.429 103  -0.799
 PolyIC Mechanical - No_PolyIC Temperature     0.21258 0.495 103   0.429
 PolyIC Mechanical - PolyIC Temperature       -0.13770 0.495 103  -0.278
 No_PolyIC Temperature - PolyIC Temperature   -0.35028 0.429 103  -0.817
 p.value
 &lt;0.0001
  0.9993
  0.9792
  0.9994
  0.8584
  0.0001
  0.0043
  0.0001
  0.0020
  0.9984
  1.0000
  0.9671
  0.9981
  0.9998
  0.9639

P value adjustment: tukey method for comparing a family of 6 estimates 

**ATP Synthase - Pairwise Contrast Summary:**

Significant contrasts (p ≤ 0.05):
- No_PolyIC Control - PolyIC Control (p = 0 )
- PolyIC Control - No_PolyIC Mechanical (p = 1e-04 )
- PolyIC Control - PolyIC Mechanical (p = 0.0043 )
- PolyIC Control - No_PolyIC Temperature (p = 1e-04 )
- PolyIC Control - PolyIC Temperature (p = 0.002 )</code></pre>
</div>
</div>
</section>
<section id="atp-synthase-anova-interpretation" class="level3">
<h3 class="anchored" data-anchor-id="atp-synthase-anova-interpretation">ATP Synthase ANOVA Interpretation</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>anova_results <span class="ot">&lt;-</span> <span class="fu">summary</span>(model_atp)[[<span class="dv">1</span>]]</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Fix: Trim whitespace from rownames</span></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(anova_results) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">rownames</span>(anova_results))</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>levene_results <span class="ot">&lt;-</span> <span class="fu">leveneTest</span>(model_atp<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"ATP.Synthase"</span>))</span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**ATP Synthase ANOVA Results:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"This ANOVA evaluates whether PolyIC treatment and/or stress type affect ATP Synthase expression (delta Cq).</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**Treatment Effects (ANOVA):**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Check main effects and interaction</span></span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb66-16"><a href="#cb66-16" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb66-17"><a href="#cb66-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb66-18"><a href="#cb66-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb66-19"><a href="#cb66-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-20"><a href="#cb66-20" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb66-21"><a href="#cb66-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb66-22"><a href="#cb66-22" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb66-23"><a href="#cb66-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb66-24"><a href="#cb66-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb66-25"><a href="#cb66-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-26"><a href="#cb66-26" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb66-27"><a href="#cb66-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb66-28"><a href="#cb66-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → The effect of stress depends on PolyIC treatment status.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb66-29"><a href="#cb66-29" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb66-30"><a href="#cb66-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb66-31"><a href="#cb66-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb66-32"><a href="#cb66-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-33"><a href="#cb66-33" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**Assumption Checks:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb66-34"><a href="#cb66-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-35"><a href="#cb66-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Check Levene's test</span></span>
<span id="cb66-36"><a href="#cb66-36" aria-hidden="true" tabindex="-1"></a>levene_p <span class="ot">&lt;-</span> levene_results<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb66-37"><a href="#cb66-37" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(levene_p <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb66-38"><a href="#cb66-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): PASSED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ✓</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb66-39"><a href="#cb66-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances are equal across groups; ANOVA is appropriate.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb66-40"><a href="#cb66-40" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb66-41"><a href="#cb66-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): FAILED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ⚠️</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb66-42"><a href="#cb66-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances differ across groups; ANOVA results should be interpreted with caution.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb66-43"><a href="#cb66-43" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>**ATP Synthase ANOVA Results:**

This ANOVA evaluates whether PolyIC treatment and/or stress type affect ATP Synthase expression (delta Cq).

**Treatment Effects (ANOVA):**

- **PolyIC effect**: SIGNIFICANT (p = 3e-04 )
- **Stress type effect**: SIGNIFICANT (p = 0.0286 )
- **PolyIC × Stress interaction**: SIGNIFICANT (p = 0.0025 )
  → The effect of stress depends on PolyIC treatment status.

**Assumption Checks:**

- **Levene's Test** (homogeneity of variance): FAILED (p = 0 ) ⚠️
  → Variances differ across groups; ANOVA results should be interpreted with caution.</code></pre>
</div>
</div>
</section>
</section>
<section id="citrate-synthase" class="level2">
<h2 class="anchored" data-anchor-id="citrate-synthase">Citrate Synthase</h2>
<p>Citrate synthase is important for energy production in the TCA cycle and is linked to the electron transport chain. It is also used as an enzyme marker for intact mitochondria.</p>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>model_citrate <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"Citrate.Synthase"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model_citrate)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                    Df Sum Sq Mean Sq F value   Pr(&gt;F)    
polyIC               1  32.28   32.28   35.11 4.00e-08 ***
stress_type          2  19.94    9.97   10.85 5.23e-05 ***
polyIC:stress_type   2  20.88   10.44   11.36 3.43e-05 ***
Residuals          105  96.53    0.92                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<section id="assumption-checks-for-citrate-synthase" class="level3">
<h3 class="anchored" data-anchor-id="assumption-checks-for-citrate-synthase">Assumption Checks for Citrate Synthase</h3>
<p><strong>QQ Plot (Quantile-Quantile Plot)</strong>: Tests normality assumption of residuals.</p>
<ul>
<li><strong>What to look for</strong>: Points should fall along the diagonal reference line</li>
<li><strong>Good fit</strong>: Points closely follow the line, indicating normally distributed residuals</li>
<li><strong>Violations</strong>: Systematic deviations (S-curves, heavy tails) suggest non-normality</li>
<li><strong>Minor deviations</strong> at the ends are usually acceptable</li>
</ul>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model_citrate<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/anova-citrate-synthase-qq-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 45 67</code></pre>
</div>
</div>
<p><strong>Levene’s Test</strong>: Tests homogeneity of variance (equal variances across groups).</p>
<ul>
<li><strong>Null hypothesis</strong>: Variances are equal across all treatment groups</li>
<li><strong>p-value &gt; 0.05</strong>: Variances are homogeneous (assumption met) ✓</li>
<li><strong>p-value ≤ 0.05</strong>: Variances differ significantly (assumption violated)</li>
</ul>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model_citrate<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"Citrate.Synthase"</span>))</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a><span class="co"># If significant effects found, run post-hoc tests</span></span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_citrate)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>  emm_citrate <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_citrate, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a>  pairs_citrate <span class="ot">&lt;-</span> <span class="fu">pairs</span>(emm_citrate)</span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pairs_citrate)</span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify significant contrasts</span></span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a>  pairs_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pairs_citrate)</span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a>  sig_contrasts <span class="ot">&lt;-</span> pairs_df[pairs_df<span class="sc">$</span>p.value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ]</span>
<span id="cb72-13"><a href="#cb72-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb72-14"><a href="#cb72-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**Citrate Synthase - Pairwise Contrast Summary:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb72-15"><a href="#cb72-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(sig_contrasts) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb72-16"><a href="#cb72-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Significant contrasts (p ≤ 0.05):</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb72-17"><a href="#cb72-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(sig_contrasts)) {</span>
<span id="cb72-18"><a href="#cb72-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">"-"</span>, sig_contrasts<span class="sc">$</span>contrast[i], <span class="st">"(p ="</span>, <span class="fu">round</span>(sig_contrasts<span class="sc">$</span>p.value[i], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb72-19"><a href="#cb72-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb72-20"><a href="#cb72-20" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb72-21"><a href="#cb72-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"No significant pairwise contrasts found (all p &gt; 0.05).</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb72-22"><a href="#cb72-22" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb72-23"><a href="#cb72-23" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb72-24"><a href="#cb72-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"No significant effects found; post-hoc tests not performed.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb72-25"><a href="#cb72-25" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
       Df F value   Pr(&gt;F)    
group   5  23.346 1.05e-15 ***
      105                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
 contrast                                     estimate    SE  df t.ratio
 No_PolyIC Control - PolyIC Control            -2.0155 0.312 105  -6.470
 No_PolyIC Control - No_PolyIC Mechanical      -0.2950 0.303 105  -0.973
 No_PolyIC Control - PolyIC Mechanical         -1.7910 0.342 105  -5.243
 No_PolyIC Control - No_PolyIC Temperature     -0.1012 0.303 105  -0.334
 No_PolyIC Control - PolyIC Temperature        -0.1179 0.303 105  -0.389
 PolyIC Control - No_PolyIC Mechanical          1.7206 0.312 105   5.523
 PolyIC Control - PolyIC Mechanical             0.2245 0.349 105   0.643
 PolyIC Control - No_PolyIC Temperature         1.9143 0.312 105   6.145
 PolyIC Control - PolyIC Temperature            1.8976 0.312 105   6.092
 No_PolyIC Mechanical - PolyIC Mechanical      -1.4961 0.342 105  -4.380
 No_PolyIC Mechanical - No_PolyIC Temperature   0.1937 0.303 105   0.639
 No_PolyIC Mechanical - PolyIC Temperature      0.1770 0.303 105   0.584
 PolyIC Mechanical - No_PolyIC Temperature      1.6898 0.342 105   4.947
 PolyIC Mechanical - PolyIC Temperature         1.6731 0.342 105   4.898
 No_PolyIC Temperature - PolyIC Temperature    -0.0167 0.303 105  -0.055
 p.value
 &lt;0.0001
  0.9256
 &lt;0.0001
  0.9994
  0.9988
 &lt;0.0001
  0.9874
 &lt;0.0001
 &lt;0.0001
  0.0004
  0.9878
  0.9919
 &lt;0.0001
 &lt;0.0001
  1.0000

P value adjustment: tukey method for comparing a family of 6 estimates 

**Citrate Synthase - Pairwise Contrast Summary:**

Significant contrasts (p ≤ 0.05):
- No_PolyIC Control - PolyIC Control (p = 0 )
- No_PolyIC Control - PolyIC Mechanical (p = 0 )
- PolyIC Control - No_PolyIC Mechanical (p = 0 )
- PolyIC Control - No_PolyIC Temperature (p = 0 )
- PolyIC Control - PolyIC Temperature (p = 0 )
- No_PolyIC Mechanical - PolyIC Mechanical (p = 4e-04 )
- PolyIC Mechanical - No_PolyIC Temperature (p = 0 )
- PolyIC Mechanical - PolyIC Temperature (p = 1e-04 )</code></pre>
</div>
</div>
</section>
<section id="citrate-synthase-anova-interpretation" class="level3">
<h3 class="anchored" data-anchor-id="citrate-synthase-anova-interpretation">Citrate Synthase ANOVA Interpretation</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>anova_results <span class="ot">&lt;-</span> <span class="fu">summary</span>(model_citrate)[[<span class="dv">1</span>]]</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Fix: Trim whitespace from rownames</span></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(anova_results) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">rownames</span>(anova_results))</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>levene_results <span class="ot">&lt;-</span> <span class="fu">leveneTest</span>(model_citrate<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"Citrate.Synthase"</span>))</span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**Citrate Synthase ANOVA Results:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"This ANOVA evaluates whether PolyIC treatment and/or stress type affect Citrate Synthase expression (delta Cq).</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**Treatment Effects (ANOVA):**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb74-14"><a href="#cb74-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb74-15"><a href="#cb74-15" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb74-16"><a href="#cb74-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb74-17"><a href="#cb74-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb74-18"><a href="#cb74-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-19"><a href="#cb74-19" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb74-20"><a href="#cb74-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb74-21"><a href="#cb74-21" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb74-22"><a href="#cb74-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb74-23"><a href="#cb74-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb74-24"><a href="#cb74-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-25"><a href="#cb74-25" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb74-26"><a href="#cb74-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb74-27"><a href="#cb74-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → The effect of stress depends on PolyIC treatment status.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb74-28"><a href="#cb74-28" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb74-29"><a href="#cb74-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb74-30"><a href="#cb74-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb74-31"><a href="#cb74-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-32"><a href="#cb74-32" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**Assumption Checks:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb74-33"><a href="#cb74-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-34"><a href="#cb74-34" aria-hidden="true" tabindex="-1"></a>levene_p <span class="ot">&lt;-</span> levene_results<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb74-35"><a href="#cb74-35" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(levene_p <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb74-36"><a href="#cb74-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): PASSED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ✓</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb74-37"><a href="#cb74-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances are equal across groups; ANOVA is appropriate.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb74-38"><a href="#cb74-38" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb74-39"><a href="#cb74-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): FAILED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ⚠️</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb74-40"><a href="#cb74-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances differ across groups; ANOVA results should be interpreted with caution.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb74-41"><a href="#cb74-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>**Citrate Synthase ANOVA Results:**

This ANOVA evaluates whether PolyIC treatment and/or stress type affect Citrate Synthase expression (delta Cq).

**Treatment Effects (ANOVA):**

- **PolyIC effect**: SIGNIFICANT (p = 0 )
- **Stress type effect**: SIGNIFICANT (p = 1e-04 )
- **PolyIC × Stress interaction**: SIGNIFICANT (p = 0 )
  → The effect of stress depends on PolyIC treatment status.

**Assumption Checks:**

- **Levene's Test** (homogeneity of variance): FAILED (p = 0 ) ⚠️
  → Variances differ across groups; ANOVA results should be interpreted with caution.</code></pre>
</div>
</div>
</section>
</section>
<section id="dnmt1" class="level2">
<h2 class="anchored" data-anchor-id="dnmt1">DNMT1</h2>
<p>The DNMT1 gene provides instructions for making an enzyme called DNA methyltransferase 1. This enzyme is involved in DNA methylation, which is the addition of methyl groups, consisting of one carbon atom and three hydrogen atoms, to DNA molecules.</p>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>model_dnmt1 <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"DNMT1"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model_dnmt1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                   Df Sum Sq Mean Sq F value Pr(&gt;F)
polyIC              1   1.70  1.7029   1.719  0.193
stress_type         2   3.47  1.7365   1.753  0.179
polyIC:stress_type  2   1.84  0.9206   0.929  0.399
Residuals          85  84.21  0.9907               </code></pre>
</div>
</div>
<section id="assumption-checks-for-dnmt1" class="level3">
<h3 class="anchored" data-anchor-id="assumption-checks-for-dnmt1">Assumption Checks for DNMT1</h3>
<p><strong>QQ Plot (Quantile-Quantile Plot)</strong>: Tests normality assumption of residuals.</p>
<ul>
<li><strong>What to look for</strong>: Points should fall along the diagonal reference line</li>
<li><strong>Good fit</strong>: Points closely follow the line, indicating normally distributed residuals</li>
<li><strong>Violations</strong>: Systematic deviations (S-curves, heavy tails) suggest non-normality</li>
<li><strong>Minor deviations</strong> at the ends are usually acceptable</li>
</ul>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model_dnmt1<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/anova-dnmt1-qq-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 82 11</code></pre>
</div>
</div>
<p><strong>Levene’s Test</strong>: Tests homogeneity of variance (equal variances across groups).</p>
<ul>
<li><strong>Null hypothesis</strong>: Variances are equal across all treatment groups</li>
<li><strong>p-value &gt; 0.05</strong>: Variances are homogeneous (assumption met) ✓</li>
<li><strong>p-value ≤ 0.05</strong>: Variances differ significantly (assumption violated)</li>
</ul>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model_dnmt1<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"DNMT1"</span>))</span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a><span class="co"># If significant effects found, run post-hoc tests</span></span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_dnmt1)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a>  emm_dnmt1 <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_dnmt1, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a>  pairs_dnmt1 <span class="ot">&lt;-</span> <span class="fu">pairs</span>(emm_dnmt1)</span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pairs_dnmt1)</span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb80-10"><a href="#cb80-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify significant contrasts</span></span>
<span id="cb80-11"><a href="#cb80-11" aria-hidden="true" tabindex="-1"></a>  pairs_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pairs_dnmt1)</span>
<span id="cb80-12"><a href="#cb80-12" aria-hidden="true" tabindex="-1"></a>  sig_contrasts <span class="ot">&lt;-</span> pairs_df[pairs_df<span class="sc">$</span>p.value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ]</span>
<span id="cb80-13"><a href="#cb80-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb80-14"><a href="#cb80-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**DNMT1 - Pairwise Contrast Summary:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb80-15"><a href="#cb80-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(sig_contrasts) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb80-16"><a href="#cb80-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Significant contrasts (p ≤ 0.05):</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb80-17"><a href="#cb80-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(sig_contrasts)) {</span>
<span id="cb80-18"><a href="#cb80-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">"-"</span>, sig_contrasts<span class="sc">$</span>contrast[i], <span class="st">"(p ="</span>, <span class="fu">round</span>(sig_contrasts<span class="sc">$</span>p.value[i], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb80-19"><a href="#cb80-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb80-20"><a href="#cb80-20" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb80-21"><a href="#cb80-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"No significant pairwise contrasts found (all p &gt; 0.05).</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb80-22"><a href="#cb80-22" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb80-23"><a href="#cb80-23" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb80-24"><a href="#cb80-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"No significant effects found; post-hoc tests not performed.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb80-25"><a href="#cb80-25" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
      Df F value Pr(&gt;F)
group  5  0.7104 0.6172
      85               
No significant effects found; post-hoc tests not performed.</code></pre>
</div>
</div>
</section>
<section id="dnmt1-anova-interpretation" class="level3">
<h3 class="anchored" data-anchor-id="dnmt1-anova-interpretation">DNMT1 ANOVA Interpretation</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>anova_results <span class="ot">&lt;-</span> <span class="fu">summary</span>(model_dnmt1)[[<span class="dv">1</span>]]</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Fix: Trim whitespace from rownames</span></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(anova_results) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">rownames</span>(anova_results))</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>levene_results <span class="ot">&lt;-</span> <span class="fu">leveneTest</span>(model_dnmt1<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"DNMT1"</span>))</span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**DNMT1 ANOVA Results:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"This ANOVA evaluates whether PolyIC treatment and/or stress type affect DNMT1 expression (delta Cq).</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**Treatment Effects (ANOVA):**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb82-20"><a href="#cb82-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb82-21"><a href="#cb82-21" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb82-22"><a href="#cb82-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb82-23"><a href="#cb82-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb82-24"><a href="#cb82-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-25"><a href="#cb82-25" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb82-26"><a href="#cb82-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb82-27"><a href="#cb82-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → The effect of stress depends on PolyIC treatment status.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb82-28"><a href="#cb82-28" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb82-29"><a href="#cb82-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb82-30"><a href="#cb82-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb82-31"><a href="#cb82-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-32"><a href="#cb82-32" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**Assumption Checks:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb82-33"><a href="#cb82-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-34"><a href="#cb82-34" aria-hidden="true" tabindex="-1"></a>levene_p <span class="ot">&lt;-</span> levene_results<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb82-35"><a href="#cb82-35" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(levene_p <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb82-36"><a href="#cb82-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): PASSED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ✓</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb82-37"><a href="#cb82-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances are equal across groups; ANOVA is appropriate.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb82-38"><a href="#cb82-38" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb82-39"><a href="#cb82-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): FAILED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ⚠️</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb82-40"><a href="#cb82-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances differ across groups; ANOVA results should be interpreted with caution.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb82-41"><a href="#cb82-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>**DNMT1 ANOVA Results:**

This ANOVA evaluates whether PolyIC treatment and/or stress type affect DNMT1 expression (delta Cq).

**Treatment Effects (ANOVA):**

- **PolyIC effect**: Not significant (p = 0.1934 )
- **Stress type effect**: Not significant (p = 0.1795 )
- **PolyIC × Stress interaction**: Not significant (p = 0.3988 )

**Assumption Checks:**

- **Levene's Test** (homogeneity of variance): PASSED (p = 0.6172 ) ✓
  → Variances are equal across groups; ANOVA is appropriate.</code></pre>
</div>
</div>
</section>
</section>
<section id="hsp70" class="level2">
<h2 class="anchored" data-anchor-id="hsp70">HSP70</h2>
<p>Heat Shock Protein 70 (Hsp70) is a molecular chaperone that plays crucial roles in maintaining cellular protein homeostasis and protecting cells from stress.</p>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>model_hsp70 <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"HSP70"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model_hsp70)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                    Df Sum Sq Mean Sq F value   Pr(&gt;F)    
polyIC               1  129.1  129.06  48.821 2.57e-10 ***
stress_type          2  510.9  255.47  96.637  &lt; 2e-16 ***
polyIC:stress_type   2    9.0    4.48   1.695    0.189    
Residuals          106  280.2    2.64                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<section id="assumption-checks-for-hsp70" class="level3">
<h3 class="anchored" data-anchor-id="assumption-checks-for-hsp70">Assumption Checks for HSP70</h3>
<p><strong>QQ Plot (Quantile-Quantile Plot)</strong>: Tests normality assumption of residuals.</p>
<ul>
<li><strong>What to look for</strong>: Points should fall along the diagonal reference line</li>
<li><strong>Good fit</strong>: Points closely follow the line, indicating normally distributed residuals</li>
<li><strong>Violations</strong>: Systematic deviations (S-curves, heavy tails) suggest non-normality</li>
<li><strong>Minor deviations</strong> at the ends are usually acceptable</li>
</ul>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model_hsp70<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/anova-hsp70-qq-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 112  33</code></pre>
</div>
</div>
<p><strong>Levene’s Test</strong>: Tests homogeneity of variance (equal variances across groups).</p>
<ul>
<li><strong>Null hypothesis</strong>: Variances are equal across all treatment groups</li>
<li><strong>p-value &gt; 0.05</strong>: Variances are homogeneous (assumption met) ✓</li>
<li><strong>p-value ≤ 0.05</strong>: Variances differ significantly (assumption violated)</li>
</ul>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model_hsp70<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"HSP70"</span>))</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a><span class="co"># If significant effects found, run post-hoc tests</span></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_hsp70)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>  emm_hsp70 <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_hsp70, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>  pairs_hsp70 <span class="ot">&lt;-</span> <span class="fu">pairs</span>(emm_hsp70)</span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pairs_hsp70)</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify significant contrasts</span></span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a>  pairs_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pairs_hsp70)</span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>  sig_contrasts <span class="ot">&lt;-</span> pairs_df[pairs_df<span class="sc">$</span>p.value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ]</span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**HSP70 - Pairwise Contrast Summary:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb88-15"><a href="#cb88-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(sig_contrasts) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb88-16"><a href="#cb88-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Significant contrasts (p ≤ 0.05):</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb88-17"><a href="#cb88-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(sig_contrasts)) {</span>
<span id="cb88-18"><a href="#cb88-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">"-"</span>, sig_contrasts<span class="sc">$</span>contrast[i], <span class="st">"(p ="</span>, <span class="fu">round</span>(sig_contrasts<span class="sc">$</span>p.value[i], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb88-19"><a href="#cb88-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb88-20"><a href="#cb88-20" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb88-21"><a href="#cb88-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"No significant pairwise contrasts found (all p &gt; 0.05).</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb88-22"><a href="#cb88-22" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb88-23"><a href="#cb88-23" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb88-24"><a href="#cb88-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"No significant effects found; post-hoc tests not performed.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb88-25"><a href="#cb88-25" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
       Df F value Pr(&gt;F)
group   5  0.5772 0.7174
      106               
 contrast                                     estimate    SE  df t.ratio
 No_PolyIC Control - PolyIC Control            2.22096 0.528 106   4.204
 No_PolyIC Control - No_PolyIC Mechanical     -0.00376 0.514 106  -0.007
 No_PolyIC Control - PolyIC Mechanical         2.48756 0.567 106   4.391
 No_PolyIC Control - No_PolyIC Temperature     5.01595 0.514 106   9.756
 No_PolyIC Control - PolyIC Temperature        6.20008 0.514 106  12.059
 PolyIC Control - No_PolyIC Mechanical        -2.22472 0.528 106  -4.212
 PolyIC Control - PolyIC Mechanical            0.26660 0.579 106   0.460
 PolyIC Control - No_PolyIC Temperature        2.79500 0.528 106   5.291
 PolyIC Control - PolyIC Temperature           3.97913 0.528 106   7.533
 No_PolyIC Mechanical - PolyIC Mechanical      2.49132 0.567 106   4.397
 No_PolyIC Mechanical - No_PolyIC Temperature  5.01971 0.514 106   9.763
 No_PolyIC Mechanical - PolyIC Temperature     6.20385 0.514 106  12.066
 PolyIC Mechanical - No_PolyIC Temperature     2.52840 0.567 106   4.463
 PolyIC Mechanical - PolyIC Temperature        3.71253 0.567 106   6.553
 No_PolyIC Temperature - PolyIC Temperature    1.18413 0.514 106   2.303
 p.value
  0.0008
  1.0000
  0.0004
 &lt;0.0001
 &lt;0.0001
  0.0007
  0.9974
 &lt;0.0001
 &lt;0.0001
  0.0004
 &lt;0.0001
 &lt;0.0001
  0.0003
 &lt;0.0001
  0.2021

P value adjustment: tukey method for comparing a family of 6 estimates 

**HSP70 - Pairwise Contrast Summary:**

Significant contrasts (p ≤ 0.05):
- No_PolyIC Control - PolyIC Control (p = 8e-04 )
- No_PolyIC Control - PolyIC Mechanical (p = 4e-04 )
- No_PolyIC Control - No_PolyIC Temperature (p = 0 )
- No_PolyIC Control - PolyIC Temperature (p = 0 )
- PolyIC Control - No_PolyIC Mechanical (p = 7e-04 )
- PolyIC Control - No_PolyIC Temperature (p = 0 )
- PolyIC Control - PolyIC Temperature (p = 0 )
- No_PolyIC Mechanical - PolyIC Mechanical (p = 4e-04 )
- No_PolyIC Mechanical - No_PolyIC Temperature (p = 0 )
- No_PolyIC Mechanical - PolyIC Temperature (p = 0 )
- PolyIC Mechanical - No_PolyIC Temperature (p = 3e-04 )
- PolyIC Mechanical - PolyIC Temperature (p = 0 )</code></pre>
</div>
</div>
</section>
<section id="hsp70-anova-interpretation" class="level3">
<h3 class="anchored" data-anchor-id="hsp70-anova-interpretation">HSP70 ANOVA Interpretation</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>anova_results <span class="ot">&lt;-</span> <span class="fu">summary</span>(model_hsp70)[[<span class="dv">1</span>]]</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Fix: Trim whitespace from rownames</span></span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(anova_results) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">rownames</span>(anova_results))</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>levene_results <span class="ot">&lt;-</span> <span class="fu">leveneTest</span>(model_hsp70<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"HSP70"</span>))</span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**HSP70 ANOVA Results:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"This ANOVA evaluates whether PolyIC treatment and/or stress type affect HSP70 expression (delta Cq).</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**Treatment Effects (ANOVA):**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb90-12"><a href="#cb90-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-13"><a href="#cb90-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb90-14"><a href="#cb90-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb90-15"><a href="#cb90-15" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb90-16"><a href="#cb90-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb90-17"><a href="#cb90-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb90-18"><a href="#cb90-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-19"><a href="#cb90-19" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb90-20"><a href="#cb90-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb90-21"><a href="#cb90-21" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb90-22"><a href="#cb90-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb90-23"><a href="#cb90-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb90-24"><a href="#cb90-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-25"><a href="#cb90-25" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb90-26"><a href="#cb90-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb90-27"><a href="#cb90-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → The effect of stress depends on PolyIC treatment status.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb90-28"><a href="#cb90-28" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb90-29"><a href="#cb90-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb90-30"><a href="#cb90-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb90-31"><a href="#cb90-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-32"><a href="#cb90-32" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**Assumption Checks:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb90-33"><a href="#cb90-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-34"><a href="#cb90-34" aria-hidden="true" tabindex="-1"></a>levene_p <span class="ot">&lt;-</span> levene_results<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb90-35"><a href="#cb90-35" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(levene_p <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb90-36"><a href="#cb90-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): PASSED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ✓</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb90-37"><a href="#cb90-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances are equal across groups; ANOVA is appropriate.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb90-38"><a href="#cb90-38" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb90-39"><a href="#cb90-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): FAILED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ⚠️</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb90-40"><a href="#cb90-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances differ across groups; ANOVA results should be interpreted with caution.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb90-41"><a href="#cb90-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>**HSP70 ANOVA Results:**

This ANOVA evaluates whether PolyIC treatment and/or stress type affect HSP70 expression (delta Cq).

**Treatment Effects (ANOVA):**

- **PolyIC effect**: SIGNIFICANT (p = 0 )
- **Stress type effect**: SIGNIFICANT (p = 0 )
- **PolyIC × Stress interaction**: Not significant (p = 0.1886 )

**Assumption Checks:**

- **Levene's Test** (homogeneity of variance): PASSED (p = 0.7174 ) ✓
  → Variances are equal across groups; ANOVA is appropriate.</code></pre>
</div>
</div>
</section>
</section>
<section id="hsp90" class="level2">
<h2 class="anchored" data-anchor-id="hsp90">HSP90</h2>
<p>Heat shock protein 90 (Hsp90) is a molecular chaperone that helps proteins fold, mature, and remain active. Hsp90 also helps regulate signaling networks and is involved in many cellular processes.</p>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>model_hsp90 <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"HSP90"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model_hsp90)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                   Df Sum Sq Mean Sq F value   Pr(&gt;F)    
polyIC              1 10.845  10.845  48.902 5.47e-10 ***
stress_type         2  0.332   0.166   0.749    0.476    
polyIC:stress_type  2  0.257   0.129   0.581    0.562    
Residuals          86 19.072   0.222                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<section id="assumption-checks-for-hsp90" class="level3">
<h3 class="anchored" data-anchor-id="assumption-checks-for-hsp90">Assumption Checks for HSP90</h3>
<p><strong>QQ Plot (Quantile-Quantile Plot)</strong>: Tests normality assumption of residuals.</p>
<ul>
<li><strong>What to look for</strong>: Points should fall along the diagonal reference line</li>
<li><strong>Good fit</strong>: Points closely follow the line, indicating normally distributed residuals</li>
<li><strong>Violations</strong>: Systematic deviations (S-curves, heavy tails) suggest non-normality</li>
<li><strong>Minor deviations</strong> at the ends are usually acceptable</li>
</ul>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model_hsp90<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/anova-hsp90-qq-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 63 11</code></pre>
</div>
</div>
<p><strong>Levene’s Test</strong>: Tests homogeneity of variance (equal variances across groups).</p>
<ul>
<li><strong>Null hypothesis</strong>: Variances are equal across all treatment groups</li>
<li><strong>p-value &gt; 0.05</strong>: Variances are homogeneous (assumption met) ✓</li>
<li><strong>p-value ≤ 0.05</strong>: Variances differ significantly (assumption violated)</li>
</ul>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model_hsp90<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"HSP90"</span>))</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a><span class="co"># If significant effects found, run post-hoc tests</span></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_hsp90)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a>  emm_hsp90 <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_hsp90, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a>  pairs_hsp90 <span class="ot">&lt;-</span> <span class="fu">pairs</span>(emm_hsp90)</span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pairs_hsp90)</span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify significant contrasts</span></span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a>  pairs_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pairs_hsp90)</span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>  sig_contrasts <span class="ot">&lt;-</span> pairs_df[pairs_df<span class="sc">$</span>p.value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ]</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**HSP90 - Pairwise Contrast Summary:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(sig_contrasts) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Significant contrasts (p ≤ 0.05):</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(sig_contrasts)) {</span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">"-"</span>, sig_contrasts<span class="sc">$</span>contrast[i], <span class="st">"(p ="</span>, <span class="fu">round</span>(sig_contrasts<span class="sc">$</span>p.value[i], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb96-19"><a href="#cb96-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb96-20"><a href="#cb96-20" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb96-21"><a href="#cb96-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"No significant pairwise contrasts found (all p &gt; 0.05).</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb96-22"><a href="#cb96-22" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb96-23"><a href="#cb96-23" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb96-24"><a href="#cb96-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"No significant effects found; post-hoc tests not performed.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb96-25"><a href="#cb96-25" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
      Df F value Pr(&gt;F)
group  5  0.2428 0.9423
      86               
 contrast                                     estimate    SE df t.ratio p.value
 No_PolyIC Control - PolyIC Control             0.5705 0.197 86   2.896  0.0525
 No_PolyIC Control - No_PolyIC Mechanical       0.1112 0.149 86   0.747  0.9753
 No_PolyIC Control - PolyIC Mechanical          0.7415 0.258 86   2.875  0.0555
 No_PolyIC Control - No_PolyIC Temperature      0.0306 0.149 86   0.205  0.9999
 No_PolyIC Control - PolyIC Temperature         0.8525 0.149 86   5.725 &lt;0.0001
 PolyIC Control - No_PolyIC Mechanical         -0.4593 0.197 86  -2.331  0.1930
 PolyIC Control - PolyIC Mechanical             0.1710 0.288 86   0.593  0.9913
 PolyIC Control - No_PolyIC Temperature        -0.5399 0.197 86  -2.741  0.0777
 PolyIC Control - PolyIC Temperature            0.2820 0.197 86   1.432  0.7078
 No_PolyIC Mechanical - PolyIC Mechanical       0.6303 0.258 86   2.444  0.1531
 No_PolyIC Mechanical - No_PolyIC Temperature  -0.0806 0.149 86  -0.541  0.9943
 No_PolyIC Mechanical - PolyIC Temperature      0.7413 0.149 86   4.978 &lt;0.0001
 PolyIC Mechanical - No_PolyIC Temperature     -0.7109 0.258 86  -2.756  0.0748
 PolyIC Mechanical - PolyIC Temperature         0.1110 0.258 86   0.430  0.9981
 No_PolyIC Temperature - PolyIC Temperature     0.8219 0.149 86   5.519 &lt;0.0001

P value adjustment: tukey method for comparing a family of 6 estimates 

**HSP90 - Pairwise Contrast Summary:**

Significant contrasts (p ≤ 0.05):
- No_PolyIC Control - PolyIC Temperature (p = 0 )
- No_PolyIC Mechanical - PolyIC Temperature (p = 0 )
- No_PolyIC Temperature - PolyIC Temperature (p = 0 )</code></pre>
</div>
</div>
</section>
<section id="hsp90-anova-interpretation" class="level3">
<h3 class="anchored" data-anchor-id="hsp90-anova-interpretation">HSP90 ANOVA Interpretation</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>anova_results <span class="ot">&lt;-</span> <span class="fu">summary</span>(model_hsp90)[[<span class="dv">1</span>]]</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Fix: Trim whitespace from rownames</span></span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(anova_results) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">rownames</span>(anova_results))</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>levene_results <span class="ot">&lt;-</span> <span class="fu">leveneTest</span>(model_hsp90<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"HSP90"</span>))</span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-8"><a href="#cb98-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**HSP90 ANOVA Results:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb98-9"><a href="#cb98-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"This ANOVA evaluates whether PolyIC treatment and/or stress type affect HSP90 expression (delta Cq).</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb98-10"><a href="#cb98-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-11"><a href="#cb98-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**Treatment Effects (ANOVA):**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb98-12"><a href="#cb98-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-13"><a href="#cb98-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb98-14"><a href="#cb98-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb98-15"><a href="#cb98-15" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb98-16"><a href="#cb98-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb98-17"><a href="#cb98-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb98-18"><a href="#cb98-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-19"><a href="#cb98-19" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb98-20"><a href="#cb98-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb98-21"><a href="#cb98-21" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb98-22"><a href="#cb98-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb98-23"><a href="#cb98-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb98-24"><a href="#cb98-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-25"><a href="#cb98-25" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb98-26"><a href="#cb98-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb98-27"><a href="#cb98-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → The effect of stress depends on PolyIC treatment status.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb98-28"><a href="#cb98-28" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb98-29"><a href="#cb98-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb98-30"><a href="#cb98-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb98-31"><a href="#cb98-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-32"><a href="#cb98-32" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**Assumption Checks:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb98-33"><a href="#cb98-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-34"><a href="#cb98-34" aria-hidden="true" tabindex="-1"></a>levene_p <span class="ot">&lt;-</span> levene_results<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb98-35"><a href="#cb98-35" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(levene_p <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb98-36"><a href="#cb98-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): PASSED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ✓</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb98-37"><a href="#cb98-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances are equal across groups; ANOVA is appropriate.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb98-38"><a href="#cb98-38" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb98-39"><a href="#cb98-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): FAILED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ⚠️</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb98-40"><a href="#cb98-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances differ across groups; ANOVA results should be interpreted with caution.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb98-41"><a href="#cb98-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>**HSP90 ANOVA Results:**

This ANOVA evaluates whether PolyIC treatment and/or stress type affect HSP90 expression (delta Cq).

**Treatment Effects (ANOVA):**

- **PolyIC effect**: SIGNIFICANT (p = 0 )
- **Stress type effect**: Not significant (p = 0.4758 )
- **PolyIC × Stress interaction**: Not significant (p = 0.5618 )

**Assumption Checks:**

- **Levene's Test** (homogeneity of variance): PASSED (p = 0.9423 ) ✓
  → Variances are equal across groups; ANOVA is appropriate.</code></pre>
</div>
</div>
</section>
</section>
<section id="cgas" class="level2">
<h2 class="anchored" data-anchor-id="cgas">cGAS</h2>
<p>The cGAS gene is involved in several processes, including cellular response to exogenous dsRNA, positive regulation of intracellular signal transduction, and regulation of defense response.</p>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>model_cgas <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"cGAS"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model_cgas)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                   Df Sum Sq Mean Sq F value Pr(&gt;F)
polyIC              1   0.09  0.0880   0.103  0.749
stress_type         2   3.10  1.5510   1.822  0.168
polyIC:stress_type  2   0.32  0.1577   0.185  0.831
Residuals          86  73.21  0.8513               </code></pre>
</div>
</div>
<section id="assumption-checks-for-cgas" class="level3">
<h3 class="anchored" data-anchor-id="assumption-checks-for-cgas">Assumption Checks for cGAS</h3>
<p><strong>QQ Plot (Quantile-Quantile Plot)</strong>: Tests normality assumption of residuals.</p>
<ul>
<li><strong>What to look for</strong>: Points should fall along the diagonal reference line</li>
<li><strong>Good fit</strong>: Points closely follow the line, indicating normally distributed residuals</li>
<li><strong>Violations</strong>: Systematic deviations (S-curves, heavy tails) suggest non-normality</li>
<li><strong>Minor deviations</strong> at the ends are usually acceptable</li>
</ul>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model_cgas<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/anova-cgas-qq-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 68 83</code></pre>
</div>
</div>
<p><strong>Levene’s Test</strong>: Tests homogeneity of variance (equal variances across groups).</p>
<ul>
<li><strong>Null hypothesis</strong>: Variances are equal across all treatment groups</li>
<li><strong>p-value &gt; 0.05</strong>: Variances are homogeneous (assumption met) ✓</li>
<li><strong>p-value ≤ 0.05</strong>: Variances differ significantly (assumption violated)</li>
</ul>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model_cgas<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"cGAS"</span>))</span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a><span class="co"># If significant effects found, run post-hoc tests</span></span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_cgas)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a>  emm_cgas <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_cgas, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb104-7"><a href="#cb104-7" aria-hidden="true" tabindex="-1"></a>  pairs_cgas <span class="ot">&lt;-</span> <span class="fu">pairs</span>(emm_cgas)</span>
<span id="cb104-8"><a href="#cb104-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pairs_cgas)</span>
<span id="cb104-9"><a href="#cb104-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb104-10"><a href="#cb104-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify significant contrasts</span></span>
<span id="cb104-11"><a href="#cb104-11" aria-hidden="true" tabindex="-1"></a>  pairs_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pairs_cgas)</span>
<span id="cb104-12"><a href="#cb104-12" aria-hidden="true" tabindex="-1"></a>  sig_contrasts <span class="ot">&lt;-</span> pairs_df[pairs_df<span class="sc">$</span>p.value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ]</span>
<span id="cb104-13"><a href="#cb104-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb104-14"><a href="#cb104-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**cGAS - Pairwise Contrast Summary:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb104-15"><a href="#cb104-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(sig_contrasts) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb104-16"><a href="#cb104-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Significant contrasts (p ≤ 0.05):</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb104-17"><a href="#cb104-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(sig_contrasts)) {</span>
<span id="cb104-18"><a href="#cb104-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">"-"</span>, sig_contrasts<span class="sc">$</span>contrast[i], <span class="st">"(p ="</span>, <span class="fu">round</span>(sig_contrasts<span class="sc">$</span>p.value[i], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb104-19"><a href="#cb104-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb104-20"><a href="#cb104-20" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb104-21"><a href="#cb104-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"No significant pairwise contrasts found (all p &gt; 0.05).</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb104-22"><a href="#cb104-22" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb104-23"><a href="#cb104-23" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb104-24"><a href="#cb104-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"No significant effects found; post-hoc tests not performed.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb104-25"><a href="#cb104-25" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
      Df F value Pr(&gt;F)
group  5  0.7847 0.5635
      86               
No significant effects found; post-hoc tests not performed.</code></pre>
</div>
</div>
</section>
<section id="cgas-anova-interpretation" class="level3">
<h3 class="anchored" data-anchor-id="cgas-anova-interpretation">cGAS ANOVA Interpretation</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>anova_results <span class="ot">&lt;-</span> <span class="fu">summary</span>(model_cgas)[[<span class="dv">1</span>]]</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Fix: Trim whitespace from rownames</span></span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(anova_results) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">rownames</span>(anova_results))</span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>levene_results <span class="ot">&lt;-</span> <span class="fu">leveneTest</span>(model_cgas<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"cGAS"</span>))</span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**cGAS ANOVA Results:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"This ANOVA evaluates whether PolyIC treatment and/or stress type affect cGAS expression (delta Cq).</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**Treatment Effects (ANOVA):**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb106-14"><a href="#cb106-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb106-15"><a href="#cb106-15" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb106-16"><a href="#cb106-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb106-17"><a href="#cb106-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb106-18"><a href="#cb106-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-19"><a href="#cb106-19" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb106-20"><a href="#cb106-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb106-21"><a href="#cb106-21" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb106-22"><a href="#cb106-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb106-23"><a href="#cb106-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb106-24"><a href="#cb106-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-25"><a href="#cb106-25" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb106-26"><a href="#cb106-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb106-27"><a href="#cb106-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → The effect of stress depends on PolyIC treatment status.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb106-28"><a href="#cb106-28" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb106-29"><a href="#cb106-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb106-30"><a href="#cb106-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb106-31"><a href="#cb106-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-32"><a href="#cb106-32" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**Assumption Checks:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb106-33"><a href="#cb106-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-34"><a href="#cb106-34" aria-hidden="true" tabindex="-1"></a>levene_p <span class="ot">&lt;-</span> levene_results<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb106-35"><a href="#cb106-35" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(levene_p <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb106-36"><a href="#cb106-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): PASSED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ✓</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb106-37"><a href="#cb106-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances are equal across groups; ANOVA is appropriate.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb106-38"><a href="#cb106-38" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb106-39"><a href="#cb106-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): FAILED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ⚠️</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb106-40"><a href="#cb106-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances differ across groups; ANOVA results should be interpreted with caution.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb106-41"><a href="#cb106-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>**cGAS ANOVA Results:**

This ANOVA evaluates whether PolyIC treatment and/or stress type affect cGAS expression (delta Cq).

**Treatment Effects (ANOVA):**

- **PolyIC effect**: Not significant (p = 0.7486 )
- **Stress type effect**: Not significant (p = 0.1679 )
- **PolyIC × Stress interaction**: Not significant (p = 0.8312 )

**Assumption Checks:**

- **Levene's Test** (homogeneity of variance): PASSED (p = 0.5635 ) ✓
  → Variances are equal across groups; ANOVA is appropriate.</code></pre>
</div>
</div>
</section>
</section>
<section id="viperin" class="level2">
<h2 class="anchored" data-anchor-id="viperin">VIPERIN</h2>
<p>Viperin limits replication of DNA and RNA viruses.</p>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>model_viperin <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"VIPERIN"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(delta_Cq <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, <span class="at">data =</span> .)</span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model_viperin)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                   Df Sum Sq Mean Sq F value   Pr(&gt;F)    
polyIC              1  12.27  12.269  15.528 0.000165 ***
stress_type         2   0.96   0.478   0.605 0.548320    
polyIC:stress_type  2   1.09   0.544   0.689 0.505068    
Residuals          86  67.95   0.790                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<section id="assumption-checks-for-viperin" class="level3">
<h3 class="anchored" data-anchor-id="assumption-checks-for-viperin">Assumption Checks for VIPERIN</h3>
<p><strong>QQ Plot (Quantile-Quantile Plot)</strong>: Tests normality assumption of residuals.</p>
<ul>
<li><strong>What to look for</strong>: Points should fall along the diagonal reference line</li>
<li><strong>Good fit</strong>: Points closely follow the line, indicating normally distributed residuals</li>
<li><strong>Violations</strong>: Systematic deviations (S-curves, heavy tails) suggest non-normality</li>
<li><strong>Minor deviations</strong> at the ends are usually acceptable</li>
</ul>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(model_viperin<span class="sc">$</span>residuals)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/anova-viperin-qq-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 49  1</code></pre>
</div>
</div>
<p><strong>Levene’s Test</strong>: Tests homogeneity of variance (equal variances across groups).</p>
<ul>
<li><strong>Null hypothesis</strong>: Variances are equal across all treatment groups</li>
<li><strong>p-value &gt; 0.05</strong>: Variances are homogeneous (assumption met) ✓</li>
<li><strong>p-value ≤ 0.05</strong>: Variances differ significantly (assumption violated)</li>
</ul>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="fu">leveneTest</span>(model_viperin<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"VIPERIN"</span>))</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a><span class="co"># If significant effects found, run post-hoc tests</span></span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">summary</span>(model_viperin)[[<span class="dv">1</span>]][[<span class="st">"Pr(&gt;F)"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>  emm_viperin <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(model_viperin, <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type)</span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a>  pairs_viperin <span class="ot">&lt;-</span> <span class="fu">pairs</span>(emm_viperin)</span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pairs_viperin)</span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify significant contrasts</span></span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a>  pairs_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pairs_viperin)</span>
<span id="cb112-12"><a href="#cb112-12" aria-hidden="true" tabindex="-1"></a>  sig_contrasts <span class="ot">&lt;-</span> pairs_df[pairs_df<span class="sc">$</span>p.value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ]</span>
<span id="cb112-13"><a href="#cb112-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb112-14"><a href="#cb112-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**VIPERIN - Pairwise Contrast Summary:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb112-15"><a href="#cb112-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(sig_contrasts) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb112-16"><a href="#cb112-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Significant contrasts (p ≤ 0.05):</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb112-17"><a href="#cb112-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(sig_contrasts)) {</span>
<span id="cb112-18"><a href="#cb112-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">"-"</span>, sig_contrasts<span class="sc">$</span>contrast[i], <span class="st">"(p ="</span>, <span class="fu">round</span>(sig_contrasts<span class="sc">$</span>p.value[i], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb112-19"><a href="#cb112-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb112-20"><a href="#cb112-20" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb112-21"><a href="#cb112-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"No significant pairwise contrasts found (all p &gt; 0.05).</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb112-22"><a href="#cb112-22" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb112-23"><a href="#cb112-23" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb112-24"><a href="#cb112-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"No significant effects found; post-hoc tests not performed.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb112-25"><a href="#cb112-25" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Levene's Test for Homogeneity of Variance (center = median)
      Df F value Pr(&gt;F)
group  5  0.6136 0.6897
      86               
 contrast                                     estimate    SE df t.ratio p.value
 No_PolyIC Control - PolyIC Control             1.1827 0.372 86   3.181  0.0242
 No_PolyIC Control - No_PolyIC Mechanical       0.3096 0.281 86   1.101  0.8795
 No_PolyIC Control - PolyIC Mechanical          1.0839 0.487 86   2.226  0.2367
 No_PolyIC Control - No_PolyIC Temperature      0.1363 0.281 86   0.485  0.9966
 No_PolyIC Control - PolyIC Temperature         0.7747 0.281 86   2.756  0.0748
 PolyIC Control - No_PolyIC Mechanical         -0.8731 0.372 86  -2.348  0.1866
 PolyIC Control - PolyIC Mechanical            -0.0988 0.544 86  -0.182  1.0000
 PolyIC Control - No_PolyIC Temperature        -1.0464 0.372 86  -2.814  0.0647
 PolyIC Control - PolyIC Temperature           -0.4080 0.372 86  -1.097  0.8812
 No_PolyIC Mechanical - PolyIC Mechanical       0.7743 0.487 86   1.590  0.6071
 No_PolyIC Mechanical - No_PolyIC Temperature  -0.1733 0.281 86  -0.616  0.9896
 No_PolyIC Mechanical - PolyIC Temperature      0.4651 0.281 86   1.655  0.5651
 PolyIC Mechanical - No_PolyIC Temperature     -0.9476 0.487 86  -1.946  0.3817
 PolyIC Mechanical - PolyIC Temperature        -0.3092 0.487 86  -0.635  0.9880
 No_PolyIC Temperature - PolyIC Temperature     0.6384 0.281 86   2.271  0.2173

P value adjustment: tukey method for comparing a family of 6 estimates 

**VIPERIN - Pairwise Contrast Summary:**

Significant contrasts (p ≤ 0.05):
- No_PolyIC Control - PolyIC Control (p = 0.0242 )</code></pre>
</div>
</div>
</section>
<section id="viperin-anova-interpretation" class="level3">
<h3 class="anchored" data-anchor-id="viperin-anova-interpretation">VIPERIN ANOVA Interpretation</h3>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>anova_results <span class="ot">&lt;-</span> <span class="fu">summary</span>(model_viperin)[[<span class="dv">1</span>]]</span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Fix: Trim whitespace from rownames</span></span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(anova_results) <span class="ot">&lt;-</span> <span class="fu">trimws</span>(<span class="fu">rownames</span>(anova_results))</span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>levene_results <span class="ot">&lt;-</span> <span class="fu">leveneTest</span>(model_viperin<span class="sc">$</span>residuals <span class="sc">~</span> polyIC <span class="sc">*</span> stress_type, </span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">data =</span> delta_Cq_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Target <span class="sc">==</span> <span class="st">"VIPERIN"</span>))</span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**VIPERIN ANOVA Results:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"This ANOVA evaluates whether PolyIC treatment and/or stress type affect VIPERIN expression (delta Cq).</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"**Treatment Effects (ANOVA):**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb114-12"><a href="#cb114-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-13"><a href="#cb114-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb114-14"><a href="#cb114-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb114-15"><a href="#cb114-15" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb114-16"><a href="#cb114-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb114-17"><a href="#cb114-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb114-18"><a href="#cb114-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-19"><a href="#cb114-19" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb114-20"><a href="#cb114-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb114-21"><a href="#cb114-21" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb114-22"><a href="#cb114-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Stress type effect**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb114-23"><a href="#cb114-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb114-24"><a href="#cb114-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-25"><a href="#cb114-25" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>] <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb114-26"><a href="#cb114-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: SIGNIFICANT (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb114-27"><a href="#cb114-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → The effect of stress depends on PolyIC treatment status.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb114-28"><a href="#cb114-28" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb114-29"><a href="#cb114-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **PolyIC × Stress interaction**: Not significant (p ="</span>, <span class="fu">round</span>(anova_results[<span class="st">"polyIC:stress_type"</span>, <span class="st">"Pr(&gt;F)"</span>], <span class="dv">4</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb114-30"><a href="#cb114-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb114-31"><a href="#cb114-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-32"><a href="#cb114-32" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">**Assumption Checks:**</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb114-33"><a href="#cb114-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-34"><a href="#cb114-34" aria-hidden="true" tabindex="-1"></a>levene_p <span class="ot">&lt;-</span> levene_results<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb114-35"><a href="#cb114-35" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(levene_p <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb114-36"><a href="#cb114-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): PASSED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ✓</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb114-37"><a href="#cb114-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances are equal across groups; ANOVA is appropriate.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb114-38"><a href="#cb114-38" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb114-39"><a href="#cb114-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"- **Levene's Test** (homogeneity of variance): FAILED (p ="</span>, <span class="fu">round</span>(levene_p, <span class="dv">4</span>), <span class="st">") ⚠️</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb114-40"><a href="#cb114-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  → Variances differ across groups; ANOVA results should be interpreted with caution.</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb114-41"><a href="#cb114-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>**VIPERIN ANOVA Results:**

This ANOVA evaluates whether PolyIC treatment and/or stress type affect VIPERIN expression (delta Cq).

**Treatment Effects (ANOVA):**

- **PolyIC effect**: SIGNIFICANT (p = 2e-04 )
- **Stress type effect**: Not significant (p = 0.5483 )
- **PolyIC × Stress interaction**: Not significant (p = 0.5051 )

**Assumption Checks:**

- **Levene's Test** (homogeneity of variance): PASSED (p = 0.6897 ) ✓
  → Variances are equal across groups; ANOVA is appropriate.</code></pre>
</div>
</div>
</section>
</section>
<section id="summary-plots" class="level2">
<h2 class="anchored" data-anchor-id="summary-plots">Summary Plots</h2>
<p>Visualize delta Cq values by treatment group for each target.</p>
<div class="cell" data-hold="true">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate Compact Letter Display (CLD) for each gene with significant effects</span></span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a><span class="co"># CLD shows which groups are significantly different: groups sharing a letter are NOT significantly different</span></span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>cld_annotations <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a><span class="co"># ATP Synthase</span></span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">exists</span>(<span class="st">"emm_atp"</span>)) {</span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a>  cld_atp <span class="ot">&lt;-</span> <span class="fu">cld</span>(emm_atp, <span class="at">Letters =</span> letters, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb116-9"><a href="#cb116-9" aria-hidden="true" tabindex="-1"></a>  cld_atp_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(cld_atp)</span>
<span id="cb116-10"><a href="#cb116-10" aria-hidden="true" tabindex="-1"></a>  cld_atp_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="st">"ATP.Synthase"</span></span>
<span id="cb116-11"><a href="#cb116-11" aria-hidden="true" tabindex="-1"></a>  cld_atp_df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">trimws</span>(cld_atp_df<span class="sc">$</span>.group)</span>
<span id="cb116-12"><a href="#cb116-12" aria-hidden="true" tabindex="-1"></a>  cld_atp_df <span class="ot">&lt;-</span> cld_atp_df[, <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"polyIC"</span>, <span class="st">"stress_type"</span>, <span class="st">"label"</span>)]</span>
<span id="cb116-13"><a href="#cb116-13" aria-hidden="true" tabindex="-1"></a>  cld_annotations <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cld_annotations, cld_atp_df)</span>
<span id="cb116-14"><a href="#cb116-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb116-15"><a href="#cb116-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-16"><a href="#cb116-16" aria-hidden="true" tabindex="-1"></a><span class="co"># HSP70</span></span>
<span id="cb116-17"><a href="#cb116-17" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">exists</span>(<span class="st">"emm_hsp70"</span>)) {</span>
<span id="cb116-18"><a href="#cb116-18" aria-hidden="true" tabindex="-1"></a>  cld_hsp70 <span class="ot">&lt;-</span> <span class="fu">cld</span>(emm_hsp70, <span class="at">Letters =</span> letters, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb116-19"><a href="#cb116-19" aria-hidden="true" tabindex="-1"></a>  cld_hsp70_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(cld_hsp70)</span>
<span id="cb116-20"><a href="#cb116-20" aria-hidden="true" tabindex="-1"></a>  cld_hsp70_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="st">"HSP70"</span></span>
<span id="cb116-21"><a href="#cb116-21" aria-hidden="true" tabindex="-1"></a>  cld_hsp70_df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">trimws</span>(cld_hsp70_df<span class="sc">$</span>.group)</span>
<span id="cb116-22"><a href="#cb116-22" aria-hidden="true" tabindex="-1"></a>  cld_hsp70_df <span class="ot">&lt;-</span> cld_hsp70_df[, <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"polyIC"</span>, <span class="st">"stress_type"</span>, <span class="st">"label"</span>)]</span>
<span id="cb116-23"><a href="#cb116-23" aria-hidden="true" tabindex="-1"></a>  cld_annotations <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cld_annotations, cld_hsp70_df)</span>
<span id="cb116-24"><a href="#cb116-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb116-25"><a href="#cb116-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-26"><a href="#cb116-26" aria-hidden="true" tabindex="-1"></a><span class="co"># HSP90</span></span>
<span id="cb116-27"><a href="#cb116-27" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">exists</span>(<span class="st">"emm_hsp90"</span>)) {</span>
<span id="cb116-28"><a href="#cb116-28" aria-hidden="true" tabindex="-1"></a>  cld_hsp90 <span class="ot">&lt;-</span> <span class="fu">cld</span>(emm_hsp90, <span class="at">Letters =</span> letters, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb116-29"><a href="#cb116-29" aria-hidden="true" tabindex="-1"></a>  cld_hsp90_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(cld_hsp90)</span>
<span id="cb116-30"><a href="#cb116-30" aria-hidden="true" tabindex="-1"></a>  cld_hsp90_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="st">"HSP90"</span></span>
<span id="cb116-31"><a href="#cb116-31" aria-hidden="true" tabindex="-1"></a>  cld_hsp90_df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">trimws</span>(cld_hsp90_df<span class="sc">$</span>.group)</span>
<span id="cb116-32"><a href="#cb116-32" aria-hidden="true" tabindex="-1"></a>  cld_hsp90_df <span class="ot">&lt;-</span> cld_hsp90_df[, <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"polyIC"</span>, <span class="st">"stress_type"</span>, <span class="st">"label"</span>)]</span>
<span id="cb116-33"><a href="#cb116-33" aria-hidden="true" tabindex="-1"></a>  cld_annotations <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cld_annotations, cld_hsp90_df)</span>
<span id="cb116-34"><a href="#cb116-34" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb116-35"><a href="#cb116-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-36"><a href="#cb116-36" aria-hidden="true" tabindex="-1"></a><span class="co"># VIPERIN</span></span>
<span id="cb116-37"><a href="#cb116-37" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">exists</span>(<span class="st">"emm_viperin"</span>)) {</span>
<span id="cb116-38"><a href="#cb116-38" aria-hidden="true" tabindex="-1"></a>  cld_viperin <span class="ot">&lt;-</span> <span class="fu">cld</span>(emm_viperin, <span class="at">Letters =</span> letters, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb116-39"><a href="#cb116-39" aria-hidden="true" tabindex="-1"></a>  cld_viperin_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(cld_viperin)</span>
<span id="cb116-40"><a href="#cb116-40" aria-hidden="true" tabindex="-1"></a>  cld_viperin_df<span class="sc">$</span>Target <span class="ot">&lt;-</span> <span class="st">"VIPERIN"</span></span>
<span id="cb116-41"><a href="#cb116-41" aria-hidden="true" tabindex="-1"></a>  cld_viperin_df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">trimws</span>(cld_viperin_df<span class="sc">$</span>.group)</span>
<span id="cb116-42"><a href="#cb116-42" aria-hidden="true" tabindex="-1"></a>  cld_viperin_df <span class="ot">&lt;-</span> cld_viperin_df[, <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"polyIC"</span>, <span class="st">"stress_type"</span>, <span class="st">"label"</span>)]</span>
<span id="cb116-43"><a href="#cb116-43" aria-hidden="true" tabindex="-1"></a>  cld_annotations <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cld_annotations, cld_viperin_df)</span>
<span id="cb116-44"><a href="#cb116-44" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb116-45"><a href="#cb116-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-46"><a href="#cb116-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate y positions for annotations (above max values)</span></span>
<span id="cb116-47"><a href="#cb116-47" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(cld_annotations) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb116-48"><a href="#cb116-48" aria-hidden="true" tabindex="-1"></a>  y_positions <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb116-49"><a href="#cb116-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Target, stress_type, polyIC) <span class="sc">%&gt;%</span></span>
<span id="cb116-50"><a href="#cb116-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">max_y =</span> <span class="fu">max</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb116-51"><a href="#cb116-51" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb116-52"><a href="#cb116-52" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate data range for each gene to use proportional offset</span></span>
<span id="cb116-53"><a href="#cb116-53" aria-hidden="true" tabindex="-1"></a>  y_range_per_gene <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb116-54"><a href="#cb116-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Target) <span class="sc">%&gt;%</span></span>
<span id="cb116-55"><a href="#cb116-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">y_range =</span> <span class="fu">max</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">-</span> <span class="fu">min</span>(delta_Cq, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb116-56"><a href="#cb116-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb116-57"><a href="#cb116-57" aria-hidden="true" tabindex="-1"></a>  cld_annotations <span class="ot">&lt;-</span> cld_annotations <span class="sc">%&gt;%</span></span>
<span id="cb116-58"><a href="#cb116-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(y_positions, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Target"</span>, <span class="st">"stress_type"</span>, <span class="st">"polyIC"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb116-59"><a href="#cb116-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(y_range_per_gene, <span class="at">by =</span> <span class="st">"Target"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb116-60"><a href="#cb116-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">y_pos =</span> max_y <span class="sc">+</span> y_range <span class="sc">*</span> <span class="fl">0.08</span>)  <span class="co"># 8% of data range above max</span></span>
<span id="cb116-61"><a href="#cb116-61" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb116-62"><a href="#cb116-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-63"><a href="#cb116-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Create plot</span></span>
<span id="cb116-64"><a href="#cb116-64" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> delta_Cq_df <span class="sc">%&gt;%</span></span>
<span id="cb116-65"><a href="#cb116-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> stress_type, <span class="at">y =</span> delta_Cq, <span class="at">fill =</span> polyIC)) <span class="sc">+</span></span>
<span id="cb116-66"><a href="#cb116-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.8</span>), <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb116-67"><a href="#cb116-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.8</span>), <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb116-68"><a href="#cb116-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Target, <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb116-69"><a href="#cb116-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"No_PolyIC"</span> <span class="ot">=</span> <span class="st">"darkgray"</span>, <span class="st">"PolyIC"</span> <span class="ot">=</span> <span class="st">"salmon"</span>)) <span class="sc">+</span></span>
<span id="cb116-70"><a href="#cb116-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.15</span>))) <span class="sc">+</span>  <span class="co"># Extra space at top for annotations</span></span>
<span id="cb116-71"><a href="#cb116-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb116-72"><a href="#cb116-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb116-73"><a href="#cb116-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb116-74"><a href="#cb116-74" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Delta Cq by Treatment Group with ANOVA Significance"</span>,</span>
<span id="cb116-75"><a href="#cb116-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Stress Type"</span>,</span>
<span id="cb116-76"><a href="#cb116-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Delta Cq"</span>,</span>
<span id="cb116-77"><a href="#cb116-77" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"PolyIC Treatment"</span></span>
<span id="cb116-78"><a href="#cb116-78" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb116-79"><a href="#cb116-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb116-80"><a href="#cb116-80" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb116-81"><a href="#cb116-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb116-82"><a href="#cb116-82" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb116-83"><a href="#cb116-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-84"><a href="#cb116-84" aria-hidden="true" tabindex="-1"></a><span class="co"># Add CLD annotations if any exist</span></span>
<span id="cb116-85"><a href="#cb116-85" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">nrow</span>(cld_annotations) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb116-86"><a href="#cb116-86" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> p <span class="sc">+</span> </span>
<span id="cb116-87"><a href="#cb116-87" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> cld_annotations, </span>
<span id="cb116-88"><a href="#cb116-88" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> stress_type, <span class="at">y =</span> y_pos, <span class="at">label =</span> label, <span class="at">group =</span> polyIC),</span>
<span id="cb116-89"><a href="#cb116-89" aria-hidden="true" tabindex="-1"></a>              <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.8</span>), <span class="at">size =</span> <span class="dv">4</span>, <span class="at">vjust =</span> <span class="dv">0</span>, </span>
<span id="cb116-90"><a href="#cb116-90" aria-hidden="true" tabindex="-1"></a>              <span class="at">color =</span> <span class="st">"darkorange"</span>, <span class="at">fontface =</span> <span class="st">"bold"</span>) <span class="sc">+</span></span>
<span id="cb116-91"><a href="#cb116-91" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">"Letters show statistical groupings (p ≤ 0.05). Groups sharing a letter are NOT significantly different from each other"</span>,</span>
<span id="cb116-92"><a href="#cb116-92" aria-hidden="true" tabindex="-1"></a>         <span class="at">caption =</span> <span class="st">"Compact Letter Display shown for: ATP Synthase, HSP70, HSP90, and VIPERIN. See individual ANOVA sections for detailed p-values."</span>)</span>
<span id="cb116-93"><a href="#cb116-93" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb116-94"><a href="#cb116-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-95"><a href="#cb116-95" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/anova-summary-plots-1.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="summary" class="level1">
<h1>SUMMARY</h1>
<p>This analysis examined gene expression responses in <em>Magallana gigas</em> to PolyIC treatment (an immune stimulant mimicking viral dsRNA) and different stress types (Temperature, Mechanical, Control) using qPCR. Eight genes were analyzed: ATP Synthase, Citrate Synthase, DNMT1, HSP70, HSP90, cGAS, and VIPERIN. Expression levels were normalized to GAPDH and analyzed using two-way ANOVA (PolyIC × Stress Type) with post-hoc pairwise comparisons.</p>
<section id="gene-specific-results" class="level2">
<h2 class="anchored" data-anchor-id="gene-specific-results">Gene-Specific Results</h2>
<section id="atp-synthase-1" class="level3">
<h3 class="anchored" data-anchor-id="atp-synthase-1">ATP Synthase</h3>
<p><strong>Function:</strong> Mitochondrial enzyme synthesizing ATP; marker of energy metabolism.</p>
<p><strong>ANOVA Results:</strong> - <strong>PolyIC effect:</strong> SIGNIFICANT (p = 7.33e-06) *<strong> - </strong>Stress type effect:** SIGNIFICANT (p = 0.04) * - <strong>PolyIC × Stress interaction:</strong> Not significant (p = 0.47)</p>
<p><strong>Assumption Checks:</strong> - <strong>Levene’s Test:</strong> FAILED (p = 0.0036) - variances are not homogeneous across groups ⚠️</p>
<p><strong>Significant Pairwise Contrasts (p ≤ 0.05):</strong> - No_PolyIC Control vs PolyIC Mechanical (p = 0.007) - No_PolyIC Control vs PolyIC Temperature (p = 0.0001) - No_PolyIC Mechanical vs PolyIC Temperature (p = 0.0195) - No_PolyIC Temperature vs PolyIC Temperature (p = 0.0158)</p>
<p><strong>Key Findings:</strong> - PolyIC treatment significantly affects ATP Synthase expression - The “stress type effect” is driven primarily by <strong>PolyIC-treated samples under stress</strong> (all significant contrasts involve PolyIC groups) - No significant differences between non-PolyIC control and non-PolyIC stressed groups - Effects are additive rather than interactive (no interaction between PolyIC and stress) - PolyIC + Temperature shows the strongest effect (lowest p-value), followed by PolyIC + Mechanical - <strong>Caveat:</strong> Heterogeneous variances suggest results should be interpreted cautiously</p>
</section>
<section id="citrate-synthase-1" class="level3">
<h3 class="anchored" data-anchor-id="citrate-synthase-1">Citrate Synthase</h3>
<p><strong>Function:</strong> TCA cycle enzyme; marker of mitochondrial density and oxidative capacity.</p>
<p><strong>ANOVA Results:</strong> - <strong>PolyIC effect:</strong> Not significant (p = 0.693) - <strong>Stress type effect:</strong> Marginally non-significant (p = 0.0688) - <strong>PolyIC × Stress interaction:</strong> Not significant (p = 0.464)</p>
<p><strong>Assumption Checks:</strong> - <strong>Levene’s Test:</strong> PASSED (p = 0.956) ✓</p>
<p><strong>Key Findings:</strong> - Citrate Synthase expression is stable across PolyIC treatments - Stress type shows a trend toward significance (p = 0.0688) but does not reach the p ≤ 0.05 threshold - No significant pairwise contrasts detected (all p &gt; 0.05) - This near-significant trend without specific contrasts suggests subtle, distributed differences across stress types rather than a strong specific effect - Mitochondrial density/function appears relatively unaffected by these experimental conditions</p>
</section>
<section id="dnmt1-1" class="level3">
<h3 class="anchored" data-anchor-id="dnmt1-1">DNMT1</h3>
<p><strong>Function:</strong> DNA methyltransferase enzyme; maintains DNA methylation patterns and epigenetic regulation.</p>
<p><strong>ANOVA Results:</strong> - <strong>PolyIC effect:</strong> Not significant (p = 0.193) - <strong>Stress type effect:</strong> Not significant (p = 0.179) - <strong>PolyIC × Stress interaction:</strong> Not significant (p = 0.399)</p>
<p><strong>Assumption Checks:</strong> - <strong>Levene’s Test:</strong> PASSED (p = 0.617) ✓</p>
<p><strong>Key Findings:</strong> - DNMT1 expression does not respond significantly to PolyIC or stress treatments - DNA methylation machinery appears constitutively expressed under these conditions - No evidence of stress-induced epigenetic remodeling at the transcriptional level</p>
</section>
<section id="hsp70-1" class="level3">
<h3 class="anchored" data-anchor-id="hsp70-1">HSP70</h3>
<p><strong>Function:</strong> Molecular chaperone; critical for protein folding and cellular stress response.</p>
<p><strong>ANOVA Results:</strong> - <strong>PolyIC effect:</strong> SIGNIFICANT (p = 5.7e-13) <em><strong> - </strong>Stress type effect:<strong> SIGNIFICANT (p &lt; 2e-16) </strong></em> - <strong>PolyIC × Stress interaction:</strong> Not significant (p = 0.831)</p>
<p><strong>Assumption Checks:</strong> - <strong>Levene’s Test:</strong> PASSED (p = 0.745) ✓</p>
<p><strong>Significant Pairwise Contrasts (p ≤ 0.05):</strong> - No_PolyIC Control vs No_PolyIC Temperature (p &lt; 0.0001) - No_PolyIC Control vs PolyIC Temperature (p &lt; 0.0001) - PolyIC Control vs No_PolyIC Temperature (p &lt; 0.0001) - PolyIC Control vs PolyIC Temperature (p &lt; 0.0001) - No_PolyIC Mechanical vs No_PolyIC Temperature (p &lt; 0.0001) - No_PolyIC Mechanical vs PolyIC Temperature (p &lt; 0.0001) - PolyIC Mechanical vs No_PolyIC Temperature (p = 0.0017) - PolyIC Mechanical vs PolyIC Temperature (p &lt; 0.0001)</p>
<p><strong>Key Findings:</strong> - HSP70 shows extremely strong responses to both PolyIC and stress type - <strong>Temperature stress is the primary driver:</strong> All 8 significant contrasts involve temperature-stressed groups vs.&nbsp;non-temperature groups - Both PolyIC and non-PolyIC oysters show dramatic HSP70 upregulation under temperature stress compared to controls or mechanical stress - Mechanical stress does NOT significantly induce HSP70 (no significant Control vs Mechanical contrasts) - No interaction suggests PolyIC and temperature stress act independently (additive effects) - This is the classic heat shock protein response to thermal stress</p>
</section>
<section id="hsp90-1" class="level3">
<h3 class="anchored" data-anchor-id="hsp90-1">HSP90</h3>
<p><strong>Function:</strong> Molecular chaperone; regulates protein maturation and signaling networks.</p>
<p><strong>ANOVA Results:</strong> - <strong>PolyIC effect:</strong> SIGNIFICANT (p = 5.47e-10) *<strong> - </strong>Stress type effect:<strong> Not significant (p = 0.476) - </strong>PolyIC × Stress interaction:** Not significant (p = 0.562)</p>
<p><strong>Assumption Checks:</strong> - <strong>Levene’s Test:</strong> PASSED (p = 0.942) ✓</p>
<p><strong>Significant Pairwise Contrasts (p ≤ 0.05):</strong> - No_PolyIC Control vs PolyIC Temperature (p &lt; 0.0001) - No_PolyIC Mechanical vs PolyIC Temperature (p &lt; 0.0001) - No_PolyIC Temperature vs PolyIC Temperature (p &lt; 0.0001)</p>
<p><strong>Key Findings:</strong> - PolyIC has a highly significant main effect on HSP90 expression - Unlike HSP70, the overall stress type main effect is not significant - <strong>All three significant contrasts specifically involve PolyIC + Temperature:</strong> - PolyIC Temperature differs from all three non-PolyIC groups (Control, Mechanical, Temperature) - No other treatment combinations show significant differences - The pattern shows <strong>PolyIC + Temperature creates a synergistic-like response</strong> despite the non-significant interaction term (possibly due to sample size/power) - PolyIC alone (control), temperature alone, and mechanical stress (with/without PolyIC) do NOT significantly induce HSP90 - HSP90 appears to require the combination of immune stimulation (PolyIC) AND thermal stress</p>
</section>
<section id="cgas-1" class="level3">
<h3 class="anchored" data-anchor-id="cgas-1">cGAS</h3>
<p><strong>Function:</strong> Cytosolic DNA sensor; involved in innate immune response and dsRNA detection.</p>
<p><strong>ANOVA Results:</strong> - <strong>PolyIC effect:</strong> Not significant (p = 0.749) - <strong>Stress type effect:</strong> Not significant (p = 0.168) - <strong>PolyIC × Stress interaction:</strong> Not significant (p = 0.831)</p>
<p><strong>Assumption Checks:</strong> - <strong>Levene’s Test:</strong> PASSED (p = 0.564) ✓</p>
<p><strong>Key Findings:</strong> - cGAS does not respond significantly to PolyIC treatment - No significant response to stress type either - This is unexpected given cGAS’s role in detecting foreign nucleic acids - May suggest: (1) different immune sensing pathways activated, (2) timing issues (response may occur earlier/later), or (3) PolyIC recognition occurs through alternative receptors in oysters</p>
</section>
<section id="viperin-1" class="level3">
<h3 class="anchored" data-anchor-id="viperin-1">VIPERIN</h3>
<p><strong>Function:</strong> Antiviral effector protein; restricts replication of DNA and RNA viruses.</p>
<p><strong>ANOVA Results:</strong> - <strong>PolyIC effect:</strong> SIGNIFICANT (p = 0.000165) *<strong> - </strong>Stress type effect:<strong> Not significant (p = 0.548) - </strong>PolyIC × Stress interaction:** Not significant (p = 0.505)</p>
<p><strong>Assumption Checks:</strong> - <strong>Levene’s Test:</strong> PASSED (p = 0.690) ✓</p>
<p><strong>Significant Pairwise Contrasts (p ≤ 0.05):</strong> - No_PolyIC Control vs PolyIC Control (p = 0.0242)</p>
<p><strong>Key Findings:</strong> - VIPERIN shows a clear response to PolyIC treatment as an immune stimulant - The only significant contrast is <strong>No_PolyIC Control vs PolyIC Control</strong> (p = 0.024) - <strong>PolyIC alone is sufficient to induce VIPERIN expression</strong> - no additional stress needed - Notably, PolyIC + stress combinations (mechanical/temperature) are NOT significantly different from PolyIC control - This indicates the VIPERIN response is triggered by PolyIC recognition itself, independent of other stressors - Stress type (temperature/mechanical) does not enhance or suppress the PolyIC-induced VIPERIN response - As an antiviral effector, VIPERIN appropriately recognizes PolyIC as a viral mimic</p>
</section>
</section>
</section>
<section id="r-session-info" class="level1">
<h1>R SESSION INFO</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.5.2 (2025-10-31)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 24.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.12.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.12.0  LAPACK version 3.12.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/Los_Angeles
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] multcomp_1.4-29 TH.data_1.1-5   MASS_7.3-65     survival_3.8-6 
 [5] mvtnorm_1.3-3   emmeans_2.0.1   car_3.1-3       carData_3.0-6  
 [9] lubridate_1.9.4 forcats_1.0.0   stringr_1.5.1   dplyr_1.1.4    
[13] purrr_1.0.4     readr_2.1.5     tidyr_1.3.1     tibble_3.2.1   
[17] ggplot2_3.5.2   tidyverse_2.0.0

loaded via a namespace (and not attached):
 [1] utf8_1.2.4          sandwich_3.1-1      generics_0.1.3     
 [4] stringi_1.8.7       lattice_0.22-7      hms_1.1.3          
 [7] digest_0.6.37       magrittr_2.0.3      evaluate_1.0.3     
[10] grid_4.5.2          timechange_0.3.0    estimability_1.5.1 
[13] RColorBrewer_1.1-3  fastmap_1.2.0       Matrix_1.7-4       
[16] jsonlite_2.0.0      Formula_1.2-5       scales_1.4.0       
[19] codetools_0.2-20    abind_1.4-8         cli_3.6.5          
[22] rlang_1.1.6         splines_4.5.2       withr_3.0.2        
[25] yaml_2.3.10         multcompView_0.1-10 tools_4.5.2        
[28] tzdb_0.5.0          vctrs_0.6.5         R6_2.6.1           
[31] zoo_1.8-15          lifecycle_1.0.4     htmlwidgets_1.6.4  
[34] pkgconfig_2.0.3     pillar_1.10.2       gtable_0.3.6       
[37] glue_1.8.0          xfun_0.52           tidyselect_1.2.1   
[40] knitr_1.50          xtable_1.8-4        farver_2.1.2       
[43] htmltools_0.5.8.1   labeling_0.4.3      rmarkdown_2.29     
[46] compiler_4.5.2     </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/robertslab\.github\.io\/sams-notebook");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>