<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sam White">
<meta name="dcterms.date" content="2025-10-08">

<title>Data Exploration - E5 Timeseries Multi-species Multi-omics Using Barnacle – Sam’s Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark-b53751a350365c71b6c909e95f209ed1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-e518e50a9807db09b51ea894c4c266f8.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 2,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 100,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-7483T61T11"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-7483T61T11', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Sam’s Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" target="_blank">
 <span class="dropdown-text">Rstudio Server - Raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu/Atumefaciens/" target="_blank">
 <span class="dropdown-text">My Gannet Directory</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" target="_blank">
 <span class="dropdown-text">Steven’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shedurkin.github.io/Roberts-LabNotebook/" target="_blank">
 <span class="dropdown-text">Kathleen’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://meganewing.github.io/mewing-notebook/" target="_blank">
 <span class="dropdown-text">Megan’s Notebook</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/sams-notebook" target="_blank">
 <span class="dropdown-text">Notebook repo</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" target="_blank">
 <span class="dropdown-text">Lab Organization (GitHub)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" target="_blank">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" target="_blank"> <i class="bi bi-calendar-week" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-right">
      <h1 class="title">Data Exploration - E5 Timeseries Multi-species Multi-omics Using Barnacle</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2025</div>
                <div class="quarto-category">barnacle</div>
                <div class="quarto-category">Acropora pulchra</div>
                <div class="quarto-category">Porites evermanni</div>
                <div class="quarto-category">Pocillopora tuahiniensis</div>
                <div class="quarto-category">E5</div>
                <div class="quarto-category">coral</div>
                <div class="quarto-category">timeseries_molecular</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-right">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Sam White </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">October 8, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#intro" id="toc-intro" class="nav-link active" data-scroll-target="#intro">INTRO</a></li>
  <li><a href="#background" id="toc-background" class="nav-link" data-scroll-target="#background">1 BACKGROUND</a>
  <ul>
  <li><a href="#input-files" id="toc-input-files" class="nav-link" data-scroll-target="#input-files">1.1 Input files</a></li>
  <li><a href="#output-files" id="toc-output-files" class="nav-link" data-scroll-target="#output-files">1.2 Output files</a></li>
  <li><a href="#notes-assumptions" id="toc-notes-assumptions" class="nav-link" data-scroll-target="#notes-assumptions">1.3 Notes / assumptions:</a></li>
  </ul></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup">2 SETUP</a>
  <ul>
  <li><a href="#libraries" id="toc-libraries" class="nav-link" data-scroll-target="#libraries">2.1 Libraries</a></li>
  <li><a href="#set-r-variables" id="toc-set-r-variables" class="nav-link" data-scroll-target="#set-r-variables">2.2 Set R variables</a></li>
  <li><a href="#load-barnacle-conda-environment" id="toc-load-barnacle-conda-environment" class="nav-link" data-scroll-target="#load-barnacle-conda-environment">2.3 Load barnacle conda environment</a></li>
  </ul></li>
  <li><a href="#data-prep" id="toc-data-prep" class="nav-link" data-scroll-target="#data-prep">3 DATA PREP</a>
  <ul>
  <li><a href="#load-ortholog-groups-data" id="toc-load-ortholog-groups-data" class="nav-link" data-scroll-target="#load-ortholog-groups-data">3.1 Load ortholog groups data</a></li>
  <li><a href="#extract-ortholog-expression-data" id="toc-extract-ortholog-expression-data" class="nav-link" data-scroll-target="#extract-ortholog-expression-data">3.2 Extract ortholog expression data</a>
  <ul>
  <li><a href="#load-gene-count-matrices" id="toc-load-gene-count-matrices" class="nav-link" data-scroll-target="#load-gene-count-matrices">3.2.1 Load gene count matrices</a></li>
  <li><a href="#filter-ortholog-groups-for-complete-three-way-matches" id="toc-filter-ortholog-groups-for-complete-three-way-matches" class="nav-link" data-scroll-target="#filter-ortholog-groups-for-complete-three-way-matches">3.2.2 Filter ortholog groups for complete three-way matches</a></li>
  <li><a href="#filter-for-expression-data-availability" id="toc-filter-for-expression-data-availability" class="nav-link" data-scroll-target="#filter-for-expression-data-availability">3.2.3 Filter for expression data availability</a></li>
  <li><a href="#gene-id-cleaning-examples" id="toc-gene-id-cleaning-examples" class="nav-link" data-scroll-target="#gene-id-cleaning-examples">3.2.4 Gene ID cleaning examples</a></li>
  <li><a href="#clean-gene-ids-for-matching" id="toc-clean-gene-ids-for-matching" class="nav-link" data-scroll-target="#clean-gene-ids-for-matching">3.2.5 Clean gene IDs for matching</a></li>
  <li><a href="#create-ortholog-expression-data-with-group_id-mapping" id="toc-create-ortholog-expression-data-with-group_id-mapping" class="nav-link" data-scroll-target="#create-ortholog-expression-data-with-group_id-mapping">3.2.6 Create ortholog expression data with group_id mapping</a></li>
  <li><a href="#write-ortholog-expression-data" id="toc-write-ortholog-expression-data" class="nav-link" data-scroll-target="#write-ortholog-expression-data">3.2.7 Write ortholog expression data</a></li>
  <li><a href="#column-structure-analysis" id="toc-column-structure-analysis" class="nav-link" data-scroll-target="#column-structure-analysis">3.2.8 Column structure analysis</a></li>
  <li><a href="#summary-statistics" id="toc-summary-statistics" class="nav-link" data-scroll-target="#summary-statistics">3.2.9 Summary statistics</a></li>
  <li><a href="#diagnostic-analysis" id="toc-diagnostic-analysis" class="nav-link" data-scroll-target="#diagnostic-analysis">3.2.10 Diagnostic analysis</a></li>
  <li><a href="#preview-expression-data" id="toc-preview-expression-data" class="nav-link" data-scroll-target="#preview-expression-data">3.2.11 Preview expression data</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#barnacle-analysis" id="toc-barnacle-analysis" class="nav-link" data-scroll-target="#barnacle-analysis">4 BARNACLE ANALYSIS</a>
  <ul>
  <li><a href="#load-expression-data" id="toc-load-expression-data" class="nav-link" data-scroll-target="#load-expression-data">4.1 Load expression data</a></li>
  <li><a href="#normalize-data-with-sctransform" id="toc-normalize-data-with-sctransform" class="nav-link" data-scroll-target="#normalize-data-with-sctransform">4.2 Normalize data with sctransform</a></li>
  <li><a href="#export-normalized-data-for-python-analysis" id="toc-export-normalized-data-for-python-analysis" class="nav-link" data-scroll-target="#export-normalized-data-for-python-analysis">4.3 Export normalized data for Python analysis</a></li>
  <li><a href="#create-tensor-dataset-in-python" id="toc-create-tensor-dataset-in-python" class="nav-link" data-scroll-target="#create-tensor-dataset-in-python">4.4 Create tensor dataset in Python</a></li>
  <li><a href="#parse-sample-information" id="toc-parse-sample-information" class="nav-link" data-scroll-target="#parse-sample-information">4.5 Parse sample information</a></li>
  <li><a href="#create-3d-tensor-genes-species_samples-timepoints" id="toc-create-3d-tensor-genes-species_samples-timepoints" class="nav-link" data-scroll-target="#create-3d-tensor-genes-species_samples-timepoints">4.6 Create 3D tensor (genes × species_samples × timepoints)</a></li>
  <li><a href="#run-barnacle-sparse-tensor-decomposition" id="toc-run-barnacle-sparse-tensor-decomposition" class="nav-link" data-scroll-target="#run-barnacle-sparse-tensor-decomposition">4.7 Run Barnacle sparse tensor decomposition</a></li>
  <li><a href="#analyze-and-interpret-results" id="toc-analyze-and-interpret-results" class="nav-link" data-scroll-target="#analyze-and-interpret-results">4.8 Analyze and interpret results</a></li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary">4.9 Summary</a></li>
  <li><a href="#save-results" id="toc-save-results" class="nav-link" data-scroll-target="#save-results">4.10 Save results</a></li>
  <li><a href="#visualize-barnacle-results" id="toc-visualize-barnacle-results" class="nav-link" data-scroll-target="#visualize-barnacle-results">4.11 Visualize Barnacle results</a>
  <ul>
  <li><a href="#visualization-setup" id="toc-visualization-setup" class="nav-link" data-scroll-target="#visualization-setup">4.11.1 Visualization setup</a></li>
  <li><a href="#component-weights" id="toc-component-weights" class="nav-link" data-scroll-target="#component-weights">4.11.2 Component weights</a></li>
  <li><a href="#timepoint-loadings" id="toc-timepoint-loadings" class="nav-link" data-scroll-target="#timepoint-loadings">4.11.3 Timepoint loadings</a></li>
  <li><a href="#sample-factor-heatmap" id="toc-sample-factor-heatmap" class="nav-link" data-scroll-target="#sample-factor-heatmap">4.11.4 Sample-factor heatmap</a></li>
  <li><a href="#pca-of-sample-factor-loadings" id="toc-pca-of-sample-factor-loadings" class="nav-link" data-scroll-target="#pca-of-sample-factor-loadings">4.11.5 PCA of sample-factor loadings</a></li>
  <li><a href="#top-orthologs-per-component" id="toc-top-orthologs-per-component" class="nav-link" data-scroll-target="#top-orthologs-per-component">4.11.6 Top orthologs per component</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#system-info" id="toc-system-info" class="nav-link" data-scroll-target="#system-info">5 SYSTEM INFO</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">6 REFERENCES</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-right" id="quarto-document-content">





<section id="intro" class="level1">
<h1>INTRO</h1>
<p>This notebook is an introductory exploration of the E5 coral <a href="https://github.com/urol-e5/timeseries_molecular"><code>urol-e5/timeseries_molecular</code></a> (GitHub repo) multi-species data using a tensor decomposition approach, using <a href="https://github.com/blasks/barnacle">barnacle</a> (GitHub Repo). Of note, using <a href="https://github.com/blasks/barnacle">barnacle</a> (GitHub Repo) for this multi-omics analysis greatly differs than <a href="../../../posts/2025/2025-09-19-Data-Exploration---E5-Timeseries-A.pulchra-Multi-omics-Using-MOFA2/index.html">the analysis I messed around with using MOFA2</a> (notebook entry).</p>
<p><a href="https://github.com/blasks/barnacle">Barnacle</a> (GitHub Repo) utilizes tensor decomposition. As such, the analysis is constrained, due to the fact that the tensors must have the same dimensions as each other. <a href="https://biofam.github.io/MOFA2/index.html">MOFA2</a> is not limited by data dimensionality. So, for this barnacle analysis, we’re only using gene expression as a metric, with the goal being to identify genes/components (factors) which might be associated with components associated with some characterized physiology. Then, we can potentially look at lncRNA/miRNA/methylation in those contexts.</p>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Warning
</div>
</div>
<div class="callout-body-container callout-body">
<p>The contents below are out of date! Due to lack of convergence (a requirement of tensor decomposition to ensure data stability), I experimented with a variety of parameters (e.g.&nbsp;10,000 iterations, 100 components, reduce lambdas, etc. to attempt to achieve convergence; all failed.) and have not rendered as markdown.</p>
<p>Review the updated R Markdown code for changes, including modifications that fixed <code>sctransform</code>:</p>
<p><a href="https://github.com/urol-e5/timeseries_molecular/blob/d88c40082bb7d2902eb847dd2fcad7476cae2928/M-multi-species/scripts/13.00-multiomics-barnacle.Rmd">https://github.com/urol-e5/timeseries_molecular/blob/d88c40082bb7d2902eb847dd2fcad7476cae2928/M-multi-species/scripts/13.00-multiomics-barnacle.Rmd</a> (commit <code>d88c400</code>)</p>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The contents below are from knitted markdown <a href="https://github.com/urol-e5/timeseries_molecular/blob/5e572b67dd86c49154d1627993ac01be81b4b368/M-multi-species/scripts/13.00-multiomics-barnacle.md"><code>13.00-multiomics-barnacle.Rmd</code></a> (commit <code>5e572b6</code>).</p>
</div>
</div>
</section>
<section id="background" class="level1">
<h1>1 BACKGROUND</h1>
<p>This analysis prepares three-species ortholog expression matrices, normalizes the data, constructs a multi‑omics tensor (genes × combined species-samples × timepoints), and runs a sparse tensor decomposition using the <a href="https://github.com/blasks/barnacle">barnacle</a> (Blaskowski et al.&nbsp;2024) workflow to discover shared gene/time/species patterns.</p>
<p>Key steps performed by this script:</p>
<ul>
<li>Load annotated ortholog groups and per-species count matrices.</li>
<li>Filter ortholog groups to retain complete three‑way matches with available expression data.</li>
<li>Extract per‑species expression matrices mapped to ortholog group IDs.</li>
<li>Normalize counts (preferred: <code>sctransform</code>; fallback: log2(CPM + 1)).</li>
<li>Build a 3D tensor combining species/sample and timepoint dimensions.</li>
<li>Run <a href="https://github.com/blasks/barnacle">barnacle</a> (Blaskowski et al.&nbsp;2024) sparse CP decomposition and save factor matrices, metadata and figures.</li>
</ul>
<section id="input-files" class="level2">
<h2 class="anchored" data-anchor-id="input-files">1.1 Input files</h2>
<ul>
<li><p><code>../output/12-ortho-annot/ortholog_groups_annotated.csv</code> : annotated ortholog groups with columns for <code>group_id</code>, <code>apul</code>, <code>peve</code>, <code>ptua</code>, <code>type</code>, etc.</p></li>
<li><p><code>../../D-Apul/output/02.20-D-Apul-RNAseq-alignment-HiSat2/apul-gene_count_matrix.csv</code><br>
Apul gene count matrix (contains <code>gene_id</code> plus sample columns).</p></li>
<li><p><code>../../E-Peve/output/02.20-E-Peve-RNAseq-alignment-HiSat2/peve-gene_count_matrix.csv</code><br>
Peve gene count matrix.</p></li>
<li><p><code>../../F-Ptua/output/02.20-F-Ptua-RNAseq-alignment-HiSat2/ptua-gene_count_matrix.csv</code><br>
Ptua gene count matrix.</p></li>
<li><p>(Also referenced) transcript-level matrices: <code>apul-transcript_count_matrix.csv</code>, <code>peve-transcript_count_matrix.csv</code>, <code>ptua-transcript_count_matrix.csv</code> when needed.</p></li>
</ul>
</section>
<section id="output-files" class="level2">
<h2 class="anchored" data-anchor-id="output-files">1.2 Output files</h2>
<ul>
<li><code>apul_ortholog_expression.csv</code>, <code>peve_ortholog_expression.csv</code>, <code>ptua_ortholog_expression.csv</code> : per‑species expression matrices aligned to ortholog <code>group_id</code>.</li>
<li><code>apul_normalized_expression.csv</code>, <code>peve_normalized_expression.csv</code>, <code>ptua_normalized_expression.csv</code> : normalized expression matrices (sctransform or log2(CPM+1)).</li>
<li><code>multiomics_tensor.npy</code> : saved NumPy array of the 3D tensor used for decomposition (genes × combined_samples × timepoints).</li>
<li><code>barnacle_factors/</code> directory containing:
<ul>
<li><code>gene_factors.csv</code> : gene loadings per component (genes × components).</li>
<li><code>sample_factors.csv</code> : combined sample (species_sample) loadings per component with <code>Species</code> and <code>Sample_ID</code> metadata.</li>
<li><code>time_factors.csv</code> : timepoint loadings per component.</li>
<li><code>component_weights.csv</code> : component weights / importance.</li>
<li><code>sample_mapping.csv</code> : mapping of combined sample indices to species and sample IDs.</li>
<li><code>metadata.json</code> : analysis parameters and tensor metadata (shape, rank, lambdas, convergence, etc.).</li>
<li><code>figures/</code> : generated visualizations (component weights, time loadings, sample heatmap, PCA, top ortholog plots).</li>
</ul></li>
</ul>
</section>
<section id="notes-assumptions" class="level2">
<h2 class="anchored" data-anchor-id="notes-assumptions">1.3 Notes / assumptions:</h2>
<ul>
<li>Sample column names are parsed expecting a dot-separated format with a <code>TP#</code> timepoint token (e.g., <code>ACR.139.TP1</code>).</li>
<li>Apul ortholog IDs in the ortholog table include transcript suffixes (e.g., <code>-T1</code>) which are removed for matching to the gene count matrix.</li>
<li>Missing values in the tensor are handled by substitution (current workflow fills NaNs with zeros before decomposition).</li>
</ul>
</section>
</section>
<section id="setup" class="level1">
<h1>2 SETUP</h1>
<section id="libraries" class="level2">
<h2 class="anchored" data-anchor-id="libraries">2.1 Libraries</h2>
</section>
<section id="set-r-variables" class="level2">
<h2 class="anchored" data-anchor-id="set-r-variables">2.2 Set R variables</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># OUTPUT DIRECTORY</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>output_dir <span class="ot">&lt;-</span> <span class="st">"../output/13.00-multiomics-barnacle"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#INPUT FILE(S)</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>ortholog_groups_file <span class="ot">&lt;-</span> <span class="st">"../output/12-ortho-annot/ortholog_groups_annotated.csv"</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Transcript count matrices for each species</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>apul_transcript_matrix_file <span class="ot">&lt;-</span> <span class="st">"../../D-Apul/output/02.20-D-Apul-RNAseq-alignment-HiSat2/apul-transcript_count_matrix.csv"</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>peve_transcript_matrix_file <span class="ot">&lt;-</span> <span class="st">"../../E-Peve/output/02.20-E-Peve-RNAseq-alignment-HiSat2/peve-transcript_count_matrix.csv"</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>ptua_transcript_matrix_file <span class="ot">&lt;-</span> <span class="st">"../../F-Ptua/output/02.20-F-Ptua-RNAseq-alignment-HiSat2/ptua-transcript_count_matrix.csv"</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co"># CONDA</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>conda_env_name <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"/home/sam/programs/mambaforge/envs/barnacle_py311_env"</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>conda_path <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"/opt/anaconda/anaconda3/bin/conda"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="load-barnacle-conda-environment" class="level2">
<h2 class="anchored" data-anchor-id="load-barnacle-conda-environment">2.3 Load <a href="https://github.com/blasks/barnacle">barnacle</a> conda environment</h2>
<p>If this is successful, the first line of output should show that the Python being used is the one in your <a href="https://github.com/blasks/barnacle">barnacle</a> (Blaskowski et al.&nbsp;2024) conda environment path.</p>
<p>E.g.</p>
<p><code>python:         /home/sam/programs/mambaforge/envs/barnacle_py311_env/bin/python</code></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">use_condaenv</span>(<span class="at">condaenv =</span> conda_env_name, <span class="at">conda =</span> conda_path)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">py_config</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## python:         /home/sam/programs/mambaforge/envs/barnacle_py311_env/bin/python
## libpython:      /home/sam/programs/mambaforge/envs/barnacle_py311_env/lib/libpython3.11.so
## pythonhome:     /home/sam/programs/mambaforge/envs/barnacle_py311_env:/home/sam/programs/mambaforge/envs/barnacle_py311_env
## version:        3.11.13 | packaged by conda-forge | (main, Jun  4 2025, 14:48:23) [GCC 13.3.0]
## numpy:          /home/sam/programs/mambaforge/envs/barnacle_py311_env/lib/python3.11/site-packages/numpy
## numpy_version:  1.26.4
## 
## NOTE: Python version was forced by use_python() function</code></pre>
</section>
</section>
<section id="data-prep" class="level1">
<h1>3 DATA PREP</h1>
<section id="load-ortholog-groups-data" class="level2">
<h2 class="anchored" data-anchor-id="load-ortholog-groups-data">3.1 Load ortholog groups data</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in the ortholog groups data</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>ortholog_groups <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(ortholog_groups_file)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Display basic info about the data</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Dimensions of ortholog groups data:"</span>, <span class="fu">dim</span>(ortholog_groups), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Column names:"</span>, <span class="fu">colnames</span>(ortholog_groups), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ortholog_groups)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ortholog_groups)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Dimensions of ortholog groups data: 18372 23 
## 
## Column names: group_id apul peve ptua type avg_identity query accession id reviewed protein_name organism pident length evalue bitscore title go_ids go_bp go_cc go_mf goslim_ids goslim_names 
## 
##   group_id          apul          peve
## 1 OG_00001 FUN_000185-T1 Peve_00037402
## 2 OG_00002 FUN_000189-T1 Peve_00038462
## 3 OG_00003 FUN_000190-T1 Peve_00038463
## 4 OG_00004 FUN_000191-T1 Peve_00038464
## 5 OG_00005 FUN_000192-T1 Peve_00038466
## 6 OG_00006 FUN_000193-T1 Peve_00038467
##                                            ptua      type avg_identity
## 1 Pocillopora_meandrina_HIv1___RNAseq.g28886.t1 three_way     71.05900
## 2 Pocillopora_meandrina_HIv1___RNAseq.g28888.t3 three_way     68.72583
## 3 Pocillopora_meandrina_HIv1___RNAseq.g28889.t1 three_way     55.19750
## 4 Pocillopora_meandrina_HIv1___RNAseq.g28890.t1 three_way     72.04700
## 5 Pocillopora_meandrina_HIv1___RNAseq.g28893.t1 three_way     81.15867
## 6 Pocillopora_meandrina_HIv1___RNAseq.g28894.t1 three_way     70.21933
##           query accession         id reviewed
## 1 FUN_000185-T1    Q32LQ0 AMPE_BOVIN       NA
## 2 FUN_000189-T1    Q9DCT6                  NA
## 3                                          NA
## 4 FUN_000191-T1    Q95108                  NA
## 5 FUN_000192-T1    Q1LWX3                  NA
## 6 FUN_000193-T1    Q5I0K7                  NA
##                                                                               protein_name
## 1 Glutamyl aminopeptidase (EAP) (EC 3.4.11.7) (Aminopeptidase A) (AP-A) (CD antigen CD249)
## 2                                                                                         
## 3                                                                                         
## 4                                                                                         
## 5                                                                                         
## 6                                                                                         
##              organism pident length   evalue bitscore
## 1 Bos taurus (Bovine)   41.5    881 1.4e-197    691.4
## 2                       38.0    171  4.6e-18     92.8
## 3                         NA     NA       NA       NA
## 4                       44.1    136  3.3e-29    129.4
## 5                       54.3    188  3.5e-53    209.5
## 6                       53.2    158  3.4e-41    169.5
##                                                                                                                    title
## 1                                  sp|Q32LQ0|AMPE_BOVIN Glutamyl aminopeptidase OS=Bos taurus OX=9913 GN=ENPEP PE=2 SV=1
## 2        sp|Q9DCT6|BAP18_MOUSE BPTF-associated chromatin complex component 1 OS=Mus musculus OX=10090 GN=Bacc1 PE=1 SV=1
## 3                                                                                                                       
## 4                               sp|Q95108|THIOM_BOVIN Thioredoxin, mitochondrial OS=Bos taurus OX=9913 GN=TXN2 PE=1 SV=2
## 5            sp|Q1LWX3|NTAQ1_DANRE Protein N-terminal glutamine amidohydrolase OS=Danio rerio OX=7955 GN=ntaq1 PE=2 SV=1
## 6 sp|Q5I0K7|ALG13_RAT UDP-N-acetylglucosamine transferase subunit ALG13 OS=Rattus norvegicus OX=10116 GN=Alg13 PE=1 SV=1
##                                                                                                                                                                   go_ids
## 1 GO:0003081; GO:0004177; GO:0004230; GO:0005615; GO:0005737; GO:0005886; GO:0006508; GO:0008217; GO:0008270; GO:0008283; GO:0016477; GO:0042277; GO:0043171; GO:0070006
## 2                                                                                                                                                                       
## 3                                                                                                                                                                       
## 4                                                                                                                                                                       
## 5                                                                                                                                                                       
## 6                                                                                                                                                                       
##                                                                                                                                                                                                                                                                    go_bp
## 1 cell migration [GO:0016477]; cell population proliferation [GO:0008283]; peptide catabolic process [GO:0043171]; proteolysis [GO:0006508]; regulation of blood pressure [GO:0008217]; regulation of systemic arterial blood pressure by renin-angiotensin [GO:0003081]
## 2                                                                                                                                                                                                                                                                       
## 3                                                                                                                                                                                                                                                                       
## 4                                                                                                                                                                                                                                                                       
## 5                                                                                                                                                                                                                                                                       
## 6                                                                                                                                                                                                                                                                       
##                                                                                    go_cc
## 1 cytoplasm [GO:0005737]; extracellular space [GO:0005615]; plasma membrane [GO:0005886]
## 2                                                                                       
## 3                                                                                       
## 4                                                                                       
## 5                                                                                       
## 6                                                                                       
##                                                                                                                                                                                           go_mf
## 1 aminopeptidase activity [GO:0004177]; glutamyl aminopeptidase activity [GO:0004230]; metalloaminopeptidase activity [GO:0070006]; peptide binding [GO:0042277]; zinc ion binding [GO:0008270]
## 2                                                                                                                                                                                              
## 3                                                                                                                                                                                              
## 4                                                                                                                                                                                              
## 5                                                                                                                                                                                              
## 6                                                                                                                                                                                              
##                                                                           goslim_ids
## 1 GO:0003824; GO:0005615; GO:0005886; GO:0016787; GO:0048870; GO:0050886; GO:0140096
## 2                                                                                   
## 3                                                                                   
## 4                                                                                   
## 5                                                                                   
## 6                                                                                   
##                                                                                                                                              goslim_names
## 1 catalytic activity; extracellular space; plasma membrane; hydrolase activity; cell motility; endocrine process; catalytic activity, acting on a protein
## 2                                                                                                                                                        
## 3                                                                                                                                                        
## 4                                                                                                                                                        
## 5                                                                                                                                                        
## 6                                                                                                                                                        
## 'data.frame':    18372 obs. of  23 variables:
##  $ group_id    : chr  "OG_00001" "OG_00002" "OG_00003" "OG_00004" ...
##  $ apul        : chr  "FUN_000185-T1" "FUN_000189-T1" "FUN_000190-T1" "FUN_000191-T1" ...
##  $ peve        : chr  "Peve_00037402" "Peve_00038462" "Peve_00038463" "Peve_00038464" ...
##  $ ptua        : chr  "Pocillopora_meandrina_HIv1___RNAseq.g28886.t1" "Pocillopora_meandrina_HIv1___RNAseq.g28888.t3" "Pocillopora_meandrina_HIv1___RNAseq.g28889.t1" "Pocillopora_meandrina_HIv1___RNAseq.g28890.t1" ...
##  $ type        : chr  "three_way" "three_way" "three_way" "three_way" ...
##  $ avg_identity: num  71.1 68.7 55.2 72 81.2 ...
##  $ query       : chr  "FUN_000185-T1" "FUN_000189-T1" "" "FUN_000191-T1" ...
##  $ accession   : chr  "Q32LQ0" "Q9DCT6" "" "Q95108" ...
##  $ id          : chr  "AMPE_BOVIN" "" "" "" ...
##  $ reviewed    : logi  NA NA NA NA NA NA ...
##  $ protein_name: chr  "Glutamyl aminopeptidase (EAP) (EC 3.4.11.7) (Aminopeptidase A) (AP-A) (CD antigen CD249)" "" "" "" ...
##  $ organism    : chr  "Bos taurus (Bovine)" "" "" "" ...
##  $ pident      : num  41.5 38 NA 44.1 54.3 53.2 60.2 38.5 44.7 NA ...
##  $ length      : num  881 171 NA 136 188 158 447 148 488 NA ...
##  $ evalue      : num  1.4e-197 4.6e-18 NA 3.3e-29 3.5e-53 ...
##  $ bitscore    : num  691.4 92.8 NA 129.4 209.5 ...
##  $ title       : chr  "sp|Q32LQ0|AMPE_BOVIN Glutamyl aminopeptidase OS=Bos taurus OX=9913 GN=ENPEP PE=2 SV=1" "sp|Q9DCT6|BAP18_MOUSE BPTF-associated chromatin complex component 1 OS=Mus musculus OX=10090 GN=Bacc1 PE=1 SV=1" "" "sp|Q95108|THIOM_BOVIN Thioredoxin, mitochondrial OS=Bos taurus OX=9913 GN=TXN2 PE=1 SV=2" ...
##  $ go_ids      : chr  "GO:0003081; GO:0004177; GO:0004230; GO:0005615; GO:0005737; GO:0005886; GO:0006508; GO:0008217; GO:0008270; GO:"| __truncated__ "" "" "" ...
##  $ go_bp       : chr  "cell migration [GO:0016477]; cell population proliferation [GO:0008283]; peptide catabolic process [GO:0043171]"| __truncated__ "" "" "" ...
##  $ go_cc       : chr  "cytoplasm [GO:0005737]; extracellular space [GO:0005615]; plasma membrane [GO:0005886]" "" "" "" ...
##  $ go_mf       : chr  "aminopeptidase activity [GO:0004177]; glutamyl aminopeptidase activity [GO:0004230]; metalloaminopeptidase acti"| __truncated__ "" "" "" ...
##  $ goslim_ids  : chr  "GO:0003824; GO:0005615; GO:0005886; GO:0016787; GO:0048870; GO:0050886; GO:0140096" "" "" "" ...
##  $ goslim_names: chr  "catalytic activity; extracellular space; plasma membrane; hydrolase activity; cell motility; endocrine process;"| __truncated__ "" "" "" ...</code></pre>
</section>
<section id="extract-ortholog-expression-data" class="level2">
<h2 class="anchored" data-anchor-id="extract-ortholog-expression-data">3.2 Extract ortholog expression data</h2>
<p>Now let’s extract expression data for genes that are present in the ortholog groups. We’ll use the gene count matrices with gene IDs to properly map the data.</p>
<section id="load-gene-count-matrices" class="level3">
<h3 class="anchored" data-anchor-id="load-gene-count-matrices">3.2.1 Load gene count matrices</h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define file paths for gene count matrices</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>apul_gene_matrix_file <span class="ot">&lt;-</span> <span class="st">"../../D-Apul/output/02.20-D-Apul-RNAseq-alignment-HiSat2/apul-gene_count_matrix.csv"</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>peve_gene_matrix_file <span class="ot">&lt;-</span> <span class="st">"../../E-Peve/output/02.20-E-Peve-RNAseq-alignment-HiSat2/peve-gene_count_matrix.csv"</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>ptua_gene_matrix_file <span class="ot">&lt;-</span> <span class="st">"../../F-Ptua/output/02.20-F-Ptua-RNAseq-alignment-HiSat2/ptua-gene_count_matrix.csv"</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Load gene count matrices for each species</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Loading gene count matrices...</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>apul_gene_matrix <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(apul_gene_matrix_file)</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Apul gene matrix dimensions:"</span>, <span class="fu">dim</span>(apul_gene_matrix), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>peve_gene_matrix <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(peve_gene_matrix_file)</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Peve gene matrix dimensions:"</span>, <span class="fu">dim</span>(peve_gene_matrix), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>ptua_gene_matrix <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(ptua_gene_matrix_file)</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ptua gene matrix dimensions:"</span>, <span class="fu">dim</span>(ptua_gene_matrix), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Loading gene count matrices...
## 
## Apul gene matrix dimensions: 44371 41 
## Peve gene matrix dimensions: 40389 39 
## Ptua gene matrix dimensions: 31840 40</code></pre>
</section>
<section id="filter-ortholog-groups-for-complete-three-way-matches" class="level3">
<h3 class="anchored" data-anchor-id="filter-ortholog-groups-for-complete-three-way-matches">3.2.2 Filter ortholog groups for complete three-way matches</h3>
<div class="sourceCode" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Filtering for complete three-way ortholog groups...</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only rows where all three species have entries (no NA values or empty strings)</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>complete_ortholog_groups <span class="ot">&lt;-</span> ortholog_groups[<span class="fu">nzchar</span>(ortholog_groups<span class="sc">$</span>apul) <span class="sc">&amp;</span> </span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">nzchar</span>(ortholog_groups<span class="sc">$</span>peve) <span class="sc">&amp;</span> </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">nzchar</span>(ortholog_groups<span class="sc">$</span>ptua), ]</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Total ortholog groups:"</span>, <span class="fu">nrow</span>(ortholog_groups), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Complete three-way ortholog groups:"</span>, <span class="fu">nrow</span>(complete_ortholog_groups), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Filtering for complete three-way ortholog groups...
## Total ortholog groups: 18372 
## Complete three-way ortholog groups: 10381</code></pre>
</section>
<section id="filter-for-expression-data-availability" class="level3">
<h3 class="anchored" data-anchor-id="filter-for-expression-data-availability">3.2.3 Filter for expression data availability</h3>
<div class="sourceCode" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Filtering ortholog groups to ensure all genes have expression data...</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean gene IDs to check against expression data</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># For Apul: remove -T[n] suffix from ortholog groups to match gene matrix format</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>apul_ortholog_genes_check <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-T[0-9]+$"</span>, <span class="st">""</span>, complete_ortholog_groups<span class="sc">$</span>apul)</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co"># For Peve and Ptua: use as-is (will clean gene matrix IDs later)</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>peve_ortholog_genes_check <span class="ot">&lt;-</span> complete_ortholog_groups<span class="sc">$</span>peve</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>ptua_ortholog_genes_check <span class="ot">&lt;-</span> complete_ortholog_groups<span class="sc">$</span>ptua</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Check which genes are present in expression data</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="co"># (Note: We need to clean gene matrix IDs to match)</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>apul_gene_matrix_ids <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^gene-"</span>, <span class="st">""</span>, apul_gene_matrix<span class="sc">$</span>gene_id)  <span class="co"># Remove gene- prefix if present</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>peve_gene_matrix_ids <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^gene-"</span>, <span class="st">""</span>, peve_gene_matrix<span class="sc">$</span>gene_id)  <span class="co"># Remove gene- prefix</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>ptua_gene_matrix_ids <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^gene-"</span>, <span class="st">""</span>, ptua_gene_matrix<span class="sc">$</span>gene_id)  <span class="co"># Remove gene- prefix</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Find which ortholog genes are present in each species' expression data</span></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>apul_present <span class="ot">&lt;-</span> apul_ortholog_genes_check <span class="sc">%in%</span> apul_gene_matrix_ids</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>peve_present <span class="ot">&lt;-</span> peve_ortholog_genes_check <span class="sc">%in%</span> peve_gene_matrix_ids</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>ptua_present <span class="ot">&lt;-</span> ptua_ortholog_genes_check <span class="sc">%in%</span> ptua_gene_matrix_ids</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only ortholog groups where all three species have expression data</span></span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>expression_complete_mask <span class="ot">&lt;-</span> apul_present <span class="sc">&amp;</span> peve_present <span class="sc">&amp;</span> ptua_present</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>complete_ortholog_groups <span class="ot">&lt;-</span> complete_ortholog_groups[expression_complete_mask, ]</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ortholog groups after filtering for expression data availability:"</span>, <span class="fu">nrow</span>(complete_ortholog_groups), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Filtering ortholog groups to ensure all genes have expression data...
## Ortholog groups after filtering for expression data availability: 10258</code></pre>
</section>
<section id="gene-id-cleaning-examples" class="level3">
<h3 class="anchored" data-anchor-id="gene-id-cleaning-examples">3.2.4 Gene ID cleaning examples</h3>
<div class="sourceCode" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">=== GENE ID CLEANING EXAMPLES ===</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Apul (clean ortholog groups to match gene matrix):</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ortholog groups original:"</span>, <span class="fu">head</span>(complete_ortholog_groups<span class="sc">$</span>apul, <span class="dv">3</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ortholog groups cleaned:"</span>, <span class="fu">head</span>(<span class="fu">gsub</span>(<span class="st">"-T[0-9]+$"</span>, <span class="st">""</span>, complete_ortholog_groups<span class="sc">$</span>apul), <span class="dv">3</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Gene matrix (target format):"</span>, <span class="fu">head</span>(apul_gene_matrix<span class="sc">$</span>gene_id, <span class="dv">3</span>), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Peve (clean gene matrix to match ortholog groups):</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ortholog groups (target format):"</span>, <span class="fu">head</span>(complete_ortholog_groups<span class="sc">$</span>peve, <span class="dv">3</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>) </span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Gene matrix original:"</span>, <span class="fu">head</span>(peve_gene_matrix<span class="sc">$</span>gene_id, <span class="dv">3</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Gene matrix cleaned:"</span>, <span class="fu">head</span>(peve_gene_matrix<span class="sc">$</span>gene_id_clean, <span class="dv">3</span>), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ptua (clean gene matrix to match ortholog groups):</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ortholog groups (target format):"</span>, <span class="fu">head</span>(complete_ortholog_groups<span class="sc">$</span>ptua, <span class="dv">3</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Gene matrix original:"</span>, <span class="fu">head</span>(ptua_gene_matrix<span class="sc">$</span>gene_id, <span class="dv">3</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Gene matrix cleaned:"</span>, <span class="fu">head</span>(ptua_gene_matrix<span class="sc">$</span>gene_id_clean, <span class="dv">3</span>), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
## === GENE ID CLEANING EXAMPLES ===
## Apul (clean ortholog groups to match gene matrix):
## Ortholog groups original: FUN_000185-T1 FUN_000189-T1 FUN_000190-T1 
## Ortholog groups cleaned: FUN_000185 FUN_000189 FUN_000190 
## Gene matrix (target format): FUN_002326 FUN_002315 FUN_002316 
## 
## Peve (clean gene matrix to match ortholog groups):
## Ortholog groups (target format): Peve_00037402 Peve_00038462 Peve_00038463 
## Gene matrix original: gene-Peve_00000032 gene-Peve_00000122 gene-Peve_00000008 
## Gene matrix cleaned: 
## 
## Ptua (clean gene matrix to match ortholog groups):
## Ortholog groups (target format): Pocillopora_meandrina_HIv1___RNAseq.g28886.t1 Pocillopora_meandrina_HIv1___RNAseq.g28888.t3 Pocillopora_meandrina_HIv1___RNAseq.g28889.t1 
## Gene matrix original: gene-Pocillopora_meandrina_HIv1___RNAseq.g20905.t1 gene-Pocillopora_meandrina_HIv1___RNAseq.g20902.t1 gene-Pocillopora_meandrina_HIv1___RNAseq.g20903.t1 
## Gene matrix cleaned:</code></pre>
</section>
<section id="clean-gene-ids-for-matching" class="level3">
<h3 class="anchored" data-anchor-id="clean-gene-ids-for-matching">3.2.5 Clean gene IDs for matching</h3>
<div class="sourceCode" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Cleaning gene matrix IDs to match ortholog group format...</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co"># For Apul: ortholog groups have "FUN_000185-T1", gene matrix has "FUN_002326"</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co"># We need to remove "-T1" from ortholog groups to match gene matrix</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>apul_ortholog_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">gsub</span>(<span class="st">"-T[0-9]+$"</span>, <span class="st">""</span>, complete_ortholog_groups<span class="sc">$</span>apul))</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co"># For Peve and Ptua: keep ortholog groups as-is and clean gene matrix</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>peve_ortholog_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(complete_ortholog_groups<span class="sc">$</span>peve)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>ptua_ortholog_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(complete_ortholog_groups<span class="sc">$</span>ptua)</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean gene matrix IDs accordingly</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Apul: gene matrix already in correct format (no cleaning needed)</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>apul_gene_matrix<span class="sc">$</span>gene_id_clean <span class="ot">&lt;-</span> apul_gene_matrix<span class="sc">$</span>gene_id</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Peve: gene matrix has "gene-Peve_00000032", ortholog groups have "Peve_00037402"  </span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="co"># So we need to remove "gene-" prefix from gene matrix</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>peve_gene_matrix<span class="sc">$</span>gene_id_clean <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^gene-"</span>, <span class="st">""</span>, peve_gene_matrix<span class="sc">$</span>gene_id)</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Ptua: gene matrix has "gene-Pocillopora_meandrina_HIv1___RNAseq.g20905.t1", </span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a><span class="co"># ortholog groups have "Pocillopora_meandrina_HIv1___RNAseq.g28886.t1"</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a><span class="co"># So we just need to remove "gene-" prefix from gene matrix</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>ptua_gene_matrix<span class="sc">$</span>gene_id_clean <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^gene-"</span>, <span class="st">""</span>, ptua_gene_matrix<span class="sc">$</span>gene_id)</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Apul ortholog genes (complete groups only):"</span>, <span class="fu">length</span>(apul_ortholog_genes), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Peve ortholog genes (complete groups only):"</span>, <span class="fu">length</span>(peve_ortholog_genes), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ptua ortholog genes (complete groups only):"</span>, <span class="fu">length</span>(ptua_ortholog_genes), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Cleaning gene matrix IDs to match ortholog group format...
## Apul ortholog genes (complete groups only): 10223 
## Peve ortholog genes (complete groups only): 10223 
## Ptua ortholog genes (complete groups only): 10223</code></pre>
</section>
<section id="create-ortholog-expression-data-with-group_id-mapping" class="level3">
<h3 class="anchored" data-anchor-id="create-ortholog-expression-data-with-group_id-mapping">3.2.6 Create ortholog expression data with group_id mapping</h3>
<div class="sourceCode" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Creating ortholog expression data with proper group_id mapping...</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to extract expression data for a species using ortholog group mapping</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>extract_species_expression <span class="ot">&lt;-</span> <span class="cf">function</span>(ortholog_groups_df, gene_matrix, species_col, species_name) {</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Processing"</span>, species_name, <span class="st">"...</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove duplicate group_ids, keeping first occurrence of each</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  Input ortholog groups:"</span>, <span class="fu">nrow</span>(ortholog_groups_df), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  unique_groups <span class="ot">&lt;-</span> ortholog_groups_df[<span class="sc">!</span><span class="fu">duplicated</span>(ortholog_groups_df<span class="sc">$</span>group_id), ]</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  After removing duplicates:"</span>, <span class="fu">nrow</span>(unique_groups), <span class="st">"unique group_ids</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create results data frame starting with group_id</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  result_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">group_id =</span> unique_groups<span class="sc">$</span>group_id, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get expression columns (exclude gene_id and gene_id_clean columns)</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>  expr_cols <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">colnames</span>(gene_matrix), <span class="fu">c</span>(<span class="st">"gene_id"</span>, <span class="st">"gene_id_clean"</span>))</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Initialize expression columns with NA</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(col <span class="cf">in</span> expr_cols) {</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>    result_df[[col]] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># For each ortholog group, find the corresponding gene and extract expression</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(unique_groups))) {</span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>    target_gene <span class="ot">&lt;-</span> unique_groups[[species_col]][i]</span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Clean target gene for matching</span></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(species_name <span class="sc">==</span> <span class="st">"Apul"</span>) {</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Remove transcript suffix for Apul</span></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>      target_gene_clean <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-T[0-9]+$"</span>, <span class="st">""</span>, target_gene)</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>      matching_rows <span class="ot">&lt;-</span> <span class="fu">which</span>(gene_matrix<span class="sc">$</span>gene_id_clean <span class="sc">==</span> target_gene_clean)</span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>      <span class="co"># For Peve and Ptua, match cleaned gene_id</span></span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>      matching_rows <span class="ot">&lt;-</span> <span class="fu">which</span>(gene_matrix<span class="sc">$</span>gene_id_clean <span class="sc">==</span> target_gene)</span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(matching_rows) <span class="sc">==</span> <span class="dv">1</span>) {</span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Single match - copy expression data</span></span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span>(col <span class="cf">in</span> expr_cols) {</span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a>        result_df[i, col] <span class="ot">&lt;-</span> gene_matrix[matching_rows, col]</span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span>(<span class="fu">length</span>(matching_rows) <span class="sc">&gt;</span> <span class="dv">1</span>) {</span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Multiple matches - take first gene (no averaging)</span></span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">"  Group"</span>, unique_groups<span class="sc">$</span>group_id[i], <span class="st">"has"</span>, <span class="fu">length</span>(matching_rows), <span class="st">"gene matches - using first match</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a>      first_match <span class="ot">&lt;-</span> matching_rows[<span class="dv">1</span>]</span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span>(col <span class="cf">in</span> expr_cols) {</span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a>        result_df[i, col] <span class="ot">&lt;-</span> gene_matrix[first_match, col]</span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a>      <span class="co"># No match - leave as NA</span></span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">"  Warning: No expression data found for group"</span>, unique_groups<span class="sc">$</span>group_id[i], <span class="st">"gene"</span>, target_gene, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-55"><a href="#cb16-55" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove rows with all NA expression values</span></span>
<span id="cb16-56"><a href="#cb16-56" aria-hidden="true" tabindex="-1"></a>  expr_na_mask <span class="ot">&lt;-</span> <span class="fu">apply</span>(result_df[, expr_cols, <span class="at">drop =</span> <span class="cn">FALSE</span>], <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="fu">all</span>(<span class="fu">is.na</span>(x)))</span>
<span id="cb16-57"><a href="#cb16-57" aria-hidden="true" tabindex="-1"></a>  result_df <span class="ot">&lt;-</span> result_df[<span class="sc">!</span>expr_na_mask, ]</span>
<span id="cb16-58"><a href="#cb16-58" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-59"><a href="#cb16-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  Final dimensions:"</span>, <span class="fu">nrow</span>(result_df), <span class="st">"ortholog groups x"</span>, <span class="fu">ncol</span>(result_df)<span class="sc">-</span><span class="dv">1</span>, <span class="st">"samples</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb16-60"><a href="#cb16-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  Removed"</span>, <span class="fu">sum</span>(expr_na_mask), <span class="st">"groups with no expression data</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb16-61"><a href="#cb16-61" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-62"><a href="#cb16-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(result_df)</span>
<span id="cb16-63"><a href="#cb16-63" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-64"><a href="#cb16-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-65"><a href="#cb16-65" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract expression data for each species using the ortholog group mapping</span></span>
<span id="cb16-66"><a href="#cb16-66" aria-hidden="true" tabindex="-1"></a>apul_ortholog_expression <span class="ot">&lt;-</span> <span class="fu">extract_species_expression</span>(complete_ortholog_groups, apul_gene_matrix, <span class="st">"apul"</span>, <span class="st">"Apul"</span>)</span>
<span id="cb16-67"><a href="#cb16-67" aria-hidden="true" tabindex="-1"></a>peve_ortholog_expression <span class="ot">&lt;-</span> <span class="fu">extract_species_expression</span>(complete_ortholog_groups, peve_gene_matrix, <span class="st">"peve"</span>, <span class="st">"Peve"</span>)</span>
<span id="cb16-68"><a href="#cb16-68" aria-hidden="true" tabindex="-1"></a>ptua_ortholog_expression <span class="ot">&lt;-</span> <span class="fu">extract_species_expression</span>(complete_ortholog_groups, ptua_gene_matrix, <span class="st">"ptua"</span>, <span class="st">"Ptua"</span>)</span>
<span id="cb16-69"><a href="#cb16-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-70"><a href="#cb16-70" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"=== FINAL ORTHOLOG EXPRESSION DATA DIMENSIONS ===</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb16-71"><a href="#cb16-71" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Apul:"</span>, <span class="fu">nrow</span>(apul_ortholog_expression), <span class="st">"ortholog groups x"</span>, <span class="fu">ncol</span>(apul_ortholog_expression)<span class="sc">-</span><span class="dv">1</span>, <span class="st">"samples</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb16-72"><a href="#cb16-72" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Peve:"</span>, <span class="fu">nrow</span>(peve_ortholog_expression), <span class="st">"ortholog groups x"</span>, <span class="fu">ncol</span>(peve_ortholog_expression)<span class="sc">-</span><span class="dv">1</span>, <span class="st">"samples</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb16-73"><a href="#cb16-73" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ptua:"</span>, <span class="fu">nrow</span>(ptua_ortholog_expression), <span class="st">"ortholog groups x"</span>, <span class="fu">ncol</span>(ptua_ortholog_expression)<span class="sc">-</span><span class="dv">1</span>, <span class="st">"samples</span><span class="sc">\n\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Creating ortholog expression data with proper group_id mapping...
## Processing Apul ...
##   Input ortholog groups: 10258 
##   After removing duplicates: 10223 unique group_ids
##   Final dimensions: 10223 ortholog groups x 40 samples
##   Removed 0 groups with no expression data
## 
## Processing Peve ...
##   Input ortholog groups: 10258 
##   After removing duplicates: 10223 unique group_ids
##   Final dimensions: 10223 ortholog groups x 38 samples
##   Removed 0 groups with no expression data
## 
## Processing Ptua ...
##   Input ortholog groups: 10258 
##   After removing duplicates: 10223 unique group_ids
##   Final dimensions: 10223 ortholog groups x 39 samples
##   Removed 0 groups with no expression data
## 
## === FINAL ORTHOLOG EXPRESSION DATA DIMENSIONS ===
## Apul: 10223 ortholog groups x 40 samples
## Peve: 10223 ortholog groups x 38 samples
## Ptua: 10223 ortholog groups x 39 samples</code></pre>
</section>
<section id="write-ortholog-expression-data" class="level3">
<h3 class="anchored" data-anchor-id="write-ortholog-expression-data">3.2.7 Write ortholog expression data</h3>
<div class="sourceCode" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Exporting ortholog expression data to CSV files...</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Define output file paths</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>apul_output_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_dir, <span class="st">"apul_ortholog_expression.csv"</span>)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>peve_output_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_dir, <span class="st">"peve_ortholog_expression.csv"</span>)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>ptua_output_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_dir, <span class="st">"ptua_ortholog_expression.csv"</span>)</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Write CSV files without quotes</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(apul_ortholog_expression, <span class="at">file =</span> apul_output_file, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Exported Apul ortholog expression data to:"</span>, apul_output_file, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(peve_ortholog_expression, <span class="at">file =</span> peve_output_file, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Exported Peve ortholog expression data to:"</span>, peve_output_file, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(ptua_ortholog_expression, <span class="at">file =</span> ptua_output_file, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Exported Ptua ortholog expression data to:"</span>, ptua_output_file, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">All ortholog expression data exported successfully!</span><span class="sc">\n\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Exporting ortholog expression data to CSV files...
## Exported Apul ortholog expression data to: ../output/13.00-multiomics-barnacle/apul_ortholog_expression.csv 
## Exported Peve ortholog expression data to: ../output/13.00-multiomics-barnacle/peve_ortholog_expression.csv 
## Exported Ptua ortholog expression data to: ../output/13.00-multiomics-barnacle/ptua_ortholog_expression.csv 
## 
## All ortholog expression data exported successfully!</code></pre>
</section>
<section id="column-structure-analysis" class="level3">
<h3 class="anchored" data-anchor-id="column-structure-analysis">3.2.8 Column structure analysis</h3>
<div class="sourceCode" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"=== COLUMN STRUCTURE ANALYSIS ===</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Apul columns:"</span>, <span class="fu">ncol</span>(apul_ortholog_expression), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Apul column names (first 10):"</span>, <span class="fu">paste</span>(<span class="fu">head</span>(<span class="fu">colnames</span>(apul_ortholog_expression), <span class="dv">10</span>), <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Apul column names (last 10):"</span>, <span class="fu">paste</span>(<span class="fu">tail</span>(<span class="fu">colnames</span>(apul_ortholog_expression), <span class="dv">10</span>), <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Peve columns:"</span>, <span class="fu">ncol</span>(peve_ortholog_expression), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Peve column names (first 10):"</span>, <span class="fu">paste</span>(<span class="fu">head</span>(<span class="fu">colnames</span>(peve_ortholog_expression), <span class="dv">10</span>), <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Peve column names (last 10):"</span>, <span class="fu">paste</span>(<span class="fu">tail</span>(<span class="fu">colnames</span>(peve_ortholog_expression), <span class="dv">10</span>), <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ptua columns:"</span>, <span class="fu">ncol</span>(ptua_ortholog_expression), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ptua column names (first 10):"</span>, <span class="fu">paste</span>(<span class="fu">head</span>(<span class="fu">colnames</span>(ptua_ortholog_expression), <span class="dv">10</span>), <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ptua column names (last 10):"</span>, <span class="fu">paste</span>(<span class="fu">tail</span>(<span class="fu">colnames</span>(ptua_ortholog_expression), <span class="dv">10</span>), <span class="at">collapse =</span> <span class="st">", "</span>), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## === COLUMN STRUCTURE ANALYSIS ===
## Apul columns: 41 
## Apul column names (first 10): group_id, ACR.139.TP1, ACR.139.TP2, ACR.139.TP3, ACR.139.TP4, ACR.145.TP1, ACR.145.TP2, ACR.145.TP3, ACR.145.TP4, ACR.150.TP1 
## Apul column names (last 10): ACR.237.TP3, ACR.237.TP4, ACR.244.TP1, ACR.244.TP2, ACR.244.TP3, ACR.244.TP4, ACR.265.TP1, ACR.265.TP2, ACR.265.TP3, ACR.265.TP4 
## 
## Peve columns: 39 
## Peve column names (first 10): group_id, POR.216.TP1, POR.216.TP2, POR.216.TP3, POR.216.TP4, POR.236.TP1, POR.236.TP2, POR.245.TP1, POR.245.TP2, POR.245.TP3 
## Peve column names (last 10): POR.73.TP3, POR.73.TP4, POR.74.TP1, POR.74.TP2, POR.74.TP3, POR.74.TP4, POR.83.TP1, POR.83.TP2, POR.83.TP3, POR.83.TP4 
## 
## Ptua columns: 40 
## Ptua column names (first 10): group_id, POC.201.TP1, POC.201.TP2, POC.201.TP3, POC.219.TP1, POC.219.TP2, POC.219.TP3, POC.219.TP4, POC.222.TP1, POC.222.TP2 
## Ptua column names (last 10): POC.52.TP3, POC.52.TP4, POC.53.TP1, POC.53.TP2, POC.53.TP3, POC.53.TP4, POC.57.TP1, POC.57.TP2, POC.57.TP3, POC.57.TP4</code></pre>
</section>
<section id="summary-statistics" class="level3">
<h3 class="anchored" data-anchor-id="summary-statistics">3.2.9 Summary statistics</h3>
<div class="sourceCode" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"=== LINE COUNTS FOR ORTHOLOG EXPRESSION DATA ===</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Apul ortholog expression with info: "</span>, <span class="fu">nrow</span>(apul_ortholog_expression), <span class="st">" rows</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Peve ortholog expression with info: "</span>, <span class="fu">nrow</span>(peve_ortholog_expression), <span class="st">" rows</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ptua ortholog expression with info: "</span>, <span class="fu">nrow</span>(ptua_ortholog_expression), <span class="st">" rows</span><span class="sc">\n\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## === LINE COUNTS FOR ORTHOLOG EXPRESSION DATA ===
## Apul ortholog expression with info:  10223  rows
## Peve ortholog expression with info:  10223  rows
## Ptua ortholog expression with info:  10223  rows</code></pre>
</section>
<section id="diagnostic-analysis" class="level3">
<h3 class="anchored" data-anchor-id="diagnostic-analysis">3.2.10 Diagnostic analysis</h3>
<div class="sourceCode" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">=== VERIFICATION: THREE-WAY ORTHOLOG COUNTS ===</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"After filtering, all species should have identical three_way ortholog counts.</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check how many genes we have for each species</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Genes found in expression data by species:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Apul ortholog expression (before adding info):"</span>, <span class="fu">nrow</span>(apul_ortholog_expression), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Peve ortholog expression (before adding info):"</span>, <span class="fu">nrow</span>(peve_ortholog_expression), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ptua ortholog expression (before adding info):"</span>, <span class="fu">nrow</span>(ptua_ortholog_expression), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Verify all ortholog groups are three-way</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Complete three-way ortholog groups available:"</span>, <span class="fu">nrow</span>(complete_ortholog_groups), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"All should be type 'three_way'? Check:"</span>, <span class="fu">table</span>(complete_ortholog_groups<span class="sc">$</span>type), <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Verify perfect gene coverage (should be 100% for all species now)</span></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>apul_coverage <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">intersect</span>(apul_ortholog_genes, apul_gene_matrix<span class="sc">$</span>gene_id_clean))</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>peve_coverage <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">intersect</span>(peve_ortholog_genes, peve_gene_matrix<span class="sc">$</span>gene_id_clean))</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>ptua_coverage <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">intersect</span>(ptua_ortholog_genes, ptua_gene_matrix<span class="sc">$</span>gene_id_clean))</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Gene coverage in expression data (should be 100% for all):</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Apul: "</span>, apul_coverage, <span class="st">"/"</span>, <span class="fu">length</span>(apul_ortholog_genes), <span class="st">" ("</span>, <span class="fu">round</span>(apul_coverage<span class="sc">/</span><span class="fu">length</span>(apul_ortholog_genes)<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%)</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Peve: "</span>, peve_coverage, <span class="st">"/"</span>, <span class="fu">length</span>(peve_ortholog_genes), <span class="st">" ("</span>, <span class="fu">round</span>(peve_coverage<span class="sc">/</span><span class="fu">length</span>(peve_ortholog_genes)<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%)</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ptua: "</span>, ptua_coverage, <span class="st">"/"</span>, <span class="fu">length</span>(ptua_ortholog_genes), <span class="st">" ("</span>, <span class="fu">round</span>(ptua_coverage<span class="sc">/</span><span class="fu">length</span>(ptua_ortholog_genes)<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%)</span><span class="sc">\n\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
## === VERIFICATION: THREE-WAY ORTHOLOG COUNTS ===
## After filtering, all species should have identical three_way ortholog counts.
## 
## Genes found in expression data by species:
## Apul ortholog expression (before adding info): 10223 
## Peve ortholog expression (before adding info): 10223 
## Ptua ortholog expression (before adding info): 10223 
## 
## Complete three-way ortholog groups available: 10258 
## All should be type 'three_way'? Check: 10258 
## 
## Gene coverage in expression data (should be 100% for all):
## Apul:  10223 / 10223  ( 100 %)
## Peve:  10223 / 10223  ( 100 %)
## Ptua:  10223 / 10223  ( 100 %)</code></pre>
</section>
<section id="preview-expression-data" class="level3">
<h3 class="anchored" data-anchor-id="preview-expression-data">3.2.11 Preview expression data</h3>
<div class="sourceCode" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">=== PREVIEW OF ORTHOLOG EXPRESSION DATA ===</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Apul ortholog expression:</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(apul_ortholog_expression)</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">Peve ortholog expression:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(peve_ortholog_expression)</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">Ptua ortholog expression:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ptua_ortholog_expression)</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
## === PREVIEW OF ORTHOLOG EXPRESSION DATA ===
## Apul ortholog expression:
## 
## 'data.frame':    10223 obs. of  41 variables:
##  $ group_id   : chr  "OG_00001" "OG_00002" "OG_00003" "OG_00004" ...
##  $ ACR.139.TP1: int  2507 1890 323 1600 295 185 1000 102 707 641 ...
##  $ ACR.139.TP2: int  2446 3030 237 1745 398 233 1186 88 1080 669 ...
##  $ ACR.139.TP3: int  1547 1409 202 3039 582 393 1098 61 793 586 ...
##  $ ACR.139.TP4: int  1917 1037 308 1843 329 304 1011 30 843 448 ...
##  $ ACR.145.TP1: int  1514 868 492 2521 435 484 1119 35 153 330 ...
##  $ ACR.145.TP2: int  1257 1930 188 1243 255 146 569 71 572 1042 ...
##  $ ACR.145.TP3: int  864 1370 356 3488 504 310 796 0 234 872 ...
##  $ ACR.145.TP4: int  992 980 345 1408 271 110 1120 63 239 265 ...
##  $ ACR.150.TP1: int  456 379 308 2502 279 257 836 59 163 181 ...
##  $ ACR.150.TP2: int  1774 4357 421 2344 388 308 849 53 415 1386 ...
##  $ ACR.150.TP3: int  2107 3236 267 1948 393 347 994 68 1036 1241 ...
##  $ ACR.150.TP4: int  2344 1484 382 1815 373 331 1212 59 1010 854 ...
##  $ ACR.173.TP1: int  511 549 394 1213 256 176 520 59 226 281 ...
##  $ ACR.173.TP2: int  1295 1658 291 869 256 159 866 113 658 566 ...
##  $ ACR.173.TP3: int  1023 1505 479 1806 311 371 878 39 707 667 ...
##  $ ACR.173.TP4: int  606 443 566 1874 281 384 674 23 322 215 ...
##  $ ACR.186.TP1: int  140 508 372 1731 193 282 465 0 161 102 ...
##  $ ACR.186.TP2: int  2046 3286 330 2320 400 343 716 75 915 1271 ...
##  $ ACR.186.TP3: int  915 1112 314 3149 270 248 395 9 363 642 ...
##  $ ACR.186.TP4: int  1943 1317 436 2586 292 253 951 18 876 505 ...
##  $ ACR.225.TP1: int  1687 467 463 1773 429 244 834 12 438 135 ...
##  $ ACR.225.TP2: int  1812 2140 292 1353 164 93 915 72 651 1095 ...
##  $ ACR.225.TP3: int  2225 2327 525 2417 479 309 1061 69 904 661 ...
##  $ ACR.225.TP4: int  850 411 324 2484 305 203 659 18 315 125 ...
##  $ ACR.229.TP1: int  3876 1026 493 2266 832 471 1699 54 847 424 ...
##  $ ACR.229.TP2: int  2471 3540 276 2826 358 272 762 93 674 1378 ...
##  $ ACR.229.TP3: int  2409 2606 399 2208 303 272 844 58 993 798 ...
##  $ ACR.229.TP4: int  756 497 842 2606 325 339 789 21 119 231 ...
##  $ ACR.237.TP1: int  1392 583 374 3052 474 399 1247 21 372 235 ...
##  $ ACR.237.TP2: int  1297 1620 231 1447 343 201 919 100 560 1073 ...
##  $ ACR.237.TP3: int  2023 3745 246 2193 341 224 1174 127 1019 1437 ...
##  $ ACR.237.TP4: int  1080 673 293 3552 367 410 1046 39 408 490 ...
##  $ ACR.244.TP1: int  998 469 376 1363 490 281 630 28 203 91 ...
##  $ ACR.244.TP2: int  1987 3116 403 2706 406 231 1268 102 695 1592 ...
##  $ ACR.244.TP3: int  2553 3142 400 2036 422 203 1227 186 1127 814 ...
##  $ ACR.244.TP4: int  752 403 494 2792 308 333 601 5 245 339 ...
##  $ ACR.265.TP1: int  1004 339 815 3730 507 336 912 30 153 255 ...
##  $ ACR.265.TP2: int  1876 3004 311 3363 568 263 1106 74 577 1454 ...
##  $ ACR.265.TP3: int  1848 1914 383 2090 484 187 1135 85 394 624 ...
##  $ ACR.265.TP4: int  697 586 686 3726 594 494 1044 34 205 351 ...
## 
## 
## 
## Peve ortholog expression:
## 'data.frame':    10223 obs. of  39 variables:
##  $ group_id   : chr  "OG_00001" "OG_00002" "OG_00003" "OG_00004" ...
##  $ POR.216.TP1: int  2254 42 894 67 51 6 217 11 256 0 ...
##  $ POR.216.TP2: int  2122 62 191 233 147 43 768 84 219 9 ...
##  $ POR.216.TP3: int  785 25 94 70 58 41 357 13 90 0 ...
##  $ POR.216.TP4: int  470 17 131 38 35 0 136 16 37 0 ...
##  $ POR.236.TP1: int  1971 45 220 251 138 45 681 38 213 11 ...
##  $ POR.236.TP2: int  849 15 61 81 61 16 300 11 83 0 ...
##  $ POR.245.TP1: int  1757 33 276 204 127 8 429 48 218 10 ...
##  $ POR.245.TP2: int  714 8 59 88 59 47 269 30 101 4 ...
##  $ POR.245.TP3: int  587 8 70 69 37 19 170 25 100 0 ...
##  $ POR.245.TP4: int  1479 90 564 91 88 18 712 72 82 12 ...
##  $ POR.260.TP1: int  2395 67 506 189 149 56 674 39 291 0 ...
##  $ POR.260.TP2: int  937 23 73 69 52 16 377 12 149 0 ...
##  $ POR.260.TP3: int  739 8 50 73 81 19 334 18 71 0 ...
##  $ POR.260.TP4: int  1196 37 337 139 49 27 327 36 144 0 ...
##  $ POR.262.TP1: int  1603 52 192 113 68 11 363 12 179 0 ...
##  $ POR.262.TP2: int  1412 92 156 215 111 25 918 41 131 0 ...
##  $ POR.262.TP3: int  1604 66 145 155 177 36 742 54 218 0 ...
##  $ POR.262.TP4: int  882 56 124 156 100 23 435 45 103 0 ...
##  $ POR.69.TP1 : int  742 44 172 58 82 8 226 40 127 0 ...
##  $ POR.69.TP2 : int  1357 26 463 89 70 10 379 20 212 8 ...
##  $ POR.69.TP3 : int  821 19 23 73 22 13 239 10 126 6 ...
##  $ POR.69.TP4 : int  387 14 67 46 7 17 114 34 118 3 ...
##  $ POR.72.TP1 : int  933 44 160 77 46 8 378 45 123 0 ...
##  $ POR.72.TP2 : int  1318 37 395 100 97 11 144 18 146 0 ...
##  $ POR.72.TP3 : int  1343 60 345 67 49 11 253 26 241 0 ...
##  $ POR.72.TP4 : int  1665 73 221 158 122 25 310 18 204 0 ...
##  $ POR.73.TP1 : int  837 25 126 52 57 12 276 21 209 0 ...
##  $ POR.73.TP2 : int  1226 22 265 38 33 10 165 29 130 0 ...
##  $ POR.73.TP3 : int  362 5 17 30 19 5 151 9 59 0 ...
##  $ POR.73.TP4 : int  716 9 49 51 19 5 137 23 115 0 ...
##  $ POR.74.TP1 : int  1590 44 288 107 106 3 273 50 131 0 ...
##  $ POR.74.TP2 : int  427 18 90 46 44 0 131 18 42 0 ...
##  $ POR.74.TP3 : int  1089 28 126 111 34 21 358 9 149 0 ...
##  $ POR.74.TP4 : int  578 3 132 50 24 4 98 0 79 0 ...
##  $ POR.83.TP1 : int  631 56 235 93 40 9 305 23 82 9 ...
##  $ POR.83.TP2 : int  999 35 252 67 13 2 101 4 159 0 ...
##  $ POR.83.TP3 : int  743 30 35 46 36 28 114 17 129 0 ...
##  $ POR.83.TP4 : int  524 13 82 30 10 0 74 4 60 0 ...
## 
## 
## 
## Ptua ortholog expression:
## 'data.frame':    10223 obs. of  40 variables:
##  $ group_id   : chr  "OG_00001" "OG_00002" "OG_00003" "OG_00004" ...
##  $ POC.201.TP1: int  1444 376 86 345 79 86 727 441 1033 53 ...
##  $ POC.201.TP2: int  794 557 139 311 135 179 1156 151 1296 112 ...
##  $ POC.201.TP3: int  171 154 75 112 40 35 257 81 205 33 ...
##  $ POC.219.TP1: int  415 496 220 480 248 143 926 282 868 56 ...
##  $ POC.219.TP2: int  365 402 87 255 118 108 556 201 625 55 ...
##  $ POC.219.TP3: int  57 113 38 93 12 4 78 56 99 0 ...
##  $ POC.219.TP4: int  187 281 60 235 21 45 343 286 366 21 ...
##  $ POC.222.TP1: int  396 638 222 651 229 406 963 408 1162 140 ...
##  $ POC.222.TP2: int  804 239 166 205 59 181 575 327 1238 44 ...
##  $ POC.222.TP3: int  1415 687 176 580 347 287 1113 208 1757 93 ...
##  $ POC.222.TP4: int  327 238 106 198 58 109 380 223 398 52 ...
##  $ POC.255.TP1: int  307 464 111 328 143 157 780 291 850 76 ...
##  $ POC.255.TP2: int  466 375 101 148 50 45 432 148 815 48 ...
##  $ POC.255.TP3: int  229 218 152 202 79 48 366 101 253 17 ...
##  $ POC.255.TP4: int  438 396 133 438 132 99 748 173 1112 18 ...
##  $ POC.259.TP1: int  733 565 142 271 93 109 790 343 876 75 ...
##  $ POC.259.TP2: int  1167 662 181 401 168 317 1228 271 1097 107 ...
##  $ POC.259.TP3: int  311 300 121 374 83 53 524 357 636 21 ...
##  $ POC.259.TP4: int  723 509 178 594 119 159 823 329 1149 62 ...
##  $ POC.40.TP1 : int  69 228 227 209 84 116 373 338 762 18 ...
##  $ POC.40.TP2 : int  155 251 132 272 60 74 355 329 854 31 ...
##  $ POC.40.TP3 : int  245 500 142 368 111 135 649 228 898 55 ...
##  $ POC.40.TP4 : int  75 324 144 288 124 134 446 235 1072 27 ...
##  $ POC.42.TP1 : int  155 312 170 236 84 136 462 315 621 24 ...
##  $ POC.42.TP2 : int  263 520 222 370 114 112 662 399 889 30 ...
##  $ POC.42.TP3 : int  201 321 125 208 72 122 347 196 798 15 ...
##  $ POC.42.TP4 : int  50 64 86 129 36 62 84 105 53 0 ...
##  $ POC.52.TP1 : int  38 64 19 61 0 0 0 34 55 6 ...
##  $ POC.52.TP2 : int  697 287 157 364 69 148 446 147 737 25 ...
##  $ POC.52.TP3 : int  769 320 104 420 102 91 623 161 893 35 ...
##  $ POC.52.TP4 : int  1741 544 162 382 228 135 1024 180 1417 64 ...
##  $ POC.53.TP1 : int  837 445 87 265 93 131 758 144 1293 23 ...
##  $ POC.53.TP2 : int  242 345 135 218 49 63 533 148 1038 35 ...
##  $ POC.53.TP3 : int  1134 629 181 448 200 321 1140 248 1174 42 ...
##  $ POC.53.TP4 : int  140 284 131 374 92 47 291 360 498 35 ...
##  $ POC.57.TP1 : int  720 369 88 317 146 242 660 437 1091 46 ...
##  $ POC.57.TP2 : int  378 141 134 184 60 78 279 304 373 18 ...
##  $ POC.57.TP3 : int  1838 465 130 404 158 172 1026 278 1303 36 ...
##  $ POC.57.TP4 : int  287 507 222 282 176 121 511 509 1048 69 ...</code></pre>
</section>
</section>
</section>
<section id="barnacle-analysis" class="level1">
<h1>4 BARNACLE ANALYSIS</h1>
<p>Based on the barnacle workflow, we need to: 1. Use <code>sctransform</code> to normalize the count data 2. Create tensors for multiomics analysis 3. Run sparse tensor decomposition</p>
<section id="load-expression-data" class="level2">
<h2 class="anchored" data-anchor-id="load-expression-data">4.1 Load expression data</h2>
<div class="sourceCode" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the exported ortholog expression data</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>apul_expr <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(output_dir, <span class="st">"apul_ortholog_expression.csv"</span>))</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>peve_expr <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(output_dir, <span class="st">"peve_ortholog_expression.csv"</span>))  </span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>ptua_expr <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(output_dir, <span class="st">"ptua_ortholog_expression.csv"</span>))</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Loaded expression data:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Apul:"</span>, <span class="fu">nrow</span>(apul_expr), <span class="st">"genes x"</span>, <span class="fu">ncol</span>(apul_expr)<span class="sc">-</span><span class="dv">1</span>, <span class="st">"samples</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Peve:"</span>, <span class="fu">nrow</span>(peve_expr), <span class="st">"genes x"</span>, <span class="fu">ncol</span>(peve_expr)<span class="sc">-</span><span class="dv">1</span>, <span class="st">"samples</span><span class="sc">\n</span><span class="st">"</span>) </span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ptua:"</span>, <span class="fu">nrow</span>(ptua_expr), <span class="st">"genes x"</span>, <span class="fu">ncol</span>(ptua_expr)<span class="sc">-</span><span class="dv">1</span>, <span class="st">"samples</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Loaded expression data:
## Apul: 10223 genes x 40 samples
## Peve: 10223 genes x 38 samples
## Ptua: 10223 genes x 39 samples</code></pre>
</section>
<section id="normalize-data-with-sctransform" class="level2">
<h2 class="anchored" data-anchor-id="normalize-data-with-sctransform">4.2 Normalize data with sctransform</h2>
<p>Following the barnacle manuscript approach, we’ll use <code>sctransform</code> to normalize each species’ data. We’ll use a bulk RNA-seq appropriate approach.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to normalize count data with sctransform for bulk RNA-seq</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>normalize_with_sctransform <span class="ot">&lt;-</span> <span class="cf">function</span>(count_data, species_name) {</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Normalizing"</span>, species_name, <span class="st">"data with sctransform...</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if we have group_id or gene_id column</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  id_col <span class="ot">&lt;-</span> <span class="cf">if</span>(<span class="st">"group_id"</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(count_data)) <span class="st">"group_id"</span> <span class="cf">else</span> <span class="st">"gene_id"</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Using"</span>, id_col, <span class="st">"as identifier column</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check for and handle duplicate group_ids/gene_ids</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>  duplicate_ids <span class="ot">&lt;-</span> count_data[[id_col]][<span class="fu">duplicated</span>(count_data[[id_col]])]</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(duplicate_ids) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Found"</span>, <span class="fu">length</span>(<span class="fu">unique</span>(duplicate_ids)), <span class="st">"duplicate"</span>, id_col, <span class="st">"values:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"  Examples:"</span>, <span class="fu">head</span>(<span class="fu">unique</span>(duplicate_ids), <span class="dv">10</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"  Note: sctransform may fail due to duplicate row names, will fall back to log2(CPM + 1)</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"No duplicate"</span>, id_col, <span class="st">"values found</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use original data without aggregation - let sctransform handle duplicates or fail</span></span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>  agg_data <span class="ot">&lt;-</span> count_data</span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract count matrix (genes as rows, samples as columns)</span></span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>  count_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(agg_data[, <span class="sc">-</span><span class="dv">1</span>])  <span class="co"># Remove id column</span></span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(count_matrix) <span class="ot">&lt;-</span> agg_data[[id_col]]</span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check for and handle problematic values</span></span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Checking data quality...</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  - Zero values:"</span>, <span class="fu">sum</span>(count_matrix <span class="sc">==</span> <span class="dv">0</span>), <span class="st">"/"</span>, <span class="fu">length</span>(count_matrix), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  - NA values:"</span>, <span class="fu">sum</span>(<span class="fu">is.na</span>(count_matrix)), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  - Infinite values:"</span>, <span class="fu">sum</span>(<span class="fu">is.infinite</span>(count_matrix)), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  - Min value:"</span>, <span class="fu">min</span>(count_matrix, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  - Max value:"</span>, <span class="fu">max</span>(count_matrix, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove genes with all zeros or very low expression</span></span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a>  gene_sums <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(count_matrix)</span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a>  keep_genes <span class="ot">&lt;-</span> gene_sums <span class="sc">&gt;</span> <span class="dv">10</span>  <span class="co"># Keep genes with total counts &gt; 10</span></span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a>  count_matrix_filtered <span class="ot">&lt;-</span> count_matrix[keep_genes, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"  - Filtered to"</span>, <span class="fu">nrow</span>(count_matrix_filtered), <span class="st">"ortholog groups (from"</span>, <span class="fu">nrow</span>(count_matrix), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Transpose for sctransform (expects cells as rows, genes as columns)</span></span>
<span id="cb30-42"><a href="#cb30-42" aria-hidden="true" tabindex="-1"></a>  count_matrix_t <span class="ot">&lt;-</span> <span class="fu">t</span>(count_matrix_filtered)</span>
<span id="cb30-43"><a href="#cb30-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-44"><a href="#cb30-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply sctransform normalization with bulk RNA-seq appropriate parameters</span></span>
<span id="cb30-45"><a href="#cb30-45" aria-hidden="true" tabindex="-1"></a>  normalized_df <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>({</span>
<span id="cb30-46"><a href="#cb30-46" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use more conservative parameters for bulk RNA-seq</span></span>
<span id="cb30-47"><a href="#cb30-47" aria-hidden="true" tabindex="-1"></a>    normalized <span class="ot">&lt;-</span> sctransform<span class="sc">::</span><span class="fu">vst</span>(</span>
<span id="cb30-48"><a href="#cb30-48" aria-hidden="true" tabindex="-1"></a>      count_matrix_t, </span>
<span id="cb30-49"><a href="#cb30-49" aria-hidden="true" tabindex="-1"></a>      <span class="at">method =</span> <span class="st">"glmGamPoi"</span>,  <span class="co"># More appropriate for bulk data</span></span>
<span id="cb30-50"><a href="#cb30-50" aria-hidden="true" tabindex="-1"></a>      <span class="at">n_genes =</span> <span class="fu">min</span>(<span class="dv">2000</span>, <span class="fu">ncol</span>(count_matrix_t)),  <span class="co"># Use fewer variable genes</span></span>
<span id="cb30-51"><a href="#cb30-51" aria-hidden="true" tabindex="-1"></a>      <span class="at">return_cell_attr =</span> <span class="cn">TRUE</span>,</span>
<span id="cb30-52"><a href="#cb30-52" aria-hidden="true" tabindex="-1"></a>      <span class="at">verbosity =</span> <span class="dv">2</span></span>
<span id="cb30-53"><a href="#cb30-53" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb30-54"><a href="#cb30-54" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-55"><a href="#cb30-55" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Extract normalized data and transpose back (genes as rows, samples as columns)</span></span>
<span id="cb30-56"><a href="#cb30-56" aria-hidden="true" tabindex="-1"></a>    normalized_data <span class="ot">&lt;-</span> <span class="fu">t</span>(normalized<span class="sc">$</span>y)</span>
<span id="cb30-57"><a href="#cb30-57" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-58"><a href="#cb30-58" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create output data frame with original gene set (fill missing with zeros)</span></span>
<span id="cb30-59"><a href="#cb30-59" aria-hidden="true" tabindex="-1"></a>    full_normalized_data <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow =</span> <span class="fu">nrow</span>(count_matrix), <span class="at">ncol =</span> <span class="fu">ncol</span>(count_matrix))</span>
<span id="cb30-60"><a href="#cb30-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(full_normalized_data) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(count_matrix)</span>
<span id="cb30-61"><a href="#cb30-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(full_normalized_data) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(count_matrix)</span>
<span id="cb30-62"><a href="#cb30-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-63"><a href="#cb30-63" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Fill in normalized values for kept genes</span></span>
<span id="cb30-64"><a href="#cb30-64" aria-hidden="true" tabindex="-1"></a>    full_normalized_data[<span class="fu">rownames</span>(normalized_data), ] <span class="ot">&lt;-</span> normalized_data</span>
<span id="cb30-65"><a href="#cb30-65" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-66"><a href="#cb30-66" aria-hidden="true" tabindex="-1"></a>    result_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb30-67"><a href="#cb30-67" aria-hidden="true" tabindex="-1"></a>      <span class="at">group_id =</span> <span class="fu">rownames</span>(full_normalized_data),</span>
<span id="cb30-68"><a href="#cb30-68" aria-hidden="true" tabindex="-1"></a>      full_normalized_data,</span>
<span id="cb30-69"><a href="#cb30-69" aria-hidden="true" tabindex="-1"></a>      <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb30-70"><a href="#cb30-70" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb30-71"><a href="#cb30-71" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-72"><a href="#cb30-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"sctransform normalization successful for"</span>, species_name, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-73"><a href="#cb30-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(result_df)</span>
<span id="cb30-74"><a href="#cb30-74" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-75"><a href="#cb30-75" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb30-76"><a href="#cb30-76" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"sctransform failed for"</span>, species_name, <span class="st">":"</span>, e<span class="sc">$</span>message, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-77"><a href="#cb30-77" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Falling back to log2(CPM + 1) normalization...</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-78"><a href="#cb30-78" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-79"><a href="#cb30-79" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Fallback: log2(CPM + 1) normalization</span></span>
<span id="cb30-80"><a href="#cb30-80" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate CPM (Counts Per Million)</span></span>
<span id="cb30-81"><a href="#cb30-81" aria-hidden="true" tabindex="-1"></a>    lib_sizes <span class="ot">&lt;-</span> <span class="fu">colSums</span>(count_matrix)</span>
<span id="cb30-82"><a href="#cb30-82" aria-hidden="true" tabindex="-1"></a>    cpm_matrix <span class="ot">&lt;-</span> <span class="fu">sweep</span>(count_matrix, <span class="dv">2</span>, lib_sizes<span class="sc">/</span><span class="fl">1e6</span>, <span class="at">FUN =</span> <span class="st">"/"</span>)</span>
<span id="cb30-83"><a href="#cb30-83" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-84"><a href="#cb30-84" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Log2 transform with pseudocount</span></span>
<span id="cb30-85"><a href="#cb30-85" aria-hidden="true" tabindex="-1"></a>    normalized_data <span class="ot">&lt;-</span> <span class="fu">log2</span>(cpm_matrix <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb30-86"><a href="#cb30-86" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-87"><a href="#cb30-87" aria-hidden="true" tabindex="-1"></a>    result_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb30-88"><a href="#cb30-88" aria-hidden="true" tabindex="-1"></a>      <span class="at">group_id =</span> <span class="fu">rownames</span>(normalized_data),</span>
<span id="cb30-89"><a href="#cb30-89" aria-hidden="true" tabindex="-1"></a>      normalized_data,</span>
<span id="cb30-90"><a href="#cb30-90" aria-hidden="true" tabindex="-1"></a>      <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb30-91"><a href="#cb30-91" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb30-92"><a href="#cb30-92" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb30-93"><a href="#cb30-93" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Log2(CPM + 1) normalization complete for"</span>, species_name, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-94"><a href="#cb30-94" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(result_df)</span>
<span id="cb30-95"><a href="#cb30-95" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb30-96"><a href="#cb30-96" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-97"><a href="#cb30-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Input dimensions:"</span>, <span class="fu">nrow</span>(count_data), <span class="st">"rows x"</span>, <span class="fu">ncol</span>(count_data)<span class="sc">-</span><span class="dv">1</span>, <span class="st">"samples</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-98"><a href="#cb30-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Output dimensions:"</span>, <span class="fu">nrow</span>(normalized_df), <span class="st">"ortholog groups x"</span>, <span class="fu">ncol</span>(normalized_df)<span class="sc">-</span><span class="dv">1</span>, <span class="st">"samples</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb30-99"><a href="#cb30-99" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-100"><a href="#cb30-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(normalized_df)</span>
<span id="cb30-101"><a href="#cb30-101" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-102"><a href="#cb30-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-103"><a href="#cb30-103" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize each species</span></span>
<span id="cb30-104"><a href="#cb30-104" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"=== STARTING NORMALIZATION ===</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb30-105"><a href="#cb30-105" aria-hidden="true" tabindex="-1"></a>apul_normalized <span class="ot">&lt;-</span> <span class="fu">normalize_with_sctransform</span>(apul_expr, <span class="st">"Apul"</span>)</span>
<span id="cb30-106"><a href="#cb30-106" aria-hidden="true" tabindex="-1"></a>peve_normalized <span class="ot">&lt;-</span> <span class="fu">normalize_with_sctransform</span>(peve_expr, <span class="st">"Peve"</span>) </span>
<span id="cb30-107"><a href="#cb30-107" aria-hidden="true" tabindex="-1"></a>ptua_normalized <span class="ot">&lt;-</span> <span class="fu">normalize_with_sctransform</span>(ptua_expr, <span class="st">"Ptua"</span>)</span>
<span id="cb30-108"><a href="#cb30-108" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"=== NORMALIZATION COMPLETE ===</span><span class="sc">\n\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## === STARTING NORMALIZATION ===
## 
## Normalizing Apul data with sctransform...
## Using group_id as identifier column
## No duplicate group_id values found
## Checking data quality...
##   - Zero values: 24719 / 408920 
##   - NA values: 0 
##   - Infinite values: 0 
##   - Min value: 0 
##   - Max value: 204262 
##   - Filtered to 10157 ortholog groups (from 10223 )
##   |                                                                              |                                                                      |   0%sctransform failed for Apul : Argument 'useNames' must be either TRUE or FALSE 
## Falling back to log2(CPM + 1) normalization...
## Log2(CPM + 1) normalization complete for Apul 
## Input dimensions: 10223 rows x 40 samples
## Output dimensions: 10223 ortholog groups x 40 samples
## 
## Normalizing Peve data with sctransform...
## Using group_id as identifier column
## No duplicate group_id values found
## Checking data quality...
##   - Zero values: 39003 / 388474 
##   - NA values: 0 
##   - Infinite values: 0 
##   - Min value: 0 
##   - Max value: 208234 
##   - Filtered to 10074 ortholog groups (from 10223 )
##   |                                                                              |                                                                      |   0%sctransform failed for Peve : Argument 'useNames' must be either TRUE or FALSE 
## Falling back to log2(CPM + 1) normalization...
## Log2(CPM + 1) normalization complete for Peve 
## Input dimensions: 10223 rows x 38 samples
## Output dimensions: 10223 ortholog groups x 38 samples
## 
## Normalizing Ptua data with sctransform...
## Using group_id as identifier column
## No duplicate group_id values found
## Checking data quality...
##   - Zero values: 20764 / 398697 
##   - NA values: 0 
##   - Infinite values: 0 
##   - Min value: 0 
##   - Max value: 204924 
##   - Filtered to 10170 ortholog groups (from 10223 )
##   |                                                                              |                                                                      |   0%sctransform failed for Ptua : Argument 'useNames' must be either TRUE or FALSE 
## Falling back to log2(CPM + 1) normalization...
## Log2(CPM + 1) normalization complete for Ptua 
## Input dimensions: 10223 rows x 39 samples
## Output dimensions: 10223 ortholog groups x 39 samples
## 
## === NORMALIZATION COMPLETE ===</code></pre>
</section>
<section id="export-normalized-data-for-python-analysis" class="level2">
<h2 class="anchored" data-anchor-id="export-normalized-data-for-python-analysis">4.3 Export normalized data for Python analysis</h2>
<div class="sourceCode" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export normalized data for Python processing</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>apul_norm_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_dir, <span class="st">"apul_normalized_expression.csv"</span>)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>peve_norm_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_dir, <span class="st">"peve_normalized_expression.csv"</span>)</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>ptua_norm_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_dir, <span class="st">"ptua_normalized_expression.csv"</span>)</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(apul_normalized, apul_norm_file, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(peve_normalized, peve_norm_file, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(ptua_normalized, ptua_norm_file, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Exported normalized data:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Apul:"</span>, apul_norm_file, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Peve:"</span>, peve_norm_file, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Ptua:"</span>, ptua_norm_file, <span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Exported normalized data:
## Apul: ../output/13.00-multiomics-barnacle/apul_normalized_expression.csv 
## Peve: ../output/13.00-multiomics-barnacle/peve_normalized_expression.csv 
## Ptua: ../output/13.00-multiomics-barnacle/ptua_normalized_expression.csv</code></pre>
</section>
<section id="create-tensor-dataset-in-python" class="level2">
<h2 class="anchored" data-anchor-id="create-tensor-dataset-in-python">4.4 Create tensor dataset in Python</h2>
<p>Now we’ll switch to Python to create the multiomics tensor and run barnacle analysis.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pathlib <span class="im">import</span> Path</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up paths</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>output_dir <span class="op">=</span> r.output_dir</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Working in output directory: </span><span class="sc">{</span>output_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Load normalized data</span></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>apul_norm <span class="op">=</span> pd.read_csv(os.path.join(output_dir, <span class="st">"apul_normalized_expression.csv"</span>))</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>peve_norm <span class="op">=</span> pd.read_csv(os.path.join(output_dir, <span class="st">"peve_normalized_expression.csv"</span>))</span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>ptua_norm <span class="op">=</span> pd.read_csv(os.path.join(output_dir, <span class="st">"ptua_normalized_expression.csv"</span>))</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Loaded normalized data:"</span>)</span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Apul: </span><span class="sc">{</span>apul_norm<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Peve: </span><span class="sc">{</span>peve_norm<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)  </span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Ptua: </span><span class="sc">{</span>ptua_norm<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Check which genes are common across all species</span></span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>apul_genes <span class="op">=</span> <span class="bu">set</span>(apul_norm[<span class="st">'group_id'</span>])</span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>peve_genes <span class="op">=</span> <span class="bu">set</span>(peve_norm[<span class="st">'group_id'</span>])</span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a>ptua_genes <span class="op">=</span> <span class="bu">set</span>(ptua_norm[<span class="st">'group_id'</span>])</span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a>common_genes <span class="op">=</span> apul_genes <span class="op">&amp;</span> peve_genes <span class="op">&amp;</span> ptua_genes</span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Common genes across all species: </span><span class="sc">{</span><span class="bu">len</span>(common_genes)<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to common genes and align gene order</span></span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a>common_genes_list <span class="op">=</span> <span class="bu">sorted</span>(<span class="bu">list</span>(common_genes))</span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a>apul_common <span class="op">=</span> apul_norm[apul_norm[<span class="st">'group_id'</span>].isin(common_genes_list)].set_index(<span class="st">'group_id'</span>).reindex(common_genes_list)</span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true" tabindex="-1"></a>peve_common <span class="op">=</span> peve_norm[peve_norm[<span class="st">'group_id'</span>].isin(common_genes_list)].set_index(<span class="st">'group_id'</span>).reindex(common_genes_list)</span>
<span id="cb34-33"><a href="#cb34-33" aria-hidden="true" tabindex="-1"></a>ptua_common <span class="op">=</span> ptua_norm[ptua_norm[<span class="st">'group_id'</span>].isin(common_genes_list)].set_index(<span class="st">'group_id'</span>).reindex(common_genes_list)</span>
<span id="cb34-34"><a href="#cb34-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-35"><a href="#cb34-35" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Filtered to common genes:"</span>)</span>
<span id="cb34-36"><a href="#cb34-36" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Apul: </span><span class="sc">{</span>apul_common<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb34-37"><a href="#cb34-37" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Peve: </span><span class="sc">{</span>peve_common<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb34-38"><a href="#cb34-38" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Ptua: </span><span class="sc">{</span>ptua_common<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Working in output directory: ../output/13.00-multiomics-barnacle
## Loaded normalized data:
## Apul: (10223, 41)
## Peve: (10223, 39)
## Ptua: (10223, 40)
## 
## Common genes across all species: 10223
## 
## Filtered to common genes:
## Apul: (10223, 40)
## Peve: (10223, 38)
## Ptua: (10223, 39)</code></pre>
</section>
<section id="parse-sample-information" class="level2">
<h2 class="anchored" data-anchor-id="parse-sample-information">4.5 Parse sample information</h2>
<div class="sourceCode" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Parse sample names to extract sample information for each species independently</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> parse_species_samples(columns, species_name):</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>    <span class="co">"""Parse sample column names for a specific species"""</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>    sample_map <span class="op">=</span> {}</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>    sample_ids <span class="op">=</span> []</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>    timepoints <span class="op">=</span> <span class="bu">set</span>()</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> col <span class="kw">in</span> columns:</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Expected format: PREFIX.NUMBER.TP# (e.g., ACR.139.TP1, POR.216.TP1, POC.201.TP1)</span></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>        parts <span class="op">=</span> col.split(<span class="st">'.'</span>)</span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="bu">len</span>(parts) <span class="op">&gt;=</span> <span class="dv">3</span> <span class="kw">and</span> parts[<span class="dv">2</span>].startswith(<span class="st">'TP'</span>):</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>            prefix <span class="op">=</span> parts[<span class="dv">0</span>]</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>            numeric_id <span class="op">=</span> parts[<span class="dv">1</span>]  </span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>            timepoint <span class="op">=</span> <span class="bu">int</span>(parts[<span class="dv">2</span>][<span class="dv">2</span>:])  <span class="co"># e.g., 1 from TP1</span></span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>            sample_id <span class="op">=</span> <span class="ss">f"</span><span class="sc">{</span>prefix<span class="sc">}</span><span class="ss">.</span><span class="sc">{</span>numeric_id<span class="sc">}</span><span class="ss">"</span>  <span class="co"># e.g., "ACR.139"</span></span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>            sample_map[(sample_id, timepoint)] <span class="op">=</span> col</span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> sample_id <span class="kw">not</span> <span class="kw">in</span> sample_ids:</span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>                sample_ids.append(sample_id)</span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>            timepoints.add(timepoint)</span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"Warning: Could not parse column name: </span><span class="sc">{</span>col<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> sample_map, sample_ids, <span class="bu">sorted</span>(timepoints)</span>
<span id="cb36-25"><a href="#cb36-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-26"><a href="#cb36-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Parse sample information for each species independently</span></span>
<span id="cb36-27"><a href="#cb36-27" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Parsing sample information for each species..."</span>)</span>
<span id="cb36-28"><a href="#cb36-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-29"><a href="#cb36-29" aria-hidden="true" tabindex="-1"></a>species_data <span class="op">=</span> {</span>
<span id="cb36-30"><a href="#cb36-30" aria-hidden="true" tabindex="-1"></a>    <span class="st">'apul'</span>: apul_common,</span>
<span id="cb36-31"><a href="#cb36-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">'peve'</span>: peve_common, </span>
<span id="cb36-32"><a href="#cb36-32" aria-hidden="true" tabindex="-1"></a>    <span class="st">'ptua'</span>: ptua_common</span>
<span id="cb36-33"><a href="#cb36-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-34"><a href="#cb36-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-35"><a href="#cb36-35" aria-hidden="true" tabindex="-1"></a>species_info <span class="op">=</span> {}</span>
<span id="cb36-36"><a href="#cb36-36" aria-hidden="true" tabindex="-1"></a>all_timepoints <span class="op">=</span> <span class="bu">set</span>()</span>
<span id="cb36-37"><a href="#cb36-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-38"><a href="#cb36-38" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> species, data <span class="kw">in</span> species_data.items():</span>
<span id="cb36-39"><a href="#cb36-39" aria-hidden="true" tabindex="-1"></a>    sample_map, sample_ids, timepoints <span class="op">=</span> parse_species_samples(data.columns, species)</span>
<span id="cb36-40"><a href="#cb36-40" aria-hidden="true" tabindex="-1"></a>    species_info[species] <span class="op">=</span> {</span>
<span id="cb36-41"><a href="#cb36-41" aria-hidden="true" tabindex="-1"></a>        <span class="st">'sample_map'</span>: sample_map,</span>
<span id="cb36-42"><a href="#cb36-42" aria-hidden="true" tabindex="-1"></a>        <span class="st">'sample_ids'</span>: sample_ids,</span>
<span id="cb36-43"><a href="#cb36-43" aria-hidden="true" tabindex="-1"></a>        <span class="st">'timepoints'</span>: timepoints,</span>
<span id="cb36-44"><a href="#cb36-44" aria-hidden="true" tabindex="-1"></a>        <span class="st">'n_samples'</span>: <span class="bu">len</span>(sample_ids)</span>
<span id="cb36-45"><a href="#cb36-45" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb36-46"><a href="#cb36-46" aria-hidden="true" tabindex="-1"></a>    all_timepoints.update(timepoints)</span>
<span id="cb36-47"><a href="#cb36-47" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-48"><a href="#cb36-48" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span>species<span class="sc">}</span><span class="ss">:"</span>)</span>
<span id="cb36-49"><a href="#cb36-49" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"  Samples: </span><span class="sc">{</span><span class="bu">len</span>(sample_ids)<span class="sc">}</span><span class="ss"> (</span><span class="sc">{</span>sample_ids[:<span class="dv">3</span>]<span class="sc">}</span><span class="ss">...)"</span>)</span>
<span id="cb36-50"><a href="#cb36-50" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"  Timepoints: </span><span class="sc">{</span>timepoints<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb36-51"><a href="#cb36-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-52"><a href="#cb36-52" aria-hidden="true" tabindex="-1"></a>common_timepoints <span class="op">=</span> <span class="bu">sorted</span>(<span class="bu">list</span>(all_timepoints))</span>
<span id="cb36-53"><a href="#cb36-53" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Timepoints found across all species: </span><span class="sc">{</span>common_timepoints<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb36-54"><a href="#cb36-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-55"><a href="#cb36-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Find the maximum number of samples to determine tensor dimensions</span></span>
<span id="cb36-56"><a href="#cb36-56" aria-hidden="true" tabindex="-1"></a>max_samples <span class="op">=</span> <span class="bu">max</span>(info[<span class="st">'n_samples'</span>] <span class="cf">for</span> info <span class="kw">in</span> species_info.values())</span>
<span id="cb36-57"><a href="#cb36-57" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Maximum samples in any species: </span><span class="sc">{</span>max_samples<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb36-58"><a href="#cb36-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-59"><a href="#cb36-59" aria-hidden="true" tabindex="-1"></a><span class="co"># Print detailed sample structure</span></span>
<span id="cb36-60"><a href="#cb36-60" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Detailed sample structure:"</span>)</span>
<span id="cb36-61"><a href="#cb36-61" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> species, info <span class="kw">in</span> species_info.items():</span>
<span id="cb36-62"><a href="#cb36-62" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="sc">{</span>species<span class="sc">}</span><span class="ss">: </span><span class="sc">{</span>info[<span class="st">'n_samples'</span>]<span class="sc">}</span><span class="ss"> samples × </span><span class="sc">{</span><span class="bu">len</span>(info[<span class="st">'timepoints'</span>])<span class="sc">}</span><span class="ss"> timepoints"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Parsing sample information for each species...
## apul:
##   Samples: 10 (['ACR.139', 'ACR.145', 'ACR.150']...)
##   Timepoints: [1, 2, 3, 4]
## peve:
##   Samples: 10 (['POR.216', 'POR.236', 'POR.245']...)
##   Timepoints: [1, 2, 3, 4]
## ptua:
##   Samples: 10 (['POC.201', 'POC.219', 'POC.222']...)
##   Timepoints: [1, 2, 3, 4]
## 
## Timepoints found across all species: [1, 2, 3, 4]
## Maximum samples in any species: 10
## 
## Detailed sample structure:
## apul: 10 samples × 4 timepoints
## peve: 10 samples × 4 timepoints
## ptua: 10 samples × 4 timepoints</code></pre>
</section>
<section id="create-3d-tensor-genes-species_samples-timepoints" class="level2">
<h2 class="anchored" data-anchor-id="create-3d-tensor-genes-species_samples-timepoints">4.6 Create 3D tensor (genes × species_samples × timepoints)</h2>
<div class="sourceCode" id="cb38"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a 3D tensor: genes × (species_samples) × timepoints</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="co"># This flattens species and samples into a single dimension that Barnacle can handle</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Creating 3D tensor by combining species and samples..."</span>)</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a><span class="co"># First, collect all actual sample-timepoint combinations that have data</span></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>all_sample_columns <span class="op">=</span> []</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>sample_labels <span class="op">=</span> []  <span class="co"># Track which sample belongs to which species</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>species_sample_map <span class="op">=</span> {}  <span class="co"># Map from combined index to (species, sample_idx, sample_id)</span></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>sample_idx <span class="op">=</span> <span class="dv">0</span></span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> species <span class="kw">in</span> [<span class="st">'apul'</span>, <span class="st">'peve'</span>, <span class="st">'ptua'</span>]:</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>    data <span class="op">=</span> species_data[species]</span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>    info <span class="op">=</span> species_info[species]</span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Processing </span><span class="sc">{</span>species<span class="sc">}</span><span class="ss">:"</span>)</span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> sample_id <span class="kw">in</span> info[<span class="st">'sample_ids'</span>]:</span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Check if this sample has data for any timepoint</span></span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a>        has_data <span class="op">=</span> <span class="va">False</span></span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a>        sample_timepoint_cols <span class="op">=</span> []</span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> timepoint <span class="kw">in</span> common_timepoints:</span>
<span id="cb38-23"><a href="#cb38-23" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (sample_id, timepoint) <span class="kw">in</span> info[<span class="st">'sample_map'</span>]:</span>
<span id="cb38-24"><a href="#cb38-24" aria-hidden="true" tabindex="-1"></a>                col_name <span class="op">=</span> info[<span class="st">'sample_map'</span>][(sample_id, timepoint)]</span>
<span id="cb38-25"><a href="#cb38-25" aria-hidden="true" tabindex="-1"></a>                sample_timepoint_cols.append(col_name)</span>
<span id="cb38-26"><a href="#cb38-26" aria-hidden="true" tabindex="-1"></a>                has_data <span class="op">=</span> <span class="va">True</span></span>
<span id="cb38-27"><a href="#cb38-27" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb38-28"><a href="#cb38-28" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> has_data:</span>
<span id="cb38-29"><a href="#cb38-29" aria-hidden="true" tabindex="-1"></a>            all_sample_columns.extend(sample_timepoint_cols)</span>
<span id="cb38-30"><a href="#cb38-30" aria-hidden="true" tabindex="-1"></a>            sample_labels.append(<span class="ss">f"</span><span class="sc">{</span>species<span class="sc">}</span><span class="ss">_</span><span class="sc">{</span>sample_id<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb38-31"><a href="#cb38-31" aria-hidden="true" tabindex="-1"></a>            species_sample_map[sample_idx] <span class="op">=</span> {</span>
<span id="cb38-32"><a href="#cb38-32" aria-hidden="true" tabindex="-1"></a>                <span class="st">'species'</span>: species,</span>
<span id="cb38-33"><a href="#cb38-33" aria-hidden="true" tabindex="-1"></a>                <span class="st">'sample_id'</span>: sample_id,</span>
<span id="cb38-34"><a href="#cb38-34" aria-hidden="true" tabindex="-1"></a>                <span class="st">'sample_idx_in_species'</span>: info[<span class="st">'sample_ids'</span>].index(sample_id)</span>
<span id="cb38-35"><a href="#cb38-35" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb38-36"><a href="#cb38-36" aria-hidden="true" tabindex="-1"></a>            sample_idx <span class="op">+=</span> <span class="dv">1</span></span>
<span id="cb38-37"><a href="#cb38-37" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  Added </span><span class="sc">{</span>sample_id<span class="sc">}</span><span class="ss"> with </span><span class="sc">{</span><span class="bu">len</span>(sample_timepoint_cols)<span class="sc">}</span><span class="ss"> timepoints"</span>)</span>
<span id="cb38-38"><a href="#cb38-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-39"><a href="#cb38-39" aria-hidden="true" tabindex="-1"></a>n_genes <span class="op">=</span> <span class="bu">len</span>(common_genes_list)</span>
<span id="cb38-40"><a href="#cb38-40" aria-hidden="true" tabindex="-1"></a>n_combined_samples <span class="op">=</span> <span class="bu">len</span>(sample_labels)</span>
<span id="cb38-41"><a href="#cb38-41" aria-hidden="true" tabindex="-1"></a>n_timepoints <span class="op">=</span> <span class="bu">len</span>(common_timepoints)</span>
<span id="cb38-42"><a href="#cb38-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-43"><a href="#cb38-43" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Creating 3D tensor with shape: (</span><span class="sc">{</span>n_genes<span class="sc">}</span><span class="ss">, </span><span class="sc">{</span>n_combined_samples<span class="sc">}</span><span class="ss">, </span><span class="sc">{</span>n_timepoints<span class="sc">}</span><span class="ss">)"</span>)</span>
<span id="cb38-44"><a href="#cb38-44" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Combined samples from all species: </span><span class="sc">{</span>n_combined_samples<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb38-45"><a href="#cb38-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-46"><a href="#cb38-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize tensor</span></span>
<span id="cb38-47"><a href="#cb38-47" aria-hidden="true" tabindex="-1"></a>tensor_3d <span class="op">=</span> np.full((n_genes, n_combined_samples, n_timepoints), np.nan)</span>
<span id="cb38-48"><a href="#cb38-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-49"><a href="#cb38-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill tensor</span></span>
<span id="cb38-50"><a href="#cb38-50" aria-hidden="true" tabindex="-1"></a>filled_count <span class="op">=</span> <span class="dv">0</span></span>
<span id="cb38-51"><a href="#cb38-51" aria-hidden="true" tabindex="-1"></a>missing_count <span class="op">=</span> <span class="dv">0</span></span>
<span id="cb38-52"><a href="#cb38-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-53"><a href="#cb38-53" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> combined_idx, sample_label <span class="kw">in</span> <span class="bu">enumerate</span>(sample_labels):</span>
<span id="cb38-54"><a href="#cb38-54" aria-hidden="true" tabindex="-1"></a>    species_info_map <span class="op">=</span> species_sample_map[combined_idx]</span>
<span id="cb38-55"><a href="#cb38-55" aria-hidden="true" tabindex="-1"></a>    species <span class="op">=</span> species_info_map[<span class="st">'species'</span>]</span>
<span id="cb38-56"><a href="#cb38-56" aria-hidden="true" tabindex="-1"></a>    sample_id <span class="op">=</span> species_info_map[<span class="st">'sample_id'</span>]</span>
<span id="cb38-57"><a href="#cb38-57" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-58"><a href="#cb38-58" aria-hidden="true" tabindex="-1"></a>    data <span class="op">=</span> species_data[species]</span>
<span id="cb38-59"><a href="#cb38-59" aria-hidden="true" tabindex="-1"></a>    info <span class="op">=</span> species_info[species]</span>
<span id="cb38-60"><a href="#cb38-60" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-61"><a href="#cb38-61" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> time_idx, timepoint <span class="kw">in</span> <span class="bu">enumerate</span>(common_timepoints):</span>
<span id="cb38-62"><a href="#cb38-62" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (sample_id, timepoint) <span class="kw">in</span> info[<span class="st">'sample_map'</span>]:</span>
<span id="cb38-63"><a href="#cb38-63" aria-hidden="true" tabindex="-1"></a>            col_name <span class="op">=</span> info[<span class="st">'sample_map'</span>][(sample_id, timepoint)]</span>
<span id="cb38-64"><a href="#cb38-64" aria-hidden="true" tabindex="-1"></a>            tensor_3d[:, combined_idx, time_idx] <span class="op">=</span> data[col_name].values</span>
<span id="cb38-65"><a href="#cb38-65" aria-hidden="true" tabindex="-1"></a>            filled_count <span class="op">+=</span> <span class="dv">1</span></span>
<span id="cb38-66"><a href="#cb38-66" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb38-67"><a href="#cb38-67" aria-hidden="true" tabindex="-1"></a>            missing_count <span class="op">+=</span> <span class="dv">1</span></span>
<span id="cb38-68"><a href="#cb38-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-69"><a href="#cb38-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Check tensor statistics</span></span>
<span id="cb38-70"><a href="#cb38-70" aria-hidden="true" tabindex="-1"></a>n_missing <span class="op">=</span> np.<span class="bu">sum</span>(np.isnan(tensor_3d))</span>
<span id="cb38-71"><a href="#cb38-71" aria-hidden="true" tabindex="-1"></a>n_total <span class="op">=</span> tensor_3d.size</span>
<span id="cb38-72"><a href="#cb38-72" aria-hidden="true" tabindex="-1"></a>n_finite <span class="op">=</span> np.<span class="bu">sum</span>(np.isfinite(tensor_3d))</span>
<span id="cb38-73"><a href="#cb38-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-74"><a href="#cb38-74" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">=== TENSOR STATISTICS ==="</span>)</span>
<span id="cb38-75"><a href="#cb38-75" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Tensor shape: </span><span class="sc">{</span>tensor_3d<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb38-76"><a href="#cb38-76" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Total elements: </span><span class="sc">{</span>n_total<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb38-77"><a href="#cb38-77" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Finite values: </span><span class="sc">{</span>n_finite<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb38-78"><a href="#cb38-78" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Missing/NaN values: </span><span class="sc">{</span>n_missing<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb38-79"><a href="#cb38-79" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Missing percentage: </span><span class="sc">{</span>n_missing <span class="op">/</span> n_total <span class="op">*</span> <span class="dv">100</span><span class="sc">:.2f}</span><span class="ss">%"</span>)</span>
<span id="cb38-80"><a href="#cb38-80" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Filled </span><span class="sc">{</span>filled_count<span class="sc">}</span><span class="ss"> sample-timepoint combinations"</span>)</span>
<span id="cb38-81"><a href="#cb38-81" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Missing </span><span class="sc">{</span>missing_count<span class="sc">}</span><span class="ss"> sample-timepoint combinations"</span>)</span>
<span id="cb38-82"><a href="#cb38-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-83"><a href="#cb38-83" aria-hidden="true" tabindex="-1"></a><span class="co"># Check non-zero values among finite values</span></span>
<span id="cb38-84"><a href="#cb38-84" aria-hidden="true" tabindex="-1"></a>finite_values <span class="op">=</span> tensor_3d[np.isfinite(tensor_3d)]</span>
<span id="cb38-85"><a href="#cb38-85" aria-hidden="true" tabindex="-1"></a>n_nonzero <span class="op">=</span> np.<span class="bu">sum</span>(finite_values <span class="op">!=</span> <span class="dv">0</span>)</span>
<span id="cb38-86"><a href="#cb38-86" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Non-zero finite values: </span><span class="sc">{</span>n_nonzero<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb38-87"><a href="#cb38-87" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Zero finite values: </span><span class="sc">{</span><span class="bu">len</span>(finite_values) <span class="op">-</span> n_nonzero<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb38-88"><a href="#cb38-88" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Sparsity among finite values: </span><span class="sc">{</span>(<span class="bu">len</span>(finite_values) <span class="op">-</span> n_nonzero) <span class="op">/</span> <span class="bu">len</span>(finite_values) <span class="op">*</span> <span class="dv">100</span><span class="sc">:.2f}</span><span class="ss">%"</span>)</span>
<span id="cb38-89"><a href="#cb38-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-90"><a href="#cb38-90" aria-hidden="true" tabindex="-1"></a><span class="co"># Save sample mapping for later interpretation</span></span>
<span id="cb38-91"><a href="#cb38-91" aria-hidden="true" tabindex="-1"></a>sample_mapping <span class="op">=</span> pd.DataFrame([</span>
<span id="cb38-92"><a href="#cb38-92" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb38-93"><a href="#cb38-93" aria-hidden="true" tabindex="-1"></a>        <span class="st">'combined_index'</span>: i,</span>
<span id="cb38-94"><a href="#cb38-94" aria-hidden="true" tabindex="-1"></a>        <span class="st">'sample_label'</span>: label,</span>
<span id="cb38-95"><a href="#cb38-95" aria-hidden="true" tabindex="-1"></a>        <span class="st">'species'</span>: species_sample_map[i][<span class="st">'species'</span>],</span>
<span id="cb38-96"><a href="#cb38-96" aria-hidden="true" tabindex="-1"></a>        <span class="st">'sample_id'</span>: species_sample_map[i][<span class="st">'sample_id'</span>]</span>
<span id="cb38-97"><a href="#cb38-97" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb38-98"><a href="#cb38-98" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i, label <span class="kw">in</span> <span class="bu">enumerate</span>(sample_labels)</span>
<span id="cb38-99"><a href="#cb38-99" aria-hidden="true" tabindex="-1"></a>])</span>
<span id="cb38-100"><a href="#cb38-100" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Sample mapping:"</span>)</span>
<span id="cb38-101"><a href="#cb38-101" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(sample_mapping.head(<span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Creating 3D tensor by combining species and samples...
## 
## Processing apul:
##   Added ACR.139 with 4 timepoints
##   Added ACR.145 with 4 timepoints
##   Added ACR.150 with 4 timepoints
##   Added ACR.173 with 4 timepoints
##   Added ACR.186 with 4 timepoints
##   Added ACR.225 with 4 timepoints
##   Added ACR.229 with 4 timepoints
##   Added ACR.237 with 4 timepoints
##   Added ACR.244 with 4 timepoints
##   Added ACR.265 with 4 timepoints
## 
## Processing peve:
##   Added POR.216 with 4 timepoints
##   Added POR.236 with 2 timepoints
##   Added POR.245 with 4 timepoints
##   Added POR.260 with 4 timepoints
##   Added POR.262 with 4 timepoints
##   Added POR.69 with 4 timepoints
##   Added POR.72 with 4 timepoints
##   Added POR.73 with 4 timepoints
##   Added POR.74 with 4 timepoints
##   Added POR.83 with 4 timepoints
## 
## Processing ptua:
##   Added POC.201 with 3 timepoints
##   Added POC.219 with 4 timepoints
##   Added POC.222 with 4 timepoints
##   Added POC.255 with 4 timepoints
##   Added POC.259 with 4 timepoints
##   Added POC.40 with 4 timepoints
##   Added POC.42 with 4 timepoints
##   Added POC.52 with 4 timepoints
##   Added POC.53 with 4 timepoints
##   Added POC.57 with 4 timepoints
## 
## Creating 3D tensor with shape: (10223, 30, 4)
## Combined samples from all species: 30
## 
## === TENSOR STATISTICS ===
## Tensor shape: (10223, 30, 4)
## Total elements: 1226760
## Finite values: 1196091
## Missing/NaN values: 30669
## Missing percentage: 2.50%
## Filled 117 sample-timepoint combinations
## Missing 3 sample-timepoint combinations
## Non-zero finite values: 1111605
## Zero finite values: 84486
## Sparsity among finite values: 7.06%
## 
## Sample mapping:
##    combined_index  sample_label species sample_id
## 0               0  apul_ACR.139    apul   ACR.139
## 1               1  apul_ACR.145    apul   ACR.145
## 2               2  apul_ACR.150    apul   ACR.150
## 3               3  apul_ACR.173    apul   ACR.173
## 4               4  apul_ACR.186    apul   ACR.186
## 5               5  apul_ACR.225    apul   ACR.225
## 6               6  apul_ACR.229    apul   ACR.229
## 7               7  apul_ACR.237    apul   ACR.237
## 8               8  apul_ACR.244    apul   ACR.244
## 9               9  apul_ACR.265    apul   ACR.265</code></pre>
</section>
<section id="run-barnacle-sparse-tensor-decomposition" class="level2">
<h2 class="anchored" data-anchor-id="run-barnacle-sparse-tensor-decomposition">4.7 Run Barnacle sparse tensor decomposition</h2>
<div class="sourceCode" id="cb40"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> barnacle.decomposition <span class="im">import</span> SparseCP</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Handle missing values by filling with zeros (could also use mean imputation)</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>tensor_filled <span class="op">=</span> np.nan_to_num(tensor_3d, nan<span class="op">=</span><span class="fl">0.0</span>)</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Running Barnacle sparse tensor decomposition..."</span>)</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Input tensor shape: </span><span class="sc">{</span>tensor_filled<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up sparse CP decomposition parameters</span></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>rank <span class="op">=</span> <span class="dv">5</span>  <span class="co"># Number of components to extract</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>lambdas <span class="op">=</span> [<span class="fl">0.1</span>, <span class="fl">0.0</span>, <span class="fl">0.1</span>]  <span class="co"># Sparsity penalties for [genes, samples, timepoints]</span></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Create model</span></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>model <span class="op">=</span> SparseCP(</span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>    rank<span class="op">=</span>rank,</span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>    lambdas<span class="op">=</span>lambdas,</span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>    nonneg_modes<span class="op">=</span>[<span class="dv">0</span>],  <span class="co"># Non-negative gene loadings only</span></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>    n_initializations<span class="op">=</span><span class="dv">3</span>,  <span class="co"># Reduced for faster testing</span></span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>    random_state<span class="op">=</span><span class="dv">42</span></span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit the model</span></span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Fitting sparse CP decomposition..."</span>)</span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a><span class="cf">try</span>:</span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a>    decomposition <span class="op">=</span> model.fit_transform(tensor_filled, verbose<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Decomposition complete!"</span>)</span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Converged: </span><span class="sc">{</span><span class="bu">hasattr</span>(model, <span class="st">'converged_'</span>) <span class="kw">and</span> model<span class="sc">.</span>converged_<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="bu">hasattr</span>(model, <span class="st">'loss_'</span>) <span class="kw">and</span> model.loss_ <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Final loss: </span><span class="sc">{</span>model<span class="sc">.</span>loss_[<span class="op">-</span><span class="dv">1</span>]<span class="sc">:.6f}</span><span class="ss">"</span>)</span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Number of iterations: </span><span class="sc">{</span><span class="bu">len</span>(model.loss_)<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">"Loss information not available"</span>)</span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Examine factor matrices</span></span>
<span id="cb40-38"><a href="#cb40-38" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Factor matrix shapes:"</span>)</span>
<span id="cb40-39"><a href="#cb40-39" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i, factor <span class="kw">in</span> <span class="bu">enumerate</span>(decomposition.factors):</span>
<span id="cb40-40"><a href="#cb40-40" aria-hidden="true" tabindex="-1"></a>        factor_names <span class="op">=</span> [<span class="st">'Genes'</span>, <span class="st">'Species_Samples'</span>, <span class="st">'Timepoints'</span>]</span>
<span id="cb40-41"><a href="#cb40-41" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Mode </span><span class="sc">{</span>i<span class="sc">}</span><span class="ss"> (</span><span class="sc">{</span>factor_names[i]<span class="sc">}</span><span class="ss">): </span><span class="sc">{</span>factor<span class="sc">.</span>shape<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb40-42"><a href="#cb40-42" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb40-43"><a href="#cb40-43" aria-hidden="true" tabindex="-1"></a>    decomposition_success <span class="op">=</span> <span class="va">True</span></span>
<span id="cb40-44"><a href="#cb40-44" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-45"><a href="#cb40-45" aria-hidden="true" tabindex="-1"></a><span class="cf">except</span> <span class="pp">Exception</span> <span class="im">as</span> e:</span>
<span id="cb40-46"><a href="#cb40-46" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Error during decomposition: </span><span class="sc">{</span>e<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb40-47"><a href="#cb40-47" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Creating dummy decomposition for downstream code..."</span>)</span>
<span id="cb40-48"><a href="#cb40-48" aria-hidden="true" tabindex="-1"></a>    decomposition_success <span class="op">=</span> <span class="va">False</span></span>
<span id="cb40-49"><a href="#cb40-49" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-50"><a href="#cb40-50" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create dummy factors for error handling</span></span>
<span id="cb40-51"><a href="#cb40-51" aria-hidden="true" tabindex="-1"></a>    <span class="kw">class</span> DummyDecomposition:</span>
<span id="cb40-52"><a href="#cb40-52" aria-hidden="true" tabindex="-1"></a>        <span class="kw">def</span> <span class="fu">__init__</span>(<span class="va">self</span>):</span>
<span id="cb40-53"><a href="#cb40-53" aria-hidden="true" tabindex="-1"></a>            <span class="va">self</span>.factors <span class="op">=</span> [</span>
<span id="cb40-54"><a href="#cb40-54" aria-hidden="true" tabindex="-1"></a>                np.random.rand(<span class="bu">len</span>(common_genes_list), rank),  <span class="co"># genes</span></span>
<span id="cb40-55"><a href="#cb40-55" aria-hidden="true" tabindex="-1"></a>                np.random.rand(<span class="bu">len</span>(sample_labels), rank),      <span class="co"># samples</span></span>
<span id="cb40-56"><a href="#cb40-56" aria-hidden="true" tabindex="-1"></a>                np.random.rand(<span class="bu">len</span>(common_timepoints), rank)   <span class="co"># timepoints</span></span>
<span id="cb40-57"><a href="#cb40-57" aria-hidden="true" tabindex="-1"></a>            ]</span>
<span id="cb40-58"><a href="#cb40-58" aria-hidden="true" tabindex="-1"></a>            <span class="va">self</span>.weights <span class="op">=</span> np.ones(rank)</span>
<span id="cb40-59"><a href="#cb40-59" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-60"><a href="#cb40-60" aria-hidden="true" tabindex="-1"></a>    decomposition <span class="op">=</span> DummyDecomposition()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Running Barnacle sparse tensor decomposition...
## Input tensor shape: (10223, 30, 4)
## Fitting sparse CP decomposition...
## 
## Beginning initialization 1 of 3
## 
## Beginning initialization 2 of 3
## 
## Beginning initialization 3 of 3
## 
## Decomposition complete!
## Converged: False
## Final loss: 850410.240223
## Number of iterations: 1000
## 
## Factor matrix shapes:
## Mode 0 (Genes): (10223, 5)
## Mode 1 (Species_Samples): (30, 5)
## Mode 2 (Timepoints): (4, 5)</code></pre>
</section>
<section id="analyze-and-interpret-results" class="level2">
<h2 class="anchored" data-anchor-id="analyze-and-interpret-results">4.8 Analyze and interpret results</h2>
<div class="sourceCode" id="cb42"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> decomposition_success:</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Extract factor matrices</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>    gene_factors <span class="op">=</span> decomposition.factors[<span class="dv">0</span>]  <span class="co"># genes × components</span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>    sample_factors <span class="op">=</span> decomposition.factors[<span class="dv">1</span>]  <span class="co"># combined_samples × components  </span></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>    time_factors <span class="op">=</span> decomposition.factors[<span class="dv">2</span>]  <span class="co"># timepoints × components</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"=== BARNACLE DECOMPOSITION RESULTS ==="</span>)</span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Number of components: </span><span class="sc">{</span>rank<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Component weights: </span><span class="sc">{</span>decomposition<span class="sc">.</span>weights<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">=== TIMEPOINT LOADINGS ==="</span>)</span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>    time_df <span class="op">=</span> pd.DataFrame(</span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>        time_factors,</span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>        index<span class="op">=</span>[<span class="ss">f'TP</span><span class="sc">{</span>tp<span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> tp <span class="kw">in</span> common_timepoints],</span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>        columns<span class="op">=</span>[<span class="ss">f'Component_</span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(rank)]</span>
<span id="cb42-16"><a href="#cb42-16" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb42-17"><a href="#cb42-17" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(time_df)</span>
<span id="cb42-18"><a href="#cb42-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-19"><a href="#cb42-19" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">=== SAMPLE LOADINGS BY SPECIES ==="</span>)</span>
<span id="cb42-20"><a href="#cb42-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Show sample loadings grouped by species</span></span>
<span id="cb42-21"><a href="#cb42-21" aria-hidden="true" tabindex="-1"></a>    sample_df_full <span class="op">=</span> pd.DataFrame(</span>
<span id="cb42-22"><a href="#cb42-22" aria-hidden="true" tabindex="-1"></a>        sample_factors,</span>
<span id="cb42-23"><a href="#cb42-23" aria-hidden="true" tabindex="-1"></a>        index<span class="op">=</span>sample_labels,</span>
<span id="cb42-24"><a href="#cb42-24" aria-hidden="true" tabindex="-1"></a>        columns<span class="op">=</span>[<span class="ss">f'Component_</span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(rank)]</span>
<span id="cb42-25"><a href="#cb42-25" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb42-26"><a href="#cb42-26" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-27"><a href="#cb42-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add species information</span></span>
<span id="cb42-28"><a href="#cb42-28" aria-hidden="true" tabindex="-1"></a>    sample_df_full[<span class="st">'Species'</span>] <span class="op">=</span> [species_sample_map[i][<span class="st">'species'</span>] <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="bu">len</span>(sample_labels))]</span>
<span id="cb42-29"><a href="#cb42-29" aria-hidden="true" tabindex="-1"></a>    sample_df_full[<span class="st">'Sample_ID'</span>] <span class="op">=</span> [species_sample_map[i][<span class="st">'sample_id'</span>] <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="bu">len</span>(sample_labels))]</span>
<span id="cb42-30"><a href="#cb42-30" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb42-31"><a href="#cb42-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Show by species</span></span>
<span id="cb42-32"><a href="#cb42-32" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> species <span class="kw">in</span> [<span class="st">'apul'</span>, <span class="st">'peve'</span>, <span class="st">'ptua'</span>]:</span>
<span id="cb42-33"><a href="#cb42-33" aria-hidden="true" tabindex="-1"></a>        species_samples <span class="op">=</span> sample_df_full[sample_df_full[<span class="st">'Species'</span>] <span class="op">==</span> species]</span>
<span id="cb42-34"><a href="#cb42-34" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="sc">{</span>species<span class="sc">.</span>upper()<span class="sc">}</span><span class="ss"> samples:"</span>)</span>
<span id="cb42-35"><a href="#cb42-35" aria-hidden="true" tabindex="-1"></a>        component_cols <span class="op">=</span> [<span class="ss">f'Component_</span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(rank)]</span>
<span id="cb42-36"><a href="#cb42-36" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(species_samples[component_cols <span class="op">+</span> [<span class="st">'Sample_ID'</span>]].head())</span>
<span id="cb42-37"><a href="#cb42-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-38"><a href="#cb42-38" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Find top orthologs for each component</span></span>
<span id="cb42-39"><a href="#cb42-39" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">=== TOP ORTHOLOGS PER COMPONENT ==="</span>)</span>
<span id="cb42-40"><a href="#cb42-40" aria-hidden="true" tabindex="-1"></a>    n_top_genes <span class="op">=</span> <span class="dv">5</span></span>
<span id="cb42-41"><a href="#cb42-41" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> comp <span class="kw">in</span> <span class="bu">range</span>(rank):</span>
<span id="cb42-42"><a href="#cb42-42" aria-hidden="true" tabindex="-1"></a>        gene_loadings <span class="op">=</span> gene_factors[:, comp]</span>
<span id="cb42-43"><a href="#cb42-43" aria-hidden="true" tabindex="-1"></a>        top_gene_indices <span class="op">=</span> np.argsort(np.<span class="bu">abs</span>(gene_loadings))[<span class="op">-</span>n_top_genes:][::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb42-44"><a href="#cb42-44" aria-hidden="true" tabindex="-1"></a>        top_genes <span class="op">=</span> [common_genes_list[i] <span class="cf">for</span> i <span class="kw">in</span> top_gene_indices]</span>
<span id="cb42-45"><a href="#cb42-45" aria-hidden="true" tabindex="-1"></a>        top_loadings <span class="op">=</span> gene_loadings[top_gene_indices]</span>
<span id="cb42-46"><a href="#cb42-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-47"><a href="#cb42-47" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Component </span><span class="sc">{</span>comp<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss"> (weight: </span><span class="sc">{</span>decomposition<span class="sc">.</span>weights[comp]<span class="sc">:.3f}</span><span class="ss">):"</span>)</span>
<span id="cb42-48"><a href="#cb42-48" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> gene, loading <span class="kw">in</span> <span class="bu">zip</span>(top_genes, top_loadings):</span>
<span id="cb42-49"><a href="#cb42-49" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  </span><span class="sc">{</span>gene<span class="sc">}</span><span class="ss">: </span><span class="sc">{</span>loading<span class="sc">:.3f}</span><span class="ss">"</span>)</span>
<span id="cb42-50"><a href="#cb42-50" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb42-51"><a href="#cb42-51" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Analyze species patterns in sample factors</span></span>
<span id="cb42-52"><a href="#cb42-52" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">=== SPECIES PATTERNS IN SAMPLE FACTORS ==="</span>)</span>
<span id="cb42-53"><a href="#cb42-53" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> comp <span class="kw">in</span> <span class="bu">range</span>(rank):</span>
<span id="cb42-54"><a href="#cb42-54" aria-hidden="true" tabindex="-1"></a>        comp_col <span class="op">=</span> <span class="ss">f'Component_</span><span class="sc">{</span>comp<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span></span>
<span id="cb42-55"><a href="#cb42-55" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Component </span><span class="sc">{</span>comp<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss"> - Average loadings by species:"</span>)</span>
<span id="cb42-56"><a href="#cb42-56" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> species <span class="kw">in</span> [<span class="st">'apul'</span>, <span class="st">'peve'</span>, <span class="st">'ptua'</span>]:</span>
<span id="cb42-57"><a href="#cb42-57" aria-hidden="true" tabindex="-1"></a>            species_loadings <span class="op">=</span> sample_df_full[sample_df_full[<span class="st">'Species'</span>] <span class="op">==</span> species][comp_col]</span>
<span id="cb42-58"><a href="#cb42-58" aria-hidden="true" tabindex="-1"></a>            avg_loading <span class="op">=</span> species_loadings.mean()</span>
<span id="cb42-59"><a href="#cb42-59" aria-hidden="true" tabindex="-1"></a>            std_loading <span class="op">=</span> species_loadings.std()</span>
<span id="cb42-60"><a href="#cb42-60" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="ss">f"  </span><span class="sc">{</span>species<span class="sc">}</span><span class="ss">: </span><span class="sc">{</span>avg_loading<span class="sc">:.3f}</span><span class="ss"> ± </span><span class="sc">{</span>std_loading<span class="sc">:.3f}</span><span class="ss">"</span>)</span>
<span id="cb42-61"><a href="#cb42-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-62"><a href="#cb42-62" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb42-63"><a href="#cb42-63" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Decomposition failed - skipping detailed analysis"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## === BARNACLE DECOMPOSITION RESULTS ===
## Number of components: 5
## Component weights: [1. 1. 1. 1. 1.]
## 
## === TIMEPOINT LOADINGS ===
##      Component_1  Component_2  Component_3  Component_4  Component_5
## TP1    -7.701441     7.248857     6.407210    -1.603194     9.432116
## TP2    -6.501151     7.048609     6.483539     2.348303     5.935557
## TP3     0.751937     6.083226     6.649038     4.224872     4.126090
## TP4     0.715524     6.361256     5.878711    -6.434467    13.710867
## 
## === SAMPLE LOADINGS BY SPECIES ===
## 
## APUL samples:
##               Component_1  Component_2  ...  Component_5  Sample_ID
## apul_ACR.139    -0.001308     0.055781  ...     0.286177    ACR.139
## apul_ACR.145    -0.020095     0.025637  ...     0.313320    ACR.145
## apul_ACR.150     0.113907     0.064111  ...     0.299476    ACR.150
## apul_ACR.173    -0.008487     0.034594  ...     0.300150    ACR.173
## apul_ACR.186     0.056992     0.048160  ...     0.306427    ACR.186
## 
## [5 rows x 6 columns]
## 
## PEVE samples:
##               Component_1  Component_2  ...  Component_5  Sample_ID
## peve_POR.216     0.127030     0.326041  ...     0.047131    POR.216
## peve_POR.236    -0.816268     0.046569  ...     0.006104    POR.236
## peve_POR.245     0.046398     0.308702  ...     0.042404    POR.245
## peve_POR.260     0.073880     0.313210  ...     0.040011    POR.260
## peve_POR.262     0.143507     0.317012  ...     0.057350    POR.262
## 
## [5 rows x 6 columns]
## 
## PTUA samples:
##               Component_1  Component_2  ...  Component_5  Sample_ID
## ptua_POC.201    -0.369146    -0.024078  ...     0.026998    POC.201
## ptua_POC.219    -0.079287     0.004088  ...     0.051066    POC.219
## ptua_POC.222    -0.001975     0.036936  ...     0.060317    POC.222
## ptua_POC.255    -0.016951     0.027903  ...     0.044414    POC.255
## ptua_POC.259    -0.057884     0.022418  ...     0.070374    POC.259
## 
## [5 rows x 6 columns]
## 
## === TOP ORTHOLOGS PER COMPONENT ===
## 
## Component 1 (weight: 1.000):
##   OG_09295: 2.483
##   OG_02088: 2.048
##   OG_04999: 2.039
##   OG_09247: 2.031
##   OG_02502: 1.911
## 
## Component 2 (weight: 1.000):
##   OG_02502: 6.268
##   OG_09295: 6.169
##   OG_02088: 6.153
##   OG_09247: 6.115
##   OG_01995: 5.971
## 
## Component 3 (weight: 1.000):
##   OG_01396: 5.810
##   OG_00843: 5.762
##   OG_09596: 5.717
##   OG_09075: 5.549
##   OG_07200: 5.528
## 
## Component 4 (weight: 1.000):
##   OG_07251: 7.120
##   OG_02560: 6.711
##   OG_08795: 6.661
##   OG_02331: 6.567
##   OG_09596: 6.558
## 
## Component 5 (weight: 1.000):
##   OG_09596: 7.402
##   OG_01995: 7.172
##   OG_07251: 7.136
##   OG_02560: 7.057
##   OG_09075: 7.022
## 
## === SPECIES PATTERNS IN SAMPLE FACTORS ===
## 
## Component 1 - Average loadings by species:
##   apul: 0.029 ± 0.043
##   peve: 0.019 ± 0.295
##   ptua: -0.026 ± 0.142
## 
## Component 2 - Average loadings by species:
##   apul: 0.039 ± 0.013
##   peve: 0.300 ± 0.090
##   ptua: 0.027 ± 0.024
## 
## Component 3 - Average loadings by species:
##   apul: -0.190 ± 0.015
##   peve: -0.039 ± 0.014
##   ptua: 0.248 ± 0.023
## 
## Component 4 - Average loadings by species:
##   apul: 0.307 ± 0.020
##   peve: 0.036 ± 0.018
##   ptua: 0.045 ± 0.042
## 
## Component 5 - Average loadings by species:
##   apul: 0.311 ± 0.013
##   peve: 0.029 ± 0.017
##   ptua: 0.045 ± 0.015</code></pre>
</section>
<section id="summary" class="level2">
<h2 class="anchored" data-anchor-id="summary">4.9 Summary</h2>
<div class="sourceCode" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"=== BARNACLE ANALYSIS SUMMARY ===</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"1. Data Preprocessing:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"   - Loaded ortholog expression data for 3 species</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"   - Normalized using sctransform for count data</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"   - Filtered to genes common across all species</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"2. Tensor Construction:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"   - Created 3D tensor: genes × (species_samples) × timepoints</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"   - Combined all species samples into single dimension</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"   - Dimensions based on timeseries experimental design</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"3. Sparse Tensor Decomposition:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"   - Applied barnacle SparseCP with sparsity constraints</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"   - Extracted latent factors representing:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"     * Gene co-expression patterns</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"     * Combined sample-species relationships</span><span class="sc">\n</span><span class="st">"</span>) </span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"     * Temporal dynamics</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"4. Results Generated:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"   - Factor matrices for each tensor mode</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"   - Component weights indicating importance</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"   - Top contributing genes per component</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb44-24"><a href="#cb44-24" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"   - Temporal and species loadings</span><span class="sc">\n\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## === BARNACLE ANALYSIS SUMMARY ===
## 
## 1. Data Preprocessing:
##    - Loaded ortholog expression data for 3 species
##    - Normalized using sctransform for count data
##    - Filtered to genes common across all species
## 
## 2. Tensor Construction:
##    - Created 3D tensor: genes × (species_samples) × timepoints
##    - Combined all species samples into single dimension
##    - Dimensions based on timeseries experimental design
## 
## 3. Sparse Tensor Decomposition:
##    - Applied barnacle SparseCP with sparsity constraints
##    - Extracted latent factors representing:
##      * Gene co-expression patterns
##      * Combined sample-species relationships
##      * Temporal dynamics
## 
## 4. Results Generated:
##    - Factor matrices for each tensor mode
##    - Component weights indicating importance
##    - Top contributing genes per component
##    - Temporal and species loadings</code></pre>
</section>
<section id="save-results" class="level2">
<h2 class="anchored" data-anchor-id="save-results">4.10 Save results</h2>
<div class="sourceCode" id="cb46"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save tensor data</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>tensor_file <span class="op">=</span> os.path.join(output_dir, <span class="st">"multiomics_tensor.npy"</span>)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>np.save(tensor_file, tensor_filled)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Saved tensor to: </span><span class="sc">{</span>tensor_file<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Save factor matrices</span></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>factors_dir <span class="op">=</span> os.path.join(output_dir, <span class="st">"barnacle_factors"</span>)</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>os.makedirs(factors_dir, exist_ok<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> decomposition_success:</span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Gene factors</span></span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>    gene_factor_df <span class="op">=</span> pd.DataFrame(</span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a>        gene_factors,</span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>        index<span class="op">=</span>common_genes_list,</span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>        columns<span class="op">=</span>[<span class="ss">f'Component_</span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(rank)]</span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a>    gene_factor_df.to_csv(os.path.join(factors_dir, <span class="st">"gene_factors.csv"</span>))</span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Sample factors with species information</span></span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a>    sample_factor_df <span class="op">=</span> pd.DataFrame(</span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a>        sample_factors,</span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a>        index<span class="op">=</span>sample_labels,</span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a>        columns<span class="op">=</span>[<span class="ss">f'Component_</span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(rank)]</span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb46-25"><a href="#cb46-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add metadata</span></span>
<span id="cb46-26"><a href="#cb46-26" aria-hidden="true" tabindex="-1"></a>    sample_factor_df[<span class="st">'Species'</span>] <span class="op">=</span> [species_sample_map[i][<span class="st">'species'</span>] <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="bu">len</span>(sample_labels))]</span>
<span id="cb46-27"><a href="#cb46-27" aria-hidden="true" tabindex="-1"></a>    sample_factor_df[<span class="st">'Sample_ID'</span>] <span class="op">=</span> [species_sample_map[i][<span class="st">'sample_id'</span>] <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="bu">len</span>(sample_labels))]</span>
<span id="cb46-28"><a href="#cb46-28" aria-hidden="true" tabindex="-1"></a>    sample_factor_df.to_csv(os.path.join(factors_dir, <span class="st">"sample_factors.csv"</span>))</span>
<span id="cb46-29"><a href="#cb46-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-30"><a href="#cb46-30" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Time factors</span></span>
<span id="cb46-31"><a href="#cb46-31" aria-hidden="true" tabindex="-1"></a>    time_df.to_csv(os.path.join(factors_dir, <span class="st">"time_factors.csv"</span>))</span>
<span id="cb46-32"><a href="#cb46-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-33"><a href="#cb46-33" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save component weights</span></span>
<span id="cb46-34"><a href="#cb46-34" aria-hidden="true" tabindex="-1"></a>    weights_df <span class="op">=</span> pd.DataFrame({</span>
<span id="cb46-35"><a href="#cb46-35" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Component'</span>: [<span class="ss">f'Component_</span><span class="sc">{</span>i<span class="op">+</span><span class="dv">1</span><span class="sc">}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(rank)],</span>
<span id="cb46-36"><a href="#cb46-36" aria-hidden="true" tabindex="-1"></a>        <span class="st">'Weight'</span>: decomposition.weights</span>
<span id="cb46-37"><a href="#cb46-37" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb46-38"><a href="#cb46-38" aria-hidden="true" tabindex="-1"></a>    weights_df.to_csv(os.path.join(factors_dir, <span class="st">"component_weights.csv"</span>), index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb46-39"><a href="#cb46-39" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb46-40"><a href="#cb46-40" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save sample mapping</span></span>
<span id="cb46-41"><a href="#cb46-41" aria-hidden="true" tabindex="-1"></a>    sample_mapping.to_csv(os.path.join(factors_dir, <span class="st">"sample_mapping.csv"</span>), index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb46-42"><a href="#cb46-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-43"><a href="#cb46-43" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save metadata</span></span>
<span id="cb46-44"><a href="#cb46-44" aria-hidden="true" tabindex="-1"></a>    metadata <span class="op">=</span> {</span>
<span id="cb46-45"><a href="#cb46-45" aria-hidden="true" tabindex="-1"></a>        <span class="st">'tensor_shape'</span>: tensor_filled.shape,</span>
<span id="cb46-46"><a href="#cb46-46" aria-hidden="true" tabindex="-1"></a>        <span class="st">'tensor_type'</span>: <span class="st">'3D_genes_species_samples_timepoints'</span>,</span>
<span id="cb46-47"><a href="#cb46-47" aria-hidden="true" tabindex="-1"></a>        <span class="st">'n_genes'</span>: <span class="bu">len</span>(common_genes_list),</span>
<span id="cb46-48"><a href="#cb46-48" aria-hidden="true" tabindex="-1"></a>        <span class="st">'n_combined_samples'</span>: <span class="bu">len</span>(sample_labels),</span>
<span id="cb46-49"><a href="#cb46-49" aria-hidden="true" tabindex="-1"></a>        <span class="st">'n_timepoints'</span>: <span class="bu">len</span>(common_timepoints),</span>
<span id="cb46-50"><a href="#cb46-50" aria-hidden="true" tabindex="-1"></a>        <span class="st">'rank'</span>: rank,</span>
<span id="cb46-51"><a href="#cb46-51" aria-hidden="true" tabindex="-1"></a>        <span class="st">'lambdas'</span>: lambdas,</span>
<span id="cb46-52"><a href="#cb46-52" aria-hidden="true" tabindex="-1"></a>        <span class="st">'decomposition_success'</span>: <span class="va">True</span>,</span>
<span id="cb46-53"><a href="#cb46-53" aria-hidden="true" tabindex="-1"></a>        <span class="st">'timepoint_order'</span>: common_timepoints,</span>
<span id="cb46-54"><a href="#cb46-54" aria-hidden="true" tabindex="-1"></a>        <span class="st">'species_info'</span>: {species: info[<span class="st">'n_samples'</span>] <span class="cf">for</span> species, info <span class="kw">in</span> species_info.items()},</span>
<span id="cb46-55"><a href="#cb46-55" aria-hidden="true" tabindex="-1"></a>        <span class="st">'gene_count'</span>: <span class="bu">len</span>(common_genes_list)</span>
<span id="cb46-56"><a href="#cb46-56" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb46-57"><a href="#cb46-57" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb46-58"><a href="#cb46-58" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="bu">hasattr</span>(model, <span class="st">'converged_'</span>):</span>
<span id="cb46-59"><a href="#cb46-59" aria-hidden="true" tabindex="-1"></a>        metadata[<span class="st">'converged'</span>] <span class="op">=</span> <span class="bu">bool</span>(model.converged_)</span>
<span id="cb46-60"><a href="#cb46-60" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="bu">hasattr</span>(model, <span class="st">'loss_'</span>) <span class="kw">and</span> model.loss_ <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb46-61"><a href="#cb46-61" aria-hidden="true" tabindex="-1"></a>        metadata[<span class="st">'final_loss'</span>] <span class="op">=</span> <span class="bu">float</span>(model.loss_[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb46-62"><a href="#cb46-62" aria-hidden="true" tabindex="-1"></a>        metadata[<span class="st">'n_iterations'</span>] <span class="op">=</span> <span class="bu">len</span>(model.loss_)</span>
<span id="cb46-63"><a href="#cb46-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-64"><a href="#cb46-64" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb46-65"><a href="#cb46-65" aria-hidden="true" tabindex="-1"></a>    metadata <span class="op">=</span> {</span>
<span id="cb46-66"><a href="#cb46-66" aria-hidden="true" tabindex="-1"></a>        <span class="st">'tensor_shape'</span>: tensor_filled.shape,</span>
<span id="cb46-67"><a href="#cb46-67" aria-hidden="true" tabindex="-1"></a>        <span class="st">'tensor_type'</span>: <span class="st">'3D_genes_species_samples_timepoints'</span>,</span>
<span id="cb46-68"><a href="#cb46-68" aria-hidden="true" tabindex="-1"></a>        <span class="st">'decomposition_success'</span>: <span class="va">False</span>,</span>
<span id="cb46-69"><a href="#cb46-69" aria-hidden="true" tabindex="-1"></a>        <span class="st">'error'</span>: <span class="st">'Decomposition failed'</span></span>
<span id="cb46-70"><a href="#cb46-70" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb46-71"><a href="#cb46-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-72"><a href="#cb46-72" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json</span>
<span id="cb46-73"><a href="#cb46-73" aria-hidden="true" tabindex="-1"></a><span class="cf">with</span> <span class="bu">open</span>(os.path.join(factors_dir, <span class="st">"metadata.json"</span>), <span class="st">'w'</span>) <span class="im">as</span> f:</span>
<span id="cb46-74"><a href="#cb46-74" aria-hidden="true" tabindex="-1"></a>    json.dump(metadata, f, indent<span class="op">=</span><span class="dv">2</span>)</span>
<span id="cb46-75"><a href="#cb46-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-76"><a href="#cb46-76" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"</span><span class="ch">\n</span><span class="ss">Saved all results to: </span><span class="sc">{</span>factors_dir<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb46-77"><a href="#cb46-77" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> decomposition_success:</span>
<span id="cb46-78"><a href="#cb46-78" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Files created:"</span>)</span>
<span id="cb46-79"><a href="#cb46-79" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"- gene_factors.csv: Gene loadings for each component"</span>)</span>
<span id="cb46-80"><a href="#cb46-80" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"- sample_factors.csv: Sample loadings for each component (with species info)"</span>)</span>
<span id="cb46-81"><a href="#cb46-81" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"- time_factors.csv: Timepoint loadings for each component"</span>)</span>
<span id="cb46-82"><a href="#cb46-82" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"- component_weights.csv: Component importance weights"</span>)</span>
<span id="cb46-83"><a href="#cb46-83" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"- sample_mapping.csv: Mapping of samples to species"</span>)</span>
<span id="cb46-84"><a href="#cb46-84" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"- metadata.json: Analysis metadata and parameters"</span>)</span>
<span id="cb46-85"><a href="#cb46-85" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb46-86"><a href="#cb46-86" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Only metadata saved due to decomposition failure"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Saved tensor to: ../output/13.00-multiomics-barnacle/multiomics_tensor.npy
## 
## Saved all results to: ../output/13.00-multiomics-barnacle/barnacle_factors
## Files created:
## - gene_factors.csv: Gene loadings for each component
## - sample_factors.csv: Sample loadings for each component (with species info)
## - time_factors.csv: Timepoint loadings for each component
## - component_weights.csv: Component importance weights
## - sample_mapping.csv: Mapping of samples to species
## - metadata.json: Analysis metadata and parameters</code></pre>
</section>
<section id="visualize-barnacle-results" class="level2">
<h2 class="anchored" data-anchor-id="visualize-barnacle-results">4.11 Visualize Barnacle results</h2>
<section id="visualization-setup" class="level3">
<h3 class="anchored" data-anchor-id="visualization-setup">4.11.1 Visualization setup</h3>
<div class="sourceCode" id="cb48"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Setup for visualizations: load libs, directories and dataframes (shared by all small chunks)</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> sklearn.decomposition <span class="im">import</span> PCA</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure factors_dir is defined (created in previous chunk)</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a><span class="cf">try</span>:</span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>    factors_dir</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a><span class="cf">except</span> <span class="pp">NameError</span>:</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>    factors_dir <span class="op">=</span> os.path.join(output_dir, <span class="st">"barnacle_factors"</span>)</span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>fig_dir <span class="op">=</span> os.path.join(factors_dir, <span class="st">"figures"</span>)</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>os.makedirs(fig_dir, exist_ok<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Creating visualizations in:"</span>, fig_dir)</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> safe_load_df(var_name, csv_name):</span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a>    <span class="cf">try</span>:</span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> <span class="bu">globals</span>()[var_name]</span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a>    <span class="cf">except</span> <span class="pp">Exception</span>:</span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a>        path <span class="op">=</span> os.path.join(factors_dir, csv_name)</span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> os.path.exists(path):</span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a>            <span class="cf">return</span> pd.read_csv(path, index_col<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a>            <span class="cf">return</span> <span class="va">None</span></span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a>gene_factor_df <span class="op">=</span> safe_load_df(<span class="st">'gene_factor_df'</span>, <span class="st">'gene_factors.csv'</span>)</span>
<span id="cb48-31"><a href="#cb48-31" aria-hidden="true" tabindex="-1"></a>sample_factor_df <span class="op">=</span> safe_load_df(<span class="st">'sample_factor_df'</span>, <span class="st">'sample_factors.csv'</span>)</span>
<span id="cb48-32"><a href="#cb48-32" aria-hidden="true" tabindex="-1"></a>time_df <span class="op">=</span> safe_load_df(<span class="st">'time_df'</span>, <span class="st">'time_factors.csv'</span>)</span>
<span id="cb48-33"><a href="#cb48-33" aria-hidden="true" tabindex="-1"></a>weights_df <span class="op">=</span> <span class="va">None</span></span>
<span id="cb48-34"><a href="#cb48-34" aria-hidden="true" tabindex="-1"></a>weights_path <span class="op">=</span> os.path.join(factors_dir, <span class="st">'component_weights.csv'</span>)</span>
<span id="cb48-35"><a href="#cb48-35" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> os.path.exists(weights_path):</span>
<span id="cb48-36"><a href="#cb48-36" aria-hidden="true" tabindex="-1"></a>    weights_df <span class="op">=</span> pd.read_csv(weights_path)</span>
<span id="cb48-37"><a href="#cb48-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-38"><a href="#cb48-38" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">'Visualization setup complete'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## '../output/13.00-multiomics-barnacle/barnacle_factors'
## Creating visualizations in: ../output/13.00-multiomics-barnacle/barnacle_factors/figures
## Visualization setup complete</code></pre>
</section>
<section id="component-weights" class="level3">
<h3 class="anchored" data-anchor-id="component-weights">4.11.2 Component weights</h3>
<div class="sourceCode" id="cb50"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Component weights barplot</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> weights_df <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span> <span class="kw">and</span> <span class="kw">not</span> weights_df.empty:</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">try</span>:</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>        plt.figure(figsize<span class="op">=</span>(<span class="dv">6</span>,<span class="dv">4</span>))</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>        sns.barplot(x<span class="op">=</span><span class="st">'Component'</span>, y<span class="op">=</span><span class="st">'Weight'</span>, data<span class="op">=</span>weights_df, palette<span class="op">=</span><span class="st">'viridis'</span>)</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>        plt.xticks(rotation<span class="op">=</span><span class="dv">45</span>)</span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="st">'Component weights'</span>)</span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>        plt.tight_layout()</span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>        out <span class="op">=</span> os.path.join(fig_dir, <span class="st">'component_weights.png'</span>)</span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>        plt.savefig(out, dpi<span class="op">=</span><span class="dv">150</span>)</span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">'Saved:'</span>, out)</span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a>        <span class="cf">try</span>:</span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a>            plt.show()</span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a>        <span class="cf">except</span> <span class="pp">Exception</span>:</span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a>            <span class="cf">pass</span></span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a>        plt.close()</span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">except</span> <span class="pp">Exception</span> <span class="im">as</span> e:</span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">'Could not plot component weights:'</span>, e)</span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'No component_weights.csv found; skipping weights plot'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-barnacle_files/figure-gfm/viz-weights-1.png" width="576"></p>
</section>
<section id="timepoint-loadings" class="level3">
<h3 class="anchored" data-anchor-id="timepoint-loadings">4.11.3 Timepoint loadings</h3>
<div class="sourceCode" id="cb51"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Timepoint loadings (line plot)</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> time_df <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span> <span class="kw">and</span> <span class="kw">not</span> time_df.empty:</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">try</span>:</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>        plt.figure(figsize<span class="op">=</span>(<span class="dv">8</span>,<span class="dv">4</span>))</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>        x <span class="op">=</span> <span class="bu">list</span>(<span class="bu">range</span>(<span class="bu">len</span>(time_df.index)))</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> col <span class="kw">in</span> time_df.columns:</span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>            plt.plot(x, time_df[col].values, marker<span class="op">=</span><span class="st">'o'</span>, label<span class="op">=</span>col)</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>        plt.xticks(x, <span class="bu">list</span>(time_df.index), rotation<span class="op">=</span><span class="dv">45</span>)</span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>        plt.xlabel(<span class="st">'Timepoint'</span>)</span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>        plt.ylabel(<span class="st">'Loading'</span>)</span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a>        plt.legend(bbox_to_anchor<span class="op">=</span>(<span class="fl">1.02</span>,<span class="dv">1</span>), loc<span class="op">=</span><span class="st">'upper left'</span>)</span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="st">'Timepoint loadings per component'</span>)</span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a>        plt.tight_layout()</span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a>        out <span class="op">=</span> os.path.join(fig_dir, <span class="st">'time_loadings.png'</span>)</span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a>        plt.savefig(out, dpi<span class="op">=</span><span class="dv">150</span>)</span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">'Saved:'</span>, out)</span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a>        <span class="cf">try</span>:</span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a>            plt.show()</span>
<span id="cb51-19"><a href="#cb51-19" aria-hidden="true" tabindex="-1"></a>        <span class="cf">except</span> <span class="pp">Exception</span>:</span>
<span id="cb51-20"><a href="#cb51-20" aria-hidden="true" tabindex="-1"></a>            <span class="cf">pass</span></span>
<span id="cb51-21"><a href="#cb51-21" aria-hidden="true" tabindex="-1"></a>        plt.close()</span>
<span id="cb51-22"><a href="#cb51-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">except</span> <span class="pp">Exception</span> <span class="im">as</span> e:</span>
<span id="cb51-23"><a href="#cb51-23" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">'Timepoint loading plot failed:'</span>, e)</span>
<span id="cb51-24"><a href="#cb51-24" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb51-25"><a href="#cb51-25" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'No time_factors available; skipping timepoint plot'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-barnacle_files/figure-gfm/viz-time-loadings-3.png" width="768"></p>
</section>
<section id="sample-factor-heatmap" class="level3">
<h3 class="anchored" data-anchor-id="sample-factor-heatmap">4.11.4 Sample-factor heatmap</h3>
<div class="sourceCode" id="cb52"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample-factor heatmap grouped by species</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> sample_factor_df <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span> <span class="kw">and</span> <span class="kw">not</span> sample_factor_df.empty:</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">try</span>:</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>        sf <span class="op">=</span> sample_factor_df.copy()</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>        <span class="co"># If metadata columns present, separate numeric factor columns</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>        meta_cols <span class="op">=</span> [c <span class="cf">for</span> c <span class="kw">in</span> [<span class="st">'Species'</span>,<span class="st">'Sample_ID'</span>] <span class="cf">if</span> c <span class="kw">in</span> sf.columns]</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>        factor_cols <span class="op">=</span> [c <span class="cf">for</span> c <span class="kw">in</span> sf.columns <span class="cf">if</span> c <span class="kw">not</span> <span class="kw">in</span> meta_cols]</span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>        species_series <span class="op">=</span> sf[<span class="st">'Species'</span>] <span class="cf">if</span> <span class="st">'Species'</span> <span class="kw">in</span> sf.columns <span class="cf">else</span> pd.Series([<span class="st">'unknown'</span>]<span class="op">*</span>sf.shape[<span class="dv">0</span>], index<span class="op">=</span>sf.index)</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>        sample_vals <span class="op">=</span> sf[factor_cols]</span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a>        <span class="co"># sort by species for display</span></span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Order samples by species, then by Sample_ID (if available) to include all samples</span></span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="st">'Sample_ID'</span> <span class="kw">in</span> sf.columns:</span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a>            order <span class="op">=</span> sf.sort_values([<span class="st">'Species'</span>,<span class="st">'Sample_ID'</span>]).index</span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>:</span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>            order <span class="op">=</span> species_series.sort_values().index</span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a>        sample_vals_sorted <span class="op">=</span> sample_vals.loc[order]</span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Create a single-axis heatmap and force display of all sample labels</span></span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a>        fig <span class="op">=</span> plt.figure(figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="bu">max</span>(<span class="dv">3</span>, sample_vals_sorted.shape[<span class="dv">0</span>]<span class="op">*</span><span class="fl">0.08</span>)))</span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a>        ax_hm <span class="op">=</span> fig.add_subplot(<span class="dv">111</span>)</span>
<span id="cb52-23"><a href="#cb52-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-24"><a href="#cb52-24" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Main heatmap with slightly shrunken colorbar</span></span>
<span id="cb52-25"><a href="#cb52-25" aria-hidden="true" tabindex="-1"></a>        h <span class="op">=</span> sns.heatmap(sample_vals_sorted, cmap<span class="op">=</span><span class="st">'vlag'</span>, center<span class="op">=</span><span class="dv">0</span>,</span>
<span id="cb52-26"><a href="#cb52-26" aria-hidden="true" tabindex="-1"></a>                        cbar_kws<span class="op">=</span>{<span class="st">'label'</span>: <span class="st">'Component loading (signed)'</span>, <span class="st">'shrink'</span>: <span class="fl">0.8</span>},</span>
<span id="cb52-27"><a href="#cb52-27" aria-hidden="true" tabindex="-1"></a>                        ax<span class="op">=</span>ax_hm)</span>
<span id="cb52-28"><a href="#cb52-28" aria-hidden="true" tabindex="-1"></a>        ax_hm.set_xlabel(<span class="st">'Components'</span>)</span>
<span id="cb52-29"><a href="#cb52-29" aria-hidden="true" tabindex="-1"></a>        ax_hm.set_ylabel(<span class="st">''</span>)</span>
<span id="cb52-30"><a href="#cb52-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-31"><a href="#cb52-31" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Force display of all sample labels (small font for legibility)</span></span>
<span id="cb52-32"><a href="#cb52-32" aria-hidden="true" tabindex="-1"></a>        <span class="cf">try</span>:</span>
<span id="cb52-33"><a href="#cb52-33" aria-hidden="true" tabindex="-1"></a>            ax_hm.set_yticks(np.arange(<span class="bu">len</span>(sample_vals_sorted.index)) <span class="op">+</span> <span class="fl">0.5</span>)</span>
<span id="cb52-34"><a href="#cb52-34" aria-hidden="true" tabindex="-1"></a>            ax_hm.set_yticklabels(sample_vals_sorted.index, fontsize<span class="op">=</span><span class="dv">6</span>)</span>
<span id="cb52-35"><a href="#cb52-35" aria-hidden="true" tabindex="-1"></a>        <span class="cf">except</span> <span class="pp">Exception</span>:</span>
<span id="cb52-36"><a href="#cb52-36" aria-hidden="true" tabindex="-1"></a>            <span class="cf">pass</span></span>
<span id="cb52-37"><a href="#cb52-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-38"><a href="#cb52-38" aria-hidden="true" tabindex="-1"></a>        plt.tight_layout()</span>
<span id="cb52-39"><a href="#cb52-39" aria-hidden="true" tabindex="-1"></a>        out <span class="op">=</span> os.path.join(fig_dir, <span class="st">'sample_factors_heatmap.png'</span>)</span>
<span id="cb52-40"><a href="#cb52-40" aria-hidden="true" tabindex="-1"></a>        plt.savefig(out, dpi<span class="op">=</span><span class="dv">150</span>)</span>
<span id="cb52-41"><a href="#cb52-41" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">'Saved:'</span>, out)</span>
<span id="cb52-42"><a href="#cb52-42" aria-hidden="true" tabindex="-1"></a>        <span class="cf">try</span>:</span>
<span id="cb52-43"><a href="#cb52-43" aria-hidden="true" tabindex="-1"></a>            plt.show()</span>
<span id="cb52-44"><a href="#cb52-44" aria-hidden="true" tabindex="-1"></a>        <span class="cf">except</span> <span class="pp">Exception</span>:</span>
<span id="cb52-45"><a href="#cb52-45" aria-hidden="true" tabindex="-1"></a>            <span class="cf">pass</span></span>
<span id="cb52-46"><a href="#cb52-46" aria-hidden="true" tabindex="-1"></a>        plt.close()</span>
<span id="cb52-47"><a href="#cb52-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-48"><a href="#cb52-48" aria-hidden="true" tabindex="-1"></a>    <span class="cf">except</span> <span class="pp">Exception</span> <span class="im">as</span> e:</span>
<span id="cb52-49"><a href="#cb52-49" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">'Sample factors plotting failed:'</span>, e)</span>
<span id="cb52-50"><a href="#cb52-50" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb52-51"><a href="#cb52-51" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'No sample_factors available; skipping sample heatmap'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-barnacle_files/figure-gfm/viz-sample-heatmap-5.png" width="960"></p>
</section>
<section id="pca-of-sample-factor-loadings" class="level3">
<h3 class="anchored" data-anchor-id="pca-of-sample-factor-loadings">4.11.5 PCA of sample-factor loadings</h3>
<div class="sourceCode" id="cb53"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA of sample factors (2D scatter)</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> sample_factor_df <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span> <span class="kw">and</span> <span class="kw">not</span> sample_factor_df.empty:</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">try</span>:</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>        sf <span class="op">=</span> sample_factor_df.copy()</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>        meta_cols <span class="op">=</span> [c <span class="cf">for</span> c <span class="kw">in</span> [<span class="st">'Species'</span>,<span class="st">'Sample_ID'</span>] <span class="cf">if</span> c <span class="kw">in</span> sf.columns]</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>        factor_cols <span class="op">=</span> [c <span class="cf">for</span> c <span class="kw">in</span> sf.columns <span class="cf">if</span> c <span class="kw">not</span> <span class="kw">in</span> meta_cols]</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>        species_series <span class="op">=</span> sf[<span class="st">'Species'</span>] <span class="cf">if</span> <span class="st">'Species'</span> <span class="kw">in</span> sf.columns <span class="cf">else</span> pd.Series([<span class="st">'unknown'</span>]<span class="op">*</span>sf.shape[<span class="dv">0</span>], index<span class="op">=</span>sf.index)</span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a>        sample_vals <span class="op">=</span> sf[factor_cols]</span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>        X <span class="op">=</span> sample_vals.fillna(<span class="dv">0</span>).values</span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a>        pca <span class="op">=</span> PCA(n_components<span class="op">=</span><span class="dv">2</span>)</span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a>        pcs <span class="op">=</span> pca.fit_transform(X)</span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a>        pcs_df <span class="op">=</span> pd.DataFrame(pcs, index<span class="op">=</span>sample_vals.index, columns<span class="op">=</span>[<span class="st">'PC1'</span>,<span class="st">'PC2'</span>])</span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a>        pcs_df[<span class="st">'Species'</span>] <span class="op">=</span> species_series.loc[pcs_df.index].values</span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a>        plt.figure(figsize<span class="op">=</span>(<span class="dv">6</span>,<span class="dv">5</span>))</span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a>        sns.scatterplot(data<span class="op">=</span>pcs_df, x<span class="op">=</span><span class="st">'PC1'</span>, y<span class="op">=</span><span class="st">'PC2'</span>, hue<span class="op">=</span><span class="st">'Species'</span>, s<span class="op">=</span><span class="dv">80</span>)</span>
<span id="cb53-18"><a href="#cb53-18" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="st">'PCA of sample-factor loadings'</span>)</span>
<span id="cb53-19"><a href="#cb53-19" aria-hidden="true" tabindex="-1"></a>        plt.tight_layout()</span>
<span id="cb53-20"><a href="#cb53-20" aria-hidden="true" tabindex="-1"></a>        out <span class="op">=</span> os.path.join(fig_dir, <span class="st">'sample_factors_pca.png'</span>)</span>
<span id="cb53-21"><a href="#cb53-21" aria-hidden="true" tabindex="-1"></a>        plt.savefig(out, dpi<span class="op">=</span><span class="dv">150</span>)</span>
<span id="cb53-22"><a href="#cb53-22" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">'Saved:'</span>, out)</span>
<span id="cb53-23"><a href="#cb53-23" aria-hidden="true" tabindex="-1"></a>        <span class="cf">try</span>:</span>
<span id="cb53-24"><a href="#cb53-24" aria-hidden="true" tabindex="-1"></a>            plt.show()</span>
<span id="cb53-25"><a href="#cb53-25" aria-hidden="true" tabindex="-1"></a>        <span class="cf">except</span> <span class="pp">Exception</span>:</span>
<span id="cb53-26"><a href="#cb53-26" aria-hidden="true" tabindex="-1"></a>            <span class="cf">pass</span></span>
<span id="cb53-27"><a href="#cb53-27" aria-hidden="true" tabindex="-1"></a>        plt.close()</span>
<span id="cb53-28"><a href="#cb53-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-29"><a href="#cb53-29" aria-hidden="true" tabindex="-1"></a>    <span class="cf">except</span> <span class="pp">Exception</span> <span class="im">as</span> e:</span>
<span id="cb53-30"><a href="#cb53-30" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">'Sample PCA plotting failed:'</span>, e)</span>
<span id="cb53-31"><a href="#cb53-31" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb53-32"><a href="#cb53-32" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'No sample_factors available; skipping sample PCA'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-barnacle_files/figure-gfm/viz-sample-pca-7.png" width="576"></p>
</section>
<section id="top-orthologs-per-component" class="level3">
<h3 class="anchored" data-anchor-id="top-orthologs-per-component">4.11.6 Top orthologs per component</h3>
<div class="sourceCode" id="cb54"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Top orthologs per component (barplots)</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> gene_factor_df <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span> <span class="kw">and</span> <span class="kw">not</span> gene_factor_df.empty:</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">try</span>:</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>        gf <span class="op">=</span> gene_factor_df.copy()</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Determine number of components</span></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>        comps <span class="op">=</span> gf.columns</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>        n_top <span class="op">=</span> <span class="dv">10</span></span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> col <span class="kw">in</span> comps:</span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>            <span class="co"># rank by absolute loading</span></span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>            s <span class="op">=</span> gf[col].<span class="bu">abs</span>().sort_values(ascending<span class="op">=</span><span class="va">False</span>).head(n_top)</span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>            plt.figure(figsize<span class="op">=</span>(<span class="dv">6</span>, <span class="bu">max</span>(<span class="dv">2</span>, n_top<span class="op">*</span><span class="fl">0.25</span>)))</span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a>            sns.barplot(x<span class="op">=</span>s.values, y<span class="op">=</span>s.index, palette<span class="op">=</span><span class="st">'magma'</span>)</span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a>            plt.xlabel(<span class="st">'Absolute loading'</span>)</span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a>            plt.ylabel(<span class="st">'Ortholog group'</span>)</span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a>            plt.title(<span class="ss">f'Top </span><span class="sc">{</span>n_top<span class="sc">}</span><span class="ss"> genes - </span><span class="sc">{</span>col<span class="sc">}</span><span class="ss">'</span>)</span>
<span id="cb54-16"><a href="#cb54-16" aria-hidden="true" tabindex="-1"></a>            plt.tight_layout()</span>
<span id="cb54-17"><a href="#cb54-17" aria-hidden="true" tabindex="-1"></a>            out <span class="op">=</span> os.path.join(fig_dir, <span class="ss">f'top_orthologs_</span><span class="sc">{</span>col<span class="sc">}</span><span class="ss">.png'</span>)</span>
<span id="cb54-18"><a href="#cb54-18" aria-hidden="true" tabindex="-1"></a>            plt.savefig(out, dpi<span class="op">=</span><span class="dv">150</span>)</span>
<span id="cb54-19"><a href="#cb54-19" aria-hidden="true" tabindex="-1"></a>            <span class="bu">print</span>(<span class="st">'Saved:'</span>, out)</span>
<span id="cb54-20"><a href="#cb54-20" aria-hidden="true" tabindex="-1"></a>            <span class="cf">try</span>:</span>
<span id="cb54-21"><a href="#cb54-21" aria-hidden="true" tabindex="-1"></a>                plt.show()</span>
<span id="cb54-22"><a href="#cb54-22" aria-hidden="true" tabindex="-1"></a>            <span class="cf">except</span> <span class="pp">Exception</span>:</span>
<span id="cb54-23"><a href="#cb54-23" aria-hidden="true" tabindex="-1"></a>                <span class="cf">pass</span></span>
<span id="cb54-24"><a href="#cb54-24" aria-hidden="true" tabindex="-1"></a>            plt.close()</span>
<span id="cb54-25"><a href="#cb54-25" aria-hidden="true" tabindex="-1"></a>    <span class="cf">except</span> <span class="pp">Exception</span> <span class="im">as</span> e:</span>
<span id="cb54-26"><a href="#cb54-26" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="st">'Top orthologs plotting failed:'</span>, e)</span>
<span id="cb54-27"><a href="#cb54-27" aria-hidden="true" tabindex="-1"></a><span class="cf">else</span>:</span>
<span id="cb54-28"><a href="#cb54-28" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'No gene_factors available; skipping gene plots'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="13.00-multiomics-barnacle_files/figure-gfm/viz-top-genes-9.png" width="576"><img src="13.00-multiomics-barnacle_files/figure-gfm/viz-top-genes-10.png" width="576"><img src="13.00-multiomics-barnacle_files/figure-gfm/viz-top-genes-11.png" width="576"><img src="13.00-multiomics-barnacle_files/figure-gfm/viz-top-genes-12.png" width="576"><img src="13.00-multiomics-barnacle_files/figure-gfm/viz-top-genes-13.png" width="576"></p>
</section>
</section>
</section>
<section id="system-info" class="level1">
<h1>5 SYSTEM INFO</h1>
<div class="sourceCode" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## R version 4.2.3 (2023-03-15)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 24.04.3 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] glmGamPoi_1.10.2  sctransform_0.4.2 reticulate_1.43.0 lubridate_1.9.4  
##  [5] forcats_1.0.0     stringr_1.5.1     dplyr_1.1.4       purrr_1.1.0      
##  [9] readr_2.1.5       tidyr_1.3.1       tibble_3.3.0      ggplot2_4.0.0    
## [13] tidyverse_2.0.0   knitr_1.50       
## 
## loaded via a namespace (and not attached):
##  [1] MatrixGenerics_1.10.0       Biobase_2.58.0             
##  [3] jsonlite_2.0.0              DelayedMatrixStats_1.20.0  
##  [5] S7_0.2.0                    stats4_4.2.3               
##  [7] GenomeInfoDbData_1.2.9      yaml_2.3.10                
##  [9] globals_0.18.0              pillar_1.11.0              
## [11] lattice_0.22-7              glue_1.8.0                 
## [13] digest_0.6.37               GenomicRanges_1.50.2       
## [15] RColorBrewer_1.1-3          XVector_0.38.0             
## [17] htmltools_0.5.8.1           Matrix_1.6-1               
## [19] plyr_1.8.9                  pkgconfig_2.0.3            
## [21] listenv_0.9.1               zlibbioc_1.44.0            
## [23] scales_1.4.0                tzdb_0.5.0                 
## [25] timechange_0.3.0            generics_0.1.4             
## [27] farver_2.1.2                IRanges_2.32.0             
## [29] withr_3.0.2                 SummarizedExperiment_1.28.0
## [31] BiocGenerics_0.44.0         cli_3.6.5                  
## [33] magrittr_2.0.4              evaluate_1.0.5             
## [35] future_1.67.0               parallelly_1.45.1          
## [37] MASS_7.3-60                 tools_4.2.3                
## [39] hms_1.1.3                   lifecycle_1.0.4            
## [41] matrixStats_1.5.0           S4Vectors_0.36.2           
## [43] DelayedArray_0.24.0         compiler_4.2.3             
## [45] GenomeInfoDb_1.34.9         rlang_1.1.6                
## [47] grid_4.2.3                  RCurl_1.98-1.17            
## [49] rstudioapi_0.17.1           bitops_1.0-9               
## [51] rmarkdown_2.29              gtable_0.3.6               
## [53] codetools_0.2-20            reshape2_1.4.4             
## [55] R6_2.6.1                    gridExtra_2.3              
## [57] fastmap_1.2.0               future.apply_1.20.0        
## [59] stringi_1.8.7               parallel_4.2.3             
## [61] Rcpp_1.1.0                  vctrs_0.6.5                
## [63] png_0.1-8                   tidyselect_1.2.1           
## [65] xfun_0.53                   sparseMatrixStats_1.10.0</code></pre>
</section>
<section id="references" class="level1">
<h1>6 REFERENCES</h1>
<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-blaskowski2024" class="csl-entry" role="listitem">
Blaskowski, Stephen, Marie Roald, Paul M. Berube, Rogier Braakman, and E. Virginia Armbrust. 2024. “Simultaneous Acclimation to Nitrogen and Iron Scarcity in Open Ocean Cyanobacteria Revealed by Sparse Tensor Decomposition of Metatranscriptomes.” <a href="http://dx.doi.org/10.1101/2024.07.15.603627" class="uri">http://dx.doi.org/10.1101/2024.07.15.603627</a>.
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/robertslab\.github\.io\/sams-notebook");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>