<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sam White">
<meta name="dcterms.date" content="2025-02-10">

<title>FastQC MultiQC - P.evermanni Raw WGBS Renaming and FastQC MultiQC E5 timeseries_molecular Project – Sam’s Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark-b53751a350365c71b6c909e95f209ed1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-e518e50a9807db09b51ea894c4c266f8.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 2,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 100,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-7483T61T11"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-7483T61T11', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Sam’s Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" target="_blank">
 <span class="dropdown-text">Rstudio Server - Raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu/Atumefaciens/" target="_blank">
 <span class="dropdown-text">My Gannet Directory</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" target="_blank">
 <span class="dropdown-text">Steven’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shedurkin.github.io/Roberts-LabNotebook/" target="_blank">
 <span class="dropdown-text">Kathleen’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://meganewing.github.io/mewing-notebook/" target="_blank">
 <span class="dropdown-text">Megan’s Notebook</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/sams-notebook" target="_blank">
 <span class="dropdown-text">Notebook repo</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" target="_blank">
 <span class="dropdown-text">Lab Organization (GitHub)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" target="_blank">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" target="_blank"> <i class="bi bi-calendar-week" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-right">
      <h1 class="title">FastQC MultiQC - P.evermanni Raw WGBS Renaming and FastQC MultiQC E5 timeseries_molecular Project</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2025</div>
                <div class="quarto-category">Porites evermanni</div>
                <div class="quarto-category">E5</div>
                <div class="quarto-category">WGBS</div>
                <div class="quarto-category">bisulfite sequencing</div>
                <div class="quarto-category">FastQC</div>
                <div class="quarto-category">MultiQC</div>
                <div class="quarto-category">timeseries_molecular</div>
                <div class="quarto-category">coral</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-right">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Sam White </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 10, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#intro" id="toc-intro" class="nav-link active" data-scroll-target="#intro">INTRO</a></li>
  <li><a href="#background" id="toc-background" class="nav-link" data-scroll-target="#background">Background</a></li>
  <li><a href="#create-a-bash-variables-file" id="toc-create-a-bash-variables-file" class="nav-link" data-scroll-target="#create-a-bash-variables-file">1 Create a Bash variables file</a></li>
  <li><a href="#download-p.evermanni-wgbs-fastqs" id="toc-download-p.evermanni-wgbs-fastqs" class="nav-link" data-scroll-target="#download-p.evermanni-wgbs-fastqs">2 Download <em>P.evermanni</em> WGBS FastQs</a>
  <ul>
  <li><a href="#inspect-metadata-file" id="toc-inspect-metadata-file" class="nav-link" data-scroll-target="#inspect-metadata-file">2.1 Inspect metadata file</a></li>
  <li><a href="#download-raw-wgbs-reads" id="toc-download-raw-wgbs-reads" class="nav-link" data-scroll-target="#download-raw-wgbs-reads">2.2 Download raw WGBS reads</a></li>
  <li><a href="#verify-raw-read-checksums" id="toc-verify-raw-read-checksums" class="nav-link" data-scroll-target="#verify-raw-read-checksums">2.3 Verify raw read checksums</a></li>
  </ul></li>
  <li><a href="#rename-fastqs-for-project-consistency" id="toc-rename-fastqs-for-project-consistency" class="nav-link" data-scroll-target="#rename-fastqs-for-project-consistency">3 Rename FastQs for Project Consistency</a></li>
  <li><a href="#fastqcmultiqc-on-raw-reads" id="toc-fastqcmultiqc-on-raw-reads" class="nav-link" data-scroll-target="#fastqcmultiqc-on-raw-reads">4 FastQC/MultiQC on raw reads</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">RESULTS</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">REFERENCES</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-right" id="quarto-document-content">





<section id="intro" class="level1">
<h1>INTRO</h1>
<p>Ran FastQC and MultiQC quality checks on the <em>P.evermanni</em> raw whole genome bisulfite sequencing (WGBS) data <a href="../../../posts/2024/2024-12-30-Data-Received---Coral-E5-WGBS-Time-Series-from-Azenta-Project-30-1067895835/index.html">received 20241230</a>, as part of <a href="https://github.com/urol-e5/timeseries_molecular">urol-e5/timeseries_molecular</a> (GitHub repo).</p>
<p>Additionally, the FastQ files have been renamed to follow the format <code>&lt;colony_id&gt;-&lt;timepoint&gt;</code>. This ensures consistency and easier identification of samples throughout the analysis process.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The contents below are from markdown knitted from <a href="https://github.com/urol-e5/timeseries_molecular/blob/e05febcc9105afaf086c573368e15b7228d2cbaf/E-Peve/code/00.20-E-Peve-WGBS-reads-FastQC-MultiQC.Rmd"><code>00.20-E-Peve-WGBS-reads-FastQC-MultiQC.Rmd</code></a> (commit <code>e05febc</code>).</p>
</div>
</div>
</section>
<section id="background" class="level1">
<h1>Background</h1>
<p>This Rmd file will download raw WGBS FastQs for <em>P.evermanni</em> and evaluate them using <a href="https://github.com/s-andrews/FastQC">FastQC</a> and <a href="https://multiqc.info/">MultiQC</a>(Ewels et al.&nbsp;2016).</p>
</section>
<section id="create-a-bash-variables-file" class="level1">
<h1>1 Create a Bash variables file</h1>
<p>This allows usage of Bash variables across R Markdown chunks.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"#### Assign Variables ####"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# Data directories"</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export timeseries_dir=/home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular'</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export output_dir_top=${timeseries_dir}/E-Peve/output/00.20-E-Peve-WGBS-reads-FastQC-MultiQC'</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export raw_reads_dir=${timeseries_dir}/E-Peve/data/wgbs-raw-fastqs'</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export raw_reads_url="https://owl.fish.washington.edu/nightingales/E5-coral-time-series/30-1067895835/"'</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# Input files"</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export metadata_file="${timeseries_dir}/M-multi-species/data/e5_DNA_Azenta_metadata.csv"'</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# Paths to programs"</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export fastqc=/home/shared/FastQC-0.12.1/fastqc'</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export multiqc=/home/sam/programs/mambaforge/bin/multiqc'</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# Set FastQ filename patterns"</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"export fastq_pattern='*.fastq.gz'"</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"export R1_fastq_pattern='*_R1_*.fastq.gz'"</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"export R2_fastq_pattern='*_R2_*.fastq.gz'"</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# Set number of CPUs to use"</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export threads=40'</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"## Inititalize arrays"</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export fastq_array_R1=()'</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export fastq_array_R2=()'</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export raw_fastqs_array=()'</span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export R1_names_array=()'</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export R2_names_array=()'</span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# Programs associative array"</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"declare -A programs_array"</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"programs_array=("</span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'[fastqc]="${fastqc}" \'</span></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'[multiqc]="${multiqc}" \'</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">")"</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"# Print formatting"</span></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">'export line="--------------------------------------------------------"'</span></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="kw">}</span> <span class="op">&gt;</span> .bashvars</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> .bashvars</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>#### Assign Variables ####

# Data directories
export timeseries_dir=/home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular
export output_dir_top=${timeseries_dir}/E-Peve/output/00.20-E-Peve-WGBS-reads-FastQC-MultiQC
export raw_reads_dir=${timeseries_dir}/E-Peve/data/wgbs-raw-fastqs
export raw_reads_url="https://owl.fish.washington.edu/nightingales/E5-coral-time-series/30-1067895835/"

# Input files
export metadata_file="${timeseries_dir}/M-multi-species/data/e5_DNA_Azenta_metadata.csv"
# Paths to programs
export fastqc=/home/shared/FastQC-0.12.1/fastqc
export multiqc=/home/sam/programs/mambaforge/bin/multiqc

# Set FastQ filename patterns
export fastq_pattern='*.fastq.gz'
export R1_fastq_pattern='*_R1_*.fastq.gz'
export R2_fastq_pattern='*_R2_*.fastq.gz'

# Set number of CPUs to use
export threads=40

## Inititalize arrays
export fastq_array_R1=()
export fastq_array_R2=()
export raw_fastqs_array=()
export R1_names_array=()
export R2_names_array=()

# Programs associative array
declare -A programs_array
programs_array=(
[fastqc]="${fastqc}" \
[multiqc]="${multiqc}" \
)

# Print formatting
export line="--------------------------------------------------------"</code></pre>
</section>
<section id="download-p.evermanni-wgbs-fastqs" class="level1">
<h1>2 Download <em>P.evermanni</em> WGBS FastQs</h1>
<section id="inspect-metadata-file" class="level2">
<h2 class="anchored" data-anchor-id="inspect-metadata-file">2.1 Inspect metadata file</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="va">${metadata_file}</span> <span class="kw">|</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="ex">column</span> <span class="at">-t</span> <span class="at">-s</span><span class="st">","</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Sample Name*  Plate         Well number  Azenta sample name  Colony ID  Timepoint  Sample Type*  Species/Strain*  Sample Buffer*    Total Amount (ng)*    Volume (µL)  Conc (ng/µL)  Purification Method*  Biosafety Level (BSL)            Method Used for Fluorescence  Method Used for Spectrophotometry  Method Used for Electrophoresis  
1             41            1            G4                  1G4        ACR-225    TP1           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  2064         80            25.8                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
2             291-20220211  2            A3                  2A3        ACR-225    TP2           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  5064         80            63.3                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
3             527           1            H5                  1H5        ACR-225    TP3           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  4232         80            52.9                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
4             911           1            C6                  1C6        ACR-225    TP4           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  4488         80            56.1                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
5             55            1            B2                  1B2        ACR-229    TP1           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  2432         80            30.4                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
6             295           1            E3                  1E3        ACR-229    TP2           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  2760         80            34.5                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
7             531           1            H3                  1H3        ACR-229    TP3           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  5752         80            71.9                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
8             931           1            F7                  1F7        ACR-229    TP4           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  6008         80            75.1                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"
9             49            1            A1                  1A1        ACR-237    TP1           Genomic DNA      Acropora pulchra  Tris buffer (8.0 pH)  1456         80            18.2                  Column clean-up (Zymo Miniprep)  None                          Qubit                              NA                               1.5% TAE agarose non-denaturing gel  "Please keep methylation libraries   as we may want to sequence more"</code></pre>
</section>
<section id="download-raw-wgbs-reads" class="level2">
<h2 class="anchored" data-anchor-id="download-raw-wgbs-reads">2.2 Download raw WGBS reads</h2>
<p>Reads are downloaded from <a href="https://owl.fish.washington.edu/nightingales/E5-coral-time-series/30-1067895835/" class="uri">https://owl.fish.washington.edu/nightingales/E5-coral-time-series/30-1067895835/</a></p>
<p>Since sequencing included multiple species, the code will also parse only those that are <em>P.evermanni</em>.</p>
<p>Some filenames are preceeded by <code>&lt;sample_number&gt;--</code> for some reason. This formatting is limited to only samples in <code>E</code> wells, so the code below uses that aspect to retrieve those samples as well.</p>
<p>The <code>--cut-dirs 3</code> command cuts the preceding directory structure (i.e.&nbsp;<code>nightingales/E5-coral-time-series/30-1067895835/</code>) so that we just end up with the reads.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Make output directory if it doesn't exist</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">--parents</span> <span class="va">${raw_reads_dir}</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create list of only P.evermanni sample names</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Some samples names are preceded by the Sample Number for some reason,</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co"># so this handles that formatting.</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="va">sample_list</span><span class="op">=</span><span class="va">$(</span><span class="fu">awk</span> <span class="at">-F</span><span class="st">","</span> <span class="st">'$11 == "Porites evermanni" { if ($9 ~ /E/) { print $8 "--" $9 } else { print $9 } }'</span> <span class="va">${timeseries_dir}</span>/M-multi-species/data/30-1067895835-WGBS-sample-submission-form.csv <span class="dt">\</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">sort</span><span class="va">)</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"</span><span class="va">${line}</span><span class="st">"</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Sample list:"</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"</span><span class="va">${sample_list}</span><span class="st">"</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"</span><span class="va">${line}</span><span class="st">"</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Use printf to format each item for use in wget</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="va">formatted_list</span><span class="op">=</span><span class="va">$(</span><span class="bu">printf</span> <span class="st">"%s_*,"</span> <span class="va">${sample_list})</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove the trailing comma</span></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a><span class="va">formatted_list</span><span class="op">=</span><span class="st">"</span><span class="va">${formatted_list</span><span class="op">%</span>,<span class="va">}</span><span class="st">"</span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Output the final wget command</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"</span><span class="va">${line}</span><span class="st">"</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Formatted wget accept list:"</span></span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"wget --accept=</span><span class="dt">\"</span><span class="va">$formatted_list</span><span class="dt">\"</span><span class="st">"</span></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"</span><span class="va">${line}</span><span class="st">"</span></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Run wget to retrieve FastQs and MD5 files</span></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="dt">\</span></span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a>--directory-prefix <span class="va">${raw_reads_dir}</span> <span class="dt">\</span></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a>--recursive <span class="dt">\</span></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a>--no-check-certificate <span class="dt">\</span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>--continue <span class="dt">\</span></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a>--cut-dirs 3 <span class="dt">\</span></span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a>--no-host-directories <span class="dt">\</span></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>--no-parent <span class="dt">\</span></span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>--quiet <span class="dt">\</span></span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>--accept=<span class="dt">\"</span><span class="va">$formatted_list</span><span class="dt">\"</span> <span class="va">${raw_reads_url}</span></span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-lh</span> <span class="st">"</span><span class="va">${raw_reads_dir}</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>--------------------------------------------------------

Sample list:

109--1E9
110--2E1
111--1E4
114--1E6
1A7
1A9
1B11
1B6
1B8
1B9
1C2
1D1
1D10
1D12
1D2
1D6
1F10
1F3
1F6
1G10
1G11
1G2
1G6
1G9
1H12
1H8
2A2
2B3
2C1
2C3
2F2
2F3
2G1
2G3
2H2
88--1E5
93--1E11
96--1E8

--------------------------------------------------------


--------------------------------------------------------

Formatted wget accept list:

wget --accept="109--1E9_*,110--2E1_*,111--1E4_*,114--1E6_*,1A7_*,1A9_*,1B11_*,1B6_*,1B8_*,1B9_*,1C2_*,1D1_*,1D10_*,1D12_*,1D2_*,1D6_*,1F10_*,1F3_*,1F6_*,1G10_*,1G11_*,1G2_*,1G6_*,1G9_*,1H12_*,1H8_*,2A2_*,2B3_*,2C1_*,2C3_*,2F2_*,2F3_*,2G1_*,2G3_*,2H2_*,88--1E5_*,93--1E11_*,96--1E8_*"

--------------------------------------------------------

total 172G
-rw-r--r-- 1 sam sam 2.0G Dec 30 08:07 110--2E1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   61 Dec 30 08:07 110--2E1_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.0G Dec 30 08:09 110--2E1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   61 Dec 30 08:09 110--2E1_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 3.0G Dec 30 08:12 111--1E4_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   61 Dec 30 08:12 111--1E4_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.9G Dec 30 08:15 111--1E4_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   61 Dec 30 08:15 111--1E4_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.1G Dec 30 08:17 114--1E6_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   61 Dec 30 08:17 114--1E6_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.0G Dec 30 08:19 114--1E6_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   61 Dec 30 08:19 114--1E6_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.7G Dec 30 09:12 1A7_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 09:12 1A7_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.6G Dec 30 09:15 1A7_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 09:15 1A7_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 3.0G Dec 30 09:24 1A9_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 09:24 1A9_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.9G Dec 30 09:28 1A9_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 09:28 1A9_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 09:38 1B11_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 09:38 1B11_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 09:40 1B11_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 09:40 1B11_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.3G Dec 30 10:15 1B6_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 10:15 1B6_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.3G Dec 30 10:18 1B6_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 10:18 1B6_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 1.9G Dec 30 10:25 1B8_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 10:25 1B8_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 1.9G Dec 30 10:27 1B8_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 10:27 1B8_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.5G Dec 30 10:29 1B9_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 10:29 1B9_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.5G Dec 30 10:32 1B9_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 10:32 1B9_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 10:57 1C2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 10:57 1C2_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 10:59 1C2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 10:59 1C2_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.1G Dec 30 11:36 1D10_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 11:36 1D10_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.1G Dec 30 11:38 1D10_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 11:38 1D10_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 11:46 1D12_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 11:46 1D12_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 11:48 1D12_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 11:48 1D12_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 3.0G Dec 30 11:51 1D1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 11:51 1D1_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.9G Dec 30 11:54 1D1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 11:54 1D1_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 11:56 1D2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 11:56 1D2_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 11:58 1D2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 11:58 1D2_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.5G Dec 30 12:16 1D6_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 12:16 1D6_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.5G Dec 30 12:19 1D6_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 12:19 1D6_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 12:36 1F10_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 12:36 1F10_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 12:38 1F10_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 12:38 1F10_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.0G Dec 30 13:00 1F3_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 13:00 1F3_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.0G Dec 30 13:02 1F3_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 13:02 1F3_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.1G Dec 30 13:15 1F6_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 13:15 1F6_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.1G Dec 30 13:17 1F6_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 13:17 1F6_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.3G Dec 30 13:32 1G10_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 13:32 1G10_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.2G Dec 30 13:34 1G10_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 13:34 1G10_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.9G Dec 30 13:37 1G11_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 13:37 1G11_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.9G Dec 30 13:39 1G11_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 13:39 1G11_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.8G Dec 30 13:51 1G2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 13:51 1G2_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.8G Dec 30 13:54 1G2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 13:54 1G2_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 1.7G Dec 30 14:10 1G6_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 14:10 1G6_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 1.7G Dec 30 14:12 1G6_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 14:12 1G6_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.2G Dec 30 14:24 1G9_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 14:24 1G9_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.2G Dec 30 14:26 1G9_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 14:26 1G9_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.9G Dec 30 14:40 1H12_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 14:40 1H12_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 3.0G Dec 30 14:43 1H12_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   57 Dec 30 14:43 1H12_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 15:24 1H8_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 15:24 1H8_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 15:27 1H8_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 15:27 1H8_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.2G Dec 30 15:41 2A2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 15:41 2A2_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.2G Dec 30 15:43 2A2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 15:43 2A2_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.8G Dec 30 15:59 2B3_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 15:59 2B3_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.9G Dec 30 16:03 2B3_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 16:03 2B3_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.1G Dec 30 16:05 2C1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 16:05 2C1_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.1G Dec 30 16:08 2C1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 16:08 2C1_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.6G Dec 30 16:17 2C3_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 16:17 2C3_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.6G Dec 30 16:20 2C3_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 16:20 2C3_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.5G Dec 30 16:43 2F2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 16:43 2F2_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.6G Dec 30 16:46 2F2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 16:46 2F2_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 16:49 2F3_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 16:49 2F3_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.4G Dec 30 16:51 2F3_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 16:51 2F3_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.6G Dec 30 16:54 2G1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 16:54 2G1_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.6G Dec 30 16:57 2G1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 16:57 2G1_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 1.9G Dec 30 17:05 2G3_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 17:05 2G3_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 1.9G Dec 30 17:07 2G3_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 17:07 2G3_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.8G Dec 30 17:17 2H2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 17:17 2H2_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.8G Dec 30 17:20 2H2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   56 Dec 30 17:20 2H2_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.8G Dec 30 18:09 88--1E5_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   60 Dec 30 18:09 88--1E5_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.7G Dec 30 18:13 88--1E5_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   60 Dec 30 18:13 88--1E5_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.6G Dec 30 18:16 93--1E11_R1_001.fastq.gz
-rw-r--r-- 1 sam sam   61 Dec 30 18:16 93--1E11_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam 2.5G Dec 30 18:18 93--1E11_R2_001.fastq.gz
-rw-r--r-- 1 sam sam   61 Dec 30 18:18 93--1E11_R2_001.fastq.gz.md5</code></pre>
</section>
<section id="verify-raw-read-checksums" class="level2">
<h2 class="anchored" data-anchor-id="verify-raw-read-checksums">2.3 Verify raw read checksums</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="st">"</span><span class="va">${raw_reads_dir}</span><span class="st">"</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file <span class="kw">in</span> <span class="pp">*</span>.md5</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">md5sum</span> <span class="at">--check</span> <span class="st">"</span><span class="va">${file}</span><span class="st">"</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>./110--2E1_R1_001.fastq.gz: OK
./110--2E1_R2_001.fastq.gz: OK
./111--1E4_R1_001.fastq.gz: OK
./111--1E4_R2_001.fastq.gz: OK
./114--1E6_R1_001.fastq.gz: OK
./114--1E6_R2_001.fastq.gz: OK
./1A7_R1_001.fastq.gz: OK
./1A7_R2_001.fastq.gz: OK
./1A9_R1_001.fastq.gz: OK
./1A9_R2_001.fastq.gz: OK
./1B11_R1_001.fastq.gz: OK
./1B11_R2_001.fastq.gz: OK
./1B6_R1_001.fastq.gz: OK
./1B6_R2_001.fastq.gz: OK
./1B8_R1_001.fastq.gz: OK
./1B8_R2_001.fastq.gz: OK
./1B9_R1_001.fastq.gz: OK
./1B9_R2_001.fastq.gz: OK
./1C2_R1_001.fastq.gz: OK
./1C2_R2_001.fastq.gz: OK
./1D10_R1_001.fastq.gz: OK
./1D10_R2_001.fastq.gz: OK
./1D12_R1_001.fastq.gz: OK
./1D12_R2_001.fastq.gz: OK
./1D1_R1_001.fastq.gz: OK
./1D1_R2_001.fastq.gz: OK
./1D2_R1_001.fastq.gz: OK
./1D2_R2_001.fastq.gz: OK
./1D6_R1_001.fastq.gz: OK
./1D6_R2_001.fastq.gz: OK
./1F10_R1_001.fastq.gz: OK
./1F10_R2_001.fastq.gz: OK
./1F3_R1_001.fastq.gz: OK
./1F3_R2_001.fastq.gz: OK
./1F6_R1_001.fastq.gz: OK
./1F6_R2_001.fastq.gz: OK
./1G10_R1_001.fastq.gz: OK
./1G10_R2_001.fastq.gz: OK
./1G11_R1_001.fastq.gz: OK
./1G11_R2_001.fastq.gz: OK
./1G2_R1_001.fastq.gz: OK
./1G2_R2_001.fastq.gz: OK
./1G6_R1_001.fastq.gz: OK
./1G6_R2_001.fastq.gz: OK
./1G9_R1_001.fastq.gz: OK
./1G9_R2_001.fastq.gz: OK
./1H12_R1_001.fastq.gz: OK
./1H12_R2_001.fastq.gz: OK
./1H8_R1_001.fastq.gz: OK
./1H8_R2_001.fastq.gz: OK
./2A2_R1_001.fastq.gz: OK
./2A2_R2_001.fastq.gz: OK
./2B3_R1_001.fastq.gz: OK
./2B3_R2_001.fastq.gz: OK
./2C1_R1_001.fastq.gz: OK
./2C1_R2_001.fastq.gz: OK
./2C3_R1_001.fastq.gz: OK
./2C3_R2_001.fastq.gz: OK
./2F2_R1_001.fastq.gz: OK
./2F2_R2_001.fastq.gz: OK
./2F3_R1_001.fastq.gz: OK
./2F3_R2_001.fastq.gz: OK
./2G1_R1_001.fastq.gz: OK
./2G1_R2_001.fastq.gz: OK
./2G3_R1_001.fastq.gz: OK
./2G3_R2_001.fastq.gz: OK
./2H2_R1_001.fastq.gz: OK
./2H2_R2_001.fastq.gz: OK
./88--1E5_R1_001.fastq.gz: OK
./88--1E5_R2_001.fastq.gz: OK
./93--1E11_R1_001.fastq.gz: OK
./93--1E11_R2_001.fastq.gz: OK</code></pre>
</section>
</section>
<section id="rename-fastqs-for-project-consistency" class="level1">
<h1>3 Rename FastQs for Project Consistency</h1>
<p>New naming format will be <code>&lt;colony_id&gt;-&lt;timepoint&gt;</code>.</p>
<p>Use of a temp file, instead of working directory from the metadata file can avoid potential issues with character/text formatting from the input CSV.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="st">"</span><span class="va">${raw_reads_dir}</span><span class="st">"</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Create an associative array to store the mapping</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="bu">declare</span> <span class="at">-A</span> <span class="va">number_to_colony_timepoint</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a temporary file to store the mapping</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="va">temp_mapping_file</span><span class="op">=</span><span class="va">$(</span><span class="fu">mktemp</span><span class="va">)</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the metadata file and populate the temporary file using awk</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Maps Azenta sample name to colony_id and timepoint.</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="at">-F,</span> <span class="st">'NR &gt; 1 { print $5, $6 "-" $7 }'</span> <span class="st">"</span><span class="va">$metadata_file</span><span class="st">"</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">$temp_mapping_file</span><span class="st">"</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the temporary file and populate the associative array</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="bu">read</span> <span class="at">-r</span> <span class="va">Number</span> <span class="va">colony_timepoint</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>    <span class="va">number_to_colony_timepoint</span><span class="op">[</span><span class="st">"</span><span class="va">$Number</span><span class="st">"</span><span class="op">]=</span><span class="st">"</span><span class="va">$colony_timepoint</span><span class="st">"</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span> <span class="op">&lt;</span> <span class="st">"</span><span class="va">$temp_mapping_file</span><span class="st">"</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove the temporary file</span></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> <span class="st">"</span><span class="va">$temp_mapping_file</span><span class="st">"</span></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the associative array for debugging</span></span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Number to colony_id-Timepoint mapping:"</span></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> key <span class="kw">in</span> <span class="st">"</span><span class="va">${</span><span class="op">!</span><span class="va">number_to_colony_timepoint</span><span class="op">[@]</span><span class="va">}</span><span class="st">"</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"</span><span class="va">$key</span><span class="st"> -&gt; </span><span class="va">${number_to_colony_timepoint</span><span class="op">[</span><span class="va">$key</span><span class="op">]</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate over the FastQ files in the current directory</span></span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> fastq_file <span class="kw">in</span> <span class="pp">*</span>.fastq.gz<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Extract the sample number from the filename</span></span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>    <span class="va">sample_number</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="st">"</span><span class="va">$fastq_file</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="at">-E</span> <span class="st">'s/^([0-9]+--)?([A-Za-z0-9]+).*/\2/'</span><span class="va">)</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if the sample number exists in the associative array</span></span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">[[</span> <span class="ot">-n</span> <span class="st">"</span><span class="va">${number_to_colony_timepoint</span><span class="op">[</span><span class="va">$sample_number</span><span class="op">]</span><span class="va">}</span><span class="st">"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>        <span class="va">new_sample_name</span><span class="op">=</span><span class="st">"</span><span class="va">${number_to_colony_timepoint</span><span class="op">[</span><span class="va">$sample_number</span><span class="op">]</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>        <span class="va">new_filename</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="st">"</span><span class="va">$fastq_file</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="at">-E</span> <span class="st">"s/^([0-9]+--)?</span><span class="va">$sample_number</span><span class="st">/</span><span class="va">$new_sample_name</span><span class="st">/"</span><span class="va">)</span></span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">$fastq_file</span><span class="st">"</span> <span class="ot">!=</span> <span class="st">"</span><span class="va">$new_filename</span><span class="st">"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>            <span class="fu">mv</span> <span class="st">"</span><span class="va">$fastq_file</span><span class="st">"</span> <span class="st">"</span><span class="va">$new_filename</span><span class="st">"</span></span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Renamed </span><span class="va">$fastq_file</span><span class="st"> to </span><span class="va">$new_filename</span><span class="st">"</span></span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span></span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"No renaming needed for </span><span class="va">$fastq_file</span><span class="st">"</span></span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a>        <span class="cf">fi</span></span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span></span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>        <span class="bu">echo</span> <span class="st">"Sample number '</span><span class="va">$sample_number</span><span class="st">' not found in metadata."</span></span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Debugging: Print all keys in the associative array with quotes</span></span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>        <span class="bu">echo</span> <span class="st">"Available keys in the associative array:"</span></span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> key <span class="kw">in</span> <span class="st">"</span><span class="va">${</span><span class="op">!</span><span class="va">number_to_colony_timepoint</span><span class="op">[@]</span><span class="va">}</span><span class="st">"</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb9-52"><a href="#cb9-52" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"'</span><span class="va">$key</span><span class="st">'"</span></span>
<span id="cb9-53"><a href="#cb9-53" aria-hidden="true" tabindex="-1"></a>        <span class="cf">done</span></span>
<span id="cb9-54"><a href="#cb9-54" aria-hidden="true" tabindex="-1"></a>    <span class="cf">fi</span></span>
<span id="cb9-55"><a href="#cb9-55" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Number to colony_id-Timepoint mapping:

1B3 -&gt; SR_1-440-NA


1B2 -&gt; ACR-229-TP1


1B1 -&gt; ACR-150-TP3


1B7 -&gt; POC-40-TP3


1B6 -&gt; POR-245-TP2


1B5 -&gt; ACR-244-TP2


1B4 -&gt; ACR-244-TP4


1B9 -&gt; POR-236-TP2


1B8 -&gt; POR-69-TP1


1D5 -&gt; POC-57-TP1


1D4 -&gt; POC-52-TP3


1C1 -&gt; POC-222-TP3


1C2 -&gt; POR-83-TP4


1D7 -&gt; ACR-186-TP4


1D6 -&gt; POR-245-TP4


1C3 -&gt; POC-53-TP3


1D1 -&gt; POR-216-TP2


1B12 -&gt; POC-219-TP4


1C4 -&gt; POC-201-TP2


1C5 -&gt; POC-52-TP4


1B10 -&gt; ACR-150-TP1


1D3 -&gt; ACR-237-TP4


1C6 -&gt; ACR-225-TP4


1B11 -&gt; POR-69-TP4


1D2 -&gt; POR-260-TP1


1C7 -&gt; POC-53-TP1


1C8 -&gt; ACR-145-TP2


1C9 -&gt; ACR-237-TP3


2C3 -&gt; POR-73-TP1


1D9 -&gt; ACR-139-TP3


2C2 -&gt; ACR-145-TP3


1D8 -&gt; ACR-139-TP4


2C1 -&gt; POR-216-TP1


1E2 -&gt; POC-42-TP3


1E3 -&gt; ACR-229-TP2


1E1 -&gt; POC-259-TP4


1E6 -&gt; POR-83-TP1


1E7 -&gt; ACR-237-TP2


1E4 -&gt; POR-74-TP2


1E5 -&gt; POR-245-TP3


1E11 -&gt; POR-260-TP4


1E10 -&gt; POC-259-TP3


1E8 -&gt; POR-262-TP3


1E9 -&gt; POR-73-TP4


1E12 -&gt; POC-222-TP2


2B1 -&gt; SR_3-440B-NA


2B2 -&gt; POC-40-TP2


2B3 -&gt; POR-262-TP4


2E1 -&gt; POR-74-TP1


2E3 -&gt; POC-255-TP4


2E2 -&gt; ACR-186-TP3


1H1 -&gt; POC-57-TP4


1H3 -&gt; ACR-229-TP3


1H2 -&gt; ACR-173-TP3


2D2 -&gt; ACR-173-TP2


1H5 -&gt; ACR-225-TP3


2D3 -&gt; POC-52-TP1


1H4 -&gt; POC-201-TP1


1H7 -&gt; POC-219-TP1


1F9 -&gt; ACR-265-TP2


1F8 -&gt; POC-53-TP4


2D1 -&gt; ACR-150-TP4


1H6 -&gt; ACR-150-TP2


1H9 -&gt; ACR-173-TP1


1F7 -&gt; ACR-229-TP4


1H8 -&gt; POR-74-TP4


1F6 -&gt; POR-260-TP2


1F5 -&gt; ACR-145-TP4


1F4 -&gt; POC-255-TP1


1F3 -&gt; POR-83-TP3


1D10 -&gt; POR-69-TP2


1D11 -&gt; SR_2-84-NA


1F2 -&gt; POC-57-TP3


1D12 -&gt; POR-73-TP3


1F1 -&gt; ACR-244-TP1


2G3 -&gt; POR-74-TP3


2G2 -&gt; ACR-265-TP1


2G1 -&gt; POR-245-TP1


1G8 -&gt; POC-222-TP1


1G9 -&gt; POR-260-TP3


1G12 -&gt; POC-40-TP1


1G11 -&gt; POR-216-TP3


1G10 -&gt; POR-72-TP3


1G4 -&gt; ACR-225-TP1


1G5 -&gt; ACR-173-TP4


1G6 -&gt; POR-236-TP1


1G7 -&gt; POC-222-TP4


1G1 -&gt; POC-57-TP2


1G2 -&gt; POR-216-TP4


1G3 -&gt; POC-259-TP1


2F1 -&gt; POC-255-TP3


2F2 -&gt; POR-83-TP2


2F3 -&gt; POR-72-TP4


2A1 -&gt; POC-40-TP4


1A11 -&gt; POC-201-TP3


1A10 -&gt; ACR-139-TP1


2A3 -&gt; ACR-225-TP2


2A2 -&gt; POR-72-TP2


1A12 -&gt; ACR-186-TP2


2H2 -&gt; POR-262-TP2


2H3 -&gt; ACR-244-TP3


2H1 -&gt; POC-42-TP1


1H10 -&gt; POC-219-TP2


1A6 -&gt; ACR-265-TP3


1A7 -&gt; POR-262-TP1


1C12 -&gt; POC-52-TP2


1H11 -&gt; ACR-265-TP4


1H12 -&gt; POR-73-TP2


1C11 -&gt; POC-42-TP4


1A4 -&gt; POC-259-TP2


1C10 -&gt; POC-53-TP2


1A5 -&gt; ACR-139-TP2


1A2 -&gt; POC-219-TP3


1A3 -&gt; ACR-145-TP1


1A1 -&gt; ACR-237-TP1


1F12 -&gt; POC-255-TP2


1F10 -&gt; POR-72-TP1


1F11 -&gt; POC-42-TP2


1A8 -&gt; ACR-186-TP1


1A9 -&gt; POR-69-TP3

Renamed 110--2E1_R1_001.fastq.gz to POR-74-TP1_R1_001.fastq.gz
Renamed 110--2E1_R2_001.fastq.gz to POR-74-TP1_R2_001.fastq.gz
Renamed 111--1E4_R1_001.fastq.gz to POR-74-TP2_R1_001.fastq.gz
Renamed 111--1E4_R2_001.fastq.gz to POR-74-TP2_R2_001.fastq.gz
Renamed 114--1E6_R1_001.fastq.gz to POR-83-TP1_R1_001.fastq.gz
Renamed 114--1E6_R2_001.fastq.gz to POR-83-TP1_R2_001.fastq.gz
Renamed 1A7_R1_001.fastq.gz to POR-262-TP1_R1_001.fastq.gz
Renamed 1A7_R2_001.fastq.gz to POR-262-TP1_R2_001.fastq.gz
Renamed 1A9_R1_001.fastq.gz to POR-69-TP3_R1_001.fastq.gz
Renamed 1A9_R2_001.fastq.gz to POR-69-TP3_R2_001.fastq.gz
Renamed 1B11_R1_001.fastq.gz to POR-69-TP4_R1_001.fastq.gz
Renamed 1B11_R2_001.fastq.gz to POR-69-TP4_R2_001.fastq.gz
Renamed 1B6_R1_001.fastq.gz to POR-245-TP2_R1_001.fastq.gz
Renamed 1B6_R2_001.fastq.gz to POR-245-TP2_R2_001.fastq.gz
Renamed 1B8_R1_001.fastq.gz to POR-69-TP1_R1_001.fastq.gz
Renamed 1B8_R2_001.fastq.gz to POR-69-TP1_R2_001.fastq.gz
Renamed 1B9_R1_001.fastq.gz to POR-236-TP2_R1_001.fastq.gz
Renamed 1B9_R2_001.fastq.gz to POR-236-TP2_R2_001.fastq.gz
Renamed 1C2_R1_001.fastq.gz to POR-83-TP4_R1_001.fastq.gz
Renamed 1C2_R2_001.fastq.gz to POR-83-TP4_R2_001.fastq.gz
Renamed 1D10_R1_001.fastq.gz to POR-69-TP2_R1_001.fastq.gz
Renamed 1D10_R2_001.fastq.gz to POR-69-TP2_R2_001.fastq.gz
Renamed 1D12_R1_001.fastq.gz to POR-73-TP3_R1_001.fastq.gz
Renamed 1D12_R2_001.fastq.gz to POR-73-TP3_R2_001.fastq.gz
Renamed 1D1_R1_001.fastq.gz to POR-216-TP2_R1_001.fastq.gz
Renamed 1D1_R2_001.fastq.gz to POR-216-TP2_R2_001.fastq.gz
Renamed 1D2_R1_001.fastq.gz to POR-260-TP1_R1_001.fastq.gz
Renamed 1D2_R2_001.fastq.gz to POR-260-TP1_R2_001.fastq.gz
Renamed 1D6_R1_001.fastq.gz to POR-245-TP4_R1_001.fastq.gz
Renamed 1D6_R2_001.fastq.gz to POR-245-TP4_R2_001.fastq.gz
Renamed 1F10_R1_001.fastq.gz to POR-72-TP1_R1_001.fastq.gz
Renamed 1F10_R2_001.fastq.gz to POR-72-TP1_R2_001.fastq.gz
Renamed 1F3_R1_001.fastq.gz to POR-83-TP3_R1_001.fastq.gz
Renamed 1F3_R2_001.fastq.gz to POR-83-TP3_R2_001.fastq.gz
Renamed 1F6_R1_001.fastq.gz to POR-260-TP2_R1_001.fastq.gz
Renamed 1F6_R2_001.fastq.gz to POR-260-TP2_R2_001.fastq.gz
Renamed 1G10_R1_001.fastq.gz to POR-72-TP3_R1_001.fastq.gz
Renamed 1G10_R2_001.fastq.gz to POR-72-TP3_R2_001.fastq.gz
Renamed 1G11_R1_001.fastq.gz to POR-216-TP3_R1_001.fastq.gz
Renamed 1G11_R2_001.fastq.gz to POR-216-TP3_R2_001.fastq.gz
Renamed 1G2_R1_001.fastq.gz to POR-216-TP4_R1_001.fastq.gz
Renamed 1G2_R2_001.fastq.gz to POR-216-TP4_R2_001.fastq.gz
Renamed 1G6_R1_001.fastq.gz to POR-236-TP1_R1_001.fastq.gz
Renamed 1G6_R2_001.fastq.gz to POR-236-TP1_R2_001.fastq.gz
Renamed 1G9_R1_001.fastq.gz to POR-260-TP3_R1_001.fastq.gz
Renamed 1G9_R2_001.fastq.gz to POR-260-TP3_R2_001.fastq.gz
Renamed 1H12_R1_001.fastq.gz to POR-73-TP2_R1_001.fastq.gz
Renamed 1H12_R2_001.fastq.gz to POR-73-TP2_R2_001.fastq.gz
Renamed 1H8_R1_001.fastq.gz to POR-74-TP4_R1_001.fastq.gz
Renamed 1H8_R2_001.fastq.gz to POR-74-TP4_R2_001.fastq.gz
Renamed 2A2_R1_001.fastq.gz to POR-72-TP2_R1_001.fastq.gz
Renamed 2A2_R2_001.fastq.gz to POR-72-TP2_R2_001.fastq.gz
Renamed 2B3_R1_001.fastq.gz to POR-262-TP4_R1_001.fastq.gz
Renamed 2B3_R2_001.fastq.gz to POR-262-TP4_R2_001.fastq.gz
Renamed 2C1_R1_001.fastq.gz to POR-216-TP1_R1_001.fastq.gz
Renamed 2C1_R2_001.fastq.gz to POR-216-TP1_R2_001.fastq.gz
Renamed 2C3_R1_001.fastq.gz to POR-73-TP1_R1_001.fastq.gz
Renamed 2C3_R2_001.fastq.gz to POR-73-TP1_R2_001.fastq.gz
Renamed 2F2_R1_001.fastq.gz to POR-83-TP2_R1_001.fastq.gz
Renamed 2F2_R2_001.fastq.gz to POR-83-TP2_R2_001.fastq.gz
Renamed 2F3_R1_001.fastq.gz to POR-72-TP4_R1_001.fastq.gz
Renamed 2F3_R2_001.fastq.gz to POR-72-TP4_R2_001.fastq.gz
Renamed 2G1_R1_001.fastq.gz to POR-245-TP1_R1_001.fastq.gz
Renamed 2G1_R2_001.fastq.gz to POR-245-TP1_R2_001.fastq.gz
Renamed 2G3_R1_001.fastq.gz to POR-74-TP3_R1_001.fastq.gz
Renamed 2G3_R2_001.fastq.gz to POR-74-TP3_R2_001.fastq.gz
Renamed 2H2_R1_001.fastq.gz to POR-262-TP2_R1_001.fastq.gz
Renamed 2H2_R2_001.fastq.gz to POR-262-TP2_R2_001.fastq.gz
Renamed 88--1E5_R1_001.fastq.gz to POR-245-TP3_R1_001.fastq.gz
Renamed 88--1E5_R2_001.fastq.gz to POR-245-TP3_R2_001.fastq.gz
Renamed 93--1E11_R1_001.fastq.gz to POR-260-TP4_R1_001.fastq.gz
Renamed 93--1E11_R2_001.fastq.gz to POR-260-TP4_R2_001.fastq.gz</code></pre>
</section>
<section id="fastqcmultiqc-on-raw-reads" class="level1">
<h1>4 FastQC/MultiQC on raw reads</h1>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load bash variables into memory</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> .bashvars</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">############ RUN FASTQC ############</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create array of trimmed FastQs</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="va">raw_fastqs_array</span><span class="op">=</span><span class="va">(${raw_reads_dir}</span>/<span class="va">${fastq_pattern})</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Pass array contents to new variable as space-delimited list</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="va">raw_fastqc_list</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="st">"</span><span class="va">${raw_fastqs_array</span><span class="op">[*]</span><span class="va">}</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Beginning FastQC on raw reads..."</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Run FastQC</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a><span class="co">### </span><span class="al">NOTE</span><span class="co">: Do NOT quote raw_fastqc_list</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="va">${programs_array</span><span class="op">[</span>fastqc<span class="op">]</span><span class="va">}</span> <span class="dt">\</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>--threads <span class="va">${threads}</span> <span class="dt">\</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>--outdir <span class="va">${raw_reads_dir}</span> <span class="dt">\</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>--quiet <span class="dt">\</span></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a><span class="va">${raw_fastqc_list}</span></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"FastQC on raw reads complete!"</span></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a><span class="co">############ </span><span class="re">END</span><span class="co"> FASTQC ############</span></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a><span class="co">############ RUN MULTIQC ############</span></span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Beginning MultiQC on raw FastQC..."</span></span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a><span class="va">${programs_array</span><span class="op">[</span>multiqc<span class="op">]</span><span class="va">}</span> <span class="dt">\</span></span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a><span class="va">${raw_reads_dir}</span> <span class="dt">\</span></span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>--interactive <span class="dt">\</span></span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>-o <span class="va">${raw_reads_dir}</span></span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"MultiQC on raw FastQs complete."</span></span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a><span class="co">############ </span><span class="re">END</span><span class="co"> MULTIQC ############</span></span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Removing FastQC zip files."</span></span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> <span class="va">${raw_reads_dir}</span>/<span class="pp">*</span>.zip</span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"FastQC zip files removed."</span></span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a><span class="co"># View directory contents</span></span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-lh</span> <span class="va">${raw_reads_dir}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>Beginning FastQC on raw reads...

application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
application/gzip
FastQC on raw reads complete!

Beginning MultiQC on raw FastQC...


  /// MultiQC 🔍 | v1.14

|           multiqc | MultiQC Version v1.27 now available!
|           multiqc | Search path : /home/shared/8TB_HDD_01/sam/gitrepos/urol-e5/timeseries_molecular/E-Peve/data/wgbs-raw-fastqs
|         searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 288/288  
|            fastqc | Found 72 reports
|           multiqc | Compressing plot data
|           multiqc | Report      : ../data/wgbs-raw-fastqs/multiqc_report.html
|           multiqc | Data        : ../data/wgbs-raw-fastqs/multiqc_data
|           multiqc | MultiQC complete

MultiQC on raw FastQs complete.

Removing FastQC zip files.

FastQC zip files removed.

total 172G
-rw-r--r-- 1 sam sam   61 Dec 30 08:07 110--2E1_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   61 Dec 30 08:09 110--2E1_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   61 Dec 30 08:12 111--1E4_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   61 Dec 30 08:15 111--1E4_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   61 Dec 30 08:17 114--1E6_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   61 Dec 30 08:19 114--1E6_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 09:12 1A7_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 09:15 1A7_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 09:24 1A9_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 09:28 1A9_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 09:38 1B11_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 09:40 1B11_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 10:15 1B6_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 10:18 1B6_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 10:25 1B8_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 10:27 1B8_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 10:29 1B9_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 10:32 1B9_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 10:57 1C2_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 10:59 1C2_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 11:36 1D10_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 11:38 1D10_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 11:46 1D12_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 11:48 1D12_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 11:51 1D1_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 11:54 1D1_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 11:56 1D2_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 11:58 1D2_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 12:16 1D6_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 12:19 1D6_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 12:36 1F10_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 12:38 1F10_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 13:00 1F3_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 13:02 1F3_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 13:15 1F6_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 13:17 1F6_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 13:32 1G10_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 13:34 1G10_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 13:37 1G11_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 13:39 1G11_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 13:51 1G2_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 13:54 1G2_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 14:10 1G6_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 14:12 1G6_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 14:24 1G9_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 14:26 1G9_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 14:40 1H12_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   57 Dec 30 14:43 1H12_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 15:24 1H8_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 15:27 1H8_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 15:41 2A2_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 15:43 2A2_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 15:59 2B3_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 16:03 2B3_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 16:05 2C1_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 16:08 2C1_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 16:17 2C3_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 16:20 2C3_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 16:43 2F2_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 16:46 2F2_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 16:49 2F3_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 16:51 2F3_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 16:54 2G1_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 16:57 2G1_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 17:05 2G3_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 17:07 2G3_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 17:17 2H2_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   56 Dec 30 17:20 2H2_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   60 Dec 30 18:09 88--1E5_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   60 Dec 30 18:13 88--1E5_R2_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   61 Dec 30 18:16 93--1E11_R1_001.fastq.gz.md5
-rw-r--r-- 1 sam sam   61 Dec 30 18:18 93--1E11_R2_001.fastq.gz.md5
drwxr-xr-x 2 sam sam 4.0K Feb 11 09:35 multiqc_data
-rw-r--r-- 1 sam sam 2.1M Feb 11 09:35 multiqc_report.html
-rw-r--r-- 1 sam sam 587K Feb 11 09:25 POR-216-TP1_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.1G Dec 30 16:05 POR-216-TP1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 593K Feb 11 09:23 POR-216-TP1_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.1G Dec 30 16:08 POR-216-TP1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 590K Feb 11 09:28 POR-216-TP2_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 3.0G Dec 30 11:51 POR-216-TP2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 609K Feb 11 09:28 POR-216-TP2_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.9G Dec 30 11:54 POR-216-TP2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 587K Feb 11 09:28 POR-216-TP3_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.9G Dec 30 13:37 POR-216-TP3_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 605K Feb 11 09:28 POR-216-TP3_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.9G Dec 30 13:39 POR-216-TP3_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 586K Feb 11 09:25 POR-216-TP4_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.8G Dec 30 13:51 POR-216-TP4_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 604K Feb 11 09:28 POR-216-TP4_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.8G Dec 30 13:54 POR-216-TP4_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 588K Feb 11 09:23 POR-236-TP1_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 1.7G Dec 30 14:10 POR-236-TP1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 609K Feb 11 09:22 POR-236-TP1_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 1.7G Dec 30 14:12 POR-236-TP1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 587K Feb 11 09:26 POR-236-TP2_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.5G Dec 30 10:29 POR-236-TP2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 606K Feb 11 09:27 POR-236-TP2_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.5G Dec 30 10:32 POR-236-TP2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 585K Feb 11 09:27 POR-245-TP1_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.6G Dec 30 16:54 POR-245-TP1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 592K Feb 11 09:24 POR-245-TP1_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.6G Dec 30 16:57 POR-245-TP1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 588K Feb 11 09:24 POR-245-TP2_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.3G Dec 30 10:15 POR-245-TP2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 608K Feb 11 09:25 POR-245-TP2_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.3G Dec 30 10:18 POR-245-TP2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 584K Feb 11 09:28 POR-245-TP3_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.8G Dec 30 18:09 POR-245-TP3_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 601K Feb 11 09:25 POR-245-TP3_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.7G Dec 30 18:13 POR-245-TP3_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 585K Feb 11 09:25 POR-245-TP4_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.5G Dec 30 12:16 POR-245-TP4_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 602K Feb 11 09:27 POR-245-TP4_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.5G Dec 30 12:19 POR-245-TP4_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 582K Feb 11 09:24 POR-260-TP1_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 11:56 POR-260-TP1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 599K Feb 11 09:25 POR-260-TP1_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 11:58 POR-260-TP1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 589K Feb 11 09:23 POR-260-TP2_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.1G Dec 30 13:15 POR-260-TP2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 609K Feb 11 09:25 POR-260-TP2_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.1G Dec 30 13:17 POR-260-TP2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 588K Feb 11 09:25 POR-260-TP3_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.2G Dec 30 14:24 POR-260-TP3_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 607K Feb 11 09:25 POR-260-TP3_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.2G Dec 30 14:26 POR-260-TP3_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 587K Feb 11 09:26 POR-260-TP4_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.6G Dec 30 18:16 POR-260-TP4_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 604K Feb 11 09:25 POR-260-TP4_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.5G Dec 30 18:18 POR-260-TP4_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 584K Feb 11 09:27 POR-262-TP1_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.7G Dec 30 09:12 POR-262-TP1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 601K Feb 11 09:27 POR-262-TP1_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.6G Dec 30 09:15 POR-262-TP1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 580K Feb 11 09:28 POR-262-TP2_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.8G Dec 30 17:17 POR-262-TP2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 589K Feb 11 09:28 POR-262-TP2_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.8G Dec 30 17:20 POR-262-TP2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 578K Feb 11 09:29 POR-262-TP4_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.8G Dec 30 15:59 POR-262-TP4_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 588K Feb 11 09:29 POR-262-TP4_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.9G Dec 30 16:03 POR-262-TP4_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 590K Feb 11 09:24 POR-69-TP1_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 1.9G Dec 30 10:25 POR-69-TP1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 607K Feb 11 09:24 POR-69-TP1_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 1.9G Dec 30 10:27 POR-69-TP1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 590K Feb 11 09:24 POR-69-TP2_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.1G Dec 30 11:36 POR-69-TP2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 608K Feb 11 09:25 POR-69-TP2_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.1G Dec 30 11:38 POR-69-TP2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 588K Feb 11 09:29 POR-69-TP3_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 3.0G Dec 30 09:24 POR-69-TP3_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 606K Feb 11 09:28 POR-69-TP3_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.9G Dec 30 09:28 POR-69-TP3_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 592K Feb 11 09:31 POR-69-TP4_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 09:38 POR-69-TP4_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 608K Feb 11 09:31 POR-69-TP4_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 09:40 POR-69-TP4_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 585K Feb 11 09:32 POR-72-TP1_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 12:36 POR-72-TP1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 603K Feb 11 09:32 POR-72-TP1_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 12:38 POR-72-TP1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 584K Feb 11 09:31 POR-72-TP2_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.2G Dec 30 15:41 POR-72-TP2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 594K Feb 11 09:32 POR-72-TP2_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.2G Dec 30 15:43 POR-72-TP2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 588K Feb 11 09:33 POR-72-TP3_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.3G Dec 30 13:32 POR-72-TP3_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 603K Feb 11 09:32 POR-72-TP3_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.2G Dec 30 13:34 POR-72-TP3_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 583K Feb 11 09:33 POR-72-TP4_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 16:49 POR-72-TP4_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 593K Feb 11 09:33 POR-72-TP4_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 16:51 POR-72-TP4_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 582K Feb 11 09:34 POR-73-TP1_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.6G Dec 30 16:17 POR-73-TP1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 591K Feb 11 09:34 POR-73-TP1_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.6G Dec 30 16:20 POR-73-TP1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 578K Feb 11 09:34 POR-73-TP2_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.9G Dec 30 14:40 POR-73-TP2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 590K Feb 11 09:34 POR-73-TP2_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 3.0G Dec 30 14:43 POR-73-TP2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 585K Feb 11 09:33 POR-73-TP3_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 11:46 POR-73-TP3_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 605K Feb 11 09:33 POR-73-TP3_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 11:48 POR-73-TP3_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 583K Feb 11 09:33 POR-74-TP1_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.0G Dec 30 08:07 POR-74-TP1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 590K Feb 11 09:32 POR-74-TP1_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.0G Dec 30 08:09 POR-74-TP1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 585K Feb 11 09:34 POR-74-TP2_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 3.0G Dec 30 08:12 POR-74-TP2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 603K Feb 11 09:34 POR-74-TP2_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.9G Dec 30 08:15 POR-74-TP2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 582K Feb 11 09:32 POR-74-TP3_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 1.9G Dec 30 17:05 POR-74-TP3_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 591K Feb 11 09:33 POR-74-TP3_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 1.9G Dec 30 17:07 POR-74-TP3_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 587K Feb 11 09:33 POR-74-TP4_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 15:24 POR-74-TP4_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 603K Feb 11 09:34 POR-74-TP4_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 15:27 POR-74-TP4_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 584K Feb 11 09:33 POR-83-TP1_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.1G Dec 30 08:17 POR-83-TP1_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 603K Feb 11 09:34 POR-83-TP1_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.0G Dec 30 08:19 POR-83-TP1_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 586K Feb 11 09:34 POR-83-TP2_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.5G Dec 30 16:43 POR-83-TP2_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 595K Feb 11 09:35 POR-83-TP2_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.6G Dec 30 16:46 POR-83-TP2_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 588K Feb 11 09:33 POR-83-TP3_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.0G Dec 30 13:00 POR-83-TP3_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 605K Feb 11 09:34 POR-83-TP3_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.0G Dec 30 13:02 POR-83-TP3_R2_001.fastq.gz
-rw-r--r-- 1 sam sam 586K Feb 11 09:34 POR-83-TP4_R1_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 10:57 POR-83-TP4_R1_001.fastq.gz
-rw-r--r-- 1 sam sam 603K Feb 11 09:34 POR-83-TP4_R2_001_fastqc.html
-rw-r--r-- 1 sam sam 2.4G Dec 30 10:59 POR-83-TP4_R2_001.fastq.gz</code></pre>
</section>
<section id="results" class="level1">
<h1>RESULTS</h1>
<p>MultiQC report (HTML):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/gitrepos/urol-e5/timeseries_molecular/E-Peve/data/wgbs-raw-fastqs/multiqc_report.html">E-Peve/data/wgbs-raw-fastqs/multiqc_report.html</a></li>
</ul>
<p>Overall, sequencing looks pretty good. The “Per Base Sequence Content” analysis by <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"><code>FastQC</code></a> suggests these libraries are considered <em>directional</em>, plus confirmation by Azenta that libraries were constructed using <a href="https://www.neb.com/en-us/products/e7120-nebnext-enzymatic-methyl-seq-kit">NEBNext® Enzymatic Methyl-seq Kit</a> which generates directional libraries.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./20250210-peve-multiqc-plot-sequence-content.png" class="img-fluid figure-img" alt="Screenshot of MultiQC per base sequence content plot showing low C content - indicative of directional WGBS libraries"></p>
<figcaption>Screenshot of MultiQC per base sequence content plot showing low C content - indicative of directional WGBS libraries</figcaption>
</figure>
</div>
</section>
<section id="references" class="level1">
<h1>REFERENCES</h1>
<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-ewels2016" class="csl-entry" role="listitem">
Ewels, Philip, Måns Magnusson, Sverker Lundin, and Max Käller. 2016. “MultiQC: Summarize Analysis Results for Multiple Tools and Samples in a Single Report.” <em>Bioinformatics</em> 32 (19): 3047–48. <a href="https://doi.org/10.1093/bioinformatics/btw354" class="uri">https://doi.org/10.1093/bioinformatics/btw354</a>.
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/robertslab\.github\.io\/sams-notebook");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>