<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sam White">
<meta name="dcterms.date" content="2024-02-01">

<title>Sam’s Notebook - Daily Bits - February 2024</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-7483T61T11"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-7483T61T11', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Sam’s Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" rel="" target="_blank">
 <span class="dropdown-text">Rstudio Server - Raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu/Atumefaciens/" rel="" target="_blank">
 <span class="dropdown-text">My Gannet Directory</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" rel="" target="_blank">
 <span class="dropdown-text">Steven’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://shedurkin.github.io/Roberts-LabNotebook/" rel="" target="_blank">
 <span class="dropdown-text">Kathleen’s Notebook</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://meganewing.github.io/mewing-notebook/" rel="" target="_blank">
 <span class="dropdown-text">Megan’s Notebook</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/sams-notebook" rel="" target="_blank">
 <span class="dropdown-text">Notebook repo</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" rel="" target="_blank">
 <span class="dropdown-text">Lab Organization (GitHub)</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" rel="" target="_blank">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" rel="" target="_blank"><i class="bi bi-calendar-week" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../../index.xml" rel="" target=""><i class="bi bi-rss" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-right">
      <h1 class="title">Daily Bits - February 2024</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2024</div>
                <div class="quarto-category">Daily Bits</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Sam White </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 1, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#section" id="toc-section" class="nav-link active" data-scroll-target="#section">20240222</a>
  <ul>
  <li><a href="#lab-meeting" id="toc-lab-meeting" class="nav-link" data-scroll-target="#lab-meeting">Lab Meeting</a></li>
  <li><a href="#ceabigr-github-repo" id="toc-ceabigr-github-repo" class="nav-link" data-scroll-target="#ceabigr-github-repo">CEABIGR (GitHub repo)</a></li>
  <li><a href="#miscellaneous" id="toc-miscellaneous" class="nav-link" data-scroll-target="#miscellaneous">Miscellaneous</a></li>
  </ul></li>
  <li><a href="#section-1" id="toc-section-1" class="nav-link" data-scroll-target="#section-1">20240221</a>
  <ul>
  <li><a href="#miscellaneous-1" id="toc-miscellaneous-1" class="nav-link" data-scroll-target="#miscellaneous-1">Miscellaneous</a></li>
  <li><a href="#ceabigr-github-repo-1" id="toc-ceabigr-github-repo-1" class="nav-link" data-scroll-target="#ceabigr-github-repo-1">CEABIGR (GitHub repo)</a></li>
  </ul></li>
  <li><a href="#section-2" id="toc-section-2" class="nav-link" data-scroll-target="#section-2">20240220</a>
  <ul>
  <li><a href="#ceabigr-github-repo-2" id="toc-ceabigr-github-repo-2" class="nav-link" data-scroll-target="#ceabigr-github-repo-2">CEABIGR (GitHub repo)</a></li>
  </ul></li>
  <li><a href="#section-3" id="toc-section-3" class="nav-link" data-scroll-target="#section-3">20240216</a>
  <ul>
  <li><a href="#coral-e5-github-repo" id="toc-coral-e5-github-repo" class="nav-link" data-scroll-target="#coral-e5-github-repo">Coral E5 (GitHub repo)</a></li>
  </ul></li>
  <li><a href="#section-4" id="toc-section-4" class="nav-link" data-scroll-target="#section-4">20240215</a>
  <ul>
  <li><a href="#coral-e5-github-repo-1" id="toc-coral-e5-github-repo-1" class="nav-link" data-scroll-target="#coral-e5-github-repo-1">Coral E5 (GitHub repo)</a></li>
  </ul></li>
  <li><a href="#section-5" id="toc-section-5" class="nav-link" data-scroll-target="#section-5">20240214</a>
  <ul>
  <li><a href="#coral-e5-github-repo-2" id="toc-coral-e5-github-repo-2" class="nav-link" data-scroll-target="#coral-e5-github-repo-2">Coral E5 (GitHub repo)</a></li>
  </ul></li>
  <li><a href="#section-6" id="toc-section-6" class="nav-link" data-scroll-target="#section-6">20240213</a>
  <ul>
  <li><a href="#coral-e5-github-repo-3" id="toc-coral-e5-github-repo-3" class="nav-link" data-scroll-target="#coral-e5-github-repo-3">Coral E5 (GitHub repo)</a></li>
  </ul></li>
  <li><a href="#section-7" id="toc-section-7" class="nav-link" data-scroll-target="#section-7">20240212</a></li>
  <li><a href="#in-lab" id="toc-in-lab" class="nav-link" data-scroll-target="#in-lab">In lab</a>
  <ul>
  <li><a href="#coral-e5-github-repo-4" id="toc-coral-e5-github-repo-4" class="nav-link" data-scroll-target="#coral-e5-github-repo-4">Coral E5 (GitHub repo)</a></li>
  </ul></li>
  <li><a href="#section-8" id="toc-section-8" class="nav-link" data-scroll-target="#section-8">20240209</a></li>
  <li><a href="#science-hour" id="toc-science-hour" class="nav-link" data-scroll-target="#science-hour">Science Hour</a>
  <ul>
  <li><a href="#coral-e5-github-repo-5" id="toc-coral-e5-github-repo-5" class="nav-link" data-scroll-target="#coral-e5-github-repo-5">Coral E5 (GitHub repo)</a></li>
  </ul></li>
  <li><a href="#section-9" id="toc-section-9" class="nav-link" data-scroll-target="#section-9">20240208</a>
  <ul>
  <li><a href="#lab-meeting-1" id="toc-lab-meeting-1" class="nav-link" data-scroll-target="#lab-meeting-1">Lab meeting</a></li>
  <li><a href="#ceabigr-github-repo-3" id="toc-ceabigr-github-repo-3" class="nav-link" data-scroll-target="#ceabigr-github-repo-3">CEABIGR (GitHub repo)</a></li>
  </ul></li>
  <li><a href="#section-10" id="toc-section-10" class="nav-link" data-scroll-target="#section-10">20240207</a>
  <ul>
  <li><a href="#coral-e5-github-repo-6" id="toc-coral-e5-github-repo-6" class="nav-link" data-scroll-target="#coral-e5-github-repo-6">Coral E5 (GitHub repo)</a></li>
  </ul></li>
  <li><a href="#section-11" id="toc-section-11" class="nav-link" data-scroll-target="#section-11">20240206</a>
  <ul>
  <li><a href="#coral-e5-github-repo-7" id="toc-coral-e5-github-repo-7" class="nav-link" data-scroll-target="#coral-e5-github-repo-7">Coral E5 (GitHub repo)</a></li>
  <li><a href="#ceabigr-github-repo-4" id="toc-ceabigr-github-repo-4" class="nav-link" data-scroll-target="#ceabigr-github-repo-4">CEABIGR (GitHub repo)</a></li>
  </ul></li>
  <li><a href="#section-12" id="toc-section-12" class="nav-link" data-scroll-target="#section-12">20240205</a>
  <ul>
  <li><a href="#coral-e5-github-repo-8" id="toc-coral-e5-github-repo-8" class="nav-link" data-scroll-target="#coral-e5-github-repo-8">Coral E5 (GitHub repo)</a></li>
  </ul></li>
  <li><a href="#section-13" id="toc-section-13" class="nav-link" data-scroll-target="#section-13">20240202</a>
  <ul>
  <li><a href="#coral-e5-github-repo-9" id="toc-coral-e5-github-repo-9" class="nav-link" data-scroll-target="#coral-e5-github-repo-9">Coral E5 (GitHub repo)</a></li>
  <li><a href="#science-hour-1" id="toc-science-hour-1" class="nav-link" data-scroll-target="#science-hour-1">Science Hour</a></li>
  </ul></li>
  <li><a href="#section-14" id="toc-section-14" class="nav-link" data-scroll-target="#section-14">20240201</a>
  <ul>
  <li><a href="#in-lab-1" id="toc-in-lab-1" class="nav-link" data-scroll-target="#in-lab-1">In Lab</a></li>
  <li><a href="#coral-e5-github-repo-10" id="toc-coral-e5-github-repo-10" class="nav-link" data-scroll-target="#coral-e5-github-repo-10">Coral E5 (GitHub repo)</a></li>
  <li><a href="#ceabigr-github-repo-5" id="toc-ceabigr-github-repo-5" class="nav-link" data-scroll-target="#ceabigr-github-repo-5">CEABIGR (GitHub repo)</a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-right" id="quarto-document-content">




<section id="section" class="level2">
<h2 class="anchored" data-anchor-id="section">20240222</h2>
<section id="lab-meeting" class="level3">
<h3 class="anchored" data-anchor-id="lab-meeting">Lab Meeting</h3>
<ul>
<li>Discussed Pub-a-thon 2024</li>
</ul>
</section>
<section id="ceabigr-github-repo" class="level3">
<h3 class="anchored" data-anchor-id="ceabigr-github-repo"><a href="https://github.com/sr320/ceabigr">CEABIGR</a> (GitHub repo)</h3>
<ul>
<li><p>Reviewed recent GitHub Issues</p></li>
<li><p>Worked on creating canonical gene/transcript counts table.</p></li>
</ul>
</section>
<section id="miscellaneous" class="level3">
<h3 class="anchored" data-anchor-id="miscellaneous">Miscellaneous</h3>
<ul>
<li><p>Updated <a href="https://b.link/nightingales"><code>Nightingales</code></a> (Google Sheet) spreadsheet with Cod RNA-seq info.</p></li>
<li><p>Raven was bogged down by some weird rogue <code>git</code> command, so had to reboot.</p></li>
</ul>
</section>
</section>
<section id="section-1" class="level2">
<h2 class="anchored" data-anchor-id="section-1">20240221</h2>
<section id="miscellaneous-1" class="level3">
<h3 class="anchored" data-anchor-id="miscellaneous-1">Miscellaneous</h3>
<ul>
<li><p><a href="https://github.com/RobertsLab/resources/issues/1806">Set Ariana up with Gannet access</a> (GitHub Issue)</p></li>
<li><p>Received <a href="../../../posts/2024/2024-02-21-Data-Received---Pacific-cod-RNA-seq-Azenta-Project-30-943133806/index.html">Pacific cod RNA-seq data</a></p></li>
</ul>
</section>
<section id="ceabigr-github-repo-1" class="level3">
<h3 class="anchored" data-anchor-id="ceabigr-github-repo-1"><a href="https://github.com/sr320/ceabigr">CEABIGR</a> (GitHub repo)</h3>
<ul>
<li>Generated volcano plots</li>
</ul>
<div class="quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://raw.githubusercontent.com/sr320/ceabigr/main/output/18-differential-gene-expression-ballgown/figures/volcano_plot-DEGs-females-CvE.png" class="img-fluid figure-img" alt="Volcano plot of gene expression in females. No genes are differentially expressed. Grey points are genes with p-values <=0.05. Black points are genes with p/q-values > 0.05."></p>
<figcaption class="figure-caption">Volcano plot of gene expression in females. No genes are differentially expressed. Grey points are genes with p-values &lt;=0.05. Black points are genes with p/q-values &gt; 0.05.</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://raw.githubusercontent.com/sr320/ceabigr/main/output/18-differential-gene-expression-ballgown/figures/volcano_plot-DETs-females-CvE.png" class="img-fluid figure-img" alt="Volcano plot of transcript expression in females. A single transcript is differentially expressed and is rendered as a blue point and labelled with its associated geneID. Grey points are transcripts with p-values <=0.05. Black points are transcripts with p/q-values > 0.05."></p>
<figcaption class="figure-caption">Volcano plot of transcript expression in females. A single transcript is differentially expressed and is rendered as a blue point and labelled with its associated geneID. Grey points are transcripts with p-values &lt;=0.05. Black points are transcripts with p/q-values &gt; 0.05.</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://raw.githubusercontent.com/sr320/ceabigr/main/output/18-differential-gene-expression-ballgown/figures/volcano_plot-DEGs-males-CvE.png" class="img-fluid figure-img" alt="Volcano plot of gene expression in males. No genes are differentially expressed. Grey points are genes with p-values <=0.05. Black points are genes with p/q-values > 0.05."></p>
<figcaption class="figure-caption">Volcano plot of gene expression in males. No genes are differentially expressed. Grey points are genes with p-values &lt;=0.05. Black points are genes with p/q-values &gt; 0.05.</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://raw.githubusercontent.com/sr320/ceabigr/main/output/18-differential-gene-expression-ballgown/figures/volcano_plot-DETs-males-CvE.png" class="img-fluid figure-img" alt="Volcano plot of transcript expression in males. No transcripts are differentially expressed. Grey points are transcripts with p-values <=0.05. Black points are transcripts with p/q-values > 0.05."></p>
<figcaption class="figure-caption">Volcano plot of transcript expression in males. No transcripts are differentially expressed. Grey points are transcripts with p-values &lt;=0.05. Black points are transcripts with p/q-values &gt; 0.05</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
</section>
<section id="section-2" class="level2">
<h2 class="anchored" data-anchor-id="section-2">20240220</h2>
<section id="ceabigr-github-repo-2" class="level3">
<h3 class="anchored" data-anchor-id="ceabigr-github-repo-2"><a href="https://github.com/sr320/ceabigr">CEABIGR</a> (GitHub repo)</h3>
<ul>
<li>Started working on generating volcano plots.</li>
</ul>
</section>
</section>
<section id="section-3" class="level2">
<h2 class="anchored" data-anchor-id="section-3">20240216</h2>
<section id="coral-e5-github-repo" class="level3">
<h3 class="anchored" data-anchor-id="coral-e5-github-repo"><a href="https://github.com/urol-e5/deep-dive">Coral E5</a> (GitHub repo)</h3>
<ul>
<li>Finished running <a href="https://github.com/MikeAxtell/ShortStack"><code>ShortStack</code></a> variations with newly trimmed <em>P.meandrina</em> sRNA-seq.</li>
</ul>
</section>
</section>
<section id="section-4" class="level2">
<h2 class="anchored" data-anchor-id="section-4">20240215</h2>
<section id="coral-e5-github-repo-1" class="level3">
<h3 class="anchored" data-anchor-id="coral-e5-github-repo-1"><a href="https://github.com/urol-e5/deep-dive">Coral E5</a> (GitHub repo)</h3>
<ul>
<li>Ran <a href="https://github.com/MikeAxtell/ShortStack"><code>ShortStack</code></a> variations with newly trimmed <em>P.meandrina</em> sRNA-seq.</li>
</ul>
</section>
</section>
<section id="section-5" class="level2">
<h2 class="anchored" data-anchor-id="section-5">20240214</h2>
<section id="coral-e5-github-repo-2" class="level3">
<h3 class="anchored" data-anchor-id="coral-e5-github-repo-2"><a href="https://github.com/urol-e5/deep-dive">Coral E5</a> (GitHub repo)</h3>
<ul>
<li>Ran <a href="https://github.com/MikeAxtell/ShortStack"><code>ShortStack</code></a> variations with newly trimmed <a href="http://www.coralsoftheworld.org/species_factsheets/species_factsheet_summary/acropora-pulchra/"><em>Acropora pulchra</em></a> sRNA-seq.</li>
</ul>
</section>
</section>
<section id="section-6" class="level2">
<h2 class="anchored" data-anchor-id="section-6">20240213</h2>
<section id="coral-e5-github-repo-3" class="level3">
<h3 class="anchored" data-anchor-id="coral-e5-github-repo-3"><a href="https://github.com/urol-e5/deep-dive">Coral E5</a> (GitHub repo)</h3>
<ul>
<li>Cranked out new trimming on <em>P.meandrina</em> sRNA-seq samples.</li>
</ul>
</section>
</section>
<section id="section-7" class="level2">
<h2 class="anchored" data-anchor-id="section-7">20240212</h2>
</section>
<section id="in-lab" class="level2">
<h2 class="anchored" data-anchor-id="in-lab">In lab</h2>
<ul>
<li>Reviewed reverse transcription and qPCR with Eric.</li>
</ul>
<section id="coral-e5-github-repo-4" class="level3">
<h3 class="anchored" data-anchor-id="coral-e5-github-repo-4"><a href="https://github.com/urol-e5/deep-dive">Coral E5</a> (GitHub repo)</h3>
<ul>
<li>Started running new trimming on <em>A.pulchra</em> sRNA-seq samples (preivously, had just tried with <em>P.evermanni</em>).</li>
</ul>
</section>
</section>
<section id="section-8" class="level2">
<h2 class="anchored" data-anchor-id="section-8">20240209</h2>
</section>
<section id="science-hour" class="level2">
<h2 class="anchored" data-anchor-id="science-hour">Science Hour</h2>
<ul>
<li>Steven and I looked at and discussed the exon expression patterns of individual genes and compared those patterns to the WGCNA outputs which Ariana generated at yesterday’s meeting. Definitely a few individuals whose exon usage seems to deviate a great deal from that WGCNA module assignment…</li>
</ul>
<section id="coral-e5-github-repo-5" class="level3">
<h3 class="anchored" data-anchor-id="coral-e5-github-repo-5"><a href="https://github.com/urol-e5/deep-dive">Coral E5</a> (GitHub repo)</h3>
<ul>
<li>Continued to work on implementing new trimming/merging parameters, but <a href="https://github.com/MikeAxtell/ShortStack/tree/master">ShortStack</a> keeps encountering an <a href="https://github.com/MikeAxtell/ShortStack/issues/146">error</a> (GitHub Issue). I’m hoping the developer can lend some insight. This is weird, because these samples, albeit trimmed with a different program and used as paired-end reads (only trying R1 or merged reads at this point), were successfully analyzed previously…</li>
</ul>
</section>
</section>
<section id="section-9" class="level2">
<h2 class="anchored" data-anchor-id="section-9">20240208</h2>
<section id="lab-meeting-1" class="level3">
<h3 class="anchored" data-anchor-id="lab-meeting-1">Lab meeting</h3>
</section>
<section id="ceabigr-github-repo-3" class="level3">
<h3 class="anchored" data-anchor-id="ceabigr-github-repo-3"><a href="https://github.com/sr320/ceabigr">CEABIGR</a> (GitHub repo)</h3>
<ul>
<li>Special, extended session with Arianna in attendance. She ran some <a href="https://cran.r-project.org/web/packages/WGCNA/WGCNA.pdf">WGCNA</a> (PDF) analysis to try to see how exon expression patterns varied between control/exposed samples. Very cool!</li>
</ul>
</section>
</section>
<section id="section-10" class="level2">
<h2 class="anchored" data-anchor-id="section-10">20240207</h2>
<section id="coral-e5-github-repo-6" class="level3">
<h3 class="anchored" data-anchor-id="coral-e5-github-repo-6"><a href="https://github.com/urol-e5/deep-dive">Coral E5</a> (GitHub repo)</h3>
<ul>
<li><p>Finished messing around with trimming/merging stuff. Here’s the <a href="https://robertslab.github.io/sams-notebook/posts/2024/2024-02-02-sRNA-seq-Paired-Read-Merging-Exploratios---Coral-E5-Deep-Dive/">notebook entry</a>.</p></li>
<li><p>Began implementing new trimming/merging.</p></li>
</ul>
</section>
</section>
<section id="section-11" class="level2">
<h2 class="anchored" data-anchor-id="section-11">20240206</h2>
<section id="coral-e5-github-repo-7" class="level3">
<h3 class="anchored" data-anchor-id="coral-e5-github-repo-7"><a href="https://github.com/urol-e5/deep-dive">Coral E5</a> (GitHub repo)</h3>
<ul>
<li>Continued to explore trimming and merging strategies for sRNA-seq data, using <a href="https://github.com/OpenGene/fastp"><code>fastp</code></a> alone, BBMerge alone, and/or combining the two approaches.</li>
</ul>
</section>
<section id="ceabigr-github-repo-4" class="level3">
<h3 class="anchored" data-anchor-id="ceabigr-github-repo-4"><a href="https://github.com/sr320/ceabigr">CEABIGR</a> (GitHub repo)</h3>
<ul>
<li>Spent a few hours on a video chat with Steven and Yaamini going through exon fold-change code/analysis.</li>
</ul>
</section>
</section>
<section id="section-12" class="level2">
<h2 class="anchored" data-anchor-id="section-12">20240205</h2>
<section id="coral-e5-github-repo-8" class="level3">
<h3 class="anchored" data-anchor-id="coral-e5-github-repo-8"><a href="https://github.com/urol-e5/deep-dive">Coral E5</a> (GitHub repo)</h3>
<ul>
<li>Continued to explore trimming and merging strategies for sRNA-seq data, using <a href="https://github.com/OpenGene/fastp"><code>fastp</code></a> alone, BBMerge alone, and/or combining the two approaches.</li>
</ul>
</section>
</section>
<section id="section-13" class="level2">
<h2 class="anchored" data-anchor-id="section-13">20240202</h2>
<section id="coral-e5-github-repo-9" class="level3">
<h3 class="anchored" data-anchor-id="coral-e5-github-repo-9"><a href="https://github.com/urol-e5/deep-dive">Coral E5</a> (GitHub repo)</h3>
</section>
<section id="science-hour-1" class="level3">
<h3 class="anchored" data-anchor-id="science-hour-1">Science Hour</h3>
<ul>
<li>Steven and I played around with exon fold-change data.</li>
</ul>
</section>
</section>
<section id="section-14" class="level2">
<h2 class="anchored" data-anchor-id="section-14">20240201</h2>
<section id="in-lab-1" class="level3">
<h3 class="anchored" data-anchor-id="in-lab-1">In Lab</h3>
<ul>
<li><p>Lab meeting.</p></li>
<li><p>Helped Eric get started with RNA quantification using Qubit.</p></li>
</ul>
</section>
<section id="coral-e5-github-repo-10" class="level3">
<h3 class="anchored" data-anchor-id="coral-e5-github-repo-10"><a href="https://github.com/urol-e5/deep-dive">Coral E5</a> (GitHub repo)</h3>
<ul>
<li><p>Tested out more R1/R2 merging/trimming using <a href="https://github.com/OpenGene/fastp"><code>fastp</code></a>.</p>
<ul>
<li>Merging is similar to BBMerge results - i.e.&nbsp;a (relatively) small number of reads are actually merging, despite the fact that these should be short insert sizes and R1/R2 reads should overlap…</li>
</ul></li>
<li><p>Installed <a href="https://github.com/HCGB-IGTP/XICRA">XICRA</a> <span class="citation" data-cites="sanchezherrero2021">(<a href="#ref-sanchezherrero2021" role="doc-biblioref">Sanchez Herrero et al. 2021</a>)</span> on Raven to identify isomiRs in sRNA-seq data. Possibly to get around the merging “issues” experienced with BBMerge and <a href="https://github.com/OpenGene/fastp"><code>fastp</code></a>.</p>
<ul>
<li><p>Installation was a bit bumpy, due to some weird Conda/Mamba “flexible solve” problem, but got that straightened out with <code>conda config --set channel_priority flexible</code>.</p></li>
<li><p>Installation test(s) had issues. I created the following GitHub Issues on the developer’s site in hopes of getting help:</p></li>
<li><p><a href="https://github.com/HCGB-IGTP/XICRA/issues/47"><code>installer.sh error</code></a></p></li>
<li><p><a href="https://github.com/HCGB-IGTP/XICRA/issues/42">“No samples were retrieved” in test data</a></p></li>
</ul></li>
</ul>
</section>
<section id="ceabigr-github-repo-5" class="level3">
<h3 class="anchored" data-anchor-id="ceabigr-github-repo-5"><a href="https://github.com/sr320/ceabigr">CEABIGR</a> (GitHub repo)</h3>
<ul>
<li>Need to generate R<sup>2</sup> values and slope of exon expression for all genes, Exons 1 - 6.</li>
</ul>



</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-sanchezherrero2021" class="csl-entry" role="listitem">
Sanchez Herrero, Jose Francisco, Raquel Pluvinet, Antonio Luna de Haro, and Lauro Sumoy. 2021. <span>“Paired-End Small RNA Sequencing Reveals a Possible Overestimation in the isomiR Sequence Repertoire Previously Reported from Conventional Single Read Data Analysis.”</span> <em>BMC Bioinformatics</em> 22 (1). <a href="https://doi.org/10.1186/s12859-021-04128-1">https://doi.org/10.1186/s12859-021-04128-1</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>