---
author: Sam White
toc-title: Contents
toc-depth: 5
toc-location: left
title: Bisulfite PCR - Testing Primer Annealing Gradient with Lake Trout Bisulfite-treated DNA
date: '2024-08-28'
draft: false
engine: knitr
categories: 
  - "2024"
  - "BS-PCR"
  - "bisulfite"
  - "PCR"
  - "DNA"
  - "Lake trout"
  - "Salvelinus namaycush"
---
## INTRO

::: {.callout-important}
This PCR was done incorrectly! The [cycling program](#pcr-cycling) was setup incorrectly. Notebook post is being left for posterity.
:::

Well, nothing like starting off a notebook post with a callout indicating this was done incorrectly... I'll leave all of this here anyway.

After previously [designing bisulfite PCR primers for the C1Q gene on 20240709](../2024-07-29-Primer-Design---Lake-Trout-C1q-Gene-Bisulfite-Sequencing-Primers-Using-Primer3/index.qmd), it was time to actually test them out. Due to their length, low complexity, and low melting temperatures (T<sub>m</sub>) I opted to test these out on a pool of the [_Salvelinus namaycush_ gDNA isolated on 20240712](../2024-07-12-DNA-Isolation-and-Quantification---Lake-Trout-Liver/index.qmd), using a temperature gradient.

## MATERIALS & METHODS

### DNA Pooling

1uL of each sample was used to create a pool of gDNA.

### PCR Reactions

PCR reactions were run using [EpiMark Hot Start Taq DNA Polymerase](https://www.neb.com/en-us/products/m0490-epimark-hot-start-taq-dna-polymerase) (New England Biolabs), according to the [manufacturer's protocol](https://www.neb.com/en-us/protocols/2012/09/25/epimark-hot-start-taq-dna-polymerase-guidelines-for-pcr-reactions-m0490), in a total reaction volume of 25uL, using 1uL of the pooled gDNA.

Since I was testing three different annealing temps (45<sup>o</sup>C, 50<sup>o</sup>C, and 55<sup>o</sup>C), I ran reactions in duplicate for each temperature (i.e. a total of six reactions).

### PCR Cycling

Annealing temperature gradient PCR was performed in a CFX Connect (BioRad).

Cycling params:

::: {.callout-warning}
The cycling program is incorrect and never actually goes through cycling!!!
:::

![Screenshot of cycling program, also sdhowing temperature gradient.](./20240828-snam-bs-pcr-gradient-range.png){#fig-PCR-parameters fig-alt="PCR cycling parameters."}


1. 95<sup>o</sup>C - 30s
2. 95<sup>o</sup>C - 15s
3. 45 - 55<sup>o</sup>C - 20s
4. 68<sup>o</sup>C - 4m
5. 68<sup>o</sup>C - 5m
6. 4<sup>o</sup>C - FOREVER
7. Go to Step 2.

Samples were loaded in the following positions:

![PCR tube placement in thermacylcer](./20240828-snam-bs-pcr-tube-placement.jpg){#fig-tubes fig-alt="Tube placement in thermalcylcer."}

PCR reactions were loaded on a low-TAE, 1.0% agarose gel, with ethidium bromide. Gel was run at 107V for ~45mins and then imaged.

O'GeneRuler (ThermoFisher) Ladder was used (5uL) for size reference:

![Ladder](../2024-05-31-Agarose-Gel---G.macrocephalus-gDNA-from-20240521/ogene-ruler.jpeg){#fig-ladder fig-alt="O'GeneRuler" .lightbox}

## RESULTS

No amplification.


![Agarose gel with pairs of lanes marked with PCR annealing temperatures of 45<sup>o</sup>C, 50<sup>o</sup>C, and 55<sup>o</sup>C](./20240828-snam-bs-pcr-gel.jpg){#fig-gel fig-alt="Agarose gel with labels of of 45<sup>o</sup>C, 50<sup>o</sup>C, and 55<sup>o</sup>C, indicating annealing temperatures"}
