---
author: Sam White
toc-title: Contents
toc-depth: 5
toc-location: left
date: 2016-04-28 05:40:12+00:00
layout: post
slug: data-management-o-lurida-raw-bgi-gbs-fastq-data
title: Data Management - O.lurida Raw BGI GBS FASTQ Data
categories:
  - 2016
  - Genotype-by-sequencing at BGI
  - Olympia oyster reciprocal transplant
tags:
  - GBS
  - genotype-by-sequencing
  - ipython notebook
  - jupyter notebook
  - olympia oyster
  - Ostrea lurida
---

BGI had previously supplied us with demultiplexed GBS FASTQ files. However, they had not provided us with the information/data on how those files were created. I contacted them and they've given us the two original FASTQ files, as well as the library index file and corresponding script they used for demultiplexing all of the files. The Jupyter (iPython) notebook below updates our [checksum](https://owl.fish.washington.edu/nightingales/O_lurida/20160223_gbs/checksums.md5) and [readme](http://owl.fish.washington.edu/nightingales/O_lurida/20160223_gbs/readme.md) files in our server directory that's hosting the files: [http://owl.fish.washington.edu/nightingales/O_lurida/20160223_gbs/](http://owl.fish.washington.edu/nightingales/O_lurida/20160223_gbs/)

See Jupyter Notebook below for processing details.

Jupyter Notebook: [20160427_Oly_GBS_data_management.ipynb](https://github.com/sr320/LabDocs/blob/master/jupyter_nbs/sam/20160427_Oly_GBS_data_management.ipynb)

<iframe src="https://nbviewer.org/github/sr320/LabDocs/blob/master/jupyter_nbs/sam/20160427_Oly_GBS_data_management.ipynb" width="100%" height="2000" scrolling="yes"></iframe>
