---
author: Sam White
toc-title: Contents
toc-depth: 5
toc-location: left
date: 2017-03-07 17:46:56+00:00
layout: post
slug: fastqc-oly-bgi-gbs-raw-illumina-data-demultiplexed
title: FASTQC - Oly BGI GBS Raw Illumina Data Demultiplexed
categories:
  - "2017"
  - "Genotype-by-sequencing at BGI"
  - "BGI"
  - "docker"
  - "FASTQC"
  - "GBS"
  - "genotype-by-sequencing"
  - "jupyter notebook"
  - "olympia oyster"
  - "Ostrea lurida"
  - "ostrich"
---

[Last week, I ran the two raw FASTQ files through FastQC](https://robertslab.github.io/sams-notebook/posts/2017/2017-03-03-fastqc-oly-bgi-gbs-raw-illumina-data/). As expected, FastQC detected "errors". These errors are due to the presence of adapter sequences, barcodes, and the use of a restriction enzyme (ApeKI) in library preparation. In summary, it's not surprising that FastQC was not please with the data because it's expecting a "standard" library prep that's already been trimmed and demultiplexed.

However, just for comparison, I ran the demultiplexed files through FastQC. The Jupyter notebook is linked (GitHub) and embedded below. I recommend viewing the Jupyter notebook on GitHub for easier viewing.

Results:

Pretty much the same, but with slight improvements due to removal of adapter and barcode sequences. The restriction site still leads to FastQC to report errors, which is expected.

Links to all of the FastQC output files are linked at the bottom of the notebook.

Jupyter notebook (GitHub): [20170306_docker_fastqc_demultiplexed_bgi_oly_gbs.ipynb
](https://github.com/sr320/LabDocs/blob/master/jupyter_nbs/sam/20170306_docker_fastqc_demultiplexed_bgi_oly_gbs.ipynb)
